// oci-objectstorage/lib/model/s-se-customer-key-details.js
"use strict";
/**
 * Object Storage Service API
 * Use Object Storage and Archive Storage APIs to manage buckets, objects, and related resources.
For more information, see [Overview of Object Storage](https://docs.oracle.com/iaas/Content/Object/Concepts/objectstorageoverview.htm) and
[Overview of Archive Storage](https://docs.oracle.com/iaas/Content/Archive/Concepts/archivestorageoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: opc_casper_users_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SSECustomerKeyDetails = void 0;
var SSECustomerKeyDetails;
(function (SSECustomerKeyDetails) {
    let Algorithm;
    (function (Algorithm) {
        Algorithm["Aes256"] = "AES256";
    })(Algorithm = SSECustomerKeyDetails.Algorithm || (SSECustomerKeyDetails.Algorithm = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SSECustomerKeyDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SSECustomerKeyDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(SSECustomerKeyDetails = exports.SSECustomerKeyDetails || (exports.SSECustomerKeyDetails = {}));
