// oci-identitydomains/lib/model/app.js
"use strict";
/**
 * Identity Domains API
 * Use the Identity Domains API to manage resources within an identity domain, for example, users, dynamic resource groups, groups, and identity providers. For information about managing resources within identity domains, see [Identity and Access Management (with identity domains)](https://docs.oracle.com/iaas/Content/Identity/home.htm).
Use this pattern to construct endpoints for identity domains: {@code https://<domainURL>/admin/v1/}. See [Finding an Identity Domain URL](https://docs.oracle.com/en-us/iaas/Content/Identity/api-getstarted/locate-identity-domain-url.htm) to locate the domain URL you need.
Use the table of contents and search tool to explore the Identity Domains API.
 * OpenAPI spec version: v1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.App = void 0;
const model = __importStar(require("../model"));
var App;
(function (App) {
    let IdcsPreventedOperations;
    (function (IdcsPreventedOperations) {
        IdcsPreventedOperations["Replace"] = "replace";
        IdcsPreventedOperations["Update"] = "update";
        IdcsPreventedOperations["Delete"] = "delete";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        IdcsPreventedOperations["UnknownValue"] = "UNKNOWN_VALUE";
    })(IdcsPreventedOperations = App.IdcsPreventedOperations || (App.IdcsPreventedOperations = {}));
    let LoginMechanism;
    (function (LoginMechanism) {
        LoginMechanism["Oidc"] = "OIDC";
        LoginMechanism["Saml"] = "SAML";
        LoginMechanism["Formfill"] = "FORMFILL";
        LoginMechanism["Radius"] = "RADIUS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LoginMechanism["UnknownValue"] = "UNKNOWN_VALUE";
    })(LoginMechanism = App.LoginMechanism || (App.LoginMechanism = {}));
    let ClientType;
    (function (ClientType) {
        ClientType["Confidential"] = "confidential";
        ClientType["Public"] = "public";
        ClientType["Trusted"] = "trusted";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ClientType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ClientType = App.ClientType || (App.ClientType = {}));
    let AllowedOperations;
    (function (AllowedOperations) {
        AllowedOperations["Introspect"] = "introspect";
        AllowedOperations["OnBehalfOfUser"] = "onBehalfOfUser";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AllowedOperations["UnknownValue"] = "UNKNOWN_VALUE";
    })(AllowedOperations = App.AllowedOperations || (App.AllowedOperations = {}));
    let ClientIPChecking;
    (function (ClientIPChecking) {
        ClientIPChecking["Anywhere"] = "anywhere";
        ClientIPChecking["Whitelisted"] = "whitelisted";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ClientIPChecking["UnknownValue"] = "UNKNOWN_VALUE";
    })(ClientIPChecking = App.ClientIPChecking || (App.ClientIPChecking = {}));
    let TrustScope;
    (function (TrustScope) {
        TrustScope["Explicit"] = "Explicit";
        TrustScope["Account"] = "Account";
        TrustScope["Tags"] = "Tags";
        TrustScope["Default"] = "Default";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TrustScope["UnknownValue"] = "UNKNOWN_VALUE";
    })(TrustScope = App.TrustScope || (App.TrustScope = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "meta": obj.meta ? model.Meta.getJsonObj(obj.meta) : undefined,
            "idcsCreatedBy": obj.idcsCreatedBy
                ? model.IdcsCreatedBy.getJsonObj(obj.idcsCreatedBy)
                : undefined,
            "idcsLastModifiedBy": obj.idcsLastModifiedBy
                ? model.IdcsLastModifiedBy.getJsonObj(obj.idcsLastModifiedBy)
                : undefined,
            "tags": obj.tags
                ? obj.tags.map(item => {
                    return model.Tags.getJsonObj(item);
                })
                : undefined,
            "radiusPolicy": obj.radiusPolicy
                ? model.AppRadiusPolicy.getJsonObj(obj.radiusPolicy)
                : undefined,
            "appsNetworkPerimeters": obj.appsNetworkPerimeters
                ? obj.appsNetworkPerimeters.map(item => {
                    return model.AppAppsNetworkPerimeters.getJsonObj(item);
                })
                : undefined,
            "cloudControlProperties": obj.cloudControlProperties
                ? obj.cloudControlProperties.map(item => {
                    return model.AppCloudControlProperties.getJsonObj(item);
                })
                : undefined,
            "editableAttributes": obj.editableAttributes
                ? obj.editableAttributes.map(item => {
                    return model.AppEditableAttributes.getJsonObj(item);
                })
                : undefined,
            "termsOfUse": obj.termsOfUse ? model.AppTermsOfUse.getJsonObj(obj.termsOfUse) : undefined,
            "protectableSecondaryAudiences": obj.protectableSecondaryAudiences
                ? obj.protectableSecondaryAudiences.map(item => {
                    return model.AppProtectableSecondaryAudiences.getJsonObj(item);
                })
                : undefined,
            "idpPolicy": obj.idpPolicy ? model.AppIdpPolicy.getJsonObj(obj.idpPolicy) : undefined,
            "allowedTags": obj.allowedTags
                ? obj.allowedTags.map(item => {
                    return model.AppAllowedTags.getJsonObj(item);
                })
                : undefined,
            "appSignonPolicy": obj.appSignonPolicy
                ? model.AppAppSignonPolicy.getJsonObj(obj.appSignonPolicy)
                : undefined,
            "trustPolicies": obj.trustPolicies
                ? obj.trustPolicies.map(item => {
                    return model.AppTrustPolicies.getJsonObj(item);
                })
                : undefined,
            "signonPolicy": obj.signonPolicy
                ? model.AppSignonPolicy.getJsonObj(obj.signonPolicy)
                : undefined,
            "identityProviders": obj.identityProviders
                ? obj.identityProviders.map(item => {
                    return model.AppIdentityProviders.getJsonObj(item);
                })
                : undefined,
            "accounts": obj.accounts
                ? obj.accounts.map(item => {
                    return model.AppAccounts.getJsonObj(item);
                })
                : undefined,
            "grants": obj.grants
                ? obj.grants.map(item => {
                    return model.AppGrants.getJsonObj(item);
                })
                : undefined,
            "serviceParams": obj.serviceParams
                ? obj.serviceParams.map(item => {
                    return model.AppServiceParams.getJsonObj(item);
                })
                : undefined,
            "attrRenderingMetadata": obj.attrRenderingMetadata
                ? obj.attrRenderingMetadata.map(item => {
                    return model.AppAttrRenderingMetadata.getJsonObj(item);
                })
                : undefined,
            "basedOnTemplate": obj.basedOnTemplate
                ? model.AppBasedOnTemplate.getJsonObj(obj.basedOnTemplate)
                : undefined,
            "grantedAppRoles": obj.grantedAppRoles
                ? obj.grantedAppRoles.map(item => {
                    return model.AppGrantedAppRoles.getJsonObj(item);
                })
                : undefined,
            "samlServiceProvider": obj.samlServiceProvider
                ? model.AppSamlServiceProvider.getJsonObj(obj.samlServiceProvider)
                : undefined,
            "allowedScopes": obj.allowedScopes
                ? obj.allowedScopes.map(item => {
                    return model.AppAllowedScopes.getJsonObj(item);
                })
                : undefined,
            "certificates": obj.certificates
                ? obj.certificates.map(item => {
                    return model.AppCertificates.getJsonObj(item);
                })
                : undefined,
            "aliasApps": obj.aliasApps
                ? obj.aliasApps.map(item => {
                    return model.AppAliasApps.getJsonObj(item);
                })
                : undefined,
            "asOPCService": obj.asOPCService
                ? model.AppAsOPCService.getJsonObj(obj.asOPCService)
                : undefined,
            "adminRoles": obj.adminRoles
                ? obj.adminRoles.map(item => {
                    return model.AppAdminRoles.getJsonObj(item);
                })
                : undefined,
            "userRoles": obj.userRoles
                ? obj.userRoles.map(item => {
                    return model.AppUserRoles.getJsonObj(item);
                })
                : undefined,
            "scopes": obj.scopes
                ? obj.scopes.map(item => {
                    return model.AppScopes.getJsonObj(item);
                })
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:radiusApp:App": obj.urnIetfParamsScimSchemasOracleIdcsExtensionRadiusAppApp
                ? model.AppExtensionRadiusAppApp.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionRadiusAppApp)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:samlServiceProvider:App": obj.urnIetfParamsScimSchemasOracleIdcsExtensionSamlServiceProviderApp
                ? model.AppExtensionSamlServiceProviderApp.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionSamlServiceProviderApp)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:webTierPolicy:App": obj.urnIetfParamsScimSchemasOracleIdcsExtensionWebTierPolicyApp
                ? model.AppExtensionWebTierPolicyApp.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionWebTierPolicyApp)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:managedapp:App": obj.urnIetfParamsScimSchemasOracleIdcsExtensionManagedappApp
                ? model.AppExtensionManagedappApp.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionManagedappApp)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:formFillAppTemplate:AppTemplate": obj.urnIetfParamsScimSchemasOracleIdcsExtensionFormFillAppTemplateAppTemplate
                ? model.AppExtensionFormFillAppTemplateAppTemplate.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionFormFillAppTemplateAppTemplate)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:opcService:App": obj.urnIetfParamsScimSchemasOracleIdcsExtensionOpcServiceApp
                ? model.AppExtensionOpcServiceApp.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionOpcServiceApp)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:kerberosRealm:App": obj.urnIetfParamsScimSchemasOracleIdcsExtensionKerberosRealmApp
                ? model.AppExtensionKerberosRealmApp.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionKerberosRealmApp)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:requestable:App": obj.urnIetfParamsScimSchemasOracleIdcsExtensionRequestableApp
                ? model.AppExtensionRequestableApp.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionRequestableApp)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:formFillApp:App": obj.urnIetfParamsScimSchemasOracleIdcsExtensionFormFillAppApp
                ? model.AppExtensionFormFillAppApp.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionFormFillAppApp)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:dbcs:App": obj.urnIetfParamsScimSchemasOracleIdcsExtensionDbcsApp
                ? model.AppExtensionDbcsApp.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionDbcsApp)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:enterpriseApp:App": obj.urnIetfParamsScimSchemasOracleIdcsExtensionEnterpriseAppApp
                ? model.AppExtensionEnterpriseAppApp.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionEnterpriseAppApp)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:OCITags": obj.urnIetfParamsScimSchemasOracleIdcsExtensionOciTags
                ? model.ExtensionOCITags.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionOciTags)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:multicloudServiceApp:App": obj.urnIetfParamsScimSchemasOracleIdcsExtensionMulticloudServiceAppApp
                ? model.AppExtensionMulticloudServiceAppApp.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionMulticloudServiceAppApp)
                : undefined
        });
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionRadiusAppApp;
        delete jsonObj
            .urnIetfParamsScimSchemasOracleIdcsExtensionSamlServiceProviderApp;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionWebTierPolicyApp;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionManagedappApp;
        delete jsonObj
            .urnIetfParamsScimSchemasOracleIdcsExtensionFormFillAppTemplateAppTemplate;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionOpcServiceApp;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionKerberosRealmApp;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionRequestableApp;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionFormFillAppApp;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionDbcsApp;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionEnterpriseAppApp;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionOciTags;
        delete jsonObj
            .urnIetfParamsScimSchemasOracleIdcsExtensionMulticloudServiceAppApp;
        return jsonObj;
    }
    App.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "meta": obj.meta ? model.Meta.getDeserializedJsonObj(obj.meta) : undefined,
            "idcsCreatedBy": obj.idcsCreatedBy
                ? model.IdcsCreatedBy.getDeserializedJsonObj(obj.idcsCreatedBy)
                : undefined,
            "idcsLastModifiedBy": obj.idcsLastModifiedBy
                ? model.IdcsLastModifiedBy.getDeserializedJsonObj(obj.idcsLastModifiedBy)
                : undefined,
            "tags": obj.tags
                ? obj.tags.map(item => {
                    return model.Tags.getDeserializedJsonObj(item);
                })
                : undefined,
            "radiusPolicy": obj.radiusPolicy
                ? model.AppRadiusPolicy.getDeserializedJsonObj(obj.radiusPolicy)
                : undefined,
            "appsNetworkPerimeters": obj.appsNetworkPerimeters
                ? obj.appsNetworkPerimeters.map(item => {
                    return model.AppAppsNetworkPerimeters.getDeserializedJsonObj(item);
                })
                : undefined,
            "cloudControlProperties": obj.cloudControlProperties
                ? obj.cloudControlProperties.map(item => {
                    return model.AppCloudControlProperties.getDeserializedJsonObj(item);
                })
                : undefined,
            "editableAttributes": obj.editableAttributes
                ? obj.editableAttributes.map(item => {
                    return model.AppEditableAttributes.getDeserializedJsonObj(item);
                })
                : undefined,
            "termsOfUse": obj.termsOfUse
                ? model.AppTermsOfUse.getDeserializedJsonObj(obj.termsOfUse)
                : undefined,
            "protectableSecondaryAudiences": obj.protectableSecondaryAudiences
                ? obj.protectableSecondaryAudiences.map(item => {
                    return model.AppProtectableSecondaryAudiences.getDeserializedJsonObj(item);
                })
                : undefined,
            "idpPolicy": obj.idpPolicy
                ? model.AppIdpPolicy.getDeserializedJsonObj(obj.idpPolicy)
                : undefined,
            "allowedTags": obj.allowedTags
                ? obj.allowedTags.map(item => {
                    return model.AppAllowedTags.getDeserializedJsonObj(item);
                })
                : undefined,
            "appSignonPolicy": obj.appSignonPolicy
                ? model.AppAppSignonPolicy.getDeserializedJsonObj(obj.appSignonPolicy)
                : undefined,
            "trustPolicies": obj.trustPolicies
                ? obj.trustPolicies.map(item => {
                    return model.AppTrustPolicies.getDeserializedJsonObj(item);
                })
                : undefined,
            "signonPolicy": obj.signonPolicy
                ? model.AppSignonPolicy.getDeserializedJsonObj(obj.signonPolicy)
                : undefined,
            "identityProviders": obj.identityProviders
                ? obj.identityProviders.map(item => {
                    return model.AppIdentityProviders.getDeserializedJsonObj(item);
                })
                : undefined,
            "accounts": obj.accounts
                ? obj.accounts.map(item => {
                    return model.AppAccounts.getDeserializedJsonObj(item);
                })
                : undefined,
            "grants": obj.grants
                ? obj.grants.map(item => {
                    return model.AppGrants.getDeserializedJsonObj(item);
                })
                : undefined,
            "serviceParams": obj.serviceParams
                ? obj.serviceParams.map(item => {
                    return model.AppServiceParams.getDeserializedJsonObj(item);
                })
                : undefined,
            "attrRenderingMetadata": obj.attrRenderingMetadata
                ? obj.attrRenderingMetadata.map(item => {
                    return model.AppAttrRenderingMetadata.getDeserializedJsonObj(item);
                })
                : undefined,
            "basedOnTemplate": obj.basedOnTemplate
                ? model.AppBasedOnTemplate.getDeserializedJsonObj(obj.basedOnTemplate)
                : undefined,
            "grantedAppRoles": obj.grantedAppRoles
                ? obj.grantedAppRoles.map(item => {
                    return model.AppGrantedAppRoles.getDeserializedJsonObj(item);
                })
                : undefined,
            "samlServiceProvider": obj.samlServiceProvider
                ? model.AppSamlServiceProvider.getDeserializedJsonObj(obj.samlServiceProvider)
                : undefined,
            "allowedScopes": obj.allowedScopes
                ? obj.allowedScopes.map(item => {
                    return model.AppAllowedScopes.getDeserializedJsonObj(item);
                })
                : undefined,
            "certificates": obj.certificates
                ? obj.certificates.map(item => {
                    return model.AppCertificates.getDeserializedJsonObj(item);
                })
                : undefined,
            "aliasApps": obj.aliasApps
                ? obj.aliasApps.map(item => {
                    return model.AppAliasApps.getDeserializedJsonObj(item);
                })
                : undefined,
            "asOPCService": obj.asOPCService
                ? model.AppAsOPCService.getDeserializedJsonObj(obj.asOPCService)
                : undefined,
            "adminRoles": obj.adminRoles
                ? obj.adminRoles.map(item => {
                    return model.AppAdminRoles.getDeserializedJsonObj(item);
                })
                : undefined,
            "userRoles": obj.userRoles
                ? obj.userRoles.map(item => {
                    return model.AppUserRoles.getDeserializedJsonObj(item);
                })
                : undefined,
            "scopes": obj.scopes
                ? obj.scopes.map(item => {
                    return model.AppScopes.getDeserializedJsonObj(item);
                })
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionRadiusAppApp": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:radiusApp:App"]
                ? model.AppExtensionRadiusAppApp.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:radiusApp:App"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionSamlServiceProviderApp": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:samlServiceProvider:App"]
                ? model.AppExtensionSamlServiceProviderApp.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:samlServiceProvider:App"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionWebTierPolicyApp": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:webTierPolicy:App"]
                ? model.AppExtensionWebTierPolicyApp.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:webTierPolicy:App"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionManagedappApp": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:managedapp:App"]
                ? model.AppExtensionManagedappApp.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:managedapp:App"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionFormFillAppTemplateAppTemplate": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:formFillAppTemplate:AppTemplate"]
                ? model.AppExtensionFormFillAppTemplateAppTemplate.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:formFillAppTemplate:AppTemplate"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionOpcServiceApp": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:opcService:App"]
                ? model.AppExtensionOpcServiceApp.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:opcService:App"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionKerberosRealmApp": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:kerberosRealm:App"]
                ? model.AppExtensionKerberosRealmApp.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:kerberosRealm:App"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionRequestableApp": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:requestable:App"]
                ? model.AppExtensionRequestableApp.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:requestable:App"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionFormFillAppApp": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:formFillApp:App"]
                ? model.AppExtensionFormFillAppApp.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:formFillApp:App"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionDbcsApp": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:dbcs:App"]
                ? model.AppExtensionDbcsApp.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:dbcs:App"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionEnterpriseAppApp": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:enterpriseApp:App"]
                ? model.AppExtensionEnterpriseAppApp.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:enterpriseApp:App"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionOciTags": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:OCITags"]
                ? model.ExtensionOCITags.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:OCITags"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionMulticloudServiceAppApp": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:multicloudServiceApp:App"]
                ? model.AppExtensionMulticloudServiceAppApp.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:multicloudServiceApp:App"])
                : undefined
        });
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:radiusApp:App"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:samlServiceProvider:App"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:webTierPolicy:App"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:managedapp:App"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:formFillAppTemplate:AppTemplate"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:opcService:App"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:kerberosRealm:App"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:requestable:App"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:formFillApp:App"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:dbcs:App"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:enterpriseApp:App"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:OCITags"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:multicloudServiceApp:App"];
        return jsonObj;
    }
    App.getDeserializedJsonObj = getDeserializedJsonObj;
})(App = exports.App || (exports.App = {}));
