// oci-identitydomains/lib/model/grant.js
"use strict";
/**
 * Identity Domains API
 * Use the Identity Domains API to manage resources within an identity domain, for example, users, dynamic resource groups, groups, and identity providers. For information about managing resources within identity domains, see [Identity and Access Management (with identity domains)](https://docs.oracle.com/iaas/Content/Identity/home.htm).
Use this pattern to construct endpoints for identity domains: {@code https://<domainURL>/admin/v1/}. See [Finding an Identity Domain URL](https://docs.oracle.com/en-us/iaas/Content/Identity/api-getstarted/locate-identity-domain-url.htm) to locate the domain URL you need.
Use the table of contents and search tool to explore the Identity Domains API.
 * OpenAPI spec version: v1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Grant = void 0;
const model = __importStar(require("../model"));
var Grant;
(function (Grant) {
    let IdcsPreventedOperations;
    (function (IdcsPreventedOperations) {
        IdcsPreventedOperations["Replace"] = "replace";
        IdcsPreventedOperations["Update"] = "update";
        IdcsPreventedOperations["Delete"] = "delete";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        IdcsPreventedOperations["UnknownValue"] = "UNKNOWN_VALUE";
    })(IdcsPreventedOperations = Grant.IdcsPreventedOperations || (Grant.IdcsPreventedOperations = {}));
    let GrantMechanism;
    (function (GrantMechanism) {
        GrantMechanism["ImportApproleMembers"] = "IMPORT_APPROLE_MEMBERS";
        GrantMechanism["AdministratorToUser"] = "ADMINISTRATOR_TO_USER";
        GrantMechanism["AdministratorToDelegatedUser"] = "ADMINISTRATOR_TO_DELEGATED_USER";
        GrantMechanism["AdministratorToGroup"] = "ADMINISTRATOR_TO_GROUP";
        GrantMechanism["ServiceManagerToUser"] = "SERVICE_MANAGER_TO_USER";
        GrantMechanism["AdministratorToApp"] = "ADMINISTRATOR_TO_APP";
        GrantMechanism["ServiceManagerToApp"] = "SERVICE_MANAGER_TO_APP";
        GrantMechanism["OpcInfraToApp"] = "OPC_INFRA_TO_APP";
        GrantMechanism["GroupMembership"] = "GROUP_MEMBERSHIP";
        GrantMechanism["ImportGrants"] = "IMPORT_GRANTS";
        GrantMechanism["SyncToUser"] = "SYNC_TO_USER";
        GrantMechanism["AccessRequest"] = "ACCESS_REQUEST";
        GrantMechanism["AppEntitlementCollection"] = "APP_ENTITLEMENT_COLLECTION";
        GrantMechanism["AdministratorToDynamicResourceGroup"] = "ADMINISTRATOR_TO_DYNAMIC_RESOURCE_GROUP";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        GrantMechanism["UnknownValue"] = "UNKNOWN_VALUE";
    })(GrantMechanism = Grant.GrantMechanism || (Grant.GrantMechanism = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "meta": obj.meta ? model.Meta.getJsonObj(obj.meta) : undefined,
            "idcsCreatedBy": obj.idcsCreatedBy
                ? model.IdcsCreatedBy.getJsonObj(obj.idcsCreatedBy)
                : undefined,
            "idcsLastModifiedBy": obj.idcsLastModifiedBy
                ? model.IdcsLastModifiedBy.getJsonObj(obj.idcsLastModifiedBy)
                : undefined,
            "tags": obj.tags
                ? obj.tags.map(item => {
                    return model.Tags.getJsonObj(item);
                })
                : undefined,
            "appEntitlementCollection": obj.appEntitlementCollection
                ? model.GrantAppEntitlementCollection.getJsonObj(obj.appEntitlementCollection)
                : undefined,
            "grantor": obj.grantor ? model.GrantGrantor.getJsonObj(obj.grantor) : undefined,
            "grantee": obj.grantee ? model.GrantGrantee.getJsonObj(obj.grantee) : undefined,
            "app": obj.app ? model.GrantApp.getJsonObj(obj.app) : undefined,
            "entitlement": obj.entitlement
                ? model.GrantEntitlement.getJsonObj(obj.entitlement)
                : undefined
        });
        return jsonObj;
    }
    Grant.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "meta": obj.meta ? model.Meta.getDeserializedJsonObj(obj.meta) : undefined,
            "idcsCreatedBy": obj.idcsCreatedBy
                ? model.IdcsCreatedBy.getDeserializedJsonObj(obj.idcsCreatedBy)
                : undefined,
            "idcsLastModifiedBy": obj.idcsLastModifiedBy
                ? model.IdcsLastModifiedBy.getDeserializedJsonObj(obj.idcsLastModifiedBy)
                : undefined,
            "tags": obj.tags
                ? obj.tags.map(item => {
                    return model.Tags.getDeserializedJsonObj(item);
                })
                : undefined,
            "appEntitlementCollection": obj.appEntitlementCollection
                ? model.GrantAppEntitlementCollection.getDeserializedJsonObj(obj.appEntitlementCollection)
                : undefined,
            "grantor": obj.grantor ? model.GrantGrantor.getDeserializedJsonObj(obj.grantor) : undefined,
            "grantee": obj.grantee ? model.GrantGrantee.getDeserializedJsonObj(obj.grantee) : undefined,
            "app": obj.app ? model.GrantApp.getDeserializedJsonObj(obj.app) : undefined,
            "entitlement": obj.entitlement
                ? model.GrantEntitlement.getDeserializedJsonObj(obj.entitlement)
                : undefined
        });
        return jsonObj;
    }
    Grant.getDeserializedJsonObj = getDeserializedJsonObj;
})(Grant = exports.Grant || (exports.Grant = {}));
