// oci-identitydomains/lib/model/index.js
"use strict";
/**
 * Identity Domains API
 * Use the Identity Domains API to manage resources within an identity domain, for example, users, dynamic resource groups, groups, and identity providers. For information about managing resources within identity domains, see [Identity and Access Management (with identity domains)](https://docs.oracle.com/iaas/Content/Identity/home.htm).
Use this pattern to construct endpoints for identity domains: {@code https://<domainURL>/admin/v1/}. See [Finding an Identity Domain URL](https://docs.oracle.com/en-us/iaas/Content/Identity/api-getstarted/locate-identity-domain-url.htm) to locate the domain URL you need.
Use the table of contents and search tool to explore the Identity Domains API.
 * OpenAPI spec version: v1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppFlatFileBundleConfigurationProperties = exports.AppExtensionWebTierPolicyApp = exports.AppExtensionSamlServiceProviderApp = exports.AppExtensionRequestableApp = exports.AppExtensionRadiusAppApp = exports.AppExtensionOpcServiceApp = exports.AppExtensionMulticloudServiceAppApp = exports.AppExtensionManagedappApp = exports.AppExtensionKerberosRealmApp = exports.AppExtensionFormFillAppTemplateAppTemplate = exports.AppExtensionFormFillAppApp = exports.AppExtensionEnterpriseAppApp = exports.AppExtensionDbcsApp = exports.AppEditableAttributes = exports.AppDomainApp = exports.AppDenyAuthzPolicy = exports.AppConnectorBundle = exports.AppCloudControlProperties = exports.AppCertificates = exports.AppBundlePoolConfiguration = exports.AppBundleConfigurationProperties = exports.AppBasedOnTemplate = exports.AppAttrRenderingMetadata = exports.AppAsOPCService = exports.AppAppsNetworkPerimeters = exports.AppAppSignonPolicy = exports.AppAppResources = exports.AppAllowedTags = exports.AppAllowedScopes = exports.AppAllowAuthzPolicy = exports.AppAliasApps = exports.AppAdminRoles = exports.AppAccounts = exports.App = exports.ApiKeys = exports.ApiKeyUser = exports.ApiKeySearchRequest = exports.ApiKey = exports.Addresses = exports.AccountRecoverySettings = exports.AccountRecoverySetting = exports.AccountMgmtInfos = exports.AccountMgmtInfoUserWalletArtifact = exports.AccountMgmtInfoSearchRequest = exports.AccountMgmtInfoResourceType = exports.AccountMgmtInfoOwner = exports.AccountMgmtInfoObjectClass = exports.AccountMgmtInfoMatchingOwners = exports.AccountMgmtInfoApp = exports.AccountMgmtInfo = void 0;
exports.AuthenticationFactorSettingsClientAppSettings = exports.AuthenticationFactorSettingsBypassCodeSettings = exports.AuthenticationFactorSettings = exports.AuthenticationFactorSetting = exports.AuthTokens = exports.AuthTokenUser = exports.AuthTokenSearchRequest = exports.AuthToken = exports.AttributeSets = exports.Apps = exports.ApprovalWorkflows = exports.ApprovalWorkflowSteps = exports.ApprovalWorkflowStepApprovers = exports.ApprovalWorkflowStep = exports.ApprovalWorkflowMaxDuration = exports.ApprovalWorkflowAssignments = exports.ApprovalWorkflowAssignmentAssignedTo = exports.ApprovalWorkflowAssignmentApprovalWorkflow = exports.ApprovalWorkflowAssignment = exports.ApprovalWorkflowApprovalWorkflowSteps = exports.ApprovalWorkflow = exports.AppUserRoles = exports.AppUserAssertionAttributes = exports.AppTrustPolicies = exports.AppThreeLeggedOAuthCredential = exports.AppTermsOfUse = exports.AppStatusChanger = exports.AppSignonPolicy = exports.AppServiceParams = exports.AppSearchRequest = exports.AppScopes = exports.AppSamlServiceProvider = exports.AppRoles = exports.AppRoleSearchRequest = exports.AppRoleMembers = exports.AppRoleApp = exports.AppRole = exports.AppRadiusPolicy = exports.AppProtectableSecondaryAudiences = exports.AppOutboundAssertionAttributes = exports.AppObjectClasses = exports.AppIdpPolicy = exports.AppIdentityProviders = exports.AppIdentityBridges = exports.AppGroupMembershipToReturn = exports.AppGroupAssertionAttributes = exports.AppGrants = exports.AppGrantedAppRoles = exports.AppFormFillUrlMatch = exports.AppFlatFileConnectorBundle = void 0;
exports.DynamicResourceGroupGrants = exports.DynamicResourceGroupDynamicGroupAppRoles = exports.DynamicResourceGroup = exports.DefinedTags = exports.CustomerSecretKeys = exports.CustomerSecretKeyUser = exports.CustomerSecretKeySearchRequest = exports.CustomerSecretKey = exports.Conditions = exports.ConditionSearchRequest = exports.ConditionExtensionOciconsolesignonpolicyconsentPolicy = exports.Condition = exports.CloudGates = exports.CloudGateServers = exports.CloudGateServerSearchRequest = exports.CloudGateServerCloudGate = exports.CloudGateServer = exports.CloudGateSearchRequest = exports.CloudGateOauthClient = exports.CloudGateMappings = exports.CloudGateMappingUpstreamServerGroup = exports.CloudGateMappingServer = exports.CloudGateMappingSearchRequest = exports.CloudGateMappingGatewayApp = exports.CloudGateMappingCloudGate = exports.CloudGateMapping = exports.CloudGateExtUpstreamServers = exports.CloudGateExtUpstreamServerGroups = exports.CloudGateExtServers = exports.CloudGateExtMappings = exports.CloudGate = exports.BrandingSettingsLoginTexts = exports.BrandingSettingsImages = exports.BrandingSettingsDefaultLoginTexts = exports.BrandingSettingsDefaultImages = exports.BrandingSettingsDefaultCompanyNames = exports.BrandingSettingsCompanyNames = exports.BrandingSettings = exports.BrandingSetting = exports.AuthenticationFactorsRemoverUser = exports.AuthenticationFactorsRemover = exports.AuthenticationFactorSettingsTotpSettings = exports.AuthenticationFactorSettingsThirdPartyFactor = exports.AuthenticationFactorSettingsSearchRequest = exports.AuthenticationFactorSettingsNotificationSettings = exports.AuthenticationFactorSettingsIdentityStoreSettings = exports.AuthenticationFactorSettingsEndpointRestrictions = exports.AuthenticationFactorSettingsEmailSettings = exports.AuthenticationFactorSettingsDuoSecuritySettings = exports.AuthenticationFactorSettingsCompliancePolicy = void 0;
exports.GroupExtSyncedFromApp = exports.GroupExtPasswordPolicy = exports.GroupExtOwners = exports.GroupExtInstanceLevelSchemaNames = exports.GroupExtGrants = exports.GroupExtDomainLevelSchemaNames = exports.GroupExtAppRoles = exports.Group = exports.Grants = exports.GrantSearchRequest = exports.GrantGrantor = exports.GrantGrantee = exports.GrantEntitlement = exports.GrantAppEntitlementCollection = exports.GrantApp = exports.Grant = exports.FreeformTags = exports.ExtensionX509IdentityProvider = exports.ExtensionUserUser = exports.ExtensionUserStateUser = exports.ExtensionUserCredentialsUser = exports.ExtensionThirdPartyAuthenticationFactorSettings = exports.ExtensionTermsOfUseUser = exports.ExtensionSocialIdentityProvider = exports.ExtensionSocialAccountUser = exports.ExtensionSffUser = exports.ExtensionSelfRegistrationUser = exports.ExtensionSelfChangeUser = exports.ExtensionSecurityQuestionsUser = exports.ExtensionRequestableGroup = exports.ExtensionPosixUser = exports.ExtensionPosixGroup = exports.ExtensionPasswordlessUser = exports.ExtensionPasswordStateUser = exports.ExtensionOCITags = exports.ExtensionMfaUser = exports.ExtensionMessagesError = exports.ExtensionMeUser = exports.ExtensionKerberosUserUser = exports.ExtensionGroupGroup = exports.ExtensionFidoAuthenticationFactorSettings = exports.ExtensionEnterprise20User = exports.ExtensionDynamicGroup = exports.ExtensionDbcsGroup = exports.ExtensionDbUserUser = exports.ExtensionDbCredentialsUser = exports.ExtensionCapabilitiesUser = exports.ExtensionAdaptiveUser = exports.DynamicResourceGroups = exports.DynamicResourceGroupSearchRequest = void 0;
exports.MyApps = exports.MyAppUserWalletArtifact = exports.MyAppSearchRequest = exports.MyAppOwner = exports.MyAppApp = exports.MyApp = exports.MyApiKeys = exports.MyApiKeyUser = exports.MyApiKey = exports.Meta = exports.MeX509Certificates = exports.MeRoles = exports.MePhotos = exports.MePhoneNumbers = exports.MePasswordChanger = exports.MeName = exports.MeIms = exports.MeGroups = exports.MeEntitlements = exports.MeEmails = exports.Me = exports.KmsiSettingsSearchRequest = exports.KmsiSettings = exports.KmsiSetting = exports.IdentitySettingsTokens = exports.IdentitySettingsSearchRequest = exports.IdentitySettingsPOSIXUid = exports.IdentitySettingsPOSIXGid = exports.IdentitySettingsMyProfile = exports.IdentitySettings = exports.IdentitySetting = exports.IdentityProviders = exports.IdentityProviderSearchRequest = exports.IdentityProviderRelayIdpParamMappings = exports.IdentityProviderJitUserProvGroupMappings = exports.IdentityProviderJitUserProvAttributes = exports.IdentityProviderJitUserProvAssignedGroups = exports.IdentityProviderJitProvAssignedGroups = exports.IdentityProviderCorrelationPolicy = exports.IdentityProvider = exports.IdentityPropagationTrusts = exports.IdentityPropagationTrustKeytab = exports.IdentityPropagationTrustImpersonationServiceUsers = exports.IdentityPropagationTrust = exports.IdcsPreventedOperations = exports.IdcsLastModifiedBy = exports.IdcsCreatedBy = exports.Groups = exports.GroupSearchRequest = exports.GroupMembers = void 0;
exports.MySupportAccountUser = exports.MySupportAccount = exports.MySmtpCredentials = exports.MySmtpCredentialUser = exports.MySmtpCredential = exports.MyRequests = exports.MyRequestableGroups = exports.MyRequestableGroupSearchRequest = exports.MyRequestableGroupMembers = exports.MyRequestableGroup = exports.MyRequestSearchRequest = exports.MyRequestRequestor = exports.MyRequestRequesting = exports.MyRequestApprovalDetails = exports.MyRequest = exports.MyPendingApprovals = exports.MyPendingApproval = exports.MyOAuth2ClientCredentials = exports.MyOAuth2ClientCredentialUser = exports.MyOAuth2ClientCredentialScopes = exports.MyOAuth2ClientCredential = exports.MyGroups = exports.MyGroupSearchRequest = exports.MyGroupMembers = exports.MyGroup = exports.MyDevices = exports.MyDeviceUser = exports.MyDeviceThirdPartyFactor = exports.MyDevicePushNotificationTarget = exports.MyDeviceNonCompliances = exports.MyDeviceAuthenticationFactors = exports.MyDeviceAdditionalAttributes = exports.MyDevice = exports.MyCustomerSecretKeys = exports.MyCustomerSecretKeyUser = exports.MyCustomerSecretKey = exports.MyCompletedApprovals = exports.MyCompletedApproval = exports.MyAuthenticationFactorsRemoverUser = exports.MyAuthenticationFactorsRemover = exports.MyAuthenticationFactorValidatorThirdPartyFactor = exports.MyAuthenticationFactorValidatorSecurityQuestions = exports.MyAuthenticationFactorValidatorAdditionalAttributes = exports.MyAuthenticationFactorValidator = exports.MyAuthenticationFactorInitiatorThirdPartyFactor = exports.MyAuthenticationFactorInitiatorAdditionalAttributes = exports.MyAuthenticationFactorInitiator = exports.MyAuthTokens = exports.MyAuthTokenUser = exports.MyAuthToken = void 0;
exports.ResourceTypeSchemaAttributes = exports.ResourceTypeSchemaAttributeSearchRequest = exports.ResourceTypeSchemaAttribute = exports.PolicySearchRequest = exports.PolicyRules = exports.PolicyPolicyType = exports.PolicyExtensionOciconsolesignonpolicyconsentPolicy = exports.Policy = exports.Policies = exports.PatchOp = exports.PasswordPolicySearchRequest = exports.PasswordPolicyGroups = exports.PasswordPolicyConfiguredPasswordPolicyRules = exports.PasswordPolicy = exports.PasswordPolicies = exports.Operations = exports.OciConsoleSignOnPolicyConsents = exports.OciConsoleSignOnPolicyConsentSearchRequest = exports.OciConsoleSignOnPolicyConsentPolicyResource = exports.OciConsoleSignOnPolicyConsentModifiedResource = exports.OciConsoleSignOnPolicyConsentConsentSignedBy = exports.OciConsoleSignOnPolicyConsent = exports.OAuthPartnerCertificates = exports.OAuthPartnerCertificateSearchRequest = exports.OAuthPartnerCertificate = exports.OAuthClientCertificates = exports.OAuthClientCertificateSearchRequest = exports.OAuthClientCertificate = exports.OAuth2ClientCredentials = exports.OAuth2ClientCredentialUser = exports.OAuth2ClientCredentialSearchRequest = exports.OAuth2ClientCredentialScopes = exports.OAuth2ClientCredential = exports.NotificationSettingsSearchRequest = exports.NotificationSettingsFromEmailAddress = exports.NotificationSettingsEventSettings = exports.NotificationSettings = exports.NotificationSetting = exports.NetworkPerimeters = exports.NetworkPerimeterSearchRequest = exports.NetworkPerimeterIpAddresses = exports.NetworkPerimeter = exports.MyUserDbCredentialsUser = exports.MyUserDbCredentials = exports.MyUserDbCredential = exports.MyTrustedUserAgents = exports.MyTrustedUserAgentUser = exports.MyTrustedUserAgentTrustedFactors = exports.MyTrustedUserAgent = exports.MySupportAccounts = void 0;
exports.SettingsTenantCustomClaims = exports.SettingsSearchRequest = exports.SettingsPurgeConfigs = exports.SettingsLoginTexts = exports.SettingsImages = exports.SettingsDefaultLoginTexts = exports.SettingsDefaultImages = exports.SettingsDefaultCompanyNames = exports.SettingsCompanyNames = exports.SettingsCloudGateCorsSettings = exports.SettingsCertificateValidation = exports.Settings = exports.Setting = exports.SelfRegistrationProfiles = exports.SelfRegistrationProfileUserAttributes = exports.SelfRegistrationProfileSearchRequest = exports.SelfRegistrationProfileHeaderText = exports.SelfRegistrationProfileFooterText = exports.SelfRegistrationProfileEmailTemplate = exports.SelfRegistrationProfileDisplayName = exports.SelfRegistrationProfileDefaultGroups = exports.SelfRegistrationProfileConsentText = exports.SelfRegistrationProfileAfterSubmitText = exports.SelfRegistrationProfile = exports.SecurityQuestions = exports.SecurityQuestionSettingsSearchRequest = exports.SecurityQuestionSettings = exports.SecurityQuestionSetting = exports.SecurityQuestionSearchRequest = exports.SecurityQuestionQuestionText = exports.SecurityQuestion = exports.Schemas = exports.SchemaSubAttributes = exports.SchemaSearchRequest = exports.SchemaResolveValueUsingResourceType = exports.SchemaReferencedResourceTypeUniqueAttributeNameMappings = exports.SchemaLocalizedDisplayName = exports.SchemaLocalizedCanonicalValues = exports.SchemaIdcsCsvAttributeNameMappings = exports.SchemaIdcsComplexAttributeNameMappings = exports.SchemaAttributes = exports.Schema = exports.Rules = exports.RuleSearchRequest = exports.RuleReturn = exports.RulePolicyType = exports.RuleExtensionOciconsolesignonpolicyconsentPolicy = exports.RuleConditionGroup = exports.Rule = exports.RestoreOciConsolePolicy = void 0;
exports.UserExtSupportAccounts = exports.UserExtSocialAccounts = exports.UserExtSmtpCredentials = exports.UserExtSelfRegistrationProfile = exports.UserExtSecQuestions = exports.UserExtRiskScores = exports.UserExtRecoveryLocked = exports.UserExtRealmUsers = exports.UserExtPreferredDevice = exports.UserExtPasswordVerifiers = exports.UserExtOAuth2ClientCredentials = exports.UserExtManager = exports.UserExtLocked = exports.UserExtIdcsAppRolesLimitedToGroups = exports.UserExtGrants = exports.UserExtFactorIdentifier = exports.UserExtDevices = exports.UserExtDelegatedAuthenticationTargetApp = exports.UserExtDbCredentials = exports.UserExtCustomerSecretKeys = exports.UserExtBypassCodes = exports.UserExtAuthTokens = exports.UserExtApplicablePasswordPolicy = exports.UserExtApplicableAuthenticationTargetApp = exports.UserExtAppRoles = exports.UserExtApiKeys = exports.UserExtAccounts = exports.UserEntitlements = exports.UserEmails = exports.UserDbCredentialsUser = exports.UserDbCredentialsSearchRequest = exports.UserDbCredentials = exports.UserDbCredential = exports.UserCapabilitiesChanger = exports.UserAttributesSettingsSearchRequest = exports.UserAttributesSettingsAttributeSettings = exports.UserAttributesSettings = exports.UserAttributesSetting = exports.User = exports.Tags = exports.SortOrder = exports.SocialIdentityProviders = exports.SocialIdentityProviderSearchRequest = exports.SocialIdentityProviderRelayIdpParamMappings = exports.SocialIdentityProviderJitProvAssignedGroups = exports.SocialIdentityProvider = exports.SmtpCredentials = exports.SmtpCredentialUser = exports.SmtpCredentialSearchRequest = exports.SmtpCredential = void 0;
exports.Users = exports.UserX509Certificates = exports.UserStatusChanger = exports.UserSearchRequest = exports.UserRoles = exports.UserPhotos = exports.UserPhoneNumbers = exports.UserPasswordResetterUserToken = exports.UserPasswordResetter = exports.UserPasswordChanger = exports.UserName = exports.UserIms = exports.UserGroups = exports.UserExtUserToken = exports.UserExtTrustedUserAgents = exports.UserExtTermsOfUseConsents = exports.UserExtSyncedFromApp = void 0;
const AccountMgmtInfo = __importStar(require("./account-mgmt-info"));
exports.AccountMgmtInfo = AccountMgmtInfo.AccountMgmtInfo;
const AccountMgmtInfoApp = __importStar(require("./account-mgmt-info-app"));
exports.AccountMgmtInfoApp = AccountMgmtInfoApp.AccountMgmtInfoApp;
const AccountMgmtInfoMatchingOwners = __importStar(require("./account-mgmt-info-matching-owners"));
exports.AccountMgmtInfoMatchingOwners = AccountMgmtInfoMatchingOwners.AccountMgmtInfoMatchingOwners;
const AccountMgmtInfoObjectClass = __importStar(require("./account-mgmt-info-object-class"));
exports.AccountMgmtInfoObjectClass = AccountMgmtInfoObjectClass.AccountMgmtInfoObjectClass;
const AccountMgmtInfoOwner = __importStar(require("./account-mgmt-info-owner"));
exports.AccountMgmtInfoOwner = AccountMgmtInfoOwner.AccountMgmtInfoOwner;
const AccountMgmtInfoResourceType = __importStar(require("./account-mgmt-info-resource-type"));
exports.AccountMgmtInfoResourceType = AccountMgmtInfoResourceType.AccountMgmtInfoResourceType;
const AccountMgmtInfoSearchRequest = __importStar(require("./account-mgmt-info-search-request"));
exports.AccountMgmtInfoSearchRequest = AccountMgmtInfoSearchRequest.AccountMgmtInfoSearchRequest;
const AccountMgmtInfoUserWalletArtifact = __importStar(require("./account-mgmt-info-user-wallet-artifact"));
exports.AccountMgmtInfoUserWalletArtifact = AccountMgmtInfoUserWalletArtifact.AccountMgmtInfoUserWalletArtifact;
const AccountMgmtInfos = __importStar(require("./account-mgmt-infos"));
exports.AccountMgmtInfos = AccountMgmtInfos.AccountMgmtInfos;
const AccountRecoverySetting = __importStar(require("./account-recovery-setting"));
exports.AccountRecoverySetting = AccountRecoverySetting.AccountRecoverySetting;
const AccountRecoverySettings = __importStar(require("./account-recovery-settings"));
exports.AccountRecoverySettings = AccountRecoverySettings.AccountRecoverySettings;
const Addresses = __importStar(require("./addresses"));
exports.Addresses = Addresses.Addresses;
const ApiKey = __importStar(require("./api-key"));
exports.ApiKey = ApiKey.ApiKey;
const ApiKeySearchRequest = __importStar(require("./api-key-search-request"));
exports.ApiKeySearchRequest = ApiKeySearchRequest.ApiKeySearchRequest;
const ApiKeyUser = __importStar(require("./api-key-user"));
exports.ApiKeyUser = ApiKeyUser.ApiKeyUser;
const ApiKeys = __importStar(require("./api-keys"));
exports.ApiKeys = ApiKeys.ApiKeys;
const App = __importStar(require("./app"));
exports.App = App.App;
const AppAccounts = __importStar(require("./app-accounts"));
exports.AppAccounts = AppAccounts.AppAccounts;
const AppAdminRoles = __importStar(require("./app-admin-roles"));
exports.AppAdminRoles = AppAdminRoles.AppAdminRoles;
const AppAliasApps = __importStar(require("./app-alias-apps"));
exports.AppAliasApps = AppAliasApps.AppAliasApps;
const AppAllowAuthzPolicy = __importStar(require("./app-allow-authz-policy"));
exports.AppAllowAuthzPolicy = AppAllowAuthzPolicy.AppAllowAuthzPolicy;
const AppAllowedScopes = __importStar(require("./app-allowed-scopes"));
exports.AppAllowedScopes = AppAllowedScopes.AppAllowedScopes;
const AppAllowedTags = __importStar(require("./app-allowed-tags"));
exports.AppAllowedTags = AppAllowedTags.AppAllowedTags;
const AppAppResources = __importStar(require("./app-app-resources"));
exports.AppAppResources = AppAppResources.AppAppResources;
const AppAppSignonPolicy = __importStar(require("./app-app-signon-policy"));
exports.AppAppSignonPolicy = AppAppSignonPolicy.AppAppSignonPolicy;
const AppAppsNetworkPerimeters = __importStar(require("./app-apps-network-perimeters"));
exports.AppAppsNetworkPerimeters = AppAppsNetworkPerimeters.AppAppsNetworkPerimeters;
const AppAsOPCService = __importStar(require("./app-as-op-cservice"));
exports.AppAsOPCService = AppAsOPCService.AppAsOPCService;
const AppAttrRenderingMetadata = __importStar(require("./app-attr-rendering-metadata"));
exports.AppAttrRenderingMetadata = AppAttrRenderingMetadata.AppAttrRenderingMetadata;
const AppBasedOnTemplate = __importStar(require("./app-based-on-template"));
exports.AppBasedOnTemplate = AppBasedOnTemplate.AppBasedOnTemplate;
const AppBundleConfigurationProperties = __importStar(require("./app-bundle-configuration-properties"));
exports.AppBundleConfigurationProperties = AppBundleConfigurationProperties.AppBundleConfigurationProperties;
const AppBundlePoolConfiguration = __importStar(require("./app-bundle-pool-configuration"));
exports.AppBundlePoolConfiguration = AppBundlePoolConfiguration.AppBundlePoolConfiguration;
const AppCertificates = __importStar(require("./app-certificates"));
exports.AppCertificates = AppCertificates.AppCertificates;
const AppCloudControlProperties = __importStar(require("./app-cloud-control-properties"));
exports.AppCloudControlProperties = AppCloudControlProperties.AppCloudControlProperties;
const AppConnectorBundle = __importStar(require("./app-connector-bundle"));
exports.AppConnectorBundle = AppConnectorBundle.AppConnectorBundle;
const AppDenyAuthzPolicy = __importStar(require("./app-deny-authz-policy"));
exports.AppDenyAuthzPolicy = AppDenyAuthzPolicy.AppDenyAuthzPolicy;
const AppDomainApp = __importStar(require("./app-domain-app"));
exports.AppDomainApp = AppDomainApp.AppDomainApp;
const AppEditableAttributes = __importStar(require("./app-editable-attributes"));
exports.AppEditableAttributes = AppEditableAttributes.AppEditableAttributes;
const AppExtensionDbcsApp = __importStar(require("./app-extension-dbcs-app"));
exports.AppExtensionDbcsApp = AppExtensionDbcsApp.AppExtensionDbcsApp;
const AppExtensionEnterpriseAppApp = __importStar(require("./app-extension-enterprise-app-app"));
exports.AppExtensionEnterpriseAppApp = AppExtensionEnterpriseAppApp.AppExtensionEnterpriseAppApp;
const AppExtensionFormFillAppApp = __importStar(require("./app-extension-form-fill-app-app"));
exports.AppExtensionFormFillAppApp = AppExtensionFormFillAppApp.AppExtensionFormFillAppApp;
const AppExtensionFormFillAppTemplateAppTemplate = __importStar(require("./app-extension-form-fill-app-template-app-template"));
exports.AppExtensionFormFillAppTemplateAppTemplate = AppExtensionFormFillAppTemplateAppTemplate.AppExtensionFormFillAppTemplateAppTemplate;
const AppExtensionKerberosRealmApp = __importStar(require("./app-extension-kerberos-realm-app"));
exports.AppExtensionKerberosRealmApp = AppExtensionKerberosRealmApp.AppExtensionKerberosRealmApp;
const AppExtensionManagedappApp = __importStar(require("./app-extension-managedapp-app"));
exports.AppExtensionManagedappApp = AppExtensionManagedappApp.AppExtensionManagedappApp;
const AppExtensionMulticloudServiceAppApp = __importStar(require("./app-extension-multicloud-service-app-app"));
exports.AppExtensionMulticloudServiceAppApp = AppExtensionMulticloudServiceAppApp.AppExtensionMulticloudServiceAppApp;
const AppExtensionOpcServiceApp = __importStar(require("./app-extension-opc-service-app"));
exports.AppExtensionOpcServiceApp = AppExtensionOpcServiceApp.AppExtensionOpcServiceApp;
const AppExtensionRadiusAppApp = __importStar(require("./app-extension-radius-app-app"));
exports.AppExtensionRadiusAppApp = AppExtensionRadiusAppApp.AppExtensionRadiusAppApp;
const AppExtensionRequestableApp = __importStar(require("./app-extension-requestable-app"));
exports.AppExtensionRequestableApp = AppExtensionRequestableApp.AppExtensionRequestableApp;
const AppExtensionSamlServiceProviderApp = __importStar(require("./app-extension-saml-service-provider-app"));
exports.AppExtensionSamlServiceProviderApp = AppExtensionSamlServiceProviderApp.AppExtensionSamlServiceProviderApp;
const AppExtensionWebTierPolicyApp = __importStar(require("./app-extension-web-tier-policy-app"));
exports.AppExtensionWebTierPolicyApp = AppExtensionWebTierPolicyApp.AppExtensionWebTierPolicyApp;
const AppFlatFileBundleConfigurationProperties = __importStar(require("./app-flat-file-bundle-configuration-properties"));
exports.AppFlatFileBundleConfigurationProperties = AppFlatFileBundleConfigurationProperties.AppFlatFileBundleConfigurationProperties;
const AppFlatFileConnectorBundle = __importStar(require("./app-flat-file-connector-bundle"));
exports.AppFlatFileConnectorBundle = AppFlatFileConnectorBundle.AppFlatFileConnectorBundle;
const AppFormFillUrlMatch = __importStar(require("./app-form-fill-url-match"));
exports.AppFormFillUrlMatch = AppFormFillUrlMatch.AppFormFillUrlMatch;
const AppGrantedAppRoles = __importStar(require("./app-granted-app-roles"));
exports.AppGrantedAppRoles = AppGrantedAppRoles.AppGrantedAppRoles;
const AppGrants = __importStar(require("./app-grants"));
exports.AppGrants = AppGrants.AppGrants;
const AppGroupAssertionAttributes = __importStar(require("./app-group-assertion-attributes"));
exports.AppGroupAssertionAttributes = AppGroupAssertionAttributes.AppGroupAssertionAttributes;
const AppGroupMembershipToReturn = __importStar(require("./app-group-membership-to-return"));
exports.AppGroupMembershipToReturn = AppGroupMembershipToReturn.AppGroupMembershipToReturn;
const AppIdentityBridges = __importStar(require("./app-identity-bridges"));
exports.AppIdentityBridges = AppIdentityBridges.AppIdentityBridges;
const AppIdentityProviders = __importStar(require("./app-identity-providers"));
exports.AppIdentityProviders = AppIdentityProviders.AppIdentityProviders;
const AppIdpPolicy = __importStar(require("./app-idp-policy"));
exports.AppIdpPolicy = AppIdpPolicy.AppIdpPolicy;
const AppObjectClasses = __importStar(require("./app-object-classes"));
exports.AppObjectClasses = AppObjectClasses.AppObjectClasses;
const AppOutboundAssertionAttributes = __importStar(require("./app-outbound-assertion-attributes"));
exports.AppOutboundAssertionAttributes = AppOutboundAssertionAttributes.AppOutboundAssertionAttributes;
const AppProtectableSecondaryAudiences = __importStar(require("./app-protectable-secondary-audiences"));
exports.AppProtectableSecondaryAudiences = AppProtectableSecondaryAudiences.AppProtectableSecondaryAudiences;
const AppRadiusPolicy = __importStar(require("./app-radius-policy"));
exports.AppRadiusPolicy = AppRadiusPolicy.AppRadiusPolicy;
const AppRole = __importStar(require("./app-role"));
exports.AppRole = AppRole.AppRole;
const AppRoleApp = __importStar(require("./app-role-app"));
exports.AppRoleApp = AppRoleApp.AppRoleApp;
const AppRoleMembers = __importStar(require("./app-role-members"));
exports.AppRoleMembers = AppRoleMembers.AppRoleMembers;
const AppRoleSearchRequest = __importStar(require("./app-role-search-request"));
exports.AppRoleSearchRequest = AppRoleSearchRequest.AppRoleSearchRequest;
const AppRoles = __importStar(require("./app-roles"));
exports.AppRoles = AppRoles.AppRoles;
const AppSamlServiceProvider = __importStar(require("./app-saml-service-provider"));
exports.AppSamlServiceProvider = AppSamlServiceProvider.AppSamlServiceProvider;
const AppScopes = __importStar(require("./app-scopes"));
exports.AppScopes = AppScopes.AppScopes;
const AppSearchRequest = __importStar(require("./app-search-request"));
exports.AppSearchRequest = AppSearchRequest.AppSearchRequest;
const AppServiceParams = __importStar(require("./app-service-params"));
exports.AppServiceParams = AppServiceParams.AppServiceParams;
const AppSignonPolicy = __importStar(require("./app-signon-policy"));
exports.AppSignonPolicy = AppSignonPolicy.AppSignonPolicy;
const AppStatusChanger = __importStar(require("./app-status-changer"));
exports.AppStatusChanger = AppStatusChanger.AppStatusChanger;
const AppTermsOfUse = __importStar(require("./app-terms-of-use"));
exports.AppTermsOfUse = AppTermsOfUse.AppTermsOfUse;
const AppThreeLeggedOAuthCredential = __importStar(require("./app-three-legged-oauth-credential"));
exports.AppThreeLeggedOAuthCredential = AppThreeLeggedOAuthCredential.AppThreeLeggedOAuthCredential;
const AppTrustPolicies = __importStar(require("./app-trust-policies"));
exports.AppTrustPolicies = AppTrustPolicies.AppTrustPolicies;
const AppUserAssertionAttributes = __importStar(require("./app-user-assertion-attributes"));
exports.AppUserAssertionAttributes = AppUserAssertionAttributes.AppUserAssertionAttributes;
const AppUserRoles = __importStar(require("./app-user-roles"));
exports.AppUserRoles = AppUserRoles.AppUserRoles;
const ApprovalWorkflow = __importStar(require("./approval-workflow"));
exports.ApprovalWorkflow = ApprovalWorkflow.ApprovalWorkflow;
const ApprovalWorkflowApprovalWorkflowSteps = __importStar(require("./approval-workflow-approval-workflow-steps"));
exports.ApprovalWorkflowApprovalWorkflowSteps = ApprovalWorkflowApprovalWorkflowSteps.ApprovalWorkflowApprovalWorkflowSteps;
const ApprovalWorkflowAssignment = __importStar(require("./approval-workflow-assignment"));
exports.ApprovalWorkflowAssignment = ApprovalWorkflowAssignment.ApprovalWorkflowAssignment;
const ApprovalWorkflowAssignmentApprovalWorkflow = __importStar(require("./approval-workflow-assignment-approval-workflow"));
exports.ApprovalWorkflowAssignmentApprovalWorkflow = ApprovalWorkflowAssignmentApprovalWorkflow.ApprovalWorkflowAssignmentApprovalWorkflow;
const ApprovalWorkflowAssignmentAssignedTo = __importStar(require("./approval-workflow-assignment-assigned-to"));
exports.ApprovalWorkflowAssignmentAssignedTo = ApprovalWorkflowAssignmentAssignedTo.ApprovalWorkflowAssignmentAssignedTo;
const ApprovalWorkflowAssignments = __importStar(require("./approval-workflow-assignments"));
exports.ApprovalWorkflowAssignments = ApprovalWorkflowAssignments.ApprovalWorkflowAssignments;
const ApprovalWorkflowMaxDuration = __importStar(require("./approval-workflow-max-duration"));
exports.ApprovalWorkflowMaxDuration = ApprovalWorkflowMaxDuration.ApprovalWorkflowMaxDuration;
const ApprovalWorkflowStep = __importStar(require("./approval-workflow-step"));
exports.ApprovalWorkflowStep = ApprovalWorkflowStep.ApprovalWorkflowStep;
const ApprovalWorkflowStepApprovers = __importStar(require("./approval-workflow-step-approvers"));
exports.ApprovalWorkflowStepApprovers = ApprovalWorkflowStepApprovers.ApprovalWorkflowStepApprovers;
const ApprovalWorkflowSteps = __importStar(require("./approval-workflow-steps"));
exports.ApprovalWorkflowSteps = ApprovalWorkflowSteps.ApprovalWorkflowSteps;
const ApprovalWorkflows = __importStar(require("./approval-workflows"));
exports.ApprovalWorkflows = ApprovalWorkflows.ApprovalWorkflows;
const Apps = __importStar(require("./apps"));
exports.Apps = Apps.Apps;
const AttributeSets = __importStar(require("./attribute-sets"));
exports.AttributeSets = AttributeSets.AttributeSets;
const AuthToken = __importStar(require("./auth-token"));
exports.AuthToken = AuthToken.AuthToken;
const AuthTokenSearchRequest = __importStar(require("./auth-token-search-request"));
exports.AuthTokenSearchRequest = AuthTokenSearchRequest.AuthTokenSearchRequest;
const AuthTokenUser = __importStar(require("./auth-token-user"));
exports.AuthTokenUser = AuthTokenUser.AuthTokenUser;
const AuthTokens = __importStar(require("./auth-tokens"));
exports.AuthTokens = AuthTokens.AuthTokens;
const AuthenticationFactorSetting = __importStar(require("./authentication-factor-setting"));
exports.AuthenticationFactorSetting = AuthenticationFactorSetting.AuthenticationFactorSetting;
const AuthenticationFactorSettings = __importStar(require("./authentication-factor-settings"));
exports.AuthenticationFactorSettings = AuthenticationFactorSettings.AuthenticationFactorSettings;
const AuthenticationFactorSettingsBypassCodeSettings = __importStar(require("./authentication-factor-settings-bypass-code-settings"));
exports.AuthenticationFactorSettingsBypassCodeSettings = AuthenticationFactorSettingsBypassCodeSettings.AuthenticationFactorSettingsBypassCodeSettings;
const AuthenticationFactorSettingsClientAppSettings = __importStar(require("./authentication-factor-settings-client-app-settings"));
exports.AuthenticationFactorSettingsClientAppSettings = AuthenticationFactorSettingsClientAppSettings.AuthenticationFactorSettingsClientAppSettings;
const AuthenticationFactorSettingsCompliancePolicy = __importStar(require("./authentication-factor-settings-compliance-policy"));
exports.AuthenticationFactorSettingsCompliancePolicy = AuthenticationFactorSettingsCompliancePolicy.AuthenticationFactorSettingsCompliancePolicy;
const AuthenticationFactorSettingsDuoSecuritySettings = __importStar(require("./authentication-factor-settings-duo-security-settings"));
exports.AuthenticationFactorSettingsDuoSecuritySettings = AuthenticationFactorSettingsDuoSecuritySettings.AuthenticationFactorSettingsDuoSecuritySettings;
const AuthenticationFactorSettingsEmailSettings = __importStar(require("./authentication-factor-settings-email-settings"));
exports.AuthenticationFactorSettingsEmailSettings = AuthenticationFactorSettingsEmailSettings.AuthenticationFactorSettingsEmailSettings;
const AuthenticationFactorSettingsEndpointRestrictions = __importStar(require("./authentication-factor-settings-endpoint-restrictions"));
exports.AuthenticationFactorSettingsEndpointRestrictions = AuthenticationFactorSettingsEndpointRestrictions.AuthenticationFactorSettingsEndpointRestrictions;
const AuthenticationFactorSettingsIdentityStoreSettings = __importStar(require("./authentication-factor-settings-identity-store-settings"));
exports.AuthenticationFactorSettingsIdentityStoreSettings = AuthenticationFactorSettingsIdentityStoreSettings.AuthenticationFactorSettingsIdentityStoreSettings;
const AuthenticationFactorSettingsNotificationSettings = __importStar(require("./authentication-factor-settings-notification-settings"));
exports.AuthenticationFactorSettingsNotificationSettings = AuthenticationFactorSettingsNotificationSettings.AuthenticationFactorSettingsNotificationSettings;
const AuthenticationFactorSettingsSearchRequest = __importStar(require("./authentication-factor-settings-search-request"));
exports.AuthenticationFactorSettingsSearchRequest = AuthenticationFactorSettingsSearchRequest.AuthenticationFactorSettingsSearchRequest;
const AuthenticationFactorSettingsThirdPartyFactor = __importStar(require("./authentication-factor-settings-third-party-factor"));
exports.AuthenticationFactorSettingsThirdPartyFactor = AuthenticationFactorSettingsThirdPartyFactor.AuthenticationFactorSettingsThirdPartyFactor;
const AuthenticationFactorSettingsTotpSettings = __importStar(require("./authentication-factor-settings-totp-settings"));
exports.AuthenticationFactorSettingsTotpSettings = AuthenticationFactorSettingsTotpSettings.AuthenticationFactorSettingsTotpSettings;
const AuthenticationFactorsRemover = __importStar(require("./authentication-factors-remover"));
exports.AuthenticationFactorsRemover = AuthenticationFactorsRemover.AuthenticationFactorsRemover;
const AuthenticationFactorsRemoverUser = __importStar(require("./authentication-factors-remover-user"));
exports.AuthenticationFactorsRemoverUser = AuthenticationFactorsRemoverUser.AuthenticationFactorsRemoverUser;
const BrandingSetting = __importStar(require("./branding-setting"));
exports.BrandingSetting = BrandingSetting.BrandingSetting;
const BrandingSettings = __importStar(require("./branding-settings"));
exports.BrandingSettings = BrandingSettings.BrandingSettings;
const BrandingSettingsCompanyNames = __importStar(require("./branding-settings-company-names"));
exports.BrandingSettingsCompanyNames = BrandingSettingsCompanyNames.BrandingSettingsCompanyNames;
const BrandingSettingsDefaultCompanyNames = __importStar(require("./branding-settings-default-company-names"));
exports.BrandingSettingsDefaultCompanyNames = BrandingSettingsDefaultCompanyNames.BrandingSettingsDefaultCompanyNames;
const BrandingSettingsDefaultImages = __importStar(require("./branding-settings-default-images"));
exports.BrandingSettingsDefaultImages = BrandingSettingsDefaultImages.BrandingSettingsDefaultImages;
const BrandingSettingsDefaultLoginTexts = __importStar(require("./branding-settings-default-login-texts"));
exports.BrandingSettingsDefaultLoginTexts = BrandingSettingsDefaultLoginTexts.BrandingSettingsDefaultLoginTexts;
const BrandingSettingsImages = __importStar(require("./branding-settings-images"));
exports.BrandingSettingsImages = BrandingSettingsImages.BrandingSettingsImages;
const BrandingSettingsLoginTexts = __importStar(require("./branding-settings-login-texts"));
exports.BrandingSettingsLoginTexts = BrandingSettingsLoginTexts.BrandingSettingsLoginTexts;
const CloudGate = __importStar(require("./cloud-gate"));
exports.CloudGate = CloudGate.CloudGate;
const CloudGateExtMappings = __importStar(require("./cloud-gate-ext-mappings"));
exports.CloudGateExtMappings = CloudGateExtMappings.CloudGateExtMappings;
const CloudGateExtServers = __importStar(require("./cloud-gate-ext-servers"));
exports.CloudGateExtServers = CloudGateExtServers.CloudGateExtServers;
const CloudGateExtUpstreamServerGroups = __importStar(require("./cloud-gate-ext-upstream-server-groups"));
exports.CloudGateExtUpstreamServerGroups = CloudGateExtUpstreamServerGroups.CloudGateExtUpstreamServerGroups;
const CloudGateExtUpstreamServers = __importStar(require("./cloud-gate-ext-upstream-servers"));
exports.CloudGateExtUpstreamServers = CloudGateExtUpstreamServers.CloudGateExtUpstreamServers;
const CloudGateMapping = __importStar(require("./cloud-gate-mapping"));
exports.CloudGateMapping = CloudGateMapping.CloudGateMapping;
const CloudGateMappingCloudGate = __importStar(require("./cloud-gate-mapping-cloud-gate"));
exports.CloudGateMappingCloudGate = CloudGateMappingCloudGate.CloudGateMappingCloudGate;
const CloudGateMappingGatewayApp = __importStar(require("./cloud-gate-mapping-gateway-app"));
exports.CloudGateMappingGatewayApp = CloudGateMappingGatewayApp.CloudGateMappingGatewayApp;
const CloudGateMappingSearchRequest = __importStar(require("./cloud-gate-mapping-search-request"));
exports.CloudGateMappingSearchRequest = CloudGateMappingSearchRequest.CloudGateMappingSearchRequest;
const CloudGateMappingServer = __importStar(require("./cloud-gate-mapping-server"));
exports.CloudGateMappingServer = CloudGateMappingServer.CloudGateMappingServer;
const CloudGateMappingUpstreamServerGroup = __importStar(require("./cloud-gate-mapping-upstream-server-group"));
exports.CloudGateMappingUpstreamServerGroup = CloudGateMappingUpstreamServerGroup.CloudGateMappingUpstreamServerGroup;
const CloudGateMappings = __importStar(require("./cloud-gate-mappings"));
exports.CloudGateMappings = CloudGateMappings.CloudGateMappings;
const CloudGateOauthClient = __importStar(require("./cloud-gate-oauth-client"));
exports.CloudGateOauthClient = CloudGateOauthClient.CloudGateOauthClient;
const CloudGateSearchRequest = __importStar(require("./cloud-gate-search-request"));
exports.CloudGateSearchRequest = CloudGateSearchRequest.CloudGateSearchRequest;
const CloudGateServer = __importStar(require("./cloud-gate-server"));
exports.CloudGateServer = CloudGateServer.CloudGateServer;
const CloudGateServerCloudGate = __importStar(require("./cloud-gate-server-cloud-gate"));
exports.CloudGateServerCloudGate = CloudGateServerCloudGate.CloudGateServerCloudGate;
const CloudGateServerSearchRequest = __importStar(require("./cloud-gate-server-search-request"));
exports.CloudGateServerSearchRequest = CloudGateServerSearchRequest.CloudGateServerSearchRequest;
const CloudGateServers = __importStar(require("./cloud-gate-servers"));
exports.CloudGateServers = CloudGateServers.CloudGateServers;
const CloudGates = __importStar(require("./cloud-gates"));
exports.CloudGates = CloudGates.CloudGates;
const Condition = __importStar(require("./condition"));
exports.Condition = Condition.Condition;
const ConditionExtensionOciconsolesignonpolicyconsentPolicy = __importStar(require("./condition-extension-ociconsolesignonpolicyconsent-policy"));
exports.ConditionExtensionOciconsolesignonpolicyconsentPolicy = ConditionExtensionOciconsolesignonpolicyconsentPolicy.ConditionExtensionOciconsolesignonpolicyconsentPolicy;
const ConditionSearchRequest = __importStar(require("./condition-search-request"));
exports.ConditionSearchRequest = ConditionSearchRequest.ConditionSearchRequest;
const Conditions = __importStar(require("./conditions"));
exports.Conditions = Conditions.Conditions;
const CustomerSecretKey = __importStar(require("./customer-secret-key"));
exports.CustomerSecretKey = CustomerSecretKey.CustomerSecretKey;
const CustomerSecretKeySearchRequest = __importStar(require("./customer-secret-key-search-request"));
exports.CustomerSecretKeySearchRequest = CustomerSecretKeySearchRequest.CustomerSecretKeySearchRequest;
const CustomerSecretKeyUser = __importStar(require("./customer-secret-key-user"));
exports.CustomerSecretKeyUser = CustomerSecretKeyUser.CustomerSecretKeyUser;
const CustomerSecretKeys = __importStar(require("./customer-secret-keys"));
exports.CustomerSecretKeys = CustomerSecretKeys.CustomerSecretKeys;
const DefinedTags = __importStar(require("./defined-tags"));
exports.DefinedTags = DefinedTags.DefinedTags;
const DynamicResourceGroup = __importStar(require("./dynamic-resource-group"));
exports.DynamicResourceGroup = DynamicResourceGroup.DynamicResourceGroup;
const DynamicResourceGroupDynamicGroupAppRoles = __importStar(require("./dynamic-resource-group-dynamic-group-app-roles"));
exports.DynamicResourceGroupDynamicGroupAppRoles = DynamicResourceGroupDynamicGroupAppRoles.DynamicResourceGroupDynamicGroupAppRoles;
const DynamicResourceGroupGrants = __importStar(require("./dynamic-resource-group-grants"));
exports.DynamicResourceGroupGrants = DynamicResourceGroupGrants.DynamicResourceGroupGrants;
const DynamicResourceGroupSearchRequest = __importStar(require("./dynamic-resource-group-search-request"));
exports.DynamicResourceGroupSearchRequest = DynamicResourceGroupSearchRequest.DynamicResourceGroupSearchRequest;
const DynamicResourceGroups = __importStar(require("./dynamic-resource-groups"));
exports.DynamicResourceGroups = DynamicResourceGroups.DynamicResourceGroups;
const ExtensionAdaptiveUser = __importStar(require("./extension-adaptive-user"));
exports.ExtensionAdaptiveUser = ExtensionAdaptiveUser.ExtensionAdaptiveUser;
const ExtensionCapabilitiesUser = __importStar(require("./extension-capabilities-user"));
exports.ExtensionCapabilitiesUser = ExtensionCapabilitiesUser.ExtensionCapabilitiesUser;
const ExtensionDbCredentialsUser = __importStar(require("./extension-db-credentials-user"));
exports.ExtensionDbCredentialsUser = ExtensionDbCredentialsUser.ExtensionDbCredentialsUser;
const ExtensionDbUserUser = __importStar(require("./extension-db-user-user"));
exports.ExtensionDbUserUser = ExtensionDbUserUser.ExtensionDbUserUser;
const ExtensionDbcsGroup = __importStar(require("./extension-dbcs-group"));
exports.ExtensionDbcsGroup = ExtensionDbcsGroup.ExtensionDbcsGroup;
const ExtensionDynamicGroup = __importStar(require("./extension-dynamic-group"));
exports.ExtensionDynamicGroup = ExtensionDynamicGroup.ExtensionDynamicGroup;
const ExtensionEnterprise20User = __importStar(require("./extension-enterprise20-user"));
exports.ExtensionEnterprise20User = ExtensionEnterprise20User.ExtensionEnterprise20User;
const ExtensionFidoAuthenticationFactorSettings = __importStar(require("./extension-fido-authentication-factor-settings"));
exports.ExtensionFidoAuthenticationFactorSettings = ExtensionFidoAuthenticationFactorSettings.ExtensionFidoAuthenticationFactorSettings;
const ExtensionGroupGroup = __importStar(require("./extension-group-group"));
exports.ExtensionGroupGroup = ExtensionGroupGroup.ExtensionGroupGroup;
const ExtensionKerberosUserUser = __importStar(require("./extension-kerberos-user-user"));
exports.ExtensionKerberosUserUser = ExtensionKerberosUserUser.ExtensionKerberosUserUser;
const ExtensionMeUser = __importStar(require("./extension-me-user"));
exports.ExtensionMeUser = ExtensionMeUser.ExtensionMeUser;
const ExtensionMessagesError = __importStar(require("./extension-messages-error"));
exports.ExtensionMessagesError = ExtensionMessagesError.ExtensionMessagesError;
const ExtensionMfaUser = __importStar(require("./extension-mfa-user"));
exports.ExtensionMfaUser = ExtensionMfaUser.ExtensionMfaUser;
const ExtensionOCITags = __importStar(require("./extension-oc-itags"));
exports.ExtensionOCITags = ExtensionOCITags.ExtensionOCITags;
const ExtensionPasswordStateUser = __importStar(require("./extension-password-state-user"));
exports.ExtensionPasswordStateUser = ExtensionPasswordStateUser.ExtensionPasswordStateUser;
const ExtensionPasswordlessUser = __importStar(require("./extension-passwordless-user"));
exports.ExtensionPasswordlessUser = ExtensionPasswordlessUser.ExtensionPasswordlessUser;
const ExtensionPosixGroup = __importStar(require("./extension-posix-group"));
exports.ExtensionPosixGroup = ExtensionPosixGroup.ExtensionPosixGroup;
const ExtensionPosixUser = __importStar(require("./extension-posix-user"));
exports.ExtensionPosixUser = ExtensionPosixUser.ExtensionPosixUser;
const ExtensionRequestableGroup = __importStar(require("./extension-requestable-group"));
exports.ExtensionRequestableGroup = ExtensionRequestableGroup.ExtensionRequestableGroup;
const ExtensionSecurityQuestionsUser = __importStar(require("./extension-security-questions-user"));
exports.ExtensionSecurityQuestionsUser = ExtensionSecurityQuestionsUser.ExtensionSecurityQuestionsUser;
const ExtensionSelfChangeUser = __importStar(require("./extension-self-change-user"));
exports.ExtensionSelfChangeUser = ExtensionSelfChangeUser.ExtensionSelfChangeUser;
const ExtensionSelfRegistrationUser = __importStar(require("./extension-self-registration-user"));
exports.ExtensionSelfRegistrationUser = ExtensionSelfRegistrationUser.ExtensionSelfRegistrationUser;
const ExtensionSffUser = __importStar(require("./extension-sff-user"));
exports.ExtensionSffUser = ExtensionSffUser.ExtensionSffUser;
const ExtensionSocialAccountUser = __importStar(require("./extension-social-account-user"));
exports.ExtensionSocialAccountUser = ExtensionSocialAccountUser.ExtensionSocialAccountUser;
const ExtensionSocialIdentityProvider = __importStar(require("./extension-social-identity-provider"));
exports.ExtensionSocialIdentityProvider = ExtensionSocialIdentityProvider.ExtensionSocialIdentityProvider;
const ExtensionTermsOfUseUser = __importStar(require("./extension-terms-of-use-user"));
exports.ExtensionTermsOfUseUser = ExtensionTermsOfUseUser.ExtensionTermsOfUseUser;
const ExtensionThirdPartyAuthenticationFactorSettings = __importStar(require("./extension-third-party-authentication-factor-settings"));
exports.ExtensionThirdPartyAuthenticationFactorSettings = ExtensionThirdPartyAuthenticationFactorSettings.ExtensionThirdPartyAuthenticationFactorSettings;
const ExtensionUserCredentialsUser = __importStar(require("./extension-user-credentials-user"));
exports.ExtensionUserCredentialsUser = ExtensionUserCredentialsUser.ExtensionUserCredentialsUser;
const ExtensionUserStateUser = __importStar(require("./extension-user-state-user"));
exports.ExtensionUserStateUser = ExtensionUserStateUser.ExtensionUserStateUser;
const ExtensionUserUser = __importStar(require("./extension-user-user"));
exports.ExtensionUserUser = ExtensionUserUser.ExtensionUserUser;
const ExtensionX509IdentityProvider = __importStar(require("./extension-x509-identity-provider"));
exports.ExtensionX509IdentityProvider = ExtensionX509IdentityProvider.ExtensionX509IdentityProvider;
const FreeformTags = __importStar(require("./freeform-tags"));
exports.FreeformTags = FreeformTags.FreeformTags;
const Grant = __importStar(require("./grant"));
exports.Grant = Grant.Grant;
const GrantApp = __importStar(require("./grant-app"));
exports.GrantApp = GrantApp.GrantApp;
const GrantAppEntitlementCollection = __importStar(require("./grant-app-entitlement-collection"));
exports.GrantAppEntitlementCollection = GrantAppEntitlementCollection.GrantAppEntitlementCollection;
const GrantEntitlement = __importStar(require("./grant-entitlement"));
exports.GrantEntitlement = GrantEntitlement.GrantEntitlement;
const GrantGrantee = __importStar(require("./grant-grantee"));
exports.GrantGrantee = GrantGrantee.GrantGrantee;
const GrantGrantor = __importStar(require("./grant-grantor"));
exports.GrantGrantor = GrantGrantor.GrantGrantor;
const GrantSearchRequest = __importStar(require("./grant-search-request"));
exports.GrantSearchRequest = GrantSearchRequest.GrantSearchRequest;
const Grants = __importStar(require("./grants"));
exports.Grants = Grants.Grants;
const Group = __importStar(require("./group"));
exports.Group = Group.Group;
const GroupExtAppRoles = __importStar(require("./group-ext-app-roles"));
exports.GroupExtAppRoles = GroupExtAppRoles.GroupExtAppRoles;
const GroupExtDomainLevelSchemaNames = __importStar(require("./group-ext-domain-level-schema-names"));
exports.GroupExtDomainLevelSchemaNames = GroupExtDomainLevelSchemaNames.GroupExtDomainLevelSchemaNames;
const GroupExtGrants = __importStar(require("./group-ext-grants"));
exports.GroupExtGrants = GroupExtGrants.GroupExtGrants;
const GroupExtInstanceLevelSchemaNames = __importStar(require("./group-ext-instance-level-schema-names"));
exports.GroupExtInstanceLevelSchemaNames = GroupExtInstanceLevelSchemaNames.GroupExtInstanceLevelSchemaNames;
const GroupExtOwners = __importStar(require("./group-ext-owners"));
exports.GroupExtOwners = GroupExtOwners.GroupExtOwners;
const GroupExtPasswordPolicy = __importStar(require("./group-ext-password-policy"));
exports.GroupExtPasswordPolicy = GroupExtPasswordPolicy.GroupExtPasswordPolicy;
const GroupExtSyncedFromApp = __importStar(require("./group-ext-synced-from-app"));
exports.GroupExtSyncedFromApp = GroupExtSyncedFromApp.GroupExtSyncedFromApp;
const GroupMembers = __importStar(require("./group-members"));
exports.GroupMembers = GroupMembers.GroupMembers;
const GroupSearchRequest = __importStar(require("./group-search-request"));
exports.GroupSearchRequest = GroupSearchRequest.GroupSearchRequest;
const Groups = __importStar(require("./groups"));
exports.Groups = Groups.Groups;
const IdcsCreatedBy = __importStar(require("./idcs-created-by"));
exports.IdcsCreatedBy = IdcsCreatedBy.IdcsCreatedBy;
const IdcsLastModifiedBy = __importStar(require("./idcs-last-modified-by"));
exports.IdcsLastModifiedBy = IdcsLastModifiedBy.IdcsLastModifiedBy;
const IdcsPreventedOperations = __importStar(require("./idcs-prevented-operations"));
exports.IdcsPreventedOperations = IdcsPreventedOperations.IdcsPreventedOperations;
const IdentityPropagationTrust = __importStar(require("./identity-propagation-trust"));
exports.IdentityPropagationTrust = IdentityPropagationTrust.IdentityPropagationTrust;
const IdentityPropagationTrustImpersonationServiceUsers = __importStar(require("./identity-propagation-trust-impersonation-service-users"));
exports.IdentityPropagationTrustImpersonationServiceUsers = IdentityPropagationTrustImpersonationServiceUsers.IdentityPropagationTrustImpersonationServiceUsers;
const IdentityPropagationTrustKeytab = __importStar(require("./identity-propagation-trust-keytab"));
exports.IdentityPropagationTrustKeytab = IdentityPropagationTrustKeytab.IdentityPropagationTrustKeytab;
const IdentityPropagationTrusts = __importStar(require("./identity-propagation-trusts"));
exports.IdentityPropagationTrusts = IdentityPropagationTrusts.IdentityPropagationTrusts;
const IdentityProvider = __importStar(require("./identity-provider"));
exports.IdentityProvider = IdentityProvider.IdentityProvider;
const IdentityProviderCorrelationPolicy = __importStar(require("./identity-provider-correlation-policy"));
exports.IdentityProviderCorrelationPolicy = IdentityProviderCorrelationPolicy.IdentityProviderCorrelationPolicy;
const IdentityProviderJitProvAssignedGroups = __importStar(require("./identity-provider-jit-prov-assigned-groups"));
exports.IdentityProviderJitProvAssignedGroups = IdentityProviderJitProvAssignedGroups.IdentityProviderJitProvAssignedGroups;
const IdentityProviderJitUserProvAssignedGroups = __importStar(require("./identity-provider-jit-user-prov-assigned-groups"));
exports.IdentityProviderJitUserProvAssignedGroups = IdentityProviderJitUserProvAssignedGroups.IdentityProviderJitUserProvAssignedGroups;
const IdentityProviderJitUserProvAttributes = __importStar(require("./identity-provider-jit-user-prov-attributes"));
exports.IdentityProviderJitUserProvAttributes = IdentityProviderJitUserProvAttributes.IdentityProviderJitUserProvAttributes;
const IdentityProviderJitUserProvGroupMappings = __importStar(require("./identity-provider-jit-user-prov-group-mappings"));
exports.IdentityProviderJitUserProvGroupMappings = IdentityProviderJitUserProvGroupMappings.IdentityProviderJitUserProvGroupMappings;
const IdentityProviderRelayIdpParamMappings = __importStar(require("./identity-provider-relay-idp-param-mappings"));
exports.IdentityProviderRelayIdpParamMappings = IdentityProviderRelayIdpParamMappings.IdentityProviderRelayIdpParamMappings;
const IdentityProviderSearchRequest = __importStar(require("./identity-provider-search-request"));
exports.IdentityProviderSearchRequest = IdentityProviderSearchRequest.IdentityProviderSearchRequest;
const IdentityProviders = __importStar(require("./identity-providers"));
exports.IdentityProviders = IdentityProviders.IdentityProviders;
const IdentitySetting = __importStar(require("./identity-setting"));
exports.IdentitySetting = IdentitySetting.IdentitySetting;
const IdentitySettings = __importStar(require("./identity-settings"));
exports.IdentitySettings = IdentitySettings.IdentitySettings;
const IdentitySettingsMyProfile = __importStar(require("./identity-settings-my-profile"));
exports.IdentitySettingsMyProfile = IdentitySettingsMyProfile.IdentitySettingsMyProfile;
const IdentitySettingsPOSIXGid = __importStar(require("./identity-settings-po-si-xgid"));
exports.IdentitySettingsPOSIXGid = IdentitySettingsPOSIXGid.IdentitySettingsPOSIXGid;
const IdentitySettingsPOSIXUid = __importStar(require("./identity-settings-po-si-xuid"));
exports.IdentitySettingsPOSIXUid = IdentitySettingsPOSIXUid.IdentitySettingsPOSIXUid;
const IdentitySettingsSearchRequest = __importStar(require("./identity-settings-search-request"));
exports.IdentitySettingsSearchRequest = IdentitySettingsSearchRequest.IdentitySettingsSearchRequest;
const IdentitySettingsTokens = __importStar(require("./identity-settings-tokens"));
exports.IdentitySettingsTokens = IdentitySettingsTokens.IdentitySettingsTokens;
const KmsiSetting = __importStar(require("./kmsi-setting"));
exports.KmsiSetting = KmsiSetting.KmsiSetting;
const KmsiSettings = __importStar(require("./kmsi-settings"));
exports.KmsiSettings = KmsiSettings.KmsiSettings;
const KmsiSettingsSearchRequest = __importStar(require("./kmsi-settings-search-request"));
exports.KmsiSettingsSearchRequest = KmsiSettingsSearchRequest.KmsiSettingsSearchRequest;
const Me = __importStar(require("./me"));
exports.Me = Me.Me;
const MeEmails = __importStar(require("./me-emails"));
exports.MeEmails = MeEmails.MeEmails;
const MeEntitlements = __importStar(require("./me-entitlements"));
exports.MeEntitlements = MeEntitlements.MeEntitlements;
const MeGroups = __importStar(require("./me-groups"));
exports.MeGroups = MeGroups.MeGroups;
const MeIms = __importStar(require("./me-ims"));
exports.MeIms = MeIms.MeIms;
const MeName = __importStar(require("./me-name"));
exports.MeName = MeName.MeName;
const MePasswordChanger = __importStar(require("./me-password-changer"));
exports.MePasswordChanger = MePasswordChanger.MePasswordChanger;
const MePhoneNumbers = __importStar(require("./me-phone-numbers"));
exports.MePhoneNumbers = MePhoneNumbers.MePhoneNumbers;
const MePhotos = __importStar(require("./me-photos"));
exports.MePhotos = MePhotos.MePhotos;
const MeRoles = __importStar(require("./me-roles"));
exports.MeRoles = MeRoles.MeRoles;
const MeX509Certificates = __importStar(require("./me-x509-certificates"));
exports.MeX509Certificates = MeX509Certificates.MeX509Certificates;
const Meta = __importStar(require("./meta"));
exports.Meta = Meta.Meta;
const MyApiKey = __importStar(require("./my-api-key"));
exports.MyApiKey = MyApiKey.MyApiKey;
const MyApiKeyUser = __importStar(require("./my-api-key-user"));
exports.MyApiKeyUser = MyApiKeyUser.MyApiKeyUser;
const MyApiKeys = __importStar(require("./my-api-keys"));
exports.MyApiKeys = MyApiKeys.MyApiKeys;
const MyApp = __importStar(require("./my-app"));
exports.MyApp = MyApp.MyApp;
const MyAppApp = __importStar(require("./my-app-app"));
exports.MyAppApp = MyAppApp.MyAppApp;
const MyAppOwner = __importStar(require("./my-app-owner"));
exports.MyAppOwner = MyAppOwner.MyAppOwner;
const MyAppSearchRequest = __importStar(require("./my-app-search-request"));
exports.MyAppSearchRequest = MyAppSearchRequest.MyAppSearchRequest;
const MyAppUserWalletArtifact = __importStar(require("./my-app-user-wallet-artifact"));
exports.MyAppUserWalletArtifact = MyAppUserWalletArtifact.MyAppUserWalletArtifact;
const MyApps = __importStar(require("./my-apps"));
exports.MyApps = MyApps.MyApps;
const MyAuthToken = __importStar(require("./my-auth-token"));
exports.MyAuthToken = MyAuthToken.MyAuthToken;
const MyAuthTokenUser = __importStar(require("./my-auth-token-user"));
exports.MyAuthTokenUser = MyAuthTokenUser.MyAuthTokenUser;
const MyAuthTokens = __importStar(require("./my-auth-tokens"));
exports.MyAuthTokens = MyAuthTokens.MyAuthTokens;
const MyAuthenticationFactorInitiator = __importStar(require("./my-authentication-factor-initiator"));
exports.MyAuthenticationFactorInitiator = MyAuthenticationFactorInitiator.MyAuthenticationFactorInitiator;
const MyAuthenticationFactorInitiatorAdditionalAttributes = __importStar(require("./my-authentication-factor-initiator-additional-attributes"));
exports.MyAuthenticationFactorInitiatorAdditionalAttributes = MyAuthenticationFactorInitiatorAdditionalAttributes.MyAuthenticationFactorInitiatorAdditionalAttributes;
const MyAuthenticationFactorInitiatorThirdPartyFactor = __importStar(require("./my-authentication-factor-initiator-third-party-factor"));
exports.MyAuthenticationFactorInitiatorThirdPartyFactor = MyAuthenticationFactorInitiatorThirdPartyFactor.MyAuthenticationFactorInitiatorThirdPartyFactor;
const MyAuthenticationFactorValidator = __importStar(require("./my-authentication-factor-validator"));
exports.MyAuthenticationFactorValidator = MyAuthenticationFactorValidator.MyAuthenticationFactorValidator;
const MyAuthenticationFactorValidatorAdditionalAttributes = __importStar(require("./my-authentication-factor-validator-additional-attributes"));
exports.MyAuthenticationFactorValidatorAdditionalAttributes = MyAuthenticationFactorValidatorAdditionalAttributes.MyAuthenticationFactorValidatorAdditionalAttributes;
const MyAuthenticationFactorValidatorSecurityQuestions = __importStar(require("./my-authentication-factor-validator-security-questions"));
exports.MyAuthenticationFactorValidatorSecurityQuestions = MyAuthenticationFactorValidatorSecurityQuestions.MyAuthenticationFactorValidatorSecurityQuestions;
const MyAuthenticationFactorValidatorThirdPartyFactor = __importStar(require("./my-authentication-factor-validator-third-party-factor"));
exports.MyAuthenticationFactorValidatorThirdPartyFactor = MyAuthenticationFactorValidatorThirdPartyFactor.MyAuthenticationFactorValidatorThirdPartyFactor;
const MyAuthenticationFactorsRemover = __importStar(require("./my-authentication-factors-remover"));
exports.MyAuthenticationFactorsRemover = MyAuthenticationFactorsRemover.MyAuthenticationFactorsRemover;
const MyAuthenticationFactorsRemoverUser = __importStar(require("./my-authentication-factors-remover-user"));
exports.MyAuthenticationFactorsRemoverUser = MyAuthenticationFactorsRemoverUser.MyAuthenticationFactorsRemoverUser;
const MyCompletedApproval = __importStar(require("./my-completed-approval"));
exports.MyCompletedApproval = MyCompletedApproval.MyCompletedApproval;
const MyCompletedApprovals = __importStar(require("./my-completed-approvals"));
exports.MyCompletedApprovals = MyCompletedApprovals.MyCompletedApprovals;
const MyCustomerSecretKey = __importStar(require("./my-customer-secret-key"));
exports.MyCustomerSecretKey = MyCustomerSecretKey.MyCustomerSecretKey;
const MyCustomerSecretKeyUser = __importStar(require("./my-customer-secret-key-user"));
exports.MyCustomerSecretKeyUser = MyCustomerSecretKeyUser.MyCustomerSecretKeyUser;
const MyCustomerSecretKeys = __importStar(require("./my-customer-secret-keys"));
exports.MyCustomerSecretKeys = MyCustomerSecretKeys.MyCustomerSecretKeys;
const MyDevice = __importStar(require("./my-device"));
exports.MyDevice = MyDevice.MyDevice;
const MyDeviceAdditionalAttributes = __importStar(require("./my-device-additional-attributes"));
exports.MyDeviceAdditionalAttributes = MyDeviceAdditionalAttributes.MyDeviceAdditionalAttributes;
const MyDeviceAuthenticationFactors = __importStar(require("./my-device-authentication-factors"));
exports.MyDeviceAuthenticationFactors = MyDeviceAuthenticationFactors.MyDeviceAuthenticationFactors;
const MyDeviceNonCompliances = __importStar(require("./my-device-non-compliances"));
exports.MyDeviceNonCompliances = MyDeviceNonCompliances.MyDeviceNonCompliances;
const MyDevicePushNotificationTarget = __importStar(require("./my-device-push-notification-target"));
exports.MyDevicePushNotificationTarget = MyDevicePushNotificationTarget.MyDevicePushNotificationTarget;
const MyDeviceThirdPartyFactor = __importStar(require("./my-device-third-party-factor"));
exports.MyDeviceThirdPartyFactor = MyDeviceThirdPartyFactor.MyDeviceThirdPartyFactor;
const MyDeviceUser = __importStar(require("./my-device-user"));
exports.MyDeviceUser = MyDeviceUser.MyDeviceUser;
const MyDevices = __importStar(require("./my-devices"));
exports.MyDevices = MyDevices.MyDevices;
const MyGroup = __importStar(require("./my-group"));
exports.MyGroup = MyGroup.MyGroup;
const MyGroupMembers = __importStar(require("./my-group-members"));
exports.MyGroupMembers = MyGroupMembers.MyGroupMembers;
const MyGroupSearchRequest = __importStar(require("./my-group-search-request"));
exports.MyGroupSearchRequest = MyGroupSearchRequest.MyGroupSearchRequest;
const MyGroups = __importStar(require("./my-groups"));
exports.MyGroups = MyGroups.MyGroups;
const MyOAuth2ClientCredential = __importStar(require("./my-oauth2-client-credential"));
exports.MyOAuth2ClientCredential = MyOAuth2ClientCredential.MyOAuth2ClientCredential;
const MyOAuth2ClientCredentialScopes = __importStar(require("./my-oauth2-client-credential-scopes"));
exports.MyOAuth2ClientCredentialScopes = MyOAuth2ClientCredentialScopes.MyOAuth2ClientCredentialScopes;
const MyOAuth2ClientCredentialUser = __importStar(require("./my-oauth2-client-credential-user"));
exports.MyOAuth2ClientCredentialUser = MyOAuth2ClientCredentialUser.MyOAuth2ClientCredentialUser;
const MyOAuth2ClientCredentials = __importStar(require("./my-oauth2-client-credentials"));
exports.MyOAuth2ClientCredentials = MyOAuth2ClientCredentials.MyOAuth2ClientCredentials;
const MyPendingApproval = __importStar(require("./my-pending-approval"));
exports.MyPendingApproval = MyPendingApproval.MyPendingApproval;
const MyPendingApprovals = __importStar(require("./my-pending-approvals"));
exports.MyPendingApprovals = MyPendingApprovals.MyPendingApprovals;
const MyRequest = __importStar(require("./my-request"));
exports.MyRequest = MyRequest.MyRequest;
const MyRequestApprovalDetails = __importStar(require("./my-request-approval-details"));
exports.MyRequestApprovalDetails = MyRequestApprovalDetails.MyRequestApprovalDetails;
const MyRequestRequesting = __importStar(require("./my-request-requesting"));
exports.MyRequestRequesting = MyRequestRequesting.MyRequestRequesting;
const MyRequestRequestor = __importStar(require("./my-request-requestor"));
exports.MyRequestRequestor = MyRequestRequestor.MyRequestRequestor;
const MyRequestSearchRequest = __importStar(require("./my-request-search-request"));
exports.MyRequestSearchRequest = MyRequestSearchRequest.MyRequestSearchRequest;
const MyRequestableGroup = __importStar(require("./my-requestable-group"));
exports.MyRequestableGroup = MyRequestableGroup.MyRequestableGroup;
const MyRequestableGroupMembers = __importStar(require("./my-requestable-group-members"));
exports.MyRequestableGroupMembers = MyRequestableGroupMembers.MyRequestableGroupMembers;
const MyRequestableGroupSearchRequest = __importStar(require("./my-requestable-group-search-request"));
exports.MyRequestableGroupSearchRequest = MyRequestableGroupSearchRequest.MyRequestableGroupSearchRequest;
const MyRequestableGroups = __importStar(require("./my-requestable-groups"));
exports.MyRequestableGroups = MyRequestableGroups.MyRequestableGroups;
const MyRequests = __importStar(require("./my-requests"));
exports.MyRequests = MyRequests.MyRequests;
const MySmtpCredential = __importStar(require("./my-smtp-credential"));
exports.MySmtpCredential = MySmtpCredential.MySmtpCredential;
const MySmtpCredentialUser = __importStar(require("./my-smtp-credential-user"));
exports.MySmtpCredentialUser = MySmtpCredentialUser.MySmtpCredentialUser;
const MySmtpCredentials = __importStar(require("./my-smtp-credentials"));
exports.MySmtpCredentials = MySmtpCredentials.MySmtpCredentials;
const MySupportAccount = __importStar(require("./my-support-account"));
exports.MySupportAccount = MySupportAccount.MySupportAccount;
const MySupportAccountUser = __importStar(require("./my-support-account-user"));
exports.MySupportAccountUser = MySupportAccountUser.MySupportAccountUser;
const MySupportAccounts = __importStar(require("./my-support-accounts"));
exports.MySupportAccounts = MySupportAccounts.MySupportAccounts;
const MyTrustedUserAgent = __importStar(require("./my-trusted-user-agent"));
exports.MyTrustedUserAgent = MyTrustedUserAgent.MyTrustedUserAgent;
const MyTrustedUserAgentTrustedFactors = __importStar(require("./my-trusted-user-agent-trusted-factors"));
exports.MyTrustedUserAgentTrustedFactors = MyTrustedUserAgentTrustedFactors.MyTrustedUserAgentTrustedFactors;
const MyTrustedUserAgentUser = __importStar(require("./my-trusted-user-agent-user"));
exports.MyTrustedUserAgentUser = MyTrustedUserAgentUser.MyTrustedUserAgentUser;
const MyTrustedUserAgents = __importStar(require("./my-trusted-user-agents"));
exports.MyTrustedUserAgents = MyTrustedUserAgents.MyTrustedUserAgents;
const MyUserDbCredential = __importStar(require("./my-user-db-credential"));
exports.MyUserDbCredential = MyUserDbCredential.MyUserDbCredential;
const MyUserDbCredentials = __importStar(require("./my-user-db-credentials"));
exports.MyUserDbCredentials = MyUserDbCredentials.MyUserDbCredentials;
const MyUserDbCredentialsUser = __importStar(require("./my-user-db-credentials-user"));
exports.MyUserDbCredentialsUser = MyUserDbCredentialsUser.MyUserDbCredentialsUser;
const NetworkPerimeter = __importStar(require("./network-perimeter"));
exports.NetworkPerimeter = NetworkPerimeter.NetworkPerimeter;
const NetworkPerimeterIpAddresses = __importStar(require("./network-perimeter-ip-addresses"));
exports.NetworkPerimeterIpAddresses = NetworkPerimeterIpAddresses.NetworkPerimeterIpAddresses;
const NetworkPerimeterSearchRequest = __importStar(require("./network-perimeter-search-request"));
exports.NetworkPerimeterSearchRequest = NetworkPerimeterSearchRequest.NetworkPerimeterSearchRequest;
const NetworkPerimeters = __importStar(require("./network-perimeters"));
exports.NetworkPerimeters = NetworkPerimeters.NetworkPerimeters;
const NotificationSetting = __importStar(require("./notification-setting"));
exports.NotificationSetting = NotificationSetting.NotificationSetting;
const NotificationSettings = __importStar(require("./notification-settings"));
exports.NotificationSettings = NotificationSettings.NotificationSettings;
const NotificationSettingsEventSettings = __importStar(require("./notification-settings-event-settings"));
exports.NotificationSettingsEventSettings = NotificationSettingsEventSettings.NotificationSettingsEventSettings;
const NotificationSettingsFromEmailAddress = __importStar(require("./notification-settings-from-email-address"));
exports.NotificationSettingsFromEmailAddress = NotificationSettingsFromEmailAddress.NotificationSettingsFromEmailAddress;
const NotificationSettingsSearchRequest = __importStar(require("./notification-settings-search-request"));
exports.NotificationSettingsSearchRequest = NotificationSettingsSearchRequest.NotificationSettingsSearchRequest;
const OAuth2ClientCredential = __importStar(require("./o-auth2-client-credential"));
exports.OAuth2ClientCredential = OAuth2ClientCredential.OAuth2ClientCredential;
const OAuth2ClientCredentialScopes = __importStar(require("./o-auth2-client-credential-scopes"));
exports.OAuth2ClientCredentialScopes = OAuth2ClientCredentialScopes.OAuth2ClientCredentialScopes;
const OAuth2ClientCredentialSearchRequest = __importStar(require("./o-auth2-client-credential-search-request"));
exports.OAuth2ClientCredentialSearchRequest = OAuth2ClientCredentialSearchRequest.OAuth2ClientCredentialSearchRequest;
const OAuth2ClientCredentialUser = __importStar(require("./o-auth2-client-credential-user"));
exports.OAuth2ClientCredentialUser = OAuth2ClientCredentialUser.OAuth2ClientCredentialUser;
const OAuth2ClientCredentials = __importStar(require("./o-auth2-client-credentials"));
exports.OAuth2ClientCredentials = OAuth2ClientCredentials.OAuth2ClientCredentials;
const OAuthClientCertificate = __importStar(require("./o-auth-client-certificate"));
exports.OAuthClientCertificate = OAuthClientCertificate.OAuthClientCertificate;
const OAuthClientCertificateSearchRequest = __importStar(require("./o-auth-client-certificate-search-request"));
exports.OAuthClientCertificateSearchRequest = OAuthClientCertificateSearchRequest.OAuthClientCertificateSearchRequest;
const OAuthClientCertificates = __importStar(require("./o-auth-client-certificates"));
exports.OAuthClientCertificates = OAuthClientCertificates.OAuthClientCertificates;
const OAuthPartnerCertificate = __importStar(require("./o-auth-partner-certificate"));
exports.OAuthPartnerCertificate = OAuthPartnerCertificate.OAuthPartnerCertificate;
const OAuthPartnerCertificateSearchRequest = __importStar(require("./o-auth-partner-certificate-search-request"));
exports.OAuthPartnerCertificateSearchRequest = OAuthPartnerCertificateSearchRequest.OAuthPartnerCertificateSearchRequest;
const OAuthPartnerCertificates = __importStar(require("./o-auth-partner-certificates"));
exports.OAuthPartnerCertificates = OAuthPartnerCertificates.OAuthPartnerCertificates;
const OciConsoleSignOnPolicyConsent = __importStar(require("./oci-console-sign-on-policy-consent"));
exports.OciConsoleSignOnPolicyConsent = OciConsoleSignOnPolicyConsent.OciConsoleSignOnPolicyConsent;
const OciConsoleSignOnPolicyConsentConsentSignedBy = __importStar(require("./oci-console-sign-on-policy-consent-consent-signed-by"));
exports.OciConsoleSignOnPolicyConsentConsentSignedBy = OciConsoleSignOnPolicyConsentConsentSignedBy.OciConsoleSignOnPolicyConsentConsentSignedBy;
const OciConsoleSignOnPolicyConsentModifiedResource = __importStar(require("./oci-console-sign-on-policy-consent-modified-resource"));
exports.OciConsoleSignOnPolicyConsentModifiedResource = OciConsoleSignOnPolicyConsentModifiedResource.OciConsoleSignOnPolicyConsentModifiedResource;
const OciConsoleSignOnPolicyConsentPolicyResource = __importStar(require("./oci-console-sign-on-policy-consent-policy-resource"));
exports.OciConsoleSignOnPolicyConsentPolicyResource = OciConsoleSignOnPolicyConsentPolicyResource.OciConsoleSignOnPolicyConsentPolicyResource;
const OciConsoleSignOnPolicyConsentSearchRequest = __importStar(require("./oci-console-sign-on-policy-consent-search-request"));
exports.OciConsoleSignOnPolicyConsentSearchRequest = OciConsoleSignOnPolicyConsentSearchRequest.OciConsoleSignOnPolicyConsentSearchRequest;
const OciConsoleSignOnPolicyConsents = __importStar(require("./oci-console-sign-on-policy-consents"));
exports.OciConsoleSignOnPolicyConsents = OciConsoleSignOnPolicyConsents.OciConsoleSignOnPolicyConsents;
const Operations = __importStar(require("./operations"));
exports.Operations = Operations.Operations;
const PasswordPolicies = __importStar(require("./password-policies"));
exports.PasswordPolicies = PasswordPolicies.PasswordPolicies;
const PasswordPolicy = __importStar(require("./password-policy"));
exports.PasswordPolicy = PasswordPolicy.PasswordPolicy;
const PasswordPolicyConfiguredPasswordPolicyRules = __importStar(require("./password-policy-configured-password-policy-rules"));
exports.PasswordPolicyConfiguredPasswordPolicyRules = PasswordPolicyConfiguredPasswordPolicyRules.PasswordPolicyConfiguredPasswordPolicyRules;
const PasswordPolicyGroups = __importStar(require("./password-policy-groups"));
exports.PasswordPolicyGroups = PasswordPolicyGroups.PasswordPolicyGroups;
const PasswordPolicySearchRequest = __importStar(require("./password-policy-search-request"));
exports.PasswordPolicySearchRequest = PasswordPolicySearchRequest.PasswordPolicySearchRequest;
const PatchOp = __importStar(require("./patch-op"));
exports.PatchOp = PatchOp.PatchOp;
const Policies = __importStar(require("./policies"));
exports.Policies = Policies.Policies;
const Policy = __importStar(require("./policy"));
exports.Policy = Policy.Policy;
const PolicyExtensionOciconsolesignonpolicyconsentPolicy = __importStar(require("./policy-extension-ociconsolesignonpolicyconsent-policy"));
exports.PolicyExtensionOciconsolesignonpolicyconsentPolicy = PolicyExtensionOciconsolesignonpolicyconsentPolicy.PolicyExtensionOciconsolesignonpolicyconsentPolicy;
const PolicyPolicyType = __importStar(require("./policy-policy-type"));
exports.PolicyPolicyType = PolicyPolicyType.PolicyPolicyType;
const PolicyRules = __importStar(require("./policy-rules"));
exports.PolicyRules = PolicyRules.PolicyRules;
const PolicySearchRequest = __importStar(require("./policy-search-request"));
exports.PolicySearchRequest = PolicySearchRequest.PolicySearchRequest;
const ResourceTypeSchemaAttribute = __importStar(require("./resource-type-schema-attribute"));
exports.ResourceTypeSchemaAttribute = ResourceTypeSchemaAttribute.ResourceTypeSchemaAttribute;
const ResourceTypeSchemaAttributeSearchRequest = __importStar(require("./resource-type-schema-attribute-search-request"));
exports.ResourceTypeSchemaAttributeSearchRequest = ResourceTypeSchemaAttributeSearchRequest.ResourceTypeSchemaAttributeSearchRequest;
const ResourceTypeSchemaAttributes = __importStar(require("./resource-type-schema-attributes"));
exports.ResourceTypeSchemaAttributes = ResourceTypeSchemaAttributes.ResourceTypeSchemaAttributes;
const RestoreOciConsolePolicy = __importStar(require("./restore-oci-console-policy"));
exports.RestoreOciConsolePolicy = RestoreOciConsolePolicy.RestoreOciConsolePolicy;
const Rule = __importStar(require("./rule"));
exports.Rule = Rule.Rule;
const RuleConditionGroup = __importStar(require("./rule-condition-group"));
exports.RuleConditionGroup = RuleConditionGroup.RuleConditionGroup;
const RuleExtensionOciconsolesignonpolicyconsentPolicy = __importStar(require("./rule-extension-ociconsolesignonpolicyconsent-policy"));
exports.RuleExtensionOciconsolesignonpolicyconsentPolicy = RuleExtensionOciconsolesignonpolicyconsentPolicy.RuleExtensionOciconsolesignonpolicyconsentPolicy;
const RulePolicyType = __importStar(require("./rule-policy-type"));
exports.RulePolicyType = RulePolicyType.RulePolicyType;
const RuleReturn = __importStar(require("./rule-return"));
exports.RuleReturn = RuleReturn.RuleReturn;
const RuleSearchRequest = __importStar(require("./rule-search-request"));
exports.RuleSearchRequest = RuleSearchRequest.RuleSearchRequest;
const Rules = __importStar(require("./rules"));
exports.Rules = Rules.Rules;
const Schema = __importStar(require("./schema"));
exports.Schema = Schema.Schema;
const SchemaAttributes = __importStar(require("./schema-attributes"));
exports.SchemaAttributes = SchemaAttributes.SchemaAttributes;
const SchemaIdcsComplexAttributeNameMappings = __importStar(require("./schema-idcs-complex-attribute-name-mappings"));
exports.SchemaIdcsComplexAttributeNameMappings = SchemaIdcsComplexAttributeNameMappings.SchemaIdcsComplexAttributeNameMappings;
const SchemaIdcsCsvAttributeNameMappings = __importStar(require("./schema-idcs-csv-attribute-name-mappings"));
exports.SchemaIdcsCsvAttributeNameMappings = SchemaIdcsCsvAttributeNameMappings.SchemaIdcsCsvAttributeNameMappings;
const SchemaLocalizedCanonicalValues = __importStar(require("./schema-localized-canonical-values"));
exports.SchemaLocalizedCanonicalValues = SchemaLocalizedCanonicalValues.SchemaLocalizedCanonicalValues;
const SchemaLocalizedDisplayName = __importStar(require("./schema-localized-display-name"));
exports.SchemaLocalizedDisplayName = SchemaLocalizedDisplayName.SchemaLocalizedDisplayName;
const SchemaReferencedResourceTypeUniqueAttributeNameMappings = __importStar(require("./schema-referenced-resource-type-unique-attribute-name-mappings"));
exports.SchemaReferencedResourceTypeUniqueAttributeNameMappings = SchemaReferencedResourceTypeUniqueAttributeNameMappings.SchemaReferencedResourceTypeUniqueAttributeNameMappings;
const SchemaResolveValueUsingResourceType = __importStar(require("./schema-resolve-value-using-resource-type"));
exports.SchemaResolveValueUsingResourceType = SchemaResolveValueUsingResourceType.SchemaResolveValueUsingResourceType;
const SchemaSearchRequest = __importStar(require("./schema-search-request"));
exports.SchemaSearchRequest = SchemaSearchRequest.SchemaSearchRequest;
const SchemaSubAttributes = __importStar(require("./schema-sub-attributes"));
exports.SchemaSubAttributes = SchemaSubAttributes.SchemaSubAttributes;
const Schemas = __importStar(require("./schemas"));
exports.Schemas = Schemas.Schemas;
const SecurityQuestion = __importStar(require("./security-question"));
exports.SecurityQuestion = SecurityQuestion.SecurityQuestion;
const SecurityQuestionQuestionText = __importStar(require("./security-question-question-text"));
exports.SecurityQuestionQuestionText = SecurityQuestionQuestionText.SecurityQuestionQuestionText;
const SecurityQuestionSearchRequest = __importStar(require("./security-question-search-request"));
exports.SecurityQuestionSearchRequest = SecurityQuestionSearchRequest.SecurityQuestionSearchRequest;
const SecurityQuestionSetting = __importStar(require("./security-question-setting"));
exports.SecurityQuestionSetting = SecurityQuestionSetting.SecurityQuestionSetting;
const SecurityQuestionSettings = __importStar(require("./security-question-settings"));
exports.SecurityQuestionSettings = SecurityQuestionSettings.SecurityQuestionSettings;
const SecurityQuestionSettingsSearchRequest = __importStar(require("./security-question-settings-search-request"));
exports.SecurityQuestionSettingsSearchRequest = SecurityQuestionSettingsSearchRequest.SecurityQuestionSettingsSearchRequest;
const SecurityQuestions = __importStar(require("./security-questions"));
exports.SecurityQuestions = SecurityQuestions.SecurityQuestions;
const SelfRegistrationProfile = __importStar(require("./self-registration-profile"));
exports.SelfRegistrationProfile = SelfRegistrationProfile.SelfRegistrationProfile;
const SelfRegistrationProfileAfterSubmitText = __importStar(require("./self-registration-profile-after-submit-text"));
exports.SelfRegistrationProfileAfterSubmitText = SelfRegistrationProfileAfterSubmitText.SelfRegistrationProfileAfterSubmitText;
const SelfRegistrationProfileConsentText = __importStar(require("./self-registration-profile-consent-text"));
exports.SelfRegistrationProfileConsentText = SelfRegistrationProfileConsentText.SelfRegistrationProfileConsentText;
const SelfRegistrationProfileDefaultGroups = __importStar(require("./self-registration-profile-default-groups"));
exports.SelfRegistrationProfileDefaultGroups = SelfRegistrationProfileDefaultGroups.SelfRegistrationProfileDefaultGroups;
const SelfRegistrationProfileDisplayName = __importStar(require("./self-registration-profile-display-name"));
exports.SelfRegistrationProfileDisplayName = SelfRegistrationProfileDisplayName.SelfRegistrationProfileDisplayName;
const SelfRegistrationProfileEmailTemplate = __importStar(require("./self-registration-profile-email-template"));
exports.SelfRegistrationProfileEmailTemplate = SelfRegistrationProfileEmailTemplate.SelfRegistrationProfileEmailTemplate;
const SelfRegistrationProfileFooterText = __importStar(require("./self-registration-profile-footer-text"));
exports.SelfRegistrationProfileFooterText = SelfRegistrationProfileFooterText.SelfRegistrationProfileFooterText;
const SelfRegistrationProfileHeaderText = __importStar(require("./self-registration-profile-header-text"));
exports.SelfRegistrationProfileHeaderText = SelfRegistrationProfileHeaderText.SelfRegistrationProfileHeaderText;
const SelfRegistrationProfileSearchRequest = __importStar(require("./self-registration-profile-search-request"));
exports.SelfRegistrationProfileSearchRequest = SelfRegistrationProfileSearchRequest.SelfRegistrationProfileSearchRequest;
const SelfRegistrationProfileUserAttributes = __importStar(require("./self-registration-profile-user-attributes"));
exports.SelfRegistrationProfileUserAttributes = SelfRegistrationProfileUserAttributes.SelfRegistrationProfileUserAttributes;
const SelfRegistrationProfiles = __importStar(require("./self-registration-profiles"));
exports.SelfRegistrationProfiles = SelfRegistrationProfiles.SelfRegistrationProfiles;
const Setting = __importStar(require("./setting"));
exports.Setting = Setting.Setting;
const Settings = __importStar(require("./settings"));
exports.Settings = Settings.Settings;
const SettingsCertificateValidation = __importStar(require("./settings-certificate-validation"));
exports.SettingsCertificateValidation = SettingsCertificateValidation.SettingsCertificateValidation;
const SettingsCloudGateCorsSettings = __importStar(require("./settings-cloud-gate-cors-settings"));
exports.SettingsCloudGateCorsSettings = SettingsCloudGateCorsSettings.SettingsCloudGateCorsSettings;
const SettingsCompanyNames = __importStar(require("./settings-company-names"));
exports.SettingsCompanyNames = SettingsCompanyNames.SettingsCompanyNames;
const SettingsDefaultCompanyNames = __importStar(require("./settings-default-company-names"));
exports.SettingsDefaultCompanyNames = SettingsDefaultCompanyNames.SettingsDefaultCompanyNames;
const SettingsDefaultImages = __importStar(require("./settings-default-images"));
exports.SettingsDefaultImages = SettingsDefaultImages.SettingsDefaultImages;
const SettingsDefaultLoginTexts = __importStar(require("./settings-default-login-texts"));
exports.SettingsDefaultLoginTexts = SettingsDefaultLoginTexts.SettingsDefaultLoginTexts;
const SettingsImages = __importStar(require("./settings-images"));
exports.SettingsImages = SettingsImages.SettingsImages;
const SettingsLoginTexts = __importStar(require("./settings-login-texts"));
exports.SettingsLoginTexts = SettingsLoginTexts.SettingsLoginTexts;
const SettingsPurgeConfigs = __importStar(require("./settings-purge-configs"));
exports.SettingsPurgeConfigs = SettingsPurgeConfigs.SettingsPurgeConfigs;
const SettingsSearchRequest = __importStar(require("./settings-search-request"));
exports.SettingsSearchRequest = SettingsSearchRequest.SettingsSearchRequest;
const SettingsTenantCustomClaims = __importStar(require("./settings-tenant-custom-claims"));
exports.SettingsTenantCustomClaims = SettingsTenantCustomClaims.SettingsTenantCustomClaims;
const SmtpCredential = __importStar(require("./smtp-credential"));
exports.SmtpCredential = SmtpCredential.SmtpCredential;
const SmtpCredentialSearchRequest = __importStar(require("./smtp-credential-search-request"));
exports.SmtpCredentialSearchRequest = SmtpCredentialSearchRequest.SmtpCredentialSearchRequest;
const SmtpCredentialUser = __importStar(require("./smtp-credential-user"));
exports.SmtpCredentialUser = SmtpCredentialUser.SmtpCredentialUser;
const SmtpCredentials = __importStar(require("./smtp-credentials"));
exports.SmtpCredentials = SmtpCredentials.SmtpCredentials;
const SocialIdentityProvider = __importStar(require("./social-identity-provider"));
exports.SocialIdentityProvider = SocialIdentityProvider.SocialIdentityProvider;
const SocialIdentityProviderJitProvAssignedGroups = __importStar(require("./social-identity-provider-jit-prov-assigned-groups"));
exports.SocialIdentityProviderJitProvAssignedGroups = SocialIdentityProviderJitProvAssignedGroups.SocialIdentityProviderJitProvAssignedGroups;
const SocialIdentityProviderRelayIdpParamMappings = __importStar(require("./social-identity-provider-relay-idp-param-mappings"));
exports.SocialIdentityProviderRelayIdpParamMappings = SocialIdentityProviderRelayIdpParamMappings.SocialIdentityProviderRelayIdpParamMappings;
const SocialIdentityProviderSearchRequest = __importStar(require("./social-identity-provider-search-request"));
exports.SocialIdentityProviderSearchRequest = SocialIdentityProviderSearchRequest.SocialIdentityProviderSearchRequest;
const SocialIdentityProviders = __importStar(require("./social-identity-providers"));
exports.SocialIdentityProviders = SocialIdentityProviders.SocialIdentityProviders;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Tags = __importStar(require("./tags"));
exports.Tags = Tags.Tags;
const User = __importStar(require("./user"));
exports.User = User.User;
const UserAttributesSetting = __importStar(require("./user-attributes-setting"));
exports.UserAttributesSetting = UserAttributesSetting.UserAttributesSetting;
const UserAttributesSettings = __importStar(require("./user-attributes-settings"));
exports.UserAttributesSettings = UserAttributesSettings.UserAttributesSettings;
const UserAttributesSettingsAttributeSettings = __importStar(require("./user-attributes-settings-attribute-settings"));
exports.UserAttributesSettingsAttributeSettings = UserAttributesSettingsAttributeSettings.UserAttributesSettingsAttributeSettings;
const UserAttributesSettingsSearchRequest = __importStar(require("./user-attributes-settings-search-request"));
exports.UserAttributesSettingsSearchRequest = UserAttributesSettingsSearchRequest.UserAttributesSettingsSearchRequest;
const UserCapabilitiesChanger = __importStar(require("./user-capabilities-changer"));
exports.UserCapabilitiesChanger = UserCapabilitiesChanger.UserCapabilitiesChanger;
const UserDbCredential = __importStar(require("./user-db-credential"));
exports.UserDbCredential = UserDbCredential.UserDbCredential;
const UserDbCredentials = __importStar(require("./user-db-credentials"));
exports.UserDbCredentials = UserDbCredentials.UserDbCredentials;
const UserDbCredentialsSearchRequest = __importStar(require("./user-db-credentials-search-request"));
exports.UserDbCredentialsSearchRequest = UserDbCredentialsSearchRequest.UserDbCredentialsSearchRequest;
const UserDbCredentialsUser = __importStar(require("./user-db-credentials-user"));
exports.UserDbCredentialsUser = UserDbCredentialsUser.UserDbCredentialsUser;
const UserEmails = __importStar(require("./user-emails"));
exports.UserEmails = UserEmails.UserEmails;
const UserEntitlements = __importStar(require("./user-entitlements"));
exports.UserEntitlements = UserEntitlements.UserEntitlements;
const UserExtAccounts = __importStar(require("./user-ext-accounts"));
exports.UserExtAccounts = UserExtAccounts.UserExtAccounts;
const UserExtApiKeys = __importStar(require("./user-ext-api-keys"));
exports.UserExtApiKeys = UserExtApiKeys.UserExtApiKeys;
const UserExtAppRoles = __importStar(require("./user-ext-app-roles"));
exports.UserExtAppRoles = UserExtAppRoles.UserExtAppRoles;
const UserExtApplicableAuthenticationTargetApp = __importStar(require("./user-ext-applicable-authentication-target-app"));
exports.UserExtApplicableAuthenticationTargetApp = UserExtApplicableAuthenticationTargetApp.UserExtApplicableAuthenticationTargetApp;
const UserExtApplicablePasswordPolicy = __importStar(require("./user-ext-applicable-password-policy"));
exports.UserExtApplicablePasswordPolicy = UserExtApplicablePasswordPolicy.UserExtApplicablePasswordPolicy;
const UserExtAuthTokens = __importStar(require("./user-ext-auth-tokens"));
exports.UserExtAuthTokens = UserExtAuthTokens.UserExtAuthTokens;
const UserExtBypassCodes = __importStar(require("./user-ext-bypass-codes"));
exports.UserExtBypassCodes = UserExtBypassCodes.UserExtBypassCodes;
const UserExtCustomerSecretKeys = __importStar(require("./user-ext-customer-secret-keys"));
exports.UserExtCustomerSecretKeys = UserExtCustomerSecretKeys.UserExtCustomerSecretKeys;
const UserExtDbCredentials = __importStar(require("./user-ext-db-credentials"));
exports.UserExtDbCredentials = UserExtDbCredentials.UserExtDbCredentials;
const UserExtDelegatedAuthenticationTargetApp = __importStar(require("./user-ext-delegated-authentication-target-app"));
exports.UserExtDelegatedAuthenticationTargetApp = UserExtDelegatedAuthenticationTargetApp.UserExtDelegatedAuthenticationTargetApp;
const UserExtDevices = __importStar(require("./user-ext-devices"));
exports.UserExtDevices = UserExtDevices.UserExtDevices;
const UserExtFactorIdentifier = __importStar(require("./user-ext-factor-identifier"));
exports.UserExtFactorIdentifier = UserExtFactorIdentifier.UserExtFactorIdentifier;
const UserExtGrants = __importStar(require("./user-ext-grants"));
exports.UserExtGrants = UserExtGrants.UserExtGrants;
const UserExtIdcsAppRolesLimitedToGroups = __importStar(require("./user-ext-idcs-app-roles-limited-to-groups"));
exports.UserExtIdcsAppRolesLimitedToGroups = UserExtIdcsAppRolesLimitedToGroups.UserExtIdcsAppRolesLimitedToGroups;
const UserExtLocked = __importStar(require("./user-ext-locked"));
exports.UserExtLocked = UserExtLocked.UserExtLocked;
const UserExtManager = __importStar(require("./user-ext-manager"));
exports.UserExtManager = UserExtManager.UserExtManager;
const UserExtOAuth2ClientCredentials = __importStar(require("./user-ext-oauth2-client-credentials"));
exports.UserExtOAuth2ClientCredentials = UserExtOAuth2ClientCredentials.UserExtOAuth2ClientCredentials;
const UserExtPasswordVerifiers = __importStar(require("./user-ext-password-verifiers"));
exports.UserExtPasswordVerifiers = UserExtPasswordVerifiers.UserExtPasswordVerifiers;
const UserExtPreferredDevice = __importStar(require("./user-ext-preferred-device"));
exports.UserExtPreferredDevice = UserExtPreferredDevice.UserExtPreferredDevice;
const UserExtRealmUsers = __importStar(require("./user-ext-realm-users"));
exports.UserExtRealmUsers = UserExtRealmUsers.UserExtRealmUsers;
const UserExtRecoveryLocked = __importStar(require("./user-ext-recovery-locked"));
exports.UserExtRecoveryLocked = UserExtRecoveryLocked.UserExtRecoveryLocked;
const UserExtRiskScores = __importStar(require("./user-ext-risk-scores"));
exports.UserExtRiskScores = UserExtRiskScores.UserExtRiskScores;
const UserExtSecQuestions = __importStar(require("./user-ext-sec-questions"));
exports.UserExtSecQuestions = UserExtSecQuestions.UserExtSecQuestions;
const UserExtSelfRegistrationProfile = __importStar(require("./user-ext-self-registration-profile"));
exports.UserExtSelfRegistrationProfile = UserExtSelfRegistrationProfile.UserExtSelfRegistrationProfile;
const UserExtSmtpCredentials = __importStar(require("./user-ext-smtp-credentials"));
exports.UserExtSmtpCredentials = UserExtSmtpCredentials.UserExtSmtpCredentials;
const UserExtSocialAccounts = __importStar(require("./user-ext-social-accounts"));
exports.UserExtSocialAccounts = UserExtSocialAccounts.UserExtSocialAccounts;
const UserExtSupportAccounts = __importStar(require("./user-ext-support-accounts"));
exports.UserExtSupportAccounts = UserExtSupportAccounts.UserExtSupportAccounts;
const UserExtSyncedFromApp = __importStar(require("./user-ext-synced-from-app"));
exports.UserExtSyncedFromApp = UserExtSyncedFromApp.UserExtSyncedFromApp;
const UserExtTermsOfUseConsents = __importStar(require("./user-ext-terms-of-use-consents"));
exports.UserExtTermsOfUseConsents = UserExtTermsOfUseConsents.UserExtTermsOfUseConsents;
const UserExtTrustedUserAgents = __importStar(require("./user-ext-trusted-user-agents"));
exports.UserExtTrustedUserAgents = UserExtTrustedUserAgents.UserExtTrustedUserAgents;
const UserExtUserToken = __importStar(require("./user-ext-user-token"));
exports.UserExtUserToken = UserExtUserToken.UserExtUserToken;
const UserGroups = __importStar(require("./user-groups"));
exports.UserGroups = UserGroups.UserGroups;
const UserIms = __importStar(require("./user-ims"));
exports.UserIms = UserIms.UserIms;
const UserName = __importStar(require("./user-name"));
exports.UserName = UserName.UserName;
const UserPasswordChanger = __importStar(require("./user-password-changer"));
exports.UserPasswordChanger = UserPasswordChanger.UserPasswordChanger;
const UserPasswordResetter = __importStar(require("./user-password-resetter"));
exports.UserPasswordResetter = UserPasswordResetter.UserPasswordResetter;
const UserPasswordResetterUserToken = __importStar(require("./user-password-resetter-user-token"));
exports.UserPasswordResetterUserToken = UserPasswordResetterUserToken.UserPasswordResetterUserToken;
const UserPhoneNumbers = __importStar(require("./user-phone-numbers"));
exports.UserPhoneNumbers = UserPhoneNumbers.UserPhoneNumbers;
const UserPhotos = __importStar(require("./user-photos"));
exports.UserPhotos = UserPhotos.UserPhotos;
const UserRoles = __importStar(require("./user-roles"));
exports.UserRoles = UserRoles.UserRoles;
const UserSearchRequest = __importStar(require("./user-search-request"));
exports.UserSearchRequest = UserSearchRequest.UserSearchRequest;
const UserStatusChanger = __importStar(require("./user-status-changer"));
exports.UserStatusChanger = UserStatusChanger.UserStatusChanger;
const UserX509Certificates = __importStar(require("./user-x509-certificates"));
exports.UserX509Certificates = UserX509Certificates.UserX509Certificates;
const Users = __importStar(require("./users"));
exports.Users = Users.Users;
