// oci-identitydomains/lib/model/schema-sub-attributes.js
"use strict";
/**
 * Identity Domains API
 * Use the Identity Domains API to manage resources within an identity domain, for example, users, dynamic resource groups, groups, and identity providers. For information about managing resources within identity domains, see [Identity and Access Management (with identity domains)](https://docs.oracle.com/iaas/Content/Identity/home.htm).
Use this pattern to construct endpoints for identity domains: {@code https://<domainURL>/admin/v1/}. See [Finding an Identity Domain URL](https://docs.oracle.com/en-us/iaas/Content/Identity/api-getstarted/locate-identity-domain-url.htm) to locate the domain URL you need.
Use the table of contents and search tool to explore the Identity Domains API.
 * OpenAPI spec version: v1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaSubAttributes = void 0;
var SchemaSubAttributes;
(function (SchemaSubAttributes) {
    let Type;
    (function (Type) {
        Type["String"] = "string";
        Type["Complex"] = "complex";
        Type["Boolean"] = "boolean";
        Type["Decimal"] = "decimal";
        Type["Integer"] = "integer";
        Type["DateTime"] = "dateTime";
        Type["Reference"] = "reference";
        Type["Binary"] = "binary";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = SchemaSubAttributes.Type || (SchemaSubAttributes.Type = {}));
    let Mutability;
    (function (Mutability) {
        Mutability["ReadOnly"] = "readOnly";
        Mutability["ReadWrite"] = "readWrite";
        Mutability["Immutable"] = "immutable";
        Mutability["WriteOnly"] = "writeOnly";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Mutability["UnknownValue"] = "UNKNOWN_VALUE";
    })(Mutability = SchemaSubAttributes.Mutability || (SchemaSubAttributes.Mutability = {}));
    let Returned;
    (function (Returned) {
        Returned["Always"] = "always";
        Returned["Never"] = "never";
        Returned["Default"] = "default";
        Returned["Request"] = "request";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Returned["UnknownValue"] = "UNKNOWN_VALUE";
    })(Returned = SchemaSubAttributes.Returned || (SchemaSubAttributes.Returned = {}));
    let Uniqueness;
    (function (Uniqueness) {
        Uniqueness["None"] = "none";
        Uniqueness["Server"] = "server";
        Uniqueness["Global"] = "global";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Uniqueness["UnknownValue"] = "UNKNOWN_VALUE";
    })(Uniqueness = SchemaSubAttributes.Uniqueness || (SchemaSubAttributes.Uniqueness = {}));
    let IdcsSensitive;
    (function (IdcsSensitive) {
        IdcsSensitive["Encrypt"] = "encrypt";
        IdcsSensitive["Hash"] = "hash";
        IdcsSensitive["HashSc"] = "hash_sc";
        IdcsSensitive["Checksum"] = "checksum";
        IdcsSensitive["None"] = "none";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        IdcsSensitive["UnknownValue"] = "UNKNOWN_VALUE";
    })(IdcsSensitive = SchemaSubAttributes.IdcsSensitive || (SchemaSubAttributes.IdcsSensitive = {}));
    let IdcsuiWidget;
    (function (IdcsuiWidget) {
        IdcsuiWidget["Inputtext"] = "inputtext";
        IdcsuiWidget["Checkbox"] = "checkbox";
        IdcsuiWidget["Textarea"] = "textarea";
        IdcsuiWidget["Combobox"] = "combobox";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        IdcsuiWidget["UnknownValue"] = "UNKNOWN_VALUE";
    })(IdcsuiWidget = SchemaSubAttributes.IdcsuiWidget || (SchemaSubAttributes.IdcsuiWidget = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SchemaSubAttributes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SchemaSubAttributes.getDeserializedJsonObj = getDeserializedJsonObj;
})(SchemaSubAttributes = exports.SchemaSubAttributes || (exports.SchemaSubAttributes = {}));
