// oci-databasetools/lib/model/key-store-type-generic-jdbc.js
"use strict";
/**
 * Database Tools
 * Use the Database Tools API to manage connections, private endpoints, and work requests in the Database Tools service.
 * OpenAPI spec version: 20201005
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeyStoreTypeGenericJdbc = void 0;
/**
 * The key store type.
 **/
var KeyStoreTypeGenericJdbc;
(function (KeyStoreTypeGenericJdbc) {
    KeyStoreTypeGenericJdbc["JavaKeyStore"] = "JAVA_KEY_STORE";
    KeyStoreTypeGenericJdbc["JavaTrustStore"] = "JAVA_TRUST_STORE";
    KeyStoreTypeGenericJdbc["Pkcs12"] = "PKCS12";
    KeyStoreTypeGenericJdbc["Sso"] = "SSO";
    KeyStoreTypeGenericJdbc["ClientCertificatePem"] = "CLIENT_CERTIFICATE_PEM";
    KeyStoreTypeGenericJdbc["ClientPrivateKeyPem"] = "CLIENT_PRIVATE_KEY_PEM";
    KeyStoreTypeGenericJdbc["CaCertificatePem"] = "CA_CERTIFICATE_PEM";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    KeyStoreTypeGenericJdbc["UnknownValue"] = "UNKNOWN_VALUE";
})(KeyStoreTypeGenericJdbc = exports.KeyStoreTypeGenericJdbc || (exports.KeyStoreTypeGenericJdbc = {}));
(function (KeyStoreTypeGenericJdbc) {
    function getJsonObj(obj) {
        return obj;
    }
    KeyStoreTypeGenericJdbc.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    KeyStoreTypeGenericJdbc.getDeserializedJsonObj = getDeserializedJsonObj;
})(KeyStoreTypeGenericJdbc = exports.KeyStoreTypeGenericJdbc || (exports.KeyStoreTypeGenericJdbc = {}));
