// oci-databasetools/lib/model/key-store-type-postgresql.js
"use strict";
/**
 * Database Tools
 * Use the Database Tools API to manage connections, private endpoints, and work requests in the Database Tools service.
 * OpenAPI spec version: 20201005
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeyStoreTypePostgresql = void 0;
/**
 * The key store type.
 **/
var KeyStoreTypePostgresql;
(function (KeyStoreTypePostgresql) {
    KeyStoreTypePostgresql["ClientCertificatePem"] = "CLIENT_CERTIFICATE_PEM";
    KeyStoreTypePostgresql["ClientPrivateKeyPem"] = "CLIENT_PRIVATE_KEY_PEM";
    KeyStoreTypePostgresql["CaCertificatePem"] = "CA_CERTIFICATE_PEM";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    KeyStoreTypePostgresql["UnknownValue"] = "UNKNOWN_VALUE";
})(KeyStoreTypePostgresql = exports.KeyStoreTypePostgresql || (exports.KeyStoreTypePostgresql = {}));
(function (KeyStoreTypePostgresql) {
    function getJsonObj(obj) {
        return obj;
    }
    KeyStoreTypePostgresql.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    KeyStoreTypePostgresql.getDeserializedJsonObj = getDeserializedJsonObj;
})(KeyStoreTypePostgresql = exports.KeyStoreTypePostgresql || (exports.KeyStoreTypePostgresql = {}));
