// oci-databasetools/lib/model/proxy-authentication-type.js
"use strict";
/**
 * Database Tools
 * Use the Database Tools API to manage connections, private endpoints, and work requests in the Database Tools service.
 * OpenAPI spec version: 20201005
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProxyAuthenticationType = void 0;
/**
 * The proxy authentication type.
 **/
var ProxyAuthenticationType;
(function (ProxyAuthenticationType) {
    ProxyAuthenticationType["UserName"] = "USER_NAME";
    ProxyAuthenticationType["NoProxy"] = "NO_PROXY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ProxyAuthenticationType["UnknownValue"] = "UNKNOWN_VALUE";
})(ProxyAuthenticationType = exports.ProxyAuthenticationType || (exports.ProxyAuthenticationType = {}));
(function (ProxyAuthenticationType) {
    function getJsonObj(obj) {
        return obj;
    }
    ProxyAuthenticationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ProxyAuthenticationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ProxyAuthenticationType = exports.ProxyAuthenticationType || (exports.ProxyAuthenticationType = {}));
