// oci-databasetools/lib/model/remove-resource-lock-details.js
"use strict";
/**
 * Database Tools
 * Use the Database Tools API to manage connections, private endpoints, and work requests in the Database Tools service.
 * OpenAPI spec version: 20201005
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemoveResourceLockDetails = void 0;
var RemoveResourceLockDetails;
(function (RemoveResourceLockDetails) {
    let Type;
    (function (Type) {
        Type["Full"] = "FULL";
        Type["Delete"] = "DELETE";
    })(Type = RemoveResourceLockDetails.Type || (RemoveResourceLockDetails.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RemoveResourceLockDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RemoveResourceLockDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(RemoveResourceLockDetails = exports.RemoveResourceLockDetails || (exports.RemoveResourceLockDetails = {}));
