// oci-databasetools/lib/model/runtime-support.js
"use strict";
/**
 * Database Tools
 * Use the Database Tools API to manage connections, private endpoints, and work requests in the Database Tools service.
 * OpenAPI spec version: 20201005
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuntimeSupport = void 0;
/**
 * Specifies whether a connection is supported by the Database Tools Runtime.
 **/
var RuntimeSupport;
(function (RuntimeSupport) {
    RuntimeSupport["Supported"] = "SUPPORTED";
    RuntimeSupport["Unsupported"] = "UNSUPPORTED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RuntimeSupport["UnknownValue"] = "UNKNOWN_VALUE";
})(RuntimeSupport = exports.RuntimeSupport || (exports.RuntimeSupport = {}));
(function (RuntimeSupport) {
    function getJsonObj(obj) {
        return obj;
    }
    RuntimeSupport.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RuntimeSupport.getDeserializedJsonObj = getDeserializedJsonObj;
})(RuntimeSupport = exports.RuntimeSupport || (exports.RuntimeSupport = {}));
