// oci-database/lib/request/list-gi-version-minor-versions-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListGiVersionMinorVersionsRequest = void 0;
var ListGiVersionMinorVersionsRequest;
(function (ListGiVersionMinorVersionsRequest) {
    let ShapeFamily;
    (function (ShapeFamily) {
        ShapeFamily["Singlenode"] = "SINGLENODE";
        ShapeFamily["Yoda"] = "YODA";
        ShapeFamily["Virtualmachine"] = "VIRTUALMACHINE";
        ShapeFamily["Exadata"] = "EXADATA";
        ShapeFamily["Exacc"] = "EXACC";
        ShapeFamily["ExadbXs"] = "EXADB_XS";
    })(ShapeFamily = ListGiVersionMinorVersionsRequest.ShapeFamily || (ListGiVersionMinorVersionsRequest.ShapeFamily = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Version"] = "VERSION";
    })(SortBy = ListGiVersionMinorVersionsRequest.SortBy || (ListGiVersionMinorVersionsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListGiVersionMinorVersionsRequest.SortOrder || (ListGiVersionMinorVersionsRequest.SortOrder = {}));
})(ListGiVersionMinorVersionsRequest = exports.ListGiVersionMinorVersionsRequest || (exports.ListGiVersionMinorVersionsRequest = {}));
