// oci-database/lib/request/list-exadb-vm-cluster-updates-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListExadbVmClusterUpdatesRequest = void 0;
var ListExadbVmClusterUpdatesRequest;
(function (ListExadbVmClusterUpdatesRequest) {
    let UpdateType;
    (function (UpdateType) {
        UpdateType["GiUpgrade"] = "GI_UPGRADE";
        UpdateType["GiPatch"] = "GI_PATCH";
        UpdateType["OsUpdate"] = "OS_UPDATE";
    })(UpdateType = ListExadbVmClusterUpdatesRequest.UpdateType || (ListExadbVmClusterUpdatesRequest.UpdateType = {}));
})(ListExadbVmClusterUpdatesRequest = exports.ListExadbVmClusterUpdatesRequest || (exports.ListExadbVmClusterUpdatesRequest = {}));
