// oci-networkfirewall/lib/request/index.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20230501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListUrlListsRequest = exports.ListTunnelInspectionRulesRequest = exports.ListServicesRequest = exports.ListServiceListsRequest = exports.ListSecurityRulesRequest = exports.ListNetworkFirewallsRequest = exports.ListNetworkFirewallPoliciesRequest = exports.ListNatRulesRequest = exports.ListMappedSecretsRequest = exports.ListDecryptionRulesRequest = exports.ListDecryptionProfilesRequest = exports.ListApplicationsRequest = exports.ListApplicationGroupsRequest = exports.ListAddressListsRequest = void 0;
const ListAddressListsRequest = __importStar(require("./list-address-lists-request"));
exports.ListAddressListsRequest = ListAddressListsRequest.ListAddressListsRequest;
const ListApplicationGroupsRequest = __importStar(require("./list-application-groups-request"));
exports.ListApplicationGroupsRequest = ListApplicationGroupsRequest.ListApplicationGroupsRequest;
const ListApplicationsRequest = __importStar(require("./list-applications-request"));
exports.ListApplicationsRequest = ListApplicationsRequest.ListApplicationsRequest;
const ListDecryptionProfilesRequest = __importStar(require("./list-decryption-profiles-request"));
exports.ListDecryptionProfilesRequest = ListDecryptionProfilesRequest.ListDecryptionProfilesRequest;
const ListDecryptionRulesRequest = __importStar(require("./list-decryption-rules-request"));
exports.ListDecryptionRulesRequest = ListDecryptionRulesRequest.ListDecryptionRulesRequest;
const ListMappedSecretsRequest = __importStar(require("./list-mapped-secrets-request"));
exports.ListMappedSecretsRequest = ListMappedSecretsRequest.ListMappedSecretsRequest;
const ListNatRulesRequest = __importStar(require("./list-nat-rules-request"));
exports.ListNatRulesRequest = ListNatRulesRequest.ListNatRulesRequest;
const ListNetworkFirewallPoliciesRequest = __importStar(require("./list-network-firewall-policies-request"));
exports.ListNetworkFirewallPoliciesRequest = ListNetworkFirewallPoliciesRequest.ListNetworkFirewallPoliciesRequest;
const ListNetworkFirewallsRequest = __importStar(require("./list-network-firewalls-request"));
exports.ListNetworkFirewallsRequest = ListNetworkFirewallsRequest.ListNetworkFirewallsRequest;
const ListSecurityRulesRequest = __importStar(require("./list-security-rules-request"));
exports.ListSecurityRulesRequest = ListSecurityRulesRequest.ListSecurityRulesRequest;
const ListServiceListsRequest = __importStar(require("./list-service-lists-request"));
exports.ListServiceListsRequest = ListServiceListsRequest.ListServiceListsRequest;
const ListServicesRequest = __importStar(require("./list-services-request"));
exports.ListServicesRequest = ListServicesRequest.ListServicesRequest;
const ListTunnelInspectionRulesRequest = __importStar(require("./list-tunnel-inspection-rules-request"));
exports.ListTunnelInspectionRulesRequest = ListTunnelInspectionRulesRequest.ListTunnelInspectionRulesRequest;
const ListUrlListsRequest = __importStar(require("./list-url-lists-request"));
exports.ListUrlListsRequest = ListUrlListsRequest.ListUrlListsRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
