// oci-jms/lib/model/java-server-sort-by.js
"use strict";
/**
 * Java Management Service Fleets API
 * The APIs for the [Fleet Management](https://docs.oracle.com/en-us/iaas/jms/doc/fleet-management.html) feature of Java Management Service to monitor and manage the usage of Java in your enterprise. Use these APIs to manage fleets, configure managed instances to report to fleets, and gain insights into the Java workloads running on these instances by carrying out basic and advanced features.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JavaServerSortBy = void 0;
/**
 * The field to sort Java Servers.
 *
 **/
var JavaServerSortBy;
(function (JavaServerSortBy) {
    JavaServerSortBy["ServerName"] = "serverName";
    JavaServerSortBy["ServerVersion"] = "serverVersion";
    JavaServerSortBy["ServerInstanceCount"] = "serverInstanceCount";
    JavaServerSortBy["ApproximateDeployedApplicationCount"] = "approximateDeployedApplicationCount";
    JavaServerSortBy["TimeFirstSeen"] = "timeFirstSeen";
    JavaServerSortBy["TimeLastSeen"] = "timeLastSeen";
})(JavaServerSortBy = exports.JavaServerSortBy || (exports.JavaServerSortBy = {}));
(function (JavaServerSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    JavaServerSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    JavaServerSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(JavaServerSortBy = exports.JavaServerSortBy || (exports.JavaServerSortBy = {}));
