// oci-opensearch/lib/model/ccc-mode.js
"use strict";
/**
 * OpenSearch Service API
 * The OpenSearch service API provides access to OCI Search Service with OpenSearch.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CccMode = void 0;
/**
 * Mode for the cross cluster connection
 **/
var CccMode;
(function (CccMode) {
    CccMode["SearchOnly"] = "SEARCH_ONLY";
    CccMode["ReplicationOnly"] = "REPLICATION_ONLY";
    CccMode["SearchAndReplication"] = "SEARCH_AND_REPLICATION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CccMode["UnknownValue"] = "UNKNOWN_VALUE";
})(CccMode = exports.CccMode || (exports.CccMode = {}));
(function (CccMode) {
    function getJsonObj(obj) {
        return obj;
    }
    CccMode.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CccMode.getDeserializedJsonObj = getDeserializedJsonObj;
})(CccMode = exports.CccMode || (exports.CccMode = {}));
