// oci-opensearch/lib/model/index.js
"use strict";
/**
 * OpenSearch Service API
 * The OpenSearch service API provides access to OCI Search Service with OpenSearch.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReverseConnectionEndpoint = exports.RestoreOpensearchClusterDetails = exports.RestoreOpensearchClusterBackupDetails = exports.ResizeOpensearchClusterVerticalDetails = exports.ResizeOpensearchClusterHorizontalDetails = exports.ReclaimClusterDetails = exports.OutboundClusterSummary = exports.OutboundClusterConfig = exports.OperationType = exports.OperationStatus = exports.OpensearchVersionsSummary = exports.OpensearchVersionsCollection = exports.OpensearchPipelineReverseConnectionEndpoint = exports.OpensearchClusterSummary = exports.OpensearchClusterPipelineSummary = exports.OpensearchClusterPipelineCollection = exports.OpensearchClusterPipeline = exports.OpensearchClusterInternalDetails = exports.OpensearchClusterCollection = exports.OpensearchClusterBackupSummary = exports.OpensearchClusterBackupCollection = exports.OpensearchClusterBackup = exports.OpensearchCluster = exports.MasterNodeHostType = exports.MaintenanceState = exports.MaintenanceNotificationType = exports.MaintenanceNotificationResponse = exports.MaintenanceNotificationFailure = exports.MaintenanceNotificationDetails = exports.MaintenanceDetails = exports.GetManifestResponse = exports.ForcePatchPipelineDetails = exports.ForcePatchClusterDetails = exports.ExportOpensearchClusterBackupDetails = exports.DeleteBlockVolumesForNamespaceDetails = exports.DataNodeHostType = exports.CustomerLoggingDetails = exports.CreateOpensearchClusterPipelineDetails = exports.CreateOpensearchClusterDetails = exports.CreateMaintenanceDetails = exports.ConfigureOutboundClusterDetails = exports.ChangeOpensearchClusterCompartmentDetails = exports.ChangeOpensearchClusterBackupCompartmentDetails = exports.CccMode = exports.BenchmarkDetails = exports.BackupState = exports.BackupPolicy = exports.BackupOpensearchClusterDetails = exports.BackupEventDetails = exports.ActionType = void 0;
exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.UpgradeType = exports.UpgradeOpenSearchClusterDetails = exports.UpdatePipelineStatusDetails = exports.UpdateOpensearchClusterPipelineDetails = exports.UpdateOpensearchClusterDetails = exports.UpdateOpensearchClusterBackupDetails = exports.UpdateMaintenanceDetails = exports.UpdateClusterStatusDetails = exports.UpdateClusterSpecsDetails = exports.UpdateClusterHardenedImageDetails = exports.UpdateCheckinDetails = exports.SortOrder = exports.ShapesDetails = exports.SecuritySamlConfig = exports.SecurityMode = exports.SearchNodeHostType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const BackupEventDetails = __importStar(require("./backup-event-details"));
exports.BackupEventDetails = BackupEventDetails.BackupEventDetails;
const BackupOpensearchClusterDetails = __importStar(require("./backup-opensearch-cluster-details"));
exports.BackupOpensearchClusterDetails = BackupOpensearchClusterDetails.BackupOpensearchClusterDetails;
const BackupPolicy = __importStar(require("./backup-policy"));
exports.BackupPolicy = BackupPolicy.BackupPolicy;
const BackupState = __importStar(require("./backup-state"));
exports.BackupState = BackupState.BackupState;
const BenchmarkDetails = __importStar(require("./benchmark-details"));
exports.BenchmarkDetails = BenchmarkDetails.BenchmarkDetails;
const CccMode = __importStar(require("./ccc-mode"));
exports.CccMode = CccMode.CccMode;
const ChangeOpensearchClusterBackupCompartmentDetails = __importStar(require("./change-opensearch-cluster-backup-compartment-details"));
exports.ChangeOpensearchClusterBackupCompartmentDetails = ChangeOpensearchClusterBackupCompartmentDetails.ChangeOpensearchClusterBackupCompartmentDetails;
const ChangeOpensearchClusterCompartmentDetails = __importStar(require("./change-opensearch-cluster-compartment-details"));
exports.ChangeOpensearchClusterCompartmentDetails = ChangeOpensearchClusterCompartmentDetails.ChangeOpensearchClusterCompartmentDetails;
const ConfigureOutboundClusterDetails = __importStar(require("./configure-outbound-cluster-details"));
exports.ConfigureOutboundClusterDetails = ConfigureOutboundClusterDetails.ConfigureOutboundClusterDetails;
const CreateMaintenanceDetails = __importStar(require("./create-maintenance-details"));
exports.CreateMaintenanceDetails = CreateMaintenanceDetails.CreateMaintenanceDetails;
const CreateOpensearchClusterDetails = __importStar(require("./create-opensearch-cluster-details"));
exports.CreateOpensearchClusterDetails = CreateOpensearchClusterDetails.CreateOpensearchClusterDetails;
const CreateOpensearchClusterPipelineDetails = __importStar(require("./create-opensearch-cluster-pipeline-details"));
exports.CreateOpensearchClusterPipelineDetails = CreateOpensearchClusterPipelineDetails.CreateOpensearchClusterPipelineDetails;
const CustomerLoggingDetails = __importStar(require("./customer-logging-details"));
exports.CustomerLoggingDetails = CustomerLoggingDetails.CustomerLoggingDetails;
const DataNodeHostType = __importStar(require("./data-node-host-type"));
exports.DataNodeHostType = DataNodeHostType.DataNodeHostType;
const DeleteBlockVolumesForNamespaceDetails = __importStar(require("./delete-block-volumes-for-namespace-details"));
exports.DeleteBlockVolumesForNamespaceDetails = DeleteBlockVolumesForNamespaceDetails.DeleteBlockVolumesForNamespaceDetails;
const ExportOpensearchClusterBackupDetails = __importStar(require("./export-opensearch-cluster-backup-details"));
exports.ExportOpensearchClusterBackupDetails = ExportOpensearchClusterBackupDetails.ExportOpensearchClusterBackupDetails;
const ForcePatchClusterDetails = __importStar(require("./force-patch-cluster-details"));
exports.ForcePatchClusterDetails = ForcePatchClusterDetails.ForcePatchClusterDetails;
const ForcePatchPipelineDetails = __importStar(require("./force-patch-pipeline-details"));
exports.ForcePatchPipelineDetails = ForcePatchPipelineDetails.ForcePatchPipelineDetails;
const GetManifestResponse = __importStar(require("./get-manifest-response"));
exports.GetManifestResponse = GetManifestResponse.GetManifestResponse;
const MaintenanceDetails = __importStar(require("./maintenance-details"));
exports.MaintenanceDetails = MaintenanceDetails.MaintenanceDetails;
const MaintenanceNotificationDetails = __importStar(require("./maintenance-notification-details"));
exports.MaintenanceNotificationDetails = MaintenanceNotificationDetails.MaintenanceNotificationDetails;
const MaintenanceNotificationFailure = __importStar(require("./maintenance-notification-failure"));
exports.MaintenanceNotificationFailure = MaintenanceNotificationFailure.MaintenanceNotificationFailure;
const MaintenanceNotificationResponse = __importStar(require("./maintenance-notification-response"));
exports.MaintenanceNotificationResponse = MaintenanceNotificationResponse.MaintenanceNotificationResponse;
const MaintenanceNotificationType = __importStar(require("./maintenance-notification-type"));
exports.MaintenanceNotificationType = MaintenanceNotificationType.MaintenanceNotificationType;
const MaintenanceState = __importStar(require("./maintenance-state"));
exports.MaintenanceState = MaintenanceState.MaintenanceState;
const MasterNodeHostType = __importStar(require("./master-node-host-type"));
exports.MasterNodeHostType = MasterNodeHostType.MasterNodeHostType;
const OpensearchCluster = __importStar(require("./opensearch-cluster"));
exports.OpensearchCluster = OpensearchCluster.OpensearchCluster;
const OpensearchClusterBackup = __importStar(require("./opensearch-cluster-backup"));
exports.OpensearchClusterBackup = OpensearchClusterBackup.OpensearchClusterBackup;
const OpensearchClusterBackupCollection = __importStar(require("./opensearch-cluster-backup-collection"));
exports.OpensearchClusterBackupCollection = OpensearchClusterBackupCollection.OpensearchClusterBackupCollection;
const OpensearchClusterBackupSummary = __importStar(require("./opensearch-cluster-backup-summary"));
exports.OpensearchClusterBackupSummary = OpensearchClusterBackupSummary.OpensearchClusterBackupSummary;
const OpensearchClusterCollection = __importStar(require("./opensearch-cluster-collection"));
exports.OpensearchClusterCollection = OpensearchClusterCollection.OpensearchClusterCollection;
const OpensearchClusterInternalDetails = __importStar(require("./opensearch-cluster-internal-details"));
exports.OpensearchClusterInternalDetails = OpensearchClusterInternalDetails.OpensearchClusterInternalDetails;
const OpensearchClusterPipeline = __importStar(require("./opensearch-cluster-pipeline"));
exports.OpensearchClusterPipeline = OpensearchClusterPipeline.OpensearchClusterPipeline;
const OpensearchClusterPipelineCollection = __importStar(require("./opensearch-cluster-pipeline-collection"));
exports.OpensearchClusterPipelineCollection = OpensearchClusterPipelineCollection.OpensearchClusterPipelineCollection;
const OpensearchClusterPipelineSummary = __importStar(require("./opensearch-cluster-pipeline-summary"));
exports.OpensearchClusterPipelineSummary = OpensearchClusterPipelineSummary.OpensearchClusterPipelineSummary;
const OpensearchClusterSummary = __importStar(require("./opensearch-cluster-summary"));
exports.OpensearchClusterSummary = OpensearchClusterSummary.OpensearchClusterSummary;
const OpensearchPipelineReverseConnectionEndpoint = __importStar(require("./opensearch-pipeline-reverse-connection-endpoint"));
exports.OpensearchPipelineReverseConnectionEndpoint = OpensearchPipelineReverseConnectionEndpoint.OpensearchPipelineReverseConnectionEndpoint;
const OpensearchVersionsCollection = __importStar(require("./opensearch-versions-collection"));
exports.OpensearchVersionsCollection = OpensearchVersionsCollection.OpensearchVersionsCollection;
const OpensearchVersionsSummary = __importStar(require("./opensearch-versions-summary"));
exports.OpensearchVersionsSummary = OpensearchVersionsSummary.OpensearchVersionsSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OutboundClusterConfig = __importStar(require("./outbound-cluster-config"));
exports.OutboundClusterConfig = OutboundClusterConfig.OutboundClusterConfig;
const OutboundClusterSummary = __importStar(require("./outbound-cluster-summary"));
exports.OutboundClusterSummary = OutboundClusterSummary.OutboundClusterSummary;
const ReclaimClusterDetails = __importStar(require("./reclaim-cluster-details"));
exports.ReclaimClusterDetails = ReclaimClusterDetails.ReclaimClusterDetails;
const ResizeOpensearchClusterHorizontalDetails = __importStar(require("./resize-opensearch-cluster-horizontal-details"));
exports.ResizeOpensearchClusterHorizontalDetails = ResizeOpensearchClusterHorizontalDetails.ResizeOpensearchClusterHorizontalDetails;
const ResizeOpensearchClusterVerticalDetails = __importStar(require("./resize-opensearch-cluster-vertical-details"));
exports.ResizeOpensearchClusterVerticalDetails = ResizeOpensearchClusterVerticalDetails.ResizeOpensearchClusterVerticalDetails;
const RestoreOpensearchClusterBackupDetails = __importStar(require("./restore-opensearch-cluster-backup-details"));
exports.RestoreOpensearchClusterBackupDetails = RestoreOpensearchClusterBackupDetails.RestoreOpensearchClusterBackupDetails;
const RestoreOpensearchClusterDetails = __importStar(require("./restore-opensearch-cluster-details"));
exports.RestoreOpensearchClusterDetails = RestoreOpensearchClusterDetails.RestoreOpensearchClusterDetails;
const ReverseConnectionEndpoint = __importStar(require("./reverse-connection-endpoint"));
exports.ReverseConnectionEndpoint = ReverseConnectionEndpoint.ReverseConnectionEndpoint;
const SearchNodeHostType = __importStar(require("./search-node-host-type"));
exports.SearchNodeHostType = SearchNodeHostType.SearchNodeHostType;
const SecurityMode = __importStar(require("./security-mode"));
exports.SecurityMode = SecurityMode.SecurityMode;
const SecuritySamlConfig = __importStar(require("./security-saml-config"));
exports.SecuritySamlConfig = SecuritySamlConfig.SecuritySamlConfig;
const ShapesDetails = __importStar(require("./shapes-details"));
exports.ShapesDetails = ShapesDetails.ShapesDetails;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateCheckinDetails = __importStar(require("./update-checkin-details"));
exports.UpdateCheckinDetails = UpdateCheckinDetails.UpdateCheckinDetails;
const UpdateClusterHardenedImageDetails = __importStar(require("./update-cluster-hardened-image-details"));
exports.UpdateClusterHardenedImageDetails = UpdateClusterHardenedImageDetails.UpdateClusterHardenedImageDetails;
const UpdateClusterSpecsDetails = __importStar(require("./update-cluster-specs-details"));
exports.UpdateClusterSpecsDetails = UpdateClusterSpecsDetails.UpdateClusterSpecsDetails;
const UpdateClusterStatusDetails = __importStar(require("./update-cluster-status-details"));
exports.UpdateClusterStatusDetails = UpdateClusterStatusDetails.UpdateClusterStatusDetails;
const UpdateMaintenanceDetails = __importStar(require("./update-maintenance-details"));
exports.UpdateMaintenanceDetails = UpdateMaintenanceDetails.UpdateMaintenanceDetails;
const UpdateOpensearchClusterBackupDetails = __importStar(require("./update-opensearch-cluster-backup-details"));
exports.UpdateOpensearchClusterBackupDetails = UpdateOpensearchClusterBackupDetails.UpdateOpensearchClusterBackupDetails;
const UpdateOpensearchClusterDetails = __importStar(require("./update-opensearch-cluster-details"));
exports.UpdateOpensearchClusterDetails = UpdateOpensearchClusterDetails.UpdateOpensearchClusterDetails;
const UpdateOpensearchClusterPipelineDetails = __importStar(require("./update-opensearch-cluster-pipeline-details"));
exports.UpdateOpensearchClusterPipelineDetails = UpdateOpensearchClusterPipelineDetails.UpdateOpensearchClusterPipelineDetails;
const UpdatePipelineStatusDetails = __importStar(require("./update-pipeline-status-details"));
exports.UpdatePipelineStatusDetails = UpdatePipelineStatusDetails.UpdatePipelineStatusDetails;
const UpgradeOpenSearchClusterDetails = __importStar(require("./upgrade-open-search-cluster-details"));
exports.UpgradeOpenSearchClusterDetails = UpgradeOpenSearchClusterDetails.UpgradeOpenSearchClusterDetails;
const UpgradeType = __importStar(require("./upgrade-type"));
exports.UpgradeType = UpgradeType.UpgradeType;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
