// oci-opensearch/lib/model/maintenance-notification-type.js
"use strict";
/**
 * OpenSearch Service API
 * The OpenSearch service API provides access to OCI Search Service with OpenSearch.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceNotificationType = void 0;
/**
 * Maintenance Notification Type
 **/
var MaintenanceNotificationType;
(function (MaintenanceNotificationType) {
    MaintenanceNotificationType["ScheduledMaintenanceWithReleaseNotes"] = "SCHEDULED_MAINTENANCE_WITH_RELEASE_NOTES";
    MaintenanceNotificationType["ScheduledMaintenanceWithoutReleaseNotes"] = "SCHEDULED_MAINTENANCE_WITHOUT_RELEASE_NOTES";
    MaintenanceNotificationType["RescheduledMaintenance"] = "RESCHEDULED_MAINTENANCE";
    MaintenanceNotificationType["ScheduledMaintenanceReminder"] = "SCHEDULED_MAINTENANCE_REMINDER";
    MaintenanceNotificationType["ScheduledMaintenanceCompleted"] = "SCHEDULED_MAINTENANCE_COMPLETED";
})(MaintenanceNotificationType = exports.MaintenanceNotificationType || (exports.MaintenanceNotificationType = {}));
(function (MaintenanceNotificationType) {
    function getJsonObj(obj) {
        return obj;
    }
    MaintenanceNotificationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MaintenanceNotificationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaintenanceNotificationType = exports.MaintenanceNotificationType || (exports.MaintenanceNotificationType = {}));
