// oci-opensearch/lib/model/maintenance-state.js
"use strict";
/**
 * OpenSearch Service API
 * The OpenSearch service API provides access to OCI Search Service with OpenSearch.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceState = void 0;
/**
 * State of the maintenance activity
 **/
var MaintenanceState;
(function (MaintenanceState) {
    MaintenanceState["Scheduled"] = "SCHEDULED";
    MaintenanceState["Rescheduled"] = "RESCHEDULED";
    MaintenanceState["ToBeScheduled"] = "TO_BE_SCHEDULED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MaintenanceState["UnknownValue"] = "UNKNOWN_VALUE";
})(MaintenanceState = exports.MaintenanceState || (exports.MaintenanceState = {}));
(function (MaintenanceState) {
    function getJsonObj(obj) {
        return obj;
    }
    MaintenanceState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MaintenanceState.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaintenanceState = exports.MaintenanceState || (exports.MaintenanceState = {}));
