// oci-opensearch/lib/model/opensearch-cluster-pipeline.js
"use strict";
/**
 * OpenSearch Service API
 * The OpenSearch service API provides access to OCI Search Service with OpenSearch.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpensearchClusterPipeline = void 0;
const model = __importStar(require("../model"));
var OpensearchClusterPipeline;
(function (OpensearchClusterPipeline) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = OpensearchClusterPipeline.LifecycleState || (OpensearchClusterPipeline.LifecycleState = {}));
    let PipelineMode;
    (function (PipelineMode) {
        PipelineMode["Running"] = "RUNNING";
        PipelineMode["Stopped"] = "STOPPED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PipelineMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(PipelineMode = OpensearchClusterPipeline.PipelineMode || (OpensearchClusterPipeline.PipelineMode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "reverseConnectionEndpoints": obj.reverseConnectionEndpoints
                ? obj.reverseConnectionEndpoints.map(item => {
                    return model.OpensearchPipelineReverseConnectionEndpoint.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    OpensearchClusterPipeline.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "reverseConnectionEndpoints": obj.reverseConnectionEndpoints
                ? obj.reverseConnectionEndpoints.map(item => {
                    return model.OpensearchPipelineReverseConnectionEndpoint.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    OpensearchClusterPipeline.getDeserializedJsonObj = getDeserializedJsonObj;
})(OpensearchClusterPipeline = exports.OpensearchClusterPipeline || (exports.OpensearchClusterPipeline = {}));
