// oci-opensearch/lib/model/search-node-host-type.js
"use strict";
/**
 * OpenSearch Service API
 * The OpenSearch service API provides access to OCI Search Service with OpenSearch.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchNodeHostType = void 0;
/**
 * The instance type for the cluster's search nodes.
 **/
var SearchNodeHostType;
(function (SearchNodeHostType) {
    SearchNodeHostType["Flex"] = "FLEX";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SearchNodeHostType["UnknownValue"] = "UNKNOWN_VALUE";
})(SearchNodeHostType = exports.SearchNodeHostType || (exports.SearchNodeHostType = {}));
(function (SearchNodeHostType) {
    function getJsonObj(obj) {
        return obj;
    }
    SearchNodeHostType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SearchNodeHostType.getDeserializedJsonObj = getDeserializedJsonObj;
})(SearchNodeHostType = exports.SearchNodeHostType || (exports.SearchNodeHostType = {}));
