// oci-opensearch/lib/model/upgrade-type.js
"use strict";
/**
 * OpenSearch Service API
 * The OpenSearch service API provides access to OCI Search Service with OpenSearch.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpgradeType = void 0;
/**
 * Upgrade Open Search cluster upgrade type(major or minor)
 **/
var UpgradeType;
(function (UpgradeType) {
    UpgradeType["Minor"] = "MINOR";
    UpgradeType["Major"] = "MAJOR";
})(UpgradeType = exports.UpgradeType || (exports.UpgradeType = {}));
(function (UpgradeType) {
    function getJsonObj(obj) {
        return obj;
    }
    UpgradeType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    UpgradeType.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpgradeType = exports.UpgradeType || (exports.UpgradeType = {}));
