// oci-fleetappsmanagement/lib/model/announcement-sort-by.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnnouncementSortBy = void 0;
/**
 * The field to sort AnnouncementSummary.
 *
 **/
var AnnouncementSortBy;
(function (AnnouncementSortBy) {
    AnnouncementSortBy["TimeReleased"] = "timeReleased";
    AnnouncementSortBy["Summary"] = "summary";
})(AnnouncementSortBy = exports.AnnouncementSortBy || (exports.AnnouncementSortBy = {}));
(function (AnnouncementSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    AnnouncementSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AnnouncementSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(AnnouncementSortBy = exports.AnnouncementSortBy || (exports.AnnouncementSortBy = {}));
