// oci-fleetappsmanagement/lib/model/catalog-item-sort-by.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CatalogItemSortBy = void 0;
/**
 * The sort by to use. At present, the values are either time created ({@code timeCreated}), name ({@code displayName}) or timeBackfillLastChecked.
 *
 **/
var CatalogItemSortBy;
(function (CatalogItemSortBy) {
    CatalogItemSortBy["TimeCreated"] = "timeCreated";
    CatalogItemSortBy["DisplayName"] = "displayName";
    CatalogItemSortBy["TimeBackfillLastChecked"] = "timeBackfillLastChecked";
})(CatalogItemSortBy = exports.CatalogItemSortBy || (exports.CatalogItemSortBy = {}));
(function (CatalogItemSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    CatalogItemSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CatalogItemSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(CatalogItemSortBy = exports.CatalogItemSortBy || (exports.CatalogItemSortBy = {}));
