// oci-fleetappsmanagement/lib/model/compliance-rule-severity.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComplianceRuleSeverity = void 0;
/**
 * Possible Compliance Rule severities.
 **/
var ComplianceRuleSeverity;
(function (ComplianceRuleSeverity) {
    ComplianceRuleSeverity["Critical"] = "CRITICAL";
    ComplianceRuleSeverity["High"] = "HIGH";
    ComplianceRuleSeverity["Medium"] = "MEDIUM";
    ComplianceRuleSeverity["Low"] = "LOW";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ComplianceRuleSeverity["UnknownValue"] = "UNKNOWN_VALUE";
})(ComplianceRuleSeverity = exports.ComplianceRuleSeverity || (exports.ComplianceRuleSeverity = {}));
(function (ComplianceRuleSeverity) {
    function getJsonObj(obj) {
        return obj;
    }
    ComplianceRuleSeverity.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ComplianceRuleSeverity.getDeserializedJsonObj = getDeserializedJsonObj;
})(ComplianceRuleSeverity = exports.ComplianceRuleSeverity || (exports.ComplianceRuleSeverity = {}));
