// oci-fleetappsmanagement/lib/model/fleet.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Fleet = void 0;
const model = __importStar(require("../model"));
var Fleet;
(function (Fleet) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Fleet.LifecycleState || (Fleet.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "details": obj.details ? model.FleetDetails.getJsonObj(obj.details) : undefined,
            "resourceSelection": obj.resourceSelection
                ? model.ResourceSelection.getJsonObj(obj.resourceSelection)
                : undefined,
            "notificationPreferences": obj.notificationPreferences
                ? obj.notificationPreferences.map(item => {
                    return model.NotificationPreference.getJsonObj(item);
                })
                : undefined,
            "resources": obj.resources
                ? obj.resources.map(item => {
                    return model.AssociatedFleetResourceDetails.getJsonObj(item);
                })
                : undefined,
            "properties": obj.properties
                ? obj.properties.map(item => {
                    return model.AssociatedFleetPropertyDetails.getJsonObj(item);
                })
                : undefined,
            "credentials": obj.credentials
                ? obj.credentials.map(item => {
                    return model.AssociatedFleetCredentialDetails.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    Fleet.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "details": obj.details ? model.FleetDetails.getDeserializedJsonObj(obj.details) : undefined,
            "resourceSelection": obj.resourceSelection
                ? model.ResourceSelection.getDeserializedJsonObj(obj.resourceSelection)
                : undefined,
            "notificationPreferences": obj.notificationPreferences
                ? obj.notificationPreferences.map(item => {
                    return model.NotificationPreference.getDeserializedJsonObj(item);
                })
                : undefined,
            "resources": obj.resources
                ? obj.resources.map(item => {
                    return model.AssociatedFleetResourceDetails.getDeserializedJsonObj(item);
                })
                : undefined,
            "properties": obj.properties
                ? obj.properties.map(item => {
                    return model.AssociatedFleetPropertyDetails.getDeserializedJsonObj(item);
                })
                : undefined,
            "credentials": obj.credentials
                ? obj.credentials.map(item => {
                    return model.AssociatedFleetCredentialDetails.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    Fleet.getDeserializedJsonObj = getDeserializedJsonObj;
})(Fleet = exports.Fleet || (exports.Fleet = {}));
