// oci-fleetappsmanagement/lib/model/job-status.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobStatus = void 0;
/**
 * Possible Job status.
 **/
var JobStatus;
(function (JobStatus) {
    JobStatus["Accepted"] = "ACCEPTED";
    JobStatus["Waiting"] = "WAITING";
    JobStatus["InProgress"] = "IN_PROGRESS";
    JobStatus["Failed"] = "FAILED";
    JobStatus["Succeeded"] = "SUCCEEDED";
    JobStatus["Canceled"] = "CANCELED";
    JobStatus["Skipped"] = "SKIPPED";
    JobStatus["Ignored"] = "IGNORED";
    JobStatus["NotApplicable"] = "NOT_APPLICABLE";
    JobStatus["Aborted"] = "ABORTED";
    JobStatus["TimedOut"] = "TIMED_OUT";
    JobStatus["Paused"] = "PAUSED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    JobStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(JobStatus = exports.JobStatus || (exports.JobStatus = {}));
(function (JobStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    JobStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    JobStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(JobStatus = exports.JobStatus || (exports.JobStatus = {}));
