// oci-fleetappsmanagement/lib/model/operation-type.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateFleet"] = "CREATE_FLEET";
    OperationType["UpdateFleet"] = "UPDATE_FLEET";
    OperationType["DeleteFleet"] = "DELETE_FLEET";
    OperationType["MoveFleet"] = "MOVE_FLEET";
    OperationType["ConfirmTarget"] = "CONFIRM_TARGET";
    OperationType["GenerateCompliance"] = "GENERATE_COMPLIANCE";
    OperationType["RequestTargetDiscovery"] = "REQUEST_TARGET_DISCOVERY";
    OperationType["ValidateResource"] = "VALIDATE_RESOURCE";
    OperationType["CreateCredential"] = "CREATE_CREDENTIAL";
    OperationType["UpdateCredential"] = "UPDATE_CREDENTIAL";
    OperationType["DeleteCredential"] = "DELETE_CREDENTIAL";
    OperationType["CreateSchedule"] = "CREATE_SCHEDULE";
    OperationType["UpdateSchedule"] = "UPDATE_SCHEDULE";
    OperationType["UpdateMaintenanceWindow"] = "UPDATE_MAINTENANCE_WINDOW";
    OperationType["DeleteMaintenanceWindow"] = "DELETE_MAINTENANCE_WINDOW";
    OperationType["CreateFleetResource"] = "CREATE_FLEET_RESOURCE";
    OperationType["UpdateFleetResource"] = "UPDATE_FLEET_RESOURCE";
    OperationType["DeleteFleetResource"] = "DELETE_FLEET_RESOURCE";
    OperationType["CreateFamsOnboarding"] = "CREATE_FAMS_ONBOARDING";
    OperationType["CreateRunbook"] = "CREATE_RUNBOOK";
    OperationType["UpdateRunbook"] = "UPDATE_RUNBOOK";
    OperationType["DeleteRunbook"] = "DELETE_RUNBOOK";
    OperationType["PublishRunbook"] = "PUBLISH_RUNBOOK";
    OperationType["MoveRunbook"] = "MOVE_RUNBOOK";
    OperationType["CreateRunbookVersion"] = "CREATE_RUNBOOK_VERSION";
    OperationType["UpdateRunbookVersion"] = "UPDATE_RUNBOOK_VERSION";
    OperationType["DeleteRunbookVersion"] = "DELETE_RUNBOOK_VERSION";
    OperationType["PublishRunbookVersion"] = "PUBLISH_RUNBOOK_VERSION";
    OperationType["MoveTask"] = "MOVE_TASK";
    OperationType["UpdateTask"] = "UPDATE_TASK";
    OperationType["DeleteTask"] = "DELETE_TASK";
    OperationType["UpdateFamsOnboarding"] = "UPDATE_FAMS_ONBOARDING";
    OperationType["DeleteFamsOnboarding"] = "DELETE_FAMS_ONBOARDING";
    OperationType["CreateCompliancePolicyRule"] = "CREATE_COMPLIANCE_POLICY_RULE";
    OperationType["UpdateCompliancePolicyRule"] = "UPDATE_COMPLIANCE_POLICY_RULE";
    OperationType["DeleteCompliancePolicyRule"] = "DELETE_COMPLIANCE_POLICY_RULE";
    OperationType["UpdatePatch"] = "UPDATE_PATCH";
    OperationType["DeletePatch"] = "DELETE_PATCH";
    OperationType["MovePatch"] = "MOVE_PATCH";
    OperationType["ManageJobExecution"] = "MANAGE_JOB_EXECUTION";
    OperationType["DeletePlatformConfiguration"] = "DELETE_PLATFORM_CONFIGURATION";
    OperationType["UpdatePlatformConfiguration"] = "UPDATE_PLATFORM_CONFIGURATION";
    OperationType["MovePlatformConfiguration"] = "MOVE_PLATFORM_CONFIGURATION";
    OperationType["CreatePlatformConfiguration"] = "CREATE_PLATFORM_CONFIGURATION";
    OperationType["MoveProperty"] = "MOVE_PROPERTY";
    OperationType["CreateCatalogItem"] = "CREATE_CATALOG_ITEM";
    OperationType["UpdateCatalogItem"] = "UPDATE_CATALOG_ITEM";
    OperationType["DeleteCatalogItem"] = "DELETE_CATALOG_ITEM";
    OperationType["MoveCatalogItem"] = "MOVE_CATALOG_ITEM";
    OperationType["CloneCatalogItem"] = "CLONE_CATALOG_ITEM";
    OperationType["CreateProvision"] = "CREATE_PROVISION";
    OperationType["UpdateProvision"] = "UPDATE_PROVISION";
    OperationType["DeleteProvision"] = "DELETE_PROVISION";
    OperationType["MoveProvision"] = "MOVE_PROVISION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
