// oci-fleetappsmanagement/lib/model/patch-severity.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchSeverity = void 0;
/**
 * Possible Patch severities.
 **/
var PatchSeverity;
(function (PatchSeverity) {
    PatchSeverity["Critical"] = "CRITICAL";
    PatchSeverity["High"] = "HIGH";
    PatchSeverity["Medium"] = "MEDIUM";
    PatchSeverity["Low"] = "LOW";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PatchSeverity["UnknownValue"] = "UNKNOWN_VALUE";
})(PatchSeverity = exports.PatchSeverity || (exports.PatchSeverity = {}));
(function (PatchSeverity) {
    function getJsonObj(obj) {
        return obj;
    }
    PatchSeverity.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PatchSeverity.getDeserializedJsonObj = getDeserializedJsonObj;
})(PatchSeverity = exports.PatchSeverity || (exports.PatchSeverity = {}));
