// oci-fleetappsmanagement/lib/model/platform-specific-artifact.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlatformSpecificArtifact = void 0;
const model = __importStar(require("../model"));
var PlatformSpecificArtifact;
(function (PlatformSpecificArtifact) {
    let OsType;
    (function (OsType) {
        OsType["Windows"] = "WINDOWS";
        OsType["Linux"] = "LINUX";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OsType["UnknownValue"] = "UNKNOWN_VALUE";
    })(OsType = PlatformSpecificArtifact.OsType || (PlatformSpecificArtifact.OsType = {}));
    let Architecture;
    (function (Architecture) {
        Architecture["Arm64"] = "ARM_64";
        Architecture["X64"] = "X64";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Architecture["UnknownValue"] = "UNKNOWN_VALUE";
    })(Architecture = PlatformSpecificArtifact.Architecture || (PlatformSpecificArtifact.Architecture = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "content": obj.content ? model.PatchFileContentDetails.getJsonObj(obj.content) : undefined
        });
        return jsonObj;
    }
    PlatformSpecificArtifact.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "content": obj.content
                ? model.PatchFileContentDetails.getDeserializedJsonObj(obj.content)
                : undefined
        });
        return jsonObj;
    }
    PlatformSpecificArtifact.getDeserializedJsonObj = getDeserializedJsonObj;
})(PlatformSpecificArtifact = exports.PlatformSpecificArtifact || (exports.PlatformSpecificArtifact = {}));
