// oci-redis/lib/request/index.js
"use strict";
/**
 * OCI Cache API
 * Use the OCI Cache API to create and manage clusters. A cluster is a memory-based storage solution. For more information, see [OCI Cache](https://docs.oracle.com/iaas/Content/ocicache/home.htm).
 * OpenAPI spec version: 20220315
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListRedisClustersRequest = exports.ListRedisClusterNodesRequest = exports.ListAttachedOciCacheUsersRequest = exports.ListOciCacheUsersRequest = exports.ListAttachedRedisClustersRequest = void 0;
const ListAttachedRedisClustersRequest = __importStar(require("./list-attached-redis-clusters-request"));
exports.ListAttachedRedisClustersRequest = ListAttachedRedisClustersRequest.ListAttachedRedisClustersRequest;
const ListOciCacheUsersRequest = __importStar(require("./list-oci-cache-users-request"));
exports.ListOciCacheUsersRequest = ListOciCacheUsersRequest.ListOciCacheUsersRequest;
const ListAttachedOciCacheUsersRequest = __importStar(require("./list-attached-oci-cache-users-request"));
exports.ListAttachedOciCacheUsersRequest = ListAttachedOciCacheUsersRequest.ListAttachedOciCacheUsersRequest;
const ListRedisClusterNodesRequest = __importStar(require("./list-redis-cluster-nodes-request"));
exports.ListRedisClusterNodesRequest = ListRedisClusterNodesRequest.ListRedisClusterNodesRequest;
const ListRedisClustersRequest = __importStar(require("./list-redis-clusters-request"));
exports.ListRedisClustersRequest = ListRedisClustersRequest.ListRedisClustersRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
