// oci-vnmonitoring/lib/model/virtual-circuit-ip-mtu.js
"use strict";
/**
 * Network Monitoring API
 * Use the Network Monitoring API to troubleshoot routing and security issues for resources such as virtual cloud networks (VCNs) and compute instances. For more information, see the console
documentation for the [Network Path Analyzer](https://docs.oracle.com/iaas/Content/Network/Concepts/path_analyzer.htm) tool.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualCircuitIpMtu = void 0;
/**
 * The layer 3 IP MTU to use with this virtual circuit.
 **/
var VirtualCircuitIpMtu;
(function (VirtualCircuitIpMtu) {
    VirtualCircuitIpMtu["Mtu1500"] = "MTU_1500";
    VirtualCircuitIpMtu["Mtu9000"] = "MTU_9000";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    VirtualCircuitIpMtu["UnknownValue"] = "UNKNOWN_VALUE";
})(VirtualCircuitIpMtu = exports.VirtualCircuitIpMtu || (exports.VirtualCircuitIpMtu = {}));
(function (VirtualCircuitIpMtu) {
    function getJsonObj(obj) {
        return obj;
    }
    VirtualCircuitIpMtu.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    VirtualCircuitIpMtu.getDeserializedJsonObj = getDeserializedJsonObj;
})(VirtualCircuitIpMtu = exports.VirtualCircuitIpMtu || (exports.VirtualCircuitIpMtu = {}));
