// oci-generativeaiagent/lib/request/index.js
"use strict";
/**
 * Generative AI Agents Management API
 * OCI Generative AI Agents is a fully managed service that combines the power of large language models (LLMs) with an intelligent retrieval system to create contextually relevant answers by searching your knowledge base, making your AI applications smart and efficient.

OCI Generative AI Agents supports several ways to onboard your data and then allows you and your customers to interact with your data using a chat interface or API.

Use the Generative AI Agents API to create and manage agents, knowledge bases, data sources, endpoints, data ingestion jobs, and work requests.

For creating and managing client chat sessions see the {@link #eNGenerative-ai-agents-clientLatest(ENGenerative-ai-agents-clientLatestRequest) eNGenerative-ai-agents-clientLatest}.

To learn more about the service, see the [Generative AI Agents documentation](https://docs.oracle.com/iaas/Content/generative-ai-agents/home.htm).

 * OpenAPI spec version: 20240531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListToolsRequest = exports.ListKnowledgeBasesRequest = exports.ListDataSourcesRequest = exports.ListDataIngestionJobsRequest = exports.ListAgentsRequest = exports.ListAgentEndpointsRequest = void 0;
const ListAgentEndpointsRequest = __importStar(require("./list-agent-endpoints-request"));
exports.ListAgentEndpointsRequest = ListAgentEndpointsRequest.ListAgentEndpointsRequest;
const ListAgentsRequest = __importStar(require("./list-agents-request"));
exports.ListAgentsRequest = ListAgentsRequest.ListAgentsRequest;
const ListDataIngestionJobsRequest = __importStar(require("./list-data-ingestion-jobs-request"));
exports.ListDataIngestionJobsRequest = ListDataIngestionJobsRequest.ListDataIngestionJobsRequest;
const ListDataSourcesRequest = __importStar(require("./list-data-sources-request"));
exports.ListDataSourcesRequest = ListDataSourcesRequest.ListDataSourcesRequest;
const ListKnowledgeBasesRequest = __importStar(require("./list-knowledge-bases-request"));
exports.ListKnowledgeBasesRequest = ListKnowledgeBasesRequest.ListKnowledgeBasesRequest;
const ListToolsRequest = __importStar(require("./list-tools-request"));
exports.ListToolsRequest = ListToolsRequest.ListToolsRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
