// oci-wlms/lib/model/index.js
"use strict";
/**
 * WebLogic Management Service API
 * WebLogic Management Service is an OCI service that enables a unified view and management of WebLogic domains
in Oracle Cloud Infrastructure. Features include on-demand patching of WebLogic domains, rollback of the
last applied patch, discovery and management of WebLogic instances on a compute host.

 * OpenAPI spec version: 20241101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateManagedInstanceDetails = exports.UpdateManagedInstanceConfigurationDetails = exports.UpdateConfigurationDetails = exports.StopWlsDomainDetails = exports.SetRestartOrderServerSummary = exports.SetRestartOrderServerCollection = exports.SetRestartOrderDetails = exports.ServerToRestore = exports.ServerSummary = exports.ServerControlMode = exports.ServerCollection = exports.Server = exports.ScanResultSummary = exports.ScanResultCollection = exports.RestoreWlsDomainDetails = exports.RestartWlsDomainDetails = exports.ResourceInventory = exports.RequiredPolicySummary = exports.RequiredPolicyCollection = exports.PatchReadinessStatus = exports.Patch = exports.OperationType = exports.OperationStatus = exports.MiddlewareBinaryBackupContent = exports.MiddlewareBackupPatch = exports.ManagedInstanceSummary = exports.ManagedInstanceConfiguration = exports.ManagedInstanceCollection = exports.ManagedInstance = exports.InstalledPatchSummary = exports.InstalledPatchCollection = exports.InstallLatestPatchesOnWlsDomainDetails = exports.DefaultError = exports.CreateAgreementRecordDetails = exports.Configuration = exports.ChangeWlsDomainCompartmentDetails = exports.BackupType = exports.BackupToRestore = exports.BackupSummary = exports.BackupContentType = exports.BackupContent = exports.BackupCollection = exports.Backup = exports.ApplicablePatchSummary = exports.ApplicablePatchCollection = exports.AgreementRecordSummary = exports.AgreementRecordCollection = exports.AgreementRecord = exports.Agreement = exports.ActionType = void 0;
exports.BinaryBackupContent = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.WlsDomainSummary = exports.WlsDomainCredentialSummary = exports.WlsDomainCredentialCollection = exports.WlsDomainCredential = exports.WlsDomainConfiguration = exports.WlsDomainCollection = exports.WlsDomain = exports.UpdateWlsDomainDetails = exports.UpdateWlsDomainCredentialDetails = exports.UpdateWlsDomainConfigurationDetails = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const Agreement = __importStar(require("./agreement"));
exports.Agreement = Agreement.Agreement;
const AgreementRecord = __importStar(require("./agreement-record"));
exports.AgreementRecord = AgreementRecord.AgreementRecord;
const AgreementRecordCollection = __importStar(require("./agreement-record-collection"));
exports.AgreementRecordCollection = AgreementRecordCollection.AgreementRecordCollection;
const AgreementRecordSummary = __importStar(require("./agreement-record-summary"));
exports.AgreementRecordSummary = AgreementRecordSummary.AgreementRecordSummary;
const ApplicablePatchCollection = __importStar(require("./applicable-patch-collection"));
exports.ApplicablePatchCollection = ApplicablePatchCollection.ApplicablePatchCollection;
const ApplicablePatchSummary = __importStar(require("./applicable-patch-summary"));
exports.ApplicablePatchSummary = ApplicablePatchSummary.ApplicablePatchSummary;
const Backup = __importStar(require("./backup"));
exports.Backup = Backup.Backup;
const BackupCollection = __importStar(require("./backup-collection"));
exports.BackupCollection = BackupCollection.BackupCollection;
const BackupContent = __importStar(require("./backup-content"));
exports.BackupContent = BackupContent.BackupContent;
const BackupContentType = __importStar(require("./backup-content-type"));
exports.BackupContentType = BackupContentType.BackupContentType;
const BackupSummary = __importStar(require("./backup-summary"));
exports.BackupSummary = BackupSummary.BackupSummary;
const BackupToRestore = __importStar(require("./backup-to-restore"));
exports.BackupToRestore = BackupToRestore.BackupToRestore;
const BackupType = __importStar(require("./backup-type"));
exports.BackupType = BackupType.BackupType;
const ChangeWlsDomainCompartmentDetails = __importStar(require("./change-wls-domain-compartment-details"));
exports.ChangeWlsDomainCompartmentDetails = ChangeWlsDomainCompartmentDetails.ChangeWlsDomainCompartmentDetails;
const Configuration = __importStar(require("./configuration"));
exports.Configuration = Configuration.Configuration;
const CreateAgreementRecordDetails = __importStar(require("./create-agreement-record-details"));
exports.CreateAgreementRecordDetails = CreateAgreementRecordDetails.CreateAgreementRecordDetails;
const DefaultError = __importStar(require("./default-error"));
exports.DefaultError = DefaultError.DefaultError;
const InstallLatestPatchesOnWlsDomainDetails = __importStar(require("./install-latest-patches-on-wls-domain-details"));
exports.InstallLatestPatchesOnWlsDomainDetails = InstallLatestPatchesOnWlsDomainDetails.InstallLatestPatchesOnWlsDomainDetails;
const InstalledPatchCollection = __importStar(require("./installed-patch-collection"));
exports.InstalledPatchCollection = InstalledPatchCollection.InstalledPatchCollection;
const InstalledPatchSummary = __importStar(require("./installed-patch-summary"));
exports.InstalledPatchSummary = InstalledPatchSummary.InstalledPatchSummary;
const ManagedInstance = __importStar(require("./managed-instance"));
exports.ManagedInstance = ManagedInstance.ManagedInstance;
const ManagedInstanceCollection = __importStar(require("./managed-instance-collection"));
exports.ManagedInstanceCollection = ManagedInstanceCollection.ManagedInstanceCollection;
const ManagedInstanceConfiguration = __importStar(require("./managed-instance-configuration"));
exports.ManagedInstanceConfiguration = ManagedInstanceConfiguration.ManagedInstanceConfiguration;
const ManagedInstanceSummary = __importStar(require("./managed-instance-summary"));
exports.ManagedInstanceSummary = ManagedInstanceSummary.ManagedInstanceSummary;
const MiddlewareBackupPatch = __importStar(require("./middleware-backup-patch"));
exports.MiddlewareBackupPatch = MiddlewareBackupPatch.MiddlewareBackupPatch;
const MiddlewareBinaryBackupContent = __importStar(require("./middleware-binary-backup-content"));
exports.MiddlewareBinaryBackupContent = MiddlewareBinaryBackupContent.MiddlewareBinaryBackupContent;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const Patch = __importStar(require("./patch"));
exports.Patch = Patch.Patch;
const PatchReadinessStatus = __importStar(require("./patch-readiness-status"));
exports.PatchReadinessStatus = PatchReadinessStatus.PatchReadinessStatus;
const RequiredPolicyCollection = __importStar(require("./required-policy-collection"));
exports.RequiredPolicyCollection = RequiredPolicyCollection.RequiredPolicyCollection;
const RequiredPolicySummary = __importStar(require("./required-policy-summary"));
exports.RequiredPolicySummary = RequiredPolicySummary.RequiredPolicySummary;
const ResourceInventory = __importStar(require("./resource-inventory"));
exports.ResourceInventory = ResourceInventory.ResourceInventory;
const RestartWlsDomainDetails = __importStar(require("./restart-wls-domain-details"));
exports.RestartWlsDomainDetails = RestartWlsDomainDetails.RestartWlsDomainDetails;
const RestoreWlsDomainDetails = __importStar(require("./restore-wls-domain-details"));
exports.RestoreWlsDomainDetails = RestoreWlsDomainDetails.RestoreWlsDomainDetails;
const ScanResultCollection = __importStar(require("./scan-result-collection"));
exports.ScanResultCollection = ScanResultCollection.ScanResultCollection;
const ScanResultSummary = __importStar(require("./scan-result-summary"));
exports.ScanResultSummary = ScanResultSummary.ScanResultSummary;
const Server = __importStar(require("./server"));
exports.Server = Server.Server;
const ServerCollection = __importStar(require("./server-collection"));
exports.ServerCollection = ServerCollection.ServerCollection;
const ServerControlMode = __importStar(require("./server-control-mode"));
exports.ServerControlMode = ServerControlMode.ServerControlMode;
const ServerSummary = __importStar(require("./server-summary"));
exports.ServerSummary = ServerSummary.ServerSummary;
const ServerToRestore = __importStar(require("./server-to-restore"));
exports.ServerToRestore = ServerToRestore.ServerToRestore;
const SetRestartOrderDetails = __importStar(require("./set-restart-order-details"));
exports.SetRestartOrderDetails = SetRestartOrderDetails.SetRestartOrderDetails;
const SetRestartOrderServerCollection = __importStar(require("./set-restart-order-server-collection"));
exports.SetRestartOrderServerCollection = SetRestartOrderServerCollection.SetRestartOrderServerCollection;
const SetRestartOrderServerSummary = __importStar(require("./set-restart-order-server-summary"));
exports.SetRestartOrderServerSummary = SetRestartOrderServerSummary.SetRestartOrderServerSummary;
const StopWlsDomainDetails = __importStar(require("./stop-wls-domain-details"));
exports.StopWlsDomainDetails = StopWlsDomainDetails.StopWlsDomainDetails;
const UpdateConfigurationDetails = __importStar(require("./update-configuration-details"));
exports.UpdateConfigurationDetails = UpdateConfigurationDetails.UpdateConfigurationDetails;
const UpdateManagedInstanceConfigurationDetails = __importStar(require("./update-managed-instance-configuration-details"));
exports.UpdateManagedInstanceConfigurationDetails = UpdateManagedInstanceConfigurationDetails.UpdateManagedInstanceConfigurationDetails;
const UpdateManagedInstanceDetails = __importStar(require("./update-managed-instance-details"));
exports.UpdateManagedInstanceDetails = UpdateManagedInstanceDetails.UpdateManagedInstanceDetails;
const UpdateWlsDomainConfigurationDetails = __importStar(require("./update-wls-domain-configuration-details"));
exports.UpdateWlsDomainConfigurationDetails = UpdateWlsDomainConfigurationDetails.UpdateWlsDomainConfigurationDetails;
const UpdateWlsDomainCredentialDetails = __importStar(require("./update-wls-domain-credential-details"));
exports.UpdateWlsDomainCredentialDetails = UpdateWlsDomainCredentialDetails.UpdateWlsDomainCredentialDetails;
const UpdateWlsDomainDetails = __importStar(require("./update-wls-domain-details"));
exports.UpdateWlsDomainDetails = UpdateWlsDomainDetails.UpdateWlsDomainDetails;
const WlsDomain = __importStar(require("./wls-domain"));
exports.WlsDomain = WlsDomain.WlsDomain;
const WlsDomainCollection = __importStar(require("./wls-domain-collection"));
exports.WlsDomainCollection = WlsDomainCollection.WlsDomainCollection;
const WlsDomainConfiguration = __importStar(require("./wls-domain-configuration"));
exports.WlsDomainConfiguration = WlsDomainConfiguration.WlsDomainConfiguration;
const WlsDomainCredential = __importStar(require("./wls-domain-credential"));
exports.WlsDomainCredential = WlsDomainCredential.WlsDomainCredential;
const WlsDomainCredentialCollection = __importStar(require("./wls-domain-credential-collection"));
exports.WlsDomainCredentialCollection = WlsDomainCredentialCollection.WlsDomainCredentialCollection;
const WlsDomainCredentialSummary = __importStar(require("./wls-domain-credential-summary"));
exports.WlsDomainCredentialSummary = WlsDomainCredentialSummary.WlsDomainCredentialSummary;
const WlsDomainSummary = __importStar(require("./wls-domain-summary"));
exports.WlsDomainSummary = WlsDomainSummary.WlsDomainSummary;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const BinaryBackupContent = __importStar(require("./binary-backup-content"));
exports.BinaryBackupContent = BinaryBackupContent.BinaryBackupContent;
