// oci-globallydistributeddatabase/lib/model/create-dedicated-sharded-database.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage distributed databases.
 * OpenAPI spec version: 20230301
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDedicatedShardedDatabase = void 0;
const model = __importStar(require("../model"));
var CreateDedicatedShardedDatabase;
(function (CreateDedicatedShardedDatabase) {
    let DbWorkload;
    (function (DbWorkload) {
        DbWorkload["Oltp"] = "OLTP";
        DbWorkload["Dw"] = "DW";
    })(DbWorkload = CreateDedicatedShardedDatabase.DbWorkload || (CreateDedicatedShardedDatabase.DbWorkload = {}));
    let ShardingMethod;
    (function (ShardingMethod) {
        ShardingMethod["User"] = "USER";
        ShardingMethod["System"] = "SYSTEM";
    })(ShardingMethod = CreateDedicatedShardedDatabase.ShardingMethod || (CreateDedicatedShardedDatabase.ShardingMethod = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.CreateShardedDatabaseDetails.getJsonObj(obj))), {
            "shardDetails": obj.shardDetails
                ? obj.shardDetails.map(item => {
                    return model.CreateDedicatedShardDetail.getJsonObj(item);
                })
                : undefined,
            "catalogDetails": obj.catalogDetails
                ? obj.catalogDetails.map(item => {
                    return model.CreateDedicatedCatalogDetail.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    CreateDedicatedShardedDatabase.getJsonObj = getJsonObj;
    CreateDedicatedShardedDatabase.dbDeploymentType = "DEDICATED";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.CreateShardedDatabaseDetails.getDeserializedJsonObj(obj))), {
            "shardDetails": obj.shardDetails
                ? obj.shardDetails.map(item => {
                    return model.CreateDedicatedShardDetail.getDeserializedJsonObj(item);
                })
                : undefined,
            "catalogDetails": obj.catalogDetails
                ? obj.catalogDetails.map(item => {
                    return model.CreateDedicatedCatalogDetail.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    CreateDedicatedShardedDatabase.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateDedicatedShardedDatabase = exports.CreateDedicatedShardedDatabase || (exports.CreateDedicatedShardedDatabase = {}));
