// oci-globallydistributeddatabase/lib/model/dedicated-sharded-database.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage distributed databases.
 * OpenAPI spec version: 20230301
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DedicatedShardedDatabase = void 0;
const model = __importStar(require("../model"));
var DedicatedShardedDatabase;
(function (DedicatedShardedDatabase) {
    let ReplicationMethod;
    (function (ReplicationMethod) {
        ReplicationMethod["Raft"] = "RAFT";
        ReplicationMethod["Dg"] = "DG";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ReplicationMethod["UnknownValue"] = "UNKNOWN_VALUE";
    })(ReplicationMethod = DedicatedShardedDatabase.ReplicationMethod || (DedicatedShardedDatabase.ReplicationMethod = {}));
    let DbWorkload;
    (function (DbWorkload) {
        DbWorkload["Oltp"] = "OLTP";
        DbWorkload["Dw"] = "DW";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DbWorkload["UnknownValue"] = "UNKNOWN_VALUE";
    })(DbWorkload = DedicatedShardedDatabase.DbWorkload || (DedicatedShardedDatabase.DbWorkload = {}));
    let ShardingMethod;
    (function (ShardingMethod) {
        ShardingMethod["User"] = "USER";
        ShardingMethod["System"] = "SYSTEM";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ShardingMethod["UnknownValue"] = "UNKNOWN_VALUE";
    })(ShardingMethod = DedicatedShardedDatabase.ShardingMethod || (DedicatedShardedDatabase.ShardingMethod = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.ShardedDatabase.getJsonObj(obj))), {
            "connectionStrings": obj.connectionStrings
                ? model.ConnectionString.getJsonObj(obj.connectionStrings)
                : undefined,
            "gsms": obj.gsms
                ? obj.gsms.map(item => {
                    return model.GsmDetails.getJsonObj(item);
                })
                : undefined,
            "shardDetails": obj.shardDetails
                ? obj.shardDetails.map(item => {
                    return model.DedicatedShardDetails.getJsonObj(item);
                })
                : undefined,
            "catalogDetails": obj.catalogDetails
                ? obj.catalogDetails.map(item => {
                    return model.DedicatedCatalogDetails.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DedicatedShardedDatabase.getJsonObj = getJsonObj;
    DedicatedShardedDatabase.dbDeploymentType = "DEDICATED";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.ShardedDatabase.getDeserializedJsonObj(obj))), {
            "connectionStrings": obj.connectionStrings
                ? model.ConnectionString.getDeserializedJsonObj(obj.connectionStrings)
                : undefined,
            "gsms": obj.gsms
                ? obj.gsms.map(item => {
                    return model.GsmDetails.getDeserializedJsonObj(item);
                })
                : undefined,
            "shardDetails": obj.shardDetails
                ? obj.shardDetails.map(item => {
                    return model.DedicatedShardDetails.getDeserializedJsonObj(item);
                })
                : undefined,
            "catalogDetails": obj.catalogDetails
                ? obj.catalogDetails.map(item => {
                    return model.DedicatedCatalogDetails.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DedicatedShardedDatabase.getDeserializedJsonObj = getDeserializedJsonObj;
})(DedicatedShardedDatabase = exports.DedicatedShardedDatabase || (exports.DedicatedShardedDatabase = {}));
