// oci-globallydistributeddatabase/lib/model/gsm-details.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage distributed databases.
 * OpenAPI spec version: 20230301
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GsmDetails = void 0;
var GsmDetails;
(function (GsmDetails) {
    let Status;
    (function (Status) {
        Status["Failed"] = "FAILED";
        Status["Deleting"] = "DELETING";
        Status["Deleted"] = "DELETED";
        Status["Updating"] = "UPDATING";
        Status["Creating"] = "CREATING";
        Status["Created"] = "CREATED";
        Status["ReadyForConfiguration"] = "READY_FOR_CONFIGURATION";
        Status["Configured"] = "CONFIGURED";
        Status["NeedsAttention"] = "NEEDS_ATTENTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = GsmDetails.Status || (GsmDetails.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GsmDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GsmDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(GsmDetails = exports.GsmDetails || (exports.GsmDetails = {}));
