// oci-globallydistributeddatabase/lib/model/index.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage distributed databases.
 * OpenAPI spec version: 20230301
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrevalidateCreatePayload = exports.PatchRemoveInstruction = exports.PatchMergeInstruction = exports.PatchInsertInstruction = exports.DedicatedShardedDatabaseSummary = exports.DedicatedShardedDatabase = exports.CreateDedicatedShardedDatabase = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UploadSignedCertificateAndGenerateWalletDetails = exports.UpdateShardedDatabaseDetails = exports.UpdatePrivateEndpointDetails = exports.SortOrder = exports.ShardedDatabaseSummary = exports.ShardedDatabaseCollection = exports.ShardedDatabase = exports.ShardableCloudAutonomousVmClusters = exports.ShardableCloudAutonomousVmClusterSummary = exports.PrivateEndpointSummary = exports.PrivateEndpointCollection = exports.PrivateEndpoint = exports.PrevalidateShardedDatabaseResult = exports.PrevalidateShardedDatabaseDetails = exports.PrevalidatePayload = exports.PatchShardedDatabaseDetails = exports.PatchInstruction = exports.OperationType = exports.OperationStatus = exports.GsmDetails = exports.GenerateWalletDetails = exports.FetchShardableCloudAutonomousVmClustersDetails = exports.DedicatedShardOrCatalogEncryptionKeyDetails = exports.DedicatedShardDetails = exports.DedicatedCatalogDetails = exports.CreateShardedDatabaseDetails = exports.CreatePrivateEndpointDetails = exports.CreateDedicatedShardDetail = exports.CreateDedicatedCatalogDetail = exports.ConnectionString = exports.ConfigureShardedDatabaseGsmsDetails = exports.ChangeShardedDatabaseCompartmentDetails = exports.ChangePrivateEndpointCompartmentDetails = exports.ActionType = void 0;
exports.PrevalidatePatchPayload = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangePrivateEndpointCompartmentDetails = __importStar(require("./change-private-endpoint-compartment-details"));
exports.ChangePrivateEndpointCompartmentDetails = ChangePrivateEndpointCompartmentDetails.ChangePrivateEndpointCompartmentDetails;
const ChangeShardedDatabaseCompartmentDetails = __importStar(require("./change-sharded-database-compartment-details"));
exports.ChangeShardedDatabaseCompartmentDetails = ChangeShardedDatabaseCompartmentDetails.ChangeShardedDatabaseCompartmentDetails;
const ConfigureShardedDatabaseGsmsDetails = __importStar(require("./configure-sharded-database-gsms-details"));
exports.ConfigureShardedDatabaseGsmsDetails = ConfigureShardedDatabaseGsmsDetails.ConfigureShardedDatabaseGsmsDetails;
const ConnectionString = __importStar(require("./connection-string"));
exports.ConnectionString = ConnectionString.ConnectionString;
const CreateDedicatedCatalogDetail = __importStar(require("./create-dedicated-catalog-detail"));
exports.CreateDedicatedCatalogDetail = CreateDedicatedCatalogDetail.CreateDedicatedCatalogDetail;
const CreateDedicatedShardDetail = __importStar(require("./create-dedicated-shard-detail"));
exports.CreateDedicatedShardDetail = CreateDedicatedShardDetail.CreateDedicatedShardDetail;
const CreatePrivateEndpointDetails = __importStar(require("./create-private-endpoint-details"));
exports.CreatePrivateEndpointDetails = CreatePrivateEndpointDetails.CreatePrivateEndpointDetails;
const CreateShardedDatabaseDetails = __importStar(require("./create-sharded-database-details"));
exports.CreateShardedDatabaseDetails = CreateShardedDatabaseDetails.CreateShardedDatabaseDetails;
const DedicatedCatalogDetails = __importStar(require("./dedicated-catalog-details"));
exports.DedicatedCatalogDetails = DedicatedCatalogDetails.DedicatedCatalogDetails;
const DedicatedShardDetails = __importStar(require("./dedicated-shard-details"));
exports.DedicatedShardDetails = DedicatedShardDetails.DedicatedShardDetails;
const DedicatedShardOrCatalogEncryptionKeyDetails = __importStar(require("./dedicated-shard-or-catalog-encryption-key-details"));
exports.DedicatedShardOrCatalogEncryptionKeyDetails = DedicatedShardOrCatalogEncryptionKeyDetails.DedicatedShardOrCatalogEncryptionKeyDetails;
const FetchShardableCloudAutonomousVmClustersDetails = __importStar(require("./fetch-shardable-cloud-autonomous-vm-clusters-details"));
exports.FetchShardableCloudAutonomousVmClustersDetails = FetchShardableCloudAutonomousVmClustersDetails.FetchShardableCloudAutonomousVmClustersDetails;
const GenerateWalletDetails = __importStar(require("./generate-wallet-details"));
exports.GenerateWalletDetails = GenerateWalletDetails.GenerateWalletDetails;
const GsmDetails = __importStar(require("./gsm-details"));
exports.GsmDetails = GsmDetails.GsmDetails;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PatchInstruction = __importStar(require("./patch-instruction"));
exports.PatchInstruction = PatchInstruction.PatchInstruction;
const PatchShardedDatabaseDetails = __importStar(require("./patch-sharded-database-details"));
exports.PatchShardedDatabaseDetails = PatchShardedDatabaseDetails.PatchShardedDatabaseDetails;
const PrevalidatePayload = __importStar(require("./prevalidate-payload"));
exports.PrevalidatePayload = PrevalidatePayload.PrevalidatePayload;
const PrevalidateShardedDatabaseDetails = __importStar(require("./prevalidate-sharded-database-details"));
exports.PrevalidateShardedDatabaseDetails = PrevalidateShardedDatabaseDetails.PrevalidateShardedDatabaseDetails;
const PrevalidateShardedDatabaseResult = __importStar(require("./prevalidate-sharded-database-result"));
exports.PrevalidateShardedDatabaseResult = PrevalidateShardedDatabaseResult.PrevalidateShardedDatabaseResult;
const PrivateEndpoint = __importStar(require("./private-endpoint"));
exports.PrivateEndpoint = PrivateEndpoint.PrivateEndpoint;
const PrivateEndpointCollection = __importStar(require("./private-endpoint-collection"));
exports.PrivateEndpointCollection = PrivateEndpointCollection.PrivateEndpointCollection;
const PrivateEndpointSummary = __importStar(require("./private-endpoint-summary"));
exports.PrivateEndpointSummary = PrivateEndpointSummary.PrivateEndpointSummary;
const ShardableCloudAutonomousVmClusterSummary = __importStar(require("./shardable-cloud-autonomous-vm-cluster-summary"));
exports.ShardableCloudAutonomousVmClusterSummary = ShardableCloudAutonomousVmClusterSummary.ShardableCloudAutonomousVmClusterSummary;
const ShardableCloudAutonomousVmClusters = __importStar(require("./shardable-cloud-autonomous-vm-clusters"));
exports.ShardableCloudAutonomousVmClusters = ShardableCloudAutonomousVmClusters.ShardableCloudAutonomousVmClusters;
const ShardedDatabase = __importStar(require("./sharded-database"));
exports.ShardedDatabase = ShardedDatabase.ShardedDatabase;
const ShardedDatabaseCollection = __importStar(require("./sharded-database-collection"));
exports.ShardedDatabaseCollection = ShardedDatabaseCollection.ShardedDatabaseCollection;
const ShardedDatabaseSummary = __importStar(require("./sharded-database-summary"));
exports.ShardedDatabaseSummary = ShardedDatabaseSummary.ShardedDatabaseSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdatePrivateEndpointDetails = __importStar(require("./update-private-endpoint-details"));
exports.UpdatePrivateEndpointDetails = UpdatePrivateEndpointDetails.UpdatePrivateEndpointDetails;
const UpdateShardedDatabaseDetails = __importStar(require("./update-sharded-database-details"));
exports.UpdateShardedDatabaseDetails = UpdateShardedDatabaseDetails.UpdateShardedDatabaseDetails;
const UploadSignedCertificateAndGenerateWalletDetails = __importStar(require("./upload-signed-certificate-and-generate-wallet-details"));
exports.UploadSignedCertificateAndGenerateWalletDetails = UploadSignedCertificateAndGenerateWalletDetails.UploadSignedCertificateAndGenerateWalletDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const CreateDedicatedShardedDatabase = __importStar(require("./create-dedicated-sharded-database"));
exports.CreateDedicatedShardedDatabase = CreateDedicatedShardedDatabase.CreateDedicatedShardedDatabase;
const DedicatedShardedDatabase = __importStar(require("./dedicated-sharded-database"));
exports.DedicatedShardedDatabase = DedicatedShardedDatabase.DedicatedShardedDatabase;
const DedicatedShardedDatabaseSummary = __importStar(require("./dedicated-sharded-database-summary"));
exports.DedicatedShardedDatabaseSummary = DedicatedShardedDatabaseSummary.DedicatedShardedDatabaseSummary;
const PatchInsertInstruction = __importStar(require("./patch-insert-instruction"));
exports.PatchInsertInstruction = PatchInsertInstruction.PatchInsertInstruction;
const PatchMergeInstruction = __importStar(require("./patch-merge-instruction"));
exports.PatchMergeInstruction = PatchMergeInstruction.PatchMergeInstruction;
const PatchRemoveInstruction = __importStar(require("./patch-remove-instruction"));
exports.PatchRemoveInstruction = PatchRemoveInstruction.PatchRemoveInstruction;
const PrevalidateCreatePayload = __importStar(require("./prevalidate-create-payload"));
exports.PrevalidateCreatePayload = PrevalidateCreatePayload.PrevalidateCreatePayload;
const PrevalidatePatchPayload = __importStar(require("./prevalidate-patch-payload"));
exports.PrevalidatePatchPayload = PrevalidatePatchPayload.PrevalidatePatchPayload;
