// oci-globallydistributeddatabase/lib/model/shardable-cloud-autonomous-vm-cluster-summary.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage distributed databases.
 * OpenAPI spec version: 20230301
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShardableCloudAutonomousVmClusterSummary = void 0;
var ShardableCloudAutonomousVmClusterSummary;
(function (ShardableCloudAutonomousVmClusterSummary) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Unavailable"] = "UNAVAILABLE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ShardableCloudAutonomousVmClusterSummary.LifecycleState || (ShardableCloudAutonomousVmClusterSummary.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ShardableCloudAutonomousVmClusterSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ShardableCloudAutonomousVmClusterSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(ShardableCloudAutonomousVmClusterSummary = exports.ShardableCloudAutonomousVmClusterSummary || (exports.ShardableCloudAutonomousVmClusterSummary = {}));
