// oci-databasemanagement/lib/model/attention-log-summary.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttentionLogSummary = void 0;
var AttentionLogSummary;
(function (AttentionLogSummary) {
    let MessageUrgency;
    (function (MessageUrgency) {
        MessageUrgency["Immediate"] = "IMMEDIATE";
        MessageUrgency["Soon"] = "SOON";
        MessageUrgency["Deferrable"] = "DEFERRABLE";
        MessageUrgency["Info"] = "INFO";
        MessageUrgency["Other"] = "OTHER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MessageUrgency["UnknownValue"] = "UNKNOWN_VALUE";
    })(MessageUrgency = AttentionLogSummary.MessageUrgency || (AttentionLogSummary.MessageUrgency = {}));
    let MessageType;
    (function (MessageType) {
        MessageType["Unknown"] = "UNKNOWN";
        MessageType["IncidentError"] = "INCIDENT_ERROR";
        MessageType["Error"] = "ERROR";
        MessageType["Warning"] = "WARNING";
        MessageType["Notification"] = "NOTIFICATION";
        MessageType["Trace"] = "TRACE";
        MessageType["Other"] = "OTHER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MessageType["UnknownValue"] = "UNKNOWN_VALUE";
    })(MessageType = AttentionLogSummary.MessageType || (AttentionLogSummary.MessageType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttentionLogSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttentionLogSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(AttentionLogSummary = exports.AttentionLogSummary || (exports.AttentionLogSummary = {}));
