// oci-databasemanagement/lib/model/automatic-capture-filter.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutomaticCaptureFilter = void 0;
var AutomaticCaptureFilter;
(function (AutomaticCaptureFilter) {
    let Name;
    (function (Name) {
        Name["AutoCaptureSqlText"] = "AUTO_CAPTURE_SQL_TEXT";
        Name["AutoCaptureParsingSchemaName"] = "AUTO_CAPTURE_PARSING_SCHEMA_NAME";
        Name["AutoCaptureModule"] = "AUTO_CAPTURE_MODULE";
        Name["AutoCaptureAction"] = "AUTO_CAPTURE_ACTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Name["UnknownValue"] = "UNKNOWN_VALUE";
    })(Name = AutomaticCaptureFilter.Name || (AutomaticCaptureFilter.Name = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AutomaticCaptureFilter.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AutomaticCaptureFilter.getDeserializedJsonObj = getDeserializedJsonObj;
})(AutomaticCaptureFilter = exports.AutomaticCaptureFilter || (exports.AutomaticCaptureFilter = {}));
