// oci-databasemanagement/lib/model/database-password-credential-details.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabasePasswordCredentialDetails = void 0;
const model = __importStar(require("../model"));
var DatabasePasswordCredentialDetails;
(function (DatabasePasswordCredentialDetails) {
    let Role;
    (function (Role) {
        Role["Normal"] = "NORMAL";
        Role["Sysdba"] = "SYSDBA";
        Role["Sysdg"] = "SYSDG";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Role["UnknownValue"] = "UNKNOWN_VALUE";
    })(Role = DatabasePasswordCredentialDetails.Role || (DatabasePasswordCredentialDetails.Role = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.DatabaseCredentialDetails.getJsonObj(obj))), {});
        return jsonObj;
    }
    DatabasePasswordCredentialDetails.getJsonObj = getJsonObj;
    DatabasePasswordCredentialDetails.credentialType = "PASSWORD";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.DatabaseCredentialDetails.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    DatabasePasswordCredentialDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabasePasswordCredentialDetails = exports.DatabasePasswordCredentialDetails || (exports.DatabasePasswordCredentialDetails = {}));
