// oci-databasemanagement/lib/model/database-status.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseStatus = void 0;
/**
 * The status of the Oracle Database. Indicates whether the status of the database is UP, DOWN, or UNKNOWN at the current time.
 **/
var DatabaseStatus;
(function (DatabaseStatus) {
    DatabaseStatus["Up"] = "UP";
    DatabaseStatus["Down"] = "DOWN";
    DatabaseStatus["Unknown"] = "UNKNOWN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DatabaseStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(DatabaseStatus = exports.DatabaseStatus || (exports.DatabaseStatus = {}));
(function (DatabaseStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    DatabaseStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DatabaseStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabaseStatus = exports.DatabaseStatus || (exports.DatabaseStatus = {}));
