// oci-databasemanagement/lib/model/db-management-feature.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbManagementFeature = void 0;
/**
 * The list of the database management supported features:
 *    - DIAGNOSTICS_AND_MANAGEMENT: Diagnostics and management.
 *    - DB_LIFECYCLE_MANAGEMENT: Database lifecycle management.
 *    - SQLWATCH: Real application testing.
 *
 **/
var DbManagementFeature;
(function (DbManagementFeature) {
    DbManagementFeature["DiagnosticsAndManagement"] = "DIAGNOSTICS_AND_MANAGEMENT";
    DbManagementFeature["DbLifecycleManagement"] = "DB_LIFECYCLE_MANAGEMENT";
    DbManagementFeature["Sqlwatch"] = "SQLWATCH";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DbManagementFeature["UnknownValue"] = "UNKNOWN_VALUE";
})(DbManagementFeature = exports.DbManagementFeature || (exports.DbManagementFeature = {}));
(function (DbManagementFeature) {
    function getJsonObj(obj) {
        return obj;
    }
    DbManagementFeature.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DbManagementFeature.getDeserializedJsonObj = getDeserializedJsonObj;
})(DbManagementFeature = exports.DbManagementFeature || (exports.DbManagementFeature = {}));
