// oci-databasemanagement/lib/model/db-role.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbRole = void 0;
/**
 * The Database role of the Managed Database.
 **/
var DbRole;
(function (DbRole) {
    DbRole["SnapshotStandby"] = "SNAPSHOT_STANDBY";
    DbRole["LogicalStandby"] = "LOGICAL_STANDBY";
    DbRole["PhysicalStandby"] = "PHYSICAL_STANDBY";
    DbRole["Primary"] = "PRIMARY";
    DbRole["FarSync"] = "FAR_SYNC";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DbRole["UnknownValue"] = "UNKNOWN_VALUE";
})(DbRole = exports.DbRole || (exports.DbRole = {}));
(function (DbRole) {
    function getJsonObj(obj) {
        return obj;
    }
    DbRole.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DbRole.getDeserializedJsonObj = getDeserializedJsonObj;
})(DbRole = exports.DbRole || (exports.DbRole = {}));
