// oci-databasemanagement/lib/model/iorm-plan-status-enum.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IormPlanStatusEnum = void 0;
/**
 * The status of the IORM plan of an Exadata storage server.
 * The OTHER status is used when Exadata adds a new value, and OTHER should not be used as input
 * when to make change to IORM plan.
 *
 **/
var IormPlanStatusEnum;
(function (IormPlanStatusEnum) {
    IormPlanStatusEnum["Active"] = "ACTIVE";
    IormPlanStatusEnum["Inactive"] = "INACTIVE";
    IormPlanStatusEnum["Other"] = "OTHER";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    IormPlanStatusEnum["UnknownValue"] = "UNKNOWN_VALUE";
})(IormPlanStatusEnum = exports.IormPlanStatusEnum || (exports.IormPlanStatusEnum = {}));
(function (IormPlanStatusEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    IormPlanStatusEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    IormPlanStatusEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(IormPlanStatusEnum = exports.IormPlanStatusEnum || (exports.IormPlanStatusEnum = {}));
