// oci-databasemanagement/lib/model/load-sql-tuning-set-details.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoadSqlTuningSetDetails = void 0;
const model = __importStar(require("../model"));
var LoadSqlTuningSetDetails;
(function (LoadSqlTuningSetDetails) {
    let LoadType;
    (function (LoadType) {
        LoadType["IncrementalCursorCache"] = "INCREMENTAL_CURSOR_CACHE";
        LoadType["CurrentCursorCache"] = "CURRENT_CURSOR_CACHE";
        LoadType["Awr"] = "AWR";
    })(LoadType = LoadSqlTuningSetDetails.LoadType || (LoadSqlTuningSetDetails.LoadType = {}));
    let RecursiveSql;
    (function (RecursiveSql) {
        RecursiveSql["HasRecursiveSql"] = "HAS_RECURSIVE_SQL";
        RecursiveSql["NoRecursiveSql"] = "NO_RECURSIVE_SQL";
    })(RecursiveSql = LoadSqlTuningSetDetails.RecursiveSql || (LoadSqlTuningSetDetails.RecursiveSql = {}));
    let CaptureOption;
    (function (CaptureOption) {
        CaptureOption["Insert"] = "INSERT";
        CaptureOption["Update"] = "UPDATE";
        CaptureOption["Merge"] = "MERGE";
    })(CaptureOption = LoadSqlTuningSetDetails.CaptureOption || (LoadSqlTuningSetDetails.CaptureOption = {}));
    let CaptureMode;
    (function (CaptureMode) {
        CaptureMode["ModeReplaceOldStats"] = "MODE_REPLACE_OLD_STATS";
        CaptureMode["ModeAccumulateStats"] = "MODE_ACCUMULATE_STATS";
    })(CaptureMode = LoadSqlTuningSetDetails.CaptureMode || (LoadSqlTuningSetDetails.CaptureMode = {}));
    let LoadOption;
    (function (LoadOption) {
        LoadOption["Insert"] = "INSERT";
        LoadOption["Update"] = "UPDATE";
        LoadOption["Merge"] = "MERGE";
    })(LoadOption = LoadSqlTuningSetDetails.LoadOption || (LoadSqlTuningSetDetails.LoadOption = {}));
    let UpdateOption;
    (function (UpdateOption) {
        UpdateOption["Replace"] = "REPLACE";
        UpdateOption["Accumulate"] = "ACCUMULATE";
    })(UpdateOption = LoadSqlTuningSetDetails.UpdateOption || (LoadSqlTuningSetDetails.UpdateOption = {}));
    let UpdateCondition;
    (function (UpdateCondition) {
        UpdateCondition["Old"] = "OLD";
        UpdateCondition["New"] = "NEW";
        UpdateCondition["Null"] = "NULL";
    })(UpdateCondition = LoadSqlTuningSetDetails.UpdateCondition || (LoadSqlTuningSetDetails.UpdateCondition = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "credentialDetails": obj.credentialDetails
                ? model.SqlTuningSetAdminCredentialDetails.getJsonObj(obj.credentialDetails)
                : undefined,
            "databaseCredential": obj.databaseCredential
                ? model.DatabaseCredentialDetails.getJsonObj(obj.databaseCredential)
                : undefined
        });
        return jsonObj;
    }
    LoadSqlTuningSetDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "credentialDetails": obj.credentialDetails
                ? model.SqlTuningSetAdminCredentialDetails.getDeserializedJsonObj(obj.credentialDetails)
                : undefined,
            "databaseCredential": obj.databaseCredential
                ? model.DatabaseCredentialDetails.getDeserializedJsonObj(obj.databaseCredential)
                : undefined
        });
        return jsonObj;
    }
    LoadSqlTuningSetDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(LoadSqlTuningSetDetails = exports.LoadSqlTuningSetDetails || (exports.LoadSqlTuningSetDetails = {}));
