// oci-databasemanagement/lib/model/management-option.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagementOption = void 0;
/**
 * The management option used when enabling Database Management.
* <p>
- BASIC: Basic database management for Oracle Databases that includes key monitoring metrics and basic Performance Hub features.
* - ADVANCED: Advanced database management for Oracle Databases that includes basic database management, fleet management, advanced Performance Hub and Managed Database features.
*
**/
var ManagementOption;
(function (ManagementOption) {
    ManagementOption["Basic"] = "BASIC";
    ManagementOption["Advanced"] = "ADVANCED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ManagementOption["UnknownValue"] = "UNKNOWN_VALUE";
})(ManagementOption = exports.ManagementOption || (exports.ManagementOption = {}));
(function (ManagementOption) {
    function getJsonObj(obj) {
        return obj;
    }
    ManagementOption.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ManagementOption.getDeserializedJsonObj = getDeserializedJsonObj;
})(ManagementOption = exports.ManagementOption || (exports.ManagementOption = {}));
