// oci-databasemanagement/lib/model/mds-type.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MdsType = void 0;
/**
 * The type of MySQL Database System.
 **/
var MdsType;
(function (MdsType) {
    MdsType["Ha"] = "HA";
    MdsType["Heatwave"] = "HEATWAVE";
    MdsType["Standalone"] = "STANDALONE";
})(MdsType = exports.MdsType || (exports.MdsType = {}));
(function (MdsType) {
    function getJsonObj(obj) {
        return obj;
    }
    MdsType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MdsType.getDeserializedJsonObj = getDeserializedJsonObj;
})(MdsType = exports.MdsType || (exports.MdsType = {}));
