// oci-databasemanagement/lib/model/my-sql-network-protocol-type.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MySqlNetworkProtocolType = void 0;
/**
 * The network protocol for MySQL Database.
 **/
var MySqlNetworkProtocolType;
(function (MySqlNetworkProtocolType) {
    MySqlNetworkProtocolType["Tcp"] = "TCP";
    MySqlNetworkProtocolType["Tcps"] = "TCPS";
    MySqlNetworkProtocolType["Sockets"] = "SOCKETS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MySqlNetworkProtocolType["UnknownValue"] = "UNKNOWN_VALUE";
})(MySqlNetworkProtocolType = exports.MySqlNetworkProtocolType || (exports.MySqlNetworkProtocolType = {}));
(function (MySqlNetworkProtocolType) {
    function getJsonObj(obj) {
        return obj;
    }
    MySqlNetworkProtocolType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MySqlNetworkProtocolType.getDeserializedJsonObj = getDeserializedJsonObj;
})(MySqlNetworkProtocolType = exports.MySqlNetworkProtocolType || (exports.MySqlNetworkProtocolType = {}));
