// oci-databasemanagement/lib/model/my-sql-cred-type.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MySqlCredType = void 0;
/**
 * The credential type for MySQL Database.
 **/
var MySqlCredType;
(function (MySqlCredType) {
    MySqlCredType["MysqlExternalNonSslCredentials"] = "MYSQL_EXTERNAL_NON_SSL_CREDENTIALS";
    MySqlCredType["MysqlExternalSslCredentials"] = "MYSQL_EXTERNAL_SSL_CREDENTIALS";
    MySqlCredType["MysqlExternalSocketCredentials"] = "MYSQL_EXTERNAL_SOCKET_CREDENTIALS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MySqlCredType["UnknownValue"] = "UNKNOWN_VALUE";
})(MySqlCredType = exports.MySqlCredType || (exports.MySqlCredType = {}));
(function (MySqlCredType) {
    function getJsonObj(obj) {
        return obj;
    }
    MySqlCredType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MySqlCredType.getDeserializedJsonObj = getDeserializedJsonObj;
})(MySqlCredType = exports.MySqlCredType || (exports.MySqlCredType = {}));
