// oci-databasemanagement/lib/model/my-sql-configuration-data-summary.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MySqlConfigurationDataSummary = void 0;
var MySqlConfigurationDataSummary;
(function (MySqlConfigurationDataSummary) {
    let Source;
    (function (Source) {
        Source["Compiled"] = "COMPILED";
        Source["Global"] = "GLOBAL";
        Source["Server"] = "SERVER";
        Source["Explicit"] = "EXPLICIT";
        Source["Extra"] = "EXTRA";
        Source["User"] = "USER";
        Source["Login"] = "LOGIN";
        Source["CommandLine"] = "COMMAND_LINE";
        Source["Persisted"] = "PERSISTED";
        Source["Dynamic"] = "DYNAMIC";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Source["UnknownValue"] = "UNKNOWN_VALUE";
    })(Source = MySqlConfigurationDataSummary.Source || (MySqlConfigurationDataSummary.Source = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MySqlConfigurationDataSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MySqlConfigurationDataSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(MySqlConfigurationDataSummary = exports.MySqlConfigurationDataSummary || (exports.MySqlConfigurationDataSummary = {}));
