// oci-databasemanagement/lib/model/named-credential-content-type.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NamedCredentialContentType = void 0;
/**
 * The possible content types of a named credential.
 **/
var NamedCredentialContentType;
(function (NamedCredentialContentType) {
    NamedCredentialContentType["Basic"] = "BASIC";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NamedCredentialContentType["UnknownValue"] = "UNKNOWN_VALUE";
})(NamedCredentialContentType = exports.NamedCredentialContentType || (exports.NamedCredentialContentType = {}));
(function (NamedCredentialContentType) {
    function getJsonObj(obj) {
        return obj;
    }
    NamedCredentialContentType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    NamedCredentialContentType.getDeserializedJsonObj = getDeserializedJsonObj;
})(NamedCredentialContentType = exports.NamedCredentialContentType || (exports.NamedCredentialContentType = {}));
