// oci-databasemanagement/lib/model/ranking-measure.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RankingMeasure = void 0;
/**
 * Defines an ORDER BY clause on the selected SQL represented by SQLSET_ROW object.
 * Possible values are defined in the enum structure below.
 *
 **/
var RankingMeasure;
(function (RankingMeasure) {
    RankingMeasure["ElapsedTime"] = "ELAPSED_TIME";
    RankingMeasure["CpuTime"] = "CPU_TIME";
    RankingMeasure["OptimizerCost"] = "OPTIMIZER_COST";
    RankingMeasure["BufferGets"] = "BUFFER_GETS";
    RankingMeasure["DiskReads"] = "DISK_READS";
    RankingMeasure["DirectWrites"] = "DIRECT_WRITES";
})(RankingMeasure = exports.RankingMeasure || (exports.RankingMeasure = {}));
(function (RankingMeasure) {
    function getJsonObj(obj) {
        return obj;
    }
    RankingMeasure.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RankingMeasure.getDeserializedJsonObj = getDeserializedJsonObj;
})(RankingMeasure = exports.RankingMeasure || (exports.RankingMeasure = {}));
