// oci-databasemanagement/lib/model/rest-credential.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RestCredential = void 0;
var RestCredential;
(function (RestCredential) {
    let SslTrustStoreType;
    (function (SslTrustStoreType) {
        SslTrustStoreType["Jks"] = "JKS";
        SslTrustStoreType["Bcfks"] = "BCFKS";
    })(SslTrustStoreType = RestCredential.SslTrustStoreType || (RestCredential.SslTrustStoreType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RestCredential.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RestCredential.getDeserializedJsonObj = getDeserializedJsonObj;
})(RestCredential = exports.RestCredential || (exports.RestCredential = {}));
