// oci-databasemanagement/lib/model/sql-plan-baseline-summary.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlPlanBaselineSummary = void 0;
var SqlPlanBaselineSummary;
(function (SqlPlanBaselineSummary) {
    let Enabled;
    (function (Enabled) {
        Enabled["Yes"] = "YES";
        Enabled["No"] = "NO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Enabled["UnknownValue"] = "UNKNOWN_VALUE";
    })(Enabled = SqlPlanBaselineSummary.Enabled || (SqlPlanBaselineSummary.Enabled = {}));
    let Accepted;
    (function (Accepted) {
        Accepted["Yes"] = "YES";
        Accepted["No"] = "NO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Accepted["UnknownValue"] = "UNKNOWN_VALUE";
    })(Accepted = SqlPlanBaselineSummary.Accepted || (SqlPlanBaselineSummary.Accepted = {}));
    let Fixed;
    (function (Fixed) {
        Fixed["Yes"] = "YES";
        Fixed["No"] = "NO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Fixed["UnknownValue"] = "UNKNOWN_VALUE";
    })(Fixed = SqlPlanBaselineSummary.Fixed || (SqlPlanBaselineSummary.Fixed = {}));
    let Reproduced;
    (function (Reproduced) {
        Reproduced["Yes"] = "YES";
        Reproduced["No"] = "NO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Reproduced["UnknownValue"] = "UNKNOWN_VALUE";
    })(Reproduced = SqlPlanBaselineSummary.Reproduced || (SqlPlanBaselineSummary.Reproduced = {}));
    let AutoPurge;
    (function (AutoPurge) {
        AutoPurge["Yes"] = "YES";
        AutoPurge["No"] = "NO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AutoPurge["UnknownValue"] = "UNKNOWN_VALUE";
    })(AutoPurge = SqlPlanBaselineSummary.AutoPurge || (SqlPlanBaselineSummary.AutoPurge = {}));
    let Adaptive;
    (function (Adaptive) {
        Adaptive["Yes"] = "YES";
        Adaptive["No"] = "NO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Adaptive["UnknownValue"] = "UNKNOWN_VALUE";
    })(Adaptive = SqlPlanBaselineSummary.Adaptive || (SqlPlanBaselineSummary.Adaptive = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SqlPlanBaselineSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SqlPlanBaselineSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlPlanBaselineSummary = exports.SqlPlanBaselineSummary || (exports.SqlPlanBaselineSummary = {}));
