// oci-databasemanagement/lib/model/sql-plan-baseline-job-summary.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlPlanBaselineJobSummary = void 0;
var SqlPlanBaselineJobSummary;
(function (SqlPlanBaselineJobSummary) {
    let Type;
    (function (Type) {
        Type["Load"] = "LOAD";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = SqlPlanBaselineJobSummary.Type || (SqlPlanBaselineJobSummary.Type = {}));
    let Status;
    (function (Status) {
        Status["Succeeded"] = "SUCCEEDED";
        Status["Scheduled"] = "SCHEDULED";
        Status["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = SqlPlanBaselineJobSummary.Status || (SqlPlanBaselineJobSummary.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SqlPlanBaselineJobSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SqlPlanBaselineJobSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlPlanBaselineJobSummary = exports.SqlPlanBaselineJobSummary || (exports.SqlPlanBaselineJobSummary = {}));
