// oci-databasemanagement/lib/model/sql-plan-baseline-origin.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlPlanBaselineOrigin = void 0;
/**
 * The origin of the SQL plan baseline.
 **/
var SqlPlanBaselineOrigin;
(function (SqlPlanBaselineOrigin) {
    SqlPlanBaselineOrigin["AddmSqltune"] = "ADDM_SQLTUNE";
    SqlPlanBaselineOrigin["AutoCapture"] = "AUTO_CAPTURE";
    SqlPlanBaselineOrigin["AutoSqltune"] = "AUTO_SQLTUNE";
    SqlPlanBaselineOrigin["EvolveAutoIndexLoad"] = "EVOLVE_AUTO_INDEX_LOAD";
    SqlPlanBaselineOrigin["EvolveCreateFromAdaptive"] = "EVOLVE_CREATE_FROM_ADAPTIVE";
    SqlPlanBaselineOrigin["EvolveLoadFromSts"] = "EVOLVE_LOAD_FROM_STS";
    SqlPlanBaselineOrigin["EvolveLoadFromAwr"] = "EVOLVE_LOAD_FROM_AWR";
    SqlPlanBaselineOrigin["EvolveLoadFromCursorCache"] = "EVOLVE_LOAD_FROM_CURSOR_CACHE";
    SqlPlanBaselineOrigin["ManualLoad"] = "MANUAL_LOAD";
    SqlPlanBaselineOrigin["ManualLoadFromAwr"] = "MANUAL_LOAD_FROM_AWR";
    SqlPlanBaselineOrigin["ManualLoadFromCursorCache"] = "MANUAL_LOAD_FROM_CURSOR_CACHE";
    SqlPlanBaselineOrigin["ManualLoadFromSts"] = "MANUAL_LOAD_FROM_STS";
    SqlPlanBaselineOrigin["ManualSqltune"] = "MANUAL_SQLTUNE";
    SqlPlanBaselineOrigin["StoredOutline"] = "STORED_OUTLINE";
    SqlPlanBaselineOrigin["Unknown"] = "UNKNOWN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SqlPlanBaselineOrigin["UnknownValue"] = "UNKNOWN_VALUE";
})(SqlPlanBaselineOrigin = exports.SqlPlanBaselineOrigin || (exports.SqlPlanBaselineOrigin = {}));
(function (SqlPlanBaselineOrigin) {
    function getJsonObj(obj) {
        return obj;
    }
    SqlPlanBaselineOrigin.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SqlPlanBaselineOrigin.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlPlanBaselineOrigin = exports.SqlPlanBaselineOrigin || (exports.SqlPlanBaselineOrigin = {}));
