// oci-announcementsservice/lib/model/comms-manager-name.js
"use strict";
/**
 * Announcements Service API
 * Manage Oracle Cloud Infrastructure console announcements.
 * OpenAPI spec version: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommsManagerName = void 0;
/**
 * Name of the comms manager team that manages Notifications to this service.
 **/
var CommsManagerName;
(function (CommsManagerName) {
    CommsManagerName["Cn"] = "CN";
    CommsManagerName["Fusion"] = "FUSION";
    CommsManagerName["As"] = "AS";
    CommsManagerName["Erf"] = "ERF";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CommsManagerName["UnknownValue"] = "UNKNOWN_VALUE";
})(CommsManagerName = exports.CommsManagerName || (exports.CommsManagerName = {}));
(function (CommsManagerName) {
    function getJsonObj(obj) {
        return obj;
    }
    CommsManagerName.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CommsManagerName.getDeserializedJsonObj = getDeserializedJsonObj;
})(CommsManagerName = exports.CommsManagerName || (exports.CommsManagerName = {}));
