// oci-announcementsservice/lib/model/platform-type.js
"use strict";
/**
 * Announcements Service API
 * Manage Oracle Cloud Infrastructure console announcements.
 * OpenAPI spec version: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlatformType = void 0;
/**
 * Platform type this announcement is related to
 **/
var PlatformType;
(function (PlatformType) {
    PlatformType["Iaas"] = "IAAS";
    PlatformType["Saas"] = "SAAS";
    PlatformType["Paas"] = "PAAS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PlatformType["UnknownValue"] = "UNKNOWN_VALUE";
})(PlatformType = exports.PlatformType || (exports.PlatformType = {}));
(function (PlatformType) {
    function getJsonObj(obj) {
        return obj;
    }
    PlatformType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PlatformType.getDeserializedJsonObj = getDeserializedJsonObj;
})(PlatformType = exports.PlatformType || (exports.PlatformType = {}));
