// oci-lustrefilestorage/lib/model/lustre-file-system.js
"use strict";
/**
 * File Storage with Lustre API
 * Use the File Storage with Lustre API to manage Lustre file systems and related resources. For more information, see [File Storage with Lustre](https://docs.oracle.com/iaas/Content/lustre/home.htm).
 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LustreFileSystem = void 0;
const model = __importStar(require("../model"));
var LustreFileSystem;
(function (LustreFileSystem) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = LustreFileSystem.LifecycleState || (LustreFileSystem.LifecycleState = {}));
    let PerformanceTier;
    (function (PerformanceTier) {
        PerformanceTier["MbpsPerTb125"] = "MBPS_PER_TB_125";
        PerformanceTier["MbpsPerTb250"] = "MBPS_PER_TB_250";
        PerformanceTier["MbpsPerTb500"] = "MBPS_PER_TB_500";
        PerformanceTier["MbpsPerTb1000"] = "MBPS_PER_TB_1000";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PerformanceTier["UnknownValue"] = "UNKNOWN_VALUE";
    })(PerformanceTier = LustreFileSystem.PerformanceTier || (LustreFileSystem.PerformanceTier = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "maintenanceWindow": obj.maintenanceWindow
                ? model.MaintenanceWindow.getJsonObj(obj.maintenanceWindow)
                : undefined,
            "rootSquashConfiguration": obj.rootSquashConfiguration
                ? model.RootSquashConfiguration.getJsonObj(obj.rootSquashConfiguration)
                : undefined
        });
        return jsonObj;
    }
    LustreFileSystem.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "maintenanceWindow": obj.maintenanceWindow
                ? model.MaintenanceWindow.getDeserializedJsonObj(obj.maintenanceWindow)
                : undefined,
            "rootSquashConfiguration": obj.rootSquashConfiguration
                ? model.RootSquashConfiguration.getDeserializedJsonObj(obj.rootSquashConfiguration)
                : undefined
        });
        return jsonObj;
    }
    LustreFileSystem.getDeserializedJsonObj = getDeserializedJsonObj;
})(LustreFileSystem = exports.LustreFileSystem || (exports.LustreFileSystem = {}));
