// oci-lustrefilestorage/lib/model/operation-type.js
"use strict";
/**
 * File Storage with Lustre API
 * Use the File Storage with Lustre API to manage Lustre file systems and related resources. For more information, see [File Storage with Lustre](https://docs.oracle.com/iaas/Content/lustre/home.htm).
 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * The asynchronous operation tracked by the work request.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateLustreFileSystem"] = "CREATE_LUSTRE_FILE_SYSTEM";
    OperationType["UpdateLustreFileSystem"] = "UPDATE_LUSTRE_FILE_SYSTEM";
    OperationType["DeleteLustreFileSystem"] = "DELETE_LUSTRE_FILE_SYSTEM";
    OperationType["MoveLustreFileSystem"] = "MOVE_LUSTRE_FILE_SYSTEM";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
