// oci-opsi/lib/model/actionable-insights-content-types-resource.js
"use strict";
/**
 * Ops Insights API
 * Use the Ops Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Ops Insights](https://docs.oracle.com/iaas/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionableInsightsContentTypesResource = void 0;
/**
 * Supported resources for actionable insights content type.
 **/
var ActionableInsightsContentTypesResource;
(function (ActionableInsightsContentTypesResource) {
    ActionableInsightsContentTypesResource["NewHighs"] = "NEW_HIGHS";
    ActionableInsightsContentTypesResource["BigChanges"] = "BIG_CHANGES";
    ActionableInsightsContentTypesResource["CurrentInventory"] = "CURRENT_INVENTORY";
    ActionableInsightsContentTypesResource["InventoryChanges"] = "INVENTORY_CHANGES";
    ActionableInsightsContentTypesResource["FleetStatistics"] = "FLEET_STATISTICS";
    ActionableInsightsContentTypesResource["FleetAnalysisSummaryDbCount"] = "FLEET_ANALYSIS_SUMMARY_DB_COUNT";
    ActionableInsightsContentTypesResource["FleetAnalysisSummarySqlAnalyzedCount"] = "FLEET_ANALYSIS_SUMMARY_SQL_ANALYZED_COUNT";
    ActionableInsightsContentTypesResource["FleetAnalysisSummaryNewSqlCount"] = "FLEET_ANALYSIS_SUMMARY_NEW_SQL_COUNT";
    ActionableInsightsContentTypesResource["FleetAnalysisSummaryBusiestDb"] = "FLEET_ANALYSIS_SUMMARY_BUSIEST_DB";
    ActionableInsightsContentTypesResource["FleetAnalysisDegradingSqlCount"] = "FLEET_ANALYSIS_DEGRADING_SQL_COUNT";
    ActionableInsightsContentTypesResource["FleetAnalysisDegradingSqlByDb"] = "FLEET_ANALYSIS_DEGRADING_SQL_BY_DB";
    ActionableInsightsContentTypesResource["FleetAnalysisDegradingSqlBySqlId"] = "FLEET_ANALYSIS_DEGRADING_SQL_BY_SQL_ID";
    ActionableInsightsContentTypesResource["FleetAnalysisPlanChangesCount"] = "FLEET_ANALYSIS_PLAN_CHANGES_COUNT";
    ActionableInsightsContentTypesResource["FleetAnalysisPlanChangesDbMostChanges"] = "FLEET_ANALYSIS_PLAN_CHANGES_DB_MOST_CHANGES";
    ActionableInsightsContentTypesResource["FleetAnalysisPlanChangesBySqlIdImproved"] = "FLEET_ANALYSIS_PLAN_CHANGES_BY_SQL_ID_IMPROVED";
    ActionableInsightsContentTypesResource["FleetAnalysisPlanChangesBySqlIdDegraded"] = "FLEET_ANALYSIS_PLAN_CHANGES_BY_SQL_ID_DEGRADED";
    ActionableInsightsContentTypesResource["FleetAnalysisInvalidationStormsCount"] = "FLEET_ANALYSIS_INVALIDATION_STORMS_COUNT";
    ActionableInsightsContentTypesResource["FleetAnalysisInvalidationStormsHighest"] = "FLEET_ANALYSIS_INVALIDATION_STORMS_HIGHEST";
    ActionableInsightsContentTypesResource["FleetAnalysisCursorSharingIssuesCount"] = "FLEET_ANALYSIS_CURSOR_SHARING_ISSUES_COUNT";
    ActionableInsightsContentTypesResource["FleetAnalysisCursorSharingIssuesByDb"] = "FLEET_ANALYSIS_CURSOR_SHARING_ISSUES_BY_DB";
    ActionableInsightsContentTypesResource["FleetAnalysisCursorSharingIssuesBySql"] = "FLEET_ANALYSIS_CURSOR_SHARING_ISSUES_BY_SQL";
    ActionableInsightsContentTypesResource["PerformanceDegradationSummaryDbCount"] = "PERFORMANCE_DEGRADATION_SUMMARY_DB_COUNT";
    ActionableInsightsContentTypesResource["PerformanceDegradationSummarySqlAnalyzedCount"] = "PERFORMANCE_DEGRADATION_SUMMARY_SQL_ANALYZED_COUNT";
    ActionableInsightsContentTypesResource["PerformanceDegradationSummarySqlPerformanceTrendsCount"] = "PERFORMANCE_DEGRADATION_SUMMARY_SQL_PERFORMANCE_TRENDS_COUNT";
    ActionableInsightsContentTypesResource["PerformanceDegradationSummaryDegradedSqlCount"] = "PERFORMANCE_DEGRADATION_SUMMARY_DEGRADED_SQL_COUNT";
    ActionableInsightsContentTypesResource["PerformanceDegradationSummaryImprovedSqlCount"] = "PERFORMANCE_DEGRADATION_SUMMARY_IMPROVED_SQL_COUNT";
    ActionableInsightsContentTypesResource["PerformanceDegradationDbDegradedCount"] = "PERFORMANCE_DEGRADATION_DB_DEGRADED_COUNT";
    ActionableInsightsContentTypesResource["PerformanceDegradationSqlDegradedTable"] = "PERFORMANCE_DEGRADATION_SQL_DEGRADED_TABLE";
    ActionableInsightsContentTypesResource["PlanChangesSummaryDbCount"] = "PLAN_CHANGES_SUMMARY_DB_COUNT";
    ActionableInsightsContentTypesResource["PlanChangesSummarySqlAnalyzedCount"] = "PLAN_CHANGES_SUMMARY_SQL_ANALYZED_COUNT";
    ActionableInsightsContentTypesResource["PlanChangesSummaryPlanChangesCount"] = "PLAN_CHANGES_SUMMARY_PLAN_CHANGES_COUNT";
    ActionableInsightsContentTypesResource["PlanChangesSummaryImprovementsCount"] = "PLAN_CHANGES_SUMMARY_IMPROVEMENTS_COUNT";
    ActionableInsightsContentTypesResource["PlanChangesSummaryDegradationCount"] = "PLAN_CHANGES_SUMMARY_DEGRADATION_COUNT";
    ActionableInsightsContentTypesResource["PlanChangesTopPlanChangesTable"] = "PLAN_CHANGES_TOP_PLAN_CHANGES_TABLE";
    ActionableInsightsContentTypesResource["TopDbSummaryDbCount"] = "TOP_DB_SUMMARY_DB_COUNT";
    ActionableInsightsContentTypesResource["TopDbSummarySqlAnalyzedCount"] = "TOP_DB_SUMMARY_SQL_ANALYZED_COUNT";
    ActionableInsightsContentTypesResource["TopDbSummaryBusiestDb"] = "TOP_DB_SUMMARY_BUSIEST_DB";
    ActionableInsightsContentTypesResource["TopTable"] = "TOP_TABLE";
    ActionableInsightsContentTypesResource["CollectionDelayCount"] = "COLLECTION_DELAY_COUNT";
    ActionableInsightsContentTypesResource["CollectionDelayPreviousWeekCount"] = "COLLECTION_DELAY_PREVIOUS_WEEK_COUNT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ActionableInsightsContentTypesResource["UnknownValue"] = "UNKNOWN_VALUE";
})(ActionableInsightsContentTypesResource = exports.ActionableInsightsContentTypesResource || (exports.ActionableInsightsContentTypesResource = {}));
(function (ActionableInsightsContentTypesResource) {
    function getJsonObj(obj) {
        return obj;
    }
    ActionableInsightsContentTypesResource.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ActionableInsightsContentTypesResource.getDeserializedJsonObj = getDeserializedJsonObj;
})(ActionableInsightsContentTypesResource = exports.ActionableInsightsContentTypesResource || (exports.ActionableInsightsContentTypesResource = {}));
