// oci-opsi/lib/model/awr-hub-source-type.js
"use strict";
/**
 * Ops Insights API
 * Use the Ops Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Ops Insights](https://docs.oracle.com/iaas/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AwrHubSourceType = void 0;
/**
 * source types of the databases
 **/
var AwrHubSourceType;
(function (AwrHubSourceType) {
    AwrHubSourceType["AdwS"] = "ADW_S";
    AwrHubSourceType["AtpS"] = "ATP_S";
    AwrHubSourceType["AdwD"] = "ADW_D";
    AwrHubSourceType["AtpD"] = "ATP_D";
    AwrHubSourceType["ExternalPdb"] = "EXTERNAL_PDB";
    AwrHubSourceType["ExternalNoncdb"] = "EXTERNAL_NONCDB";
    AwrHubSourceType["ComanagedVmCdb"] = "COMANAGED_VM_CDB";
    AwrHubSourceType["ComanagedVmPdb"] = "COMANAGED_VM_PDB";
    AwrHubSourceType["ComanagedVmNoncdb"] = "COMANAGED_VM_NONCDB";
    AwrHubSourceType["ComanagedBmCdb"] = "COMANAGED_BM_CDB";
    AwrHubSourceType["ComanagedBmPdb"] = "COMANAGED_BM_PDB";
    AwrHubSourceType["ComanagedBmNoncdb"] = "COMANAGED_BM_NONCDB";
    AwrHubSourceType["ComanagedExacsCdb"] = "COMANAGED_EXACS_CDB";
    AwrHubSourceType["ComanagedExacsPdb"] = "COMANAGED_EXACS_PDB";
    AwrHubSourceType["ComanagedExacsNoncdb"] = "COMANAGED_EXACS_NONCDB";
    AwrHubSourceType["Undefined"] = "UNDEFINED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AwrHubSourceType["UnknownValue"] = "UNKNOWN_VALUE";
})(AwrHubSourceType = exports.AwrHubSourceType || (exports.AwrHubSourceType = {}));
(function (AwrHubSourceType) {
    function getJsonObj(obj) {
        return obj;
    }
    AwrHubSourceType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AwrHubSourceType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AwrHubSourceType = exports.AwrHubSourceType || (exports.AwrHubSourceType = {}));
