// oci-opsi/lib/model/awr-hub-source-lifecycle-state.js
"use strict";
/**
 * Ops Insights API
 * Use the Ops Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Ops Insights](https://docs.oracle.com/iaas/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AwrHubSourceLifecycleState = void 0;
/**
 * Possible lifecycle states.
 **/
var AwrHubSourceLifecycleState;
(function (AwrHubSourceLifecycleState) {
    AwrHubSourceLifecycleState["Creating"] = "CREATING";
    AwrHubSourceLifecycleState["Updating"] = "UPDATING";
    AwrHubSourceLifecycleState["Active"] = "ACTIVE";
    AwrHubSourceLifecycleState["Deleting"] = "DELETING";
    AwrHubSourceLifecycleState["Deleted"] = "DELETED";
    AwrHubSourceLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AwrHubSourceLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(AwrHubSourceLifecycleState = exports.AwrHubSourceLifecycleState || (exports.AwrHubSourceLifecycleState = {}));
(function (AwrHubSourceLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    AwrHubSourceLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AwrHubSourceLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(AwrHubSourceLifecycleState = exports.AwrHubSourceLifecycleState || (exports.AwrHubSourceLifecycleState = {}));
