// oci-opsi/lib/model/create-macs-managed-cloud-database-insight-details.js
"use strict";
/**
 * Ops Insights API
 * Use the Ops Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Ops Insights](https://docs.oracle.com/iaas/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateMacsManagedCloudDatabaseInsightDetails = void 0;
const model = __importStar(require("../model"));
var CreateMacsManagedCloudDatabaseInsightDetails;
(function (CreateMacsManagedCloudDatabaseInsightDetails) {
    let DeploymentType;
    (function (DeploymentType) {
        DeploymentType["VirtualMachine"] = "VIRTUAL_MACHINE";
        DeploymentType["BareMetal"] = "BARE_METAL";
        DeploymentType["Exacc"] = "EXACC";
        DeploymentType["Exacs"] = "EXACS";
    })(DeploymentType = CreateMacsManagedCloudDatabaseInsightDetails.DeploymentType || (CreateMacsManagedCloudDatabaseInsightDetails.DeploymentType = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.CreateDatabaseInsightDetails.getJsonObj(obj))), {
            "connectionDetails": obj.connectionDetails
                ? model.ConnectionDetails.getJsonObj(obj.connectionDetails)
                : undefined,
            "connectionCredentialDetails": obj.connectionCredentialDetails
                ? model.CredentialDetails.getJsonObj(obj.connectionCredentialDetails)
                : undefined
        });
        return jsonObj;
    }
    CreateMacsManagedCloudDatabaseInsightDetails.getJsonObj = getJsonObj;
    CreateMacsManagedCloudDatabaseInsightDetails.entitySource = "MACS_MANAGED_CLOUD_DATABASE";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.CreateDatabaseInsightDetails.getDeserializedJsonObj(obj))), {
            "connectionDetails": obj.connectionDetails
                ? model.ConnectionDetails.getDeserializedJsonObj(obj.connectionDetails)
                : undefined,
            "connectionCredentialDetails": obj.connectionCredentialDetails
                ? model.CredentialDetails.getDeserializedJsonObj(obj.connectionCredentialDetails)
                : undefined
        });
        return jsonObj;
    }
    CreateMacsManagedCloudDatabaseInsightDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateMacsManagedCloudDatabaseInsightDetails = exports.CreateMacsManagedCloudDatabaseInsightDetails || (exports.CreateMacsManagedCloudDatabaseInsightDetails = {}));
