// oci-opsi/lib/model/exadata-vm-cluster-type.js
"use strict";
/**
 * Ops Insights API
 * Use the Ops Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Ops Insights](https://docs.oracle.com/iaas/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExadataVmClusterType = void 0;
/**
 * OCI Exadata VM Cluster Type
 **/
var ExadataVmClusterType;
(function (ExadataVmClusterType) {
    ExadataVmClusterType["VmCluster"] = "vmCluster";
})(ExadataVmClusterType = exports.ExadataVmClusterType || (exports.ExadataVmClusterType = {}));
(function (ExadataVmClusterType) {
    function getJsonObj(obj) {
        return obj;
    }
    ExadataVmClusterType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ExadataVmClusterType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExadataVmClusterType = exports.ExadataVmClusterType || (exports.ExadataVmClusterType = {}));
