// oci-opsi/lib/model/news-frequency.js
"use strict";
/**
 * Ops Insights API
 * Use the Ops Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Ops Insights](https://docs.oracle.com/iaas/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NewsFrequency = void 0;
/**
 * Frequency in which the news report will be sent.
 * WEEKLY: Receive an OPSI news report every week.
 * DAILY: Receive an OPSI news report every day.
 * HOURLY: Receive an actionable insights report every hour if there are significant changes to the customer's resources.
 *
 **/
var NewsFrequency;
(function (NewsFrequency) {
    NewsFrequency["Weekly"] = "WEEKLY";
    NewsFrequency["Daily"] = "DAILY";
    NewsFrequency["Hourly"] = "HOURLY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NewsFrequency["UnknownValue"] = "UNKNOWN_VALUE";
})(NewsFrequency = exports.NewsFrequency || (exports.NewsFrequency = {}));
(function (NewsFrequency) {
    function getJsonObj(obj) {
        return obj;
    }
    NewsFrequency.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    NewsFrequency.getDeserializedJsonObj = getDeserializedJsonObj;
})(NewsFrequency = exports.NewsFrequency || (exports.NewsFrequency = {}));
