// oci-oda/lib/model/entity-type.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityType = void 0;
/**
 * The valid set of entity type values.
 **/
var EntityType;
(function (EntityType) {
    EntityType["Composite"] = "COMPOSITE";
    EntityType["EnumValues"] = "ENUM_VALUES";
})(EntityType = exports.EntityType || (exports.EntityType = {}));
(function (EntityType) {
    function getJsonObj(obj) {
        return obj;
    }
    EntityType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    EntityType.getDeserializedJsonObj = getDeserializedJsonObj;
})(EntityType = exports.EntityType || (exports.EntityType = {}));
