// oci-oda/lib/model/entity-attribute-type.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityAttributeType = void 0;
/**
 * The valid set of entity attribute type values.
 **/
var EntityAttributeType;
(function (EntityAttributeType) {
    EntityAttributeType["Text"] = "TEXT";
    EntityAttributeType["Number"] = "NUMBER";
    EntityAttributeType["DateTime"] = "DATE_TIME";
    EntityAttributeType["Date"] = "DATE";
    EntityAttributeType["Entity"] = "ENTITY";
    EntityAttributeType["CompositeEntity"] = "COMPOSITE_ENTITY";
    EntityAttributeType["AttributeReference"] = "ATTRIBUTE_REFERENCE";
    EntityAttributeType["Boolean"] = "BOOLEAN";
})(EntityAttributeType = exports.EntityAttributeType || (exports.EntityAttributeType = {}));
(function (EntityAttributeType) {
    function getJsonObj(obj) {
        return obj;
    }
    EntityAttributeType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    EntityAttributeType.getDeserializedJsonObj = getDeserializedJsonObj;
})(EntityAttributeType = exports.EntityAttributeType || (exports.EntityAttributeType = {}));
