// oci-oda/lib/model/temporal-preference.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemporalPreference = void 0;
/**
 * The valid set of temporal preference values.
 **/
var TemporalPreference;
(function (TemporalPreference) {
    TemporalPreference["Past"] = "PAST";
    TemporalPreference["Future"] = "FUTURE";
    TemporalPreference["Nearest"] = "NEAREST";
})(TemporalPreference = exports.TemporalPreference || (exports.TemporalPreference = {}));
(function (TemporalPreference) {
    function getJsonObj(obj) {
        return obj;
    }
    TemporalPreference.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TemporalPreference.getDeserializedJsonObj = getDeserializedJsonObj;
})(TemporalPreference = exports.TemporalPreference || (exports.TemporalPreference = {}));
