// oci-oda/lib/model/train-skill-parameter-type.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrainSkillParameterType = void 0;
/**
 * The valid set of parameter type values for training a skill.
 **/
var TrainSkillParameterType;
(function (TrainSkillParameterType) {
    TrainSkillParameterType["QueryEntity"] = "QUERY_ENTITY";
})(TrainSkillParameterType = exports.TrainSkillParameterType || (exports.TrainSkillParameterType = {}));
(function (TrainSkillParameterType) {
    function getJsonObj(obj) {
        return obj;
    }
    TrainSkillParameterType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TrainSkillParameterType.getDeserializedJsonObj = getDeserializedJsonObj;
})(TrainSkillParameterType = exports.TrainSkillParameterType || (exports.TrainSkillParameterType = {}));
