// oci-wlms/lib/request/index.js
"use strict";
/**
 * WebLogic Management Service API
 * WebLogic Management Service is an OCI service that enables a unified view and management of WebLogic domains
in Oracle Cloud Infrastructure. Features include on-demand patching of WebLogic domains, rollback of the
last applied patch, discovery and management of WebLogic instances on a compute host.

 * OpenAPI spec version: 20241101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateWlsDomainCredentialRequest = exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListWlsDomainsSharingMiddlewaresRequest = exports.ListWlsDomainsRequest = exports.ListWlsDomainServersRequest = exports.ListWlsDomainServerInstalledPatchesRequest = exports.ListWlsDomainServerBackupsRequest = exports.ListWlsDomainScanResultsRequest = exports.ListManagedInstancesRequest = exports.ListManagedInstanceServersRequest = exports.ListManagedInstanceServerInstalledPatchesRequest = exports.ListManagedInstanceScanResultsRequest = exports.ListApplicablePatchesRequest = exports.ListAgreementRecordsRequest = exports.GetWlsDomainCredentialRequest = void 0;
const GetWlsDomainCredentialRequest = __importStar(require("./get-wls-domain-credential-request"));
exports.GetWlsDomainCredentialRequest = GetWlsDomainCredentialRequest.GetWlsDomainCredentialRequest;
const ListAgreementRecordsRequest = __importStar(require("./list-agreement-records-request"));
exports.ListAgreementRecordsRequest = ListAgreementRecordsRequest.ListAgreementRecordsRequest;
const ListApplicablePatchesRequest = __importStar(require("./list-applicable-patches-request"));
exports.ListApplicablePatchesRequest = ListApplicablePatchesRequest.ListApplicablePatchesRequest;
const ListManagedInstanceScanResultsRequest = __importStar(require("./list-managed-instance-scan-results-request"));
exports.ListManagedInstanceScanResultsRequest = ListManagedInstanceScanResultsRequest.ListManagedInstanceScanResultsRequest;
const ListManagedInstanceServerInstalledPatchesRequest = __importStar(require("./list-managed-instance-server-installed-patches-request"));
exports.ListManagedInstanceServerInstalledPatchesRequest = ListManagedInstanceServerInstalledPatchesRequest.ListManagedInstanceServerInstalledPatchesRequest;
const ListManagedInstanceServersRequest = __importStar(require("./list-managed-instance-servers-request"));
exports.ListManagedInstanceServersRequest = ListManagedInstanceServersRequest.ListManagedInstanceServersRequest;
const ListManagedInstancesRequest = __importStar(require("./list-managed-instances-request"));
exports.ListManagedInstancesRequest = ListManagedInstancesRequest.ListManagedInstancesRequest;
const ListWlsDomainScanResultsRequest = __importStar(require("./list-wls-domain-scan-results-request"));
exports.ListWlsDomainScanResultsRequest = ListWlsDomainScanResultsRequest.ListWlsDomainScanResultsRequest;
const ListWlsDomainServerBackupsRequest = __importStar(require("./list-wls-domain-server-backups-request"));
exports.ListWlsDomainServerBackupsRequest = ListWlsDomainServerBackupsRequest.ListWlsDomainServerBackupsRequest;
const ListWlsDomainServerInstalledPatchesRequest = __importStar(require("./list-wls-domain-server-installed-patches-request"));
exports.ListWlsDomainServerInstalledPatchesRequest = ListWlsDomainServerInstalledPatchesRequest.ListWlsDomainServerInstalledPatchesRequest;
const ListWlsDomainServersRequest = __importStar(require("./list-wls-domain-servers-request"));
exports.ListWlsDomainServersRequest = ListWlsDomainServersRequest.ListWlsDomainServersRequest;
const ListWlsDomainsRequest = __importStar(require("./list-wls-domains-request"));
exports.ListWlsDomainsRequest = ListWlsDomainsRequest.ListWlsDomainsRequest;
const ListWlsDomainsSharingMiddlewaresRequest = __importStar(require("./list-wls-domains-sharing-middlewares-request"));
exports.ListWlsDomainsSharingMiddlewaresRequest = ListWlsDomainsSharingMiddlewaresRequest.ListWlsDomainsSharingMiddlewaresRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
const UpdateWlsDomainCredentialRequest = __importStar(require("./update-wls-domain-credential-request"));
exports.UpdateWlsDomainCredentialRequest = UpdateWlsDomainCredentialRequest.UpdateWlsDomainCredentialRequest;
