// oci-wlms/lib/request/list-wls-domains-sharing-middlewares-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20241101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWlsDomainsSharingMiddlewaresRequest = void 0;
var ListWlsDomainsSharingMiddlewaresRequest;
(function (ListWlsDomainsSharingMiddlewaresRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListWlsDomainsSharingMiddlewaresRequest.SortOrder || (ListWlsDomainsSharingMiddlewaresRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListWlsDomainsSharingMiddlewaresRequest.SortBy || (ListWlsDomainsSharingMiddlewaresRequest.SortBy = {}));
})(ListWlsDomainsSharingMiddlewaresRequest = exports.ListWlsDomainsSharingMiddlewaresRequest || (exports.ListWlsDomainsSharingMiddlewaresRequest = {}));
