// oci-wlms/lib/request/list-wls-domain-server-installed-patches-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20241101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWlsDomainServerInstalledPatchesRequest = void 0;
var ListWlsDomainServerInstalledPatchesRequest;
(function (ListWlsDomainServerInstalledPatchesRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListWlsDomainServerInstalledPatchesRequest.SortOrder || (ListWlsDomainServerInstalledPatchesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListWlsDomainServerInstalledPatchesRequest.SortBy || (ListWlsDomainServerInstalledPatchesRequest.SortBy = {}));
})(ListWlsDomainServerInstalledPatchesRequest = exports.ListWlsDomainServerInstalledPatchesRequest || (exports.ListWlsDomainServerInstalledPatchesRequest = {}));
