// oci-wlms/lib/request/list-wls-domains-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20241101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWlsDomainsRequest = void 0;
var ListWlsDomainsRequest;
(function (ListWlsDomainsRequest) {
    let WeblogicVersion;
    (function (WeblogicVersion) {
        WeblogicVersion["V12214"] = "v12.2.1.4";
        WeblogicVersion["V14110"] = "v14.1.1.0";
        WeblogicVersion["V14120"] = "v14.1.2.0";
    })(WeblogicVersion = ListWlsDomainsRequest.WeblogicVersion || (ListWlsDomainsRequest.WeblogicVersion = {}));
    let MiddlewareType;
    (function (MiddlewareType) {
        MiddlewareType["Fmw"] = "FMW";
        MiddlewareType["Wls"] = "WLS";
    })(MiddlewareType = ListWlsDomainsRequest.MiddlewareType || (ListWlsDomainsRequest.MiddlewareType = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListWlsDomainsRequest.SortOrder || (ListWlsDomainsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListWlsDomainsRequest.SortBy || (ListWlsDomainsRequest.SortBy = {}));
})(ListWlsDomainsRequest = exports.ListWlsDomainsRequest || (exports.ListWlsDomainsRequest = {}));
