// oci-wlms/lib/request/list-managed-instance-scan-results-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20241101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListManagedInstanceScanResultsRequest = void 0;
var ListManagedInstanceScanResultsRequest;
(function (ListManagedInstanceScanResultsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListManagedInstanceScanResultsRequest.SortOrder || (ListManagedInstanceScanResultsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeOfServerCheck"] = "timeOfServerCheck";
        SortBy["ServerName"] = "serverName";
    })(SortBy = ListManagedInstanceScanResultsRequest.SortBy || (ListManagedInstanceScanResultsRequest.SortBy = {}));
})(ListManagedInstanceScanResultsRequest = exports.ListManagedInstanceScanResultsRequest || (exports.ListManagedInstanceScanResultsRequest = {}));
