// oci-wlms/lib/request/list-managed-instance-server-installed-patches-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20241101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListManagedInstanceServerInstalledPatchesRequest = void 0;
var ListManagedInstanceServerInstalledPatchesRequest;
(function (ListManagedInstanceServerInstalledPatchesRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListManagedInstanceServerInstalledPatchesRequest.SortOrder || (ListManagedInstanceServerInstalledPatchesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListManagedInstanceServerInstalledPatchesRequest.SortBy || (ListManagedInstanceServerInstalledPatchesRequest.SortBy = {}));
})(ListManagedInstanceServerInstalledPatchesRequest = exports.ListManagedInstanceServerInstalledPatchesRequest || (exports.ListManagedInstanceServerInstalledPatchesRequest = {}));
