// oci-securityattribute/lib/model/bulk-edit-security-attribute-operation-details.js
"use strict";
/**
 * Security Attribute API
 * Use the Security Attributes API to manage security attributes and security attribute namespaces. For more information, see the documentation for [Security Attributes](https://docs.oracle.com/iaas/Content/zero-trust-packet-routing/managing-security-attributes.htm) and [Security Attribute Namespaces](https://docs.oracle.com/iaas/Content/zero-trust-packet-routing/managing-security-attribute-namespaces.htm).
 * OpenAPI spec version: 20240815
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BulkEditSecurityAttributeOperationDetails = void 0;
var BulkEditSecurityAttributeOperationDetails;
(function (BulkEditSecurityAttributeOperationDetails) {
    let OperationType;
    (function (OperationType) {
        OperationType["AddWhereAbsent"] = "ADD_WHERE_ABSENT";
        OperationType["SetWherePresent"] = "SET_WHERE_PRESENT";
        OperationType["AddOrSet"] = "ADD_OR_SET";
        OperationType["Remove"] = "REMOVE";
    })(OperationType = BulkEditSecurityAttributeOperationDetails.OperationType || (BulkEditSecurityAttributeOperationDetails.OperationType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BulkEditSecurityAttributeOperationDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BulkEditSecurityAttributeOperationDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(BulkEditSecurityAttributeOperationDetails = exports.BulkEditSecurityAttributeOperationDetails || (exports.BulkEditSecurityAttributeOperationDetails = {}));
