// oci-securityattribute/lib/model/index.js
"use strict";
/**
 * Security Attribute API
 * Use the Security Attributes API to manage security attributes and security attribute namespaces. For more information, see the documentation for [Security Attributes](https://docs.oracle.com/iaas/Content/zero-trust-packet-routing/managing-security-attributes.htm) and [Security Attribute Namespaces](https://docs.oracle.com/iaas/Content/zero-trust-packet-routing/managing-security-attribute-namespaces.htm).
 * OpenAPI spec version: 20240815
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumSecurityAttributeValidator = exports.DefaultSecurityAttributeValidator = exports.WorkRequestStatus = exports.WorkRequestResource = exports.WorkRequestOperationType = exports.UpdateSecurityAttributeNamespaceDetails = exports.UpdateSecurityAttributeDetails = exports.SortOrder = exports.SecurityAttributeWorkRequestSummary = exports.SecurityAttributeWorkRequestLogSummary = exports.SecurityAttributeWorkRequestErrorSummary = exports.SecurityAttributeWorkRequest = exports.SecurityAttributeSummary = exports.SecurityAttributeNamespaceSummary = exports.SecurityAttributeNamespace = exports.SecurityAttribute = exports.CreateSecurityAttributeNamespaceDetails = exports.CreateSecurityAttributeDetails = exports.ChangeSecurityAttributeNamespaceCompartmentDetails = exports.BulkEditSecurityAttributeOperationDetails = exports.BulkEditSecurityAttributeDetails = exports.BulkEditResource = exports.BulkDeleteSecurityAttributesDetails = exports.BaseSecurityAttributeValidator = void 0;
const BaseSecurityAttributeValidator = __importStar(require("./base-security-attribute-validator"));
exports.BaseSecurityAttributeValidator = BaseSecurityAttributeValidator.BaseSecurityAttributeValidator;
const BulkDeleteSecurityAttributesDetails = __importStar(require("./bulk-delete-security-attributes-details"));
exports.BulkDeleteSecurityAttributesDetails = BulkDeleteSecurityAttributesDetails.BulkDeleteSecurityAttributesDetails;
const BulkEditResource = __importStar(require("./bulk-edit-resource"));
exports.BulkEditResource = BulkEditResource.BulkEditResource;
const BulkEditSecurityAttributeDetails = __importStar(require("./bulk-edit-security-attribute-details"));
exports.BulkEditSecurityAttributeDetails = BulkEditSecurityAttributeDetails.BulkEditSecurityAttributeDetails;
const BulkEditSecurityAttributeOperationDetails = __importStar(require("./bulk-edit-security-attribute-operation-details"));
exports.BulkEditSecurityAttributeOperationDetails = BulkEditSecurityAttributeOperationDetails.BulkEditSecurityAttributeOperationDetails;
const ChangeSecurityAttributeNamespaceCompartmentDetails = __importStar(require("./change-security-attribute-namespace-compartment-details"));
exports.ChangeSecurityAttributeNamespaceCompartmentDetails = ChangeSecurityAttributeNamespaceCompartmentDetails.ChangeSecurityAttributeNamespaceCompartmentDetails;
const CreateSecurityAttributeDetails = __importStar(require("./create-security-attribute-details"));
exports.CreateSecurityAttributeDetails = CreateSecurityAttributeDetails.CreateSecurityAttributeDetails;
const CreateSecurityAttributeNamespaceDetails = __importStar(require("./create-security-attribute-namespace-details"));
exports.CreateSecurityAttributeNamespaceDetails = CreateSecurityAttributeNamespaceDetails.CreateSecurityAttributeNamespaceDetails;
const SecurityAttribute = __importStar(require("./security-attribute"));
exports.SecurityAttribute = SecurityAttribute.SecurityAttribute;
const SecurityAttributeNamespace = __importStar(require("./security-attribute-namespace"));
exports.SecurityAttributeNamespace = SecurityAttributeNamespace.SecurityAttributeNamespace;
const SecurityAttributeNamespaceSummary = __importStar(require("./security-attribute-namespace-summary"));
exports.SecurityAttributeNamespaceSummary = SecurityAttributeNamespaceSummary.SecurityAttributeNamespaceSummary;
const SecurityAttributeSummary = __importStar(require("./security-attribute-summary"));
exports.SecurityAttributeSummary = SecurityAttributeSummary.SecurityAttributeSummary;
const SecurityAttributeWorkRequest = __importStar(require("./security-attribute-work-request"));
exports.SecurityAttributeWorkRequest = SecurityAttributeWorkRequest.SecurityAttributeWorkRequest;
const SecurityAttributeWorkRequestErrorSummary = __importStar(require("./security-attribute-work-request-error-summary"));
exports.SecurityAttributeWorkRequestErrorSummary = SecurityAttributeWorkRequestErrorSummary.SecurityAttributeWorkRequestErrorSummary;
const SecurityAttributeWorkRequestLogSummary = __importStar(require("./security-attribute-work-request-log-summary"));
exports.SecurityAttributeWorkRequestLogSummary = SecurityAttributeWorkRequestLogSummary.SecurityAttributeWorkRequestLogSummary;
const SecurityAttributeWorkRequestSummary = __importStar(require("./security-attribute-work-request-summary"));
exports.SecurityAttributeWorkRequestSummary = SecurityAttributeWorkRequestSummary.SecurityAttributeWorkRequestSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateSecurityAttributeDetails = __importStar(require("./update-security-attribute-details"));
exports.UpdateSecurityAttributeDetails = UpdateSecurityAttributeDetails.UpdateSecurityAttributeDetails;
const UpdateSecurityAttributeNamespaceDetails = __importStar(require("./update-security-attribute-namespace-details"));
exports.UpdateSecurityAttributeNamespaceDetails = UpdateSecurityAttributeNamespaceDetails.UpdateSecurityAttributeNamespaceDetails;
const WorkRequestOperationType = __importStar(require("./work-request-operation-type"));
exports.WorkRequestOperationType = WorkRequestOperationType.WorkRequestOperationType;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestStatus = __importStar(require("./work-request-status"));
exports.WorkRequestStatus = WorkRequestStatus.WorkRequestStatus;
const DefaultSecurityAttributeValidator = __importStar(require("./default-security-attribute-validator"));
exports.DefaultSecurityAttributeValidator = DefaultSecurityAttributeValidator.DefaultSecurityAttributeValidator;
const EnumSecurityAttributeValidator = __importStar(require("./enum-security-attribute-validator"));
exports.EnumSecurityAttributeValidator = EnumSecurityAttributeValidator.EnumSecurityAttributeValidator;
