// oci-securityattribute/lib/model/work-request-status.js
"use strict";
/**
 * Security Attribute API
 * Use the Security Attributes API to manage security attributes and security attribute namespaces. For more information, see the documentation for [Security Attributes](https://docs.oracle.com/iaas/Content/zero-trust-packet-routing/managing-security-attributes.htm) and [Security Attribute Namespaces](https://docs.oracle.com/iaas/Content/zero-trust-packet-routing/managing-security-attribute-namespaces.htm).
 * OpenAPI spec version: 20240815
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestStatus = void 0;
/**
 * The current status of the work request.
 **/
var WorkRequestStatus;
(function (WorkRequestStatus) {
    WorkRequestStatus["Accepted"] = "ACCEPTED";
    WorkRequestStatus["InProgress"] = "IN_PROGRESS";
    WorkRequestStatus["Failed"] = "FAILED";
    WorkRequestStatus["Succeeded"] = "SUCCEEDED";
    WorkRequestStatus["PartiallySucceeded"] = "PARTIALLY_SUCCEEDED";
    WorkRequestStatus["Canceling"] = "CANCELING";
    WorkRequestStatus["Canceled"] = "CANCELED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestStatus = exports.WorkRequestStatus || (exports.WorkRequestStatus = {}));
(function (WorkRequestStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    WorkRequestStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WorkRequestStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkRequestStatus = exports.WorkRequestStatus || (exports.WorkRequestStatus = {}));
