// oci-jmsjavadownloads/lib/model/aggregation-sort-by.js
"use strict";
/**
 * Java Management Service Download API
 * The APIs for the <a href="https://docs.oracle.com/en-us/iaas/jms/doc/java-download.html">Java Download</a> feature of Java Management Service.
 * OpenAPI spec version: 20230601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregationSortBy = void 0;
/**
 * The field to sort the Java download aggregation.
 **/
var AggregationSortBy;
(function (AggregationSortBy) {
    AggregationSortBy["FamilyVersion"] = "FAMILY_VERSION";
    AggregationSortBy["DownloadCount"] = "DOWNLOAD_COUNT";
})(AggregationSortBy = exports.AggregationSortBy || (exports.AggregationSortBy = {}));
(function (AggregationSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    AggregationSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AggregationSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(AggregationSortBy = exports.AggregationSortBy || (exports.AggregationSortBy = {}));
