// oci-jmsjavadownloads/lib/model/download-url-type.js
"use strict";
/**
 * Java Management Service Download API
 * The APIs for the <a href="https://docs.oracle.com/en-us/iaas/jms/doc/java-download.html">Java Download</a> feature of Java Management Service.
 * OpenAPI spec version: 20230601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DownloadUrlType = void 0;
/**
 * Supported types for the download URL.
 **/
var DownloadUrlType;
(function (DownloadUrlType) {
    DownloadUrlType["Oss"] = "OSS";
    DownloadUrlType["Cdn"] = "CDN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DownloadUrlType["UnknownValue"] = "UNKNOWN_VALUE";
})(DownloadUrlType = exports.DownloadUrlType || (exports.DownloadUrlType = {}));
(function (DownloadUrlType) {
    function getJsonObj(obj) {
        return obj;
    }
    DownloadUrlType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DownloadUrlType.getDeserializedJsonObj = getDeserializedJsonObj;
})(DownloadUrlType = exports.DownloadUrlType || (exports.DownloadUrlType = {}));
