// oci-jmsjavadownloads/lib/model/index.js
"use strict";
/**
 * Java Management Service Download API
 * The APIs for the <a href="https://docs.oracle.com/en-us/iaas/jms/doc/java-download.html">Java Download</a> feature of Java Management Service.
 * OpenAPI spec version: 20230601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateJavaLicenseAcceptanceRecordDetails = exports.UpdateJavaDownloadTokenDetails = exports.TokenSortBy = exports.TokenLifecycleDetails = exports.SortOrder = exports.RequestSummarizedJavaDownloadCountsDetails = exports.Principal = exports.OperationType = exports.OperationStatus = exports.LifecycleState = exports.LicenseType = exports.LicenseSortBy = exports.LicenseAcceptanceStatus = exports.LicenseAcceptanceSortBy = exports.JavaLicenseSummary = exports.JavaLicenseCollection = exports.JavaLicenseAcceptanceRecordSummary = exports.JavaLicenseAcceptanceRecordCollection = exports.JavaLicenseAcceptanceRecord = exports.JavaLicense = exports.JavaDownloadTokenSummary = exports.JavaDownloadTokenCollection = exports.JavaDownloadToken = exports.JavaDownloadReportSummary = exports.JavaDownloadReportSortBy = exports.JavaDownloadReportFormat = exports.JavaDownloadReportCollection = exports.JavaDownloadReport = exports.JavaDownloadRecordSummary = exports.JavaDownloadRecordSortBy = exports.JavaDownloadRecordCollection = exports.JavaDownloadRecord = exports.JavaDownloadCountAggregationType = exports.JavaDownloadCountAggregationCollection = exports.JavaDownloadCountAggregation = exports.GenerateArtifactDownloadUrlDetails = exports.DownloadUrlType = exports.DownloadUrl = exports.CreateJavaLicenseAcceptanceRecordDetails = exports.CreateJavaDownloadTokenDetails = exports.CreateJavaDownloadReportDetails = exports.ChecksumType = exports.AggregationSortBy = exports.ActionType = void 0;
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AggregationSortBy = __importStar(require("./aggregation-sort-by"));
exports.AggregationSortBy = AggregationSortBy.AggregationSortBy;
const ChecksumType = __importStar(require("./checksum-type"));
exports.ChecksumType = ChecksumType.ChecksumType;
const CreateJavaDownloadReportDetails = __importStar(require("./create-java-download-report-details"));
exports.CreateJavaDownloadReportDetails = CreateJavaDownloadReportDetails.CreateJavaDownloadReportDetails;
const CreateJavaDownloadTokenDetails = __importStar(require("./create-java-download-token-details"));
exports.CreateJavaDownloadTokenDetails = CreateJavaDownloadTokenDetails.CreateJavaDownloadTokenDetails;
const CreateJavaLicenseAcceptanceRecordDetails = __importStar(require("./create-java-license-acceptance-record-details"));
exports.CreateJavaLicenseAcceptanceRecordDetails = CreateJavaLicenseAcceptanceRecordDetails.CreateJavaLicenseAcceptanceRecordDetails;
const DownloadUrl = __importStar(require("./download-url"));
exports.DownloadUrl = DownloadUrl.DownloadUrl;
const DownloadUrlType = __importStar(require("./download-url-type"));
exports.DownloadUrlType = DownloadUrlType.DownloadUrlType;
const GenerateArtifactDownloadUrlDetails = __importStar(require("./generate-artifact-download-url-details"));
exports.GenerateArtifactDownloadUrlDetails = GenerateArtifactDownloadUrlDetails.GenerateArtifactDownloadUrlDetails;
const JavaDownloadCountAggregation = __importStar(require("./java-download-count-aggregation"));
exports.JavaDownloadCountAggregation = JavaDownloadCountAggregation.JavaDownloadCountAggregation;
const JavaDownloadCountAggregationCollection = __importStar(require("./java-download-count-aggregation-collection"));
exports.JavaDownloadCountAggregationCollection = JavaDownloadCountAggregationCollection.JavaDownloadCountAggregationCollection;
const JavaDownloadCountAggregationType = __importStar(require("./java-download-count-aggregation-type"));
exports.JavaDownloadCountAggregationType = JavaDownloadCountAggregationType.JavaDownloadCountAggregationType;
const JavaDownloadRecord = __importStar(require("./java-download-record"));
exports.JavaDownloadRecord = JavaDownloadRecord.JavaDownloadRecord;
const JavaDownloadRecordCollection = __importStar(require("./java-download-record-collection"));
exports.JavaDownloadRecordCollection = JavaDownloadRecordCollection.JavaDownloadRecordCollection;
const JavaDownloadRecordSortBy = __importStar(require("./java-download-record-sort-by"));
exports.JavaDownloadRecordSortBy = JavaDownloadRecordSortBy.JavaDownloadRecordSortBy;
const JavaDownloadRecordSummary = __importStar(require("./java-download-record-summary"));
exports.JavaDownloadRecordSummary = JavaDownloadRecordSummary.JavaDownloadRecordSummary;
const JavaDownloadReport = __importStar(require("./java-download-report"));
exports.JavaDownloadReport = JavaDownloadReport.JavaDownloadReport;
const JavaDownloadReportCollection = __importStar(require("./java-download-report-collection"));
exports.JavaDownloadReportCollection = JavaDownloadReportCollection.JavaDownloadReportCollection;
const JavaDownloadReportFormat = __importStar(require("./java-download-report-format"));
exports.JavaDownloadReportFormat = JavaDownloadReportFormat.JavaDownloadReportFormat;
const JavaDownloadReportSortBy = __importStar(require("./java-download-report-sort-by"));
exports.JavaDownloadReportSortBy = JavaDownloadReportSortBy.JavaDownloadReportSortBy;
const JavaDownloadReportSummary = __importStar(require("./java-download-report-summary"));
exports.JavaDownloadReportSummary = JavaDownloadReportSummary.JavaDownloadReportSummary;
const JavaDownloadToken = __importStar(require("./java-download-token"));
exports.JavaDownloadToken = JavaDownloadToken.JavaDownloadToken;
const JavaDownloadTokenCollection = __importStar(require("./java-download-token-collection"));
exports.JavaDownloadTokenCollection = JavaDownloadTokenCollection.JavaDownloadTokenCollection;
const JavaDownloadTokenSummary = __importStar(require("./java-download-token-summary"));
exports.JavaDownloadTokenSummary = JavaDownloadTokenSummary.JavaDownloadTokenSummary;
const JavaLicense = __importStar(require("./java-license"));
exports.JavaLicense = JavaLicense.JavaLicense;
const JavaLicenseAcceptanceRecord = __importStar(require("./java-license-acceptance-record"));
exports.JavaLicenseAcceptanceRecord = JavaLicenseAcceptanceRecord.JavaLicenseAcceptanceRecord;
const JavaLicenseAcceptanceRecordCollection = __importStar(require("./java-license-acceptance-record-collection"));
exports.JavaLicenseAcceptanceRecordCollection = JavaLicenseAcceptanceRecordCollection.JavaLicenseAcceptanceRecordCollection;
const JavaLicenseAcceptanceRecordSummary = __importStar(require("./java-license-acceptance-record-summary"));
exports.JavaLicenseAcceptanceRecordSummary = JavaLicenseAcceptanceRecordSummary.JavaLicenseAcceptanceRecordSummary;
const JavaLicenseCollection = __importStar(require("./java-license-collection"));
exports.JavaLicenseCollection = JavaLicenseCollection.JavaLicenseCollection;
const JavaLicenseSummary = __importStar(require("./java-license-summary"));
exports.JavaLicenseSummary = JavaLicenseSummary.JavaLicenseSummary;
const LicenseAcceptanceSortBy = __importStar(require("./license-acceptance-sort-by"));
exports.LicenseAcceptanceSortBy = LicenseAcceptanceSortBy.LicenseAcceptanceSortBy;
const LicenseAcceptanceStatus = __importStar(require("./license-acceptance-status"));
exports.LicenseAcceptanceStatus = LicenseAcceptanceStatus.LicenseAcceptanceStatus;
const LicenseSortBy = __importStar(require("./license-sort-by"));
exports.LicenseSortBy = LicenseSortBy.LicenseSortBy;
const LicenseType = __importStar(require("./license-type"));
exports.LicenseType = LicenseType.LicenseType;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const Principal = __importStar(require("./principal"));
exports.Principal = Principal.Principal;
const RequestSummarizedJavaDownloadCountsDetails = __importStar(require("./request-summarized-java-download-counts-details"));
exports.RequestSummarizedJavaDownloadCountsDetails = RequestSummarizedJavaDownloadCountsDetails.RequestSummarizedJavaDownloadCountsDetails;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const TokenLifecycleDetails = __importStar(require("./token-lifecycle-details"));
exports.TokenLifecycleDetails = TokenLifecycleDetails.TokenLifecycleDetails;
const TokenSortBy = __importStar(require("./token-sort-by"));
exports.TokenSortBy = TokenSortBy.TokenSortBy;
const UpdateJavaDownloadTokenDetails = __importStar(require("./update-java-download-token-details"));
exports.UpdateJavaDownloadTokenDetails = UpdateJavaDownloadTokenDetails.UpdateJavaDownloadTokenDetails;
const UpdateJavaLicenseAcceptanceRecordDetails = __importStar(require("./update-java-license-acceptance-record-details"));
exports.UpdateJavaLicenseAcceptanceRecordDetails = UpdateJavaLicenseAcceptanceRecordDetails.UpdateJavaLicenseAcceptanceRecordDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
