// oci-delegateaccesscontrol/lib/model/operation-type.js
"use strict";
/**
 * Oracle Delegate Access Control API
 * Oracle Delegate Access Control allows ExaCC and ExaCS customers to delegate management of their Exadata resources operators outside their tenancies.
With Delegate Access Control, Support Providers can deliver managed services using comprehensive and robust tooling built on the OCI platform.
Customers maintain control over who has access to the delegated resources in their tenancy and what actions can be taken.
Enterprises managing resources across multiple tenants can use Delegate Access Control to streamline management tasks.
Using logging service, customers can view a near real-time audit report of all actions performed by a Service Provider operator.

 * OpenAPI spec version: 20230801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateDelegationControl"] = "CREATE_DELEGATION_CONTROL";
    OperationType["UpdateDelegationControl"] = "UPDATE_DELEGATION_CONTROL";
    OperationType["DeleteDelegationControl"] = "DELETE_DELEGATION_CONTROL";
    OperationType["MoveDelegationControl"] = "MOVE_DELEGATION_CONTROL";
    OperationType["CreateDelegatedResourceAccessRequest"] = "CREATE_DELEGATED_RESOURCE_ACCESS_REQUEST";
    OperationType["ApproveDelegatedResourceAccessRequest"] = "APPROVE_DELEGATED_RESOURCE_ACCESS_REQUEST";
    OperationType["RejectDelegatedResourceAccessRequest"] = "REJECT_DELEGATED_RESOURCE_ACCESS_REQUEST";
    OperationType["RevokeDelegatedResourceAccessRequest"] = "REVOKE_DELEGATED_RESOURCE_ACCESS_REQUEST";
    OperationType["CreateDelegationSubscription"] = "CREATE_DELEGATION_SUBSCRIPTION";
    OperationType["UpdateDelegationSubscription"] = "UPDATE_DELEGATION_SUBSCRIPTION";
    OperationType["DeleteDelegationSubscription"] = "DELETE_DELEGATION_SUBSCRIPTION";
    OperationType["MoveDelegationSubscription"] = "MOVE_DELEGATION_SUBSCRIPTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
