// oci-generativeaiagentruntime/lib/model/index.js
"use strict";
/**
 * Generative AI Agents Client API
 * OCI Generative AI Agents is a fully managed service that combines the power of large language models (LLMs) with an intelligent retrieval system to create contextually relevant answers by searching your knowledge base, making your AI applications smart and efficient.

OCI Generative AI Agents supports several ways to onboard your data and then allows you and your customers to interact with your data using a chat interface or API.

Use the Generative AI Agents Client API to create and manage client chat sessions. A session represents an interactive conversation initiated by a user through an API to engage with an agent. It involves a series of exchanges where the user sends queries or prompts, and the agent responds with relevant information, actions, or assistance based on the user's input. The session persists for the duration of the interaction, maintaining context and continuity to provide coherent and meaningful responses throughout the conversation.

For creating and managing agents, knowledge bases, data sources, endpoints, and data ingestion jobs see the {@link #eNGenerative-ai-agentsLatest(ENGenerative-ai-agentsLatestRequest) eNGenerative-ai-agentsLatest}.

To learn more about the service, see the [Generative AI Agents documentation](https://docs.oracle.com/iaas/Content/generative-ai-agents/home.htm).

 * OpenAPI spec version: 20240531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ToolInvocationTrace = exports.RetrievalTrace = exports.PlanningTrace = exports.OciOpenSearchSourceLocation = exports.OciObjectStorageSourceLocation = exports.OciDatabaseSourceLocation = exports.HumanApprovalRequiredAction = exports.HumanApprovalPerformedAction = exports.GenerationTrace = exports.FunctionCallingRequiredAction = exports.FunctionCallingPerformedAction = exports.ExecutionTrace = exports.ErrorTrace = exports.UsageDetails = exports.Usage = exports.UpdateSessionDetails = exports.Trace = exports.SourceLocation = exports.SourceDetails = exports.Session = exports.RetrieveMetadataDetails = exports.RequiredAction = exports.PerformedAction = exports.ParagraphCitation = exports.Paragraph = exports.ModelParams = exports.ModelDetails = exports.MetadataFilter = exports.MessageContent = exports.Message = exports.KnowledgeBaseMetadataValue = exports.KnowledgeBaseMetadataSummary = exports.FunctionCall = exports.CreateSessionDetails = exports.Citation = exports.ChatResult = exports.ChatDetails = void 0;
const ChatDetails = __importStar(require("./chat-details"));
exports.ChatDetails = ChatDetails.ChatDetails;
const ChatResult = __importStar(require("./chat-result"));
exports.ChatResult = ChatResult.ChatResult;
const Citation = __importStar(require("./citation"));
exports.Citation = Citation.Citation;
const CreateSessionDetails = __importStar(require("./create-session-details"));
exports.CreateSessionDetails = CreateSessionDetails.CreateSessionDetails;
const FunctionCall = __importStar(require("./function-call"));
exports.FunctionCall = FunctionCall.FunctionCall;
const KnowledgeBaseMetadataSummary = __importStar(require("./knowledge-base-metadata-summary"));
exports.KnowledgeBaseMetadataSummary = KnowledgeBaseMetadataSummary.KnowledgeBaseMetadataSummary;
const KnowledgeBaseMetadataValue = __importStar(require("./knowledge-base-metadata-value"));
exports.KnowledgeBaseMetadataValue = KnowledgeBaseMetadataValue.KnowledgeBaseMetadataValue;
const Message = __importStar(require("./message"));
exports.Message = Message.Message;
const MessageContent = __importStar(require("./message-content"));
exports.MessageContent = MessageContent.MessageContent;
const MetadataFilter = __importStar(require("./metadata-filter"));
exports.MetadataFilter = MetadataFilter.MetadataFilter;
const ModelDetails = __importStar(require("./model-details"));
exports.ModelDetails = ModelDetails.ModelDetails;
const ModelParams = __importStar(require("./model-params"));
exports.ModelParams = ModelParams.ModelParams;
const Paragraph = __importStar(require("./paragraph"));
exports.Paragraph = Paragraph.Paragraph;
const ParagraphCitation = __importStar(require("./paragraph-citation"));
exports.ParagraphCitation = ParagraphCitation.ParagraphCitation;
const PerformedAction = __importStar(require("./performed-action"));
exports.PerformedAction = PerformedAction.PerformedAction;
const RequiredAction = __importStar(require("./required-action"));
exports.RequiredAction = RequiredAction.RequiredAction;
const RetrieveMetadataDetails = __importStar(require("./retrieve-metadata-details"));
exports.RetrieveMetadataDetails = RetrieveMetadataDetails.RetrieveMetadataDetails;
const Session = __importStar(require("./session"));
exports.Session = Session.Session;
const SourceDetails = __importStar(require("./source-details"));
exports.SourceDetails = SourceDetails.SourceDetails;
const SourceLocation = __importStar(require("./source-location"));
exports.SourceLocation = SourceLocation.SourceLocation;
const Trace = __importStar(require("./trace"));
exports.Trace = Trace.Trace;
const UpdateSessionDetails = __importStar(require("./update-session-details"));
exports.UpdateSessionDetails = UpdateSessionDetails.UpdateSessionDetails;
const Usage = __importStar(require("./usage"));
exports.Usage = Usage.Usage;
const UsageDetails = __importStar(require("./usage-details"));
exports.UsageDetails = UsageDetails.UsageDetails;
const ErrorTrace = __importStar(require("./error-trace"));
exports.ErrorTrace = ErrorTrace.ErrorTrace;
const ExecutionTrace = __importStar(require("./execution-trace"));
exports.ExecutionTrace = ExecutionTrace.ExecutionTrace;
const FunctionCallingPerformedAction = __importStar(require("./function-calling-performed-action"));
exports.FunctionCallingPerformedAction = FunctionCallingPerformedAction.FunctionCallingPerformedAction;
const FunctionCallingRequiredAction = __importStar(require("./function-calling-required-action"));
exports.FunctionCallingRequiredAction = FunctionCallingRequiredAction.FunctionCallingRequiredAction;
const GenerationTrace = __importStar(require("./generation-trace"));
exports.GenerationTrace = GenerationTrace.GenerationTrace;
const HumanApprovalPerformedAction = __importStar(require("./human-approval-performed-action"));
exports.HumanApprovalPerformedAction = HumanApprovalPerformedAction.HumanApprovalPerformedAction;
const HumanApprovalRequiredAction = __importStar(require("./human-approval-required-action"));
exports.HumanApprovalRequiredAction = HumanApprovalRequiredAction.HumanApprovalRequiredAction;
const OciDatabaseSourceLocation = __importStar(require("./oci-database-source-location"));
exports.OciDatabaseSourceLocation = OciDatabaseSourceLocation.OciDatabaseSourceLocation;
const OciObjectStorageSourceLocation = __importStar(require("./oci-object-storage-source-location"));
exports.OciObjectStorageSourceLocation = OciObjectStorageSourceLocation.OciObjectStorageSourceLocation;
const OciOpenSearchSourceLocation = __importStar(require("./oci-open-search-source-location"));
exports.OciOpenSearchSourceLocation = OciOpenSearchSourceLocation.OciOpenSearchSourceLocation;
const PlanningTrace = __importStar(require("./planning-trace"));
exports.PlanningTrace = PlanningTrace.PlanningTrace;
const RetrievalTrace = __importStar(require("./retrieval-trace"));
exports.RetrievalTrace = RetrievalTrace.RetrievalTrace;
const ToolInvocationTrace = __importStar(require("./tool-invocation-trace"));
exports.ToolInvocationTrace = ToolInvocationTrace.ToolInvocationTrace;
