// oci-emaildataplane/lib/model/index.js
"use strict";
/**
 * Email Delivery Submission API
 * Use the Email Delivery API to send high-volume and application-generated emails.
For more information, see [Overview of the Email Delivery Service](https://docs.oracle.com/iaas/Content/Email/Concepts/overview.htm).

 * OpenAPI spec version: 20220926
 * Contact: email-dev_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubmitEmailDetails = exports.Sender = exports.Recipients = exports.EmailSubmittedResponse = exports.EmailAddress = void 0;
const EmailAddress = __importStar(require("./email-address"));
exports.EmailAddress = EmailAddress.EmailAddress;
const EmailSubmittedResponse = __importStar(require("./email-submitted-response"));
exports.EmailSubmittedResponse = EmailSubmittedResponse.EmailSubmittedResponse;
const Recipients = __importStar(require("./recipients"));
exports.Recipients = Recipients.Recipients;
const Sender = __importStar(require("./sender"));
exports.Sender = Sender.Sender;
const SubmitEmailDetails = __importStar(require("./submit-email-details"));
exports.SubmitEmailDetails = SubmitEmailDetails.SubmitEmailDetails;
