// oci-tenantmanagercontrolplane/lib/model/billing-model.js
"use strict";
/**
 * Organizations API
 * Use the Organizations API to consolidate multiple OCI tenancies into an organization, and centrally manage your tenancies and organization resources. For more information, see [Organization Management Overview](https://docs.oracle.com/iaas/Content/General/Concepts/organization_management_overview.htm).
 * OpenAPI spec version: 20230401
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingModel = void 0;
/**
 * Billing model.
 **/
var BillingModel;
(function (BillingModel) {
    BillingModel["Commitment"] = "COMMITMENT";
    BillingModel["Paygo"] = "PAYGO";
    BillingModel["Promotion"] = "PROMOTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    BillingModel["UnknownValue"] = "UNKNOWN_VALUE";
})(BillingModel = exports.BillingModel || (exports.BillingModel = {}));
(function (BillingModel) {
    function getJsonObj(obj) {
        return obj;
    }
    BillingModel.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    BillingModel.getDeserializedJsonObj = getDeserializedJsonObj;
})(BillingModel = exports.BillingModel || (exports.BillingModel = {}));
