// oci-tenantmanagercontrolplane/lib/model/classic-subscription-lifecycle-state.js
"use strict";
/**
 * Organizations API
 * Use the Organizations API to consolidate multiple OCI tenancies into an organization, and centrally manage your tenancies and organization resources. For more information, see [Organization Management Overview](https://docs.oracle.com/iaas/Content/General/Concepts/organization_management_overview.htm).
 * OpenAPI spec version: 20230401
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClassicSubscriptionLifecycleState = void 0;
/**
 * Lifecycle state of a V1 (legacy schema version) subscription.
 **/
var ClassicSubscriptionLifecycleState;
(function (ClassicSubscriptionLifecycleState) {
    ClassicSubscriptionLifecycleState["Creating"] = "CREATING";
    ClassicSubscriptionLifecycleState["Active"] = "ACTIVE";
    ClassicSubscriptionLifecycleState["Inactive"] = "INACTIVE";
    ClassicSubscriptionLifecycleState["Updating"] = "UPDATING";
    ClassicSubscriptionLifecycleState["Deleting"] = "DELETING";
    ClassicSubscriptionLifecycleState["Deleted"] = "DELETED";
    ClassicSubscriptionLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ClassicSubscriptionLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(ClassicSubscriptionLifecycleState = exports.ClassicSubscriptionLifecycleState || (exports.ClassicSubscriptionLifecycleState = {}));
(function (ClassicSubscriptionLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    ClassicSubscriptionLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ClassicSubscriptionLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(ClassicSubscriptionLifecycleState = exports.ClassicSubscriptionLifecycleState || (exports.ClassicSubscriptionLifecycleState = {}));
