// oci-marketplacepublisher/lib/model/index.js
"use strict";
/**
 * MarketplacePublisherService API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Market = exports.ListingType = exports.ListingSummary = exports.ListingRevisionSummary = exports.ListingRevisionPackageSummary = exports.ListingRevisionPackageCollection = exports.ListingRevisionPackage = exports.ListingRevisionNoteSummary = exports.ListingRevisionNoteCollection = exports.ListingRevisionNote = exports.ListingRevisionIconAttachment = exports.ListingRevisionCollection = exports.ListingRevisionAttachmentSummary = exports.ListingRevisionAttachmentCollection = exports.ListingRevisionAttachment = exports.ListingRevision = exports.ListingCollection = exports.Listing = exports.LanguageItem = exports.HelmChartImageDetails = exports.CustomField = exports.CreateVideoDetails = exports.CreateTermDetails = exports.CreateOfferDetails = exports.CreateListingRevisionPackageDetails = exports.CreateListingRevisionNoteDetails = exports.CreateListingRevisionDetails = exports.CreateListingRevisionAttachmentDetails = exports.CreateListingDetails = exports.CreateHelmChartImageDetails = exports.CreateContainerImageDetails = exports.CreateAttachmentDetails = exports.CreateArtifactDetails = exports.ContainerImageDetails = exports.Contact = exports.ChangeTermCompartmentDetails = exports.ChangeListingCompartmentDetails = exports.ChangeArtifactCompartmentDetails = exports.CategorySummary = exports.CategoryCollection = exports.Category = exports.BuyerInformation = exports.AttachmentSummary = exports.AttachmentCollection = exports.Attachment = exports.ArtifactTypeEnum = exports.ArtifactSummary = exports.ArtifactCollection = exports.Artifact = exports.ActionType = void 0;
exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.VersionDetails = exports.ValidationStatus = exports.UploadData = exports.UpdateTermVersionDetails = exports.UpdateTermDetails = exports.UpdateOfferDetails = exports.UpdateListingRevisionPackageDetails = exports.UpdateListingRevisionDetails = exports.UpdateListingRevisionAttachmentDetails = exports.UpdateListingDetails = exports.UpdateHelmChartImageDetails = exports.UpdateContainerImageDetails = exports.UpdateArtifactDetails = exports.TermVersionSummary = exports.TermVersionCollection = exports.TermVersionAttachment = exports.TermVersion = exports.TermSummary = exports.TermCollection = exports.Term = exports.SupportContact = exports.SubmitListingRevisionForReviewDetails = exports.SortOrder = exports.SellerInformation = exports.ResourceBundle = exports.PublisherSummary = exports.PublisherCollection = exports.Publisher = exports.PublishListingRevisionAsPrivateDetails = exports.PublicationStatus = exports.ProductSummary = exports.ProductCollection = exports.Product = exports.PrivateOfferAccountDetails = exports.Pricing = exports.PackageType = exports.OpnMembership = exports.OperationType = exports.OperationStatus = exports.OfferSummary = exports.OfferInternalDetail = exports.OfferCollection = exports.Offer = exports.NamedLink = exports.MarketSummary = exports.MarketCollection = void 0;
exports.VideoAttachment = exports.UpdateVideoAttachmentDetails = exports.UpdateScreenShotAttachmentDetails = exports.UpdateRelatedDocumentAttachmentDetails = exports.UpdateKubernetesImageArtifactDetails = exports.UpdateContainerImageArtifactDetails = exports.ScreenShotAttachment = exports.RelatedDocumentAttachment = exports.KubernetesImageArtifact = exports.HelmChartPackage = exports.CreateVideoAttachmentDetails = exports.CreateScreenShotAttachmentDetails = exports.CreateRelatedDocumentAttachmentDetails = exports.CreateKubernetesImageArtifactDetails = exports.CreateContainerImageArtifactDetails = exports.ContainerPackage = exports.ContainerImageArtifact = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const Artifact = __importStar(require("./artifact"));
exports.Artifact = Artifact.Artifact;
const ArtifactCollection = __importStar(require("./artifact-collection"));
exports.ArtifactCollection = ArtifactCollection.ArtifactCollection;
const ArtifactSummary = __importStar(require("./artifact-summary"));
exports.ArtifactSummary = ArtifactSummary.ArtifactSummary;
const ArtifactTypeEnum = __importStar(require("./artifact-type-enum"));
exports.ArtifactTypeEnum = ArtifactTypeEnum.ArtifactTypeEnum;
const Attachment = __importStar(require("./attachment"));
exports.Attachment = Attachment.Attachment;
const AttachmentCollection = __importStar(require("./attachment-collection"));
exports.AttachmentCollection = AttachmentCollection.AttachmentCollection;
const AttachmentSummary = __importStar(require("./attachment-summary"));
exports.AttachmentSummary = AttachmentSummary.AttachmentSummary;
const BuyerInformation = __importStar(require("./buyer-information"));
exports.BuyerInformation = BuyerInformation.BuyerInformation;
const Category = __importStar(require("./category"));
exports.Category = Category.Category;
const CategoryCollection = __importStar(require("./category-collection"));
exports.CategoryCollection = CategoryCollection.CategoryCollection;
const CategorySummary = __importStar(require("./category-summary"));
exports.CategorySummary = CategorySummary.CategorySummary;
const ChangeArtifactCompartmentDetails = __importStar(require("./change-artifact-compartment-details"));
exports.ChangeArtifactCompartmentDetails = ChangeArtifactCompartmentDetails.ChangeArtifactCompartmentDetails;
const ChangeListingCompartmentDetails = __importStar(require("./change-listing-compartment-details"));
exports.ChangeListingCompartmentDetails = ChangeListingCompartmentDetails.ChangeListingCompartmentDetails;
const ChangeTermCompartmentDetails = __importStar(require("./change-term-compartment-details"));
exports.ChangeTermCompartmentDetails = ChangeTermCompartmentDetails.ChangeTermCompartmentDetails;
const Contact = __importStar(require("./contact"));
exports.Contact = Contact.Contact;
const ContainerImageDetails = __importStar(require("./container-image-details"));
exports.ContainerImageDetails = ContainerImageDetails.ContainerImageDetails;
const CreateArtifactDetails = __importStar(require("./create-artifact-details"));
exports.CreateArtifactDetails = CreateArtifactDetails.CreateArtifactDetails;
const CreateAttachmentDetails = __importStar(require("./create-attachment-details"));
exports.CreateAttachmentDetails = CreateAttachmentDetails.CreateAttachmentDetails;
const CreateContainerImageDetails = __importStar(require("./create-container-image-details"));
exports.CreateContainerImageDetails = CreateContainerImageDetails.CreateContainerImageDetails;
const CreateHelmChartImageDetails = __importStar(require("./create-helm-chart-image-details"));
exports.CreateHelmChartImageDetails = CreateHelmChartImageDetails.CreateHelmChartImageDetails;
const CreateListingDetails = __importStar(require("./create-listing-details"));
exports.CreateListingDetails = CreateListingDetails.CreateListingDetails;
const CreateListingRevisionAttachmentDetails = __importStar(require("./create-listing-revision-attachment-details"));
exports.CreateListingRevisionAttachmentDetails = CreateListingRevisionAttachmentDetails.CreateListingRevisionAttachmentDetails;
const CreateListingRevisionDetails = __importStar(require("./create-listing-revision-details"));
exports.CreateListingRevisionDetails = CreateListingRevisionDetails.CreateListingRevisionDetails;
const CreateListingRevisionNoteDetails = __importStar(require("./create-listing-revision-note-details"));
exports.CreateListingRevisionNoteDetails = CreateListingRevisionNoteDetails.CreateListingRevisionNoteDetails;
const CreateListingRevisionPackageDetails = __importStar(require("./create-listing-revision-package-details"));
exports.CreateListingRevisionPackageDetails = CreateListingRevisionPackageDetails.CreateListingRevisionPackageDetails;
const CreateOfferDetails = __importStar(require("./create-offer-details"));
exports.CreateOfferDetails = CreateOfferDetails.CreateOfferDetails;
const CreateTermDetails = __importStar(require("./create-term-details"));
exports.CreateTermDetails = CreateTermDetails.CreateTermDetails;
const CreateVideoDetails = __importStar(require("./create-video-details"));
exports.CreateVideoDetails = CreateVideoDetails.CreateVideoDetails;
const CustomField = __importStar(require("./custom-field"));
exports.CustomField = CustomField.CustomField;
const HelmChartImageDetails = __importStar(require("./helm-chart-image-details"));
exports.HelmChartImageDetails = HelmChartImageDetails.HelmChartImageDetails;
const LanguageItem = __importStar(require("./language-item"));
exports.LanguageItem = LanguageItem.LanguageItem;
const Listing = __importStar(require("./listing"));
exports.Listing = Listing.Listing;
const ListingCollection = __importStar(require("./listing-collection"));
exports.ListingCollection = ListingCollection.ListingCollection;
const ListingRevision = __importStar(require("./listing-revision"));
exports.ListingRevision = ListingRevision.ListingRevision;
const ListingRevisionAttachment = __importStar(require("./listing-revision-attachment"));
exports.ListingRevisionAttachment = ListingRevisionAttachment.ListingRevisionAttachment;
const ListingRevisionAttachmentCollection = __importStar(require("./listing-revision-attachment-collection"));
exports.ListingRevisionAttachmentCollection = ListingRevisionAttachmentCollection.ListingRevisionAttachmentCollection;
const ListingRevisionAttachmentSummary = __importStar(require("./listing-revision-attachment-summary"));
exports.ListingRevisionAttachmentSummary = ListingRevisionAttachmentSummary.ListingRevisionAttachmentSummary;
const ListingRevisionCollection = __importStar(require("./listing-revision-collection"));
exports.ListingRevisionCollection = ListingRevisionCollection.ListingRevisionCollection;
const ListingRevisionIconAttachment = __importStar(require("./listing-revision-icon-attachment"));
exports.ListingRevisionIconAttachment = ListingRevisionIconAttachment.ListingRevisionIconAttachment;
const ListingRevisionNote = __importStar(require("./listing-revision-note"));
exports.ListingRevisionNote = ListingRevisionNote.ListingRevisionNote;
const ListingRevisionNoteCollection = __importStar(require("./listing-revision-note-collection"));
exports.ListingRevisionNoteCollection = ListingRevisionNoteCollection.ListingRevisionNoteCollection;
const ListingRevisionNoteSummary = __importStar(require("./listing-revision-note-summary"));
exports.ListingRevisionNoteSummary = ListingRevisionNoteSummary.ListingRevisionNoteSummary;
const ListingRevisionPackage = __importStar(require("./listing-revision-package"));
exports.ListingRevisionPackage = ListingRevisionPackage.ListingRevisionPackage;
const ListingRevisionPackageCollection = __importStar(require("./listing-revision-package-collection"));
exports.ListingRevisionPackageCollection = ListingRevisionPackageCollection.ListingRevisionPackageCollection;
const ListingRevisionPackageSummary = __importStar(require("./listing-revision-package-summary"));
exports.ListingRevisionPackageSummary = ListingRevisionPackageSummary.ListingRevisionPackageSummary;
const ListingRevisionSummary = __importStar(require("./listing-revision-summary"));
exports.ListingRevisionSummary = ListingRevisionSummary.ListingRevisionSummary;
const ListingSummary = __importStar(require("./listing-summary"));
exports.ListingSummary = ListingSummary.ListingSummary;
const ListingType = __importStar(require("./listing-type"));
exports.ListingType = ListingType.ListingType;
const Market = __importStar(require("./market"));
exports.Market = Market.Market;
const MarketCollection = __importStar(require("./market-collection"));
exports.MarketCollection = MarketCollection.MarketCollection;
const MarketSummary = __importStar(require("./market-summary"));
exports.MarketSummary = MarketSummary.MarketSummary;
const NamedLink = __importStar(require("./named-link"));
exports.NamedLink = NamedLink.NamedLink;
const Offer = __importStar(require("./offer"));
exports.Offer = Offer.Offer;
const OfferCollection = __importStar(require("./offer-collection"));
exports.OfferCollection = OfferCollection.OfferCollection;
const OfferInternalDetail = __importStar(require("./offer-internal-detail"));
exports.OfferInternalDetail = OfferInternalDetail.OfferInternalDetail;
const OfferSummary = __importStar(require("./offer-summary"));
exports.OfferSummary = OfferSummary.OfferSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OpnMembership = __importStar(require("./opn-membership"));
exports.OpnMembership = OpnMembership.OpnMembership;
const PackageType = __importStar(require("./package-type"));
exports.PackageType = PackageType.PackageType;
const Pricing = __importStar(require("./pricing"));
exports.Pricing = Pricing.Pricing;
const PrivateOfferAccountDetails = __importStar(require("./private-offer-account-details"));
exports.PrivateOfferAccountDetails = PrivateOfferAccountDetails.PrivateOfferAccountDetails;
const Product = __importStar(require("./product"));
exports.Product = Product.Product;
const ProductCollection = __importStar(require("./product-collection"));
exports.ProductCollection = ProductCollection.ProductCollection;
const ProductSummary = __importStar(require("./product-summary"));
exports.ProductSummary = ProductSummary.ProductSummary;
const PublicationStatus = __importStar(require("./publication-status"));
exports.PublicationStatus = PublicationStatus.PublicationStatus;
const PublishListingRevisionAsPrivateDetails = __importStar(require("./publish-listing-revision-as-private-details"));
exports.PublishListingRevisionAsPrivateDetails = PublishListingRevisionAsPrivateDetails.PublishListingRevisionAsPrivateDetails;
const Publisher = __importStar(require("./publisher"));
exports.Publisher = Publisher.Publisher;
const PublisherCollection = __importStar(require("./publisher-collection"));
exports.PublisherCollection = PublisherCollection.PublisherCollection;
const PublisherSummary = __importStar(require("./publisher-summary"));
exports.PublisherSummary = PublisherSummary.PublisherSummary;
const ResourceBundle = __importStar(require("./resource-bundle"));
exports.ResourceBundle = ResourceBundle.ResourceBundle;
const SellerInformation = __importStar(require("./seller-information"));
exports.SellerInformation = SellerInformation.SellerInformation;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const SubmitListingRevisionForReviewDetails = __importStar(require("./submit-listing-revision-for-review-details"));
exports.SubmitListingRevisionForReviewDetails = SubmitListingRevisionForReviewDetails.SubmitListingRevisionForReviewDetails;
const SupportContact = __importStar(require("./support-contact"));
exports.SupportContact = SupportContact.SupportContact;
const Term = __importStar(require("./term"));
exports.Term = Term.Term;
const TermCollection = __importStar(require("./term-collection"));
exports.TermCollection = TermCollection.TermCollection;
const TermSummary = __importStar(require("./term-summary"));
exports.TermSummary = TermSummary.TermSummary;
const TermVersion = __importStar(require("./term-version"));
exports.TermVersion = TermVersion.TermVersion;
const TermVersionAttachment = __importStar(require("./term-version-attachment"));
exports.TermVersionAttachment = TermVersionAttachment.TermVersionAttachment;
const TermVersionCollection = __importStar(require("./term-version-collection"));
exports.TermVersionCollection = TermVersionCollection.TermVersionCollection;
const TermVersionSummary = __importStar(require("./term-version-summary"));
exports.TermVersionSummary = TermVersionSummary.TermVersionSummary;
const UpdateArtifactDetails = __importStar(require("./update-artifact-details"));
exports.UpdateArtifactDetails = UpdateArtifactDetails.UpdateArtifactDetails;
const UpdateContainerImageDetails = __importStar(require("./update-container-image-details"));
exports.UpdateContainerImageDetails = UpdateContainerImageDetails.UpdateContainerImageDetails;
const UpdateHelmChartImageDetails = __importStar(require("./update-helm-chart-image-details"));
exports.UpdateHelmChartImageDetails = UpdateHelmChartImageDetails.UpdateHelmChartImageDetails;
const UpdateListingDetails = __importStar(require("./update-listing-details"));
exports.UpdateListingDetails = UpdateListingDetails.UpdateListingDetails;
const UpdateListingRevisionAttachmentDetails = __importStar(require("./update-listing-revision-attachment-details"));
exports.UpdateListingRevisionAttachmentDetails = UpdateListingRevisionAttachmentDetails.UpdateListingRevisionAttachmentDetails;
const UpdateListingRevisionDetails = __importStar(require("./update-listing-revision-details"));
exports.UpdateListingRevisionDetails = UpdateListingRevisionDetails.UpdateListingRevisionDetails;
const UpdateListingRevisionPackageDetails = __importStar(require("./update-listing-revision-package-details"));
exports.UpdateListingRevisionPackageDetails = UpdateListingRevisionPackageDetails.UpdateListingRevisionPackageDetails;
const UpdateOfferDetails = __importStar(require("./update-offer-details"));
exports.UpdateOfferDetails = UpdateOfferDetails.UpdateOfferDetails;
const UpdateTermDetails = __importStar(require("./update-term-details"));
exports.UpdateTermDetails = UpdateTermDetails.UpdateTermDetails;
const UpdateTermVersionDetails = __importStar(require("./update-term-version-details"));
exports.UpdateTermVersionDetails = UpdateTermVersionDetails.UpdateTermVersionDetails;
const UploadData = __importStar(require("./upload-data"));
exports.UploadData = UploadData.UploadData;
const ValidationStatus = __importStar(require("./validation-status"));
exports.ValidationStatus = ValidationStatus.ValidationStatus;
const VersionDetails = __importStar(require("./version-details"));
exports.VersionDetails = VersionDetails.VersionDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const ContainerImageArtifact = __importStar(require("./container-image-artifact"));
exports.ContainerImageArtifact = ContainerImageArtifact.ContainerImageArtifact;
const ContainerPackage = __importStar(require("./container-package"));
exports.ContainerPackage = ContainerPackage.ContainerPackage;
const CreateContainerImageArtifactDetails = __importStar(require("./create-container-image-artifact-details"));
exports.CreateContainerImageArtifactDetails = CreateContainerImageArtifactDetails.CreateContainerImageArtifactDetails;
const CreateKubernetesImageArtifactDetails = __importStar(require("./create-kubernetes-image-artifact-details"));
exports.CreateKubernetesImageArtifactDetails = CreateKubernetesImageArtifactDetails.CreateKubernetesImageArtifactDetails;
const CreateRelatedDocumentAttachmentDetails = __importStar(require("./create-related-document-attachment-details"));
exports.CreateRelatedDocumentAttachmentDetails = CreateRelatedDocumentAttachmentDetails.CreateRelatedDocumentAttachmentDetails;
const CreateScreenShotAttachmentDetails = __importStar(require("./create-screen-shot-attachment-details"));
exports.CreateScreenShotAttachmentDetails = CreateScreenShotAttachmentDetails.CreateScreenShotAttachmentDetails;
const CreateVideoAttachmentDetails = __importStar(require("./create-video-attachment-details"));
exports.CreateVideoAttachmentDetails = CreateVideoAttachmentDetails.CreateVideoAttachmentDetails;
const HelmChartPackage = __importStar(require("./helm-chart-package"));
exports.HelmChartPackage = HelmChartPackage.HelmChartPackage;
const KubernetesImageArtifact = __importStar(require("./kubernetes-image-artifact"));
exports.KubernetesImageArtifact = KubernetesImageArtifact.KubernetesImageArtifact;
const RelatedDocumentAttachment = __importStar(require("./related-document-attachment"));
exports.RelatedDocumentAttachment = RelatedDocumentAttachment.RelatedDocumentAttachment;
const ScreenShotAttachment = __importStar(require("./screen-shot-attachment"));
exports.ScreenShotAttachment = ScreenShotAttachment.ScreenShotAttachment;
const UpdateContainerImageArtifactDetails = __importStar(require("./update-container-image-artifact-details"));
exports.UpdateContainerImageArtifactDetails = UpdateContainerImageArtifactDetails.UpdateContainerImageArtifactDetails;
const UpdateKubernetesImageArtifactDetails = __importStar(require("./update-kubernetes-image-artifact-details"));
exports.UpdateKubernetesImageArtifactDetails = UpdateKubernetesImageArtifactDetails.UpdateKubernetesImageArtifactDetails;
const UpdateRelatedDocumentAttachmentDetails = __importStar(require("./update-related-document-attachment-details"));
exports.UpdateRelatedDocumentAttachmentDetails = UpdateRelatedDocumentAttachmentDetails.UpdateRelatedDocumentAttachmentDetails;
const UpdateScreenShotAttachmentDetails = __importStar(require("./update-screen-shot-attachment-details"));
exports.UpdateScreenShotAttachmentDetails = UpdateScreenShotAttachmentDetails.UpdateScreenShotAttachmentDetails;
const UpdateVideoAttachmentDetails = __importStar(require("./update-video-attachment-details"));
exports.UpdateVideoAttachmentDetails = UpdateVideoAttachmentDetails.UpdateVideoAttachmentDetails;
const VideoAttachment = __importStar(require("./video-attachment"));
exports.VideoAttachment = VideoAttachment.VideoAttachment;
