// oci-marketplacepublisher/lib/model/listing-revision-package.js
"use strict";
/**
 * MarketplacePublisherService API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingRevisionPackage = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var ListingRevisionPackage;
(function (ListingRevisionPackage) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ListingRevisionPackage.LifecycleState || (ListingRevisionPackage.LifecycleState = {}));
    let Status;
    (function (Status) {
        Status["New"] = "NEW";
        Status["PublishInProgress"] = "PUBLISH_IN_PROGRESS";
        Status["UnpublishInProgress"] = "UNPUBLISH_IN_PROGRESS";
        Status["PublishFailed"] = "PUBLISH_FAILED";
        Status["Published"] = "PUBLISHED";
        Status["PublishedAsPrivate"] = "PUBLISHED_AS_PRIVATE";
        Status["Unpublished"] = "UNPUBLISHED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = ListingRevisionPackage.Status || (ListingRevisionPackage.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "packageType" in obj && obj.packageType) {
            switch (obj.packageType) {
                case "HELM_CHART":
                    return model.HelmChartPackage.getJsonObj(jsonObj, true);
                case "CONTAINER_IMAGE":
                    return model.ContainerPackage.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.packageType}`);
            }
        }
        return jsonObj;
    }
    ListingRevisionPackage.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "packageType" in obj && obj.packageType) {
            switch (obj.packageType) {
                case "HELM_CHART":
                    return model.HelmChartPackage.getDeserializedJsonObj(jsonObj, true);
                case "CONTAINER_IMAGE":
                    return model.ContainerPackage.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.packageType}`);
            }
        }
        return jsonObj;
    }
    ListingRevisionPackage.getDeserializedJsonObj = getDeserializedJsonObj;
})(ListingRevisionPackage = exports.ListingRevisionPackage || (exports.ListingRevisionPackage = {}));
