// oci-marketplacepublisher/lib/model/listing-revision-note.js
"use strict";
/**
 * MarketplacePublisherService API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingRevisionNote = void 0;
var ListingRevisionNote;
(function (ListingRevisionNote) {
    let NoteSource;
    (function (NoteSource) {
        NoteSource["Publisher"] = "PUBLISHER";
        NoteSource["Administrator"] = "ADMINISTRATOR";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        NoteSource["UnknownValue"] = "UNKNOWN_VALUE";
    })(NoteSource = ListingRevisionNote.NoteSource || (ListingRevisionNote.NoteSource = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ListingRevisionNote.LifecycleState || (ListingRevisionNote.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ListingRevisionNote.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ListingRevisionNote.getDeserializedJsonObj = getDeserializedJsonObj;
})(ListingRevisionNote = exports.ListingRevisionNote || (exports.ListingRevisionNote = {}));
