// oci-marketplacepublisher/lib/model/listing-revision-attachment.js
"use strict";
/**
 * MarketplacePublisherService API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingRevisionAttachment = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var ListingRevisionAttachment;
(function (ListingRevisionAttachment) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ListingRevisionAttachment.LifecycleState || (ListingRevisionAttachment.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "attachmentType" in obj && obj.attachmentType) {
            switch (obj.attachmentType) {
                case "RELATED_DOCUMENT":
                    return model.RelatedDocumentAttachment.getJsonObj(jsonObj, true);
                case "SCREENSHOT":
                    return model.ScreenShotAttachment.getJsonObj(jsonObj, true);
                case "VIDEO":
                    return model.VideoAttachment.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.attachmentType}`);
            }
        }
        return jsonObj;
    }
    ListingRevisionAttachment.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "attachmentType" in obj && obj.attachmentType) {
            switch (obj.attachmentType) {
                case "RELATED_DOCUMENT":
                    return model.RelatedDocumentAttachment.getDeserializedJsonObj(jsonObj, true);
                case "SCREENSHOT":
                    return model.ScreenShotAttachment.getDeserializedJsonObj(jsonObj, true);
                case "VIDEO":
                    return model.VideoAttachment.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.attachmentType}`);
            }
        }
        return jsonObj;
    }
    ListingRevisionAttachment.getDeserializedJsonObj = getDeserializedJsonObj;
})(ListingRevisionAttachment = exports.ListingRevisionAttachment || (exports.ListingRevisionAttachment = {}));
