// oci-marketplacepublisher/lib/model/operation-type.js
"use strict";
/**
 * MarketplacePublisherService API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateOffer"] = "CREATE_OFFER";
    OperationType["UpdateOffer"] = "UPDATE_OFFER";
    OperationType["DeleteOffer"] = "DELETE_OFFER";
    OperationType["MoveOffer"] = "MOVE_OFFER";
    OperationType["DeleteOfferAttachment"] = "DELETE_OFFER_ATTACHMENT";
    OperationType["PublishListingRevisionPackage"] = "PUBLISH_LISTING_REVISION_PACKAGE";
    OperationType["UnpublishListingRevisionPackage"] = "UNPUBLISH_LISTING_REVISION_PACKAGE";
    OperationType["PublishListingRevision"] = "PUBLISH_LISTING_REVISION";
    OperationType["PublishListingRevisionAsPrivate"] = "PUBLISH_LISTING_REVISION_AS_PRIVATE";
    OperationType["WithdrawListingRevision"] = "WITHDRAW_LISTING_REVISION";
    OperationType["CloneListingRevision"] = "CLONE_LISTING_REVISION";
    OperationType["CascadingDeleteListing"] = "CASCADING_DELETE_LISTING";
    OperationType["CascadingDeleteListingRevision"] = "CASCADING_DELETE_LISTING_REVISION";
    OperationType["MarkAsDefaultListingRevisionPackage"] = "MARK_AS_DEFAULT_LISTING_REVISION_PACKAGE";
    OperationType["ChangeListingCompartment"] = "CHANGE_LISTING_COMPARTMENT";
    OperationType["CreateArtifact"] = "CREATE_ARTIFACT";
    OperationType["ValidateAndPublishArtifact"] = "VALIDATE_AND_PUBLISH_ARTIFACT";
    OperationType["ChangeArtifactCompartment"] = "CHANGE_ARTIFACT_COMPARTMENT";
    OperationType["ChangeTermCompartment"] = "CHANGE_TERM_COMPARTMENT";
    OperationType["DeleteArtifact"] = "DELETE_ARTIFACT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
