// oci-marketplacepublisher/lib/model/publication-status.js
"use strict";
/**
 * MarketplacePublisherService API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicationStatus = void 0;
/**
 * Publication status for the artifact.
 *
 **/
var PublicationStatus;
(function (PublicationStatus) {
    PublicationStatus["PublicationInProgress"] = "PUBLICATION_IN_PROGRESS";
    PublicationStatus["PublicationCompleted"] = "PUBLICATION_COMPLETED";
    PublicationStatus["PublicationFailed"] = "PUBLICATION_FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PublicationStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(PublicationStatus = exports.PublicationStatus || (exports.PublicationStatus = {}));
(function (PublicationStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    PublicationStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PublicationStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(PublicationStatus = exports.PublicationStatus || (exports.PublicationStatus = {}));
