// oci-marketplacepublisher/lib/model/validation-status.js
"use strict";
/**
 * MarketplacePublisherService API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationStatus = void 0;
/**
 * Validation status for the artifact.
 *
 **/
var ValidationStatus;
(function (ValidationStatus) {
    ValidationStatus["ValidationInProgress"] = "VALIDATION_IN_PROGRESS";
    ValidationStatus["ValidationFailed"] = "VALIDATION_FAILED";
    ValidationStatus["ValidationCompleted"] = "VALIDATION_COMPLETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ValidationStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(ValidationStatus = exports.ValidationStatus || (exports.ValidationStatus = {}));
(function (ValidationStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    ValidationStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ValidationStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(ValidationStatus = exports.ValidationStatus || (exports.ValidationStatus = {}));
