// oci-apmtraces/lib/request/list-scheduled-queries-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListScheduledQueriesRequest = void 0;
var ListScheduledQueriesRequest;
(function (ListScheduledQueriesRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["ScheduledQueryName"] = "SCHEDULED_QUERY_NAME";
        SortBy["ScheduledQueryType"] = "SCHEDULED_QUERY_TYPE";
        SortBy["ScheduledQuerySubType"] = "SCHEDULED_QUERY_SUB_TYPE";
        SortBy["ScheduledQueryNextRun"] = "SCHEDULED_QUERY_NEXT_RUN";
        SortBy["ScheduledQuerySchedule"] = "SCHEDULED_QUERY_SCHEDULE";
    })(SortBy = ListScheduledQueriesRequest.SortBy || (ListScheduledQueriesRequest.SortBy = {}));
})(ListScheduledQueriesRequest = exports.ListScheduledQueriesRequest || (exports.ListScheduledQueriesRequest = {}));
