// oci-opsi/lib/request/list-awr-hub-objects-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAwrHubObjectsRequest = void 0;
var ListAwrHubObjectsRequest;
(function (ListAwrHubObjectsRequest) {
    let Fields;
    (function (Fields) {
        Fields["Name"] = "name";
        Fields["Size"] = "size";
        Fields["Etag"] = "etag";
        Fields["TimeCreated"] = "timeCreated";
        Fields["Md5"] = "md5";
        Fields["ArchivalState"] = "archivalState";
        Fields["TimeModified"] = "timeModified";
        Fields["StorageTier"] = "storageTier";
    })(Fields = ListAwrHubObjectsRequest.Fields || (ListAwrHubObjectsRequest.Fields = {}));
})(ListAwrHubObjectsRequest = exports.ListAwrHubObjectsRequest || (exports.ListAwrHubObjectsRequest = {}));
