// oci-opsi/lib/request/list-warehouse-data-objects-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWarehouseDataObjectsRequest = void 0;
var ListWarehouseDataObjectsRequest;
(function (ListWarehouseDataObjectsRequest) {
    let WarehouseType;
    (function (WarehouseType) {
        WarehouseType["AwrHubs"] = "awrHubs";
    })(WarehouseType = ListWarehouseDataObjectsRequest.WarehouseType || (ListWarehouseDataObjectsRequest.WarehouseType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["DataObjectType"] = "dataObjectType";
        SortBy["Name"] = "name";
        SortBy["Owner"] = "owner";
    })(SortBy = ListWarehouseDataObjectsRequest.SortBy || (ListWarehouseDataObjectsRequest.SortBy = {}));
    let SummaryField;
    (function (SummaryField) {
        SummaryField["Details"] = "details";
    })(SummaryField = ListWarehouseDataObjectsRequest.SummaryField || (ListWarehouseDataObjectsRequest.SummaryField = {}));
})(ListWarehouseDataObjectsRequest = exports.ListWarehouseDataObjectsRequest || (exports.ListWarehouseDataObjectsRequest = {}));
