// oci-datasafe/lib/model/access-type.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessType = void 0;
/**
 * The type of access a user has on a table/view.
 **/
var AccessType;
(function (AccessType) {
    AccessType["Select"] = "SELECT";
    AccessType["Update"] = "UPDATE";
    AccessType["Insert"] = "INSERT";
    AccessType["Delete"] = "DELETE";
    AccessType["Owner"] = "OWNER";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AccessType["UnknownValue"] = "UNKNOWN_VALUE";
})(AccessType = exports.AccessType || (exports.AccessType = {}));
(function (AccessType) {
    function getJsonObj(obj) {
        return obj;
    }
    AccessType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AccessType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AccessType = exports.AccessType || (exports.AccessType = {}));
