// oci-datasafe/lib/model/create-sql-collection-details.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSqlCollectionDetails = void 0;
var CreateSqlCollectionDetails;
(function (CreateSqlCollectionDetails) {
    let Status;
    (function (Status) {
        Status["Enabled"] = "ENABLED";
        Status["Disabled"] = "DISABLED";
    })(Status = CreateSqlCollectionDetails.Status || (CreateSqlCollectionDetails.Status = {}));
    let SqlLevel;
    (function (SqlLevel) {
        SqlLevel["UserIssuedSql"] = "USER_ISSUED_SQL";
        SqlLevel["AllSql"] = "ALL_SQL";
    })(SqlLevel = CreateSqlCollectionDetails.SqlLevel || (CreateSqlCollectionDetails.SqlLevel = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateSqlCollectionDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateSqlCollectionDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateSqlCollectionDetails = exports.CreateSqlCollectionDetails || (exports.CreateSqlCollectionDetails = {}));
