// oci-datasafe/lib/model/finding-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FindingLifecycleState = void 0;
/**
 * The current state of the finding.
 **/
var FindingLifecycleState;
(function (FindingLifecycleState) {
    FindingLifecycleState["Active"] = "ACTIVE";
    FindingLifecycleState["Updating"] = "UPDATING";
    FindingLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    FindingLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    FindingLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(FindingLifecycleState = exports.FindingLifecycleState || (exports.FindingLifecycleState = {}));
(function (FindingLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    FindingLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    FindingLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(FindingLifecycleState = exports.FindingLifecycleState || (exports.FindingLifecycleState = {}));
