// oci-datasafe/lib/model/masking-policy-columns-info.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaskingPolicyColumnsInfo = void 0;
var MaskingPolicyColumnsInfo;
(function (MaskingPolicyColumnsInfo) {
    let ObjectType;
    (function (ObjectType) {
        ObjectType["Table"] = "TABLE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ObjectType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ObjectType = MaskingPolicyColumnsInfo.ObjectType || (MaskingPolicyColumnsInfo.ObjectType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MaskingPolicyColumnsInfo.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MaskingPolicyColumnsInfo.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaskingPolicyColumnsInfo = exports.MaskingPolicyColumnsInfo || (exports.MaskingPolicyColumnsInfo = {}));
