// oci-datasafe/lib/model/masking-error-summary.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaskingErrorSummary = void 0;
var MaskingErrorSummary;
(function (MaskingErrorSummary) {
    let StepName;
    (function (StepName) {
        StepName["ExecuteMasking"] = "EXECUTE_MASKING";
        StepName["PreMasking"] = "PRE_MASKING";
        StepName["PostMasking"] = "POST_MASKING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        StepName["UnknownValue"] = "UNKNOWN_VALUE";
    })(StepName = MaskingErrorSummary.StepName || (MaskingErrorSummary.StepName = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MaskingErrorSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MaskingErrorSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaskingErrorSummary = exports.MaskingErrorSummary || (exports.MaskingErrorSummary = {}));
