// oci-datasafe/lib/model/masking-policy-health-report-log-summary.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaskingPolicyHealthReportLogSummary = void 0;
var MaskingPolicyHealthReportLogSummary;
(function (MaskingPolicyHealthReportLogSummary) {
    let MessageType;
    (function (MessageType) {
        MessageType["Pass"] = "PASS";
        MessageType["Warning"] = "WARNING";
        MessageType["Error"] = "ERROR";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MessageType["UnknownValue"] = "UNKNOWN_VALUE";
    })(MessageType = MaskingPolicyHealthReportLogSummary.MessageType || (MaskingPolicyHealthReportLogSummary.MessageType = {}));
    let HealthCheckType;
    (function (HealthCheckType) {
        HealthCheckType["InvalidObjectCheck"] = "INVALID_OBJECT_CHECK";
        HealthCheckType["PrivilegeCheck"] = "PRIVILEGE_CHECK";
        HealthCheckType["TablespaceCheck"] = "TABLESPACE_CHECK";
        HealthCheckType["DatabaseOrSystemTriggersCheck"] = "DATABASE_OR_SYSTEM_TRIGGERS_CHECK";
        HealthCheckType["UndoTablespaceCheck"] = "UNDO_TABLESPACE_CHECK";
        HealthCheckType["StateStatsCheck"] = "STATE_STATS_CHECK";
        HealthCheckType["OlsPolicyCheck"] = "OLS_POLICY_CHECK";
        HealthCheckType["VpdPolicyCheck"] = "VPD_POLICY_CHECK";
        HealthCheckType["DvEnableCheck"] = "DV_ENABLE_CHECK";
        HealthCheckType["DeColSizeCheck"] = "DE_COL_SIZE_CHECK";
        HealthCheckType["RedactionPolicyCheck"] = "REDACTION_POLICY_CHECK";
        HealthCheckType["ActiveMaskJobCheck"] = "ACTIVE_MASK_JOB_CHECK";
        HealthCheckType["TargetValidationCheck"] = "TARGET_VALIDATION_CHECK";
        HealthCheckType["DeterministicEncryptionFormatCheck"] = "DETERMINISTIC_ENCRYPTION_FORMAT_CHECK";
        HealthCheckType["ColumnExistCheck"] = "COLUMN_EXIST_CHECK";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        HealthCheckType["UnknownValue"] = "UNKNOWN_VALUE";
    })(HealthCheckType = MaskingPolicyHealthReportLogSummary.HealthCheckType || (MaskingPolicyHealthReportLogSummary.HealthCheckType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MaskingPolicyHealthReportLogSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MaskingPolicyHealthReportLogSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaskingPolicyHealthReportLogSummary = exports.MaskingPolicyHealthReportLogSummary || (exports.MaskingPolicyHealthReportLogSummary = {}));
