// oci-datasafe/lib/model/privilege-name.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivilegeName = void 0;
/**
 * The name of the privilege granted to user.
 **/
var PrivilegeName;
(function (PrivilegeName) {
    PrivilegeName["Select"] = "SELECT";
    PrivilegeName["Update"] = "UPDATE";
    PrivilegeName["Insert"] = "INSERT";
    PrivilegeName["Delete"] = "DELETE";
    PrivilegeName["Read"] = "READ";
    PrivilegeName["Owner"] = "OWNER";
    PrivilegeName["Index"] = "INDEX";
    PrivilegeName["SelectAnyTable"] = "SELECT_ANY_TABLE";
    PrivilegeName["UpdateAnyTable"] = "UPDATE_ANY_TABLE";
    PrivilegeName["InsertAnyTable"] = "INSERT_ANY_TABLE";
    PrivilegeName["DeleteAnyTable"] = "DELETE_ANY_TABLE";
    PrivilegeName["ReadAnyTable"] = "READ_ANY_TABLE";
    PrivilegeName["CreateAnyIndex"] = "CREATE_ANY_INDEX";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PrivilegeName["UnknownValue"] = "UNKNOWN_VALUE";
})(PrivilegeName = exports.PrivilegeName || (exports.PrivilegeName = {}));
(function (PrivilegeName) {
    function getJsonObj(obj) {
        return obj;
    }
    PrivilegeName.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PrivilegeName.getDeserializedJsonObj = getDeserializedJsonObj;
})(PrivilegeName = exports.PrivilegeName || (exports.PrivilegeName = {}));
