// oci-datasafe/lib/model/referential-relation-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReferentialRelationLifecycleState = void 0;
/**
 * The current state of a referential relation.
 **/
var ReferentialRelationLifecycleState;
(function (ReferentialRelationLifecycleState) {
    ReferentialRelationLifecycleState["Creating"] = "CREATING";
    ReferentialRelationLifecycleState["Active"] = "ACTIVE";
    ReferentialRelationLifecycleState["Updating"] = "UPDATING";
    ReferentialRelationLifecycleState["Deleting"] = "DELETING";
    ReferentialRelationLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ReferentialRelationLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(ReferentialRelationLifecycleState = exports.ReferentialRelationLifecycleState || (exports.ReferentialRelationLifecycleState = {}));
(function (ReferentialRelationLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    ReferentialRelationLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ReferentialRelationLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(ReferentialRelationLifecycleState = exports.ReferentialRelationLifecycleState || (exports.ReferentialRelationLifecycleState = {}));
