// oci-datasafe/lib/model/security-policy-report-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityPolicyReportLifecycleState = void 0;
/**
 * The current state of the security policy report available on the target.
 **/
var SecurityPolicyReportLifecycleState;
(function (SecurityPolicyReportLifecycleState) {
    SecurityPolicyReportLifecycleState["Creating"] = "CREATING";
    SecurityPolicyReportLifecycleState["Succeeded"] = "SUCCEEDED";
    SecurityPolicyReportLifecycleState["Updating"] = "UPDATING";
    SecurityPolicyReportLifecycleState["Deleting"] = "DELETING";
    SecurityPolicyReportLifecycleState["Deleted"] = "DELETED";
    SecurityPolicyReportLifecycleState["Failed"] = "FAILED";
    SecurityPolicyReportLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SecurityPolicyReportLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SecurityPolicyReportLifecycleState = exports.SecurityPolicyReportLifecycleState || (exports.SecurityPolicyReportLifecycleState = {}));
(function (SecurityPolicyReportLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SecurityPolicyReportLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SecurityPolicyReportLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SecurityPolicyReportLifecycleState = exports.SecurityPolicyReportLifecycleState || (exports.SecurityPolicyReportLifecycleState = {}));
