// oci-datasafe/lib/model/security-policy-deployment-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityPolicyDeploymentLifecycleState = void 0;
/**
 * The current state of security policy deployment.
 **/
var SecurityPolicyDeploymentLifecycleState;
(function (SecurityPolicyDeploymentLifecycleState) {
    SecurityPolicyDeploymentLifecycleState["Creating"] = "CREATING";
    SecurityPolicyDeploymentLifecycleState["Updating"] = "UPDATING";
    SecurityPolicyDeploymentLifecycleState["Deployed"] = "DEPLOYED";
    SecurityPolicyDeploymentLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    SecurityPolicyDeploymentLifecycleState["Failed"] = "FAILED";
    SecurityPolicyDeploymentLifecycleState["Deleting"] = "DELETING";
    SecurityPolicyDeploymentLifecycleState["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SecurityPolicyDeploymentLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SecurityPolicyDeploymentLifecycleState = exports.SecurityPolicyDeploymentLifecycleState || (exports.SecurityPolicyDeploymentLifecycleState = {}));
(function (SecurityPolicyDeploymentLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SecurityPolicyDeploymentLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SecurityPolicyDeploymentLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SecurityPolicyDeploymentLifecycleState = exports.SecurityPolicyDeploymentLifecycleState || (exports.SecurityPolicyDeploymentLifecycleState = {}));
