// oci-datasafe/lib/model/security-assessment-references.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityAssessmentReferences = void 0;
/**
 * A filter to return only findings that match the specified reference. Currently supported values are STIG, CIS, and GDPR.
 * Here STIG refers to DISA STIG, CIS refers to CIS Benchmark, and GDPR refers to EU GDPR.
 *
 **/
var SecurityAssessmentReferences;
(function (SecurityAssessmentReferences) {
    SecurityAssessmentReferences["Stig"] = "STIG";
    SecurityAssessmentReferences["Cis"] = "CIS";
    SecurityAssessmentReferences["Gdpr"] = "GDPR";
})(SecurityAssessmentReferences = exports.SecurityAssessmentReferences || (exports.SecurityAssessmentReferences = {}));
(function (SecurityAssessmentReferences) {
    function getJsonObj(obj) {
        return obj;
    }
    SecurityAssessmentReferences.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SecurityAssessmentReferences.getDeserializedJsonObj = getDeserializedJsonObj;
})(SecurityAssessmentReferences = exports.SecurityAssessmentReferences || (exports.SecurityAssessmentReferences = {}));
