// oci-datasafe/lib/model/security-feature-analytics-dimensions.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityFeatureAnalyticsDimensions = void 0;
var SecurityFeatureAnalyticsDimensions;
(function (SecurityFeatureAnalyticsDimensions) {
    let SecurityFeature;
    (function (SecurityFeature) {
        SecurityFeature["UnifiedAudit"] = "UNIFIED_AUDIT";
        SecurityFeature["FineGrainedAudit"] = "FINE_GRAINED_AUDIT";
        SecurityFeature["TraditionalAudit"] = "TRADITIONAL_AUDIT";
        SecurityFeature["DatabaseVault"] = "DATABASE_VAULT";
        SecurityFeature["PrivilegeAnalysis"] = "PRIVILEGE_ANALYSIS";
        SecurityFeature["TablespaceEncryption"] = "TABLESPACE_ENCRYPTION";
        SecurityFeature["ColumnEncryption"] = "COLUMN_ENCRYPTION";
        SecurityFeature["NetworkEncryption"] = "NETWORK_ENCRYPTION";
        SecurityFeature["PasswordAuthentication"] = "PASSWORD_AUTHENTICATION";
        SecurityFeature["GlobalAuthentication"] = "GLOBAL_AUTHENTICATION";
        SecurityFeature["ExternalAuthentication"] = "EXTERNAL_AUTHENTICATION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SecurityFeature["UnknownValue"] = "UNKNOWN_VALUE";
    })(SecurityFeature = SecurityFeatureAnalyticsDimensions.SecurityFeature || (SecurityFeatureAnalyticsDimensions.SecurityFeature = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SecurityFeatureAnalyticsDimensions.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SecurityFeatureAnalyticsDimensions.getDeserializedJsonObj = getDeserializedJsonObj;
})(SecurityFeatureAnalyticsDimensions = exports.SecurityFeatureAnalyticsDimensions || (exports.SecurityFeatureAnalyticsDimensions = {}));
