// oci-datasafe/lib/model/security-policy-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityPolicyLifecycleState = void 0;
/**
 * The current state of security policies available on the target.
 **/
var SecurityPolicyLifecycleState;
(function (SecurityPolicyLifecycleState) {
    SecurityPolicyLifecycleState["Creating"] = "CREATING";
    SecurityPolicyLifecycleState["Updating"] = "UPDATING";
    SecurityPolicyLifecycleState["Active"] = "ACTIVE";
    SecurityPolicyLifecycleState["Failed"] = "FAILED";
    SecurityPolicyLifecycleState["Deleting"] = "DELETING";
    SecurityPolicyLifecycleState["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SecurityPolicyLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SecurityPolicyLifecycleState = exports.SecurityPolicyLifecycleState || (exports.SecurityPolicyLifecycleState = {}));
(function (SecurityPolicyLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SecurityPolicyLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SecurityPolicyLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SecurityPolicyLifecycleState = exports.SecurityPolicyLifecycleState || (exports.SecurityPolicyLifecycleState = {}));
