// oci-datasafe/lib/model/sensitive-type-group-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SensitiveTypeGroupLifecycleState = void 0;
/**
 * The current state of a sensitive type group
 **/
var SensitiveTypeGroupLifecycleState;
(function (SensitiveTypeGroupLifecycleState) {
    SensitiveTypeGroupLifecycleState["Creating"] = "CREATING";
    SensitiveTypeGroupLifecycleState["Active"] = "ACTIVE";
    SensitiveTypeGroupLifecycleState["Updating"] = "UPDATING";
    SensitiveTypeGroupLifecycleState["Deleting"] = "DELETING";
    SensitiveTypeGroupLifecycleState["Deleted"] = "DELETED";
    SensitiveTypeGroupLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SensitiveTypeGroupLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SensitiveTypeGroupLifecycleState = exports.SensitiveTypeGroupLifecycleState || (exports.SensitiveTypeGroupLifecycleState = {}));
(function (SensitiveTypeGroupLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SensitiveTypeGroupLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SensitiveTypeGroupLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SensitiveTypeGroupLifecycleState = exports.SensitiveTypeGroupLifecycleState || (exports.SensitiveTypeGroupLifecycleState = {}));
