// oci-datasafe/lib/model/sql-firewall-violation-summary.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlFirewallViolationSummary = void 0;
var SqlFirewallViolationSummary;
(function (SqlFirewallViolationSummary) {
    let SqlLevel;
    (function (SqlLevel) {
        SqlLevel["UserIssuedSql"] = "USER_ISSUED_SQL";
        SqlLevel["AllSql"] = "ALL_SQL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SqlLevel["UnknownValue"] = "UNKNOWN_VALUE";
    })(SqlLevel = SqlFirewallViolationSummary.SqlLevel || (SqlFirewallViolationSummary.SqlLevel = {}));
    let ViolationAction;
    (function (ViolationAction) {
        ViolationAction["Blocked"] = "BLOCKED";
        ViolationAction["Allowed"] = "ALLOWED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ViolationAction["UnknownValue"] = "UNKNOWN_VALUE";
    })(ViolationAction = SqlFirewallViolationSummary.ViolationAction || (SqlFirewallViolationSummary.ViolationAction = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SqlFirewallViolationSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SqlFirewallViolationSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlFirewallViolationSummary = exports.SqlFirewallViolationSummary || (exports.SqlFirewallViolationSummary = {}));
