// oci-datasafe/lib/model/sql-collection-summary.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlCollectionSummary = void 0;
var SqlCollectionSummary;
(function (SqlCollectionSummary) {
    let Status;
    (function (Status) {
        Status["Enabled"] = "ENABLED";
        Status["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = SqlCollectionSummary.Status || (SqlCollectionSummary.Status = {}));
    let SqlLevel;
    (function (SqlLevel) {
        SqlLevel["UserIssuedSql"] = "USER_ISSUED_SQL";
        SqlLevel["AllSql"] = "ALL_SQL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SqlLevel["UnknownValue"] = "UNKNOWN_VALUE";
    })(SqlLevel = SqlCollectionSummary.SqlLevel || (SqlCollectionSummary.SqlLevel = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SqlCollectionSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SqlCollectionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlCollectionSummary = exports.SqlCollectionSummary || (exports.SqlCollectionSummary = {}));
