// oci-datasafe/lib/model/sql-firewall-policy-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlFirewallPolicyLifecycleState = void 0;
/**
 * The current state of SQL Firewall policy.
 **/
var SqlFirewallPolicyLifecycleState;
(function (SqlFirewallPolicyLifecycleState) {
    SqlFirewallPolicyLifecycleState["Creating"] = "CREATING";
    SqlFirewallPolicyLifecycleState["Updating"] = "UPDATING";
    SqlFirewallPolicyLifecycleState["Active"] = "ACTIVE";
    SqlFirewallPolicyLifecycleState["Inactive"] = "INACTIVE";
    SqlFirewallPolicyLifecycleState["Failed"] = "FAILED";
    SqlFirewallPolicyLifecycleState["Deleting"] = "DELETING";
    SqlFirewallPolicyLifecycleState["Deleted"] = "DELETED";
    SqlFirewallPolicyLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SqlFirewallPolicyLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SqlFirewallPolicyLifecycleState = exports.SqlFirewallPolicyLifecycleState || (exports.SqlFirewallPolicyLifecycleState = {}));
(function (SqlFirewallPolicyLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SqlFirewallPolicyLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SqlFirewallPolicyLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlFirewallPolicyLifecycleState = exports.SqlFirewallPolicyLifecycleState || (exports.SqlFirewallPolicyLifecycleState = {}));
