// oci-datasafe/lib/model/sql-collection-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlCollectionLifecycleState = void 0;
/**
 * The current state of SQL collections available on the target.
 **/
var SqlCollectionLifecycleState;
(function (SqlCollectionLifecycleState) {
    SqlCollectionLifecycleState["Creating"] = "CREATING";
    SqlCollectionLifecycleState["Updating"] = "UPDATING";
    SqlCollectionLifecycleState["Collecting"] = "COLLECTING";
    SqlCollectionLifecycleState["Completed"] = "COMPLETED";
    SqlCollectionLifecycleState["Inactive"] = "INACTIVE";
    SqlCollectionLifecycleState["Failed"] = "FAILED";
    SqlCollectionLifecycleState["Deleting"] = "DELETING";
    SqlCollectionLifecycleState["Deleted"] = "DELETED";
    SqlCollectionLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SqlCollectionLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SqlCollectionLifecycleState = exports.SqlCollectionLifecycleState || (exports.SqlCollectionLifecycleState = {}));
(function (SqlCollectionLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SqlCollectionLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SqlCollectionLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlCollectionLifecycleState = exports.SqlCollectionLifecycleState || (exports.SqlCollectionLifecycleState = {}));
