// oci-datasafe/lib/model/sql-firewall-allowed-sql-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlFirewallAllowedSqlLifecycleState = void 0;
/**
 * The current state of SQL Firewall allowed SQL.
 **/
var SqlFirewallAllowedSqlLifecycleState;
(function (SqlFirewallAllowedSqlLifecycleState) {
    SqlFirewallAllowedSqlLifecycleState["Active"] = "ACTIVE";
    SqlFirewallAllowedSqlLifecycleState["Deleted"] = "DELETED";
    SqlFirewallAllowedSqlLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SqlFirewallAllowedSqlLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SqlFirewallAllowedSqlLifecycleState = exports.SqlFirewallAllowedSqlLifecycleState || (exports.SqlFirewallAllowedSqlLifecycleState = {}));
(function (SqlFirewallAllowedSqlLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SqlFirewallAllowedSqlLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SqlFirewallAllowedSqlLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlFirewallAllowedSqlLifecycleState = exports.SqlFirewallAllowedSqlLifecycleState || (exports.SqlFirewallAllowedSqlLifecycleState = {}));
