// oci-datasafe/lib/model/sql-firewall-policy-summary.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlFirewallPolicySummary = void 0;
var SqlFirewallPolicySummary;
(function (SqlFirewallPolicySummary) {
    let SqlLevel;
    (function (SqlLevel) {
        SqlLevel["UserIssuedSql"] = "USER_ISSUED_SQL";
        SqlLevel["AllSql"] = "ALL_SQL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SqlLevel["UnknownValue"] = "UNKNOWN_VALUE";
    })(SqlLevel = SqlFirewallPolicySummary.SqlLevel || (SqlFirewallPolicySummary.SqlLevel = {}));
    let Status;
    (function (Status) {
        Status["Enabled"] = "ENABLED";
        Status["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = SqlFirewallPolicySummary.Status || (SqlFirewallPolicySummary.Status = {}));
    let EnforcementScope;
    (function (EnforcementScope) {
        EnforcementScope["EnforceContext"] = "ENFORCE_CONTEXT";
        EnforcementScope["EnforceSql"] = "ENFORCE_SQL";
        EnforcementScope["EnforceAll"] = "ENFORCE_ALL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        EnforcementScope["UnknownValue"] = "UNKNOWN_VALUE";
    })(EnforcementScope = SqlFirewallPolicySummary.EnforcementScope || (SqlFirewallPolicySummary.EnforcementScope = {}));
    let ViolationAction;
    (function (ViolationAction) {
        ViolationAction["Block"] = "BLOCK";
        ViolationAction["Observe"] = "OBSERVE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ViolationAction["UnknownValue"] = "UNKNOWN_VALUE";
    })(ViolationAction = SqlFirewallPolicySummary.ViolationAction || (SqlFirewallPolicySummary.ViolationAction = {}));
    let ViolationAudit;
    (function (ViolationAudit) {
        ViolationAudit["Enabled"] = "ENABLED";
        ViolationAudit["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ViolationAudit["UnknownValue"] = "UNKNOWN_VALUE";
    })(ViolationAudit = SqlFirewallPolicySummary.ViolationAudit || (SqlFirewallPolicySummary.ViolationAudit = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SqlFirewallPolicySummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SqlFirewallPolicySummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlFirewallPolicySummary = exports.SqlFirewallPolicySummary || (exports.SqlFirewallPolicySummary = {}));
