// oci-datasafe/lib/model/update-sql-firewall-config-details.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSqlFirewallConfigDetails = void 0;
var UpdateSqlFirewallConfigDetails;
(function (UpdateSqlFirewallConfigDetails) {
    let Status;
    (function (Status) {
        Status["Enabled"] = "ENABLED";
        Status["Disabled"] = "DISABLED";
    })(Status = UpdateSqlFirewallConfigDetails.Status || (UpdateSqlFirewallConfigDetails.Status = {}));
    let ViolationLogAutoPurge;
    (function (ViolationLogAutoPurge) {
        ViolationLogAutoPurge["Enabled"] = "ENABLED";
        ViolationLogAutoPurge["Disabled"] = "DISABLED";
    })(ViolationLogAutoPurge = UpdateSqlFirewallConfigDetails.ViolationLogAutoPurge || (UpdateSqlFirewallConfigDetails.ViolationLogAutoPurge = {}));
    let ExcludeJob;
    (function (ExcludeJob) {
        ExcludeJob["Excluded"] = "EXCLUDED";
        ExcludeJob["Included"] = "INCLUDED";
    })(ExcludeJob = UpdateSqlFirewallConfigDetails.ExcludeJob || (UpdateSqlFirewallConfigDetails.ExcludeJob = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateSqlFirewallConfigDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateSqlFirewallConfigDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateSqlFirewallConfigDetails = exports.UpdateSqlFirewallConfigDetails || (exports.UpdateSqlFirewallConfigDetails = {}));
