// oci-datasafe/lib/model/update-sql-firewall-policy-details.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSqlFirewallPolicyDetails = void 0;
var UpdateSqlFirewallPolicyDetails;
(function (UpdateSqlFirewallPolicyDetails) {
    let Status;
    (function (Status) {
        Status["Enabled"] = "ENABLED";
        Status["Disabled"] = "DISABLED";
    })(Status = UpdateSqlFirewallPolicyDetails.Status || (UpdateSqlFirewallPolicyDetails.Status = {}));
    let EnforcementScope;
    (function (EnforcementScope) {
        EnforcementScope["EnforceContext"] = "ENFORCE_CONTEXT";
        EnforcementScope["EnforceSql"] = "ENFORCE_SQL";
        EnforcementScope["EnforceAll"] = "ENFORCE_ALL";
    })(EnforcementScope = UpdateSqlFirewallPolicyDetails.EnforcementScope || (UpdateSqlFirewallPolicyDetails.EnforcementScope = {}));
    let ViolationAction;
    (function (ViolationAction) {
        ViolationAction["Block"] = "BLOCK";
        ViolationAction["Observe"] = "OBSERVE";
    })(ViolationAction = UpdateSqlFirewallPolicyDetails.ViolationAction || (UpdateSqlFirewallPolicyDetails.ViolationAction = {}));
    let ViolationAudit;
    (function (ViolationAudit) {
        ViolationAudit["Enabled"] = "ENABLED";
        ViolationAudit["Disabled"] = "DISABLED";
    })(ViolationAudit = UpdateSqlFirewallPolicyDetails.ViolationAudit || (UpdateSqlFirewallPolicyDetails.ViolationAudit = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateSqlFirewallPolicyDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateSqlFirewallPolicyDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateSqlFirewallPolicyDetails = exports.UpdateSqlFirewallPolicyDetails || (exports.UpdateSqlFirewallPolicyDetails = {}));
