// oci-capacitymanagement/lib/request/list-occ-customer-groups-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20231107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListOccCustomerGroupsRequest = void 0;
var ListOccCustomerGroupsRequest;
(function (ListOccCustomerGroupsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["CompartmentId"] = "compartmentId";
        SortBy["Name"] = "name";
        SortBy["TimeCreated"] = "timeCreated";
    })(SortBy = ListOccCustomerGroupsRequest.SortBy || (ListOccCustomerGroupsRequest.SortBy = {}));
})(ListOccCustomerGroupsRequest = exports.ListOccCustomerGroupsRequest || (exports.ListOccCustomerGroupsRequest = {}));
