// oci-apigateway/lib/model/validation-request-policy.js
"use strict";
/**
 * API Gateway API
 * API for the API Gateway service. Use this API to manage gateways, deployments, and related items.
For more information, see
[Overview of API Gateway](https://docs.oracle.com/iaas/Content/APIGateway/Concepts/apigatewayoverview.htm).

 * OpenAPI spec version: 20190501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationRequestPolicy = void 0;
var ValidationRequestPolicy;
(function (ValidationRequestPolicy) {
    let ValidationMode;
    (function (ValidationMode) {
        ValidationMode["Enforcing"] = "ENFORCING";
        ValidationMode["Permissive"] = "PERMISSIVE";
        ValidationMode["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ValidationMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(ValidationMode = ValidationRequestPolicy.ValidationMode || (ValidationRequestPolicy.ValidationMode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ValidationRequestPolicy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ValidationRequestPolicy.getDeserializedJsonObj = getDeserializedJsonObj;
})(ValidationRequestPolicy = exports.ValidationRequestPolicy || (exports.ValidationRequestPolicy = {}));
