// oci-stackmonitoring/lib/model/alarm-condition-severity.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlarmConditionSeverity = void 0;
/**
 * Alarm condition severity.
 **/
var AlarmConditionSeverity;
(function (AlarmConditionSeverity) {
    AlarmConditionSeverity["Critical"] = "CRITICAL";
    AlarmConditionSeverity["Warning"] = "WARNING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AlarmConditionSeverity["UnknownValue"] = "UNKNOWN_VALUE";
})(AlarmConditionSeverity = exports.AlarmConditionSeverity || (exports.AlarmConditionSeverity = {}));
(function (AlarmConditionSeverity) {
    function getJsonObj(obj) {
        return obj;
    }
    AlarmConditionSeverity.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AlarmConditionSeverity.getDeserializedJsonObj = getDeserializedJsonObj;
})(AlarmConditionSeverity = exports.AlarmConditionSeverity || (exports.AlarmConditionSeverity = {}));
