// oci-stackmonitoring/lib/model/baselineable-metric-life-cycle-states.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaselineableMetricLifeCycleStates = void 0;
/**
 * Lifecycle state value of Baselineable Metric like ACTIVE or DELETED
 **/
var BaselineableMetricLifeCycleStates;
(function (BaselineableMetricLifeCycleStates) {
    BaselineableMetricLifeCycleStates["Active"] = "ACTIVE";
    BaselineableMetricLifeCycleStates["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    BaselineableMetricLifeCycleStates["UnknownValue"] = "UNKNOWN_VALUE";
})(BaselineableMetricLifeCycleStates = exports.BaselineableMetricLifeCycleStates || (exports.BaselineableMetricLifeCycleStates = {}));
(function (BaselineableMetricLifeCycleStates) {
    function getJsonObj(obj) {
        return obj;
    }
    BaselineableMetricLifeCycleStates.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    BaselineableMetricLifeCycleStates.getDeserializedJsonObj = getDeserializedJsonObj;
})(BaselineableMetricLifeCycleStates = exports.BaselineableMetricLifeCycleStates || (exports.BaselineableMetricLifeCycleStates = {}));
