// oci-stackmonitoring/lib/model/handler-type.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HandlerType = void 0;
/**
 * Handler type.
 **/
var HandlerType;
(function (HandlerType) {
    HandlerType["Telegraf"] = "TELEGRAF";
    HandlerType["Collectd"] = "COLLECTD";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    HandlerType["UnknownValue"] = "UNKNOWN_VALUE";
})(HandlerType = exports.HandlerType || (exports.HandlerType = {}));
(function (HandlerType) {
    function getJsonObj(obj) {
        return obj;
    }
    HandlerType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    HandlerType.getDeserializedJsonObj = getDeserializedJsonObj;
})(HandlerType = exports.HandlerType || (exports.HandlerType = {}));
