// oci-stackmonitoring/lib/model/http-response-content-types.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpResponseContentTypes = void 0;
/**
 * Possible response content types of HTTP collection method
 **/
var HttpResponseContentTypes;
(function (HttpResponseContentTypes) {
    HttpResponseContentTypes["TextPlain"] = "TEXT_PLAIN";
    HttpResponseContentTypes["TextHtml"] = "TEXT_HTML";
    HttpResponseContentTypes["ApplicationJson"] = "APPLICATION_JSON";
    HttpResponseContentTypes["ApplicationXml"] = "APPLICATION_XML";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    HttpResponseContentTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(HttpResponseContentTypes = exports.HttpResponseContentTypes || (exports.HttpResponseContentTypes = {}));
(function (HttpResponseContentTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    HttpResponseContentTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    HttpResponseContentTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(HttpResponseContentTypes = exports.HttpResponseContentTypes || (exports.HttpResponseContentTypes = {}));
