// oci-stackmonitoring/lib/model/maintenance-window-sort-by.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceWindowSortBy = void 0;
/**
 * Sort By.
 **/
var MaintenanceWindowSortBy;
(function (MaintenanceWindowSortBy) {
    MaintenanceWindowSortBy["Name"] = "NAME";
    MaintenanceWindowSortBy["StartTime"] = "START_TIME";
    MaintenanceWindowSortBy["EndTime"] = "END_TIME";
    MaintenanceWindowSortBy["TimeCreated"] = "TIME_CREATED";
    MaintenanceWindowSortBy["TimeUpdated"] = "TIME_UPDATED";
})(MaintenanceWindowSortBy = exports.MaintenanceWindowSortBy || (exports.MaintenanceWindowSortBy = {}));
(function (MaintenanceWindowSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    MaintenanceWindowSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MaintenanceWindowSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaintenanceWindowSortBy = exports.MaintenanceWindowSortBy || (exports.MaintenanceWindowSortBy = {}));
