// oci-stackmonitoring/lib/model/maintenance-window-lifecycle-state.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceWindowLifecycleState = void 0;
/**
 * Possible lifecycle states.
 *
 **/
var MaintenanceWindowLifecycleState;
(function (MaintenanceWindowLifecycleState) {
    MaintenanceWindowLifecycleState["Creating"] = "CREATING";
    MaintenanceWindowLifecycleState["Updating"] = "UPDATING";
    MaintenanceWindowLifecycleState["Inactive"] = "INACTIVE";
    MaintenanceWindowLifecycleState["Active"] = "ACTIVE";
    MaintenanceWindowLifecycleState["Deleting"] = "DELETING";
    MaintenanceWindowLifecycleState["Deleted"] = "DELETED";
    MaintenanceWindowLifecycleState["Failed"] = "FAILED";
    MaintenanceWindowLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MaintenanceWindowLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(MaintenanceWindowLifecycleState = exports.MaintenanceWindowLifecycleState || (exports.MaintenanceWindowLifecycleState = {}));
(function (MaintenanceWindowLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    MaintenanceWindowLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MaintenanceWindowLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaintenanceWindowLifecycleState = exports.MaintenanceWindowLifecycleState || (exports.MaintenanceWindowLifecycleState = {}));
