// oci-stackmonitoring/lib/model/member-reference-type.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MemberReferenceType = void 0;
/**
 * Member Reference types of Monitoring template like RESOURCE_INSTANCE or RESOURCE_TYPE or RESOURCE_GROUP
 **/
var MemberReferenceType;
(function (MemberReferenceType) {
    MemberReferenceType["ResourceInstance"] = "RESOURCE_INSTANCE";
    MemberReferenceType["ResourceType"] = "RESOURCE_TYPE";
    MemberReferenceType["ResourceGroup"] = "RESOURCE_GROUP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MemberReferenceType["UnknownValue"] = "UNKNOWN_VALUE";
})(MemberReferenceType = exports.MemberReferenceType || (exports.MemberReferenceType = {}));
(function (MemberReferenceType) {
    function getJsonObj(obj) {
        return obj;
    }
    MemberReferenceType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MemberReferenceType.getDeserializedJsonObj = getDeserializedJsonObj;
})(MemberReferenceType = exports.MemberReferenceType || (exports.MemberReferenceType = {}));
