// oci-stackmonitoring/lib/model/metric-extension-sort-by.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricExtensionSortBy = void 0;
/**
 * Sort By.
 **/
var MetricExtensionSortBy;
(function (MetricExtensionSortBy) {
    MetricExtensionSortBy["Name"] = "NAME";
    MetricExtensionSortBy["TimeCreated"] = "TIME_CREATED";
    MetricExtensionSortBy["EnabledOnResourceCount"] = "ENABLED_ON_RESOURCE_COUNT";
})(MetricExtensionSortBy = exports.MetricExtensionSortBy || (exports.MetricExtensionSortBy = {}));
(function (MetricExtensionSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    MetricExtensionSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MetricExtensionSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(MetricExtensionSortBy = exports.MetricExtensionSortBy || (exports.MetricExtensionSortBy = {}));
