// oci-stackmonitoring/lib/model/monitored-resource-sort-by.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitoredResourceSortBy = void 0;
/**
 * Sort By.
 **/
var MonitoredResourceSortBy;
(function (MonitoredResourceSortBy) {
    MonitoredResourceSortBy["Name"] = "NAME";
    MonitoredResourceSortBy["TimeCreated"] = "TIME_CREATED";
})(MonitoredResourceSortBy = exports.MonitoredResourceSortBy || (exports.MonitoredResourceSortBy = {}));
(function (MonitoredResourceSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    MonitoredResourceSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MonitoredResourceSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(MonitoredResourceSortBy = exports.MonitoredResourceSortBy || (exports.MonitoredResourceSortBy = {}));
