// oci-stackmonitoring/lib/model/monitoring-template-life-cycle-states.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitoringTemplateLifeCycleStates = void 0;
/**
 * The current lifecycle state of the monitoringTemplate
 **/
var MonitoringTemplateLifeCycleStates;
(function (MonitoringTemplateLifeCycleStates) {
    MonitoringTemplateLifeCycleStates["Creating"] = "CREATING";
    MonitoringTemplateLifeCycleStates["Active"] = "ACTIVE";
    MonitoringTemplateLifeCycleStates["Inactive"] = "INACTIVE";
    MonitoringTemplateLifeCycleStates["Updating"] = "UPDATING";
    MonitoringTemplateLifeCycleStates["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MonitoringTemplateLifeCycleStates["UnknownValue"] = "UNKNOWN_VALUE";
})(MonitoringTemplateLifeCycleStates = exports.MonitoringTemplateLifeCycleStates || (exports.MonitoringTemplateLifeCycleStates = {}));
(function (MonitoringTemplateLifeCycleStates) {
    function getJsonObj(obj) {
        return obj;
    }
    MonitoringTemplateLifeCycleStates.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MonitoringTemplateLifeCycleStates.getDeserializedJsonObj = getDeserializedJsonObj;
})(MonitoringTemplateLifeCycleStates = exports.MonitoringTemplateLifeCycleStates || (exports.MonitoringTemplateLifeCycleStates = {}));
