// oci-stackmonitoring/lib/model/monitored-resource-task-lifecycle-state.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitoredResourceTaskLifecycleState = void 0;
/**
 * Possible lifecycle states for stack monitoring resource task.
 *
 **/
var MonitoredResourceTaskLifecycleState;
(function (MonitoredResourceTaskLifecycleState) {
    MonitoredResourceTaskLifecycleState["Accepted"] = "ACCEPTED";
    MonitoredResourceTaskLifecycleState["InProgress"] = "IN_PROGRESS";
    MonitoredResourceTaskLifecycleState["Waiting"] = "WAITING";
    MonitoredResourceTaskLifecycleState["Failed"] = "FAILED";
    MonitoredResourceTaskLifecycleState["Succeeded"] = "SUCCEEDED";
    MonitoredResourceTaskLifecycleState["Canceling"] = "CANCELING";
    MonitoredResourceTaskLifecycleState["Canceled"] = "CANCELED";
    MonitoredResourceTaskLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MonitoredResourceTaskLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(MonitoredResourceTaskLifecycleState = exports.MonitoredResourceTaskLifecycleState || (exports.MonitoredResourceTaskLifecycleState = {}));
(function (MonitoredResourceTaskLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    MonitoredResourceTaskLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MonitoredResourceTaskLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(MonitoredResourceTaskLifecycleState = exports.MonitoredResourceTaskLifecycleState || (exports.MonitoredResourceTaskLifecycleState = {}));
