// oci-stackmonitoring/lib/model/operation-type.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateResources"] = "CREATE_RESOURCES";
    OperationType["UpdateResources"] = "UPDATE_RESOURCES";
    OperationType["DeleteResources"] = "DELETE_RESOURCES";
    OperationType["MoveResources"] = "MOVE_RESOURCES";
    OperationType["EnableExternalDatabase"] = "ENABLE_EXTERNAL_DATABASE";
    OperationType["DisableExternalDatabase"] = "DISABLE_EXTERNAL_DATABASE";
    OperationType["AddSourcesToAgent"] = "ADD_SOURCES_TO_AGENT";
    OperationType["EnableMetricExtension"] = "ENABLE_METRIC_EXTENSION";
    OperationType["DisableMetricExtension"] = "DISABLE_METRIC_EXTENSION";
    OperationType["TestMetricExtension"] = "TEST_METRIC_EXTENSION";
    OperationType["ApplyMonitoringTemplate"] = "APPLY_MONITORING_TEMPLATE";
    OperationType["UnapplyMonitoringTemplate"] = "UNAPPLY_MONITORING_TEMPLATE";
    OperationType["BulkAddResources"] = "BULK_ADD_RESOURCES";
    OperationType["BulkDeleteResources"] = "BULK_DELETE_RESOURCES";
    OperationType["UpdateAndPropagateTags"] = "UPDATE_AND_PROPAGATE_TAGS";
    OperationType["ImportResources"] = "IMPORT_RESOURCES";
    OperationType["CreateMaintenanceWindow"] = "CREATE_MAINTENANCE_WINDOW";
    OperationType["UpdateMaintenanceWindow"] = "UPDATE_MAINTENANCE_WINDOW";
    OperationType["DeleteMaintenanceWindow"] = "DELETE_MAINTENANCE_WINDOW";
    OperationType["StopMaintenanceWindow"] = "STOP_MAINTENANCE_WINDOW";
    OperationType["UpdateAgentReceiver"] = "UPDATE_AGENT_RECEIVER";
    OperationType["UpdateResourceTypeConfigs"] = "UPDATE_RESOURCE_TYPE_CONFIGS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
