// oci-stackmonitoring/lib/model/resource-category.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceCategory = void 0;
/**
 * Resource Category.
 **/
var ResourceCategory;
(function (ResourceCategory) {
    ResourceCategory["Application"] = "APPLICATION";
    ResourceCategory["Database"] = "DATABASE";
    ResourceCategory["Middleware"] = "MIDDLEWARE";
    ResourceCategory["Infrastructure"] = "INFRASTRUCTURE";
    ResourceCategory["Unknown"] = "UNKNOWN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ResourceCategory["UnknownValue"] = "UNKNOWN_VALUE";
})(ResourceCategory = exports.ResourceCategory || (exports.ResourceCategory = {}));
(function (ResourceCategory) {
    function getJsonObj(obj) {
        return obj;
    }
    ResourceCategory.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ResourceCategory.getDeserializedJsonObj = getDeserializedJsonObj;
})(ResourceCategory = exports.ResourceCategory || (exports.ResourceCategory = {}));
