// oci-stackmonitoring/lib/model/resource-type-lifecycle-state.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceTypeLifecycleState = void 0;
/**
 * Possible lifecycle states for monitored resource type.
 *
 **/
var ResourceTypeLifecycleState;
(function (ResourceTypeLifecycleState) {
    ResourceTypeLifecycleState["Creating"] = "CREATING";
    ResourceTypeLifecycleState["Updating"] = "UPDATING";
    ResourceTypeLifecycleState["Active"] = "ACTIVE";
    ResourceTypeLifecycleState["Deleting"] = "DELETING";
    ResourceTypeLifecycleState["Deleted"] = "DELETED";
    ResourceTypeLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ResourceTypeLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(ResourceTypeLifecycleState = exports.ResourceTypeLifecycleState || (exports.ResourceTypeLifecycleState = {}));
(function (ResourceTypeLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    ResourceTypeLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ResourceTypeLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(ResourceTypeLifecycleState = exports.ResourceTypeLifecycleState || (exports.ResourceTypeLifecycleState = {}));
