// oci-stackmonitoring/lib/model/search-monitored-resources-details.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchMonitoredResourcesDetails = void 0;
const model = __importStar(require("../model"));
var SearchMonitoredResourcesDetails;
(function (SearchMonitoredResourcesDetails) {
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "TIME_CREATED";
        SortBy["ResourceName"] = "RESOURCE_NAME";
    })(SortBy = SearchMonitoredResourcesDetails.SortBy || (SearchMonitoredResourcesDetails.SortBy = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "lifecycleStates": obj.lifecycleStates
                ? obj.lifecycleStates.map(item => {
                    return model.ResourceLifecycleState.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    SearchMonitoredResourcesDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "lifecycleStates": obj.lifecycleStates
                ? obj.lifecycleStates.map(item => {
                    return model.ResourceLifecycleState.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    SearchMonitoredResourcesDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(SearchMonitoredResourcesDetails = exports.SearchMonitoredResourcesDetails || (exports.SearchMonitoredResourcesDetails = {}));
