// oci-stackmonitoring/lib/model/source-type.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceType = void 0;
/**
 * Source Type of the resource.
 * 1. SM_MGMT_AGENT_MONITORED - Management agent monitored stack monitoring discovered resources.
 * 2. SM_REPO_ONLY - Repository only stack monitoring discovered resources.
 * 3. OCI_NATIVE - OCI Native resources imported from OCI Monitoring Service.
 * 4. PROMETHEUS - Prometheus resources imported from OCI Monitoring Service.
 * 5. TELEGRAF - Telegraf based resources.
 * 6. COLLECTD - Collectd based resources.
 *
 **/
var SourceType;
(function (SourceType) {
    SourceType["SmMgmtAgentMonitored"] = "SM_MGMT_AGENT_MONITORED";
    SourceType["SmRepoOnly"] = "SM_REPO_ONLY";
    SourceType["OciNative"] = "OCI_NATIVE";
    SourceType["Prometheus"] = "PROMETHEUS";
    SourceType["Telegraf"] = "TELEGRAF";
    SourceType["Collectd"] = "COLLECTD";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SourceType["UnknownValue"] = "UNKNOWN_VALUE";
})(SourceType = exports.SourceType || (exports.SourceType = {}));
(function (SourceType) {
    function getJsonObj(obj) {
        return obj;
    }
    SourceType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SourceType.getDeserializedJsonObj = getDeserializedJsonObj;
})(SourceType = exports.SourceType || (exports.SourceType = {}));
