// oci-stackmonitoring/lib/model/sql-query-types.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlQueryTypes = void 0;
/**
 * Type of SQL data collection method i.e. either a Statement or SQL Script File
 **/
var SqlQueryTypes;
(function (SqlQueryTypes) {
    SqlQueryTypes["Statement"] = "STATEMENT";
    SqlQueryTypes["SqlScript"] = "SQL_SCRIPT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SqlQueryTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(SqlQueryTypes = exports.SqlQueryTypes || (exports.SqlQueryTypes = {}));
(function (SqlQueryTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    SqlQueryTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SqlQueryTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlQueryTypes = exports.SqlQueryTypes || (exports.SqlQueryTypes = {}));
