// oci-stackmonitoring/lib/model/sql-out-param-types.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlOutParamTypes = void 0;
/**
 * Possible out parameter types of SQL collection method. During the update, to completely remove the out parameter, use the value NO_OUT_PARAM
 **/
var SqlOutParamTypes;
(function (SqlOutParamTypes) {
    SqlOutParamTypes["SqlCursor"] = "SQL_CURSOR";
    SqlOutParamTypes["Array"] = "ARRAY";
    SqlOutParamTypes["NoOutParam"] = "NO_OUT_PARAM";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SqlOutParamTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(SqlOutParamTypes = exports.SqlOutParamTypes || (exports.SqlOutParamTypes = {}));
(function (SqlOutParamTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    SqlOutParamTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SqlOutParamTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlOutParamTypes = exports.SqlOutParamTypes || (exports.SqlOutParamTypes = {}));
