// oci-sch/lib/model/index.js
"use strict";
/**
 * Connector Hub API
 * Use the Connector Hub API to transfer data between services in Oracle Cloud Infrastructure.
For more information about Connector Hub, see
[the Connector Hub documentation](https://docs.oracle.com/iaas/Content/connector-hub/home.htm).
Connector Hub is formerly known as Service Connector Hub.

 * OpenAPI spec version: 20200909
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoggingSourceDetailsResponse = exports.LoggingSourceDetails = exports.LoggingAnalyticsTargetDetailsResponse = exports.LoggingAnalyticsTargetDetails = exports.LogRuleTaskDetailsResponse = exports.LogRuleTaskDetails = exports.LatestStreamingCursor = exports.JmesPathDimensionValue = exports.FunctionsTargetDetailsResponse = exports.FunctionsTargetDetails = exports.FunctionTaskDetailsResponse = exports.FunctionTaskDetails = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.UpdateServiceConnectorDetails = exports.TaskDetailsResponse = exports.TaskDetails = exports.TargetDetailsResponse = exports.TargetDetails = exports.StreamingCursorDetails = exports.SourceDetailsResponse = exports.SourceDetails = exports.SortOrder = exports.ServiceConnectorSummary = exports.ServiceConnectorCollection = exports.ServiceConnector = exports.PrivateEndpointMetadata = exports.OperationType = exports.OperationStatus = exports.MonitoringSourceSelectedNamespace = exports.MonitoringSourceNamespaceDetails = exports.MonitoringSourceMetricDetails = exports.MonitoringSource = exports.LogSource = exports.LifecycleState = exports.EstimatedThroughput = exports.DimensionValueDetails = exports.DimensionDetails = exports.CreateServiceConnectorDetails = exports.ConnectorPluginSummary = exports.ConnectorPluginLifecycleState = exports.ConnectorPluginCollection = exports.ConnectorPlugin = exports.ChangeServiceConnectorCompartmentDetails = exports.ActionType = void 0;
exports.TrimHorizonStreamingCursor = exports.TargetConnectorPluginSummary = exports.TargetConnectorPlugin = exports.StreamingTargetDetailsResponse = exports.StreamingTargetDetails = exports.StreamingSourceDetailsResponse = exports.StreamingSourceDetails = exports.StaticDimensionValue = exports.SourceConnectorPluginSummary = exports.SourceConnectorPlugin = exports.PluginSourceDetailsResponse = exports.PluginSourceDetails = exports.ObjectStorageTargetDetailsResponse = exports.ObjectStorageTargetDetails = exports.NotificationsTargetDetailsResponse = exports.NotificationsTargetDetails = exports.MonitoringTargetDetailsResponse = exports.MonitoringTargetDetails = exports.MonitoringSourceSelectedNamespaceDetails = exports.MonitoringSourceDetailsResponse = exports.MonitoringSourceDetails = exports.MonitoringSourceAllMetrics = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangeServiceConnectorCompartmentDetails = __importStar(require("./change-service-connector-compartment-details"));
exports.ChangeServiceConnectorCompartmentDetails = ChangeServiceConnectorCompartmentDetails.ChangeServiceConnectorCompartmentDetails;
const ConnectorPlugin = __importStar(require("./connector-plugin"));
exports.ConnectorPlugin = ConnectorPlugin.ConnectorPlugin;
const ConnectorPluginCollection = __importStar(require("./connector-plugin-collection"));
exports.ConnectorPluginCollection = ConnectorPluginCollection.ConnectorPluginCollection;
const ConnectorPluginLifecycleState = __importStar(require("./connector-plugin-lifecycle-state"));
exports.ConnectorPluginLifecycleState = ConnectorPluginLifecycleState.ConnectorPluginLifecycleState;
const ConnectorPluginSummary = __importStar(require("./connector-plugin-summary"));
exports.ConnectorPluginSummary = ConnectorPluginSummary.ConnectorPluginSummary;
const CreateServiceConnectorDetails = __importStar(require("./create-service-connector-details"));
exports.CreateServiceConnectorDetails = CreateServiceConnectorDetails.CreateServiceConnectorDetails;
const DimensionDetails = __importStar(require("./dimension-details"));
exports.DimensionDetails = DimensionDetails.DimensionDetails;
const DimensionValueDetails = __importStar(require("./dimension-value-details"));
exports.DimensionValueDetails = DimensionValueDetails.DimensionValueDetails;
const EstimatedThroughput = __importStar(require("./estimated-throughput"));
exports.EstimatedThroughput = EstimatedThroughput.EstimatedThroughput;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const LogSource = __importStar(require("./log-source"));
exports.LogSource = LogSource.LogSource;
const MonitoringSource = __importStar(require("./monitoring-source"));
exports.MonitoringSource = MonitoringSource.MonitoringSource;
const MonitoringSourceMetricDetails = __importStar(require("./monitoring-source-metric-details"));
exports.MonitoringSourceMetricDetails = MonitoringSourceMetricDetails.MonitoringSourceMetricDetails;
const MonitoringSourceNamespaceDetails = __importStar(require("./monitoring-source-namespace-details"));
exports.MonitoringSourceNamespaceDetails = MonitoringSourceNamespaceDetails.MonitoringSourceNamespaceDetails;
const MonitoringSourceSelectedNamespace = __importStar(require("./monitoring-source-selected-namespace"));
exports.MonitoringSourceSelectedNamespace = MonitoringSourceSelectedNamespace.MonitoringSourceSelectedNamespace;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PrivateEndpointMetadata = __importStar(require("./private-endpoint-metadata"));
exports.PrivateEndpointMetadata = PrivateEndpointMetadata.PrivateEndpointMetadata;
const ServiceConnector = __importStar(require("./service-connector"));
exports.ServiceConnector = ServiceConnector.ServiceConnector;
const ServiceConnectorCollection = __importStar(require("./service-connector-collection"));
exports.ServiceConnectorCollection = ServiceConnectorCollection.ServiceConnectorCollection;
const ServiceConnectorSummary = __importStar(require("./service-connector-summary"));
exports.ServiceConnectorSummary = ServiceConnectorSummary.ServiceConnectorSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const SourceDetails = __importStar(require("./source-details"));
exports.SourceDetails = SourceDetails.SourceDetails;
const SourceDetailsResponse = __importStar(require("./source-details-response"));
exports.SourceDetailsResponse = SourceDetailsResponse.SourceDetailsResponse;
const StreamingCursorDetails = __importStar(require("./streaming-cursor-details"));
exports.StreamingCursorDetails = StreamingCursorDetails.StreamingCursorDetails;
const TargetDetails = __importStar(require("./target-details"));
exports.TargetDetails = TargetDetails.TargetDetails;
const TargetDetailsResponse = __importStar(require("./target-details-response"));
exports.TargetDetailsResponse = TargetDetailsResponse.TargetDetailsResponse;
const TaskDetails = __importStar(require("./task-details"));
exports.TaskDetails = TaskDetails.TaskDetails;
const TaskDetailsResponse = __importStar(require("./task-details-response"));
exports.TaskDetailsResponse = TaskDetailsResponse.TaskDetailsResponse;
const UpdateServiceConnectorDetails = __importStar(require("./update-service-connector-details"));
exports.UpdateServiceConnectorDetails = UpdateServiceConnectorDetails.UpdateServiceConnectorDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const FunctionTaskDetails = __importStar(require("./function-task-details"));
exports.FunctionTaskDetails = FunctionTaskDetails.FunctionTaskDetails;
const FunctionTaskDetailsResponse = __importStar(require("./function-task-details-response"));
exports.FunctionTaskDetailsResponse = FunctionTaskDetailsResponse.FunctionTaskDetailsResponse;
const FunctionsTargetDetails = __importStar(require("./functions-target-details"));
exports.FunctionsTargetDetails = FunctionsTargetDetails.FunctionsTargetDetails;
const FunctionsTargetDetailsResponse = __importStar(require("./functions-target-details-response"));
exports.FunctionsTargetDetailsResponse = FunctionsTargetDetailsResponse.FunctionsTargetDetailsResponse;
const JmesPathDimensionValue = __importStar(require("./jmes-path-dimension-value"));
exports.JmesPathDimensionValue = JmesPathDimensionValue.JmesPathDimensionValue;
const LatestStreamingCursor = __importStar(require("./latest-streaming-cursor"));
exports.LatestStreamingCursor = LatestStreamingCursor.LatestStreamingCursor;
const LogRuleTaskDetails = __importStar(require("./log-rule-task-details"));
exports.LogRuleTaskDetails = LogRuleTaskDetails.LogRuleTaskDetails;
const LogRuleTaskDetailsResponse = __importStar(require("./log-rule-task-details-response"));
exports.LogRuleTaskDetailsResponse = LogRuleTaskDetailsResponse.LogRuleTaskDetailsResponse;
const LoggingAnalyticsTargetDetails = __importStar(require("./logging-analytics-target-details"));
exports.LoggingAnalyticsTargetDetails = LoggingAnalyticsTargetDetails.LoggingAnalyticsTargetDetails;
const LoggingAnalyticsTargetDetailsResponse = __importStar(require("./logging-analytics-target-details-response"));
exports.LoggingAnalyticsTargetDetailsResponse = LoggingAnalyticsTargetDetailsResponse.LoggingAnalyticsTargetDetailsResponse;
const LoggingSourceDetails = __importStar(require("./logging-source-details"));
exports.LoggingSourceDetails = LoggingSourceDetails.LoggingSourceDetails;
const LoggingSourceDetailsResponse = __importStar(require("./logging-source-details-response"));
exports.LoggingSourceDetailsResponse = LoggingSourceDetailsResponse.LoggingSourceDetailsResponse;
const MonitoringSourceAllMetrics = __importStar(require("./monitoring-source-all-metrics"));
exports.MonitoringSourceAllMetrics = MonitoringSourceAllMetrics.MonitoringSourceAllMetrics;
const MonitoringSourceDetails = __importStar(require("./monitoring-source-details"));
exports.MonitoringSourceDetails = MonitoringSourceDetails.MonitoringSourceDetails;
const MonitoringSourceDetailsResponse = __importStar(require("./monitoring-source-details-response"));
exports.MonitoringSourceDetailsResponse = MonitoringSourceDetailsResponse.MonitoringSourceDetailsResponse;
const MonitoringSourceSelectedNamespaceDetails = __importStar(require("./monitoring-source-selected-namespace-details"));
exports.MonitoringSourceSelectedNamespaceDetails = MonitoringSourceSelectedNamespaceDetails.MonitoringSourceSelectedNamespaceDetails;
const MonitoringTargetDetails = __importStar(require("./monitoring-target-details"));
exports.MonitoringTargetDetails = MonitoringTargetDetails.MonitoringTargetDetails;
const MonitoringTargetDetailsResponse = __importStar(require("./monitoring-target-details-response"));
exports.MonitoringTargetDetailsResponse = MonitoringTargetDetailsResponse.MonitoringTargetDetailsResponse;
const NotificationsTargetDetails = __importStar(require("./notifications-target-details"));
exports.NotificationsTargetDetails = NotificationsTargetDetails.NotificationsTargetDetails;
const NotificationsTargetDetailsResponse = __importStar(require("./notifications-target-details-response"));
exports.NotificationsTargetDetailsResponse = NotificationsTargetDetailsResponse.NotificationsTargetDetailsResponse;
const ObjectStorageTargetDetails = __importStar(require("./object-storage-target-details"));
exports.ObjectStorageTargetDetails = ObjectStorageTargetDetails.ObjectStorageTargetDetails;
const ObjectStorageTargetDetailsResponse = __importStar(require("./object-storage-target-details-response"));
exports.ObjectStorageTargetDetailsResponse = ObjectStorageTargetDetailsResponse.ObjectStorageTargetDetailsResponse;
const PluginSourceDetails = __importStar(require("./plugin-source-details"));
exports.PluginSourceDetails = PluginSourceDetails.PluginSourceDetails;
const PluginSourceDetailsResponse = __importStar(require("./plugin-source-details-response"));
exports.PluginSourceDetailsResponse = PluginSourceDetailsResponse.PluginSourceDetailsResponse;
const SourceConnectorPlugin = __importStar(require("./source-connector-plugin"));
exports.SourceConnectorPlugin = SourceConnectorPlugin.SourceConnectorPlugin;
const SourceConnectorPluginSummary = __importStar(require("./source-connector-plugin-summary"));
exports.SourceConnectorPluginSummary = SourceConnectorPluginSummary.SourceConnectorPluginSummary;
const StaticDimensionValue = __importStar(require("./static-dimension-value"));
exports.StaticDimensionValue = StaticDimensionValue.StaticDimensionValue;
const StreamingSourceDetails = __importStar(require("./streaming-source-details"));
exports.StreamingSourceDetails = StreamingSourceDetails.StreamingSourceDetails;
const StreamingSourceDetailsResponse = __importStar(require("./streaming-source-details-response"));
exports.StreamingSourceDetailsResponse = StreamingSourceDetailsResponse.StreamingSourceDetailsResponse;
const StreamingTargetDetails = __importStar(require("./streaming-target-details"));
exports.StreamingTargetDetails = StreamingTargetDetails.StreamingTargetDetails;
const StreamingTargetDetailsResponse = __importStar(require("./streaming-target-details-response"));
exports.StreamingTargetDetailsResponse = StreamingTargetDetailsResponse.StreamingTargetDetailsResponse;
const TargetConnectorPlugin = __importStar(require("./target-connector-plugin"));
exports.TargetConnectorPlugin = TargetConnectorPlugin.TargetConnectorPlugin;
const TargetConnectorPluginSummary = __importStar(require("./target-connector-plugin-summary"));
exports.TargetConnectorPluginSummary = TargetConnectorPluginSummary.TargetConnectorPluginSummary;
const TrimHorizonStreamingCursor = __importStar(require("./trim-horizon-streaming-cursor"));
exports.TrimHorizonStreamingCursor = TrimHorizonStreamingCursor.TrimHorizonStreamingCursor;
