// oci-computecloudatcustomer/lib/request/list-ccc-infrastructures-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20221208
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListCccInfrastructuresRequest = void 0;
var ListCccInfrastructuresRequest;
(function (ListCccInfrastructuresRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListCccInfrastructuresRequest.AccessLevel || (ListCccInfrastructuresRequest.AccessLevel = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListCccInfrastructuresRequest.SortBy || (ListCccInfrastructuresRequest.SortBy = {}));
})(ListCccInfrastructuresRequest = exports.ListCccInfrastructuresRequest || (exports.ListCccInfrastructuresRequest = {}));
