// oci-apmtraces/lib/model/attribute-update-notes-response.js
"use strict";
/**
 * Application Performance Monitoring Trace Explorer API
 * Use the Application Performance Monitoring Trace Explorer API to query traces and associated spans in Trace Explorer. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttributeUpdateNotesResponse = void 0;
var AttributeUpdateNotesResponse;
(function (AttributeUpdateNotesResponse) {
    let OperationType;
    (function (OperationType) {
        OperationType["UpdateNotes"] = "UPDATE_NOTES";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OperationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(OperationType = AttributeUpdateNotesResponse.OperationType || (AttributeUpdateNotesResponse.OperationType = {}));
    let AttributeNameSpace;
    (function (AttributeNameSpace) {
        AttributeNameSpace["Traces"] = "TRACES";
        AttributeNameSpace["Synthetic"] = "SYNTHETIC";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AttributeNameSpace["UnknownValue"] = "UNKNOWN_VALUE";
    })(AttributeNameSpace = AttributeUpdateNotesResponse.AttributeNameSpace || (AttributeUpdateNotesResponse.AttributeNameSpace = {}));
    let AttributeStatus;
    (function (AttributeStatus) {
        AttributeStatus["AttributeNotesUpdated"] = "ATTRIBUTE_NOTES_UPDATED";
        AttributeStatus["DuplicateAttribute"] = "DUPLICATE_ATTRIBUTE";
        AttributeStatus["InvalidAttribute"] = "INVALID_ATTRIBUTE";
        AttributeStatus["AttributeNotProcessed"] = "ATTRIBUTE_NOT_PROCESSED";
        AttributeStatus["AttributeDoesNotExist"] = "ATTRIBUTE_DOES_NOT_EXIST";
        AttributeStatus["NotesTooLong"] = "NOTES_TOO_LONG";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AttributeStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(AttributeStatus = AttributeUpdateNotesResponse.AttributeStatus || (AttributeUpdateNotesResponse.AttributeStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttributeUpdateNotesResponse.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttributeUpdateNotesResponse.getDeserializedJsonObj = getDeserializedJsonObj;
})(AttributeUpdateNotesResponse = exports.AttributeUpdateNotesResponse || (exports.AttributeUpdateNotesResponse = {}));
