// oci-apmtraces/lib/model/attribute-unpin-response.js
"use strict";
/**
 * Application Performance Monitoring Trace Explorer API
 * Use the Application Performance Monitoring Trace Explorer API to query traces and associated spans in Trace Explorer. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttributeUnpinResponse = void 0;
var AttributeUnpinResponse;
(function (AttributeUnpinResponse) {
    let OperationType;
    (function (OperationType) {
        OperationType["Unpin"] = "UNPIN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OperationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(OperationType = AttributeUnpinResponse.OperationType || (AttributeUnpinResponse.OperationType = {}));
    let AttributeNameSpace;
    (function (AttributeNameSpace) {
        AttributeNameSpace["Traces"] = "TRACES";
        AttributeNameSpace["Synthetic"] = "SYNTHETIC";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AttributeNameSpace["UnknownValue"] = "UNKNOWN_VALUE";
    })(AttributeNameSpace = AttributeUnpinResponse.AttributeNameSpace || (AttributeUnpinResponse.AttributeNameSpace = {}));
    let AttributeStatus;
    (function (AttributeStatus) {
        AttributeStatus["AttributeUnpinned"] = "ATTRIBUTE_UNPINNED";
        AttributeStatus["DuplicateAttribute"] = "DUPLICATE_ATTRIBUTE";
        AttributeStatus["InvalidAttribute"] = "INVALID_ATTRIBUTE";
        AttributeStatus["AttributeNotFound"] = "ATTRIBUTE_NOT_FOUND";
        AttributeStatus["AttributeNotProcessed"] = "ATTRIBUTE_NOT_PROCESSED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AttributeStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(AttributeStatus = AttributeUnpinResponse.AttributeStatus || (AttributeUnpinResponse.AttributeStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttributeUnpinResponse.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttributeUnpinResponse.getDeserializedJsonObj = getDeserializedJsonObj;
})(AttributeUnpinResponse = exports.AttributeUnpinResponse || (exports.AttributeUnpinResponse = {}));
