// oci-apmtraces/lib/model/auto-activate-toggle-status.js
"use strict";
/**
 * Application Performance Monitoring Trace Explorer API
 * Use the Application Performance Monitoring Trace Explorer API to query traces and associated spans in Trace Explorer. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutoActivateToggleStatus = void 0;
var AutoActivateToggleStatus;
(function (AutoActivateToggleStatus) {
    let Status;
    (function (Status) {
        Status["Success"] = "SUCCESS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = AutoActivateToggleStatus.Status || (AutoActivateToggleStatus.Status = {}));
    let State;
    (function (State) {
        State["On"] = "ON";
        State["Off"] = "OFF";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        State["UnknownValue"] = "UNKNOWN_VALUE";
    })(State = AutoActivateToggleStatus.State || (AutoActivateToggleStatus.State = {}));
    let DataKey;
    (function (DataKey) {
        DataKey["PrivateDataKey"] = "PRIVATE_DATA_KEY";
        DataKey["PublicDataKey"] = "PUBLIC_DATA_KEY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DataKey["UnknownValue"] = "UNKNOWN_VALUE";
    })(DataKey = AutoActivateToggleStatus.DataKey || (AutoActivateToggleStatus.DataKey = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AutoActivateToggleStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AutoActivateToggleStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(AutoActivateToggleStatus = exports.AutoActivateToggleStatus || (exports.AutoActivateToggleStatus = {}));
