// oci-apmtraces/lib/model/bulk-unpin-attribute-detail.js
"use strict";
/**
 * Application Performance Monitoring Trace Explorer API
 * Use the Application Performance Monitoring Trace Explorer API to query traces and associated spans in Trace Explorer. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BulkUnpinAttributeDetail = void 0;
var BulkUnpinAttributeDetail;
(function (BulkUnpinAttributeDetail) {
    let AttributeNameSpace;
    (function (AttributeNameSpace) {
        AttributeNameSpace["Traces"] = "TRACES";
        AttributeNameSpace["Synthetic"] = "SYNTHETIC";
        AttributeNameSpace["Logs"] = "LOGS";
    })(AttributeNameSpace = BulkUnpinAttributeDetail.AttributeNameSpace || (BulkUnpinAttributeDetail.AttributeNameSpace = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BulkUnpinAttributeDetail.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BulkUnpinAttributeDetail.getDeserializedJsonObj = getDeserializedJsonObj;
})(BulkUnpinAttributeDetail = exports.BulkUnpinAttributeDetail || (exports.BulkUnpinAttributeDetail = {}));
