// oci-apmtraces/lib/model/bulk-update-attribute-metadata.js
"use strict";
/**
 * Application Performance Monitoring Trace Explorer API
 * Use the Application Performance Monitoring Trace Explorer API to query traces and associated spans in Trace Explorer. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BulkUpdateAttributeMetadata = void 0;
var BulkUpdateAttributeMetadata;
(function (BulkUpdateAttributeMetadata) {
    let OperationStatus;
    (function (OperationStatus) {
        OperationStatus["Success"] = "SUCCESS";
        OperationStatus["EmptyAttributeList"] = "EMPTY_ATTRIBUTE_LIST";
        OperationStatus["InvalidBulkRequest"] = "INVALID_BULK_REQUEST";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OperationStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(OperationStatus = BulkUpdateAttributeMetadata.OperationStatus || (BulkUpdateAttributeMetadata.OperationStatus = {}));
    let OperationType;
    (function (OperationType) {
        OperationType["UpdateAttributeProperties"] = "UPDATE_ATTRIBUTE_PROPERTIES";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OperationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(OperationType = BulkUpdateAttributeMetadata.OperationType || (BulkUpdateAttributeMetadata.OperationType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BulkUpdateAttributeMetadata.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BulkUpdateAttributeMetadata.getDeserializedJsonObj = getDeserializedJsonObj;
})(BulkUpdateAttributeMetadata = exports.BulkUpdateAttributeMetadata || (exports.BulkUpdateAttributeMetadata = {}));
