// oci-apmtraces/lib/model/index.js
"use strict";
/**
 * Application Performance Monitoring Trace Explorer API
 * Use the Application Performance Monitoring Trace Explorer API to query traces and associated spans in Trace Explorer. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryResultsTopologyNodeInfo = exports.QueryResultsTopologyInfo = exports.QueryResultsOrderedBySummary = exports.QueryResultsGroupedBySummary = exports.QueryResultWarning = exports.QueryResultRowTypeSummary = exports.QueryResultRow = exports.QueryResultResponse = exports.QueryResultMetadataSummary = exports.QueryDetails = exports.ObjectStorage = exports.Log = exports.LifecycleStates = exports.CustomMetric = exports.CreateScheduledQueryDetails = exports.BulkUpdateNotesStatus = exports.BulkUpdateNotesMetadata = exports.BulkUpdateAttributeStatus = exports.BulkUpdateAttributeNotesDetails = exports.BulkUpdateAttributeNotesDetail = exports.BulkUpdateAttributeMetadata = exports.BulkUpdateAttributeDetails = exports.BulkUpdateAttributeDetail = exports.BulkUnpinStatus = exports.BulkUnpinMetadata = exports.BulkUnpinAttributeDetails = exports.BulkUnpinAttributeDetail = exports.BulkPinStatus = exports.BulkPinMetadata = exports.BulkPinAttributeDetails = exports.BulkPinAttributeDetail = exports.BulkDeActivationStatus = exports.BulkDeActivationMetadata = exports.BulkDeActivateAttributeDetails = exports.BulkDeActivateAttributeDetail = exports.BulkActivationStatus = exports.BulkActivationMetadata = exports.BulkActivateAttributeDetails = exports.BulkActivateAttributeDetail = exports.AutoActivateToggleStatus = exports.AutoActivateStatus = exports.AttributeUpdateResponse = exports.AttributeUpdateNotesResponse = exports.AttributeUnpinResponse = exports.AttributeResponse = exports.AttributePinResponse = exports.AttributeMetadata = exports.Attribute = exports.AggregatedStackTrace = exports.AggregatedSnapshot = void 0;
exports.UpdateScheduledQueryDetails = exports.TraceSpanSummary = exports.TraceSnapshot = exports.TraceServiceSummary = exports.Trace = exports.ThreadSnapshot = exports.TagMetadata = exports.Tag = exports.Streaming = exports.StackTraceElement = exports.SpanSnapshot = exports.SpanLogCollection = exports.SpanLog = exports.Span = exports.SortOrder = exports.SnapshotDetail = exports.ScheduledQuerySummary = exports.ScheduledQueryRetentionCriteria = exports.ScheduledQueryProcessingType = exports.ScheduledQueryProcessingSubType = exports.ScheduledQueryProcessingConfig = exports.ScheduledQueryCollection = exports.ScheduledQuery = exports.QuickPickSummary = void 0;
const AggregatedSnapshot = __importStar(require("./aggregated-snapshot"));
exports.AggregatedSnapshot = AggregatedSnapshot.AggregatedSnapshot;
const AggregatedStackTrace = __importStar(require("./aggregated-stack-trace"));
exports.AggregatedStackTrace = AggregatedStackTrace.AggregatedStackTrace;
const Attribute = __importStar(require("./attribute"));
exports.Attribute = Attribute.Attribute;
const AttributeMetadata = __importStar(require("./attribute-metadata"));
exports.AttributeMetadata = AttributeMetadata.AttributeMetadata;
const AttributePinResponse = __importStar(require("./attribute-pin-response"));
exports.AttributePinResponse = AttributePinResponse.AttributePinResponse;
const AttributeResponse = __importStar(require("./attribute-response"));
exports.AttributeResponse = AttributeResponse.AttributeResponse;
const AttributeUnpinResponse = __importStar(require("./attribute-unpin-response"));
exports.AttributeUnpinResponse = AttributeUnpinResponse.AttributeUnpinResponse;
const AttributeUpdateNotesResponse = __importStar(require("./attribute-update-notes-response"));
exports.AttributeUpdateNotesResponse = AttributeUpdateNotesResponse.AttributeUpdateNotesResponse;
const AttributeUpdateResponse = __importStar(require("./attribute-update-response"));
exports.AttributeUpdateResponse = AttributeUpdateResponse.AttributeUpdateResponse;
const AutoActivateStatus = __importStar(require("./auto-activate-status"));
exports.AutoActivateStatus = AutoActivateStatus.AutoActivateStatus;
const AutoActivateToggleStatus = __importStar(require("./auto-activate-toggle-status"));
exports.AutoActivateToggleStatus = AutoActivateToggleStatus.AutoActivateToggleStatus;
const BulkActivateAttributeDetail = __importStar(require("./bulk-activate-attribute-detail"));
exports.BulkActivateAttributeDetail = BulkActivateAttributeDetail.BulkActivateAttributeDetail;
const BulkActivateAttributeDetails = __importStar(require("./bulk-activate-attribute-details"));
exports.BulkActivateAttributeDetails = BulkActivateAttributeDetails.BulkActivateAttributeDetails;
const BulkActivationMetadata = __importStar(require("./bulk-activation-metadata"));
exports.BulkActivationMetadata = BulkActivationMetadata.BulkActivationMetadata;
const BulkActivationStatus = __importStar(require("./bulk-activation-status"));
exports.BulkActivationStatus = BulkActivationStatus.BulkActivationStatus;
const BulkDeActivateAttributeDetail = __importStar(require("./bulk-de-activate-attribute-detail"));
exports.BulkDeActivateAttributeDetail = BulkDeActivateAttributeDetail.BulkDeActivateAttributeDetail;
const BulkDeActivateAttributeDetails = __importStar(require("./bulk-de-activate-attribute-details"));
exports.BulkDeActivateAttributeDetails = BulkDeActivateAttributeDetails.BulkDeActivateAttributeDetails;
const BulkDeActivationMetadata = __importStar(require("./bulk-de-activation-metadata"));
exports.BulkDeActivationMetadata = BulkDeActivationMetadata.BulkDeActivationMetadata;
const BulkDeActivationStatus = __importStar(require("./bulk-de-activation-status"));
exports.BulkDeActivationStatus = BulkDeActivationStatus.BulkDeActivationStatus;
const BulkPinAttributeDetail = __importStar(require("./bulk-pin-attribute-detail"));
exports.BulkPinAttributeDetail = BulkPinAttributeDetail.BulkPinAttributeDetail;
const BulkPinAttributeDetails = __importStar(require("./bulk-pin-attribute-details"));
exports.BulkPinAttributeDetails = BulkPinAttributeDetails.BulkPinAttributeDetails;
const BulkPinMetadata = __importStar(require("./bulk-pin-metadata"));
exports.BulkPinMetadata = BulkPinMetadata.BulkPinMetadata;
const BulkPinStatus = __importStar(require("./bulk-pin-status"));
exports.BulkPinStatus = BulkPinStatus.BulkPinStatus;
const BulkUnpinAttributeDetail = __importStar(require("./bulk-unpin-attribute-detail"));
exports.BulkUnpinAttributeDetail = BulkUnpinAttributeDetail.BulkUnpinAttributeDetail;
const BulkUnpinAttributeDetails = __importStar(require("./bulk-unpin-attribute-details"));
exports.BulkUnpinAttributeDetails = BulkUnpinAttributeDetails.BulkUnpinAttributeDetails;
const BulkUnpinMetadata = __importStar(require("./bulk-unpin-metadata"));
exports.BulkUnpinMetadata = BulkUnpinMetadata.BulkUnpinMetadata;
const BulkUnpinStatus = __importStar(require("./bulk-unpin-status"));
exports.BulkUnpinStatus = BulkUnpinStatus.BulkUnpinStatus;
const BulkUpdateAttributeDetail = __importStar(require("./bulk-update-attribute-detail"));
exports.BulkUpdateAttributeDetail = BulkUpdateAttributeDetail.BulkUpdateAttributeDetail;
const BulkUpdateAttributeDetails = __importStar(require("./bulk-update-attribute-details"));
exports.BulkUpdateAttributeDetails = BulkUpdateAttributeDetails.BulkUpdateAttributeDetails;
const BulkUpdateAttributeMetadata = __importStar(require("./bulk-update-attribute-metadata"));
exports.BulkUpdateAttributeMetadata = BulkUpdateAttributeMetadata.BulkUpdateAttributeMetadata;
const BulkUpdateAttributeNotesDetail = __importStar(require("./bulk-update-attribute-notes-detail"));
exports.BulkUpdateAttributeNotesDetail = BulkUpdateAttributeNotesDetail.BulkUpdateAttributeNotesDetail;
const BulkUpdateAttributeNotesDetails = __importStar(require("./bulk-update-attribute-notes-details"));
exports.BulkUpdateAttributeNotesDetails = BulkUpdateAttributeNotesDetails.BulkUpdateAttributeNotesDetails;
const BulkUpdateAttributeStatus = __importStar(require("./bulk-update-attribute-status"));
exports.BulkUpdateAttributeStatus = BulkUpdateAttributeStatus.BulkUpdateAttributeStatus;
const BulkUpdateNotesMetadata = __importStar(require("./bulk-update-notes-metadata"));
exports.BulkUpdateNotesMetadata = BulkUpdateNotesMetadata.BulkUpdateNotesMetadata;
const BulkUpdateNotesStatus = __importStar(require("./bulk-update-notes-status"));
exports.BulkUpdateNotesStatus = BulkUpdateNotesStatus.BulkUpdateNotesStatus;
const CreateScheduledQueryDetails = __importStar(require("./create-scheduled-query-details"));
exports.CreateScheduledQueryDetails = CreateScheduledQueryDetails.CreateScheduledQueryDetails;
const CustomMetric = __importStar(require("./custom-metric"));
exports.CustomMetric = CustomMetric.CustomMetric;
const LifecycleStates = __importStar(require("./lifecycle-states"));
exports.LifecycleStates = LifecycleStates.LifecycleStates;
const Log = __importStar(require("./log"));
exports.Log = Log.Log;
const ObjectStorage = __importStar(require("./object-storage"));
exports.ObjectStorage = ObjectStorage.ObjectStorage;
const QueryDetails = __importStar(require("./query-details"));
exports.QueryDetails = QueryDetails.QueryDetails;
const QueryResultMetadataSummary = __importStar(require("./query-result-metadata-summary"));
exports.QueryResultMetadataSummary = QueryResultMetadataSummary.QueryResultMetadataSummary;
const QueryResultResponse = __importStar(require("./query-result-response"));
exports.QueryResultResponse = QueryResultResponse.QueryResultResponse;
const QueryResultRow = __importStar(require("./query-result-row"));
exports.QueryResultRow = QueryResultRow.QueryResultRow;
const QueryResultRowTypeSummary = __importStar(require("./query-result-row-type-summary"));
exports.QueryResultRowTypeSummary = QueryResultRowTypeSummary.QueryResultRowTypeSummary;
const QueryResultWarning = __importStar(require("./query-result-warning"));
exports.QueryResultWarning = QueryResultWarning.QueryResultWarning;
const QueryResultsGroupedBySummary = __importStar(require("./query-results-grouped-by-summary"));
exports.QueryResultsGroupedBySummary = QueryResultsGroupedBySummary.QueryResultsGroupedBySummary;
const QueryResultsOrderedBySummary = __importStar(require("./query-results-ordered-by-summary"));
exports.QueryResultsOrderedBySummary = QueryResultsOrderedBySummary.QueryResultsOrderedBySummary;
const QueryResultsTopologyInfo = __importStar(require("./query-results-topology-info"));
exports.QueryResultsTopologyInfo = QueryResultsTopologyInfo.QueryResultsTopologyInfo;
const QueryResultsTopologyNodeInfo = __importStar(require("./query-results-topology-node-info"));
exports.QueryResultsTopologyNodeInfo = QueryResultsTopologyNodeInfo.QueryResultsTopologyNodeInfo;
const QuickPickSummary = __importStar(require("./quick-pick-summary"));
exports.QuickPickSummary = QuickPickSummary.QuickPickSummary;
const ScheduledQuery = __importStar(require("./scheduled-query"));
exports.ScheduledQuery = ScheduledQuery.ScheduledQuery;
const ScheduledQueryCollection = __importStar(require("./scheduled-query-collection"));
exports.ScheduledQueryCollection = ScheduledQueryCollection.ScheduledQueryCollection;
const ScheduledQueryProcessingConfig = __importStar(require("./scheduled-query-processing-config"));
exports.ScheduledQueryProcessingConfig = ScheduledQueryProcessingConfig.ScheduledQueryProcessingConfig;
const ScheduledQueryProcessingSubType = __importStar(require("./scheduled-query-processing-sub-type"));
exports.ScheduledQueryProcessingSubType = ScheduledQueryProcessingSubType.ScheduledQueryProcessingSubType;
const ScheduledQueryProcessingType = __importStar(require("./scheduled-query-processing-type"));
exports.ScheduledQueryProcessingType = ScheduledQueryProcessingType.ScheduledQueryProcessingType;
const ScheduledQueryRetentionCriteria = __importStar(require("./scheduled-query-retention-criteria"));
exports.ScheduledQueryRetentionCriteria = ScheduledQueryRetentionCriteria.ScheduledQueryRetentionCriteria;
const ScheduledQuerySummary = __importStar(require("./scheduled-query-summary"));
exports.ScheduledQuerySummary = ScheduledQuerySummary.ScheduledQuerySummary;
const SnapshotDetail = __importStar(require("./snapshot-detail"));
exports.SnapshotDetail = SnapshotDetail.SnapshotDetail;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Span = __importStar(require("./span"));
exports.Span = Span.Span;
const SpanLog = __importStar(require("./span-log"));
exports.SpanLog = SpanLog.SpanLog;
const SpanLogCollection = __importStar(require("./span-log-collection"));
exports.SpanLogCollection = SpanLogCollection.SpanLogCollection;
const SpanSnapshot = __importStar(require("./span-snapshot"));
exports.SpanSnapshot = SpanSnapshot.SpanSnapshot;
const StackTraceElement = __importStar(require("./stack-trace-element"));
exports.StackTraceElement = StackTraceElement.StackTraceElement;
const Streaming = __importStar(require("./streaming"));
exports.Streaming = Streaming.Streaming;
const Tag = __importStar(require("./tag"));
exports.Tag = Tag.Tag;
const TagMetadata = __importStar(require("./tag-metadata"));
exports.TagMetadata = TagMetadata.TagMetadata;
const ThreadSnapshot = __importStar(require("./thread-snapshot"));
exports.ThreadSnapshot = ThreadSnapshot.ThreadSnapshot;
const Trace = __importStar(require("./trace"));
exports.Trace = Trace.Trace;
const TraceServiceSummary = __importStar(require("./trace-service-summary"));
exports.TraceServiceSummary = TraceServiceSummary.TraceServiceSummary;
const TraceSnapshot = __importStar(require("./trace-snapshot"));
exports.TraceSnapshot = TraceSnapshot.TraceSnapshot;
const TraceSpanSummary = __importStar(require("./trace-span-summary"));
exports.TraceSpanSummary = TraceSpanSummary.TraceSpanSummary;
const UpdateScheduledQueryDetails = __importStar(require("./update-scheduled-query-details"));
exports.UpdateScheduledQueryDetails = UpdateScheduledQueryDetails.UpdateScheduledQueryDetails;
