// oci-apmtraces/lib/model/scheduled-query-processing-type.js
"use strict";
/**
 * Application Performance Monitoring Trace Explorer API
 * Use the Application Performance Monitoring Trace Explorer API to query traces and associated spans in Trace Explorer. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduledQueryProcessingType = void 0;
/**
 * A processing type for the scheduled query
 *
 **/
var ScheduledQueryProcessingType;
(function (ScheduledQueryProcessingType) {
    ScheduledQueryProcessingType["Export"] = "EXPORT";
    ScheduledQueryProcessingType["Query"] = "QUERY";
    ScheduledQueryProcessingType["Alert"] = "ALERT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ScheduledQueryProcessingType["UnknownValue"] = "UNKNOWN_VALUE";
})(ScheduledQueryProcessingType = exports.ScheduledQueryProcessingType || (exports.ScheduledQueryProcessingType = {}));
(function (ScheduledQueryProcessingType) {
    function getJsonObj(obj) {
        return obj;
    }
    ScheduledQueryProcessingType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ScheduledQueryProcessingType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ScheduledQueryProcessingType = exports.ScheduledQueryProcessingType || (exports.ScheduledQueryProcessingType = {}));
