// oci-apmtraces/lib/model/tag-metadata.js
"use strict";
/**
 * Application Performance Monitoring Trace Explorer API
 * Use the Application Performance Monitoring Trace Explorer API to query traces and associated spans in Trace Explorer. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagMetadata = void 0;
var TagMetadata;
(function (TagMetadata) {
    let TagType;
    (function (TagType) {
        TagType["String"] = "STRING";
        TagType["Numeric"] = "NUMERIC";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TagType["UnknownValue"] = "UNKNOWN_VALUE";
    })(TagType = TagMetadata.TagType || (TagMetadata.TagType = {}));
    let TagUnit;
    (function (TagUnit) {
        TagUnit["EpochTimeMs"] = "EPOCH_TIME_MS";
        TagUnit["Bytes"] = "BYTES";
        TagUnit["Count"] = "COUNT";
        TagUnit["DurationMs"] = "DURATION_MS";
        TagUnit["TraceStatus"] = "TRACE_STATUS";
        TagUnit["Percentage"] = "PERCENTAGE";
        TagUnit["None"] = "NONE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TagUnit["UnknownValue"] = "UNKNOWN_VALUE";
    })(TagUnit = TagMetadata.TagUnit || (TagMetadata.TagUnit = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    TagMetadata.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    TagMetadata.getDeserializedJsonObj = getDeserializedJsonObj;
})(TagMetadata = exports.TagMetadata || (exports.TagMetadata = {}));
