// oci-resourcescheduler/lib/model/index.js
"use strict";
/**
 * Resource Scheduler API
 * Use the Resource scheduler API to manage schedules, to perform actions on a collection of resources.
 * OpenAPI spec version: 20240430
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeCreatedResourceFilter = exports.ResourceTypeResourceFilter = exports.QueryParameter = exports.PathParameter = exports.LifecycleStateResourceFilter = exports.HeaderParameter = exports.DefinedTagsResourceFilter = exports.CompartmentIdResourceFilter = exports.BodyParameter = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateScheduleDetails = exports.SortOrder = exports.ScheduleSummary = exports.ScheduleCollection = exports.Schedule = exports.ResourceTypeCollection = exports.ResourceType = exports.ResourceFilter = exports.Resource = exports.Parameter = exports.OperationType = exports.OperationStatus = exports.DefinedTagFilterValue = exports.CreateScheduleDetails = exports.ChangeScheduleCompartmentDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangeScheduleCompartmentDetails = __importStar(require("./change-schedule-compartment-details"));
exports.ChangeScheduleCompartmentDetails = ChangeScheduleCompartmentDetails.ChangeScheduleCompartmentDetails;
const CreateScheduleDetails = __importStar(require("./create-schedule-details"));
exports.CreateScheduleDetails = CreateScheduleDetails.CreateScheduleDetails;
const DefinedTagFilterValue = __importStar(require("./defined-tag-filter-value"));
exports.DefinedTagFilterValue = DefinedTagFilterValue.DefinedTagFilterValue;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const Parameter = __importStar(require("./parameter"));
exports.Parameter = Parameter.Parameter;
const Resource = __importStar(require("./resource"));
exports.Resource = Resource.Resource;
const ResourceFilter = __importStar(require("./resource-filter"));
exports.ResourceFilter = ResourceFilter.ResourceFilter;
const ResourceType = __importStar(require("./resource-type"));
exports.ResourceType = ResourceType.ResourceType;
const ResourceTypeCollection = __importStar(require("./resource-type-collection"));
exports.ResourceTypeCollection = ResourceTypeCollection.ResourceTypeCollection;
const Schedule = __importStar(require("./schedule"));
exports.Schedule = Schedule.Schedule;
const ScheduleCollection = __importStar(require("./schedule-collection"));
exports.ScheduleCollection = ScheduleCollection.ScheduleCollection;
const ScheduleSummary = __importStar(require("./schedule-summary"));
exports.ScheduleSummary = ScheduleSummary.ScheduleSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateScheduleDetails = __importStar(require("./update-schedule-details"));
exports.UpdateScheduleDetails = UpdateScheduleDetails.UpdateScheduleDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const BodyParameter = __importStar(require("./body-parameter"));
exports.BodyParameter = BodyParameter.BodyParameter;
const CompartmentIdResourceFilter = __importStar(require("./compartment-id-resource-filter"));
exports.CompartmentIdResourceFilter = CompartmentIdResourceFilter.CompartmentIdResourceFilter;
const DefinedTagsResourceFilter = __importStar(require("./defined-tags-resource-filter"));
exports.DefinedTagsResourceFilter = DefinedTagsResourceFilter.DefinedTagsResourceFilter;
const HeaderParameter = __importStar(require("./header-parameter"));
exports.HeaderParameter = HeaderParameter.HeaderParameter;
const LifecycleStateResourceFilter = __importStar(require("./lifecycle-state-resource-filter"));
exports.LifecycleStateResourceFilter = LifecycleStateResourceFilter.LifecycleStateResourceFilter;
const PathParameter = __importStar(require("./path-parameter"));
exports.PathParameter = PathParameter.PathParameter;
const QueryParameter = __importStar(require("./query-parameter"));
exports.QueryParameter = QueryParameter.QueryParameter;
const ResourceTypeResourceFilter = __importStar(require("./resource-type-resource-filter"));
exports.ResourceTypeResourceFilter = ResourceTypeResourceFilter.ResourceTypeResourceFilter;
const TimeCreatedResourceFilter = __importStar(require("./time-created-resource-filter"));
exports.TimeCreatedResourceFilter = TimeCreatedResourceFilter.TimeCreatedResourceFilter;
