// oci-resourcescheduler/lib/model/time-created-resource-filter.js
"use strict";
/**
 * Resource Scheduler API
 * Use the Resource scheduler API to manage schedules, to perform actions on a collection of resources.
 * OpenAPI spec version: 20240430
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeCreatedResourceFilter = void 0;
const model = __importStar(require("../model"));
var TimeCreatedResourceFilter;
(function (TimeCreatedResourceFilter) {
    let Condition;
    (function (Condition) {
        Condition["Equal"] = "EQUAL";
        Condition["Before"] = "BEFORE";
        Condition["After"] = "AFTER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Condition["UnknownValue"] = "UNKNOWN_VALUE";
    })(Condition = TimeCreatedResourceFilter.Condition || (TimeCreatedResourceFilter.Condition = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.ResourceFilter.getJsonObj(obj))), {});
        return jsonObj;
    }
    TimeCreatedResourceFilter.getJsonObj = getJsonObj;
    TimeCreatedResourceFilter.attribute = "TIME_CREATED";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.ResourceFilter.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    TimeCreatedResourceFilter.getDeserializedJsonObj = getDeserializedJsonObj;
})(TimeCreatedResourceFilter = exports.TimeCreatedResourceFilter || (exports.TimeCreatedResourceFilter = {}));
