// oci-resourcescheduler/lib/model/update-schedule-details.js
"use strict";
/**
 * Resource Scheduler API
 * Use the Resource scheduler API to manage schedules, to perform actions on a collection of resources.
 * OpenAPI spec version: 20240430
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateScheduleDetails = void 0;
const model = __importStar(require("../model"));
var UpdateScheduleDetails;
(function (UpdateScheduleDetails) {
    let Action;
    (function (Action) {
        Action["StartResource"] = "START_RESOURCE";
        Action["StopResource"] = "STOP_RESOURCE";
    })(Action = UpdateScheduleDetails.Action || (UpdateScheduleDetails.Action = {}));
    let RecurrenceType;
    (function (RecurrenceType) {
        RecurrenceType["Cron"] = "CRON";
        RecurrenceType["Ical"] = "ICAL";
    })(RecurrenceType = UpdateScheduleDetails.RecurrenceType || (UpdateScheduleDetails.RecurrenceType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "resourceFilters": obj.resourceFilters
                ? obj.resourceFilters.map(item => {
                    return model.ResourceFilter.getJsonObj(item);
                })
                : undefined,
            "resources": obj.resources
                ? obj.resources.map(item => {
                    return model.Resource.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    UpdateScheduleDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "resourceFilters": obj.resourceFilters
                ? obj.resourceFilters.map(item => {
                    return model.ResourceFilter.getDeserializedJsonObj(item);
                })
                : undefined,
            "resources": obj.resources
                ? obj.resources.map(item => {
                    return model.Resource.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    UpdateScheduleDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateScheduleDetails = exports.UpdateScheduleDetails || (exports.UpdateScheduleDetails = {}));
