// oci-desktops/lib/model/create-desktop-pool-details.js
"use strict";
/**
 * Secure Desktops API
 * Create and manage cloud-hosted desktops which can be accessed from a web browser or installed client.

 * OpenAPI spec version: 20220618
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDesktopPoolDetails = void 0;
const model = __importStar(require("../model"));
var CreateDesktopPoolDetails;
(function (CreateDesktopPoolDetails) {
    let UseDedicatedVmHost;
    (function (UseDedicatedVmHost) {
        UseDedicatedVmHost["True"] = "TRUE";
        UseDedicatedVmHost["False"] = "FALSE";
        UseDedicatedVmHost["Auto"] = "AUTO";
    })(UseDedicatedVmHost = CreateDesktopPoolDetails.UseDedicatedVmHost || (CreateDesktopPoolDetails.UseDedicatedVmHost = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "shapeConfig": obj.shapeConfig
                ? model.CreateDesktopPoolShapeConfigDetails.getJsonObj(obj.shapeConfig)
                : undefined,
            "devicePolicy": obj.devicePolicy
                ? model.DesktopDevicePolicy.getJsonObj(obj.devicePolicy)
                : undefined,
            "availabilityPolicy": obj.availabilityPolicy
                ? model.DesktopAvailabilityPolicy.getJsonObj(obj.availabilityPolicy)
                : undefined,
            "image": obj.image ? model.DesktopImage.getJsonObj(obj.image) : undefined,
            "networkConfiguration": obj.networkConfiguration
                ? model.DesktopNetworkConfiguration.getJsonObj(obj.networkConfiguration)
                : undefined,
            "sessionLifecycleActions": obj.sessionLifecycleActions
                ? model.CreateDesktopPoolDesktopSessionLifecycleActions.getJsonObj(obj.sessionLifecycleActions)
                : undefined,
            "privateAccessDetails": obj.privateAccessDetails
                ? model.CreateDesktopPoolPrivateAccessDetails.getJsonObj(obj.privateAccessDetails)
                : undefined
        });
        return jsonObj;
    }
    CreateDesktopPoolDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "shapeConfig": obj.shapeConfig
                ? model.CreateDesktopPoolShapeConfigDetails.getDeserializedJsonObj(obj.shapeConfig)
                : undefined,
            "devicePolicy": obj.devicePolicy
                ? model.DesktopDevicePolicy.getDeserializedJsonObj(obj.devicePolicy)
                : undefined,
            "availabilityPolicy": obj.availabilityPolicy
                ? model.DesktopAvailabilityPolicy.getDeserializedJsonObj(obj.availabilityPolicy)
                : undefined,
            "image": obj.image ? model.DesktopImage.getDeserializedJsonObj(obj.image) : undefined,
            "networkConfiguration": obj.networkConfiguration
                ? model.DesktopNetworkConfiguration.getDeserializedJsonObj(obj.networkConfiguration)
                : undefined,
            "sessionLifecycleActions": obj.sessionLifecycleActions
                ? model.CreateDesktopPoolDesktopSessionLifecycleActions.getDeserializedJsonObj(obj.sessionLifecycleActions)
                : undefined,
            "privateAccessDetails": obj.privateAccessDetails
                ? model.CreateDesktopPoolPrivateAccessDetails.getDeserializedJsonObj(obj.privateAccessDetails)
                : undefined
        });
        return jsonObj;
    }
    CreateDesktopPoolDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateDesktopPoolDetails = exports.CreateDesktopPoolDetails || (exports.CreateDesktopPoolDetails = {}));
