// oci-desktops/lib/model/create-desktop-pool-shape-config-details.js
"use strict";
/**
 * Secure Desktops API
 * Create and manage cloud-hosted desktops which can be accessed from a web browser or installed client.

 * OpenAPI spec version: 20220618
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDesktopPoolShapeConfigDetails = void 0;
var CreateDesktopPoolShapeConfigDetails;
(function (CreateDesktopPoolShapeConfigDetails) {
    let BaselineOcpuUtilization;
    (function (BaselineOcpuUtilization) {
        BaselineOcpuUtilization["Baseline18"] = "BASELINE_1_8";
        BaselineOcpuUtilization["Baseline12"] = "BASELINE_1_2";
        BaselineOcpuUtilization["Baseline11"] = "BASELINE_1_1";
    })(BaselineOcpuUtilization = CreateDesktopPoolShapeConfigDetails.BaselineOcpuUtilization || (CreateDesktopPoolShapeConfigDetails.BaselineOcpuUtilization = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateDesktopPoolShapeConfigDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateDesktopPoolShapeConfigDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateDesktopPoolShapeConfigDetails = exports.CreateDesktopPoolShapeConfigDetails || (exports.CreateDesktopPoolShapeConfigDetails = {}));
