// oci-desktops/lib/model/desktop-device-policy.js
"use strict";
/**
 * Secure Desktops API
 * Create and manage cloud-hosted desktops which can be accessed from a web browser or installed client.

 * OpenAPI spec version: 20220618
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DesktopDevicePolicy = void 0;
var DesktopDevicePolicy;
(function (DesktopDevicePolicy) {
    let ClipboardMode;
    (function (ClipboardMode) {
        ClipboardMode["None"] = "NONE";
        ClipboardMode["Todesktop"] = "TODESKTOP";
        ClipboardMode["Fromdesktop"] = "FROMDESKTOP";
        ClipboardMode["Full"] = "FULL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ClipboardMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(ClipboardMode = DesktopDevicePolicy.ClipboardMode || (DesktopDevicePolicy.ClipboardMode = {}));
    let AudioMode;
    (function (AudioMode) {
        AudioMode["None"] = "NONE";
        AudioMode["Todesktop"] = "TODESKTOP";
        AudioMode["Fromdesktop"] = "FROMDESKTOP";
        AudioMode["Full"] = "FULL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AudioMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(AudioMode = DesktopDevicePolicy.AudioMode || (DesktopDevicePolicy.AudioMode = {}));
    let CdmMode;
    (function (CdmMode) {
        CdmMode["None"] = "NONE";
        CdmMode["Readonly"] = "READONLY";
        CdmMode["Full"] = "FULL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        CdmMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(CdmMode = DesktopDevicePolicy.CdmMode || (DesktopDevicePolicy.CdmMode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DesktopDevicePolicy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DesktopDevicePolicy.getDeserializedJsonObj = getDeserializedJsonObj;
})(DesktopDevicePolicy = exports.DesktopDevicePolicy || (exports.DesktopDevicePolicy = {}));
