// oci-desktops/lib/model/index.js
"use strict";
/**
 * Secure Desktops API
 * Create and manage cloud-hosted desktops which can be accessed from a web browser or installed client.

 * OpenAPI spec version: 20220618
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateDesktopPoolDetails = exports.UpdateDesktopPoolDesktopSessionLifecycleActions = exports.UpdateDesktopDetails = exports.OperationType = exports.OperationStatus = exports.LifecycleState = exports.InactivityConfig = exports.HostingOptions = exports.DisconnectConfig = exports.DesktopSummary = exports.DesktopSessionLifecycleActions = exports.DesktopSchedule = exports.DesktopPoolVolumeSummary = exports.DesktopPoolVolumeCollection = exports.DesktopPoolSummary = exports.DesktopPoolShapeConfig = exports.DesktopPoolPrivateAccessDetails = exports.DesktopPoolDesktopSummary = exports.DesktopPoolDesktopCollection = exports.DesktopPoolCollection = exports.DesktopPool = exports.DesktopNetworkConfiguration = exports.DesktopImage = exports.DesktopDevicePolicy = exports.DesktopCollection = exports.DesktopAvailabilityPolicy = exports.Desktop = exports.CreateDesktopPoolShapeConfigDetails = exports.CreateDesktopPoolPrivateAccessDetails = exports.CreateDesktopPoolDetails = exports.CreateDesktopPoolDesktopSessionLifecycleActions = exports.ChangeDesktopPoolCompartmentDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangeDesktopPoolCompartmentDetails = __importStar(require("./change-desktop-pool-compartment-details"));
exports.ChangeDesktopPoolCompartmentDetails = ChangeDesktopPoolCompartmentDetails.ChangeDesktopPoolCompartmentDetails;
const CreateDesktopPoolDesktopSessionLifecycleActions = __importStar(require("./create-desktop-pool-desktop-session-lifecycle-actions"));
exports.CreateDesktopPoolDesktopSessionLifecycleActions = CreateDesktopPoolDesktopSessionLifecycleActions.CreateDesktopPoolDesktopSessionLifecycleActions;
const CreateDesktopPoolDetails = __importStar(require("./create-desktop-pool-details"));
exports.CreateDesktopPoolDetails = CreateDesktopPoolDetails.CreateDesktopPoolDetails;
const CreateDesktopPoolPrivateAccessDetails = __importStar(require("./create-desktop-pool-private-access-details"));
exports.CreateDesktopPoolPrivateAccessDetails = CreateDesktopPoolPrivateAccessDetails.CreateDesktopPoolPrivateAccessDetails;
const CreateDesktopPoolShapeConfigDetails = __importStar(require("./create-desktop-pool-shape-config-details"));
exports.CreateDesktopPoolShapeConfigDetails = CreateDesktopPoolShapeConfigDetails.CreateDesktopPoolShapeConfigDetails;
const Desktop = __importStar(require("./desktop"));
exports.Desktop = Desktop.Desktop;
const DesktopAvailabilityPolicy = __importStar(require("./desktop-availability-policy"));
exports.DesktopAvailabilityPolicy = DesktopAvailabilityPolicy.DesktopAvailabilityPolicy;
const DesktopCollection = __importStar(require("./desktop-collection"));
exports.DesktopCollection = DesktopCollection.DesktopCollection;
const DesktopDevicePolicy = __importStar(require("./desktop-device-policy"));
exports.DesktopDevicePolicy = DesktopDevicePolicy.DesktopDevicePolicy;
const DesktopImage = __importStar(require("./desktop-image"));
exports.DesktopImage = DesktopImage.DesktopImage;
const DesktopNetworkConfiguration = __importStar(require("./desktop-network-configuration"));
exports.DesktopNetworkConfiguration = DesktopNetworkConfiguration.DesktopNetworkConfiguration;
const DesktopPool = __importStar(require("./desktop-pool"));
exports.DesktopPool = DesktopPool.DesktopPool;
const DesktopPoolCollection = __importStar(require("./desktop-pool-collection"));
exports.DesktopPoolCollection = DesktopPoolCollection.DesktopPoolCollection;
const DesktopPoolDesktopCollection = __importStar(require("./desktop-pool-desktop-collection"));
exports.DesktopPoolDesktopCollection = DesktopPoolDesktopCollection.DesktopPoolDesktopCollection;
const DesktopPoolDesktopSummary = __importStar(require("./desktop-pool-desktop-summary"));
exports.DesktopPoolDesktopSummary = DesktopPoolDesktopSummary.DesktopPoolDesktopSummary;
const DesktopPoolPrivateAccessDetails = __importStar(require("./desktop-pool-private-access-details"));
exports.DesktopPoolPrivateAccessDetails = DesktopPoolPrivateAccessDetails.DesktopPoolPrivateAccessDetails;
const DesktopPoolShapeConfig = __importStar(require("./desktop-pool-shape-config"));
exports.DesktopPoolShapeConfig = DesktopPoolShapeConfig.DesktopPoolShapeConfig;
const DesktopPoolSummary = __importStar(require("./desktop-pool-summary"));
exports.DesktopPoolSummary = DesktopPoolSummary.DesktopPoolSummary;
const DesktopPoolVolumeCollection = __importStar(require("./desktop-pool-volume-collection"));
exports.DesktopPoolVolumeCollection = DesktopPoolVolumeCollection.DesktopPoolVolumeCollection;
const DesktopPoolVolumeSummary = __importStar(require("./desktop-pool-volume-summary"));
exports.DesktopPoolVolumeSummary = DesktopPoolVolumeSummary.DesktopPoolVolumeSummary;
const DesktopSchedule = __importStar(require("./desktop-schedule"));
exports.DesktopSchedule = DesktopSchedule.DesktopSchedule;
const DesktopSessionLifecycleActions = __importStar(require("./desktop-session-lifecycle-actions"));
exports.DesktopSessionLifecycleActions = DesktopSessionLifecycleActions.DesktopSessionLifecycleActions;
const DesktopSummary = __importStar(require("./desktop-summary"));
exports.DesktopSummary = DesktopSummary.DesktopSummary;
const DisconnectConfig = __importStar(require("./disconnect-config"));
exports.DisconnectConfig = DisconnectConfig.DisconnectConfig;
const HostingOptions = __importStar(require("./hosting-options"));
exports.HostingOptions = HostingOptions.HostingOptions;
const InactivityConfig = __importStar(require("./inactivity-config"));
exports.InactivityConfig = InactivityConfig.InactivityConfig;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const UpdateDesktopDetails = __importStar(require("./update-desktop-details"));
exports.UpdateDesktopDetails = UpdateDesktopDetails.UpdateDesktopDetails;
const UpdateDesktopPoolDesktopSessionLifecycleActions = __importStar(require("./update-desktop-pool-desktop-session-lifecycle-actions"));
exports.UpdateDesktopPoolDesktopSessionLifecycleActions = UpdateDesktopPoolDesktopSessionLifecycleActions.UpdateDesktopPoolDesktopSessionLifecycleActions;
const UpdateDesktopPoolDetails = __importStar(require("./update-desktop-pool-details"));
exports.UpdateDesktopPoolDetails = UpdateDesktopPoolDetails.UpdateDesktopPoolDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
