// oci-desktops/lib/model/operation-type.js
"use strict";
/**
 * Secure Desktops API
 * Create and manage cloud-hosted desktops which can be accessed from a web browser or installed client.

 * OpenAPI spec version: 20220618
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateDesktopPool"] = "CREATE_DESKTOP_POOL";
    OperationType["UpdateDesktopPool"] = "UPDATE_DESKTOP_POOL";
    OperationType["DeleteDesktopPool"] = "DELETE_DESKTOP_POOL";
    OperationType["MoveDesktopPool"] = "MOVE_DESKTOP_POOL";
    OperationType["StartDesktopPool"] = "START_DESKTOP_POOL";
    OperationType["StopDesktopPool"] = "STOP_DESKTOP_POOL";
    OperationType["DeleteDesktop"] = "DELETE_DESKTOP";
    OperationType["UpdateDesktop"] = "UPDATE_DESKTOP";
    OperationType["StartDesktop"] = "START_DESKTOP";
    OperationType["StopDesktop"] = "STOP_DESKTOP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
