// oci-datacatalog/lib/model/lineage-direction.js
"use strict";
/**
 * Data Catalog API
 * Use the Data Catalog APIs to collect, organize, find, access, understand, enrich, and activate technical, business, and operational metadata.
For more information, see [Data Catalog](https://docs.oracle.com/iaas/data-catalog/home.htm).

 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LineageDirection = void 0;
/**
 * Indicates the direction for lineage fetch. It is one of upstream (or source objects), downstream (or target
 * objects), both.
 * Corresponding enums are:
 * UPSTREAM   - Source objects.
 * DOWNSTREAM - Target objects.
 * BOTH       - Both source and target objects.
 *
 **/
var LineageDirection;
(function (LineageDirection) {
    LineageDirection["Upstream"] = "UPSTREAM";
    LineageDirection["Both"] = "BOTH";
    LineageDirection["Downstream"] = "DOWNSTREAM";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    LineageDirection["UnknownValue"] = "UNKNOWN_VALUE";
})(LineageDirection = exports.LineageDirection || (exports.LineageDirection = {}));
(function (LineageDirection) {
    function getJsonObj(obj) {
        return obj;
    }
    LineageDirection.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    LineageDirection.getDeserializedJsonObj = getDeserializedJsonObj;
})(LineageDirection = exports.LineageDirection || (exports.LineageDirection = {}));
