// oci-osmanagementhub/lib/model/event.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for instances in OCI, your private data center, or 3rd-party clouds.
For more information, see [Overview of OS Management Hub](https://docs.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Event = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var Event;
(function (Event) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Event.LifecycleState || (Event.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "systemDetails": obj.systemDetails
                ? model.SystemDetails.getJsonObj(obj.systemDetails)
                : undefined
        });
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "SOFTWARE_UPDATE":
                    return model.SoftwareUpdateEvent.getJsonObj(jsonObj, true);
                case "KERNEL_OOPS":
                    return model.KernelOopsEvent.getJsonObj(jsonObj, true);
                case "MANAGEMENT_STATION":
                    return model.ManagementStationEvent.getJsonObj(jsonObj, true);
                case "SOFTWARE_SOURCE":
                    return model.SoftwareSourceEvent.getJsonObj(jsonObj, true);
                case "KERNEL_CRASH":
                    return model.KernelCrashEvent.getJsonObj(jsonObj, true);
                case "SYSADMIN":
                    return model.SysadminEvent.getJsonObj(jsonObj, true);
                case "REBOOT":
                    return model.RebootEvent.getJsonObj(jsonObj, true);
                case "EXPLOIT_ATTEMPT":
                    return model.ExploitAttemptEvent.getJsonObj(jsonObj, true);
                case "AGENT":
                    return model.AgentEvent.getJsonObj(jsonObj, true);
                case "KSPLICE_UPDATE":
                    return model.KspliceUpdateEvent.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    Event.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "systemDetails": obj.systemDetails
                ? model.SystemDetails.getDeserializedJsonObj(obj.systemDetails)
                : undefined
        });
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "SOFTWARE_UPDATE":
                    return model.SoftwareUpdateEvent.getDeserializedJsonObj(jsonObj, true);
                case "KERNEL_OOPS":
                    return model.KernelOopsEvent.getDeserializedJsonObj(jsonObj, true);
                case "MANAGEMENT_STATION":
                    return model.ManagementStationEvent.getDeserializedJsonObj(jsonObj, true);
                case "SOFTWARE_SOURCE":
                    return model.SoftwareSourceEvent.getDeserializedJsonObj(jsonObj, true);
                case "KERNEL_CRASH":
                    return model.KernelCrashEvent.getDeserializedJsonObj(jsonObj, true);
                case "SYSADMIN":
                    return model.SysadminEvent.getDeserializedJsonObj(jsonObj, true);
                case "REBOOT":
                    return model.RebootEvent.getDeserializedJsonObj(jsonObj, true);
                case "EXPLOIT_ATTEMPT":
                    return model.ExploitAttemptEvent.getDeserializedJsonObj(jsonObj, true);
                case "AGENT":
                    return model.AgentEvent.getDeserializedJsonObj(jsonObj, true);
                case "KSPLICE_UPDATE":
                    return model.KspliceUpdateEvent.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    Event.getDeserializedJsonObj = getDeserializedJsonObj;
})(Event = exports.Event || (exports.Event = {}));
