// oci-osmanagementhub/lib/model/event-type.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for instances in OCI, your private data center, or 3rd-party clouds.
For more information, see [Overview of OS Management Hub](https://docs.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventType = void 0;
/**
 * Possible event types.
 **/
var EventType;
(function (EventType) {
    EventType["KernelOops"] = "KERNEL_OOPS";
    EventType["KernelCrash"] = "KERNEL_CRASH";
    EventType["ExploitAttempt"] = "EXPLOIT_ATTEMPT";
    EventType["SoftwareUpdate"] = "SOFTWARE_UPDATE";
    EventType["KspliceUpdate"] = "KSPLICE_UPDATE";
    EventType["SoftwareSource"] = "SOFTWARE_SOURCE";
    EventType["Agent"] = "AGENT";
    EventType["ManagementStation"] = "MANAGEMENT_STATION";
    EventType["Sysadmin"] = "SYSADMIN";
    EventType["Reboot"] = "REBOOT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    EventType["UnknownValue"] = "UNKNOWN_VALUE";
})(EventType = exports.EventType || (exports.EventType = {}));
(function (EventType) {
    function getJsonObj(obj) {
        return obj;
    }
    EventType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    EventType.getDeserializedJsonObj = getDeserializedJsonObj;
})(EventType = exports.EventType || (exports.EventType = {}));
