// oci-osmanagementhub/lib/model/kernel-event-content.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for instances in OCI, your private data center, or 3rd-party clouds.
For more information, see [Overview of OS Management Hub](https://docs.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.KernelEventContent = void 0;
const model = __importStar(require("../model"));
var KernelEventContent;
(function (KernelEventContent) {
    let ContentAvailability;
    (function (ContentAvailability) {
        ContentAvailability["NotAvailable"] = "NOT_AVAILABLE";
        ContentAvailability["AvailableOnInstance"] = "AVAILABLE_ON_INSTANCE";
        ContentAvailability["AvailableOnService"] = "AVAILABLE_ON_SERVICE";
        ContentAvailability["AvailableOnInstanceAndService"] = "AVAILABLE_ON_INSTANCE_AND_SERVICE";
        ContentAvailability["AvailableOnInstanceUploadInProgress"] = "AVAILABLE_ON_INSTANCE_UPLOAD_IN_PROGRESS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ContentAvailability["UnknownValue"] = "UNKNOWN_VALUE";
    })(ContentAvailability = KernelEventContent.ContentAvailability || (KernelEventContent.ContentAvailability = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.EventContent.getJsonObj(obj))), {});
        return jsonObj;
    }
    KernelEventContent.getJsonObj = getJsonObj;
    KernelEventContent.type = "KERNEL";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.EventContent.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    KernelEventContent.getDeserializedJsonObj = getDeserializedJsonObj;
})(KernelEventContent = exports.KernelEventContent || (exports.KernelEventContent = {}));
