// oci-osmanagementhub/lib/model/management-station-event-data.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for instances in OCI, your private data center, or 3rd-party clouds.
For more information, see [Overview of OS Management Hub](https://docs.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagementStationEventData = void 0;
const model = __importStar(require("../model"));
var ManagementStationEventData;
(function (ManagementStationEventData) {
    let OperationType;
    (function (OperationType) {
        OperationType["SetManagementStationConfig"] = "SET_MANAGEMENT_STATION_CONFIG";
        OperationType["SyncManagementStationMirror"] = "SYNC_MANAGEMENT_STATION_MIRROR";
        OperationType["UpdateManagementStationSoftware"] = "UPDATE_MANAGEMENT_STATION_SOFTWARE";
        OperationType["SetManagementStationHealthState"] = "SET_MANAGEMENT_STATION_HEALTH_STATE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OperationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(OperationType = ManagementStationEventData.OperationType || (ManagementStationEventData.OperationType = {}));
    let HealthState;
    (function (HealthState) {
        HealthState["Healthy"] = "HEALTHY";
        HealthState["Unhealthy"] = "UNHEALTHY";
        HealthState["Unavailable"] = "UNAVAILABLE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        HealthState["UnknownValue"] = "UNKNOWN_VALUE";
    })(HealthState = ManagementStationEventData.HealthState || (ManagementStationEventData.HealthState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "additionalDetails": obj.additionalDetails
                ? model.WorkRequestEventDataAdditionalDetails.getJsonObj(obj.additionalDetails)
                : undefined
        });
        return jsonObj;
    }
    ManagementStationEventData.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "additionalDetails": obj.additionalDetails
                ? model.WorkRequestEventDataAdditionalDetails.getDeserializedJsonObj(obj.additionalDetails)
                : undefined
        });
        return jsonObj;
    }
    ManagementStationEventData.getDeserializedJsonObj = getDeserializedJsonObj;
})(ManagementStationEventData = exports.ManagementStationEventData || (exports.ManagementStationEventData = {}));
