// oci-osmanagementhub/lib/model/managed-instance-location.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for instances in OCI, your private data center, or 3rd-party clouds.
For more information, see [Overview of OS Management Hub](https://docs.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedInstanceLocation = void 0;
/**
 * Location of the managed instance.
 **/
var ManagedInstanceLocation;
(function (ManagedInstanceLocation) {
    ManagedInstanceLocation["OnPremise"] = "ON_PREMISE";
    ManagedInstanceLocation["OciCompute"] = "OCI_COMPUTE";
    ManagedInstanceLocation["Azure"] = "AZURE";
    ManagedInstanceLocation["Ec2"] = "EC2";
    ManagedInstanceLocation["Gcp"] = "GCP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ManagedInstanceLocation["UnknownValue"] = "UNKNOWN_VALUE";
})(ManagedInstanceLocation = exports.ManagedInstanceLocation || (exports.ManagedInstanceLocation = {}));
(function (ManagedInstanceLocation) {
    function getJsonObj(obj) {
        return obj;
    }
    ManagedInstanceLocation.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ManagedInstanceLocation.getDeserializedJsonObj = getDeserializedJsonObj;
})(ManagedInstanceLocation = exports.ManagedInstanceLocation || (exports.ManagedInstanceLocation = {}));
