// oci-osmanagementhub/lib/model/os-family.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for instances in OCI, your private data center, or 3rd-party clouds.
For more information, see [Overview of OS Management Hub](https://docs.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OsFamily = void 0;
/**
 * Type of operating system.
 **/
var OsFamily;
(function (OsFamily) {
    OsFamily["OracleLinux9"] = "ORACLE_LINUX_9";
    OsFamily["OracleLinux8"] = "ORACLE_LINUX_8";
    OsFamily["OracleLinux7"] = "ORACLE_LINUX_7";
    OsFamily["OracleLinux6"] = "ORACLE_LINUX_6";
    OsFamily["WindowsServer2016"] = "WINDOWS_SERVER_2016";
    OsFamily["WindowsServer2019"] = "WINDOWS_SERVER_2019";
    OsFamily["WindowsServer2022"] = "WINDOWS_SERVER_2022";
    OsFamily["All"] = "ALL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OsFamily["UnknownValue"] = "UNKNOWN_VALUE";
})(OsFamily = exports.OsFamily || (exports.OsFamily = {}));
(function (OsFamily) {
    function getJsonObj(obj) {
        return obj;
    }
    OsFamily.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OsFamily.getDeserializedJsonObj = getDeserializedJsonObj;
})(OsFamily = exports.OsFamily || (exports.OsFamily = {}));
