// oci-osmanagementhub/lib/model/profile-version.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for instances in OCI, your private data center, or 3rd-party clouds.
For more information, see [Overview of OS Management Hub](https://docs.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileVersion = void 0;
const model = __importStar(require("../model"));
var ProfileVersion;
(function (ProfileVersion) {
    let RegistrationType;
    (function (RegistrationType) {
        RegistrationType["OciLinux"] = "OCI_LINUX";
        RegistrationType["NonOciLinux"] = "NON_OCI_LINUX";
        RegistrationType["OciWindows"] = "OCI_WINDOWS";
        RegistrationType["AutonomousLinux"] = "AUTONOMOUS_LINUX";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        RegistrationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(RegistrationType = ProfileVersion.RegistrationType || (ProfileVersion.RegistrationType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "softwareSources": obj.softwareSources
                ? obj.softwareSources.map(item => {
                    return model.SoftwareSourceDetails.getJsonObj(item);
                })
                : undefined,
            "managedInstanceGroup": obj.managedInstanceGroup
                ? model.ManagedInstanceGroupDetails.getJsonObj(obj.managedInstanceGroup)
                : undefined,
            "lifecycleEnvironment": obj.lifecycleEnvironment
                ? model.LifecycleEnvironmentDetails.getJsonObj(obj.lifecycleEnvironment)
                : undefined,
            "lifecycleStage": obj.lifecycleStage
                ? model.LifecycleStageDetails.getJsonObj(obj.lifecycleStage)
                : undefined
        });
        return jsonObj;
    }
    ProfileVersion.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "softwareSources": obj.softwareSources
                ? obj.softwareSources.map(item => {
                    return model.SoftwareSourceDetails.getDeserializedJsonObj(item);
                })
                : undefined,
            "managedInstanceGroup": obj.managedInstanceGroup
                ? model.ManagedInstanceGroupDetails.getDeserializedJsonObj(obj.managedInstanceGroup)
                : undefined,
            "lifecycleEnvironment": obj.lifecycleEnvironment
                ? model.LifecycleEnvironmentDetails.getDeserializedJsonObj(obj.lifecycleEnvironment)
                : undefined,
            "lifecycleStage": obj.lifecycleStage
                ? model.LifecycleStageDetails.getDeserializedJsonObj(obj.lifecycleStage)
                : undefined
        });
        return jsonObj;
    }
    ProfileVersion.getDeserializedJsonObj = getDeserializedJsonObj;
})(ProfileVersion = exports.ProfileVersion || (exports.ProfileVersion = {}));
