// oci-osmanagementhub/lib/model/search-software-source-modules-details.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for instances in OCI, your private data center, or 3rd-party clouds.
For more information, see [Overview of OS Management Hub](https://docs.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchSoftwareSourceModulesDetails = void 0;
var SearchSoftwareSourceModulesDetails;
(function (SearchSoftwareSourceModulesDetails) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = SearchSoftwareSourceModulesDetails.SortOrder || (SearchSoftwareSourceModulesDetails.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Name"] = "NAME";
    })(SortBy = SearchSoftwareSourceModulesDetails.SortBy || (SearchSoftwareSourceModulesDetails.SortBy = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SearchSoftwareSourceModulesDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SearchSoftwareSourceModulesDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(SearchSoftwareSourceModulesDetails = exports.SearchSoftwareSourceModulesDetails || (exports.SearchSoftwareSourceModulesDetails = {}));
