// oci-osmanagementhub/lib/model/software-source-vendor-summary.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for instances in OCI, your private data center, or 3rd-party clouds.
For more information, see [Overview of OS Management Hub](https://docs.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SoftwareSourceVendorSummary = void 0;
var SoftwareSourceVendorSummary;
(function (SoftwareSourceVendorSummary) {
    let OsFamilies;
    (function (OsFamilies) {
        OsFamilies["OracleLinux9"] = "ORACLE_LINUX_9";
        OsFamilies["OracleLinux8"] = "ORACLE_LINUX_8";
        OsFamilies["OracleLinux7"] = "ORACLE_LINUX_7";
        OsFamilies["OracleLinux6"] = "ORACLE_LINUX_6";
        OsFamilies["WindowsServer2016"] = "WINDOWS_SERVER_2016";
        OsFamilies["WindowsServer2019"] = "WINDOWS_SERVER_2019";
        OsFamilies["WindowsServer2022"] = "WINDOWS_SERVER_2022";
        OsFamilies["All"] = "ALL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OsFamilies["UnknownValue"] = "UNKNOWN_VALUE";
    })(OsFamilies = SoftwareSourceVendorSummary.OsFamilies || (SoftwareSourceVendorSummary.OsFamilies = {}));
    let ArchTypes;
    (function (ArchTypes) {
        ArchTypes["X8664"] = "X86_64";
        ArchTypes["Aarch64"] = "AARCH64";
        ArchTypes["I686"] = "I686";
        ArchTypes["Noarch"] = "NOARCH";
        ArchTypes["Src"] = "SRC";
        ArchTypes["I386"] = "I386";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ArchTypes["UnknownValue"] = "UNKNOWN_VALUE";
    })(ArchTypes = SoftwareSourceVendorSummary.ArchTypes || (SoftwareSourceVendorSummary.ArchTypes = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SoftwareSourceVendorSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SoftwareSourceVendorSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(SoftwareSourceVendorSummary = exports.SoftwareSourceVendorSummary || (exports.SoftwareSourceVendorSummary = {}));
