// oci-osmanagementhub/lib/model/software-source.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for instances in OCI, your private data center, or 3rd-party clouds.
For more information, see [Overview of OS Management Hub](https://docs.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SoftwareSource = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var SoftwareSource;
(function (SoftwareSource) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = SoftwareSource.LifecycleState || (SoftwareSource.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "softwareSourceType" in obj && obj.softwareSourceType) {
            switch (obj.softwareSourceType) {
                case "VENDOR":
                    return model.VendorSoftwareSource.getJsonObj(jsonObj, true);
                case "THIRD_PARTY":
                    return model.ThirdPartySoftwareSource.getJsonObj(jsonObj, true);
                case "CUSTOM":
                    return model.CustomSoftwareSource.getJsonObj(jsonObj, true);
                case "VERSIONED":
                    return model.VersionedCustomSoftwareSource.getJsonObj(jsonObj, true);
                case "PRIVATE":
                    return model.PrivateSoftwareSource.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.softwareSourceType}`);
            }
        }
        return jsonObj;
    }
    SoftwareSource.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "softwareSourceType" in obj && obj.softwareSourceType) {
            switch (obj.softwareSourceType) {
                case "VENDOR":
                    return model.VendorSoftwareSource.getDeserializedJsonObj(jsonObj, true);
                case "THIRD_PARTY":
                    return model.ThirdPartySoftwareSource.getDeserializedJsonObj(jsonObj, true);
                case "CUSTOM":
                    return model.CustomSoftwareSource.getDeserializedJsonObj(jsonObj, true);
                case "VERSIONED":
                    return model.VersionedCustomSoftwareSource.getDeserializedJsonObj(jsonObj, true);
                case "PRIVATE":
                    return model.PrivateSoftwareSource.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.softwareSourceType}`);
            }
        }
        return jsonObj;
    }
    SoftwareSource.getDeserializedJsonObj = getDeserializedJsonObj;
})(SoftwareSource = exports.SoftwareSource || (exports.SoftwareSource = {}));
