// oci-osmanagementhub/lib/model/windows-update-types.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for instances in OCI, your private data center, or 3rd-party clouds.
For more information, see [Overview of OS Management Hub](https://docs.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowsUpdateTypes = void 0;
/**
 * Type of Windows software update.
 **/
var WindowsUpdateTypes;
(function (WindowsUpdateTypes) {
    WindowsUpdateTypes["Security"] = "SECURITY";
    WindowsUpdateTypes["Bugfix"] = "BUGFIX";
    WindowsUpdateTypes["Enhancement"] = "ENHANCEMENT";
    WindowsUpdateTypes["Other"] = "OTHER";
    WindowsUpdateTypes["All"] = "ALL";
})(WindowsUpdateTypes = exports.WindowsUpdateTypes || (exports.WindowsUpdateTypes = {}));
(function (WindowsUpdateTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    WindowsUpdateTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WindowsUpdateTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(WindowsUpdateTypes = exports.WindowsUpdateTypes || (exports.WindowsUpdateTypes = {}));
