// oci-mysql/lib/model/certificate-generation-type.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificateGenerationType = void 0;
/**
 * Select whether to use MySQL Database Service-managed certificate (SYSTEM) or your own certificate (BYOC).
 *
 **/
var CertificateGenerationType;
(function (CertificateGenerationType) {
    CertificateGenerationType["System"] = "SYSTEM";
    CertificateGenerationType["Byoc"] = "BYOC";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CertificateGenerationType["UnknownValue"] = "UNKNOWN_VALUE";
})(CertificateGenerationType = exports.CertificateGenerationType || (exports.CertificateGenerationType = {}));
(function (CertificateGenerationType) {
    function getJsonObj(obj) {
        return obj;
    }
    CertificateGenerationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CertificateGenerationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(CertificateGenerationType = exports.CertificateGenerationType || (exports.CertificateGenerationType = {}));
