// oci-mysql/lib/model/configuration-variables.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigurationVariables = void 0;
var ConfigurationVariables;
(function (ConfigurationVariables) {
    let CompletionType;
    (function (CompletionType) {
        CompletionType["NoChain"] = "NO_CHAIN";
        CompletionType["Chain"] = "CHAIN";
        CompletionType["Release"] = "RELEASE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        CompletionType["UnknownValue"] = "UNKNOWN_VALUE";
    })(CompletionType = ConfigurationVariables.CompletionType || (ConfigurationVariables.CompletionType = {}));
    let CharacterSetServer;
    (function (CharacterSetServer) {
        CharacterSetServer["Armscii8"] = "ARMSCII8";
        CharacterSetServer["Ascii"] = "ASCII";
        CharacterSetServer["Big5"] = "BIG5";
        CharacterSetServer["Binary"] = "BINARY";
        CharacterSetServer["Cp1250"] = "CP1250";
        CharacterSetServer["Cp1251"] = "CP1251";
        CharacterSetServer["Cp1256"] = "CP1256";
        CharacterSetServer["Cp1257"] = "CP1257";
        CharacterSetServer["Cp850"] = "CP850";
        CharacterSetServer["Cp852"] = "CP852";
        CharacterSetServer["Cp866"] = "CP866";
        CharacterSetServer["Cp932"] = "CP932";
        CharacterSetServer["Dec8"] = "DEC8";
        CharacterSetServer["Eucjpms"] = "EUCJPMS";
        CharacterSetServer["Euckr"] = "EUCKR";
        CharacterSetServer["Gb18030"] = "GB18030";
        CharacterSetServer["Gb2312"] = "GB2312";
        CharacterSetServer["Gbk"] = "GBK";
        CharacterSetServer["Geostd8"] = "GEOSTD8";
        CharacterSetServer["Greek"] = "GREEK";
        CharacterSetServer["Hebrew"] = "HEBREW";
        CharacterSetServer["Hp8"] = "HP8";
        CharacterSetServer["Keybcs2"] = "KEYBCS2";
        CharacterSetServer["Koi8R"] = "KOI8R";
        CharacterSetServer["Koi8U"] = "KOI8U";
        CharacterSetServer["Latin1"] = "LATIN1";
        CharacterSetServer["Latin2"] = "LATIN2";
        CharacterSetServer["Latin5"] = "LATIN5";
        CharacterSetServer["Latin7"] = "LATIN7";
        CharacterSetServer["Macce"] = "MACCE";
        CharacterSetServer["Macroman"] = "MACROMAN";
        CharacterSetServer["Sjis"] = "SJIS";
        CharacterSetServer["Swe7"] = "SWE7";
        CharacterSetServer["Tis620"] = "TIS620";
        CharacterSetServer["Ucs2"] = "UCS2";
        CharacterSetServer["Ujis"] = "UJIS";
        CharacterSetServer["Utf16"] = "UTF16";
        CharacterSetServer["Utf16Le"] = "UTF16LE";
        CharacterSetServer["Utf32"] = "UTF32";
        CharacterSetServer["Utf8Mb3"] = "UTF8MB3";
        CharacterSetServer["Utf8Mb4"] = "UTF8MB4";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        CharacterSetServer["UnknownValue"] = "UNKNOWN_VALUE";
    })(CharacterSetServer = ConfigurationVariables.CharacterSetServer || (ConfigurationVariables.CharacterSetServer = {}));
    let CollationServer;
    (function (CollationServer) {
        CollationServer["Armscii8GeneralCi"] = "ARMSCII8_GENERAL_CI";
        CollationServer["Armscii8Bin"] = "ARMSCII8_BIN";
        CollationServer["AsciiGeneralCi"] = "ASCII_GENERAL_CI";
        CollationServer["AsciiBin"] = "ASCII_BIN";
        CollationServer["Big5ChineseCi"] = "BIG5_CHINESE_CI";
        CollationServer["Big5Bin"] = "BIG5_BIN";
        CollationServer["Binary"] = "BINARY";
        CollationServer["Cp1250GeneralCi"] = "CP1250_GENERAL_CI";
        CollationServer["Cp1250Bin"] = "CP1250_BIN";
        CollationServer["Cp1250CroatianCi"] = "CP1250_CROATIAN_CI";
        CollationServer["Cp1250CzechCs"] = "CP1250_CZECH_CS";
        CollationServer["Cp1250PolishCi"] = "CP1250_POLISH_CI";
        CollationServer["Cp1251GeneralCi"] = "CP1251_GENERAL_CI";
        CollationServer["Cp1251Bin"] = "CP1251_BIN";
        CollationServer["Cp1251BulgarianCi"] = "CP1251_BULGARIAN_CI";
        CollationServer["Cp1251GeneralCs"] = "CP1251_GENERAL_CS";
        CollationServer["Cp1251UkrainianCi"] = "CP1251_UKRAINIAN_CI";
        CollationServer["Cp1256GeneralCi"] = "CP1256_GENERAL_CI";
        CollationServer["Cp1256Bin"] = "CP1256_BIN";
        CollationServer["Cp1257GeneralCi"] = "CP1257_GENERAL_CI";
        CollationServer["Cp1257Bin"] = "CP1257_BIN";
        CollationServer["Cp1257LithuanianCi"] = "CP1257_LITHUANIAN_CI";
        CollationServer["Cp850GeneralCi"] = "CP850_GENERAL_CI";
        CollationServer["Cp850Bin"] = "CP850_BIN";
        CollationServer["Cp852GeneralCi"] = "CP852_GENERAL_CI";
        CollationServer["Cp852Bin"] = "CP852_BIN";
        CollationServer["Cp866GeneralCi"] = "CP866_GENERAL_CI";
        CollationServer["Cp866Bin"] = "CP866_BIN";
        CollationServer["Cp932JapaneseCi"] = "CP932_JAPANESE_CI";
        CollationServer["Cp932Bin"] = "CP932_BIN";
        CollationServer["Dec8SwedishCi"] = "DEC8_SWEDISH_CI";
        CollationServer["Dec8Bin"] = "DEC8_BIN";
        CollationServer["EucjpmsJapaneseCi"] = "EUCJPMS_JAPANESE_CI";
        CollationServer["EucjpmsBin"] = "EUCJPMS_BIN";
        CollationServer["EuckrKoreanCi"] = "EUCKR_KOREAN_CI";
        CollationServer["EuckrBin"] = "EUCKR_BIN";
        CollationServer["Gb18030ChineseCi"] = "GB18030_CHINESE_CI";
        CollationServer["Gb18030Bin"] = "GB18030_BIN";
        CollationServer["Gb18030Unicode520Ci"] = "GB18030_UNICODE_520_CI";
        CollationServer["Gb2312ChineseCi"] = "GB2312_CHINESE_CI";
        CollationServer["Gb2312Bin"] = "GB2312_BIN";
        CollationServer["GbkChineseCi"] = "GBK_CHINESE_CI";
        CollationServer["GbkBin"] = "GBK_BIN";
        CollationServer["Geostd8GeneralCi"] = "GEOSTD8_GENERAL_CI";
        CollationServer["Geostd8Bin"] = "GEOSTD8_BIN";
        CollationServer["GreekGeneralCi"] = "GREEK_GENERAL_CI";
        CollationServer["GreekBin"] = "GREEK_BIN";
        CollationServer["HebrewGeneralCi"] = "HEBREW_GENERAL_CI";
        CollationServer["HebrewBin"] = "HEBREW_BIN";
        CollationServer["Hp8EnglishCi"] = "HP8_ENGLISH_CI";
        CollationServer["Hp8Bin"] = "HP8_BIN";
        CollationServer["Keybcs2GeneralCi"] = "KEYBCS2_GENERAL_CI";
        CollationServer["Keybcs2Bin"] = "KEYBCS2_BIN";
        CollationServer["Koi8RGeneralCi"] = "KOI8R_GENERAL_CI";
        CollationServer["Koi8RBin"] = "KOI8R_BIN";
        CollationServer["Koi8UGeneralCi"] = "KOI8U_GENERAL_CI";
        CollationServer["Koi8UBin"] = "KOI8U_BIN";
        CollationServer["Latin1SwedishCi"] = "LATIN1_SWEDISH_CI";
        CollationServer["Latin1Bin"] = "LATIN1_BIN";
        CollationServer["Latin1DanishCi"] = "LATIN1_DANISH_CI";
        CollationServer["Latin1GeneralCi"] = "LATIN1_GENERAL_CI";
        CollationServer["Latin1GeneralCs"] = "LATIN1_GENERAL_CS";
        CollationServer["Latin1German1Ci"] = "LATIN1_GERMAN1_CI";
        CollationServer["Latin1German2Ci"] = "LATIN1_GERMAN2_CI";
        CollationServer["Latin1SpanishCi"] = "LATIN1_SPANISH_CI";
        CollationServer["Latin2GeneralCi"] = "LATIN2_GENERAL_CI";
        CollationServer["Latin2Bin"] = "LATIN2_BIN";
        CollationServer["Latin2CroatianCi"] = "LATIN2_CROATIAN_CI";
        CollationServer["Latin2CzechCs"] = "LATIN2_CZECH_CS";
        CollationServer["Latin2HungarianCi"] = "LATIN2_HUNGARIAN_CI";
        CollationServer["Latin5TurkishCi"] = "LATIN5_TURKISH_CI";
        CollationServer["Latin5Bin"] = "LATIN5_BIN";
        CollationServer["Latin7GeneralCi"] = "LATIN7_GENERAL_CI";
        CollationServer["Latin7Bin"] = "LATIN7_BIN";
        CollationServer["Latin7EstonianCs"] = "LATIN7_ESTONIAN_CS";
        CollationServer["Latin7GeneralCs"] = "LATIN7_GENERAL_CS";
        CollationServer["MacceGeneralCi"] = "MACCE_GENERAL_CI";
        CollationServer["MacceBin"] = "MACCE_BIN";
        CollationServer["MacromanGeneralCi"] = "MACROMAN_GENERAL_CI";
        CollationServer["MacromanBin"] = "MACROMAN_BIN";
        CollationServer["SjisJapaneseCi"] = "SJIS_JAPANESE_CI";
        CollationServer["SjisBin"] = "SJIS_BIN";
        CollationServer["Swe7SwedishCi"] = "SWE7_SWEDISH_CI";
        CollationServer["Swe7Bin"] = "SWE7_BIN";
        CollationServer["Tis620ThaiCi"] = "TIS620_THAI_CI";
        CollationServer["Tis620Bin"] = "TIS620_BIN";
        CollationServer["Ucs2GeneralCi"] = "UCS2_GENERAL_CI";
        CollationServer["Ucs2Bin"] = "UCS2_BIN";
        CollationServer["Ucs2CroatianCi"] = "UCS2_CROATIAN_CI";
        CollationServer["Ucs2CzechCi"] = "UCS2_CZECH_CI";
        CollationServer["Ucs2DanishCi"] = "UCS2_DANISH_CI";
        CollationServer["Ucs2EsperantoCi"] = "UCS2_ESPERANTO_CI";
        CollationServer["Ucs2EstonianCi"] = "UCS2_ESTONIAN_CI";
        CollationServer["Ucs2GeneralMysql500Ci"] = "UCS2_GENERAL_MYSQL500_CI";
        CollationServer["Ucs2German2Ci"] = "UCS2_GERMAN2_CI";
        CollationServer["Ucs2HungarianCi"] = "UCS2_HUNGARIAN_CI";
        CollationServer["Ucs2IcelandicCi"] = "UCS2_ICELANDIC_CI";
        CollationServer["Ucs2LatvianCi"] = "UCS2_LATVIAN_CI";
        CollationServer["Ucs2LithuanianCi"] = "UCS2_LITHUANIAN_CI";
        CollationServer["Ucs2PersianCi"] = "UCS2_PERSIAN_CI";
        CollationServer["Ucs2PolishCi"] = "UCS2_POLISH_CI";
        CollationServer["Ucs2RomanianCi"] = "UCS2_ROMANIAN_CI";
        CollationServer["Ucs2RomanCi"] = "UCS2_ROMAN_CI";
        CollationServer["Ucs2SinhalaCi"] = "UCS2_SINHALA_CI";
        CollationServer["Ucs2SlovakCi"] = "UCS2_SLOVAK_CI";
        CollationServer["Ucs2SlovenianCi"] = "UCS2_SLOVENIAN_CI";
        CollationServer["Ucs2Spanish2Ci"] = "UCS2_SPANISH2_CI";
        CollationServer["Ucs2SpanishCi"] = "UCS2_SPANISH_CI";
        CollationServer["Ucs2SwedishCi"] = "UCS2_SWEDISH_CI";
        CollationServer["Ucs2TurkishCi"] = "UCS2_TURKISH_CI";
        CollationServer["Ucs2Unicode520Ci"] = "UCS2_UNICODE_520_CI";
        CollationServer["Ucs2UnicodeCi"] = "UCS2_UNICODE_CI";
        CollationServer["Ucs2VietnameseCi"] = "UCS2_VIETNAMESE_CI";
        CollationServer["UjisJapaneseCi"] = "UJIS_JAPANESE_CI";
        CollationServer["UjisBin"] = "UJIS_BIN";
        CollationServer["Utf16GeneralCi"] = "UTF16_GENERAL_CI";
        CollationServer["Utf16Bin"] = "UTF16_BIN";
        CollationServer["Utf16CroatianCi"] = "UTF16_CROATIAN_CI";
        CollationServer["Utf16CzechCi"] = "UTF16_CZECH_CI";
        CollationServer["Utf16DanishCi"] = "UTF16_DANISH_CI";
        CollationServer["Utf16EsperantoCi"] = "UTF16_ESPERANTO_CI";
        CollationServer["Utf16EstonianCi"] = "UTF16_ESTONIAN_CI";
        CollationServer["Utf16German2Ci"] = "UTF16_GERMAN2_CI";
        CollationServer["Utf16HungarianCi"] = "UTF16_HUNGARIAN_CI";
        CollationServer["Utf16IcelandicCi"] = "UTF16_ICELANDIC_CI";
        CollationServer["Utf16LatvianCi"] = "UTF16_LATVIAN_CI";
        CollationServer["Utf16LithuanianCi"] = "UTF16_LITHUANIAN_CI";
        CollationServer["Utf16PersianCi"] = "UTF16_PERSIAN_CI";
        CollationServer["Utf16PolishCi"] = "UTF16_POLISH_CI";
        CollationServer["Utf16RomanianCi"] = "UTF16_ROMANIAN_CI";
        CollationServer["Utf16RomanCi"] = "UTF16_ROMAN_CI";
        CollationServer["Utf16SinhalaCi"] = "UTF16_SINHALA_CI";
        CollationServer["Utf16SlovakCi"] = "UTF16_SLOVAK_CI";
        CollationServer["Utf16SlovenianCi"] = "UTF16_SLOVENIAN_CI";
        CollationServer["Utf16Spanish2Ci"] = "UTF16_SPANISH2_CI";
        CollationServer["Utf16SpanishCi"] = "UTF16_SPANISH_CI";
        CollationServer["Utf16SwedishCi"] = "UTF16_SWEDISH_CI";
        CollationServer["Utf16TurkishCi"] = "UTF16_TURKISH_CI";
        CollationServer["Utf16Unicode520Ci"] = "UTF16_UNICODE_520_CI";
        CollationServer["Utf16UnicodeCi"] = "UTF16_UNICODE_CI";
        CollationServer["Utf16VietnameseCi"] = "UTF16_VIETNAMESE_CI";
        CollationServer["Utf16LeGeneralCi"] = "UTF16LE_GENERAL_CI";
        CollationServer["Utf16LeBin"] = "UTF16LE_BIN";
        CollationServer["Utf32GeneralCi"] = "UTF32_GENERAL_CI";
        CollationServer["Utf32Bin"] = "UTF32_BIN";
        CollationServer["Utf32CroatianCi"] = "UTF32_CROATIAN_CI";
        CollationServer["Utf32CzechCi"] = "UTF32_CZECH_CI";
        CollationServer["Utf32DanishCi"] = "UTF32_DANISH_CI";
        CollationServer["Utf32EsperantoCi"] = "UTF32_ESPERANTO_CI";
        CollationServer["Utf32EstonianCi"] = "UTF32_ESTONIAN_CI";
        CollationServer["Utf32German2Ci"] = "UTF32_GERMAN2_CI";
        CollationServer["Utf32HungarianCi"] = "UTF32_HUNGARIAN_CI";
        CollationServer["Utf32IcelandicCi"] = "UTF32_ICELANDIC_CI";
        CollationServer["Utf32LatvianCi"] = "UTF32_LATVIAN_CI";
        CollationServer["Utf32LithuanianCi"] = "UTF32_LITHUANIAN_CI";
        CollationServer["Utf32PersianCi"] = "UTF32_PERSIAN_CI";
        CollationServer["Utf32PolishCi"] = "UTF32_POLISH_CI";
        CollationServer["Utf32RomanianCi"] = "UTF32_ROMANIAN_CI";
        CollationServer["Utf32RomanCi"] = "UTF32_ROMAN_CI";
        CollationServer["Utf32SinhalaCi"] = "UTF32_SINHALA_CI";
        CollationServer["Utf32SlovakCi"] = "UTF32_SLOVAK_CI";
        CollationServer["Utf32SlovenianCi"] = "UTF32_SLOVENIAN_CI";
        CollationServer["Utf32Spanish2Ci"] = "UTF32_SPANISH2_CI";
        CollationServer["Utf32SpanishCi"] = "UTF32_SPANISH_CI";
        CollationServer["Utf32SwedishCi"] = "UTF32_SWEDISH_CI";
        CollationServer["Utf32TurkishCi"] = "UTF32_TURKISH_CI";
        CollationServer["Utf32Unicode520Ci"] = "UTF32_UNICODE_520_CI";
        CollationServer["Utf32UnicodeCi"] = "UTF32_UNICODE_CI";
        CollationServer["Utf32VietnameseCi"] = "UTF32_VIETNAMESE_CI";
        CollationServer["Utf8GeneralCi"] = "UTF8_GENERAL_CI";
        CollationServer["Utf8Bin"] = "UTF8_BIN";
        CollationServer["Utf8CroatianCi"] = "UTF8_CROATIAN_CI";
        CollationServer["Utf8CzechCi"] = "UTF8_CZECH_CI";
        CollationServer["Utf8DanishCi"] = "UTF8_DANISH_CI";
        CollationServer["Utf8EsperantoCi"] = "UTF8_ESPERANTO_CI";
        CollationServer["Utf8EstonianCi"] = "UTF8_ESTONIAN_CI";
        CollationServer["Utf8GeneralMysql500Ci"] = "UTF8_GENERAL_MYSQL500_CI";
        CollationServer["Utf8German2Ci"] = "UTF8_GERMAN2_CI";
        CollationServer["Utf8HungarianCi"] = "UTF8_HUNGARIAN_CI";
        CollationServer["Utf8IcelandicCi"] = "UTF8_ICELANDIC_CI";
        CollationServer["Utf8LatvianCi"] = "UTF8_LATVIAN_CI";
        CollationServer["Utf8LithuanianCi"] = "UTF8_LITHUANIAN_CI";
        CollationServer["Utf8PersianCi"] = "UTF8_PERSIAN_CI";
        CollationServer["Utf8PolishCi"] = "UTF8_POLISH_CI";
        CollationServer["Utf8RomanianCi"] = "UTF8_ROMANIAN_CI";
        CollationServer["Utf8RomanCi"] = "UTF8_ROMAN_CI";
        CollationServer["Utf8SinhalaCi"] = "UTF8_SINHALA_CI";
        CollationServer["Utf8SlovakCi"] = "UTF8_SLOVAK_CI";
        CollationServer["Utf8SlovenianCi"] = "UTF8_SLOVENIAN_CI";
        CollationServer["Utf8Spanish2Ci"] = "UTF8_SPANISH2_CI";
        CollationServer["Utf8SpanishCi"] = "UTF8_SPANISH_CI";
        CollationServer["Utf8SwedishCi"] = "UTF8_SWEDISH_CI";
        CollationServer["Utf8TolowerCi"] = "UTF8_TOLOWER_CI";
        CollationServer["Utf8TurkishCi"] = "UTF8_TURKISH_CI";
        CollationServer["Utf8Unicode520Ci"] = "UTF8_UNICODE_520_CI";
        CollationServer["Utf8UnicodeCi"] = "UTF8_UNICODE_CI";
        CollationServer["Utf8VietnameseCi"] = "UTF8_VIETNAMESE_CI";
        CollationServer["Utf8Mb40900AiCi"] = "UTF8MB4_0900_AI_CI";
        CollationServer["Utf8Mb40900AsCi"] = "UTF8MB4_0900_AS_CI";
        CollationServer["Utf8Mb40900AsCs"] = "UTF8MB4_0900_AS_CS";
        CollationServer["Utf8Mb40900Bin"] = "UTF8MB4_0900_BIN";
        CollationServer["Utf8Mb4Bin"] = "UTF8MB4_BIN";
        CollationServer["Utf8Mb4CroatianCi"] = "UTF8MB4_CROATIAN_CI";
        CollationServer["Utf8Mb4Cs0900AiCi"] = "UTF8MB4_CS_0900_AI_CI";
        CollationServer["Utf8Mb4Cs0900AsCs"] = "UTF8MB4_CS_0900_AS_CS";
        CollationServer["Utf8Mb4CzechCi"] = "UTF8MB4_CZECH_CI";
        CollationServer["Utf8Mb4DanishCi"] = "UTF8MB4_DANISH_CI";
        CollationServer["Utf8Mb4Da0900AiCi"] = "UTF8MB4_DA_0900_AI_CI";
        CollationServer["Utf8Mb4Da0900AsCs"] = "UTF8MB4_DA_0900_AS_CS";
        CollationServer["Utf8Mb4DePb0900AiCi"] = "UTF8MB4_DE_PB_0900_AI_CI";
        CollationServer["Utf8Mb4DePb0900AsCs"] = "UTF8MB4_DE_PB_0900_AS_CS";
        CollationServer["Utf8Mb4Eo0900AiCi"] = "UTF8MB4_EO_0900_AI_CI";
        CollationServer["Utf8Mb4Eo0900AsCs"] = "UTF8MB4_EO_0900_AS_CS";
        CollationServer["Utf8Mb4EsperantoCi"] = "UTF8MB4_ESPERANTO_CI";
        CollationServer["Utf8Mb4EstonianCi"] = "UTF8MB4_ESTONIAN_CI";
        CollationServer["Utf8Mb4Es0900AiCi"] = "UTF8MB4_ES_0900_AI_CI";
        CollationServer["Utf8Mb4Es0900AsCs"] = "UTF8MB4_ES_0900_AS_CS";
        CollationServer["Utf8Mb4EsTrad0900AiCi"] = "UTF8MB4_ES_TRAD_0900_AI_CI";
        CollationServer["Utf8Mb4EsTrad0900AsCs"] = "UTF8MB4_ES_TRAD_0900_AS_CS";
        CollationServer["Utf8Mb4Et0900AiCi"] = "UTF8MB4_ET_0900_AI_CI";
        CollationServer["Utf8Mb4Et0900AsCs"] = "UTF8MB4_ET_0900_AS_CS";
        CollationServer["Utf8Mb4GeneralCi"] = "UTF8MB4_GENERAL_CI";
        CollationServer["Utf8Mb4German2Ci"] = "UTF8MB4_GERMAN2_CI";
        CollationServer["Utf8Mb4Hr0900AiCi"] = "UTF8MB4_HR_0900_AI_CI";
        CollationServer["Utf8Mb4Hr0900AsCs"] = "UTF8MB4_HR_0900_AS_CS";
        CollationServer["Utf8Mb4HungarianCi"] = "UTF8MB4_HUNGARIAN_CI";
        CollationServer["Utf8Mb4Hu0900AiCi"] = "UTF8MB4_HU_0900_AI_CI";
        CollationServer["Utf8Mb4Hu0900AsCs"] = "UTF8MB4_HU_0900_AS_CS";
        CollationServer["Utf8Mb4IcelandicCi"] = "UTF8MB4_ICELANDIC_CI";
        CollationServer["Utf8Mb4Is0900AiCi"] = "UTF8MB4_IS_0900_AI_CI";
        CollationServer["Utf8Mb4Is0900AsCs"] = "UTF8MB4_IS_0900_AS_CS";
        CollationServer["Utf8Mb4Ja0900AsCs"] = "UTF8MB4_JA_0900_AS_CS";
        CollationServer["Utf8Mb4Ja0900AsCsKs"] = "UTF8MB4_JA_0900_AS_CS_KS";
        CollationServer["Utf8Mb4LatvianCi"] = "UTF8MB4_LATVIAN_CI";
        CollationServer["Utf8Mb4La0900AiCi"] = "UTF8MB4_LA_0900_AI_CI";
        CollationServer["Utf8Mb4La0900AsCs"] = "UTF8MB4_LA_0900_AS_CS";
        CollationServer["Utf8Mb4LithuanianCi"] = "UTF8MB4_LITHUANIAN_CI";
        CollationServer["Utf8Mb4Lt0900AiCi"] = "UTF8MB4_LT_0900_AI_CI";
        CollationServer["Utf8Mb4Lt0900AsCs"] = "UTF8MB4_LT_0900_AS_CS";
        CollationServer["Utf8Mb4Lv0900AiCi"] = "UTF8MB4_LV_0900_AI_CI";
        CollationServer["Utf8Mb4Lv0900AsCs"] = "UTF8MB4_LV_0900_AS_CS";
        CollationServer["Utf8Mb4PersianCi"] = "UTF8MB4_PERSIAN_CI";
        CollationServer["Utf8Mb4Pl0900AiCi"] = "UTF8MB4_PL_0900_AI_CI";
        CollationServer["Utf8Mb4Pl0900AsCs"] = "UTF8MB4_PL_0900_AS_CS";
        CollationServer["Utf8Mb4PolishCi"] = "UTF8MB4_POLISH_CI";
        CollationServer["Utf8Mb4RomanianCi"] = "UTF8MB4_ROMANIAN_CI";
        CollationServer["Utf8Mb4RomanCi"] = "UTF8MB4_ROMAN_CI";
        CollationServer["Utf8Mb4Ro0900AiCi"] = "UTF8MB4_RO_0900_AI_CI";
        CollationServer["Utf8Mb4Ro0900AsCs"] = "UTF8MB4_RO_0900_AS_CS";
        CollationServer["Utf8Mb4Ru0900AiCi"] = "UTF8MB4_RU_0900_AI_CI";
        CollationServer["Utf8Mb4Ru0900AsCs"] = "UTF8MB4_RU_0900_AS_CS";
        CollationServer["Utf8Mb4SinhalaCi"] = "UTF8MB4_SINHALA_CI";
        CollationServer["Utf8Mb4Sk0900AiCi"] = "UTF8MB4_SK_0900_AI_CI";
        CollationServer["Utf8Mb4Sk0900AsCs"] = "UTF8MB4_SK_0900_AS_CS";
        CollationServer["Utf8Mb4SlovakCi"] = "UTF8MB4_SLOVAK_CI";
        CollationServer["Utf8Mb4SlovenianCi"] = "UTF8MB4_SLOVENIAN_CI";
        CollationServer["Utf8Mb4Sl0900AiCi"] = "UTF8MB4_SL_0900_AI_CI";
        CollationServer["Utf8Mb4Sl0900AsCs"] = "UTF8MB4_SL_0900_AS_CS";
        CollationServer["Utf8Mb4Spanish2Ci"] = "UTF8MB4_SPANISH2_CI";
        CollationServer["Utf8Mb4SpanishCi"] = "UTF8MB4_SPANISH_CI";
        CollationServer["Utf8Mb4Sv0900AiCi"] = "UTF8MB4_SV_0900_AI_CI";
        CollationServer["Utf8Mb4Sv0900AsCs"] = "UTF8MB4_SV_0900_AS_CS";
        CollationServer["Utf8Mb4SwedishCi"] = "UTF8MB4_SWEDISH_CI";
        CollationServer["Utf8Mb4Tr0900AiCi"] = "UTF8MB4_TR_0900_AI_CI";
        CollationServer["Utf8Mb4Tr0900AsCs"] = "UTF8MB4_TR_0900_AS_CS";
        CollationServer["Utf8Mb4TurkishCi"] = "UTF8MB4_TURKISH_CI";
        CollationServer["Utf8Mb4Unicode520Ci"] = "UTF8MB4_UNICODE_520_CI";
        CollationServer["Utf8Mb4UnicodeCi"] = "UTF8MB4_UNICODE_CI";
        CollationServer["Utf8Mb4VietnameseCi"] = "UTF8MB4_VIETNAMESE_CI";
        CollationServer["Utf8Mb4Vi0900AiCi"] = "UTF8MB4_VI_0900_AI_CI";
        CollationServer["Utf8Mb4Vi0900AsCs"] = "UTF8MB4_VI_0900_AS_CS";
        CollationServer["Utf8Mb4Zh0900AsCs"] = "UTF8MB4_ZH_0900_AS_CS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        CollationServer["UnknownValue"] = "UNKNOWN_VALUE";
    })(CollationServer = ConfigurationVariables.CollationServer || (ConfigurationVariables.CollationServer = {}));
    let DefaultAuthenticationPlugin;
    (function (DefaultAuthenticationPlugin) {
        DefaultAuthenticationPlugin["MysqlNativePassword"] = "mysql_native_password";
        DefaultAuthenticationPlugin["Sha256Password"] = "sha256_password";
        DefaultAuthenticationPlugin["CachingSha2Password"] = "caching_sha2_password";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DefaultAuthenticationPlugin["UnknownValue"] = "UNKNOWN_VALUE";
    })(DefaultAuthenticationPlugin = ConfigurationVariables.DefaultAuthenticationPlugin || (ConfigurationVariables.DefaultAuthenticationPlugin = {}));
    let TransactionIsolation;
    (function (TransactionIsolation) {
        TransactionIsolation["ReadUncommitted"] = "READ-UNCOMMITTED";
        TransactionIsolation["ReadCommited"] = "READ-COMMITED";
        TransactionIsolation["ReadCommitted"] = "READ-COMMITTED";
        TransactionIsolation["RepeatableRead"] = "REPEATABLE-READ";
        TransactionIsolation["Serializable"] = "SERIALIZABLE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TransactionIsolation["UnknownValue"] = "UNKNOWN_VALUE";
    })(TransactionIsolation = ConfigurationVariables.TransactionIsolation || (ConfigurationVariables.TransactionIsolation = {}));
    let GroupReplicationConsistency;
    (function (GroupReplicationConsistency) {
        GroupReplicationConsistency["Eventual"] = "EVENTUAL";
        GroupReplicationConsistency["BeforeOnPrimaryFailover"] = "BEFORE_ON_PRIMARY_FAILOVER";
        GroupReplicationConsistency["Before"] = "BEFORE";
        GroupReplicationConsistency["After"] = "AFTER";
        GroupReplicationConsistency["BeforeAndAfter"] = "BEFORE_AND_AFTER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        GroupReplicationConsistency["UnknownValue"] = "UNKNOWN_VALUE";
    })(GroupReplicationConsistency = ConfigurationVariables.GroupReplicationConsistency || (ConfigurationVariables.GroupReplicationConsistency = {}));
    let BinlogRowMetadata;
    (function (BinlogRowMetadata) {
        BinlogRowMetadata["Full"] = "FULL";
        BinlogRowMetadata["Minimal"] = "MINIMAL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        BinlogRowMetadata["UnknownValue"] = "UNKNOWN_VALUE";
    })(BinlogRowMetadata = ConfigurationVariables.BinlogRowMetadata || (ConfigurationVariables.BinlogRowMetadata = {}));
    let ExplainFormat;
    (function (ExplainFormat) {
        ExplainFormat["Traditional"] = "TRADITIONAL";
        ExplainFormat["Json"] = "JSON";
        ExplainFormat["Tree"] = "TREE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ExplainFormat["UnknownValue"] = "UNKNOWN_VALUE";
    })(ExplainFormat = ConfigurationVariables.ExplainFormat || (ConfigurationVariables.ExplainFormat = {}));
    let InnodbChangeBuffering;
    (function (InnodbChangeBuffering) {
        InnodbChangeBuffering["None"] = "NONE";
        InnodbChangeBuffering["Inserts"] = "INSERTS";
        InnodbChangeBuffering["Deletes"] = "DELETES";
        InnodbChangeBuffering["Changes"] = "CHANGES";
        InnodbChangeBuffering["Purges"] = "PURGES";
        InnodbChangeBuffering["All"] = "ALL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        InnodbChangeBuffering["UnknownValue"] = "UNKNOWN_VALUE";
    })(InnodbChangeBuffering = ConfigurationVariables.InnodbChangeBuffering || (ConfigurationVariables.InnodbChangeBuffering = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ConfigurationVariables.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ConfigurationVariables.getDeserializedJsonObj = getDeserializedJsonObj;
})(ConfigurationVariables = exports.ConfigurationVariables || (exports.ConfigurationVariables = {}));
