// oci-mysql/lib/model/database-management-status.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseManagementStatus = void 0;
/**
 * Whether to enable monitoring via the Database Management service.
 *
 **/
var DatabaseManagementStatus;
(function (DatabaseManagementStatus) {
    DatabaseManagementStatus["Enabled"] = "ENABLED";
    DatabaseManagementStatus["Disabled"] = "DISABLED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DatabaseManagementStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(DatabaseManagementStatus = exports.DatabaseManagementStatus || (exports.DatabaseManagementStatus = {}));
(function (DatabaseManagementStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    DatabaseManagementStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DatabaseManagementStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabaseManagementStatus = exports.DatabaseManagementStatus || (exports.DatabaseManagementStatus = {}));
