// oci-mysql/lib/model/rest-configuration-type.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RestConfigurationType = void 0;
/**
 * Select how REST is configured across the DB System instances.
 *
 **/
var RestConfigurationType;
(function (RestConfigurationType) {
    RestConfigurationType["Disabled"] = "DISABLED";
    RestConfigurationType["DbsystemOnly"] = "DBSYSTEM_ONLY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RestConfigurationType["UnknownValue"] = "UNKNOWN_VALUE";
})(RestConfigurationType = exports.RestConfigurationType || (exports.RestConfigurationType = {}));
(function (RestConfigurationType) {
    function getJsonObj(obj) {
        return obj;
    }
    RestConfigurationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RestConfigurationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(RestConfigurationType = exports.RestConfigurationType || (exports.RestConfigurationType = {}));
