// oci-usage/lib/model/usage-limit-summary.js
"use strict";
/**
 * Usage Proxy API
 * Use the Usage Proxy API to list Oracle Support Rewards, view related detailed usage information, and manage users who redeem rewards. For more information, see [Oracle Support Rewards Overview](https://docs.oracle.com/iaas/Content/Billing/Concepts/supportrewardsoverview.htm).
 * OpenAPI spec version: 20190111
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageLimitSummary = void 0;
var UsageLimitSummary;
(function (UsageLimitSummary) {
    let Action;
    (function (Action) {
        Action["QuotaBreach"] = "QUOTA_BREACH";
        Action["QuotaAlert"] = "QUOTA_ALERT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Action["UnknownValue"] = "UNKNOWN_VALUE";
    })(Action = UsageLimitSummary.Action || (UsageLimitSummary.Action = {}));
    let LimitType;
    (function (LimitType) {
        LimitType["Hard"] = "HARD";
        LimitType["Soft"] = "SOFT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LimitType["UnknownValue"] = "UNKNOWN_VALUE";
    })(LimitType = UsageLimitSummary.LimitType || (UsageLimitSummary.LimitType = {}));
    let ValueType;
    (function (ValueType) {
        ValueType["Absolute"] = "ABSOLUTE";
        ValueType["Percentage"] = "PERCENTAGE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ValueType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ValueType = UsageLimitSummary.ValueType || (UsageLimitSummary.ValueType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = UsageLimitSummary.LifecycleState || (UsageLimitSummary.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UsageLimitSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UsageLimitSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(UsageLimitSummary = exports.UsageLimitSummary || (exports.UsageLimitSummary = {}));
