// oci-cims/lib/model/error-code.js
"use strict";
/**
 * Support Management API
 * Use the Support Management API to manage support requests.
For more information, see [Getting Help and Contacting Support](https://docs.oracle.com/iaas/Content/GSG/Tasks/contactingsupport.htm).
**Note**: Before you can create service requests with this API,
complete user registration at My Oracle Cloud Support
and then ask your tenancy administrator to provide you authorization for the related user groups.

 * OpenAPI spec version: 20181231
 * Contact: oci_ops_cims_dev_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorCode = void 0;
/**
 * Service error codes that can be returned from the API calls.
 **/
var ErrorCode;
(function (ErrorCode) {
    ErrorCode["ContentEmpty"] = "CONTENT_EMPTY";
    ErrorCode["ClientException"] = "CLIENT_EXCEPTION";
    ErrorCode["InvalidFormat"] = "INVALID_FORMAT";
    ErrorCode["InvalidJsonInput"] = "INVALID_JSON_INPUT";
    ErrorCode["SslAuthorization"] = "SSL_AUTHORIZATION";
    ErrorCode["AuthFailed"] = "AUTH_FAILED";
    ErrorCode["AuthzFailed"] = "AUTHZ_FAILED";
    ErrorCode["CsiNotAuthorized"] = "CSI_NOT_AUTHORIZED";
    ErrorCode["UserPolicyNotAuthorized"] = "USER_POLICY_NOT_AUTHORIZED";
    ErrorCode["EmailNotVerified"] = "EMAIL_NOT_VERIFIED";
    ErrorCode["EmailNotFound"] = "EMAIL_NOT_FOUND";
    ErrorCode["OciEmailNotFound"] = "OCI_EMAIL_NOT_FOUND";
    ErrorCode["MosEmailNotFound"] = "MOS_EMAIL_NOT_FOUND";
    ErrorCode["IdcsEmailNotValid"] = "IDCS_EMAIL_NOT_VALID";
    ErrorCode["InvalidPath"] = "INVALID_PATH";
    ErrorCode["MethodNotAllowed"] = "METHOD_NOT_ALLOWED";
    ErrorCode["JsonProcessing"] = "JSON_PROCESSING";
    ErrorCode["GenericException"] = "GENERIC_EXCEPTION";
    ErrorCode["ExternalServiceProviderUnavailable"] = "EXTERNAL_SERVICE_PROVIDER_UNAVAILABLE";
    ErrorCode["ExternalServiceProviderTimeout"] = "EXTERNAL_SERVICE_PROVIDER_TIMEOUT";
    ErrorCode["TooManyRequests"] = "TOO_MANY_REQUESTS";
    ErrorCode["IdpScimNotSetup"] = "IDP_SCIM_NOT_SETUP";
    ErrorCode["IncidentNotFound"] = "INCIDENT_NOT_FOUND";
    ErrorCode["InvalidUserCsi"] = "INVALID_USER_CSI";
    ErrorCode["DataAlreadyExists"] = "DATA_ALREADY_EXISTS";
    ErrorCode["AuthUserNotMatching"] = "AUTH_USER_NOT_MATCHING";
    ErrorCode["ContactNotApproved"] = "CONTACT_NOT_APPROVED";
    ErrorCode["CreateProfileMosFailure"] = "CREATE_PROFILE_MOS_FAILURE";
    ErrorCode["CreateProfileCreateOssoFailure"] = "CREATE_PROFILE_CREATE_OSSO_FAILURE";
    ErrorCode["CreateProfileIdentityFailure"] = "CREATE_PROFILE_IDENTITY_FAILURE";
    ErrorCode["CreateProfileVerifyOssoFailure"] = "CREATE_PROFILE_VERIFY_OSSO_FAILURE";
    ErrorCode["ProfileAccountNotVerified"] = "PROFILE_ACCOUNT_NOT_VERIFIED";
    ErrorCode["SupportAccountNotFound"] = "SUPPORT_ACCOUNT_NOT_FOUND";
    ErrorCode["SupportAccountPendingCsiApproval"] = "SUPPORT_ACCOUNT_PENDING_CSI_APPROVAL";
    ErrorCode["FreeTierCustomerSliUnsupported"] = "FREE_TIER_CUSTOMER_SLI_UNSUPPORTED";
    ErrorCode["ProfileAccountVerified"] = "PROFILE_ACCOUNT_VERIFIED";
    ErrorCode["ProfileVerifiedCsiRequestPending"] = "PROFILE_VERIFIED_CSI_REQUEST_PENDING";
    ErrorCode["ProfileVerifiedCsiRequestNotFound"] = "PROFILE_VERIFIED_CSI_REQUEST_NOT_FOUND";
    ErrorCode["CreateProfileOrganizationNameInvalid"] = "CREATE_PROFILE_ORGANIZATION_NAME_INVALID";
    ErrorCode["CreateProfileEmailInvalid"] = "CREATE_PROFILE_EMAIL_INVALID";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ErrorCode["UnknownValue"] = "UNKNOWN_VALUE";
})(ErrorCode = exports.ErrorCode || (exports.ErrorCode = {}));
(function (ErrorCode) {
    function getJsonObj(obj) {
        return obj;
    }
    ErrorCode.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ErrorCode.getDeserializedJsonObj = getDeserializedJsonObj;
})(ErrorCode = exports.ErrorCode || (exports.ErrorCode = {}));
