// oci-generativeaiagent/lib/model/index.js
"use strict";
/**
 * Generative AI Agents Management API
 * OCI Generative AI Agents is a fully managed service that combines the power of large language models (LLMs) with an intelligent retrieval system to create contextually relevant answers by searching your knowledge base, making your AI applications smart and efficient.

OCI Generative AI Agents supports several ways to onboard your data and then allows you and your customers to interact with your data using a chat interface or API.

Use the Generative AI Agents API to create and manage agents, knowledge bases, data sources, endpoints, data ingestion jobs, and work requests.

For creating and managing client chat sessions see the {@link #eNGenerative-ai-agents-clientLatest(ENGenerative-ai-agents-clientLatestRequest) eNGenerative-ai-agents-clientLatest}.

To learn more about the service, see the [Generative AI Agents documentation](https://docs.oracle.com/iaas/Content/generative-ai-agents/home.htm).

 * OpenAPI spec version: 20240531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutputLocation = exports.OutputConfig = exports.OperationType = exports.OperationStatus = exports.ObjectStoragePrefix = exports.LlmCustomization = exports.LlmConfig = exports.KnowledgeBaseSummary = exports.KnowledgeBaseStatistics = exports.KnowledgeBaseConfig = exports.KnowledgeBaseCollection = exports.KnowledgeBase = exports.InputLocation = exports.IndexSchema = exports.IndexConfig = exports.Index = exports.HumanInputConfig = exports.HttpEndpointAuthConfig = exports.GuardrailMode = exports.GuardrailConfig = exports.Function = exports.DatabaseFunction = exports.DatabaseConnection = exports.DataSourceSummary = exports.DataSourceConfig = exports.DataSourceCollection = exports.DataSource = exports.DataIngestionJobSummary = exports.DataIngestionJobStatistics = exports.DataIngestionJobCollection = exports.DataIngestionJob = exports.CreateToolDetails = exports.CreateKnowledgeBaseDetails = exports.CreateDataSourceDetails = exports.CreateDataIngestionJobDetails = exports.CreateAgentEndpointDetails = exports.CreateAgentDetails = exports.ContentModerationGuardrailConfig = exports.ContentModerationConfig = exports.ChangeKnowledgeBaseCompartmentDetails = exports.ChangeAgentEndpointCompartmentDetails = exports.ChangeAgentCompartmentDetails = exports.ApiSchemaInputLocation = exports.AgentSummary = exports.AgentEndpointSummary = exports.AgentEndpointCollection = exports.AgentEndpoint = exports.AgentCollection = exports.Agent = exports.ActionType = void 0;
exports.SqlToolConfig = exports.RagToolConfig = exports.OciOpenSearchIndexConfig = exports.OciObjectStorageDataSourceConfig = exports.OciDatabaseConfig = exports.ObjectStoragePrefixOutputLocation = exports.ObjectStorageInputLocation = exports.InlineInputLocation = exports.IdcsSecret = exports.HttpEndpointToolConfig = exports.HttpEndpointOciResourcePrincipalAuthConfig = exports.HttpEndpointNoAuthConfig = exports.HttpEndpointIdcsAuthConfig = exports.HttpEndpointDelegatedBearerAuthConfig = exports.FunctionCallingToolConfig = exports.DefaultIndexConfig = exports.DatabaseToolConnection = exports.BasicAuthSecret = exports.ApiSchemaObjectStorageInputLocation = exports.ApiSchemaInlineInputLocation = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateToolDetails = exports.UpdateKnowledgeBaseDetails = exports.UpdateDataSourceDetails = exports.UpdateAgentEndpointDetails = exports.UpdateAgentDetails = exports.ToolSummary = exports.ToolConfig = exports.ToolCollection = exports.Tool = exports.SortOrder = exports.SessionConfig = exports.SecretDetail = exports.PromptInjectionGuardrailConfig = exports.PersonallyIdentifiableInformationGuardrailConfig = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const Agent = __importStar(require("./agent"));
exports.Agent = Agent.Agent;
const AgentCollection = __importStar(require("./agent-collection"));
exports.AgentCollection = AgentCollection.AgentCollection;
const AgentEndpoint = __importStar(require("./agent-endpoint"));
exports.AgentEndpoint = AgentEndpoint.AgentEndpoint;
const AgentEndpointCollection = __importStar(require("./agent-endpoint-collection"));
exports.AgentEndpointCollection = AgentEndpointCollection.AgentEndpointCollection;
const AgentEndpointSummary = __importStar(require("./agent-endpoint-summary"));
exports.AgentEndpointSummary = AgentEndpointSummary.AgentEndpointSummary;
const AgentSummary = __importStar(require("./agent-summary"));
exports.AgentSummary = AgentSummary.AgentSummary;
const ApiSchemaInputLocation = __importStar(require("./api-schema-input-location"));
exports.ApiSchemaInputLocation = ApiSchemaInputLocation.ApiSchemaInputLocation;
const ChangeAgentCompartmentDetails = __importStar(require("./change-agent-compartment-details"));
exports.ChangeAgentCompartmentDetails = ChangeAgentCompartmentDetails.ChangeAgentCompartmentDetails;
const ChangeAgentEndpointCompartmentDetails = __importStar(require("./change-agent-endpoint-compartment-details"));
exports.ChangeAgentEndpointCompartmentDetails = ChangeAgentEndpointCompartmentDetails.ChangeAgentEndpointCompartmentDetails;
const ChangeKnowledgeBaseCompartmentDetails = __importStar(require("./change-knowledge-base-compartment-details"));
exports.ChangeKnowledgeBaseCompartmentDetails = ChangeKnowledgeBaseCompartmentDetails.ChangeKnowledgeBaseCompartmentDetails;
const ContentModerationConfig = __importStar(require("./content-moderation-config"));
exports.ContentModerationConfig = ContentModerationConfig.ContentModerationConfig;
const ContentModerationGuardrailConfig = __importStar(require("./content-moderation-guardrail-config"));
exports.ContentModerationGuardrailConfig = ContentModerationGuardrailConfig.ContentModerationGuardrailConfig;
const CreateAgentDetails = __importStar(require("./create-agent-details"));
exports.CreateAgentDetails = CreateAgentDetails.CreateAgentDetails;
const CreateAgentEndpointDetails = __importStar(require("./create-agent-endpoint-details"));
exports.CreateAgentEndpointDetails = CreateAgentEndpointDetails.CreateAgentEndpointDetails;
const CreateDataIngestionJobDetails = __importStar(require("./create-data-ingestion-job-details"));
exports.CreateDataIngestionJobDetails = CreateDataIngestionJobDetails.CreateDataIngestionJobDetails;
const CreateDataSourceDetails = __importStar(require("./create-data-source-details"));
exports.CreateDataSourceDetails = CreateDataSourceDetails.CreateDataSourceDetails;
const CreateKnowledgeBaseDetails = __importStar(require("./create-knowledge-base-details"));
exports.CreateKnowledgeBaseDetails = CreateKnowledgeBaseDetails.CreateKnowledgeBaseDetails;
const CreateToolDetails = __importStar(require("./create-tool-details"));
exports.CreateToolDetails = CreateToolDetails.CreateToolDetails;
const DataIngestionJob = __importStar(require("./data-ingestion-job"));
exports.DataIngestionJob = DataIngestionJob.DataIngestionJob;
const DataIngestionJobCollection = __importStar(require("./data-ingestion-job-collection"));
exports.DataIngestionJobCollection = DataIngestionJobCollection.DataIngestionJobCollection;
const DataIngestionJobStatistics = __importStar(require("./data-ingestion-job-statistics"));
exports.DataIngestionJobStatistics = DataIngestionJobStatistics.DataIngestionJobStatistics;
const DataIngestionJobSummary = __importStar(require("./data-ingestion-job-summary"));
exports.DataIngestionJobSummary = DataIngestionJobSummary.DataIngestionJobSummary;
const DataSource = __importStar(require("./data-source"));
exports.DataSource = DataSource.DataSource;
const DataSourceCollection = __importStar(require("./data-source-collection"));
exports.DataSourceCollection = DataSourceCollection.DataSourceCollection;
const DataSourceConfig = __importStar(require("./data-source-config"));
exports.DataSourceConfig = DataSourceConfig.DataSourceConfig;
const DataSourceSummary = __importStar(require("./data-source-summary"));
exports.DataSourceSummary = DataSourceSummary.DataSourceSummary;
const DatabaseConnection = __importStar(require("./database-connection"));
exports.DatabaseConnection = DatabaseConnection.DatabaseConnection;
const DatabaseFunction = __importStar(require("./database-function"));
exports.DatabaseFunction = DatabaseFunction.DatabaseFunction;
const Function = __importStar(require("./function"));
exports.Function = Function.Function;
const GuardrailConfig = __importStar(require("./guardrail-config"));
exports.GuardrailConfig = GuardrailConfig.GuardrailConfig;
const GuardrailMode = __importStar(require("./guardrail-mode"));
exports.GuardrailMode = GuardrailMode.GuardrailMode;
const HttpEndpointAuthConfig = __importStar(require("./http-endpoint-auth-config"));
exports.HttpEndpointAuthConfig = HttpEndpointAuthConfig.HttpEndpointAuthConfig;
const HumanInputConfig = __importStar(require("./human-input-config"));
exports.HumanInputConfig = HumanInputConfig.HumanInputConfig;
const Index = __importStar(require("./model-index"));
exports.Index = Index.Index;
const IndexConfig = __importStar(require("./index-config"));
exports.IndexConfig = IndexConfig.IndexConfig;
const IndexSchema = __importStar(require("./index-schema"));
exports.IndexSchema = IndexSchema.IndexSchema;
const InputLocation = __importStar(require("./input-location"));
exports.InputLocation = InputLocation.InputLocation;
const KnowledgeBase = __importStar(require("./knowledge-base"));
exports.KnowledgeBase = KnowledgeBase.KnowledgeBase;
const KnowledgeBaseCollection = __importStar(require("./knowledge-base-collection"));
exports.KnowledgeBaseCollection = KnowledgeBaseCollection.KnowledgeBaseCollection;
const KnowledgeBaseConfig = __importStar(require("./knowledge-base-config"));
exports.KnowledgeBaseConfig = KnowledgeBaseConfig.KnowledgeBaseConfig;
const KnowledgeBaseStatistics = __importStar(require("./knowledge-base-statistics"));
exports.KnowledgeBaseStatistics = KnowledgeBaseStatistics.KnowledgeBaseStatistics;
const KnowledgeBaseSummary = __importStar(require("./knowledge-base-summary"));
exports.KnowledgeBaseSummary = KnowledgeBaseSummary.KnowledgeBaseSummary;
const LlmConfig = __importStar(require("./llm-config"));
exports.LlmConfig = LlmConfig.LlmConfig;
const LlmCustomization = __importStar(require("./llm-customization"));
exports.LlmCustomization = LlmCustomization.LlmCustomization;
const ObjectStoragePrefix = __importStar(require("./object-storage-prefix"));
exports.ObjectStoragePrefix = ObjectStoragePrefix.ObjectStoragePrefix;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OutputConfig = __importStar(require("./output-config"));
exports.OutputConfig = OutputConfig.OutputConfig;
const OutputLocation = __importStar(require("./output-location"));
exports.OutputLocation = OutputLocation.OutputLocation;
const PersonallyIdentifiableInformationGuardrailConfig = __importStar(require("./personally-identifiable-information-guardrail-config"));
exports.PersonallyIdentifiableInformationGuardrailConfig = PersonallyIdentifiableInformationGuardrailConfig.PersonallyIdentifiableInformationGuardrailConfig;
const PromptInjectionGuardrailConfig = __importStar(require("./prompt-injection-guardrail-config"));
exports.PromptInjectionGuardrailConfig = PromptInjectionGuardrailConfig.PromptInjectionGuardrailConfig;
const SecretDetail = __importStar(require("./secret-detail"));
exports.SecretDetail = SecretDetail.SecretDetail;
const SessionConfig = __importStar(require("./session-config"));
exports.SessionConfig = SessionConfig.SessionConfig;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Tool = __importStar(require("./tool"));
exports.Tool = Tool.Tool;
const ToolCollection = __importStar(require("./tool-collection"));
exports.ToolCollection = ToolCollection.ToolCollection;
const ToolConfig = __importStar(require("./tool-config"));
exports.ToolConfig = ToolConfig.ToolConfig;
const ToolSummary = __importStar(require("./tool-summary"));
exports.ToolSummary = ToolSummary.ToolSummary;
const UpdateAgentDetails = __importStar(require("./update-agent-details"));
exports.UpdateAgentDetails = UpdateAgentDetails.UpdateAgentDetails;
const UpdateAgentEndpointDetails = __importStar(require("./update-agent-endpoint-details"));
exports.UpdateAgentEndpointDetails = UpdateAgentEndpointDetails.UpdateAgentEndpointDetails;
const UpdateDataSourceDetails = __importStar(require("./update-data-source-details"));
exports.UpdateDataSourceDetails = UpdateDataSourceDetails.UpdateDataSourceDetails;
const UpdateKnowledgeBaseDetails = __importStar(require("./update-knowledge-base-details"));
exports.UpdateKnowledgeBaseDetails = UpdateKnowledgeBaseDetails.UpdateKnowledgeBaseDetails;
const UpdateToolDetails = __importStar(require("./update-tool-details"));
exports.UpdateToolDetails = UpdateToolDetails.UpdateToolDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const ApiSchemaInlineInputLocation = __importStar(require("./api-schema-inline-input-location"));
exports.ApiSchemaInlineInputLocation = ApiSchemaInlineInputLocation.ApiSchemaInlineInputLocation;
const ApiSchemaObjectStorageInputLocation = __importStar(require("./api-schema-object-storage-input-location"));
exports.ApiSchemaObjectStorageInputLocation = ApiSchemaObjectStorageInputLocation.ApiSchemaObjectStorageInputLocation;
const BasicAuthSecret = __importStar(require("./basic-auth-secret"));
exports.BasicAuthSecret = BasicAuthSecret.BasicAuthSecret;
const DatabaseToolConnection = __importStar(require("./database-tool-connection"));
exports.DatabaseToolConnection = DatabaseToolConnection.DatabaseToolConnection;
const DefaultIndexConfig = __importStar(require("./default-index-config"));
exports.DefaultIndexConfig = DefaultIndexConfig.DefaultIndexConfig;
const FunctionCallingToolConfig = __importStar(require("./function-calling-tool-config"));
exports.FunctionCallingToolConfig = FunctionCallingToolConfig.FunctionCallingToolConfig;
const HttpEndpointDelegatedBearerAuthConfig = __importStar(require("./http-endpoint-delegated-bearer-auth-config"));
exports.HttpEndpointDelegatedBearerAuthConfig = HttpEndpointDelegatedBearerAuthConfig.HttpEndpointDelegatedBearerAuthConfig;
const HttpEndpointIdcsAuthConfig = __importStar(require("./http-endpoint-idcs-auth-config"));
exports.HttpEndpointIdcsAuthConfig = HttpEndpointIdcsAuthConfig.HttpEndpointIdcsAuthConfig;
const HttpEndpointNoAuthConfig = __importStar(require("./http-endpoint-no-auth-config"));
exports.HttpEndpointNoAuthConfig = HttpEndpointNoAuthConfig.HttpEndpointNoAuthConfig;
const HttpEndpointOciResourcePrincipalAuthConfig = __importStar(require("./http-endpoint-oci-resource-principal-auth-config"));
exports.HttpEndpointOciResourcePrincipalAuthConfig = HttpEndpointOciResourcePrincipalAuthConfig.HttpEndpointOciResourcePrincipalAuthConfig;
const HttpEndpointToolConfig = __importStar(require("./http-endpoint-tool-config"));
exports.HttpEndpointToolConfig = HttpEndpointToolConfig.HttpEndpointToolConfig;
const IdcsSecret = __importStar(require("./idcs-secret"));
exports.IdcsSecret = IdcsSecret.IdcsSecret;
const InlineInputLocation = __importStar(require("./inline-input-location"));
exports.InlineInputLocation = InlineInputLocation.InlineInputLocation;
const ObjectStorageInputLocation = __importStar(require("./object-storage-input-location"));
exports.ObjectStorageInputLocation = ObjectStorageInputLocation.ObjectStorageInputLocation;
const ObjectStoragePrefixOutputLocation = __importStar(require("./object-storage-prefix-output-location"));
exports.ObjectStoragePrefixOutputLocation = ObjectStoragePrefixOutputLocation.ObjectStoragePrefixOutputLocation;
const OciDatabaseConfig = __importStar(require("./oci-database-config"));
exports.OciDatabaseConfig = OciDatabaseConfig.OciDatabaseConfig;
const OciObjectStorageDataSourceConfig = __importStar(require("./oci-object-storage-data-source-config"));
exports.OciObjectStorageDataSourceConfig = OciObjectStorageDataSourceConfig.OciObjectStorageDataSourceConfig;
const OciOpenSearchIndexConfig = __importStar(require("./oci-open-search-index-config"));
exports.OciOpenSearchIndexConfig = OciOpenSearchIndexConfig.OciOpenSearchIndexConfig;
const RagToolConfig = __importStar(require("./rag-tool-config"));
exports.RagToolConfig = RagToolConfig.RagToolConfig;
const SqlToolConfig = __importStar(require("./sql-tool-config"));
exports.SqlToolConfig = SqlToolConfig.SqlToolConfig;
