// oci-usageapi/lib/model/index.js
"use strict";
/**
 * Usage API
 * Use the Usage API to view your Oracle Cloud usage and costs. The API allows you to request data that meets the specified filter criteria, and to group that data by the chosen dimension. The Usage API is used by [Cost Analysis](https://docs.oracle.com/iaas/Content/Billing/Concepts/costanalysisoverview.htm), [Scheduled Reports](https://docs.oracle.com/iaas/Content/Billing/Concepts/scheduledreportoverview.htm), and [Carbon Emissions Analysis](https://docs.oracle.com/iaas/Content/General/Concepts/emissions-management.htm) in the Console. Also see [Using the Usage API](https://docs.oracle.com/iaas/Content/Billing/Concepts/costanalysisoverview.htm#cost_analysis_using_the_api) for more information.
 * OpenAPI spec version: 20200107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageCarbonEmissionsQuerySummary = exports.UsageCarbonEmissionsQueryDefinition = exports.UsageCarbonEmissionsQueryCollection = exports.UsageCarbonEmissionsQuery = exports.UsageCarbonEmissionSummary = exports.UsageCarbonEmissionAggregation = exports.UsageAggregation = exports.UpdateUsageCarbonEmissionsQueryDetails = exports.UpdateScheduleDetails = exports.UpdateQueryDetails = exports.UpdateEmailRecipientsGroupDetails = exports.UpdateCustomTableDetails = exports.Tag = exports.ScheduledRunSummary = exports.ScheduledRunCollection = exports.ScheduledRun = exports.ScheduleSummary = exports.ScheduleCollection = exports.Schedule = exports.SavedCustomTable = exports.ResultLocation = exports.RequestUsageCarbonEmissionsDetails = exports.RequestSummarizedUsagesDetails = exports.ReportQuery = exports.QuerySummary = exports.QueryProperties = exports.QueryDefinition = exports.QueryCollection = exports.Query = exports.Forecast = exports.Filter = exports.EmailRecipientsGroupSummary = exports.EmailRecipientsGroupCollection = exports.EmailRecipientsGroup = exports.EmailRecipient = exports.Dimension = exports.DateRange = exports.CustomTableSummary = exports.CustomTableCollection = exports.CustomTable = exports.CreateUsageCarbonEmissionsQueryDetails = exports.CreateScheduleDetails = exports.CreateQueryDetails = exports.CreateEmailRecipientsGroupDetails = exports.CreateCustomTableDetails = exports.CostAnalysisUI = exports.ConfigurationAggregation = exports.Configuration = exports.CleanEnergyUsage = exports.AverageCarbonEmission = void 0;
exports.StaticDateRange = exports.ObjectStorageLocation = exports.DynamicDateRange = exports.UsageSummary = exports.UsageCarbonEmissionsReportQuery = void 0;
const AverageCarbonEmission = __importStar(require("./average-carbon-emission"));
exports.AverageCarbonEmission = AverageCarbonEmission.AverageCarbonEmission;
const CleanEnergyUsage = __importStar(require("./clean-energy-usage"));
exports.CleanEnergyUsage = CleanEnergyUsage.CleanEnergyUsage;
const Configuration = __importStar(require("./configuration"));
exports.Configuration = Configuration.Configuration;
const ConfigurationAggregation = __importStar(require("./configuration-aggregation"));
exports.ConfigurationAggregation = ConfigurationAggregation.ConfigurationAggregation;
const CostAnalysisUI = __importStar(require("./cost-analysis-ui"));
exports.CostAnalysisUI = CostAnalysisUI.CostAnalysisUI;
const CreateCustomTableDetails = __importStar(require("./create-custom-table-details"));
exports.CreateCustomTableDetails = CreateCustomTableDetails.CreateCustomTableDetails;
const CreateEmailRecipientsGroupDetails = __importStar(require("./create-email-recipients-group-details"));
exports.CreateEmailRecipientsGroupDetails = CreateEmailRecipientsGroupDetails.CreateEmailRecipientsGroupDetails;
const CreateQueryDetails = __importStar(require("./create-query-details"));
exports.CreateQueryDetails = CreateQueryDetails.CreateQueryDetails;
const CreateScheduleDetails = __importStar(require("./create-schedule-details"));
exports.CreateScheduleDetails = CreateScheduleDetails.CreateScheduleDetails;
const CreateUsageCarbonEmissionsQueryDetails = __importStar(require("./create-usage-carbon-emissions-query-details"));
exports.CreateUsageCarbonEmissionsQueryDetails = CreateUsageCarbonEmissionsQueryDetails.CreateUsageCarbonEmissionsQueryDetails;
const CustomTable = __importStar(require("./custom-table"));
exports.CustomTable = CustomTable.CustomTable;
const CustomTableCollection = __importStar(require("./custom-table-collection"));
exports.CustomTableCollection = CustomTableCollection.CustomTableCollection;
const CustomTableSummary = __importStar(require("./custom-table-summary"));
exports.CustomTableSummary = CustomTableSummary.CustomTableSummary;
const DateRange = __importStar(require("./date-range"));
exports.DateRange = DateRange.DateRange;
const Dimension = __importStar(require("./dimension"));
exports.Dimension = Dimension.Dimension;
const EmailRecipient = __importStar(require("./email-recipient"));
exports.EmailRecipient = EmailRecipient.EmailRecipient;
const EmailRecipientsGroup = __importStar(require("./email-recipients-group"));
exports.EmailRecipientsGroup = EmailRecipientsGroup.EmailRecipientsGroup;
const EmailRecipientsGroupCollection = __importStar(require("./email-recipients-group-collection"));
exports.EmailRecipientsGroupCollection = EmailRecipientsGroupCollection.EmailRecipientsGroupCollection;
const EmailRecipientsGroupSummary = __importStar(require("./email-recipients-group-summary"));
exports.EmailRecipientsGroupSummary = EmailRecipientsGroupSummary.EmailRecipientsGroupSummary;
const Filter = __importStar(require("./filter"));
exports.Filter = Filter.Filter;
const Forecast = __importStar(require("./forecast"));
exports.Forecast = Forecast.Forecast;
const Query = __importStar(require("./query"));
exports.Query = Query.Query;
const QueryCollection = __importStar(require("./query-collection"));
exports.QueryCollection = QueryCollection.QueryCollection;
const QueryDefinition = __importStar(require("./query-definition"));
exports.QueryDefinition = QueryDefinition.QueryDefinition;
const QueryProperties = __importStar(require("./query-properties"));
exports.QueryProperties = QueryProperties.QueryProperties;
const QuerySummary = __importStar(require("./query-summary"));
exports.QuerySummary = QuerySummary.QuerySummary;
const ReportQuery = __importStar(require("./report-query"));
exports.ReportQuery = ReportQuery.ReportQuery;
const RequestSummarizedUsagesDetails = __importStar(require("./request-summarized-usages-details"));
exports.RequestSummarizedUsagesDetails = RequestSummarizedUsagesDetails.RequestSummarizedUsagesDetails;
const RequestUsageCarbonEmissionsDetails = __importStar(require("./request-usage-carbon-emissions-details"));
exports.RequestUsageCarbonEmissionsDetails = RequestUsageCarbonEmissionsDetails.RequestUsageCarbonEmissionsDetails;
const ResultLocation = __importStar(require("./result-location"));
exports.ResultLocation = ResultLocation.ResultLocation;
const SavedCustomTable = __importStar(require("./saved-custom-table"));
exports.SavedCustomTable = SavedCustomTable.SavedCustomTable;
const Schedule = __importStar(require("./schedule"));
exports.Schedule = Schedule.Schedule;
const ScheduleCollection = __importStar(require("./schedule-collection"));
exports.ScheduleCollection = ScheduleCollection.ScheduleCollection;
const ScheduleSummary = __importStar(require("./schedule-summary"));
exports.ScheduleSummary = ScheduleSummary.ScheduleSummary;
const ScheduledRun = __importStar(require("./scheduled-run"));
exports.ScheduledRun = ScheduledRun.ScheduledRun;
const ScheduledRunCollection = __importStar(require("./scheduled-run-collection"));
exports.ScheduledRunCollection = ScheduledRunCollection.ScheduledRunCollection;
const ScheduledRunSummary = __importStar(require("./scheduled-run-summary"));
exports.ScheduledRunSummary = ScheduledRunSummary.ScheduledRunSummary;
const Tag = __importStar(require("./tag"));
exports.Tag = Tag.Tag;
const UpdateCustomTableDetails = __importStar(require("./update-custom-table-details"));
exports.UpdateCustomTableDetails = UpdateCustomTableDetails.UpdateCustomTableDetails;
const UpdateEmailRecipientsGroupDetails = __importStar(require("./update-email-recipients-group-details"));
exports.UpdateEmailRecipientsGroupDetails = UpdateEmailRecipientsGroupDetails.UpdateEmailRecipientsGroupDetails;
const UpdateQueryDetails = __importStar(require("./update-query-details"));
exports.UpdateQueryDetails = UpdateQueryDetails.UpdateQueryDetails;
const UpdateScheduleDetails = __importStar(require("./update-schedule-details"));
exports.UpdateScheduleDetails = UpdateScheduleDetails.UpdateScheduleDetails;
const UpdateUsageCarbonEmissionsQueryDetails = __importStar(require("./update-usage-carbon-emissions-query-details"));
exports.UpdateUsageCarbonEmissionsQueryDetails = UpdateUsageCarbonEmissionsQueryDetails.UpdateUsageCarbonEmissionsQueryDetails;
const UsageAggregation = __importStar(require("./usage-aggregation"));
exports.UsageAggregation = UsageAggregation.UsageAggregation;
const UsageCarbonEmissionAggregation = __importStar(require("./usage-carbon-emission-aggregation"));
exports.UsageCarbonEmissionAggregation = UsageCarbonEmissionAggregation.UsageCarbonEmissionAggregation;
const UsageCarbonEmissionSummary = __importStar(require("./usage-carbon-emission-summary"));
exports.UsageCarbonEmissionSummary = UsageCarbonEmissionSummary.UsageCarbonEmissionSummary;
const UsageCarbonEmissionsQuery = __importStar(require("./usage-carbon-emissions-query"));
exports.UsageCarbonEmissionsQuery = UsageCarbonEmissionsQuery.UsageCarbonEmissionsQuery;
const UsageCarbonEmissionsQueryCollection = __importStar(require("./usage-carbon-emissions-query-collection"));
exports.UsageCarbonEmissionsQueryCollection = UsageCarbonEmissionsQueryCollection.UsageCarbonEmissionsQueryCollection;
const UsageCarbonEmissionsQueryDefinition = __importStar(require("./usage-carbon-emissions-query-definition"));
exports.UsageCarbonEmissionsQueryDefinition = UsageCarbonEmissionsQueryDefinition.UsageCarbonEmissionsQueryDefinition;
const UsageCarbonEmissionsQuerySummary = __importStar(require("./usage-carbon-emissions-query-summary"));
exports.UsageCarbonEmissionsQuerySummary = UsageCarbonEmissionsQuerySummary.UsageCarbonEmissionsQuerySummary;
const UsageCarbonEmissionsReportQuery = __importStar(require("./usage-carbon-emissions-report-query"));
exports.UsageCarbonEmissionsReportQuery = UsageCarbonEmissionsReportQuery.UsageCarbonEmissionsReportQuery;
const UsageSummary = __importStar(require("./usage-summary"));
exports.UsageSummary = UsageSummary.UsageSummary;
const DynamicDateRange = __importStar(require("./dynamic-date-range"));
exports.DynamicDateRange = DynamicDateRange.DynamicDateRange;
const ObjectStorageLocation = __importStar(require("./object-storage-location"));
exports.ObjectStorageLocation = ObjectStorageLocation.ObjectStorageLocation;
const StaticDateRange = __importStar(require("./static-date-range"));
exports.StaticDateRange = StaticDateRange.StaticDateRange;
