// oci-marketplace/lib/model/pricing-currency-enum.js
"use strict";
/**
 * Marketplace Service API
 * Use the Marketplace API to manage applications in Oracle Cloud Infrastructure Marketplace. For more information, see [Overview of Marketplace](https://docs.oracle.com/iaas/Content/Marketplace/Concepts/marketoverview.htm)
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricingCurrencyEnum = void 0;
/**
 * Possible values for the pricing currency code.
 **/
var PricingCurrencyEnum;
(function (PricingCurrencyEnum) {
    PricingCurrencyEnum["Usd"] = "USD";
    PricingCurrencyEnum["Cad"] = "CAD";
    PricingCurrencyEnum["Inr"] = "INR";
    PricingCurrencyEnum["Gbp"] = "GBP";
    PricingCurrencyEnum["Brl"] = "BRL";
    PricingCurrencyEnum["Jpy"] = "JPY";
    PricingCurrencyEnum["Omr"] = "OMR";
    PricingCurrencyEnum["Eur"] = "EUR";
    PricingCurrencyEnum["Chf"] = "CHF";
    PricingCurrencyEnum["Mxn"] = "MXN";
    PricingCurrencyEnum["Clp"] = "CLP";
    PricingCurrencyEnum["All"] = "ALL";
    PricingCurrencyEnum["Ars"] = "ARS";
    PricingCurrencyEnum["Aud"] = "AUD";
    PricingCurrencyEnum["Bdt"] = "BDT";
    PricingCurrencyEnum["Bam"] = "BAM";
    PricingCurrencyEnum["Bgn"] = "BGN";
    PricingCurrencyEnum["Cny"] = "CNY";
    PricingCurrencyEnum["Cop"] = "COP";
    PricingCurrencyEnum["Crc"] = "CRC";
    PricingCurrencyEnum["Hrk"] = "HRK";
    PricingCurrencyEnum["Czk"] = "CZK";
    PricingCurrencyEnum["Dkk"] = "DKK";
    PricingCurrencyEnum["Egp"] = "EGP";
    PricingCurrencyEnum["Hkd"] = "HKD";
    PricingCurrencyEnum["Huf"] = "HUF";
    PricingCurrencyEnum["Isk"] = "ISK";
    PricingCurrencyEnum["Idr"] = "IDR";
    PricingCurrencyEnum["Ils"] = "ILS";
    PricingCurrencyEnum["Jmd"] = "JMD";
    PricingCurrencyEnum["Kzt"] = "KZT";
    PricingCurrencyEnum["Kes"] = "KES";
    PricingCurrencyEnum["Krw"] = "KRW";
    PricingCurrencyEnum["Kwd"] = "KWD";
    PricingCurrencyEnum["Lbp"] = "LBP";
    PricingCurrencyEnum["Mop"] = "MOP";
    PricingCurrencyEnum["Myr"] = "MYR";
    PricingCurrencyEnum["Mvr"] = "MVR";
    PricingCurrencyEnum["Aed"] = "AED";
    PricingCurrencyEnum["Nzd"] = "NZD";
    PricingCurrencyEnum["Nok"] = "NOK";
    PricingCurrencyEnum["Pkr"] = "PKR";
    PricingCurrencyEnum["Pen"] = "PEN";
    PricingCurrencyEnum["Php"] = "PHP";
    PricingCurrencyEnum["Pln"] = "PLN";
    PricingCurrencyEnum["Qar"] = "QAR";
    PricingCurrencyEnum["Ron"] = "RON";
    PricingCurrencyEnum["Sar"] = "SAR";
    PricingCurrencyEnum["Rsd"] = "RSD";
    PricingCurrencyEnum["Sgd"] = "SGD";
    PricingCurrencyEnum["Zar"] = "ZAR";
    PricingCurrencyEnum["Sek"] = "SEK";
    PricingCurrencyEnum["Twd"] = "TWD";
    PricingCurrencyEnum["Thb"] = "THB";
    PricingCurrencyEnum["Try"] = "TRY";
    PricingCurrencyEnum["Vnd"] = "VND";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PricingCurrencyEnum["UnknownValue"] = "UNKNOWN_VALUE";
})(PricingCurrencyEnum = exports.PricingCurrencyEnum || (exports.PricingCurrencyEnum = {}));
(function (PricingCurrencyEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    PricingCurrencyEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PricingCurrencyEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(PricingCurrencyEnum = exports.PricingCurrencyEnum || (exports.PricingCurrencyEnum = {}));
