// oci-marketplace/lib/model/work-request-action-type.js
"use strict";
/**
 * Marketplace Service API
 * Use the Marketplace API to manage applications in Oracle Cloud Infrastructure Marketplace. For more information, see [Overview of Marketplace](https://docs.oracle.com/iaas/Content/Marketplace/Concepts/marketoverview.htm)
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestActionType = void 0;
/**
 * Possible work request resource actions.
 **/
var WorkRequestActionType;
(function (WorkRequestActionType) {
    WorkRequestActionType["Accepted"] = "ACCEPTED";
    WorkRequestActionType["InProgress"] = "IN_PROGRESS";
    WorkRequestActionType["Waiting"] = "WAITING";
    WorkRequestActionType["NeedsAttention"] = "NEEDS_ATTENTION";
    WorkRequestActionType["Failed"] = "FAILED";
    WorkRequestActionType["Succeeded"] = "SUCCEEDED";
    WorkRequestActionType["Canceled"] = "CANCELED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestActionType["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestActionType = exports.WorkRequestActionType || (exports.WorkRequestActionType = {}));
(function (WorkRequestActionType) {
    function getJsonObj(obj) {
        return obj;
    }
    WorkRequestActionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WorkRequestActionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkRequestActionType = exports.WorkRequestActionType || (exports.WorkRequestActionType = {}));
