// oci-announcementsservice/lib/request/list-services-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListServicesRequest = void 0;
var ListServicesRequest;
(function (ListServicesRequest) {
    let PlatformType;
    (function (PlatformType) {
        PlatformType["Iaas"] = "IAAS";
        PlatformType["Saas"] = "SAAS";
        PlatformType["Paas"] = "PAAS";
    })(PlatformType = ListServicesRequest.PlatformType || (ListServicesRequest.PlatformType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["ServiceName"] = "serviceName";
    })(SortBy = ListServicesRequest.SortBy || (ListServicesRequest.SortBy = {}));
})(ListServicesRequest = exports.ListServicesRequest || (exports.ListServicesRequest = {}));
