// oci-recovery/lib/model/deletion-schedule.js
"use strict";
/**
 * Oracle Database Autonomous Recovery Service API
 * Use Oracle Database Autonomous Recovery Service API to manage Protected Databases.
 * OpenAPI spec version: 20210216
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeletionSchedule = void 0;
/**
 * The preferred schedule to delete a protected database and its backups after you terminate the source database -
 * * If you do not specify a value, then Recovery Service schedules to delete the protected database 72 hours (3 days) after the database is terminated.
 * * If you set the option as DELETE_AFTER_RETENTION_PERIOD, then Recovery Service schedules to delete the protected database after the backup retention period expires.
 *
 **/
var DeletionSchedule;
(function (DeletionSchedule) {
    DeletionSchedule["DeleteAfterRetentionPeriod"] = "DELETE_AFTER_RETENTION_PERIOD";
    DeletionSchedule["DeleteAfter72Hours"] = "DELETE_AFTER_72_HOURS";
})(DeletionSchedule = exports.DeletionSchedule || (exports.DeletionSchedule = {}));
(function (DeletionSchedule) {
    function getJsonObj(obj) {
        return obj;
    }
    DeletionSchedule.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DeletionSchedule.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeletionSchedule = exports.DeletionSchedule || (exports.DeletionSchedule = {}));
