// oci-computecloudatcustomer/lib/model/ccc-infrastructure.js
"use strict";
/**
 * Compute Cloud@Customer API
 * Use the Compute Cloud@Customer API to manage Compute Cloud@Customer infrastructures and upgrade schedules.
For more information see [Compute Cloud@Customer documentation](https://docs.oracle.com/iaas/compute-cloud-at-customer/home.htm).

 * OpenAPI spec version: 20221208
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CccInfrastructure = void 0;
const model = __importStar(require("../model"));
var CccInfrastructure;
(function (CccInfrastructure) {
    let ConnectionState;
    (function (ConnectionState) {
        ConnectionState["Reject"] = "REJECT";
        ConnectionState["Request"] = "REQUEST";
        ConnectionState["Ready"] = "READY";
        ConnectionState["Connected"] = "CONNECTED";
        ConnectionState["Disconnected"] = "DISCONNECTED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ConnectionState["UnknownValue"] = "UNKNOWN_VALUE";
    })(ConnectionState = CccInfrastructure.ConnectionState || (CccInfrastructure.ConnectionState = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = CccInfrastructure.LifecycleState || (CccInfrastructure.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "infrastructureInventory": obj.infrastructureInventory
                ? model.CccInfrastructureInventory.getJsonObj(obj.infrastructureInventory)
                : undefined,
            "infrastructureNetworkConfiguration": obj.infrastructureNetworkConfiguration
                ? model.CccInfrastructureNetworkConfiguration.getJsonObj(obj.infrastructureNetworkConfiguration)
                : undefined,
            "upgradeInformation": obj.upgradeInformation
                ? model.CccUpgradeInformation.getJsonObj(obj.upgradeInformation)
                : undefined
        });
        return jsonObj;
    }
    CccInfrastructure.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "infrastructureInventory": obj.infrastructureInventory
                ? model.CccInfrastructureInventory.getDeserializedJsonObj(obj.infrastructureInventory)
                : undefined,
            "infrastructureNetworkConfiguration": obj.infrastructureNetworkConfiguration
                ? model.CccInfrastructureNetworkConfiguration.getDeserializedJsonObj(obj.infrastructureNetworkConfiguration)
                : undefined,
            "upgradeInformation": obj.upgradeInformation
                ? model.CccUpgradeInformation.getDeserializedJsonObj(obj.upgradeInformation)
                : undefined
        });
        return jsonObj;
    }
    CccInfrastructure.getDeserializedJsonObj = getDeserializedJsonObj;
})(CccInfrastructure = exports.CccInfrastructure || (exports.CccInfrastructure = {}));
