// oci-computecloudatcustomer/lib/model/ccc-infrastructure-routing-dynamic-details.js
"use strict";
/**
 * Compute Cloud@Customer API
 * Use the Compute Cloud@Customer API to manage Compute Cloud@Customer infrastructures and upgrade schedules.
For more information see [Compute Cloud@Customer documentation](https://docs.oracle.com/iaas/compute-cloud-at-customer/home.htm).

 * OpenAPI spec version: 20221208
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CccInfrastructureRoutingDynamicDetails = void 0;
const model = __importStar(require("../model"));
var CccInfrastructureRoutingDynamicDetails;
(function (CccInfrastructureRoutingDynamicDetails) {
    let BgpTopology;
    (function (BgpTopology) {
        BgpTopology["Triangle"] = "TRIANGLE";
        BgpTopology["Square"] = "SQUARE";
        BgpTopology["Mesh"] = "MESH";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        BgpTopology["UnknownValue"] = "UNKNOWN_VALUE";
    })(BgpTopology = CccInfrastructureRoutingDynamicDetails.BgpTopology || (CccInfrastructureRoutingDynamicDetails.BgpTopology = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "peerInformation": obj.peerInformation
                ? obj.peerInformation.map(item => {
                    return model.PeerInformation.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    CccInfrastructureRoutingDynamicDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "peerInformation": obj.peerInformation
                ? obj.peerInformation.map(item => {
                    return model.PeerInformation.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    CccInfrastructureRoutingDynamicDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CccInfrastructureRoutingDynamicDetails = exports.CccInfrastructureRoutingDynamicDetails || (exports.CccInfrastructureRoutingDynamicDetails = {}));
