// oci-computecloudatcustomer/lib/model/index.js
"use strict";
/**
 * Compute Cloud@Customer API
 * Use the Compute Cloud@Customer API to manage Compute Cloud@Customer infrastructures and upgrade schedules.
For more information see [Compute Cloud@Customer documentation](https://docs.oracle.com/iaas/compute-cloud-at-customer/home.htm).

 * OpenAPI spec version: 20221208
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateCccUpgradeScheduleDetails = exports.UpdateCccScheduleEvent = exports.UpdateCccInfrastructureDetails = exports.SortOrder = exports.PeerInformation = exports.CreateCccUpgradeScheduleDetails = exports.CreateCccScheduleEvent = exports.CreateCccInfrastructureDetails = exports.ChangeCccUpgradeScheduleCompartmentDetails = exports.ChangeCccInfrastructureCompartmentDetails = exports.CccUpgradeScheduleSummary = exports.CccUpgradeScheduleCollection = exports.CccUpgradeSchedule = exports.CccUpgradeInformation = exports.CccScheduleEvent = exports.CccInfrastructureSummary = exports.CccInfrastructureRoutingStaticDetails = exports.CccInfrastructureRoutingDynamicDetails = exports.CccInfrastructureNetworkConfiguration = exports.CccInfrastructureManagementNode = exports.CccInfrastructureInventory = exports.CccInfrastructureCollection = exports.CccInfrastructure = void 0;
const CccInfrastructure = __importStar(require("./ccc-infrastructure"));
exports.CccInfrastructure = CccInfrastructure.CccInfrastructure;
const CccInfrastructureCollection = __importStar(require("./ccc-infrastructure-collection"));
exports.CccInfrastructureCollection = CccInfrastructureCollection.CccInfrastructureCollection;
const CccInfrastructureInventory = __importStar(require("./ccc-infrastructure-inventory"));
exports.CccInfrastructureInventory = CccInfrastructureInventory.CccInfrastructureInventory;
const CccInfrastructureManagementNode = __importStar(require("./ccc-infrastructure-management-node"));
exports.CccInfrastructureManagementNode = CccInfrastructureManagementNode.CccInfrastructureManagementNode;
const CccInfrastructureNetworkConfiguration = __importStar(require("./ccc-infrastructure-network-configuration"));
exports.CccInfrastructureNetworkConfiguration = CccInfrastructureNetworkConfiguration.CccInfrastructureNetworkConfiguration;
const CccInfrastructureRoutingDynamicDetails = __importStar(require("./ccc-infrastructure-routing-dynamic-details"));
exports.CccInfrastructureRoutingDynamicDetails = CccInfrastructureRoutingDynamicDetails.CccInfrastructureRoutingDynamicDetails;
const CccInfrastructureRoutingStaticDetails = __importStar(require("./ccc-infrastructure-routing-static-details"));
exports.CccInfrastructureRoutingStaticDetails = CccInfrastructureRoutingStaticDetails.CccInfrastructureRoutingStaticDetails;
const CccInfrastructureSummary = __importStar(require("./ccc-infrastructure-summary"));
exports.CccInfrastructureSummary = CccInfrastructureSummary.CccInfrastructureSummary;
const CccScheduleEvent = __importStar(require("./ccc-schedule-event"));
exports.CccScheduleEvent = CccScheduleEvent.CccScheduleEvent;
const CccUpgradeInformation = __importStar(require("./ccc-upgrade-information"));
exports.CccUpgradeInformation = CccUpgradeInformation.CccUpgradeInformation;
const CccUpgradeSchedule = __importStar(require("./ccc-upgrade-schedule"));
exports.CccUpgradeSchedule = CccUpgradeSchedule.CccUpgradeSchedule;
const CccUpgradeScheduleCollection = __importStar(require("./ccc-upgrade-schedule-collection"));
exports.CccUpgradeScheduleCollection = CccUpgradeScheduleCollection.CccUpgradeScheduleCollection;
const CccUpgradeScheduleSummary = __importStar(require("./ccc-upgrade-schedule-summary"));
exports.CccUpgradeScheduleSummary = CccUpgradeScheduleSummary.CccUpgradeScheduleSummary;
const ChangeCccInfrastructureCompartmentDetails = __importStar(require("./change-ccc-infrastructure-compartment-details"));
exports.ChangeCccInfrastructureCompartmentDetails = ChangeCccInfrastructureCompartmentDetails.ChangeCccInfrastructureCompartmentDetails;
const ChangeCccUpgradeScheduleCompartmentDetails = __importStar(require("./change-ccc-upgrade-schedule-compartment-details"));
exports.ChangeCccUpgradeScheduleCompartmentDetails = ChangeCccUpgradeScheduleCompartmentDetails.ChangeCccUpgradeScheduleCompartmentDetails;
const CreateCccInfrastructureDetails = __importStar(require("./create-ccc-infrastructure-details"));
exports.CreateCccInfrastructureDetails = CreateCccInfrastructureDetails.CreateCccInfrastructureDetails;
const CreateCccScheduleEvent = __importStar(require("./create-ccc-schedule-event"));
exports.CreateCccScheduleEvent = CreateCccScheduleEvent.CreateCccScheduleEvent;
const CreateCccUpgradeScheduleDetails = __importStar(require("./create-ccc-upgrade-schedule-details"));
exports.CreateCccUpgradeScheduleDetails = CreateCccUpgradeScheduleDetails.CreateCccUpgradeScheduleDetails;
const PeerInformation = __importStar(require("./peer-information"));
exports.PeerInformation = PeerInformation.PeerInformation;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateCccInfrastructureDetails = __importStar(require("./update-ccc-infrastructure-details"));
exports.UpdateCccInfrastructureDetails = UpdateCccInfrastructureDetails.UpdateCccInfrastructureDetails;
const UpdateCccScheduleEvent = __importStar(require("./update-ccc-schedule-event"));
exports.UpdateCccScheduleEvent = UpdateCccScheduleEvent.UpdateCccScheduleEvent;
const UpdateCccUpgradeScheduleDetails = __importStar(require("./update-ccc-upgrade-schedule-details"));
exports.UpdateCccUpgradeScheduleDetails = UpdateCccUpgradeScheduleDetails.UpdateCccUpgradeScheduleDetails;
