// oci-networkfirewall/lib/model/app-type.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20230501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppType = void 0;
/**
 * Application Type:
 * * ICMP - An ICMP application.
 * * ICMPv6 - An ICMPv6 support application.
 *
 **/
var AppType;
(function (AppType) {
    AppType["Icmp"] = "ICMP";
    AppType["IcmpV6"] = "ICMP_V6";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AppType["UnknownValue"] = "UNKNOWN_VALUE";
})(AppType = exports.AppType || (exports.AppType = {}));
(function (AppType) {
    function getJsonObj(obj) {
        return obj;
    }
    AppType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AppType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AppType = exports.AppType || (exports.AppType = {}));
