// oci-networkfirewall/lib/model/decryption-action-type.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20230501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DecryptionActionType = void 0;
/**
 * Action:
 * * NO_DECRYPT - Matching traffic is not decrypted.
 * * DECRYPT - Matching traffic is decrypted with the specified {@code secret} according to the specified {@code decryptionProfile}.
 *
 **/
var DecryptionActionType;
(function (DecryptionActionType) {
    DecryptionActionType["NoDecrypt"] = "NO_DECRYPT";
    DecryptionActionType["Decrypt"] = "DECRYPT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DecryptionActionType["UnknownValue"] = "UNKNOWN_VALUE";
})(DecryptionActionType = exports.DecryptionActionType || (exports.DecryptionActionType = {}));
(function (DecryptionActionType) {
    function getJsonObj(obj) {
        return obj;
    }
    DecryptionActionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DecryptionActionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(DecryptionActionType = exports.DecryptionActionType || (exports.DecryptionActionType = {}));
