// oci-networkfirewall/lib/model/index.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20230501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NatRuleMatchCriteria = exports.NatRuleCollection = exports.NatRule = exports.NatConfigurationResponse = exports.NatConfigurationRequest = exports.MappedSecretType = exports.MappedSecretSummaryCollection = exports.MappedSecretSummary = exports.MappedSecret = exports.LifecycleState = exports.InspectionType = exports.InspectActionType = exports.DecryptionRuleSummaryCollection = exports.DecryptionRuleSummary = exports.DecryptionRuleMatchCriteria = exports.DecryptionRule = exports.DecryptionProfileSummaryCollection = exports.DecryptionProfileSummary = exports.DecryptionProfile = exports.DecryptionActionType = exports.CreateUrlListDetails = exports.CreateTunnelInspectionRuleDetails = exports.CreateServiceListDetails = exports.CreateServiceDetails = exports.CreateSecurityRuleDetails = exports.CreateNetworkFirewallPolicyDetails = exports.CreateNetworkFirewallDetails = exports.CreateNatRuleDetails = exports.CreateMappedSecretDetails = exports.CreateDecryptionRuleDetails = exports.CreateDecryptionProfileDetails = exports.CreateApplicationGroupDetails = exports.CreateApplicationDetails = exports.CreateAddressListDetails = exports.CloneNetworkFirewallPolicyDetails = exports.ChangeNetworkFirewallPolicyCompartmentDetails = exports.ChangeNetworkFirewallCompartmentDetails = exports.ApplyNetworkFirewallPolicyDetails = exports.ApplicationSummaryCollection = exports.ApplicationSummary = exports.ApplicationGroupSummaryCollection = exports.ApplicationGroupSummary = exports.ApplicationGroup = exports.Application = exports.AppType = exports.AddressListType = exports.AddressListSummaryCollection = exports.AddressListSummary = exports.AddressList = exports.ActionType = void 0;
exports.VxlanInspectionRuleMatchCriteria = exports.UrlPattern = exports.UrlListSummaryCollection = exports.UrlListSummary = exports.UrlList = exports.UpdateUrlListDetails = exports.UpdateTunnelInspectionRuleDetails = exports.UpdateServiceListDetails = exports.UpdateServiceDetails = exports.UpdateSecurityRuleDetails = exports.UpdateNetworkFirewallPolicyDetails = exports.UpdateNetworkFirewallDetails = exports.UpdateNatRuleDetails = exports.UpdateMappedSecretDetails = exports.UpdateDecryptionRuleDetails = exports.UpdateDecryptionProfileDetails = exports.UpdateApplicationGroupDetails = exports.UpdateApplicationDetails = exports.UpdateAddressListDetails = exports.TunnelInspectionRuleSummaryCollection = exports.TunnelInspectionRuleSummary = exports.TunnelInspectionRule = exports.TunnelInspectionProtocol = exports.TrafficInspectionType = exports.TrafficActionType = exports.SortOrder = exports.ServiceType = exports.ServiceSummaryCollection = exports.ServiceSummary = exports.ServiceListSummaryCollection = exports.ServiceListSummary = exports.ServiceList = exports.Service = exports.SecurityRuleSummaryCollection = exports.SecurityRuleSummary = exports.SecurityRuleMatchCriteria = exports.SecurityRule = exports.RulePosition = exports.PortRange = exports.OperationType = exports.OperationStatus = exports.NetworkFirewallSummary = exports.NetworkFirewallPolicySummaryCollection = exports.NetworkFirewallPolicySummary = exports.NetworkFirewallPolicy = exports.NetworkFirewallCollection = exports.NetworkFirewall = exports.NatV4ActionType = exports.NatType = exports.NatRuleSummary = void 0;
exports.VxlanInspectionRuleSummary = exports.VxlanInspectionRule = exports.VaultMappedSecret = exports.UpdateVxlanInspectionRuleDetails = exports.UpdateVaultMappedSecretDetails = exports.UpdateUdpServiceDetails = exports.UpdateTcpServiceDetails = exports.UpdateSslInboundInspectionProfileDetails = exports.UpdateSslForwardProxyProfileDetails = exports.UpdateNatV4RuleDetails = exports.UpdateIpAddressListDetails = exports.UpdateIcmpApplicationDetails = exports.UpdateIcmp6ApplicationDetails = exports.UpdateFqdnAddressListDetails = exports.UdpService = exports.TcpService = exports.SslInboundInspectionProfile = exports.SslForwardProxyProfile = exports.SimpleUrlPattern = exports.NatV4NatSummary = exports.NatV4NatRule = exports.IcmpApplicationSummary = exports.IcmpApplication = exports.Icmp6ApplicationSummary = exports.Icmp6Application = exports.CreateVxlanInspectionRuleDetails = exports.CreateVaultMappedSecretDetails = exports.CreateUdpServiceDetails = exports.CreateTcpServiceDetails = exports.CreateSslInboundInspectionProfileDetails = exports.CreateSslForwardProxyProfileDetails = exports.CreateNatV4RuleDetails = exports.CreateIcmpApplicationDetails = exports.CreateIcmp6ApplicationDetails = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.VxlanInspectionRuleProfile = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AddressList = __importStar(require("./address-list"));
exports.AddressList = AddressList.AddressList;
const AddressListSummary = __importStar(require("./address-list-summary"));
exports.AddressListSummary = AddressListSummary.AddressListSummary;
const AddressListSummaryCollection = __importStar(require("./address-list-summary-collection"));
exports.AddressListSummaryCollection = AddressListSummaryCollection.AddressListSummaryCollection;
const AddressListType = __importStar(require("./address-list-type"));
exports.AddressListType = AddressListType.AddressListType;
const AppType = __importStar(require("./app-type"));
exports.AppType = AppType.AppType;
const Application = __importStar(require("./application"));
exports.Application = Application.Application;
const ApplicationGroup = __importStar(require("./application-group"));
exports.ApplicationGroup = ApplicationGroup.ApplicationGroup;
const ApplicationGroupSummary = __importStar(require("./application-group-summary"));
exports.ApplicationGroupSummary = ApplicationGroupSummary.ApplicationGroupSummary;
const ApplicationGroupSummaryCollection = __importStar(require("./application-group-summary-collection"));
exports.ApplicationGroupSummaryCollection = ApplicationGroupSummaryCollection.ApplicationGroupSummaryCollection;
const ApplicationSummary = __importStar(require("./application-summary"));
exports.ApplicationSummary = ApplicationSummary.ApplicationSummary;
const ApplicationSummaryCollection = __importStar(require("./application-summary-collection"));
exports.ApplicationSummaryCollection = ApplicationSummaryCollection.ApplicationSummaryCollection;
const ApplyNetworkFirewallPolicyDetails = __importStar(require("./apply-network-firewall-policy-details"));
exports.ApplyNetworkFirewallPolicyDetails = ApplyNetworkFirewallPolicyDetails.ApplyNetworkFirewallPolicyDetails;
const ChangeNetworkFirewallCompartmentDetails = __importStar(require("./change-network-firewall-compartment-details"));
exports.ChangeNetworkFirewallCompartmentDetails = ChangeNetworkFirewallCompartmentDetails.ChangeNetworkFirewallCompartmentDetails;
const ChangeNetworkFirewallPolicyCompartmentDetails = __importStar(require("./change-network-firewall-policy-compartment-details"));
exports.ChangeNetworkFirewallPolicyCompartmentDetails = ChangeNetworkFirewallPolicyCompartmentDetails.ChangeNetworkFirewallPolicyCompartmentDetails;
const CloneNetworkFirewallPolicyDetails = __importStar(require("./clone-network-firewall-policy-details"));
exports.CloneNetworkFirewallPolicyDetails = CloneNetworkFirewallPolicyDetails.CloneNetworkFirewallPolicyDetails;
const CreateAddressListDetails = __importStar(require("./create-address-list-details"));
exports.CreateAddressListDetails = CreateAddressListDetails.CreateAddressListDetails;
const CreateApplicationDetails = __importStar(require("./create-application-details"));
exports.CreateApplicationDetails = CreateApplicationDetails.CreateApplicationDetails;
const CreateApplicationGroupDetails = __importStar(require("./create-application-group-details"));
exports.CreateApplicationGroupDetails = CreateApplicationGroupDetails.CreateApplicationGroupDetails;
const CreateDecryptionProfileDetails = __importStar(require("./create-decryption-profile-details"));
exports.CreateDecryptionProfileDetails = CreateDecryptionProfileDetails.CreateDecryptionProfileDetails;
const CreateDecryptionRuleDetails = __importStar(require("./create-decryption-rule-details"));
exports.CreateDecryptionRuleDetails = CreateDecryptionRuleDetails.CreateDecryptionRuleDetails;
const CreateMappedSecretDetails = __importStar(require("./create-mapped-secret-details"));
exports.CreateMappedSecretDetails = CreateMappedSecretDetails.CreateMappedSecretDetails;
const CreateNatRuleDetails = __importStar(require("./create-nat-rule-details"));
exports.CreateNatRuleDetails = CreateNatRuleDetails.CreateNatRuleDetails;
const CreateNetworkFirewallDetails = __importStar(require("./create-network-firewall-details"));
exports.CreateNetworkFirewallDetails = CreateNetworkFirewallDetails.CreateNetworkFirewallDetails;
const CreateNetworkFirewallPolicyDetails = __importStar(require("./create-network-firewall-policy-details"));
exports.CreateNetworkFirewallPolicyDetails = CreateNetworkFirewallPolicyDetails.CreateNetworkFirewallPolicyDetails;
const CreateSecurityRuleDetails = __importStar(require("./create-security-rule-details"));
exports.CreateSecurityRuleDetails = CreateSecurityRuleDetails.CreateSecurityRuleDetails;
const CreateServiceDetails = __importStar(require("./create-service-details"));
exports.CreateServiceDetails = CreateServiceDetails.CreateServiceDetails;
const CreateServiceListDetails = __importStar(require("./create-service-list-details"));
exports.CreateServiceListDetails = CreateServiceListDetails.CreateServiceListDetails;
const CreateTunnelInspectionRuleDetails = __importStar(require("./create-tunnel-inspection-rule-details"));
exports.CreateTunnelInspectionRuleDetails = CreateTunnelInspectionRuleDetails.CreateTunnelInspectionRuleDetails;
const CreateUrlListDetails = __importStar(require("./create-url-list-details"));
exports.CreateUrlListDetails = CreateUrlListDetails.CreateUrlListDetails;
const DecryptionActionType = __importStar(require("./decryption-action-type"));
exports.DecryptionActionType = DecryptionActionType.DecryptionActionType;
const DecryptionProfile = __importStar(require("./decryption-profile"));
exports.DecryptionProfile = DecryptionProfile.DecryptionProfile;
const DecryptionProfileSummary = __importStar(require("./decryption-profile-summary"));
exports.DecryptionProfileSummary = DecryptionProfileSummary.DecryptionProfileSummary;
const DecryptionProfileSummaryCollection = __importStar(require("./decryption-profile-summary-collection"));
exports.DecryptionProfileSummaryCollection = DecryptionProfileSummaryCollection.DecryptionProfileSummaryCollection;
const DecryptionRule = __importStar(require("./decryption-rule"));
exports.DecryptionRule = DecryptionRule.DecryptionRule;
const DecryptionRuleMatchCriteria = __importStar(require("./decryption-rule-match-criteria"));
exports.DecryptionRuleMatchCriteria = DecryptionRuleMatchCriteria.DecryptionRuleMatchCriteria;
const DecryptionRuleSummary = __importStar(require("./decryption-rule-summary"));
exports.DecryptionRuleSummary = DecryptionRuleSummary.DecryptionRuleSummary;
const DecryptionRuleSummaryCollection = __importStar(require("./decryption-rule-summary-collection"));
exports.DecryptionRuleSummaryCollection = DecryptionRuleSummaryCollection.DecryptionRuleSummaryCollection;
const InspectActionType = __importStar(require("./inspect-action-type"));
exports.InspectActionType = InspectActionType.InspectActionType;
const InspectionType = __importStar(require("./inspection-type"));
exports.InspectionType = InspectionType.InspectionType;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const MappedSecret = __importStar(require("./mapped-secret"));
exports.MappedSecret = MappedSecret.MappedSecret;
const MappedSecretSummary = __importStar(require("./mapped-secret-summary"));
exports.MappedSecretSummary = MappedSecretSummary.MappedSecretSummary;
const MappedSecretSummaryCollection = __importStar(require("./mapped-secret-summary-collection"));
exports.MappedSecretSummaryCollection = MappedSecretSummaryCollection.MappedSecretSummaryCollection;
const MappedSecretType = __importStar(require("./mapped-secret-type"));
exports.MappedSecretType = MappedSecretType.MappedSecretType;
const NatConfigurationRequest = __importStar(require("./nat-configuration-request"));
exports.NatConfigurationRequest = NatConfigurationRequest.NatConfigurationRequest;
const NatConfigurationResponse = __importStar(require("./nat-configuration-response"));
exports.NatConfigurationResponse = NatConfigurationResponse.NatConfigurationResponse;
const NatRule = __importStar(require("./nat-rule"));
exports.NatRule = NatRule.NatRule;
const NatRuleCollection = __importStar(require("./nat-rule-collection"));
exports.NatRuleCollection = NatRuleCollection.NatRuleCollection;
const NatRuleMatchCriteria = __importStar(require("./nat-rule-match-criteria"));
exports.NatRuleMatchCriteria = NatRuleMatchCriteria.NatRuleMatchCriteria;
const NatRuleSummary = __importStar(require("./nat-rule-summary"));
exports.NatRuleSummary = NatRuleSummary.NatRuleSummary;
const NatType = __importStar(require("./nat-type"));
exports.NatType = NatType.NatType;
const NatV4ActionType = __importStar(require("./nat-v4-action-type"));
exports.NatV4ActionType = NatV4ActionType.NatV4ActionType;
const NetworkFirewall = __importStar(require("./network-firewall"));
exports.NetworkFirewall = NetworkFirewall.NetworkFirewall;
const NetworkFirewallCollection = __importStar(require("./network-firewall-collection"));
exports.NetworkFirewallCollection = NetworkFirewallCollection.NetworkFirewallCollection;
const NetworkFirewallPolicy = __importStar(require("./network-firewall-policy"));
exports.NetworkFirewallPolicy = NetworkFirewallPolicy.NetworkFirewallPolicy;
const NetworkFirewallPolicySummary = __importStar(require("./network-firewall-policy-summary"));
exports.NetworkFirewallPolicySummary = NetworkFirewallPolicySummary.NetworkFirewallPolicySummary;
const NetworkFirewallPolicySummaryCollection = __importStar(require("./network-firewall-policy-summary-collection"));
exports.NetworkFirewallPolicySummaryCollection = NetworkFirewallPolicySummaryCollection.NetworkFirewallPolicySummaryCollection;
const NetworkFirewallSummary = __importStar(require("./network-firewall-summary"));
exports.NetworkFirewallSummary = NetworkFirewallSummary.NetworkFirewallSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PortRange = __importStar(require("./port-range"));
exports.PortRange = PortRange.PortRange;
const RulePosition = __importStar(require("./rule-position"));
exports.RulePosition = RulePosition.RulePosition;
const SecurityRule = __importStar(require("./security-rule"));
exports.SecurityRule = SecurityRule.SecurityRule;
const SecurityRuleMatchCriteria = __importStar(require("./security-rule-match-criteria"));
exports.SecurityRuleMatchCriteria = SecurityRuleMatchCriteria.SecurityRuleMatchCriteria;
const SecurityRuleSummary = __importStar(require("./security-rule-summary"));
exports.SecurityRuleSummary = SecurityRuleSummary.SecurityRuleSummary;
const SecurityRuleSummaryCollection = __importStar(require("./security-rule-summary-collection"));
exports.SecurityRuleSummaryCollection = SecurityRuleSummaryCollection.SecurityRuleSummaryCollection;
const Service = __importStar(require("./service"));
exports.Service = Service.Service;
const ServiceList = __importStar(require("./service-list"));
exports.ServiceList = ServiceList.ServiceList;
const ServiceListSummary = __importStar(require("./service-list-summary"));
exports.ServiceListSummary = ServiceListSummary.ServiceListSummary;
const ServiceListSummaryCollection = __importStar(require("./service-list-summary-collection"));
exports.ServiceListSummaryCollection = ServiceListSummaryCollection.ServiceListSummaryCollection;
const ServiceSummary = __importStar(require("./service-summary"));
exports.ServiceSummary = ServiceSummary.ServiceSummary;
const ServiceSummaryCollection = __importStar(require("./service-summary-collection"));
exports.ServiceSummaryCollection = ServiceSummaryCollection.ServiceSummaryCollection;
const ServiceType = __importStar(require("./service-type"));
exports.ServiceType = ServiceType.ServiceType;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const TrafficActionType = __importStar(require("./traffic-action-type"));
exports.TrafficActionType = TrafficActionType.TrafficActionType;
const TrafficInspectionType = __importStar(require("./traffic-inspection-type"));
exports.TrafficInspectionType = TrafficInspectionType.TrafficInspectionType;
const TunnelInspectionProtocol = __importStar(require("./tunnel-inspection-protocol"));
exports.TunnelInspectionProtocol = TunnelInspectionProtocol.TunnelInspectionProtocol;
const TunnelInspectionRule = __importStar(require("./tunnel-inspection-rule"));
exports.TunnelInspectionRule = TunnelInspectionRule.TunnelInspectionRule;
const TunnelInspectionRuleSummary = __importStar(require("./tunnel-inspection-rule-summary"));
exports.TunnelInspectionRuleSummary = TunnelInspectionRuleSummary.TunnelInspectionRuleSummary;
const TunnelInspectionRuleSummaryCollection = __importStar(require("./tunnel-inspection-rule-summary-collection"));
exports.TunnelInspectionRuleSummaryCollection = TunnelInspectionRuleSummaryCollection.TunnelInspectionRuleSummaryCollection;
const UpdateAddressListDetails = __importStar(require("./update-address-list-details"));
exports.UpdateAddressListDetails = UpdateAddressListDetails.UpdateAddressListDetails;
const UpdateApplicationDetails = __importStar(require("./update-application-details"));
exports.UpdateApplicationDetails = UpdateApplicationDetails.UpdateApplicationDetails;
const UpdateApplicationGroupDetails = __importStar(require("./update-application-group-details"));
exports.UpdateApplicationGroupDetails = UpdateApplicationGroupDetails.UpdateApplicationGroupDetails;
const UpdateDecryptionProfileDetails = __importStar(require("./update-decryption-profile-details"));
exports.UpdateDecryptionProfileDetails = UpdateDecryptionProfileDetails.UpdateDecryptionProfileDetails;
const UpdateDecryptionRuleDetails = __importStar(require("./update-decryption-rule-details"));
exports.UpdateDecryptionRuleDetails = UpdateDecryptionRuleDetails.UpdateDecryptionRuleDetails;
const UpdateMappedSecretDetails = __importStar(require("./update-mapped-secret-details"));
exports.UpdateMappedSecretDetails = UpdateMappedSecretDetails.UpdateMappedSecretDetails;
const UpdateNatRuleDetails = __importStar(require("./update-nat-rule-details"));
exports.UpdateNatRuleDetails = UpdateNatRuleDetails.UpdateNatRuleDetails;
const UpdateNetworkFirewallDetails = __importStar(require("./update-network-firewall-details"));
exports.UpdateNetworkFirewallDetails = UpdateNetworkFirewallDetails.UpdateNetworkFirewallDetails;
const UpdateNetworkFirewallPolicyDetails = __importStar(require("./update-network-firewall-policy-details"));
exports.UpdateNetworkFirewallPolicyDetails = UpdateNetworkFirewallPolicyDetails.UpdateNetworkFirewallPolicyDetails;
const UpdateSecurityRuleDetails = __importStar(require("./update-security-rule-details"));
exports.UpdateSecurityRuleDetails = UpdateSecurityRuleDetails.UpdateSecurityRuleDetails;
const UpdateServiceDetails = __importStar(require("./update-service-details"));
exports.UpdateServiceDetails = UpdateServiceDetails.UpdateServiceDetails;
const UpdateServiceListDetails = __importStar(require("./update-service-list-details"));
exports.UpdateServiceListDetails = UpdateServiceListDetails.UpdateServiceListDetails;
const UpdateTunnelInspectionRuleDetails = __importStar(require("./update-tunnel-inspection-rule-details"));
exports.UpdateTunnelInspectionRuleDetails = UpdateTunnelInspectionRuleDetails.UpdateTunnelInspectionRuleDetails;
const UpdateUrlListDetails = __importStar(require("./update-url-list-details"));
exports.UpdateUrlListDetails = UpdateUrlListDetails.UpdateUrlListDetails;
const UrlList = __importStar(require("./url-list"));
exports.UrlList = UrlList.UrlList;
const UrlListSummary = __importStar(require("./url-list-summary"));
exports.UrlListSummary = UrlListSummary.UrlListSummary;
const UrlListSummaryCollection = __importStar(require("./url-list-summary-collection"));
exports.UrlListSummaryCollection = UrlListSummaryCollection.UrlListSummaryCollection;
const UrlPattern = __importStar(require("./url-pattern"));
exports.UrlPattern = UrlPattern.UrlPattern;
const VxlanInspectionRuleMatchCriteria = __importStar(require("./vxlan-inspection-rule-match-criteria"));
exports.VxlanInspectionRuleMatchCriteria = VxlanInspectionRuleMatchCriteria.VxlanInspectionRuleMatchCriteria;
const VxlanInspectionRuleProfile = __importStar(require("./vxlan-inspection-rule-profile"));
exports.VxlanInspectionRuleProfile = VxlanInspectionRuleProfile.VxlanInspectionRuleProfile;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const CreateIcmp6ApplicationDetails = __importStar(require("./create-icmp6-application-details"));
exports.CreateIcmp6ApplicationDetails = CreateIcmp6ApplicationDetails.CreateIcmp6ApplicationDetails;
const CreateIcmpApplicationDetails = __importStar(require("./create-icmp-application-details"));
exports.CreateIcmpApplicationDetails = CreateIcmpApplicationDetails.CreateIcmpApplicationDetails;
const CreateNatV4RuleDetails = __importStar(require("./create-nat-v4-rule-details"));
exports.CreateNatV4RuleDetails = CreateNatV4RuleDetails.CreateNatV4RuleDetails;
const CreateSslForwardProxyProfileDetails = __importStar(require("./create-ssl-forward-proxy-profile-details"));
exports.CreateSslForwardProxyProfileDetails = CreateSslForwardProxyProfileDetails.CreateSslForwardProxyProfileDetails;
const CreateSslInboundInspectionProfileDetails = __importStar(require("./create-ssl-inbound-inspection-profile-details"));
exports.CreateSslInboundInspectionProfileDetails = CreateSslInboundInspectionProfileDetails.CreateSslInboundInspectionProfileDetails;
const CreateTcpServiceDetails = __importStar(require("./create-tcp-service-details"));
exports.CreateTcpServiceDetails = CreateTcpServiceDetails.CreateTcpServiceDetails;
const CreateUdpServiceDetails = __importStar(require("./create-udp-service-details"));
exports.CreateUdpServiceDetails = CreateUdpServiceDetails.CreateUdpServiceDetails;
const CreateVaultMappedSecretDetails = __importStar(require("./create-vault-mapped-secret-details"));
exports.CreateVaultMappedSecretDetails = CreateVaultMappedSecretDetails.CreateVaultMappedSecretDetails;
const CreateVxlanInspectionRuleDetails = __importStar(require("./create-vxlan-inspection-rule-details"));
exports.CreateVxlanInspectionRuleDetails = CreateVxlanInspectionRuleDetails.CreateVxlanInspectionRuleDetails;
const Icmp6Application = __importStar(require("./icmp6-application"));
exports.Icmp6Application = Icmp6Application.Icmp6Application;
const Icmp6ApplicationSummary = __importStar(require("./icmp6-application-summary"));
exports.Icmp6ApplicationSummary = Icmp6ApplicationSummary.Icmp6ApplicationSummary;
const IcmpApplication = __importStar(require("./icmp-application"));
exports.IcmpApplication = IcmpApplication.IcmpApplication;
const IcmpApplicationSummary = __importStar(require("./icmp-application-summary"));
exports.IcmpApplicationSummary = IcmpApplicationSummary.IcmpApplicationSummary;
const NatV4NatRule = __importStar(require("./nat-v4-nat-rule"));
exports.NatV4NatRule = NatV4NatRule.NatV4NatRule;
const NatV4NatSummary = __importStar(require("./nat-v4-nat-summary"));
exports.NatV4NatSummary = NatV4NatSummary.NatV4NatSummary;
const SimpleUrlPattern = __importStar(require("./simple-url-pattern"));
exports.SimpleUrlPattern = SimpleUrlPattern.SimpleUrlPattern;
const SslForwardProxyProfile = __importStar(require("./ssl-forward-proxy-profile"));
exports.SslForwardProxyProfile = SslForwardProxyProfile.SslForwardProxyProfile;
const SslInboundInspectionProfile = __importStar(require("./ssl-inbound-inspection-profile"));
exports.SslInboundInspectionProfile = SslInboundInspectionProfile.SslInboundInspectionProfile;
const TcpService = __importStar(require("./tcp-service"));
exports.TcpService = TcpService.TcpService;
const UdpService = __importStar(require("./udp-service"));
exports.UdpService = UdpService.UdpService;
const UpdateFqdnAddressListDetails = __importStar(require("./update-fqdn-address-list-details"));
exports.UpdateFqdnAddressListDetails = UpdateFqdnAddressListDetails.UpdateFqdnAddressListDetails;
const UpdateIcmp6ApplicationDetails = __importStar(require("./update-icmp6-application-details"));
exports.UpdateIcmp6ApplicationDetails = UpdateIcmp6ApplicationDetails.UpdateIcmp6ApplicationDetails;
const UpdateIcmpApplicationDetails = __importStar(require("./update-icmp-application-details"));
exports.UpdateIcmpApplicationDetails = UpdateIcmpApplicationDetails.UpdateIcmpApplicationDetails;
const UpdateIpAddressListDetails = __importStar(require("./update-ip-address-list-details"));
exports.UpdateIpAddressListDetails = UpdateIpAddressListDetails.UpdateIpAddressListDetails;
const UpdateNatV4RuleDetails = __importStar(require("./update-nat-v4-rule-details"));
exports.UpdateNatV4RuleDetails = UpdateNatV4RuleDetails.UpdateNatV4RuleDetails;
const UpdateSslForwardProxyProfileDetails = __importStar(require("./update-ssl-forward-proxy-profile-details"));
exports.UpdateSslForwardProxyProfileDetails = UpdateSslForwardProxyProfileDetails.UpdateSslForwardProxyProfileDetails;
const UpdateSslInboundInspectionProfileDetails = __importStar(require("./update-ssl-inbound-inspection-profile-details"));
exports.UpdateSslInboundInspectionProfileDetails = UpdateSslInboundInspectionProfileDetails.UpdateSslInboundInspectionProfileDetails;
const UpdateTcpServiceDetails = __importStar(require("./update-tcp-service-details"));
exports.UpdateTcpServiceDetails = UpdateTcpServiceDetails.UpdateTcpServiceDetails;
const UpdateUdpServiceDetails = __importStar(require("./update-udp-service-details"));
exports.UpdateUdpServiceDetails = UpdateUdpServiceDetails.UpdateUdpServiceDetails;
const UpdateVaultMappedSecretDetails = __importStar(require("./update-vault-mapped-secret-details"));
exports.UpdateVaultMappedSecretDetails = UpdateVaultMappedSecretDetails.UpdateVaultMappedSecretDetails;
const UpdateVxlanInspectionRuleDetails = __importStar(require("./update-vxlan-inspection-rule-details"));
exports.UpdateVxlanInspectionRuleDetails = UpdateVxlanInspectionRuleDetails.UpdateVxlanInspectionRuleDetails;
const VaultMappedSecret = __importStar(require("./vault-mapped-secret"));
exports.VaultMappedSecret = VaultMappedSecret.VaultMappedSecret;
const VxlanInspectionRule = __importStar(require("./vxlan-inspection-rule"));
exports.VxlanInspectionRule = VxlanInspectionRule.VxlanInspectionRule;
const VxlanInspectionRuleSummary = __importStar(require("./vxlan-inspection-rule-summary"));
exports.VxlanInspectionRuleSummary = VxlanInspectionRuleSummary.VxlanInspectionRuleSummary;
