// oci-networkfirewall/lib/model/inspect-action-type.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20230501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InspectActionType = void 0;
/**
 * Types of Inspect Action on the traffic flow.
 * <p>
 * INSPECT - Inspect the traffic.
 *   * INSPECT_AND_CAPTURE_LOG - Inspect and capture logs for the traffic.
 *
 **/
var InspectActionType;
(function (InspectActionType) {
    InspectActionType["Inspect"] = "INSPECT";
    InspectActionType["InspectAndCaptureLog"] = "INSPECT_AND_CAPTURE_LOG";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    InspectActionType["UnknownValue"] = "UNKNOWN_VALUE";
})(InspectActionType = exports.InspectActionType || (exports.InspectActionType = {}));
(function (InspectActionType) {
    function getJsonObj(obj) {
        return obj;
    }
    InspectActionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    InspectActionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(InspectActionType = exports.InspectActionType || (exports.InspectActionType = {}));
