// oci-networkfirewall/lib/model/inspection-type.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20230501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InspectionType = void 0;
/**
 * Type of the secrets mapped based on the policy.
 * <p>
 * {@code SSL_INBOUND_INSPECTION}: For Inbound inspection of SSL traffic.
 *   * {@code SSL_FORWARD_PROXY}: For forward proxy certificates for SSL inspection.
 *
 **/
var InspectionType;
(function (InspectionType) {
    InspectionType["SslInboundInspection"] = "SSL_INBOUND_INSPECTION";
    InspectionType["SslForwardProxy"] = "SSL_FORWARD_PROXY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    InspectionType["UnknownValue"] = "UNKNOWN_VALUE";
})(InspectionType = exports.InspectionType || (exports.InspectionType = {}));
(function (InspectionType) {
    function getJsonObj(obj) {
        return obj;
    }
    InspectionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    InspectionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(InspectionType = exports.InspectionType || (exports.InspectionType = {}));
