// oci-networkfirewall/lib/model/mapped-secret-type.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20230501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MappedSecretType = void 0;
/**
 * Types of Mapped Secrets based on the source of data for Mapped Secret.
 * * OCI_VAULT : Mapped Secrets with using data from OCI Vaults.
 *
 **/
var MappedSecretType;
(function (MappedSecretType) {
    MappedSecretType["OciVault"] = "OCI_VAULT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MappedSecretType["UnknownValue"] = "UNKNOWN_VALUE";
})(MappedSecretType = exports.MappedSecretType || (exports.MappedSecretType = {}));
(function (MappedSecretType) {
    function getJsonObj(obj) {
        return obj;
    }
    MappedSecretType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MappedSecretType.getDeserializedJsonObj = getDeserializedJsonObj;
})(MappedSecretType = exports.MappedSecretType || (exports.MappedSecretType = {}));
