// oci-networkfirewall/lib/model/nat-v4-action-type.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20230501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NatV4ActionType = void 0;
/**
 * Types of NAT action support for NATv4 type NAT.
 * <p>
 * DIPP_SRC_NAT - Dynamic-ip-port source NAT.
 *
 **/
var NatV4ActionType;
(function (NatV4ActionType) {
    NatV4ActionType["DippSrcNat"] = "DIPP_SRC_NAT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NatV4ActionType["UnknownValue"] = "UNKNOWN_VALUE";
})(NatV4ActionType = exports.NatV4ActionType || (exports.NatV4ActionType = {}));
(function (NatV4ActionType) {
    function getJsonObj(obj) {
        return obj;
    }
    NatV4ActionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    NatV4ActionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(NatV4ActionType = exports.NatV4ActionType || (exports.NatV4ActionType = {}));
