// oci-networkfirewall/lib/model/nat-type.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20230501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NatType = void 0;
/**
 * Action:
 * * NATV4 - IPv4 type NAT.
 *
 **/
var NatType;
(function (NatType) {
    NatType["Natv4"] = "NATV4";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NatType["UnknownValue"] = "UNKNOWN_VALUE";
})(NatType = exports.NatType || (exports.NatType = {}));
(function (NatType) {
    function getJsonObj(obj) {
        return obj;
    }
    NatType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    NatType.getDeserializedJsonObj = getDeserializedJsonObj;
})(NatType = exports.NatType || (exports.NatType = {}));
