// oci-networkfirewall/lib/model/traffic-action-type.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20230501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrafficActionType = void 0;
/**
 * Types of Action on the Traffic flow.
 * <p>
 * ALLOW - Allows the traffic.
 *   * DROP - Silently drops the traffic, e.g. without sending a TCP reset.
 *   * REJECT - Rejects the traffic, sending a TCP reset to client and/or server as applicable.
 *   * INSPECT - Inspects traffic for vulnerability as specified in {@code inspection}, which may result in rejection.
 *
 **/
var TrafficActionType;
(function (TrafficActionType) {
    TrafficActionType["Allow"] = "ALLOW";
    TrafficActionType["Drop"] = "DROP";
    TrafficActionType["Reject"] = "REJECT";
    TrafficActionType["Inspect"] = "INSPECT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TrafficActionType["UnknownValue"] = "UNKNOWN_VALUE";
})(TrafficActionType = exports.TrafficActionType || (exports.TrafficActionType = {}));
(function (TrafficActionType) {
    function getJsonObj(obj) {
        return obj;
    }
    TrafficActionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TrafficActionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(TrafficActionType = exports.TrafficActionType || (exports.TrafficActionType = {}));
