// oci-networkfirewall/lib/model/traffic-inspection-type.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20230501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrafficInspectionType = void 0;
/**
 * Type of inspection to affect the Traffic flow. This is only applicable if action is INSPECT.
 * <p>
 * INTRUSION_DETECTION - Intrusion Detection.
 *   * INTRUSION_PREVENTION - Intrusion Detection and Prevention. Traffic classified as potentially malicious will be rejected as described in {@code type}.
 *
 **/
var TrafficInspectionType;
(function (TrafficInspectionType) {
    TrafficInspectionType["IntrusionDetection"] = "INTRUSION_DETECTION";
    TrafficInspectionType["IntrusionPrevention"] = "INTRUSION_PREVENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TrafficInspectionType["UnknownValue"] = "UNKNOWN_VALUE";
})(TrafficInspectionType = exports.TrafficInspectionType || (exports.TrafficInspectionType = {}));
(function (TrafficInspectionType) {
    function getJsonObj(obj) {
        return obj;
    }
    TrafficInspectionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TrafficInspectionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(TrafficInspectionType = exports.TrafficInspectionType || (exports.TrafficInspectionType = {}));
