// oci-adm/lib/model/config-severity.js
"use strict";
/**
 * Application Dependency Management API
 * Use the Application Dependency Management API to create knowledge bases and vulnerability audits.  For more information, see [ADM](https://docs.oracle.com/iaas/Content/application-dependency-management/home.htm).
 * OpenAPI spec version: 20220421
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigSeverity = void 0;
/**
 * Qualitative evaluation of the severity for a vulnerability, from NONE to CRITICAL. Used to configure requests that specify severity requirements, that's why UNSET is also a valid value (unlike in VulnerabilitySeverity).
 **/
var ConfigSeverity;
(function (ConfigSeverity) {
    ConfigSeverity["Unset"] = "UNSET";
    ConfigSeverity["None"] = "NONE";
    ConfigSeverity["Low"] = "LOW";
    ConfigSeverity["Medium"] = "MEDIUM";
    ConfigSeverity["High"] = "HIGH";
    ConfigSeverity["Critical"] = "CRITICAL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ConfigSeverity["UnknownValue"] = "UNKNOWN_VALUE";
})(ConfigSeverity = exports.ConfigSeverity || (exports.ConfigSeverity = {}));
(function (ConfigSeverity) {
    function getJsonObj(obj) {
        return obj;
    }
    ConfigSeverity.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ConfigSeverity.getDeserializedJsonObj = getDeserializedJsonObj;
})(ConfigSeverity = exports.ConfigSeverity || (exports.ConfigSeverity = {}));
