// oci-adm/lib/model/detect-configuration.js
"use strict";
/**
 * Application Dependency Management API
 * Use the Application Dependency Management API to create knowledge bases and vulnerability audits.  For more information, see [ADM](https://docs.oracle.com/iaas/Content/application-dependency-management/home.htm).
 * OpenAPI spec version: 20220421
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DetectConfiguration = void 0;
var DetectConfiguration;
(function (DetectConfiguration) {
    let UpgradePolicy;
    (function (UpgradePolicy) {
        UpgradePolicy["Nearest"] = "NEAREST";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        UpgradePolicy["UnknownValue"] = "UNKNOWN_VALUE";
    })(UpgradePolicy = DetectConfiguration.UpgradePolicy || (DetectConfiguration.UpgradePolicy = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DetectConfiguration.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DetectConfiguration.getDeserializedJsonObj = getDeserializedJsonObj;
})(DetectConfiguration = exports.DetectConfiguration || (exports.DetectConfiguration = {}));
