// oci-adm/lib/model/index.js
"use strict";
/**
 * Application Dependency Management API
 * Use the Application Dependency Management API to create knowledge bases and vulnerability audits.  For more information, see [ADM](https://docs.oracle.com/iaas/Content/application-dependency-management/home.htm).
 * OpenAPI spec version: 20220421
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VulnerabilitySeverity = exports.VulnerabilityAuditSummary = exports.VulnerabilityAuditSource = exports.VulnerabilityAuditConfiguration = exports.VulnerabilityAuditCollection = exports.VulnerabilityAudit = exports.Vulnerability = exports.VerifyConfiguration = exports.UsageDataDetails = exports.UpdateVulnerabilityAuditDetails = exports.UpdateRemediationRunDetails = exports.UpdateRemediationRecipeDetails = exports.UpdateKnowledgeBaseDetails = exports.StageSummary = exports.SortOrder = exports.ScmConfiguration = exports.RemediationRunSummary = exports.RemediationRunStageType = exports.RemediationRunStageSummary = exports.RemediationRunStageCollection = exports.RemediationRunStage = exports.RemediationRunCollection = exports.RemediationRun = exports.RemediationRecipeSummary = exports.RemediationRecipeCollection = exports.RemediationRecipe = exports.PullRequestProperties = exports.PipelineProperties = exports.OperationType = exports.OperationStatus = exports.NetworkConfiguration = exports.KnowledgeBaseSummary = exports.KnowledgeBaseCollection = exports.KnowledgeBase = exports.DetectConfiguration = exports.CreateVulnerabilityAuditDetails = exports.CreateRemediationRunDetails = exports.CreateRemediationRecipeDetails = exports.CreateKnowledgeBaseDetails = exports.ConfigSeverity = exports.ChangeVulnerabilityAuditCompartmentDetails = exports.ChangeRemediationRunCompartmentDetails = exports.ChangeRemediationRecipeCompartmentDetails = exports.ChangeKnowledgeBaseCompartmentDetails = exports.ApplicationDependencyVulnerabilitySummary = exports.ApplicationDependencyVulnerabilityCollection = exports.ApplicationDependencyRecommendationSummary = exports.ApplicationDependencyRecommendationCollection = exports.ApplicationDependency = exports.ActionType = void 0;
exports.VerifyStage = exports.UsageDataViaObjectStorageTupleDetails = exports.UnknownSourceVulnerabilityAuditSource = exports.RecommendStage = exports.OciResourceVulnerabilityAuditSource = exports.OciDevOpsBuildConfiguration = exports.OciCodeRepositoryConfiguration = exports.NoneVerifyConfiguration = exports.JenkinsPipelineConfiguration = exports.GitLabPipelineConfiguration = exports.GitHubActionsConfiguration = exports.ExternalScmConfiguration = exports.ExternalResourceVulnerabilityAuditSource = exports.DetectStage = exports.ApplyStage = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ApplicationDependency = __importStar(require("./application-dependency"));
exports.ApplicationDependency = ApplicationDependency.ApplicationDependency;
const ApplicationDependencyRecommendationCollection = __importStar(require("./application-dependency-recommendation-collection"));
exports.ApplicationDependencyRecommendationCollection = ApplicationDependencyRecommendationCollection.ApplicationDependencyRecommendationCollection;
const ApplicationDependencyRecommendationSummary = __importStar(require("./application-dependency-recommendation-summary"));
exports.ApplicationDependencyRecommendationSummary = ApplicationDependencyRecommendationSummary.ApplicationDependencyRecommendationSummary;
const ApplicationDependencyVulnerabilityCollection = __importStar(require("./application-dependency-vulnerability-collection"));
exports.ApplicationDependencyVulnerabilityCollection = ApplicationDependencyVulnerabilityCollection.ApplicationDependencyVulnerabilityCollection;
const ApplicationDependencyVulnerabilitySummary = __importStar(require("./application-dependency-vulnerability-summary"));
exports.ApplicationDependencyVulnerabilitySummary = ApplicationDependencyVulnerabilitySummary.ApplicationDependencyVulnerabilitySummary;
const ChangeKnowledgeBaseCompartmentDetails = __importStar(require("./change-knowledge-base-compartment-details"));
exports.ChangeKnowledgeBaseCompartmentDetails = ChangeKnowledgeBaseCompartmentDetails.ChangeKnowledgeBaseCompartmentDetails;
const ChangeRemediationRecipeCompartmentDetails = __importStar(require("./change-remediation-recipe-compartment-details"));
exports.ChangeRemediationRecipeCompartmentDetails = ChangeRemediationRecipeCompartmentDetails.ChangeRemediationRecipeCompartmentDetails;
const ChangeRemediationRunCompartmentDetails = __importStar(require("./change-remediation-run-compartment-details"));
exports.ChangeRemediationRunCompartmentDetails = ChangeRemediationRunCompartmentDetails.ChangeRemediationRunCompartmentDetails;
const ChangeVulnerabilityAuditCompartmentDetails = __importStar(require("./change-vulnerability-audit-compartment-details"));
exports.ChangeVulnerabilityAuditCompartmentDetails = ChangeVulnerabilityAuditCompartmentDetails.ChangeVulnerabilityAuditCompartmentDetails;
const ConfigSeverity = __importStar(require("./config-severity"));
exports.ConfigSeverity = ConfigSeverity.ConfigSeverity;
const CreateKnowledgeBaseDetails = __importStar(require("./create-knowledge-base-details"));
exports.CreateKnowledgeBaseDetails = CreateKnowledgeBaseDetails.CreateKnowledgeBaseDetails;
const CreateRemediationRecipeDetails = __importStar(require("./create-remediation-recipe-details"));
exports.CreateRemediationRecipeDetails = CreateRemediationRecipeDetails.CreateRemediationRecipeDetails;
const CreateRemediationRunDetails = __importStar(require("./create-remediation-run-details"));
exports.CreateRemediationRunDetails = CreateRemediationRunDetails.CreateRemediationRunDetails;
const CreateVulnerabilityAuditDetails = __importStar(require("./create-vulnerability-audit-details"));
exports.CreateVulnerabilityAuditDetails = CreateVulnerabilityAuditDetails.CreateVulnerabilityAuditDetails;
const DetectConfiguration = __importStar(require("./detect-configuration"));
exports.DetectConfiguration = DetectConfiguration.DetectConfiguration;
const KnowledgeBase = __importStar(require("./knowledge-base"));
exports.KnowledgeBase = KnowledgeBase.KnowledgeBase;
const KnowledgeBaseCollection = __importStar(require("./knowledge-base-collection"));
exports.KnowledgeBaseCollection = KnowledgeBaseCollection.KnowledgeBaseCollection;
const KnowledgeBaseSummary = __importStar(require("./knowledge-base-summary"));
exports.KnowledgeBaseSummary = KnowledgeBaseSummary.KnowledgeBaseSummary;
const NetworkConfiguration = __importStar(require("./network-configuration"));
exports.NetworkConfiguration = NetworkConfiguration.NetworkConfiguration;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PipelineProperties = __importStar(require("./pipeline-properties"));
exports.PipelineProperties = PipelineProperties.PipelineProperties;
const PullRequestProperties = __importStar(require("./pull-request-properties"));
exports.PullRequestProperties = PullRequestProperties.PullRequestProperties;
const RemediationRecipe = __importStar(require("./remediation-recipe"));
exports.RemediationRecipe = RemediationRecipe.RemediationRecipe;
const RemediationRecipeCollection = __importStar(require("./remediation-recipe-collection"));
exports.RemediationRecipeCollection = RemediationRecipeCollection.RemediationRecipeCollection;
const RemediationRecipeSummary = __importStar(require("./remediation-recipe-summary"));
exports.RemediationRecipeSummary = RemediationRecipeSummary.RemediationRecipeSummary;
const RemediationRun = __importStar(require("./remediation-run"));
exports.RemediationRun = RemediationRun.RemediationRun;
const RemediationRunCollection = __importStar(require("./remediation-run-collection"));
exports.RemediationRunCollection = RemediationRunCollection.RemediationRunCollection;
const RemediationRunStage = __importStar(require("./remediation-run-stage"));
exports.RemediationRunStage = RemediationRunStage.RemediationRunStage;
const RemediationRunStageCollection = __importStar(require("./remediation-run-stage-collection"));
exports.RemediationRunStageCollection = RemediationRunStageCollection.RemediationRunStageCollection;
const RemediationRunStageSummary = __importStar(require("./remediation-run-stage-summary"));
exports.RemediationRunStageSummary = RemediationRunStageSummary.RemediationRunStageSummary;
const RemediationRunStageType = __importStar(require("./remediation-run-stage-type"));
exports.RemediationRunStageType = RemediationRunStageType.RemediationRunStageType;
const RemediationRunSummary = __importStar(require("./remediation-run-summary"));
exports.RemediationRunSummary = RemediationRunSummary.RemediationRunSummary;
const ScmConfiguration = __importStar(require("./scm-configuration"));
exports.ScmConfiguration = ScmConfiguration.ScmConfiguration;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const StageSummary = __importStar(require("./stage-summary"));
exports.StageSummary = StageSummary.StageSummary;
const UpdateKnowledgeBaseDetails = __importStar(require("./update-knowledge-base-details"));
exports.UpdateKnowledgeBaseDetails = UpdateKnowledgeBaseDetails.UpdateKnowledgeBaseDetails;
const UpdateRemediationRecipeDetails = __importStar(require("./update-remediation-recipe-details"));
exports.UpdateRemediationRecipeDetails = UpdateRemediationRecipeDetails.UpdateRemediationRecipeDetails;
const UpdateRemediationRunDetails = __importStar(require("./update-remediation-run-details"));
exports.UpdateRemediationRunDetails = UpdateRemediationRunDetails.UpdateRemediationRunDetails;
const UpdateVulnerabilityAuditDetails = __importStar(require("./update-vulnerability-audit-details"));
exports.UpdateVulnerabilityAuditDetails = UpdateVulnerabilityAuditDetails.UpdateVulnerabilityAuditDetails;
const UsageDataDetails = __importStar(require("./usage-data-details"));
exports.UsageDataDetails = UsageDataDetails.UsageDataDetails;
const VerifyConfiguration = __importStar(require("./verify-configuration"));
exports.VerifyConfiguration = VerifyConfiguration.VerifyConfiguration;
const Vulnerability = __importStar(require("./vulnerability"));
exports.Vulnerability = Vulnerability.Vulnerability;
const VulnerabilityAudit = __importStar(require("./vulnerability-audit"));
exports.VulnerabilityAudit = VulnerabilityAudit.VulnerabilityAudit;
const VulnerabilityAuditCollection = __importStar(require("./vulnerability-audit-collection"));
exports.VulnerabilityAuditCollection = VulnerabilityAuditCollection.VulnerabilityAuditCollection;
const VulnerabilityAuditConfiguration = __importStar(require("./vulnerability-audit-configuration"));
exports.VulnerabilityAuditConfiguration = VulnerabilityAuditConfiguration.VulnerabilityAuditConfiguration;
const VulnerabilityAuditSource = __importStar(require("./vulnerability-audit-source"));
exports.VulnerabilityAuditSource = VulnerabilityAuditSource.VulnerabilityAuditSource;
const VulnerabilityAuditSummary = __importStar(require("./vulnerability-audit-summary"));
exports.VulnerabilityAuditSummary = VulnerabilityAuditSummary.VulnerabilityAuditSummary;
const VulnerabilitySeverity = __importStar(require("./vulnerability-severity"));
exports.VulnerabilitySeverity = VulnerabilitySeverity.VulnerabilitySeverity;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const ApplyStage = __importStar(require("./apply-stage"));
exports.ApplyStage = ApplyStage.ApplyStage;
const DetectStage = __importStar(require("./detect-stage"));
exports.DetectStage = DetectStage.DetectStage;
const ExternalResourceVulnerabilityAuditSource = __importStar(require("./external-resource-vulnerability-audit-source"));
exports.ExternalResourceVulnerabilityAuditSource = ExternalResourceVulnerabilityAuditSource.ExternalResourceVulnerabilityAuditSource;
const ExternalScmConfiguration = __importStar(require("./external-scm-configuration"));
exports.ExternalScmConfiguration = ExternalScmConfiguration.ExternalScmConfiguration;
const GitHubActionsConfiguration = __importStar(require("./git-hub-actions-configuration"));
exports.GitHubActionsConfiguration = GitHubActionsConfiguration.GitHubActionsConfiguration;
const GitLabPipelineConfiguration = __importStar(require("./git-lab-pipeline-configuration"));
exports.GitLabPipelineConfiguration = GitLabPipelineConfiguration.GitLabPipelineConfiguration;
const JenkinsPipelineConfiguration = __importStar(require("./jenkins-pipeline-configuration"));
exports.JenkinsPipelineConfiguration = JenkinsPipelineConfiguration.JenkinsPipelineConfiguration;
const NoneVerifyConfiguration = __importStar(require("./none-verify-configuration"));
exports.NoneVerifyConfiguration = NoneVerifyConfiguration.NoneVerifyConfiguration;
const OciCodeRepositoryConfiguration = __importStar(require("./oci-code-repository-configuration"));
exports.OciCodeRepositoryConfiguration = OciCodeRepositoryConfiguration.OciCodeRepositoryConfiguration;
const OciDevOpsBuildConfiguration = __importStar(require("./oci-dev-ops-build-configuration"));
exports.OciDevOpsBuildConfiguration = OciDevOpsBuildConfiguration.OciDevOpsBuildConfiguration;
const OciResourceVulnerabilityAuditSource = __importStar(require("./oci-resource-vulnerability-audit-source"));
exports.OciResourceVulnerabilityAuditSource = OciResourceVulnerabilityAuditSource.OciResourceVulnerabilityAuditSource;
const RecommendStage = __importStar(require("./recommend-stage"));
exports.RecommendStage = RecommendStage.RecommendStage;
const UnknownSourceVulnerabilityAuditSource = __importStar(require("./unknown-source-vulnerability-audit-source"));
exports.UnknownSourceVulnerabilityAuditSource = UnknownSourceVulnerabilityAuditSource.UnknownSourceVulnerabilityAuditSource;
const UsageDataViaObjectStorageTupleDetails = __importStar(require("./usage-data-via-object-storage-tuple-details"));
exports.UsageDataViaObjectStorageTupleDetails = UsageDataViaObjectStorageTupleDetails.UsageDataViaObjectStorageTupleDetails;
const VerifyStage = __importStar(require("./verify-stage"));
exports.VerifyStage = VerifyStage.VerifyStage;
