// oci-adm/lib/model/remediation-run-stage.js
"use strict";
/**
 * Application Dependency Management API
 * Use the Application Dependency Management API to create knowledge bases and vulnerability audits.  For more information, see [ADM](https://docs.oracle.com/iaas/Content/application-dependency-management/home.htm).
 * OpenAPI spec version: 20220421
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemediationRunStage = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var RemediationRunStage;
(function (RemediationRunStage) {
    let Status;
    (function (Status) {
        Status["Created"] = "CREATED";
        Status["InProgress"] = "IN_PROGRESS";
        Status["Succeeded"] = "SUCCEEDED";
        Status["Failed"] = "FAILED";
        Status["Canceling"] = "CANCELING";
        Status["Canceled"] = "CANCELED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = RemediationRunStage.Status || (RemediationRunStage.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "RECOMMEND":
                    return model.RecommendStage.getJsonObj(jsonObj, true);
                case "VERIFY":
                    return model.VerifyStage.getJsonObj(jsonObj, true);
                case "APPLY":
                    return model.ApplyStage.getJsonObj(jsonObj, true);
                case "DETECT":
                    return model.DetectStage.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    RemediationRunStage.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "RECOMMEND":
                    return model.RecommendStage.getDeserializedJsonObj(jsonObj, true);
                case "VERIFY":
                    return model.VerifyStage.getDeserializedJsonObj(jsonObj, true);
                case "APPLY":
                    return model.ApplyStage.getDeserializedJsonObj(jsonObj, true);
                case "DETECT":
                    return model.DetectStage.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    RemediationRunStage.getDeserializedJsonObj = getDeserializedJsonObj;
})(RemediationRunStage = exports.RemediationRunStage || (exports.RemediationRunStage = {}));
