// oci-adm/lib/model/remediation-run.js
"use strict";
/**
 * Application Dependency Management API
 * Use the Application Dependency Management API to create knowledge bases and vulnerability audits.  For more information, see [ADM](https://docs.oracle.com/iaas/Content/application-dependency-management/home.htm).
 * OpenAPI spec version: 20220421
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemediationRun = void 0;
const model = __importStar(require("../model"));
var RemediationRun;
(function (RemediationRun) {
    let RemediationRunSource;
    (function (RemediationRunSource) {
        RemediationRunSource["Manual"] = "MANUAL";
        RemediationRunSource["KnowledgeBase"] = "KNOWLEDGE_BASE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        RemediationRunSource["UnknownValue"] = "UNKNOWN_VALUE";
    })(RemediationRunSource = RemediationRun.RemediationRunSource || (RemediationRun.RemediationRunSource = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Accepted"] = "ACCEPTED";
        LifecycleState["Canceled"] = "CANCELED";
        LifecycleState["Canceling"] = "CANCELING";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        LifecycleState["Succeeded"] = "SUCCEEDED";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = RemediationRun.LifecycleState || (RemediationRun.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "stages": obj.stages
                ? obj.stages.map(item => {
                    return model.StageSummary.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    RemediationRun.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "stages": obj.stages
                ? obj.stages.map(item => {
                    return model.StageSummary.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    RemediationRun.getDeserializedJsonObj = getDeserializedJsonObj;
})(RemediationRun = exports.RemediationRun || (exports.RemediationRun = {}));
