// oci-adm/lib/model/remediation-run-stage-type.js
"use strict";
/**
 * Application Dependency Management API
 * Use the Application Dependency Management API to create knowledge bases and vulnerability audits.  For more information, see [ADM](https://docs.oracle.com/iaas/Content/application-dependency-management/home.htm).
 * OpenAPI spec version: 20220421
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemediationRunStageType = void 0;
/**
 * Remediation Run stage type.
 **/
var RemediationRunStageType;
(function (RemediationRunStageType) {
    RemediationRunStageType["Detect"] = "DETECT";
    RemediationRunStageType["Recommend"] = "RECOMMEND";
    RemediationRunStageType["Verify"] = "VERIFY";
    RemediationRunStageType["Apply"] = "APPLY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RemediationRunStageType["UnknownValue"] = "UNKNOWN_VALUE";
})(RemediationRunStageType = exports.RemediationRunStageType || (exports.RemediationRunStageType = {}));
(function (RemediationRunStageType) {
    function getJsonObj(obj) {
        return obj;
    }
    RemediationRunStageType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RemediationRunStageType.getDeserializedJsonObj = getDeserializedJsonObj;
})(RemediationRunStageType = exports.RemediationRunStageType || (exports.RemediationRunStageType = {}));
