// oci-ospgateway/lib/model/address-quality-type.js
"use strict";
/**
 * OSP Gateway API
 * This site describes all the Rest endpoints of OSP Gateway.
 * OpenAPI spec version: 20191001
 * Contact: osp_team_oci_cam_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddressQualityType = void 0;
/**
 * Address quality type
 **/
var AddressQualityType;
(function (AddressQualityType) {
    AddressQualityType["Excellent"] = "EXCELLENT";
    AddressQualityType["Good"] = "GOOD";
    AddressQualityType["Average"] = "AVERAGE";
    AddressQualityType["Poor"] = "POOR";
    AddressQualityType["Bad"] = "BAD";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AddressQualityType["UnknownValue"] = "UNKNOWN_VALUE";
})(AddressQualityType = exports.AddressQualityType || (exports.AddressQualityType = {}));
(function (AddressQualityType) {
    function getJsonObj(obj) {
        return obj;
    }
    AddressQualityType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AddressQualityType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AddressQualityType = exports.AddressQualityType || (exports.AddressQualityType = {}));
