// oci-ospgateway/lib/model/echeck-payment-detail.js
"use strict";
/**
 * OSP Gateway API
 * This site describes all the Rest endpoints of OSP Gateway.
 * OpenAPI spec version: 20191001
 * Contact: osp_team_oci_cam_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EcheckPaymentDetail = void 0;
const model = __importStar(require("../model"));
var EcheckPaymentDetail;
(function (EcheckPaymentDetail) {
    let CardType;
    (function (CardType) {
        CardType["Saving"] = "SAVING";
        CardType["Checking"] = "CHECKING";
        CardType["CorporateChecking"] = "CORPORATE_CHECKING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        CardType["UnknownValue"] = "UNKNOWN_VALUE";
    })(CardType = EcheckPaymentDetail.CardType || (EcheckPaymentDetail.CardType = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.PaymentDetail.getJsonObj(obj))), {});
        return jsonObj;
    }
    EcheckPaymentDetail.getJsonObj = getJsonObj;
    EcheckPaymentDetail.paymentMethod = "ECHECK";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.PaymentDetail.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    EcheckPaymentDetail.getDeserializedJsonObj = getDeserializedJsonObj;
})(EcheckPaymentDetail = exports.EcheckPaymentDetail || (exports.EcheckPaymentDetail = {}));
