// oci-generativeai/lib/model/dedicated-ai-cluster.js
"use strict";
/**
 * Generative AI Service Management API
 * OCI Generative AI is a fully managed service that provides a set of state-of-the-art, customizable large language models (LLMs) that cover a wide range of use cases for text generation, summarization, and text embeddings.

Use the Generative AI service management API to create and manage {@link DedicatedAiCluster}, {@link Endpoint}, {@link Model}, and {@link WorkRequest} in the Generative AI service. For example, create a custom model by fine-tuning an out-of-the-box model using your own data, on a fine-tuning dedicated AI cluster. Then, create a hosting dedicated AI cluster with an endpoint to host your custom model.

To access your custom model endpoints, or to try the out-of-the-box models to generate text, summarize, and create text embeddings see the [Generative AI Inference API](https://docs.oracle.com/iaas/api/#/en/generative-ai-inference/latest/).

To learn more about the service, see the [Generative AI documentation](https://docs.oracle.com/iaas/Content/generative-ai/home.htm).

 * OpenAPI spec version: 20231130
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DedicatedAiCluster = void 0;
const model = __importStar(require("../model"));
var DedicatedAiCluster;
(function (DedicatedAiCluster) {
    let Type;
    (function (Type) {
        Type["Hosting"] = "HOSTING";
        Type["FineTuning"] = "FINE_TUNING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = DedicatedAiCluster.Type || (DedicatedAiCluster.Type = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = DedicatedAiCluster.LifecycleState || (DedicatedAiCluster.LifecycleState = {}));
    let UnitShape;
    (function (UnitShape) {
        UnitShape["LargeCohere"] = "LARGE_COHERE";
        UnitShape["LargeCohereV2"] = "LARGE_COHERE_V2";
        UnitShape["SmallCohere"] = "SMALL_COHERE";
        UnitShape["SmallCohereV2"] = "SMALL_COHERE_V2";
        UnitShape["SmallCohere4"] = "SMALL_COHERE_4";
        UnitShape["EmbedCohere"] = "EMBED_COHERE";
        UnitShape["Llama270"] = "LLAMA2_70";
        UnitShape["LargeGeneric"] = "LARGE_GENERIC";
        UnitShape["LargeCohereV22"] = "LARGE_COHERE_V2_2";
        UnitShape["LargeGeneric4"] = "LARGE_GENERIC_4";
        UnitShape["SmallGenericV2"] = "SMALL_GENERIC_V2";
        UnitShape["LargeGeneric2"] = "LARGE_GENERIC_2";
        UnitShape["LargeGenericV2"] = "LARGE_GENERIC_V2";
        UnitShape["LargeCohereV3"] = "LARGE_COHERE_V3";
        UnitShape["RerankCohere"] = "RERANK_COHERE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        UnitShape["UnknownValue"] = "UNKNOWN_VALUE";
    })(UnitShape = DedicatedAiCluster.UnitShape || (DedicatedAiCluster.UnitShape = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "capacity": obj.capacity
                ? model.DedicatedAiClusterCapacity.getJsonObj(obj.capacity)
                : undefined,
            "previousState": obj.previousState
                ? model.DedicatedAiCluster.getJsonObj(obj.previousState)
                : undefined
        });
        return jsonObj;
    }
    DedicatedAiCluster.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "capacity": obj.capacity
                ? model.DedicatedAiClusterCapacity.getDeserializedJsonObj(obj.capacity)
                : undefined,
            "previousState": obj.previousState
                ? model.DedicatedAiCluster.getDeserializedJsonObj(obj.previousState)
                : undefined
        });
        return jsonObj;
    }
    DedicatedAiCluster.getDeserializedJsonObj = getDeserializedJsonObj;
})(DedicatedAiCluster = exports.DedicatedAiCluster || (exports.DedicatedAiCluster = {}));
