// oci-generativeai/lib/model/operation-type.js
"use strict";
/**
 * Generative AI Service Management API
 * OCI Generative AI is a fully managed service that provides a set of state-of-the-art, customizable large language models (LLMs) that cover a wide range of use cases for text generation, summarization, and text embeddings.

Use the Generative AI service management API to create and manage {@link DedicatedAiCluster}, {@link Endpoint}, {@link Model}, and {@link WorkRequest} in the Generative AI service. For example, create a custom model by fine-tuning an out-of-the-box model using your own data, on a fine-tuning dedicated AI cluster. Then, create a hosting dedicated AI cluster with an endpoint to host your custom model.

To access your custom model endpoints, or to try the out-of-the-box models to generate text, summarize, and create text embeddings see the [Generative AI Inference API](https://docs.oracle.com/iaas/api/#/en/generative-ai-inference/latest/).

To learn more about the service, see the [Generative AI documentation](https://docs.oracle.com/iaas/Content/generative-ai/home.htm).

 * OpenAPI spec version: 20231130
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * The resources affected by this work request.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateModel"] = "CREATE_MODEL";
    OperationType["DeleteModel"] = "DELETE_MODEL";
    OperationType["MoveModel"] = "MOVE_MODEL";
    OperationType["CreateDedicatedAiCluster"] = "CREATE_DEDICATED_AI_CLUSTER";
    OperationType["DeleteDedicatedAiCluster"] = "DELETE_DEDICATED_AI_CLUSTER";
    OperationType["UpdateDedicatedAiCluster"] = "UPDATE_DEDICATED_AI_CLUSTER";
    OperationType["MoveDedicatedAiCluster"] = "MOVE_DEDICATED_AI_CLUSTER";
    OperationType["CreateEndpoint"] = "CREATE_ENDPOINT";
    OperationType["DeleteEndpoint"] = "DELETE_ENDPOINT";
    OperationType["UpdateEndpoint"] = "UPDATE_ENDPOINT";
    OperationType["MoveEndpoint"] = "MOVE_ENDPOINT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
