// oci-dblm/lib/model/additional-patches.js
"use strict";
/**
 * Dblm API
 * A description of the Dblm API
 * OpenAPI spec version: 20240102
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdditionalPatches = void 0;
var AdditionalPatches;
(function (AdditionalPatches) {
    let Category;
    (function (Category) {
        Category["Recommended"] = "RECOMMENDED";
        Category["AdditionalPatch"] = "ADDITIONAL_PATCH";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Category["UnknownValue"] = "UNKNOWN_VALUE";
    })(Category = AdditionalPatches.Category || (AdditionalPatches.Category = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AdditionalPatches.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AdditionalPatches.getDeserializedJsonObj = getDeserializedJsonObj;
})(AdditionalPatches = exports.AdditionalPatches || (exports.AdditionalPatches = {}));
