// oci-dblm/lib/model/databases-summary.js
"use strict";
/**
 * Dblm API
 * A description of the Dblm API
 * OpenAPI spec version: 20240102
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabasesSummary = void 0;
const model = __importStar(require("../model"));
var DatabasesSummary;
(function (DatabasesSummary) {
    let DatabaseType;
    (function (DatabaseType) {
        DatabaseType["Si"] = "SI";
        DatabaseType["Rac"] = "RAC";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DatabaseType["UnknownValue"] = "UNKNOWN_VALUE";
    })(DatabaseType = DatabasesSummary.DatabaseType || (DatabasesSummary.DatabaseType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = DatabasesSummary.LifecycleState || (DatabasesSummary.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "imageDetails": obj.imageDetails
                ? model.ImageDetails.getJsonObj(obj.imageDetails)
                : undefined,
            "patchComplianceDetails": obj.patchComplianceDetails
                ? model.PatchComplianceDetails.getJsonObj(obj.patchComplianceDetails)
                : undefined,
            "patchActivityDetails": obj.patchActivityDetails
                ? model.PatchActivityDetails.getJsonObj(obj.patchActivityDetails)
                : undefined,
            "additionalPatches": obj.additionalPatches
                ? obj.additionalPatches.map(item => {
                    return model.AdditionalPatches.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DatabasesSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "imageDetails": obj.imageDetails
                ? model.ImageDetails.getDeserializedJsonObj(obj.imageDetails)
                : undefined,
            "patchComplianceDetails": obj.patchComplianceDetails
                ? model.PatchComplianceDetails.getDeserializedJsonObj(obj.patchComplianceDetails)
                : undefined,
            "patchActivityDetails": obj.patchActivityDetails
                ? model.PatchActivityDetails.getDeserializedJsonObj(obj.patchActivityDetails)
                : undefined,
            "additionalPatches": obj.additionalPatches
                ? obj.additionalPatches.map(item => {
                    return model.AdditionalPatches.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DatabasesSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabasesSummary = exports.DatabasesSummary || (exports.DatabasesSummary = {}));
