// oci-dblm/lib/model/image-details.js
"use strict";
/**
 * Dblm API
 * A description of the Dblm API
 * OpenAPI spec version: 20240102
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageDetails = void 0;
var ImageDetails;
(function (ImageDetails) {
    let ImageStatus;
    (function (ImageStatus) {
        ImageStatus["Green"] = "GREEN";
        ImageStatus["Yellow"] = "YELLOW";
        ImageStatus["Red"] = "RED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ImageStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(ImageStatus = ImageDetails.ImageStatus || (ImageDetails.ImageStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ImageDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ImageDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(ImageDetails = exports.ImageDetails || (exports.ImageDetails = {}));
