// oci-dblm/lib/model/index.js
"use strict";
/**
 * Dblm API
 * A description of the Dblm API
 * OpenAPI spec version: 20240102
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.VulnerabilitySummary = exports.VulnerabilityScanSummary = exports.VulnerabilityScanCollection = exports.VulnerabilityScan = exports.VulnerabilityRiskLevel = exports.VulnerabilityResourceSummary = exports.VulnerabilityResourceCollection = exports.VulnerabilityCollection = exports.VulnerabilitiesSummary = exports.SortOrder = exports.ResourcesSeverities = exports.ResourcesPatchComplianceSummary = exports.ResourceSummary = exports.ResourceInfo = exports.PatchRecommendationsDetailsSummary = exports.PatchRecommendationSummary = exports.PatchOperationsSummary = exports.PatchDatabasesCollection = exports.PatchComplianceDetails = exports.PatchActivityDetails = exports.OperationType = exports.OperationStatus = exports.NotificationSummary = exports.NotificationCollection = exports.MetricError = exports.ImagesPatchRecommendationSummary = exports.ImageDetails = exports.HostInfo = exports.FppServer = exports.DblmVulnerability = exports.DblmPatchManagement = exports.DatabasesSummary = exports.CreateVulnerabilityScanDetails = exports.ChildPdb = exports.AggregatedVulnerabilitySummary = exports.AggregatedVulnerabilityCollection = exports.AdditionalPatches = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AdditionalPatches = __importStar(require("./additional-patches"));
exports.AdditionalPatches = AdditionalPatches.AdditionalPatches;
const AggregatedVulnerabilityCollection = __importStar(require("./aggregated-vulnerability-collection"));
exports.AggregatedVulnerabilityCollection = AggregatedVulnerabilityCollection.AggregatedVulnerabilityCollection;
const AggregatedVulnerabilitySummary = __importStar(require("./aggregated-vulnerability-summary"));
exports.AggregatedVulnerabilitySummary = AggregatedVulnerabilitySummary.AggregatedVulnerabilitySummary;
const ChildPdb = __importStar(require("./child-pdb"));
exports.ChildPdb = ChildPdb.ChildPdb;
const CreateVulnerabilityScanDetails = __importStar(require("./create-vulnerability-scan-details"));
exports.CreateVulnerabilityScanDetails = CreateVulnerabilityScanDetails.CreateVulnerabilityScanDetails;
const DatabasesSummary = __importStar(require("./databases-summary"));
exports.DatabasesSummary = DatabasesSummary.DatabasesSummary;
const DblmPatchManagement = __importStar(require("./dblm-patch-management"));
exports.DblmPatchManagement = DblmPatchManagement.DblmPatchManagement;
const DblmVulnerability = __importStar(require("./dblm-vulnerability"));
exports.DblmVulnerability = DblmVulnerability.DblmVulnerability;
const FppServer = __importStar(require("./fpp-server"));
exports.FppServer = FppServer.FppServer;
const HostInfo = __importStar(require("./host-info"));
exports.HostInfo = HostInfo.HostInfo;
const ImageDetails = __importStar(require("./image-details"));
exports.ImageDetails = ImageDetails.ImageDetails;
const ImagesPatchRecommendationSummary = __importStar(require("./images-patch-recommendation-summary"));
exports.ImagesPatchRecommendationSummary = ImagesPatchRecommendationSummary.ImagesPatchRecommendationSummary;
const MetricError = __importStar(require("./metric-error"));
exports.MetricError = MetricError.MetricError;
const NotificationCollection = __importStar(require("./notification-collection"));
exports.NotificationCollection = NotificationCollection.NotificationCollection;
const NotificationSummary = __importStar(require("./notification-summary"));
exports.NotificationSummary = NotificationSummary.NotificationSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PatchActivityDetails = __importStar(require("./patch-activity-details"));
exports.PatchActivityDetails = PatchActivityDetails.PatchActivityDetails;
const PatchComplianceDetails = __importStar(require("./patch-compliance-details"));
exports.PatchComplianceDetails = PatchComplianceDetails.PatchComplianceDetails;
const PatchDatabasesCollection = __importStar(require("./patch-databases-collection"));
exports.PatchDatabasesCollection = PatchDatabasesCollection.PatchDatabasesCollection;
const PatchOperationsSummary = __importStar(require("./patch-operations-summary"));
exports.PatchOperationsSummary = PatchOperationsSummary.PatchOperationsSummary;
const PatchRecommendationSummary = __importStar(require("./patch-recommendation-summary"));
exports.PatchRecommendationSummary = PatchRecommendationSummary.PatchRecommendationSummary;
const PatchRecommendationsDetailsSummary = __importStar(require("./patch-recommendations-details-summary"));
exports.PatchRecommendationsDetailsSummary = PatchRecommendationsDetailsSummary.PatchRecommendationsDetailsSummary;
const ResourceInfo = __importStar(require("./resource-info"));
exports.ResourceInfo = ResourceInfo.ResourceInfo;
const ResourceSummary = __importStar(require("./resource-summary"));
exports.ResourceSummary = ResourceSummary.ResourceSummary;
const ResourcesPatchComplianceSummary = __importStar(require("./resources-patch-compliance-summary"));
exports.ResourcesPatchComplianceSummary = ResourcesPatchComplianceSummary.ResourcesPatchComplianceSummary;
const ResourcesSeverities = __importStar(require("./resources-severities"));
exports.ResourcesSeverities = ResourcesSeverities.ResourcesSeverities;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const VulnerabilitiesSummary = __importStar(require("./vulnerabilities-summary"));
exports.VulnerabilitiesSummary = VulnerabilitiesSummary.VulnerabilitiesSummary;
const VulnerabilityCollection = __importStar(require("./vulnerability-collection"));
exports.VulnerabilityCollection = VulnerabilityCollection.VulnerabilityCollection;
const VulnerabilityResourceCollection = __importStar(require("./vulnerability-resource-collection"));
exports.VulnerabilityResourceCollection = VulnerabilityResourceCollection.VulnerabilityResourceCollection;
const VulnerabilityResourceSummary = __importStar(require("./vulnerability-resource-summary"));
exports.VulnerabilityResourceSummary = VulnerabilityResourceSummary.VulnerabilityResourceSummary;
const VulnerabilityRiskLevel = __importStar(require("./vulnerability-risk-level"));
exports.VulnerabilityRiskLevel = VulnerabilityRiskLevel.VulnerabilityRiskLevel;
const VulnerabilityScan = __importStar(require("./vulnerability-scan"));
exports.VulnerabilityScan = VulnerabilityScan.VulnerabilityScan;
const VulnerabilityScanCollection = __importStar(require("./vulnerability-scan-collection"));
exports.VulnerabilityScanCollection = VulnerabilityScanCollection.VulnerabilityScanCollection;
const VulnerabilityScanSummary = __importStar(require("./vulnerability-scan-summary"));
exports.VulnerabilityScanSummary = VulnerabilityScanSummary.VulnerabilityScanSummary;
const VulnerabilitySummary = __importStar(require("./vulnerability-summary"));
exports.VulnerabilitySummary = VulnerabilitySummary.VulnerabilitySummary;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
