// oci-dblm/lib/model/patch-activity-details.js
"use strict";
/**
 * Dblm API
 * A description of the Dblm API
 * OpenAPI spec version: 20240102
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchActivityDetails = void 0;
var PatchActivityDetails;
(function (PatchActivityDetails) {
    let DeployStatus;
    (function (DeployStatus) {
        DeployStatus["Scheduled"] = "SCHEDULED";
        DeployStatus["Running"] = "RUNNING";
        DeployStatus["Completed"] = "COMPLETED";
        DeployStatus["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DeployStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(DeployStatus = PatchActivityDetails.DeployStatus || (PatchActivityDetails.DeployStatus = {}));
    let UpdateStatus;
    (function (UpdateStatus) {
        UpdateStatus["Scheduled"] = "SCHEDULED";
        UpdateStatus["Running"] = "RUNNING";
        UpdateStatus["Completed"] = "COMPLETED";
        UpdateStatus["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        UpdateStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(UpdateStatus = PatchActivityDetails.UpdateStatus || (PatchActivityDetails.UpdateStatus = {}));
    let MigrateListenerStatus;
    (function (MigrateListenerStatus) {
        MigrateListenerStatus["Scheduled"] = "SCHEDULED";
        MigrateListenerStatus["Running"] = "RUNNING";
        MigrateListenerStatus["Completed"] = "COMPLETED";
        MigrateListenerStatus["Failed"] = "FAILED";
        MigrateListenerStatus["Na"] = "NA";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MigrateListenerStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(MigrateListenerStatus = PatchActivityDetails.MigrateListenerStatus || (PatchActivityDetails.MigrateListenerStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PatchActivityDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PatchActivityDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(PatchActivityDetails = exports.PatchActivityDetails || (exports.PatchActivityDetails = {}));
