// oci-dblm/lib/model/resources-severities.js
"use strict";
/**
 * Dblm API
 * A description of the Dblm API
 * OpenAPI spec version: 20240102
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourcesSeverities = void 0;
/**
 * Applicable severity types are critical, high, medium, low, info and none.
 **/
var ResourcesSeverities;
(function (ResourcesSeverities) {
    ResourcesSeverities["Critical"] = "CRITICAL";
    ResourcesSeverities["High"] = "HIGH";
    ResourcesSeverities["Medium"] = "MEDIUM";
    ResourcesSeverities["Low"] = "LOW";
    ResourcesSeverities["Info"] = "INFO";
    ResourcesSeverities["None"] = "NONE";
})(ResourcesSeverities = exports.ResourcesSeverities || (exports.ResourcesSeverities = {}));
(function (ResourcesSeverities) {
    function getJsonObj(obj) {
        return obj;
    }
    ResourcesSeverities.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ResourcesSeverities.getDeserializedJsonObj = getDeserializedJsonObj;
})(ResourcesSeverities = exports.ResourcesSeverities || (exports.ResourcesSeverities = {}));
