// oci-database/lib/model/autonomous-container-database-dataguard.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](https://docs.oracle.com/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutonomousContainerDatabaseDataguard = void 0;
var AutonomousContainerDatabaseDataguard;
(function (AutonomousContainerDatabaseDataguard) {
    let Role;
    (function (Role) {
        Role["Primary"] = "PRIMARY";
        Role["Standby"] = "STANDBY";
        Role["DisabledStandby"] = "DISABLED_STANDBY";
        Role["BackupCopy"] = "BACKUP_COPY";
        Role["SnapshotStandby"] = "SNAPSHOT_STANDBY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Role["UnknownValue"] = "UNKNOWN_VALUE";
    })(Role = AutonomousContainerDatabaseDataguard.Role || (AutonomousContainerDatabaseDataguard.Role = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["RoleChangeInProgress"] = "ROLE_CHANGE_IN_PROGRESS";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Unavailable"] = "UNAVAILABLE";
        LifecycleState["Updating"] = "UPDATING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = AutonomousContainerDatabaseDataguard.LifecycleState || (AutonomousContainerDatabaseDataguard.LifecycleState = {}));
    let ProtectionMode;
    (function (ProtectionMode) {
        ProtectionMode["MaximumAvailability"] = "MAXIMUM_AVAILABILITY";
        ProtectionMode["MaximumPerformance"] = "MAXIMUM_PERFORMANCE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ProtectionMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(ProtectionMode = AutonomousContainerDatabaseDataguard.ProtectionMode || (AutonomousContainerDatabaseDataguard.ProtectionMode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AutonomousContainerDatabaseDataguard.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AutonomousContainerDatabaseDataguard.getDeserializedJsonObj = getDeserializedJsonObj;
})(AutonomousContainerDatabaseDataguard = exports.AutonomousContainerDatabaseDataguard || (exports.AutonomousContainerDatabaseDataguard = {}));
