// oci-database/lib/model/convert-standby-autonomous-container-database-details.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](https://docs.oracle.com/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConvertStandbyAutonomousContainerDatabaseDetails = void 0;
var ConvertStandbyAutonomousContainerDatabaseDetails;
(function (ConvertStandbyAutonomousContainerDatabaseDetails) {
    let Role;
    (function (Role) {
        Role["Primary"] = "PRIMARY";
        Role["Standby"] = "STANDBY";
        Role["DisabledStandby"] = "DISABLED_STANDBY";
        Role["BackupCopy"] = "BACKUP_COPY";
        Role["SnapshotStandby"] = "SNAPSHOT_STANDBY";
    })(Role = ConvertStandbyAutonomousContainerDatabaseDetails.Role || (ConvertStandbyAutonomousContainerDatabaseDetails.Role = {}));
    let ConnectionStringsType;
    (function (ConnectionStringsType) {
        ConnectionStringsType["SnapshotServices"] = "SNAPSHOT_SERVICES";
        ConnectionStringsType["PrimaryServices"] = "PRIMARY_SERVICES";
    })(ConnectionStringsType = ConvertStandbyAutonomousContainerDatabaseDetails.ConnectionStringsType || (ConvertStandbyAutonomousContainerDatabaseDetails.ConnectionStringsType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ConvertStandbyAutonomousContainerDatabaseDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ConvertStandbyAutonomousContainerDatabaseDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(ConvertStandbyAutonomousContainerDatabaseDetails = exports.ConvertStandbyAutonomousContainerDatabaseDetails || (exports.ConvertStandbyAutonomousContainerDatabaseDetails = {}));
