// oci-database/lib/model/execution-window.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](https://docs.oracle.com/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExecutionWindow = void 0;
var ExecutionWindow;
(function (ExecutionWindow) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Created"] = "CREATED";
        LifecycleState["Scheduled"] = "SCHEDULED";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Canceled"] = "CANCELED";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Succeeded"] = "SUCCEEDED";
        LifecycleState["PartialSuccess"] = "PARTIAL_SUCCESS";
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Deleting"] = "DELETING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ExecutionWindow.LifecycleState || (ExecutionWindow.LifecycleState = {}));
    let LifecycleSubstate;
    (function (LifecycleSubstate) {
        LifecycleSubstate["DurationExceeded"] = "DURATION_EXCEEDED";
        LifecycleSubstate["MaintenanceInProgress"] = "MAINTENANCE_IN_PROGRESS";
        LifecycleSubstate["Waiting"] = "WAITING";
        LifecycleSubstate["Rescheduled"] = "RESCHEDULED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleSubstate["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleSubstate = ExecutionWindow.LifecycleSubstate || (ExecutionWindow.LifecycleSubstate = {}));
    let WindowType;
    (function (WindowType) {
        WindowType["Planned"] = "PLANNED";
        WindowType["Unplanned"] = "UNPLANNED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        WindowType["UnknownValue"] = "UNKNOWN_VALUE";
    })(WindowType = ExecutionWindow.WindowType || (ExecutionWindow.WindowType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExecutionWindow.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExecutionWindow.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExecutionWindow = exports.ExecutionWindow || (exports.ExecutionWindow = {}));
