// oci-dataintegration/lib/request/list-export-requests-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200430
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListExportRequestsRequest = void 0;
var ListExportRequestsRequest;
(function (ListExportRequestsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListExportRequestsRequest.SortOrder || (ListExportRequestsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "TIME_CREATED";
        SortBy["DisplayName"] = "DISPLAY_NAME";
        SortBy["TimeUpdated"] = "TIME_UPDATED";
    })(SortBy = ListExportRequestsRequest.SortBy || (ListExportRequestsRequest.SortBy = {}));
    let ExportStatus;
    (function (ExportStatus) {
        ExportStatus["InProgress"] = "IN_PROGRESS";
        ExportStatus["Successful"] = "SUCCESSFUL";
        ExportStatus["Queued"] = "QUEUED";
        ExportStatus["Terminating"] = "TERMINATING";
        ExportStatus["Terminated"] = "TERMINATED";
        ExportStatus["Failed"] = "FAILED";
    })(ExportStatus = ListExportRequestsRequest.ExportStatus || (ListExportRequestsRequest.ExportStatus = {}));
    let Projection;
    (function (Projection) {
        Projection["Summary"] = "SUMMARY";
        Projection["Details"] = "DETAILS";
    })(Projection = ListExportRequestsRequest.Projection || (ListExportRequestsRequest.Projection = {}));
})(ListExportRequestsRequest = exports.ListExportRequestsRequest || (exports.ListExportRequestsRequest = {}));
