// oci-stackmonitoring/lib/request/index.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchMonitoredResourceMembersRequest = exports.RequestMonitoredResourcesSummarizedCountRequest = exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListProcessSetsRequest = exports.ListMonitoringTemplatesRequest = exports.ListMonitoredResourceTypesRequest = exports.ListMonitoredResourceTasksRequest = exports.ListDiscoveryJobsRequest = exports.ListDiscoveryJobLogsRequest = exports.ListDefinedMonitoringTemplatesRequest = exports.ListConfigsRequest = exports.ListBaselineableMetricsRequest = exports.ListAlarmConditionsRequest = void 0;
const ListAlarmConditionsRequest = __importStar(require("./list-alarm-conditions-request"));
exports.ListAlarmConditionsRequest = ListAlarmConditionsRequest.ListAlarmConditionsRequest;
const ListBaselineableMetricsRequest = __importStar(require("./list-baselineable-metrics-request"));
exports.ListBaselineableMetricsRequest = ListBaselineableMetricsRequest.ListBaselineableMetricsRequest;
const ListConfigsRequest = __importStar(require("./list-configs-request"));
exports.ListConfigsRequest = ListConfigsRequest.ListConfigsRequest;
const ListDefinedMonitoringTemplatesRequest = __importStar(require("./list-defined-monitoring-templates-request"));
exports.ListDefinedMonitoringTemplatesRequest = ListDefinedMonitoringTemplatesRequest.ListDefinedMonitoringTemplatesRequest;
const ListDiscoveryJobLogsRequest = __importStar(require("./list-discovery-job-logs-request"));
exports.ListDiscoveryJobLogsRequest = ListDiscoveryJobLogsRequest.ListDiscoveryJobLogsRequest;
const ListDiscoveryJobsRequest = __importStar(require("./list-discovery-jobs-request"));
exports.ListDiscoveryJobsRequest = ListDiscoveryJobsRequest.ListDiscoveryJobsRequest;
const ListMonitoredResourceTasksRequest = __importStar(require("./list-monitored-resource-tasks-request"));
exports.ListMonitoredResourceTasksRequest = ListMonitoredResourceTasksRequest.ListMonitoredResourceTasksRequest;
const ListMonitoredResourceTypesRequest = __importStar(require("./list-monitored-resource-types-request"));
exports.ListMonitoredResourceTypesRequest = ListMonitoredResourceTypesRequest.ListMonitoredResourceTypesRequest;
const ListMonitoringTemplatesRequest = __importStar(require("./list-monitoring-templates-request"));
exports.ListMonitoringTemplatesRequest = ListMonitoringTemplatesRequest.ListMonitoringTemplatesRequest;
const ListProcessSetsRequest = __importStar(require("./list-process-sets-request"));
exports.ListProcessSetsRequest = ListProcessSetsRequest.ListProcessSetsRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
const RequestMonitoredResourcesSummarizedCountRequest = __importStar(require("./request-monitored-resources-summarized-count-request"));
exports.RequestMonitoredResourcesSummarizedCountRequest = RequestMonitoredResourcesSummarizedCountRequest.RequestMonitoredResourcesSummarizedCountRequest;
const SearchMonitoredResourceMembersRequest = __importStar(require("./search-monitored-resource-members-request"));
exports.SearchMonitoredResourceMembersRequest = SearchMonitoredResourceMembersRequest.SearchMonitoredResourceMembersRequest;
