// oci-vault/lib/model/index.js
"use strict";
/**
 * Vault Secret Management API
 * Use the Secret Management API to manage secrets and secret versions. For more information, see [Managing Secrets](https://docs.oracle.com/iaas/Content/KeyManagement/Tasks/managingsecrets.htm).
 * OpenAPI spec version: 20180608
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SshKeyGenerationContext = exports.SecretReuseRule = exports.SecretExpiryRule = exports.PassphraseGenerationContext = exports.FunctionTargetSystemDetails = exports.BytesGenerationContext = exports.Base64SecretContentDetails = exports.AdbTargetSystemDetails = exports.UpdateSecretDetails = exports.TargetSystemDetails = exports.SecretVersionSummary = exports.SecretVersion = exports.SecretSummary = exports.SecretRule = exports.SecretGenerationContext = exports.SecretContentDetails = exports.Secret = exports.ScheduleSecretVersionDeletionDetails = exports.ScheduleSecretDeletionDetails = exports.RotationConfig = exports.CreateSecretDetails = exports.ChangeSecretCompartmentDetails = void 0;
const ChangeSecretCompartmentDetails = __importStar(require("./change-secret-compartment-details"));
exports.ChangeSecretCompartmentDetails = ChangeSecretCompartmentDetails.ChangeSecretCompartmentDetails;
const CreateSecretDetails = __importStar(require("./create-secret-details"));
exports.CreateSecretDetails = CreateSecretDetails.CreateSecretDetails;
const RotationConfig = __importStar(require("./rotation-config"));
exports.RotationConfig = RotationConfig.RotationConfig;
const ScheduleSecretDeletionDetails = __importStar(require("./schedule-secret-deletion-details"));
exports.ScheduleSecretDeletionDetails = ScheduleSecretDeletionDetails.ScheduleSecretDeletionDetails;
const ScheduleSecretVersionDeletionDetails = __importStar(require("./schedule-secret-version-deletion-details"));
exports.ScheduleSecretVersionDeletionDetails = ScheduleSecretVersionDeletionDetails.ScheduleSecretVersionDeletionDetails;
const Secret = __importStar(require("./secret"));
exports.Secret = Secret.Secret;
const SecretContentDetails = __importStar(require("./secret-content-details"));
exports.SecretContentDetails = SecretContentDetails.SecretContentDetails;
const SecretGenerationContext = __importStar(require("./secret-generation-context"));
exports.SecretGenerationContext = SecretGenerationContext.SecretGenerationContext;
const SecretRule = __importStar(require("./secret-rule"));
exports.SecretRule = SecretRule.SecretRule;
const SecretSummary = __importStar(require("./secret-summary"));
exports.SecretSummary = SecretSummary.SecretSummary;
const SecretVersion = __importStar(require("./secret-version"));
exports.SecretVersion = SecretVersion.SecretVersion;
const SecretVersionSummary = __importStar(require("./secret-version-summary"));
exports.SecretVersionSummary = SecretVersionSummary.SecretVersionSummary;
const TargetSystemDetails = __importStar(require("./target-system-details"));
exports.TargetSystemDetails = TargetSystemDetails.TargetSystemDetails;
const UpdateSecretDetails = __importStar(require("./update-secret-details"));
exports.UpdateSecretDetails = UpdateSecretDetails.UpdateSecretDetails;
const AdbTargetSystemDetails = __importStar(require("./adb-target-system-details"));
exports.AdbTargetSystemDetails = AdbTargetSystemDetails.AdbTargetSystemDetails;
const Base64SecretContentDetails = __importStar(require("./base64-secret-content-details"));
exports.Base64SecretContentDetails = Base64SecretContentDetails.Base64SecretContentDetails;
const BytesGenerationContext = __importStar(require("./bytes-generation-context"));
exports.BytesGenerationContext = BytesGenerationContext.BytesGenerationContext;
const FunctionTargetSystemDetails = __importStar(require("./function-target-system-details"));
exports.FunctionTargetSystemDetails = FunctionTargetSystemDetails.FunctionTargetSystemDetails;
const PassphraseGenerationContext = __importStar(require("./passphrase-generation-context"));
exports.PassphraseGenerationContext = PassphraseGenerationContext.PassphraseGenerationContext;
const SecretExpiryRule = __importStar(require("./secret-expiry-rule"));
exports.SecretExpiryRule = SecretExpiryRule.SecretExpiryRule;
const SecretReuseRule = __importStar(require("./secret-reuse-rule"));
exports.SecretReuseRule = SecretReuseRule.SecretReuseRule;
const SshKeyGenerationContext = __importStar(require("./ssh-key-generation-context"));
exports.SshKeyGenerationContext = SshKeyGenerationContext.SshKeyGenerationContext;
