// oci-vault/lib/model/ssh-key-generation-context.js
"use strict";
/**
 * Vault Secret Management API
 * Use the Secret Management API to manage secrets and secret versions. For more information, see [Managing Secrets](https://docs.oracle.com/iaas/Content/KeyManagement/Tasks/managingsecrets.htm).
 * OpenAPI spec version: 20180608
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SshKeyGenerationContext = void 0;
const model = __importStar(require("../model"));
var SshKeyGenerationContext;
(function (SshKeyGenerationContext) {
    let GenerationTemplate;
    (function (GenerationTemplate) {
        GenerationTemplate["Rsa2048"] = "RSA_2048";
        GenerationTemplate["Rsa3072"] = "RSA_3072";
        GenerationTemplate["Rsa4096"] = "RSA_4096";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        GenerationTemplate["UnknownValue"] = "UNKNOWN_VALUE";
    })(GenerationTemplate = SshKeyGenerationContext.GenerationTemplate || (SshKeyGenerationContext.GenerationTemplate = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.SecretGenerationContext.getJsonObj(obj))), {});
        return jsonObj;
    }
    SshKeyGenerationContext.getJsonObj = getJsonObj;
    SshKeyGenerationContext.generationType = "SSH_KEY";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.SecretGenerationContext.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    SshKeyGenerationContext.getDeserializedJsonObj = getDeserializedJsonObj;
})(SshKeyGenerationContext = exports.SshKeyGenerationContext || (exports.SshKeyGenerationContext = {}));
