// oci-goldengate/lib/model/backup-schedule.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackupSchedule = void 0;
var BackupSchedule;
(function (BackupSchedule) {
    let FrequencyBackupScheduled;
    (function (FrequencyBackupScheduled) {
        FrequencyBackupScheduled["Daily"] = "DAILY";
        FrequencyBackupScheduled["Weekly"] = "WEEKLY";
        FrequencyBackupScheduled["Monthly"] = "MONTHLY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        FrequencyBackupScheduled["UnknownValue"] = "UNKNOWN_VALUE";
    })(FrequencyBackupScheduled = BackupSchedule.FrequencyBackupScheduled || (BackupSchedule.FrequencyBackupScheduled = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BackupSchedule.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BackupSchedule.getDeserializedJsonObj = getDeserializedJsonObj;
})(BackupSchedule = exports.BackupSchedule || (exports.BackupSchedule = {}));
