// oci-goldengate/lib/model/deployment-type-summary.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentTypeSummary = void 0;
var DeploymentTypeSummary;
(function (DeploymentTypeSummary) {
    let ConnectionTypes;
    (function (ConnectionTypes) {
        ConnectionTypes["Goldengate"] = "GOLDENGATE";
        ConnectionTypes["Kafka"] = "KAFKA";
        ConnectionTypes["KafkaSchemaRegistry"] = "KAFKA_SCHEMA_REGISTRY";
        ConnectionTypes["Mysql"] = "MYSQL";
        ConnectionTypes["JavaMessageService"] = "JAVA_MESSAGE_SERVICE";
        ConnectionTypes["MicrosoftSqlserver"] = "MICROSOFT_SQLSERVER";
        ConnectionTypes["OciObjectStorage"] = "OCI_OBJECT_STORAGE";
        ConnectionTypes["Oracle"] = "ORACLE";
        ConnectionTypes["AzureDataLakeStorage"] = "AZURE_DATA_LAKE_STORAGE";
        ConnectionTypes["Postgresql"] = "POSTGRESQL";
        ConnectionTypes["AzureSynapseAnalytics"] = "AZURE_SYNAPSE_ANALYTICS";
        ConnectionTypes["Snowflake"] = "SNOWFLAKE";
        ConnectionTypes["AmazonS3"] = "AMAZON_S3";
        ConnectionTypes["Hdfs"] = "HDFS";
        ConnectionTypes["OracleNosql"] = "ORACLE_NOSQL";
        ConnectionTypes["Mongodb"] = "MONGODB";
        ConnectionTypes["AmazonKinesis"] = "AMAZON_KINESIS";
        ConnectionTypes["AmazonRedshift"] = "AMAZON_REDSHIFT";
        ConnectionTypes["Db2"] = "DB2";
        ConnectionTypes["Redis"] = "REDIS";
        ConnectionTypes["Elasticsearch"] = "ELASTICSEARCH";
        ConnectionTypes["Generic"] = "GENERIC";
        ConnectionTypes["GoogleCloudStorage"] = "GOOGLE_CLOUD_STORAGE";
        ConnectionTypes["GoogleBigquery"] = "GOOGLE_BIGQUERY";
        ConnectionTypes["Databricks"] = "DATABRICKS";
        ConnectionTypes["GooglePubsub"] = "GOOGLE_PUBSUB";
        ConnectionTypes["MicrosoftFabric"] = "MICROSOFT_FABRIC";
        ConnectionTypes["Iceberg"] = "ICEBERG";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ConnectionTypes["UnknownValue"] = "UNKNOWN_VALUE";
    })(ConnectionTypes = DeploymentTypeSummary.ConnectionTypes || (DeploymentTypeSummary.ConnectionTypes = {}));
    let SupportedCapabilities;
    (function (SupportedCapabilities) {
        SupportedCapabilities["GroupToRole"] = "GROUP_TO_ROLE";
        SupportedCapabilities["BackupRestore"] = "BACKUP_RESTORE";
        SupportedCapabilities["CopyBackup"] = "COPY_BACKUP";
        SupportedCapabilities["ManualBackup"] = "MANUAL_BACKUP";
        SupportedCapabilities["Placement"] = "PLACEMENT";
        SupportedCapabilities["DisasterRecovery"] = "DISASTER_RECOVERY";
        SupportedCapabilities["ScheduleManualBackup"] = "SCHEDULE_MANUAL_BACKUP";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SupportedCapabilities["UnknownValue"] = "UNKNOWN_VALUE";
    })(SupportedCapabilities = DeploymentTypeSummary.SupportedCapabilities || (DeploymentTypeSummary.SupportedCapabilities = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DeploymentTypeSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DeploymentTypeSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeploymentTypeSummary = exports.DeploymentTypeSummary || (exports.DeploymentTypeSummary = {}));
