// oci-goldengate/lib/model/deployment-role.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentRole = void 0;
/**
 * The Possible role for the deployment which is either primary or standby.
 *
 **/
var DeploymentRole;
(function (DeploymentRole) {
    DeploymentRole["Primary"] = "PRIMARY";
    DeploymentRole["Standby"] = "STANDBY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DeploymentRole["UnknownValue"] = "UNKNOWN_VALUE";
})(DeploymentRole = exports.DeploymentRole || (exports.DeploymentRole = {}));
(function (DeploymentRole) {
    function getJsonObj(obj) {
        return obj;
    }
    DeploymentRole.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DeploymentRole.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeploymentRole = exports.DeploymentRole || (exports.DeploymentRole = {}));
