// oci-goldengate/lib/model/operation-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
* <p>
Deprecated value: GoldengateConnectionAssigmnentDelete
*
**/
var OperationType;
(function (OperationType) {
    OperationType["GoldengateDatabaseRegistrationCreate"] = "GOLDENGATE_DATABASE_REGISTRATION_CREATE";
    OperationType["GoldengateDatabaseRegistrationUpdate"] = "GOLDENGATE_DATABASE_REGISTRATION_UPDATE";
    OperationType["GoldengateDatabaseRegistrationDelete"] = "GOLDENGATE_DATABASE_REGISTRATION_DELETE";
    OperationType["GoldengateDatabaseRegistrationMove"] = "GOLDENGATE_DATABASE_REGISTRATION_MOVE";
    OperationType["GoldengateDeploymentCreate"] = "GOLDENGATE_DEPLOYMENT_CREATE";
    OperationType["GoldengateDeploymentUpdate"] = "GOLDENGATE_DEPLOYMENT_UPDATE";
    OperationType["GoldengateDeploymentDelete"] = "GOLDENGATE_DEPLOYMENT_DELETE";
    OperationType["GoldengateDeploymentMove"] = "GOLDENGATE_DEPLOYMENT_MOVE";
    OperationType["GoldengateDeploymentRestore"] = "GOLDENGATE_DEPLOYMENT_RESTORE";
    OperationType["GoldengateDeploymentStart"] = "GOLDENGATE_DEPLOYMENT_START";
    OperationType["GoldengateDeploymentStop"] = "GOLDENGATE_DEPLOYMENT_STOP";
    OperationType["GoldengateDeploymentUpgrade"] = "GOLDENGATE_DEPLOYMENT_UPGRADE";
    OperationType["GoldengateDeploymentBackupCreate"] = "GOLDENGATE_DEPLOYMENT_BACKUP_CREATE";
    OperationType["GoldengateDeploymentBackupDelete"] = "GOLDENGATE_DEPLOYMENT_BACKUP_DELETE";
    OperationType["GoldengateDeploymentBackupCancel"] = "GOLDENGATE_DEPLOYMENT_BACKUP_CANCEL";
    OperationType["GoldengateDeploymentBackupCopy"] = "GOLDENGATE_DEPLOYMENT_BACKUP_COPY";
    OperationType["GoldengateConnectionCreate"] = "GOLDENGATE_CONNECTION_CREATE";
    OperationType["GoldengateConnectionUpdate"] = "GOLDENGATE_CONNECTION_UPDATE";
    OperationType["GoldengateConnectionDelete"] = "GOLDENGATE_CONNECTION_DELETE";
    OperationType["GoldengateConnectionMove"] = "GOLDENGATE_CONNECTION_MOVE";
    OperationType["GoldengateConnectionRefresh"] = "GOLDENGATE_CONNECTION_REFRESH";
    OperationType["GoldengateConnectionAssignmentCreate"] = "GOLDENGATE_CONNECTION_ASSIGNMENT_CREATE";
    OperationType["GoldengateConnectionAssignmentDelete"] = "GOLDENGATE_CONNECTION_ASSIGNMENT_DELETE";
    OperationType["GoldengateConnectionAssigmnentDelete"] = "GOLDENGATE_CONNECTION_ASSIGMNENT_DELETE";
    OperationType["GoldengateDeploymentDiagnosticCollect"] = "GOLDENGATE_DEPLOYMENT_DIAGNOSTIC_COLLECT";
    OperationType["GoldengateDeploymentWalletExport"] = "GOLDENGATE_DEPLOYMENT_WALLET_EXPORT";
    OperationType["GoldengateDeploymentWalletImport"] = "GOLDENGATE_DEPLOYMENT_WALLET_IMPORT";
    OperationType["GoldengateDeploymentUpgradeUpgrade"] = "GOLDENGATE_DEPLOYMENT_UPGRADE_UPGRADE";
    OperationType["GoldengateDeploymentUpgradeRollback"] = "GOLDENGATE_DEPLOYMENT_UPGRADE_ROLLBACK";
    OperationType["GoldengateDeploymentUpgradeSnooze"] = "GOLDENGATE_DEPLOYMENT_UPGRADE_SNOOZE";
    OperationType["GoldengateDeploymentCertificateCreate"] = "GOLDENGATE_DEPLOYMENT_CERTIFICATE_CREATE";
    OperationType["GoldengateDeploymentCertificateDelete"] = "GOLDENGATE_DEPLOYMENT_CERTIFICATE_DELETE";
    OperationType["GoldengatePipelineCreate"] = "GOLDENGATE_PIPELINE_CREATE";
    OperationType["GoldengatePipelineStart"] = "GOLDENGATE_PIPELINE_START";
    OperationType["GoldengatePipelineStop"] = "GOLDENGATE_PIPELINE_STOP";
    OperationType["GoldengatePipelineUpdate"] = "GOLDENGATE_PIPELINE_UPDATE";
    OperationType["GoldengatePipelineDelete"] = "GOLDENGATE_PIPELINE_DELETE";
    OperationType["GoldengatePipelineMove"] = "GOLDENGATE_PIPELINE_MOVE";
    OperationType["GoldengatePipelineDiagnosticsCollect"] = "GOLDENGATE_PIPELINE_DIAGNOSTICS_COLLECT";
    OperationType["GoldengateSwitchoverDeploymentPeer"] = "GOLDENGATE_SWITCHOVER_DEPLOYMENT_PEER";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
