// oci-goldengate/lib/model/process-status-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProcessStatusType = void 0;
/**
 * The process (Extract or Replicat) status type.
 *
 **/
var ProcessStatusType;
(function (ProcessStatusType) {
    ProcessStatusType["Stopped"] = "STOPPED";
    ProcessStatusType["Running"] = "RUNNING";
    ProcessStatusType["Error"] = "ERROR";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ProcessStatusType["UnknownValue"] = "UNKNOWN_VALUE";
})(ProcessStatusType = exports.ProcessStatusType || (exports.ProcessStatusType = {}));
(function (ProcessStatusType) {
    function getJsonObj(obj) {
        return obj;
    }
    ProcessStatusType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ProcessStatusType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ProcessStatusType = exports.ProcessStatusType || (exports.ProcessStatusType = {}));
