// oci-goldengate/lib/model/replicate-ddl-error-action.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicateDdlErrorAction = void 0;
/**
 * The action to take when a DDL error occurs during replication. Active only if Replicate schema changes (canReplicateSchemaChange=true) is selected.
 *
 **/
var ReplicateDdlErrorAction;
(function (ReplicateDdlErrorAction) {
    ReplicateDdlErrorAction["Terminate"] = "TERMINATE";
    ReplicateDdlErrorAction["Discard"] = "DISCARD";
    ReplicateDdlErrorAction["Ignore"] = "IGNORE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ReplicateDdlErrorAction["UnknownValue"] = "UNKNOWN_VALUE";
})(ReplicateDdlErrorAction = exports.ReplicateDdlErrorAction || (exports.ReplicateDdlErrorAction = {}));
(function (ReplicateDdlErrorAction) {
    function getJsonObj(obj) {
        return obj;
    }
    ReplicateDdlErrorAction.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ReplicateDdlErrorAction.getDeserializedJsonObj = getDeserializedJsonObj;
})(ReplicateDdlErrorAction = exports.ReplicateDdlErrorAction || (exports.ReplicateDdlErrorAction = {}));
