// oci-goldengate/lib/model/reschedule-deployment-upgrade-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RescheduleDeploymentUpgradeType = void 0;
/**
 * The reschedule type of deploymentUpgrade.
 *
 **/
var RescheduleDeploymentUpgradeType;
(function (RescheduleDeploymentUpgradeType) {
    RescheduleDeploymentUpgradeType["RescheduleToDate"] = "RESCHEDULE_TO_DATE";
})(RescheduleDeploymentUpgradeType = exports.RescheduleDeploymentUpgradeType || (exports.RescheduleDeploymentUpgradeType = {}));
(function (RescheduleDeploymentUpgradeType) {
    function getJsonObj(obj) {
        return obj;
    }
    RescheduleDeploymentUpgradeType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RescheduleDeploymentUpgradeType.getDeserializedJsonObj = getDeserializedJsonObj;
})(RescheduleDeploymentUpgradeType = exports.RescheduleDeploymentUpgradeType || (exports.RescheduleDeploymentUpgradeType = {}));
