// oci-goldengate/lib/model/severity-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SeverityType = void 0;
/**
 * Message severity of GoldenGate APIs called for a step in a recipe.
 * https://docs.oracle.com/en/middleware/goldengate/core/23/oggra/rest-endpoints.html
 *
 **/
var SeverityType;
(function (SeverityType) {
    SeverityType["Info"] = "INFO";
    SeverityType["Error"] = "ERROR";
    SeverityType["Warning"] = "WARNING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SeverityType["UnknownValue"] = "UNKNOWN_VALUE";
})(SeverityType = exports.SeverityType || (exports.SeverityType = {}));
(function (SeverityType) {
    function getJsonObj(obj) {
        return obj;
    }
    SeverityType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SeverityType.getDeserializedJsonObj = getDeserializedJsonObj;
})(SeverityType = exports.SeverityType || (exports.SeverityType = {}));
