// oci-goldengate/lib/model/supported-capabilities.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SupportedCapabilities = void 0;
/**
 * Specifies supported Capabilities or features supported by deployment type.
 *
 **/
var SupportedCapabilities;
(function (SupportedCapabilities) {
    SupportedCapabilities["GroupToRole"] = "GROUP_TO_ROLE";
    SupportedCapabilities["BackupRestore"] = "BACKUP_RESTORE";
    SupportedCapabilities["CopyBackup"] = "COPY_BACKUP";
    SupportedCapabilities["ManualBackup"] = "MANUAL_BACKUP";
    SupportedCapabilities["Placement"] = "PLACEMENT";
    SupportedCapabilities["DisasterRecovery"] = "DISASTER_RECOVERY";
    SupportedCapabilities["ScheduleManualBackup"] = "SCHEDULE_MANUAL_BACKUP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SupportedCapabilities["UnknownValue"] = "UNKNOWN_VALUE";
})(SupportedCapabilities = exports.SupportedCapabilities || (exports.SupportedCapabilities = {}));
(function (SupportedCapabilities) {
    function getJsonObj(obj) {
        return obj;
    }
    SupportedCapabilities.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SupportedCapabilities.getDeserializedJsonObj = getDeserializedJsonObj;
})(SupportedCapabilities = exports.SupportedCapabilities || (exports.SupportedCapabilities = {}));
