// oci-goldengate/lib/model/technology-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TechnologyType = void 0;
/**
 * The technology type.
 *
 **/
var TechnologyType;
(function (TechnologyType) {
    TechnologyType["Goldengate"] = "GOLDENGATE";
    TechnologyType["Generic"] = "GENERIC";
    TechnologyType["OciAutonomousDatabase"] = "OCI_AUTONOMOUS_DATABASE";
    TechnologyType["OciAutonomousJsonDatabase"] = "OCI_AUTONOMOUS_JSON_DATABASE";
    TechnologyType["OciCacheWithRedis"] = "OCI_CACHE_WITH_REDIS";
    TechnologyType["OciMysql"] = "OCI_MYSQL";
    TechnologyType["OciObjectStorage"] = "OCI_OBJECT_STORAGE";
    TechnologyType["OciPostgresql"] = "OCI_POSTGRESQL";
    TechnologyType["OciStreaming"] = "OCI_STREAMING";
    TechnologyType["OracleDatabase"] = "ORACLE_DATABASE";
    TechnologyType["OracleExadata"] = "ORACLE_EXADATA";
    TechnologyType["OracleExadataDatabaseAtAzure"] = "ORACLE_EXADATA_DATABASE_AT_AZURE";
    TechnologyType["OracleAutonomousDatabaseAtAzure"] = "ORACLE_AUTONOMOUS_DATABASE_AT_AZURE";
    TechnologyType["OracleJsonCollection"] = "ORACLE_JSON_COLLECTION";
    TechnologyType["OracleExadataDatabaseAtGoogleCloud"] = "ORACLE_EXADATA_DATABASE_AT_GOOGLE_CLOUD";
    TechnologyType["OracleAutonomousDatabaseAtGoogleCloud"] = "ORACLE_AUTONOMOUS_DATABASE_AT_GOOGLE_CLOUD";
    TechnologyType["OracleExadataDatabaseAtAws"] = "ORACLE_EXADATA_DATABASE_AT_AWS";
    TechnologyType["OracleAutonomousDatabaseAtAws"] = "ORACLE_AUTONOMOUS_DATABASE_AT_AWS";
    TechnologyType["OracleNosql"] = "ORACLE_NOSQL";
    TechnologyType["OracleRestDataServices"] = "ORACLE_REST_DATA_SERVICES";
    TechnologyType["OracleWeblogicJms"] = "ORACLE_WEBLOGIC_JMS";
    TechnologyType["AmazonRdsOracle"] = "AMAZON_RDS_ORACLE";
    TechnologyType["AmazonRdsSqlserver"] = "AMAZON_RDS_SQLSERVER";
    TechnologyType["AmazonS3"] = "AMAZON_S3";
    TechnologyType["AmazonAuroraMysql"] = "AMAZON_AURORA_MYSQL";
    TechnologyType["AmazonAuroraPostgresql"] = "AMAZON_AURORA_POSTGRESQL";
    TechnologyType["AmazonDocumentDb"] = "AMAZON_DOCUMENT_DB";
    TechnologyType["AmazonKinesis"] = "AMAZON_KINESIS";
    TechnologyType["AmazonRedshift"] = "AMAZON_REDSHIFT";
    TechnologyType["AmazonRdsMariadb"] = "AMAZON_RDS_MARIADB";
    TechnologyType["AmazonRdsMysql"] = "AMAZON_RDS_MYSQL";
    TechnologyType["AmazonRdsPostgresql"] = "AMAZON_RDS_POSTGRESQL";
    TechnologyType["ApacheIceberg"] = "APACHE_ICEBERG";
    TechnologyType["ApacheKafka"] = "APACHE_KAFKA";
    TechnologyType["AzureCosmosDbForMongodb"] = "AZURE_COSMOS_DB_FOR_MONGODB";
    TechnologyType["AzureCosmosDbForPostgresql"] = "AZURE_COSMOS_DB_FOR_POSTGRESQL";
    TechnologyType["AzureDataLakeStorage"] = "AZURE_DATA_LAKE_STORAGE";
    TechnologyType["AzureEventHubs"] = "AZURE_EVENT_HUBS";
    TechnologyType["AzureMysql"] = "AZURE_MYSQL";
    TechnologyType["AzurePostgresql"] = "AZURE_POSTGRESQL";
    TechnologyType["AzureSqlserverManagedInstance"] = "AZURE_SQLSERVER_MANAGED_INSTANCE";
    TechnologyType["AzureSqlserverNonManagedInstance"] = "AZURE_SQLSERVER_NON_MANAGED_INSTANCE";
    TechnologyType["AzureSynapseAnalytics"] = "AZURE_SYNAPSE_ANALYTICS";
    TechnologyType["ConfluentKafka"] = "CONFLUENT_KAFKA";
    TechnologyType["ConfluentSchemaRegistry"] = "CONFLUENT_SCHEMA_REGISTRY";
    TechnologyType["Databricks"] = "DATABRICKS";
    TechnologyType["Db2I"] = "DB2_I";
    TechnologyType["Db2Zos"] = "DB2_ZOS";
    TechnologyType["Elasticsearch"] = "ELASTICSEARCH";
    TechnologyType["GoogleAlloyDbForPostgresql"] = "GOOGLE_ALLOY_DB_FOR_POSTGRESQL";
    TechnologyType["GoogleBigquery"] = "GOOGLE_BIGQUERY";
    TechnologyType["GoogleCloudStorage"] = "GOOGLE_CLOUD_STORAGE";
    TechnologyType["GoogleCloudSqlMysql"] = "GOOGLE_CLOUD_SQL_MYSQL";
    TechnologyType["GoogleCloudSqlPostgresql"] = "GOOGLE_CLOUD_SQL_POSTGRESQL";
    TechnologyType["GoogleCloudSqlSqlserver"] = "GOOGLE_CLOUD_SQL_SQLSERVER";
    TechnologyType["GooglePubsub"] = "GOOGLE_PUBSUB";
    TechnologyType["Hdfs"] = "HDFS";
    TechnologyType["Mariadb"] = "MARIADB";
    TechnologyType["MicrosoftSqlserver"] = "MICROSOFT_SQLSERVER";
    TechnologyType["MicrosoftFabricLakehouse"] = "MICROSOFT_FABRIC_LAKEHOUSE";
    TechnologyType["MicrosoftFabricMirror"] = "MICROSOFT_FABRIC_MIRROR";
    TechnologyType["Mongodb"] = "MONGODB";
    TechnologyType["MysqlServer"] = "MYSQL_SERVER";
    TechnologyType["MysqlHeatwaveOnAzure"] = "MYSQL_HEATWAVE_ON_AZURE";
    TechnologyType["MysqlHeatwaveOnAws"] = "MYSQL_HEATWAVE_ON_AWS";
    TechnologyType["PostgresqlServer"] = "POSTGRESQL_SERVER";
    TechnologyType["Redis"] = "REDIS";
    TechnologyType["Singlestoredb"] = "SINGLESTOREDB";
    TechnologyType["SinglestoredbCloud"] = "SINGLESTOREDB_CLOUD";
    TechnologyType["Snowflake"] = "SNOWFLAKE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TechnologyType["UnknownValue"] = "UNKNOWN_VALUE";
})(TechnologyType = exports.TechnologyType || (exports.TechnologyType = {}));
(function (TechnologyType) {
    function getJsonObj(obj) {
        return obj;
    }
    TechnologyType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TechnologyType.getDeserializedJsonObj = getDeserializedJsonObj;
})(TechnologyType = exports.TechnologyType || (exports.TechnologyType = {}));
