// oci-goldengate/lib/model/test-connection-assignment-result.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestConnectionAssignmentResult = void 0;
const model = __importStar(require("../model"));
var TestConnectionAssignmentResult;
(function (TestConnectionAssignmentResult) {
    let ResultType;
    (function (ResultType) {
        ResultType["Succeeded"] = "SUCCEEDED";
        ResultType["Failed"] = "FAILED";
        ResultType["TimedOut"] = "TIMED_OUT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ResultType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ResultType = TestConnectionAssignmentResult.ResultType || (TestConnectionAssignmentResult.ResultType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "error": obj.error ? model.TestConnectionAssignmentError.getJsonObj(obj.error) : undefined,
            "errors": obj.errors
                ? obj.errors.map(item => {
                    return model.TestConnectionAssignmentError.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    TestConnectionAssignmentResult.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "error": obj.error
                ? model.TestConnectionAssignmentError.getDeserializedJsonObj(obj.error)
                : undefined,
            "errors": obj.errors
                ? obj.errors.map(item => {
                    return model.TestConnectionAssignmentError.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    TestConnectionAssignmentResult.getDeserializedJsonObj = getDeserializedJsonObj;
})(TestConnectionAssignmentResult = exports.TestConnectionAssignmentResult || (exports.TestConnectionAssignmentResult = {}));
