// oci-goldengate/lib/model/test-connection-assignment-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestConnectionAssignmentType = void 0;
/**
 * The type of the test to be performed.
 *
 **/
var TestConnectionAssignmentType;
(function (TestConnectionAssignmentType) {
    TestConnectionAssignmentType["Default"] = "DEFAULT";
})(TestConnectionAssignmentType = exports.TestConnectionAssignmentType || (exports.TestConnectionAssignmentType = {}));
(function (TestConnectionAssignmentType) {
    function getJsonObj(obj) {
        return obj;
    }
    TestConnectionAssignmentType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TestConnectionAssignmentType.getDeserializedJsonObj = getDeserializedJsonObj;
})(TestConnectionAssignmentType = exports.TestConnectionAssignmentType || (exports.TestConnectionAssignmentType = {}));
