// oci-mngdmac/lib/model/index.js
"use strict";
/**
 * OCI Managed Services for Mac API
 * Use the OCI Managed Services for Mac API to create and manage orders for dedicated, partially-managed Mac servers hosted in an OCI Data Center. For more information, see the user guide documentation for the [OCI Managed Services for Mac]()
 * OpenAPI spec version: 20250320
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateNodeConfigDetails = exports.UpdateMacOrderDetails = exports.SortOrder = exports.OpsActionUpdateMacOrderDetails = exports.OpsActionUpdateMacDeviceDetails = exports.OpsActionCreateMacDeviceDetails = exports.OperationType = exports.OperationStatus = exports.NodeConfig = exports.MacOrderSummary = exports.MacOrderCollection = exports.MacOrder = exports.MacDeviceSummary = exports.MacDeviceCollection = exports.MacDevice = exports.CreateNodeConfigDetails = exports.CreateMacOrderDetails = exports.ChangeMacOrderCompartmentDetails = exports.CancelMacOrderDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const CancelMacOrderDetails = __importStar(require("./cancel-mac-order-details"));
exports.CancelMacOrderDetails = CancelMacOrderDetails.CancelMacOrderDetails;
const ChangeMacOrderCompartmentDetails = __importStar(require("./change-mac-order-compartment-details"));
exports.ChangeMacOrderCompartmentDetails = ChangeMacOrderCompartmentDetails.ChangeMacOrderCompartmentDetails;
const CreateMacOrderDetails = __importStar(require("./create-mac-order-details"));
exports.CreateMacOrderDetails = CreateMacOrderDetails.CreateMacOrderDetails;
const CreateNodeConfigDetails = __importStar(require("./create-node-config-details"));
exports.CreateNodeConfigDetails = CreateNodeConfigDetails.CreateNodeConfigDetails;
const MacDevice = __importStar(require("./mac-device"));
exports.MacDevice = MacDevice.MacDevice;
const MacDeviceCollection = __importStar(require("./mac-device-collection"));
exports.MacDeviceCollection = MacDeviceCollection.MacDeviceCollection;
const MacDeviceSummary = __importStar(require("./mac-device-summary"));
exports.MacDeviceSummary = MacDeviceSummary.MacDeviceSummary;
const MacOrder = __importStar(require("./mac-order"));
exports.MacOrder = MacOrder.MacOrder;
const MacOrderCollection = __importStar(require("./mac-order-collection"));
exports.MacOrderCollection = MacOrderCollection.MacOrderCollection;
const MacOrderSummary = __importStar(require("./mac-order-summary"));
exports.MacOrderSummary = MacOrderSummary.MacOrderSummary;
const NodeConfig = __importStar(require("./node-config"));
exports.NodeConfig = NodeConfig.NodeConfig;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OpsActionCreateMacDeviceDetails = __importStar(require("./ops-action-create-mac-device-details"));
exports.OpsActionCreateMacDeviceDetails = OpsActionCreateMacDeviceDetails.OpsActionCreateMacDeviceDetails;
const OpsActionUpdateMacDeviceDetails = __importStar(require("./ops-action-update-mac-device-details"));
exports.OpsActionUpdateMacDeviceDetails = OpsActionUpdateMacDeviceDetails.OpsActionUpdateMacDeviceDetails;
const OpsActionUpdateMacOrderDetails = __importStar(require("./ops-action-update-mac-order-details"));
exports.OpsActionUpdateMacOrderDetails = OpsActionUpdateMacOrderDetails.OpsActionUpdateMacOrderDetails;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateMacOrderDetails = __importStar(require("./update-mac-order-details"));
exports.UpdateMacOrderDetails = UpdateMacOrderDetails.UpdateMacOrderDetails;
const UpdateNodeConfigDetails = __importStar(require("./update-node-config-details"));
exports.UpdateNodeConfigDetails = UpdateNodeConfigDetails.UpdateNodeConfigDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
