// oci-mngdmac/lib/model/ops-action-update-mac-device-details.js
"use strict";
/**
 * OCI Managed Services for Mac API
 * Use the OCI Managed Services for Mac API to create and manage orders for dedicated, partially-managed Mac servers hosted in an OCI Data Center. For more information, see the user guide documentation for the [OCI Managed Services for Mac]()
 * OpenAPI spec version: 20250320
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpsActionUpdateMacDeviceDetails = void 0;
var OpsActionUpdateMacDeviceDetails;
(function (OpsActionUpdateMacDeviceDetails) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
    })(LifecycleState = OpsActionUpdateMacDeviceDetails.LifecycleState || (OpsActionUpdateMacDeviceDetails.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    OpsActionUpdateMacDeviceDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    OpsActionUpdateMacDeviceDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(OpsActionUpdateMacDeviceDetails = exports.OpsActionUpdateMacDeviceDetails || (exports.OpsActionUpdateMacDeviceDetails = {}));
