// oci-cloudbridge/lib/model/supported-cloud-region-lifecycle-state.js
"use strict";
/**
 * Oracle Cloud Bridge API
 * API for Oracle Cloud Bridge service.
 * OpenAPI spec version: 20220509
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SupportedCloudRegionLifecycleState = void 0;
/**
 * Lifecycle state of the supported cloud region.
 **/
var SupportedCloudRegionLifecycleState;
(function (SupportedCloudRegionLifecycleState) {
    SupportedCloudRegionLifecycleState["Active"] = "ACTIVE";
    SupportedCloudRegionLifecycleState["Inactive"] = "INACTIVE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SupportedCloudRegionLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SupportedCloudRegionLifecycleState = exports.SupportedCloudRegionLifecycleState || (exports.SupportedCloudRegionLifecycleState = {}));
(function (SupportedCloudRegionLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SupportedCloudRegionLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SupportedCloudRegionLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SupportedCloudRegionLifecycleState = exports.SupportedCloudRegionLifecycleState || (exports.SupportedCloudRegionLifecycleState = {}));
