// oci-budget/lib/model/processing-period-type.js
"use strict";
/**
 * Budgets API
 * Use the Budgets API to manage budgets and budget alerts. For more information, see [Budgets Overview](https://docs.oracle.com/iaas/Content/Billing/Concepts/budgetsoverview.htm).
 * OpenAPI spec version: 20190111
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProcessingPeriodType = void 0;
/**
 * The budget processing period type. Valid values are INVOICE, MONTH, and SINGLE_USE.
 **/
var ProcessingPeriodType;
(function (ProcessingPeriodType) {
    ProcessingPeriodType["Invoice"] = "INVOICE";
    ProcessingPeriodType["Month"] = "MONTH";
    ProcessingPeriodType["SingleUse"] = "SINGLE_USE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ProcessingPeriodType["UnknownValue"] = "UNKNOWN_VALUE";
})(ProcessingPeriodType = exports.ProcessingPeriodType || (exports.ProcessingPeriodType = {}));
(function (ProcessingPeriodType) {
    function getJsonObj(obj) {
        return obj;
    }
    ProcessingPeriodType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ProcessingPeriodType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ProcessingPeriodType = exports.ProcessingPeriodType || (exports.ProcessingPeriodType = {}));
