// oci-filestorage/lib/request/list-quota-rules-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20171215
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListQuotaRulesRequest = void 0;
var ListQuotaRulesRequest;
(function (ListQuotaRulesRequest) {
    let PrincipalType;
    (function (PrincipalType) {
        PrincipalType["FileSystemLevel"] = "FILE_SYSTEM_LEVEL";
        PrincipalType["DefaultGroup"] = "DEFAULT_GROUP";
        PrincipalType["DefaultUser"] = "DEFAULT_USER";
        PrincipalType["IndividualGroup"] = "INDIVIDUAL_GROUP";
        PrincipalType["IndividualUser"] = "INDIVIDUAL_USER";
    })(PrincipalType = ListQuotaRulesRequest.PrincipalType || (ListQuotaRulesRequest.PrincipalType = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListQuotaRulesRequest.SortOrder || (ListQuotaRulesRequest.SortOrder = {}));
})(ListQuotaRulesRequest = exports.ListQuotaRulesRequest || (exports.ListQuotaRulesRequest = {}));
