// oci-ocvp/lib/model/supported-host-shape-summary.js
"use strict";
/**
 * Oracle Cloud VMware Solution API
 * Use the Oracle Cloud VMware API to create SDDCs and manage ESXi hosts and software.
For more information, see [Oracle Cloud VMware Solution](https://docs.oracle.com/iaas/Content/VMware/Concepts/ocvsoverview.htm).

 * OpenAPI spec version: 20230701
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SupportedHostShapeSummary = void 0;
var SupportedHostShapeSummary;
(function (SupportedHostShapeSummary) {
    let SupportedOperations;
    (function (SupportedOperations) {
        SupportedOperations["CreateSddc"] = "CREATE_SDDC";
        SupportedOperations["DeleteSddc"] = "DELETE_SDDC";
        SupportedOperations["CreateCluster"] = "CREATE_CLUSTER";
        SupportedOperations["DeleteCluster"] = "DELETE_CLUSTER";
        SupportedOperations["CreateEsxiHost"] = "CREATE_ESXI_HOST";
        SupportedOperations["DeleteEsxiHost"] = "DELETE_ESXI_HOST";
        SupportedOperations["UpgradeHcx"] = "UPGRADE_HCX";
        SupportedOperations["DowngradeHcx"] = "DOWNGRADE_HCX";
        SupportedOperations["CancelDowngradeHcx"] = "CANCEL_DOWNGRADE_HCX";
        SupportedOperations["RefreshHcxLicenseStatus"] = "REFRESH_HCX_LICENSE_STATUS";
        SupportedOperations["SwapBilling"] = "SWAP_BILLING";
        SupportedOperations["ReplaceHost"] = "REPLACE_HOST";
        SupportedOperations["InPlaceUpgrade"] = "IN_PLACE_UPGRADE";
        SupportedOperations["CreateDatastore"] = "CREATE_DATASTORE";
        SupportedOperations["UpdateDatastore"] = "UPDATE_DATASTORE";
        SupportedOperations["AddBlockVolumeToDatastore"] = "ADD_BLOCK_VOLUME_TO_DATASTORE";
        SupportedOperations["DeleteDatastore"] = "DELETE_DATASTORE";
        SupportedOperations["CreateDatastoreCluster"] = "CREATE_DATASTORE_CLUSTER";
        SupportedOperations["UpdateDatastoreCluster"] = "UPDATE_DATASTORE_CLUSTER";
        SupportedOperations["AttachDatastoreClusterToEsxiHost"] = "ATTACH_DATASTORE_CLUSTER_TO_ESXI_HOST";
        SupportedOperations["AttachDatastoreClusterToCluster"] = "ATTACH_DATASTORE_CLUSTER_TO_CLUSTER";
        SupportedOperations["DetachDatastoreClusterFromEsxiHost"] = "DETACH_DATASTORE_CLUSTER_FROM_ESXI_HOST";
        SupportedOperations["DetachDatastoreClusterFromCluster"] = "DETACH_DATASTORE_CLUSTER_FROM_CLUSTER";
        SupportedOperations["DeleteDatastoreCluster"] = "DELETE_DATASTORE_CLUSTER";
        SupportedOperations["AddDatastoreToDatastoreCluster"] = "ADD_DATASTORE_TO_DATASTORE_CLUSTER";
        SupportedOperations["RemoveDatastoreFromDatastoreCluster"] = "REMOVE_DATASTORE_FROM_DATASTORE_CLUSTER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SupportedOperations["UnknownValue"] = "UNKNOWN_VALUE";
    })(SupportedOperations = SupportedHostShapeSummary.SupportedOperations || (SupportedHostShapeSummary.SupportedOperations = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SupportedHostShapeSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SupportedHostShapeSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(SupportedHostShapeSummary = exports.SupportedHostShapeSummary || (exports.SupportedHostShapeSummary = {}));
