// oci-dblm/lib/request/list-databases-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20240102
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDatabasesRequest = void 0;
var ListDatabasesRequest;
(function (ListDatabasesRequest) {
    let DatabaseType;
    (function (DatabaseType) {
        DatabaseType["Si"] = "SI";
        DatabaseType["Rac"] = "RAC";
    })(DatabaseType = ListDatabasesRequest.DatabaseType || (ListDatabasesRequest.DatabaseType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["Name"] = "name";
        SortBy["ResourceType"] = "resourceType";
        SortBy["Release"] = "release";
        SortBy["SubscribedImage"] = "subscribedImage";
        SortBy["PatchCompliance"] = "patchCompliance";
    })(SortBy = ListDatabasesRequest.SortBy || (ListDatabasesRequest.SortBy = {}));
    let ImageCompliance;
    (function (ImageCompliance) {
        ImageCompliance["NotSubscribed"] = "NOT_SUBSCRIBED";
        ImageCompliance["NotCompliantWithImages"] = "NOT_COMPLIANT_WITH_IMAGES";
        ImageCompliance["AllDatabases"] = "ALL_DATABASES";
    })(ImageCompliance = ListDatabasesRequest.ImageCompliance || (ListDatabasesRequest.ImageCompliance = {}));
})(ListDatabasesRequest = exports.ListDatabasesRequest || (exports.ListDatabasesRequest = {}));
