// oci-bds/lib/model/node-replace-configuration.js
"use strict";
/**
 * Big Data Service API
 * REST API for Oracle Big Data Service. Use this API to build, deploy, and manage fully elastic Big Data Service clusters. Build on Hadoop, Spark and Data Science distributions, which can be fully integrated with existing enterprise data in Oracle Database and Oracle applications.

 * OpenAPI spec version: 20190531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeReplaceConfiguration = void 0;
const model = __importStar(require("../model"));
var NodeReplaceConfiguration;
(function (NodeReplaceConfiguration) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = NodeReplaceConfiguration.LifecycleState || (NodeReplaceConfiguration.LifecycleState = {}));
    let MetricType;
    (function (MetricType) {
        MetricType["InstanceStatus"] = "INSTANCE_STATUS";
        MetricType["InstanceAccessibilityStatus"] = "INSTANCE_ACCESSIBILITY_STATUS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MetricType["UnknownValue"] = "UNKNOWN_VALUE";
    })(MetricType = NodeReplaceConfiguration.MetricType || (NodeReplaceConfiguration.MetricType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "levelTypeDetails": obj.levelTypeDetails
                ? model.LevelTypeDetails.getJsonObj(obj.levelTypeDetails)
                : undefined
        });
        return jsonObj;
    }
    NodeReplaceConfiguration.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "levelTypeDetails": obj.levelTypeDetails
                ? model.LevelTypeDetails.getDeserializedJsonObj(obj.levelTypeDetails)
                : undefined
        });
        return jsonObj;
    }
    NodeReplaceConfiguration.getDeserializedJsonObj = getDeserializedJsonObj;
})(NodeReplaceConfiguration = exports.NodeReplaceConfiguration || (exports.NodeReplaceConfiguration = {}));
