// oci-apmsynthetics/lib/request/index.js
"use strict";
/**
 * APM Availability Monitoring API
 * Use the APM Availability Monitoring API to query Scripts, Monitors, Dedicated Vantage Points and On-Premise Vantage Points resources. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkersRequest = exports.ListScriptsRequest = exports.ListPublicVantagePointsRequest = exports.ListOnPremiseVantagePointsRequest = exports.ListMonitorsRequest = exports.ListDedicatedVantagePointsRequest = void 0;
const ListDedicatedVantagePointsRequest = __importStar(require("./list-dedicated-vantage-points-request"));
exports.ListDedicatedVantagePointsRequest = ListDedicatedVantagePointsRequest.ListDedicatedVantagePointsRequest;
const ListMonitorsRequest = __importStar(require("./list-monitors-request"));
exports.ListMonitorsRequest = ListMonitorsRequest.ListMonitorsRequest;
const ListOnPremiseVantagePointsRequest = __importStar(require("./list-on-premise-vantage-points-request"));
exports.ListOnPremiseVantagePointsRequest = ListOnPremiseVantagePointsRequest.ListOnPremiseVantagePointsRequest;
const ListPublicVantagePointsRequest = __importStar(require("./list-public-vantage-points-request"));
exports.ListPublicVantagePointsRequest = ListPublicVantagePointsRequest.ListPublicVantagePointsRequest;
const ListScriptsRequest = __importStar(require("./list-scripts-request"));
exports.ListScriptsRequest = ListScriptsRequest.ListScriptsRequest;
const ListWorkersRequest = __importStar(require("./list-workers-request"));
exports.ListWorkersRequest = ListWorkersRequest.ListWorkersRequest;
