// oci-apmsynthetics/lib/request/list-workers-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkersRequest = void 0;
var ListWorkersRequest;
(function (ListWorkersRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["DisplayName"] = "displayName";
        SortBy["Name"] = "name";
        SortBy["Status"] = "status";
        SortBy["Priority"] = "priority";
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["TimeUpdated"] = "timeUpdated";
        SortBy["TimeLastSyncup"] = "timeLastSyncup";
    })(SortBy = ListWorkersRequest.SortBy || (ListWorkersRequest.SortBy = {}));
})(ListWorkersRequest = exports.ListWorkersRequest || (exports.ListWorkersRequest = {}));
