// oci-osmanagementhub/lib/request/get-managed-instance-analytic-content-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetManagedInstanceAnalyticContentRequest = void 0;
var GetManagedInstanceAnalyticContentRequest;
(function (GetManagedInstanceAnalyticContentRequest) {
    let ReportFormat;
    (function (ReportFormat) {
        ReportFormat["Csv"] = "csv";
        ReportFormat["Json"] = "json";
        ReportFormat["Xml"] = "xml";
    })(ReportFormat = GetManagedInstanceAnalyticContentRequest.ReportFormat || (GetManagedInstanceAnalyticContentRequest.ReportFormat = {}));
    let ReportType;
    (function (ReportType) {
        ReportType["Security"] = "SECURITY";
        ReportType["Bugfix"] = "BUGFIX";
        ReportType["Activity"] = "ACTIVITY";
        ReportType["All"] = "ALL";
    })(ReportType = GetManagedInstanceAnalyticContentRequest.ReportType || (GetManagedInstanceAnalyticContentRequest.ReportType = {}));
})(GetManagedInstanceAnalyticContentRequest = exports.GetManagedInstanceAnalyticContentRequest || (exports.GetManagedInstanceAnalyticContentRequest = {}));
