// oci-osmanagementhub/lib/request/list-events-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListEventsRequest = void 0;
var ListEventsRequest;
(function (ListEventsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["TimeOccurredAt"] = "timeOccurredAt";
        SortBy["TimeUpdated"] = "timeUpdated";
        SortBy["EventSummary"] = "eventSummary";
    })(SortBy = ListEventsRequest.SortBy || (ListEventsRequest.SortBy = {}));
})(ListEventsRequest = exports.ListEventsRequest || (exports.ListEventsRequest = {}));
