// oci-osmanagementhub/lib/request/list-managed-instance-group-installed-packages-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListManagedInstanceGroupInstalledPackagesRequest = void 0;
var ListManagedInstanceGroupInstalledPackagesRequest;
(function (ListManagedInstanceGroupInstalledPackagesRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["TimeInstalled"] = "timeInstalled";
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListManagedInstanceGroupInstalledPackagesRequest.SortBy || (ListManagedInstanceGroupInstalledPackagesRequest.SortBy = {}));
})(ListManagedInstanceGroupInstalledPackagesRequest = exports.ListManagedInstanceGroupInstalledPackagesRequest || (exports.ListManagedInstanceGroupInstalledPackagesRequest = {}));
