// oci-fleetappsmanagement/lib/request/index.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListStepsRequest = exports.ListSchedulerJobsRequest = exports.ListSchedulerExecutionsRequest = exports.ListSchedulerDefinitionsRequest = exports.ListScheduledFleetsRequest = exports.ListResourcesRequest = exports.ListInventoryRecordsRequest = exports.ListExecutionsRequest = exports.ListOnboardingPoliciesRequest = void 0;
const ListOnboardingPoliciesRequest = __importStar(require("./list-onboarding-policies-request"));
exports.ListOnboardingPoliciesRequest = ListOnboardingPoliciesRequest.ListOnboardingPoliciesRequest;
const ListExecutionsRequest = __importStar(require("./list-executions-request"));
exports.ListExecutionsRequest = ListExecutionsRequest.ListExecutionsRequest;
const ListInventoryRecordsRequest = __importStar(require("./list-inventory-records-request"));
exports.ListInventoryRecordsRequest = ListInventoryRecordsRequest.ListInventoryRecordsRequest;
const ListResourcesRequest = __importStar(require("./list-resources-request"));
exports.ListResourcesRequest = ListResourcesRequest.ListResourcesRequest;
const ListScheduledFleetsRequest = __importStar(require("./list-scheduled-fleets-request"));
exports.ListScheduledFleetsRequest = ListScheduledFleetsRequest.ListScheduledFleetsRequest;
const ListSchedulerDefinitionsRequest = __importStar(require("./list-scheduler-definitions-request"));
exports.ListSchedulerDefinitionsRequest = ListSchedulerDefinitionsRequest.ListSchedulerDefinitionsRequest;
const ListSchedulerExecutionsRequest = __importStar(require("./list-scheduler-executions-request"));
exports.ListSchedulerExecutionsRequest = ListSchedulerExecutionsRequest.ListSchedulerExecutionsRequest;
const ListSchedulerJobsRequest = __importStar(require("./list-scheduler-jobs-request"));
exports.ListSchedulerJobsRequest = ListSchedulerJobsRequest.ListSchedulerJobsRequest;
const ListStepsRequest = __importStar(require("./list-steps-request"));
exports.ListStepsRequest = ListStepsRequest.ListStepsRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
