// oci-fleetappsmanagement/lib/request/list-inventory-records-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListInventoryRecordsRequest = void 0;
var ListInventoryRecordsRequest;
(function (ListInventoryRecordsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["TargetName"] = "targetName";
        SortBy["TargetProductName"] = "targetProductName";
        SortBy["TargetResourceId"] = "targetResourceId";
        SortBy["OsType"] = "osType";
        SortBy["Architecture"] = "architecture";
    })(SortBy = ListInventoryRecordsRequest.SortBy || (ListInventoryRecordsRequest.SortBy = {}));
})(ListInventoryRecordsRequest = exports.ListInventoryRecordsRequest || (exports.ListInventoryRecordsRequest = {}));
