// oci-identitydataplane/lib/model/context-variable.js
"use strict";
/**
 * Identity and Access Management Data Plane API
 * APIs for managing identity data plane services. For example, use this API to create a scoped-access security token. To manage identity domains (for example, creating or deleting an identity domain) or to manage resources (for example, users and groups) within the default identity domain, see [IAM API](https://docs.oracle.com/iaas/api/#/en/identity/).
 * OpenAPI spec version: v1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContextVariable = void 0;
var ContextVariable;
(function (ContextVariable) {
    let Type;
    (function (Type) {
        Type["String"] = "STRING";
        Type["Number"] = "NUMBER";
        Type["Entity"] = "ENTITY";
        Type["Boolean"] = "BOOLEAN";
        Type["List"] = "LIST";
    })(Type = ContextVariable.Type || (ContextVariable.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ContextVariable.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ContextVariable.getDeserializedJsonObj = getDeserializedJsonObj;
})(ContextVariable = exports.ContextVariable || (exports.ContextVariable = {}));
