// oci-apmsynthetics/lib/model/database-connection-type.js
"use strict";
/**
 * APM Availability Monitoring API
 * Use the APM Availability Monitoring API to query Scripts, Monitors, Dedicated Vantage Points and On-Premise Vantage Points resources. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseConnectionType = void 0;
/**
 * Type of database connection.
 **/
var DatabaseConnectionType;
(function (DatabaseConnectionType) {
    DatabaseConnectionType["CloudWallet"] = "CLOUD_WALLET";
    DatabaseConnectionType["CustomJdbc"] = "CUSTOM_JDBC";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DatabaseConnectionType["UnknownValue"] = "UNKNOWN_VALUE";
})(DatabaseConnectionType = exports.DatabaseConnectionType || (exports.DatabaseConnectionType = {}));
(function (DatabaseConnectionType) {
    function getJsonObj(obj) {
        return obj;
    }
    DatabaseConnectionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DatabaseConnectionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabaseConnectionType = exports.DatabaseConnectionType || (exports.DatabaseConnectionType = {}));
