// oci-apmsynthetics/lib/model/dns-record-type.js
"use strict";
/**
 * APM Availability Monitoring API
 * Use the APM Availability Monitoring API to query Scripts, Monitors, Dedicated Vantage Points and On-Premise Vantage Points resources. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DnsRecordType = void 0;
/**
 * DNS record type.
 **/
var DnsRecordType;
(function (DnsRecordType) {
    DnsRecordType["A"] = "A";
    DnsRecordType["Aaaa"] = "AAAA";
    DnsRecordType["Any"] = "ANY";
    DnsRecordType["Cname"] = "CNAME";
    DnsRecordType["Dnskey"] = "DNSKEY";
    DnsRecordType["Ds"] = "DS";
    DnsRecordType["Mx"] = "MX";
    DnsRecordType["Ns"] = "NS";
    DnsRecordType["Nsec"] = "NSEC";
    DnsRecordType["NullRec"] = "NULL_REC";
    DnsRecordType["Ptr"] = "PTR";
    DnsRecordType["Rrsig"] = "RRSIG";
    DnsRecordType["Soa"] = "SOA";
    DnsRecordType["Txt"] = "TXT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DnsRecordType["UnknownValue"] = "UNKNOWN_VALUE";
})(DnsRecordType = exports.DnsRecordType || (exports.DnsRecordType = {}));
(function (DnsRecordType) {
    function getJsonObj(obj) {
        return obj;
    }
    DnsRecordType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DnsRecordType.getDeserializedJsonObj = getDeserializedJsonObj;
})(DnsRecordType = exports.DnsRecordType || (exports.DnsRecordType = {}));
