// oci-apmsynthetics/lib/model/monitor-types.js
"use strict";
/**
 * APM Availability Monitoring API
 * Use the APM Availability Monitoring API to query Scripts, Monitors, Dedicated Vantage Points and On-Premise Vantage Points resources. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorTypes = void 0;
/**
 * Monitor type.
 **/
var MonitorTypes;
(function (MonitorTypes) {
    MonitorTypes["ScriptedBrowser"] = "SCRIPTED_BROWSER";
    MonitorTypes["Browser"] = "BROWSER";
    MonitorTypes["ScriptedRest"] = "SCRIPTED_REST";
    MonitorTypes["Rest"] = "REST";
    MonitorTypes["Network"] = "NETWORK";
    MonitorTypes["Dns"] = "DNS";
    MonitorTypes["Ftp"] = "FTP";
    MonitorTypes["Sql"] = "SQL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MonitorTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(MonitorTypes = exports.MonitorTypes || (exports.MonitorTypes = {}));
(function (MonitorTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    MonitorTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MonitorTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(MonitorTypes = exports.MonitorTypes || (exports.MonitorTypes = {}));
