// oci-apmsynthetics/lib/model/on-premise-vantage-point-worker-status.js
"use strict";
/**
 * APM Availability Monitoring API
 * Use the APM Availability Monitoring API to query Scripts, Monitors, Dedicated Vantage Points and On-Premise Vantage Points resources. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OnPremiseVantagePointWorkerStatus = void 0;
/**
 * On-premise VP worker status.
 **/
var OnPremiseVantagePointWorkerStatus;
(function (OnPremiseVantagePointWorkerStatus) {
    OnPremiseVantagePointWorkerStatus["Enabled"] = "ENABLED";
    OnPremiseVantagePointWorkerStatus["Disabled"] = "DISABLED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OnPremiseVantagePointWorkerStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(OnPremiseVantagePointWorkerStatus = exports.OnPremiseVantagePointWorkerStatus || (exports.OnPremiseVantagePointWorkerStatus = {}));
(function (OnPremiseVantagePointWorkerStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    OnPremiseVantagePointWorkerStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OnPremiseVantagePointWorkerStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(OnPremiseVantagePointWorkerStatus = exports.OnPremiseVantagePointWorkerStatus || (exports.OnPremiseVantagePointWorkerStatus = {}));
