// oci-apmsynthetics/lib/model/on-premise-vantage-point-worker-type.js
"use strict";
/**
 * APM Availability Monitoring API
 * Use the APM Availability Monitoring API to query Scripts, Monitors, Dedicated Vantage Points and On-Premise Vantage Points resources. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OnPremiseVantagePointWorkerType = void 0;
/**
 * Type of the On-premise vantage point worker.
 **/
var OnPremiseVantagePointWorkerType;
(function (OnPremiseVantagePointWorkerType) {
    OnPremiseVantagePointWorkerType["Docker"] = "DOCKER";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OnPremiseVantagePointWorkerType["UnknownValue"] = "UNKNOWN_VALUE";
})(OnPremiseVantagePointWorkerType = exports.OnPremiseVantagePointWorkerType || (exports.OnPremiseVantagePointWorkerType = {}));
(function (OnPremiseVantagePointWorkerType) {
    function getJsonObj(obj) {
        return obj;
    }
    OnPremiseVantagePointWorkerType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OnPremiseVantagePointWorkerType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OnPremiseVantagePointWorkerType = exports.OnPremiseVantagePointWorkerType || (exports.OnPremiseVantagePointWorkerType = {}));
