// oci-psql/lib/model/backup.js
"use strict";
/**
 * PGSQL Control Plane API
 * Use the OCI Database with PostgreSQL API to manage resources such as database systems, database nodes, backups, and configurations.
For information, see the user guide documentation for the [service](https://docs.oracle.com/iaas/Content/postgresql/home.htm).

 * OpenAPI spec version: 20220915
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Backup = void 0;
const model = __importStar(require("../model"));
var Backup;
(function (Backup) {
    let SourceType;
    (function (SourceType) {
        SourceType["Scheduled"] = "SCHEDULED";
        SourceType["Manual"] = "MANUAL";
        SourceType["Copied"] = "COPIED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SourceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(SourceType = Backup.SourceType || (Backup.SourceType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Backup.LifecycleState || (Backup.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dbSystemDetails": obj.dbSystemDetails
                ? model.DbSystemDetails.getJsonObj(obj.dbSystemDetails)
                : undefined,
            "sourceBackupDetails": obj.sourceBackupDetails
                ? model.SourceBackupDetails.getJsonObj(obj.sourceBackupDetails)
                : undefined,
            "copyStatus": obj.copyStatus
                ? obj.copyStatus.map(item => {
                    return model.BackupCopyStatusDetails.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    Backup.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dbSystemDetails": obj.dbSystemDetails
                ? model.DbSystemDetails.getDeserializedJsonObj(obj.dbSystemDetails)
                : undefined,
            "sourceBackupDetails": obj.sourceBackupDetails
                ? model.SourceBackupDetails.getDeserializedJsonObj(obj.sourceBackupDetails)
                : undefined,
            "copyStatus": obj.copyStatus
                ? obj.copyStatus.map(item => {
                    return model.BackupCopyStatusDetails.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    Backup.getDeserializedJsonObj = getDeserializedJsonObj;
})(Backup = exports.Backup || (exports.Backup = {}));
