// oci-psql/lib/model/db-system.js
"use strict";
/**
 * PGSQL Control Plane API
 * Use the OCI Database with PostgreSQL API to manage resources such as database systems, database nodes, backups, and configurations.
For information, see the user guide documentation for the [service](https://docs.oracle.com/iaas/Content/postgresql/home.htm).

 * OpenAPI spec version: 20220915
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbSystem = void 0;
const model = __importStar(require("../model"));
var DbSystem;
(function (DbSystem) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = DbSystem.LifecycleState || (DbSystem.LifecycleState = {}));
    let SystemType;
    (function (SystemType) {
        SystemType["OciOptimizedStorage"] = "OCI_OPTIMIZED_STORAGE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SystemType["UnknownValue"] = "UNKNOWN_VALUE";
    })(SystemType = DbSystem.SystemType || (DbSystem.SystemType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "instances": obj.instances
                ? obj.instances.map(item => {
                    return model.DbInstance.getJsonObj(item);
                })
                : undefined,
            "storageDetails": obj.storageDetails
                ? model.StorageDetails.getJsonObj(obj.storageDetails)
                : undefined,
            "networkDetails": obj.networkDetails
                ? model.NetworkDetails.getJsonObj(obj.networkDetails)
                : undefined,
            "managementPolicy": obj.managementPolicy
                ? model.ManagementPolicy.getJsonObj(obj.managementPolicy)
                : undefined,
            "source": obj.source ? model.SourceDetails.getJsonObj(obj.source) : undefined
        });
        return jsonObj;
    }
    DbSystem.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "instances": obj.instances
                ? obj.instances.map(item => {
                    return model.DbInstance.getDeserializedJsonObj(item);
                })
                : undefined,
            "storageDetails": obj.storageDetails
                ? model.StorageDetails.getDeserializedJsonObj(obj.storageDetails)
                : undefined,
            "networkDetails": obj.networkDetails
                ? model.NetworkDetails.getDeserializedJsonObj(obj.networkDetails)
                : undefined,
            "managementPolicy": obj.managementPolicy
                ? model.ManagementPolicy.getDeserializedJsonObj(obj.managementPolicy)
                : undefined,
            "source": obj.source ? model.SourceDetails.getDeserializedJsonObj(obj.source) : undefined
        });
        return jsonObj;
    }
    DbSystem.getDeserializedJsonObj = getDeserializedJsonObj;
})(DbSystem = exports.DbSystem || (exports.DbSystem = {}));
