// oci-psql/lib/model/index.js
"use strict";
/**
 * PGSQL Control Plane API
 * Use the OCI Database with PostgreSQL API to manage resources such as database systems, database nodes, backups, and configurations.
For information, see the user guide documentation for the [service](https://docs.oracle.com/iaas/Content/postgresql/home.htm).

 * OpenAPI spec version: 20220915
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShapeCollection = exports.RestoreDbSystemDetails = exports.RestartDbInstanceInDbSystemDetails = exports.ResetMasterUserPasswordDetails = exports.PrimaryDbInstanceDetails = exports.PatchInstruction = exports.PatchDbSystemDetails = exports.PasswordDetails = exports.OperationType = exports.OperationStatus = exports.NetworkDetails = exports.ManagementPolicyDetails = exports.ManagementPolicy = exports.FailoverDbSystemDetails = exports.Endpoint = exports.DefaultConfigurationSummary = exports.DefaultConfigurationDetails = exports.DefaultConfigurationCollection = exports.DefaultConfiguration = exports.DefaultConfigParams = exports.DbSystemSummary = exports.DbSystemDetails = exports.DbSystemCollection = exports.DbSystem = exports.DbInstanceEndpoint = exports.DbInstance = exports.DbConfigurationOverrideCollection = exports.Credentials = exports.CreateDbSystemDetails = exports.CreateDbInstanceDetails = exports.CreateConfigurationDetails = exports.CreateBackupDetails = exports.ConnectionDetails = exports.ConfigurationSummary = exports.ConfigurationDetails = exports.ConfigurationCollection = exports.Configuration = exports.ConfigParams = exports.ConfigOverrides = exports.ChangeDbSystemCompartmentDetails = exports.ChangeConfigurationCompartmentDetails = exports.ChangeBackupCompartmentDetails = exports.BackupSummary = exports.BackupPolicy = exports.BackupCopyStatusDetails = exports.BackupCopyPolicy = exports.BackupCopyDetails = exports.BackupCollection = exports.Backup = exports.ActionType = void 0;
exports.WeeklyBackupPolicy = exports.VaultSecretPasswordDetails = exports.PlainTextPasswordDetails = exports.PatchRequireInstruction = exports.PatchReplaceInstruction = exports.PatchRemoveInstruction = exports.PatchProhibitInstruction = exports.PatchMoveInstruction = exports.PatchMergeInstruction = exports.PatchInsertInstruction = exports.OciOptimizedStorageDetails = exports.NoneSourceDetails = exports.NoneBackupPolicy = exports.MonthlyBackupPolicy = exports.DailyBackupPolicy = exports.BackupSourceDetails = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateStorageDetailsParams = exports.UpdateNetworkDetails = exports.UpdateDbSystemDetails = exports.UpdateDbSystemDbInstanceDetails = exports.UpdateDbConfigParams = exports.UpdateConfigurationDetails = exports.UpdateBackupDetails = exports.StorageDetails = exports.SourceDetails = exports.SourceBackupDetails = exports.SortOrder = exports.ShapeSummary = exports.ShapeOcpuOptions = exports.ShapeMemoryOptions = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const Backup = __importStar(require("./backup"));
exports.Backup = Backup.Backup;
const BackupCollection = __importStar(require("./backup-collection"));
exports.BackupCollection = BackupCollection.BackupCollection;
const BackupCopyDetails = __importStar(require("./backup-copy-details"));
exports.BackupCopyDetails = BackupCopyDetails.BackupCopyDetails;
const BackupCopyPolicy = __importStar(require("./backup-copy-policy"));
exports.BackupCopyPolicy = BackupCopyPolicy.BackupCopyPolicy;
const BackupCopyStatusDetails = __importStar(require("./backup-copy-status-details"));
exports.BackupCopyStatusDetails = BackupCopyStatusDetails.BackupCopyStatusDetails;
const BackupPolicy = __importStar(require("./backup-policy"));
exports.BackupPolicy = BackupPolicy.BackupPolicy;
const BackupSummary = __importStar(require("./backup-summary"));
exports.BackupSummary = BackupSummary.BackupSummary;
const ChangeBackupCompartmentDetails = __importStar(require("./change-backup-compartment-details"));
exports.ChangeBackupCompartmentDetails = ChangeBackupCompartmentDetails.ChangeBackupCompartmentDetails;
const ChangeConfigurationCompartmentDetails = __importStar(require("./change-configuration-compartment-details"));
exports.ChangeConfigurationCompartmentDetails = ChangeConfigurationCompartmentDetails.ChangeConfigurationCompartmentDetails;
const ChangeDbSystemCompartmentDetails = __importStar(require("./change-db-system-compartment-details"));
exports.ChangeDbSystemCompartmentDetails = ChangeDbSystemCompartmentDetails.ChangeDbSystemCompartmentDetails;
const ConfigOverrides = __importStar(require("./config-overrides"));
exports.ConfigOverrides = ConfigOverrides.ConfigOverrides;
const ConfigParams = __importStar(require("./config-params"));
exports.ConfigParams = ConfigParams.ConfigParams;
const Configuration = __importStar(require("./configuration"));
exports.Configuration = Configuration.Configuration;
const ConfigurationCollection = __importStar(require("./configuration-collection"));
exports.ConfigurationCollection = ConfigurationCollection.ConfigurationCollection;
const ConfigurationDetails = __importStar(require("./configuration-details"));
exports.ConfigurationDetails = ConfigurationDetails.ConfigurationDetails;
const ConfigurationSummary = __importStar(require("./configuration-summary"));
exports.ConfigurationSummary = ConfigurationSummary.ConfigurationSummary;
const ConnectionDetails = __importStar(require("./connection-details"));
exports.ConnectionDetails = ConnectionDetails.ConnectionDetails;
const CreateBackupDetails = __importStar(require("./create-backup-details"));
exports.CreateBackupDetails = CreateBackupDetails.CreateBackupDetails;
const CreateConfigurationDetails = __importStar(require("./create-configuration-details"));
exports.CreateConfigurationDetails = CreateConfigurationDetails.CreateConfigurationDetails;
const CreateDbInstanceDetails = __importStar(require("./create-db-instance-details"));
exports.CreateDbInstanceDetails = CreateDbInstanceDetails.CreateDbInstanceDetails;
const CreateDbSystemDetails = __importStar(require("./create-db-system-details"));
exports.CreateDbSystemDetails = CreateDbSystemDetails.CreateDbSystemDetails;
const Credentials = __importStar(require("./credentials"));
exports.Credentials = Credentials.Credentials;
const DbConfigurationOverrideCollection = __importStar(require("./db-configuration-override-collection"));
exports.DbConfigurationOverrideCollection = DbConfigurationOverrideCollection.DbConfigurationOverrideCollection;
const DbInstance = __importStar(require("./db-instance"));
exports.DbInstance = DbInstance.DbInstance;
const DbInstanceEndpoint = __importStar(require("./db-instance-endpoint"));
exports.DbInstanceEndpoint = DbInstanceEndpoint.DbInstanceEndpoint;
const DbSystem = __importStar(require("./db-system"));
exports.DbSystem = DbSystem.DbSystem;
const DbSystemCollection = __importStar(require("./db-system-collection"));
exports.DbSystemCollection = DbSystemCollection.DbSystemCollection;
const DbSystemDetails = __importStar(require("./db-system-details"));
exports.DbSystemDetails = DbSystemDetails.DbSystemDetails;
const DbSystemSummary = __importStar(require("./db-system-summary"));
exports.DbSystemSummary = DbSystemSummary.DbSystemSummary;
const DefaultConfigParams = __importStar(require("./default-config-params"));
exports.DefaultConfigParams = DefaultConfigParams.DefaultConfigParams;
const DefaultConfiguration = __importStar(require("./default-configuration"));
exports.DefaultConfiguration = DefaultConfiguration.DefaultConfiguration;
const DefaultConfigurationCollection = __importStar(require("./default-configuration-collection"));
exports.DefaultConfigurationCollection = DefaultConfigurationCollection.DefaultConfigurationCollection;
const DefaultConfigurationDetails = __importStar(require("./default-configuration-details"));
exports.DefaultConfigurationDetails = DefaultConfigurationDetails.DefaultConfigurationDetails;
const DefaultConfigurationSummary = __importStar(require("./default-configuration-summary"));
exports.DefaultConfigurationSummary = DefaultConfigurationSummary.DefaultConfigurationSummary;
const Endpoint = __importStar(require("./endpoint"));
exports.Endpoint = Endpoint.Endpoint;
const FailoverDbSystemDetails = __importStar(require("./failover-db-system-details"));
exports.FailoverDbSystemDetails = FailoverDbSystemDetails.FailoverDbSystemDetails;
const ManagementPolicy = __importStar(require("./management-policy"));
exports.ManagementPolicy = ManagementPolicy.ManagementPolicy;
const ManagementPolicyDetails = __importStar(require("./management-policy-details"));
exports.ManagementPolicyDetails = ManagementPolicyDetails.ManagementPolicyDetails;
const NetworkDetails = __importStar(require("./network-details"));
exports.NetworkDetails = NetworkDetails.NetworkDetails;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PasswordDetails = __importStar(require("./password-details"));
exports.PasswordDetails = PasswordDetails.PasswordDetails;
const PatchDbSystemDetails = __importStar(require("./patch-db-system-details"));
exports.PatchDbSystemDetails = PatchDbSystemDetails.PatchDbSystemDetails;
const PatchInstruction = __importStar(require("./patch-instruction"));
exports.PatchInstruction = PatchInstruction.PatchInstruction;
const PrimaryDbInstanceDetails = __importStar(require("./primary-db-instance-details"));
exports.PrimaryDbInstanceDetails = PrimaryDbInstanceDetails.PrimaryDbInstanceDetails;
const ResetMasterUserPasswordDetails = __importStar(require("./reset-master-user-password-details"));
exports.ResetMasterUserPasswordDetails = ResetMasterUserPasswordDetails.ResetMasterUserPasswordDetails;
const RestartDbInstanceInDbSystemDetails = __importStar(require("./restart-db-instance-in-db-system-details"));
exports.RestartDbInstanceInDbSystemDetails = RestartDbInstanceInDbSystemDetails.RestartDbInstanceInDbSystemDetails;
const RestoreDbSystemDetails = __importStar(require("./restore-db-system-details"));
exports.RestoreDbSystemDetails = RestoreDbSystemDetails.RestoreDbSystemDetails;
const ShapeCollection = __importStar(require("./shape-collection"));
exports.ShapeCollection = ShapeCollection.ShapeCollection;
const ShapeMemoryOptions = __importStar(require("./shape-memory-options"));
exports.ShapeMemoryOptions = ShapeMemoryOptions.ShapeMemoryOptions;
const ShapeOcpuOptions = __importStar(require("./shape-ocpu-options"));
exports.ShapeOcpuOptions = ShapeOcpuOptions.ShapeOcpuOptions;
const ShapeSummary = __importStar(require("./shape-summary"));
exports.ShapeSummary = ShapeSummary.ShapeSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const SourceBackupDetails = __importStar(require("./source-backup-details"));
exports.SourceBackupDetails = SourceBackupDetails.SourceBackupDetails;
const SourceDetails = __importStar(require("./source-details"));
exports.SourceDetails = SourceDetails.SourceDetails;
const StorageDetails = __importStar(require("./storage-details"));
exports.StorageDetails = StorageDetails.StorageDetails;
const UpdateBackupDetails = __importStar(require("./update-backup-details"));
exports.UpdateBackupDetails = UpdateBackupDetails.UpdateBackupDetails;
const UpdateConfigurationDetails = __importStar(require("./update-configuration-details"));
exports.UpdateConfigurationDetails = UpdateConfigurationDetails.UpdateConfigurationDetails;
const UpdateDbConfigParams = __importStar(require("./update-db-config-params"));
exports.UpdateDbConfigParams = UpdateDbConfigParams.UpdateDbConfigParams;
const UpdateDbSystemDbInstanceDetails = __importStar(require("./update-db-system-db-instance-details"));
exports.UpdateDbSystemDbInstanceDetails = UpdateDbSystemDbInstanceDetails.UpdateDbSystemDbInstanceDetails;
const UpdateDbSystemDetails = __importStar(require("./update-db-system-details"));
exports.UpdateDbSystemDetails = UpdateDbSystemDetails.UpdateDbSystemDetails;
const UpdateNetworkDetails = __importStar(require("./update-network-details"));
exports.UpdateNetworkDetails = UpdateNetworkDetails.UpdateNetworkDetails;
const UpdateStorageDetailsParams = __importStar(require("./update-storage-details-params"));
exports.UpdateStorageDetailsParams = UpdateStorageDetailsParams.UpdateStorageDetailsParams;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const BackupSourceDetails = __importStar(require("./backup-source-details"));
exports.BackupSourceDetails = BackupSourceDetails.BackupSourceDetails;
const DailyBackupPolicy = __importStar(require("./daily-backup-policy"));
exports.DailyBackupPolicy = DailyBackupPolicy.DailyBackupPolicy;
const MonthlyBackupPolicy = __importStar(require("./monthly-backup-policy"));
exports.MonthlyBackupPolicy = MonthlyBackupPolicy.MonthlyBackupPolicy;
const NoneBackupPolicy = __importStar(require("./none-backup-policy"));
exports.NoneBackupPolicy = NoneBackupPolicy.NoneBackupPolicy;
const NoneSourceDetails = __importStar(require("./none-source-details"));
exports.NoneSourceDetails = NoneSourceDetails.NoneSourceDetails;
const OciOptimizedStorageDetails = __importStar(require("./oci-optimized-storage-details"));
exports.OciOptimizedStorageDetails = OciOptimizedStorageDetails.OciOptimizedStorageDetails;
const PatchInsertInstruction = __importStar(require("./patch-insert-instruction"));
exports.PatchInsertInstruction = PatchInsertInstruction.PatchInsertInstruction;
const PatchMergeInstruction = __importStar(require("./patch-merge-instruction"));
exports.PatchMergeInstruction = PatchMergeInstruction.PatchMergeInstruction;
const PatchMoveInstruction = __importStar(require("./patch-move-instruction"));
exports.PatchMoveInstruction = PatchMoveInstruction.PatchMoveInstruction;
const PatchProhibitInstruction = __importStar(require("./patch-prohibit-instruction"));
exports.PatchProhibitInstruction = PatchProhibitInstruction.PatchProhibitInstruction;
const PatchRemoveInstruction = __importStar(require("./patch-remove-instruction"));
exports.PatchRemoveInstruction = PatchRemoveInstruction.PatchRemoveInstruction;
const PatchReplaceInstruction = __importStar(require("./patch-replace-instruction"));
exports.PatchReplaceInstruction = PatchReplaceInstruction.PatchReplaceInstruction;
const PatchRequireInstruction = __importStar(require("./patch-require-instruction"));
exports.PatchRequireInstruction = PatchRequireInstruction.PatchRequireInstruction;
const PlainTextPasswordDetails = __importStar(require("./plain-text-password-details"));
exports.PlainTextPasswordDetails = PlainTextPasswordDetails.PlainTextPasswordDetails;
const VaultSecretPasswordDetails = __importStar(require("./vault-secret-password-details"));
exports.VaultSecretPasswordDetails = VaultSecretPasswordDetails.VaultSecretPasswordDetails;
const WeeklyBackupPolicy = __importStar(require("./weekly-backup-policy"));
exports.WeeklyBackupPolicy = WeeklyBackupPolicy.WeeklyBackupPolicy;
