// oci-psql/lib/model/operation-type.js
"use strict";
/**
 * PGSQL Control Plane API
 * Use the OCI Database with PostgreSQL API to manage resources such as database systems, database nodes, backups, and configurations.
For information, see the user guide documentation for the [service](https://docs.oracle.com/iaas/Content/postgresql/home.htm).

 * OpenAPI spec version: 20220915
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreatePostgresqlDbSystem"] = "CREATE_POSTGRESQL_DB_SYSTEM";
    OperationType["UpdatePostgresqlDbSystem"] = "UPDATE_POSTGRESQL_DB_SYSTEM";
    OperationType["DeletePostgresqlDbSystem"] = "DELETE_POSTGRESQL_DB_SYSTEM";
    OperationType["MovePostgresqlDbSystem"] = "MOVE_POSTGRESQL_DB_SYSTEM";
    OperationType["CreatePostgresqlDbSystemBackup"] = "CREATE_POSTGRESQL_DB_SYSTEM_BACKUP";
    OperationType["UpdatePostgresqlDbSystemBackup"] = "UPDATE_POSTGRESQL_DB_SYSTEM_BACKUP";
    OperationType["DeletePostgresqlDbSystemBackup"] = "DELETE_POSTGRESQL_DB_SYSTEM_BACKUP";
    OperationType["MovePostgresqlDbSystemBackup"] = "MOVE_POSTGRESQL_DB_SYSTEM_BACKUP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
