// oci-psql/lib/model/update-db-config-params.js
"use strict";
/**
 * PGSQL Control Plane API
 * Use the OCI Database with PostgreSQL API to manage resources such as database systems, database nodes, backups, and configurations.
For information, see the user guide documentation for the [service](https://docs.oracle.com/iaas/Content/postgresql/home.htm).

 * OpenAPI spec version: 20220915
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDbConfigParams = void 0;
var UpdateDbConfigParams;
(function (UpdateDbConfigParams) {
    let ApplyConfig;
    (function (ApplyConfig) {
        ApplyConfig["Restart"] = "RESTART";
        ApplyConfig["Reload"] = "RELOAD";
    })(ApplyConfig = UpdateDbConfigParams.ApplyConfig || (UpdateDbConfigParams.ApplyConfig = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateDbConfigParams.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateDbConfigParams.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateDbConfigParams = exports.UpdateDbConfigParams || (exports.UpdateDbConfigParams = {}));
