// oci-marketplacepublisher/lib/request/index.js
"use strict";
/**
 * MarketplacePublisherService API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListOffersRequest = exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListTermsRequest = exports.ListTermVersionsRequest = exports.ListPublishersRequest = exports.ListProductsRequest = exports.ListMarketsRequest = exports.ListListingsRequest = exports.ListListingRevisionsRequest = exports.ListListingRevisionPackagesRequest = exports.ListListingRevisionNotesRequest = exports.ListListingRevisionAttachmentsRequest = exports.ListCategoriesRequest = exports.ListArtifactsRequest = exports.ListAttachmentsRequest = void 0;
const ListAttachmentsRequest = __importStar(require("./list-attachments-request"));
exports.ListAttachmentsRequest = ListAttachmentsRequest.ListAttachmentsRequest;
const ListArtifactsRequest = __importStar(require("./list-artifacts-request"));
exports.ListArtifactsRequest = ListArtifactsRequest.ListArtifactsRequest;
const ListCategoriesRequest = __importStar(require("./list-categories-request"));
exports.ListCategoriesRequest = ListCategoriesRequest.ListCategoriesRequest;
const ListListingRevisionAttachmentsRequest = __importStar(require("./list-listing-revision-attachments-request"));
exports.ListListingRevisionAttachmentsRequest = ListListingRevisionAttachmentsRequest.ListListingRevisionAttachmentsRequest;
const ListListingRevisionNotesRequest = __importStar(require("./list-listing-revision-notes-request"));
exports.ListListingRevisionNotesRequest = ListListingRevisionNotesRequest.ListListingRevisionNotesRequest;
const ListListingRevisionPackagesRequest = __importStar(require("./list-listing-revision-packages-request"));
exports.ListListingRevisionPackagesRequest = ListListingRevisionPackagesRequest.ListListingRevisionPackagesRequest;
const ListListingRevisionsRequest = __importStar(require("./list-listing-revisions-request"));
exports.ListListingRevisionsRequest = ListListingRevisionsRequest.ListListingRevisionsRequest;
const ListListingsRequest = __importStar(require("./list-listings-request"));
exports.ListListingsRequest = ListListingsRequest.ListListingsRequest;
const ListMarketsRequest = __importStar(require("./list-markets-request"));
exports.ListMarketsRequest = ListMarketsRequest.ListMarketsRequest;
const ListProductsRequest = __importStar(require("./list-products-request"));
exports.ListProductsRequest = ListProductsRequest.ListProductsRequest;
const ListPublishersRequest = __importStar(require("./list-publishers-request"));
exports.ListPublishersRequest = ListPublishersRequest.ListPublishersRequest;
const ListTermVersionsRequest = __importStar(require("./list-term-versions-request"));
exports.ListTermVersionsRequest = ListTermVersionsRequest.ListTermVersionsRequest;
const ListTermsRequest = __importStar(require("./list-terms-request"));
exports.ListTermsRequest = ListTermsRequest.ListTermsRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
const ListOffersRequest = __importStar(require("./list-offers-request"));
exports.ListOffersRequest = ListOffersRequest.ListOffersRequest;
