// oci-nosql/lib/model/index.js
"use strict";
/**
 * NoSQL Database API
 * The control plane API for NoSQL Database Cloud Service HTTPS
provides endpoints to perform NDCS operations, including creation
and deletion of tables and indexes; population and access of data
in tables; and access of table usage metrics.

 * OpenAPI spec version: 20190828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateMultiTenancyConfigurationDetails = exports.UpdateHostedConfigurationDetails = exports.MultiTenancyConfiguration = exports.HostedConfiguration = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.UpdateTableDetails = exports.UpdateRowResult = exports.UpdateRowDetails = exports.UpdateConfigurationDetails = exports.TableUsageSummary = exports.TableUsageCollection = exports.TableSummary = exports.TableLimits = exports.TableCollection = exports.Table = exports.StatementSummary = exports.Schema = exports.Row = exports.RequestUsage = exports.Replica = exports.QueryResultCollection = exports.QueryDetails = exports.PreparedStatement = exports.KmsKey = exports.IndexSummary = exports.IndexKey = exports.IndexCollection = exports.Index = exports.Identity = exports.DeleteRowResult = exports.CreateTableDetails = exports.CreateReplicaDetails = exports.CreateIndexDetails = exports.Configuration = exports.Column = exports.ChangeTableCompartmentDetails = void 0;
const ChangeTableCompartmentDetails = __importStar(require("./change-table-compartment-details"));
exports.ChangeTableCompartmentDetails = ChangeTableCompartmentDetails.ChangeTableCompartmentDetails;
const Column = __importStar(require("./column"));
exports.Column = Column.Column;
const Configuration = __importStar(require("./configuration"));
exports.Configuration = Configuration.Configuration;
const CreateIndexDetails = __importStar(require("./create-index-details"));
exports.CreateIndexDetails = CreateIndexDetails.CreateIndexDetails;
const CreateReplicaDetails = __importStar(require("./create-replica-details"));
exports.CreateReplicaDetails = CreateReplicaDetails.CreateReplicaDetails;
const CreateTableDetails = __importStar(require("./create-table-details"));
exports.CreateTableDetails = CreateTableDetails.CreateTableDetails;
const DeleteRowResult = __importStar(require("./delete-row-result"));
exports.DeleteRowResult = DeleteRowResult.DeleteRowResult;
const Identity = __importStar(require("./identity"));
exports.Identity = Identity.Identity;
const Index = __importStar(require("./model-index"));
exports.Index = Index.Index;
const IndexCollection = __importStar(require("./index-collection"));
exports.IndexCollection = IndexCollection.IndexCollection;
const IndexKey = __importStar(require("./index-key"));
exports.IndexKey = IndexKey.IndexKey;
const IndexSummary = __importStar(require("./index-summary"));
exports.IndexSummary = IndexSummary.IndexSummary;
const KmsKey = __importStar(require("./kms-key"));
exports.KmsKey = KmsKey.KmsKey;
const PreparedStatement = __importStar(require("./prepared-statement"));
exports.PreparedStatement = PreparedStatement.PreparedStatement;
const QueryDetails = __importStar(require("./query-details"));
exports.QueryDetails = QueryDetails.QueryDetails;
const QueryResultCollection = __importStar(require("./query-result-collection"));
exports.QueryResultCollection = QueryResultCollection.QueryResultCollection;
const Replica = __importStar(require("./replica"));
exports.Replica = Replica.Replica;
const RequestUsage = __importStar(require("./request-usage"));
exports.RequestUsage = RequestUsage.RequestUsage;
const Row = __importStar(require("./row"));
exports.Row = Row.Row;
const Schema = __importStar(require("./schema"));
exports.Schema = Schema.Schema;
const StatementSummary = __importStar(require("./statement-summary"));
exports.StatementSummary = StatementSummary.StatementSummary;
const Table = __importStar(require("./table"));
exports.Table = Table.Table;
const TableCollection = __importStar(require("./table-collection"));
exports.TableCollection = TableCollection.TableCollection;
const TableLimits = __importStar(require("./table-limits"));
exports.TableLimits = TableLimits.TableLimits;
const TableSummary = __importStar(require("./table-summary"));
exports.TableSummary = TableSummary.TableSummary;
const TableUsageCollection = __importStar(require("./table-usage-collection"));
exports.TableUsageCollection = TableUsageCollection.TableUsageCollection;
const TableUsageSummary = __importStar(require("./table-usage-summary"));
exports.TableUsageSummary = TableUsageSummary.TableUsageSummary;
const UpdateConfigurationDetails = __importStar(require("./update-configuration-details"));
exports.UpdateConfigurationDetails = UpdateConfigurationDetails.UpdateConfigurationDetails;
const UpdateRowDetails = __importStar(require("./update-row-details"));
exports.UpdateRowDetails = UpdateRowDetails.UpdateRowDetails;
const UpdateRowResult = __importStar(require("./update-row-result"));
exports.UpdateRowResult = UpdateRowResult.UpdateRowResult;
const UpdateTableDetails = __importStar(require("./update-table-details"));
exports.UpdateTableDetails = UpdateTableDetails.UpdateTableDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const HostedConfiguration = __importStar(require("./hosted-configuration"));
exports.HostedConfiguration = HostedConfiguration.HostedConfiguration;
const MultiTenancyConfiguration = __importStar(require("./multi-tenancy-configuration"));
exports.MultiTenancyConfiguration = MultiTenancyConfiguration.MultiTenancyConfiguration;
const UpdateHostedConfigurationDetails = __importStar(require("./update-hosted-configuration-details"));
exports.UpdateHostedConfigurationDetails = UpdateHostedConfigurationDetails.UpdateHostedConfigurationDetails;
const UpdateMultiTenancyConfigurationDetails = __importStar(require("./update-multi-tenancy-configuration-details"));
exports.UpdateMultiTenancyConfigurationDetails = UpdateMultiTenancyConfigurationDetails.UpdateMultiTenancyConfigurationDetails;
