// oci-nosql/lib/model/kms-key.js
"use strict";
/**
 * NoSQL Database API
 * The control plane API for NoSQL Database Cloud Service HTTPS
provides endpoints to perform NDCS operations, including creation
and deletion of tables and indexes; population and access of data
in tables; and access of table usage metrics.

 * OpenAPI spec version: 20190828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KmsKey = void 0;
var KmsKey;
(function (KmsKey) {
    let KmsKeyState;
    (function (KmsKeyState) {
        KmsKeyState["Updating"] = "UPDATING";
        KmsKeyState["Active"] = "ACTIVE";
        KmsKeyState["Deleted"] = "DELETED";
        KmsKeyState["Failed"] = "FAILED";
        KmsKeyState["Reverting"] = "REVERTING";
        KmsKeyState["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        KmsKeyState["UnknownValue"] = "UNKNOWN_VALUE";
    })(KmsKeyState = KmsKey.KmsKeyState || (KmsKey.KmsKeyState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    KmsKey.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    KmsKey.getDeserializedJsonObj = getDeserializedJsonObj;
})(KmsKey = exports.KmsKey || (exports.KmsKey = {}));
