// oci-analytics/lib/model/feature-bundle.js
"use strict";
/**
 * Analytics API
 * Analytics API.

 * OpenAPI spec version: 20190331
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeatureBundle = void 0;
/**
 * Feature Bundle Enum.
 *
 **/
var FeatureBundle;
(function (FeatureBundle) {
    FeatureBundle["FawPaid"] = "FAW_PAID";
    FeatureBundle["FawFree"] = "FAW_FREE";
    FeatureBundle["EeEmbedded"] = "EE_EMBEDDED";
    FeatureBundle["SeEmbedded"] = "SE_EMBEDDED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    FeatureBundle["UnknownValue"] = "UNKNOWN_VALUE";
})(FeatureBundle = exports.FeatureBundle || (exports.FeatureBundle = {}));
(function (FeatureBundle) {
    function getJsonObj(obj) {
        return obj;
    }
    FeatureBundle.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    FeatureBundle.getDeserializedJsonObj = getDeserializedJsonObj;
})(FeatureBundle = exports.FeatureBundle || (exports.FeatureBundle = {}));
