// oci-monitoring/lib/model/alarm-suppression.js
"use strict";
/**
 * Monitoring API
 * Use the Monitoring API to manage metric queries and alarms for assessing the health, capacity, and performance of your cloud resources.
Endpoints vary by operation. For PostMetricData, use the {@code telemetry-ingestion} endpoints; for all other operations, use the {@code telemetry} endpoints.
For more information, see
[the Monitoring documentation](https://docs.oracle.com/iaas/Content/Monitoring/home.htm).

 * OpenAPI spec version: 20180401
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlarmSuppression = void 0;
const model = __importStar(require("../model"));
var AlarmSuppression;
(function (AlarmSuppression) {
    let Level;
    (function (Level) {
        Level["Alarm"] = "ALARM";
        Level["Dimension"] = "DIMENSION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Level["UnknownValue"] = "UNKNOWN_VALUE";
    })(Level = AlarmSuppression.Level || (AlarmSuppression.Level = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = AlarmSuppression.LifecycleState || (AlarmSuppression.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "alarmSuppressionTarget": obj.alarmSuppressionTarget
                ? model.AlarmSuppressionTarget.getJsonObj(obj.alarmSuppressionTarget)
                : undefined,
            "suppressionConditions": obj.suppressionConditions
                ? obj.suppressionConditions.map(item => {
                    return model.SuppressionCondition.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    AlarmSuppression.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "alarmSuppressionTarget": obj.alarmSuppressionTarget
                ? model.AlarmSuppressionTarget.getDeserializedJsonObj(obj.alarmSuppressionTarget)
                : undefined,
            "suppressionConditions": obj.suppressionConditions
                ? obj.suppressionConditions.map(item => {
                    return model.SuppressionCondition.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    AlarmSuppression.getDeserializedJsonObj = getDeserializedJsonObj;
})(AlarmSuppression = exports.AlarmSuppression || (exports.AlarmSuppression = {}));
