// oci-databasemigration/lib/model/advanced-parameter-data-types.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20230518
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdvancedParameterDataTypes = void 0;
/**
 * Advanced Parameter Data types.
 *
 **/
var AdvancedParameterDataTypes;
(function (AdvancedParameterDataTypes) {
    AdvancedParameterDataTypes["String"] = "STRING";
    AdvancedParameterDataTypes["Integer"] = "INTEGER";
    AdvancedParameterDataTypes["Float"] = "FLOAT";
    AdvancedParameterDataTypes["Boolean"] = "BOOLEAN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AdvancedParameterDataTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(AdvancedParameterDataTypes = exports.AdvancedParameterDataTypes || (exports.AdvancedParameterDataTypes = {}));
(function (AdvancedParameterDataTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    AdvancedParameterDataTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AdvancedParameterDataTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(AdvancedParameterDataTypes = exports.AdvancedParameterDataTypes || (exports.AdvancedParameterDataTypes = {}));
