// oci-databasemigration/lib/model/connection.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20230518
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Connection = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var Connection;
(function (Connection) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Connection.LifecycleState || (Connection.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "ingressIps": obj.ingressIps
                ? obj.ingressIps.map(item => {
                    return model.IngressIpDetails.getJsonObj(item);
                })
                : undefined
        });
        if (obj && "connectionType" in obj && obj.connectionType) {
            switch (obj.connectionType) {
                case "MYSQL":
                    return model.MysqlConnection.getJsonObj(jsonObj, true);
                case "ORACLE":
                    return model.OracleConnection.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.connectionType}`);
            }
        }
        return jsonObj;
    }
    Connection.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "ingressIps": obj.ingressIps
                ? obj.ingressIps.map(item => {
                    return model.IngressIpDetails.getDeserializedJsonObj(item);
                })
                : undefined
        });
        if (obj && "connectionType" in obj && obj.connectionType) {
            switch (obj.connectionType) {
                case "MYSQL":
                    return model.MysqlConnection.getDeserializedJsonObj(jsonObj, true);
                case "ORACLE":
                    return model.OracleConnection.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.connectionType}`);
            }
        }
        return jsonObj;
    }
    Connection.getDeserializedJsonObj = getDeserializedJsonObj;
})(Connection = exports.Connection || (exports.Connection = {}));
