// oci-databasemigration/lib/model/data-pump-tablespace-block-sizes-in-kb.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20230518
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataPumpTablespaceBlockSizesInKb = void 0;
/**
 * Size of Oracle database blocks in KB.
 **/
var DataPumpTablespaceBlockSizesInKb;
(function (DataPumpTablespaceBlockSizesInKb) {
    DataPumpTablespaceBlockSizesInKb["Size2K"] = "SIZE_2K";
    DataPumpTablespaceBlockSizesInKb["Size8K"] = "SIZE_8K";
    DataPumpTablespaceBlockSizesInKb["Size16K"] = "SIZE_16K";
    DataPumpTablespaceBlockSizesInKb["Size32K"] = "SIZE_32K";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DataPumpTablespaceBlockSizesInKb["UnknownValue"] = "UNKNOWN_VALUE";
})(DataPumpTablespaceBlockSizesInKb = exports.DataPumpTablespaceBlockSizesInKb || (exports.DataPumpTablespaceBlockSizesInKb = {}));
(function (DataPumpTablespaceBlockSizesInKb) {
    function getJsonObj(obj) {
        return obj;
    }
    DataPumpTablespaceBlockSizesInKb.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DataPumpTablespaceBlockSizesInKb.getDeserializedJsonObj = getDeserializedJsonObj;
})(DataPumpTablespaceBlockSizesInKb = exports.DataPumpTablespaceBlockSizesInKb || (exports.DataPumpTablespaceBlockSizesInKb = {}));
