// oci-databasemigration/lib/model/handle-grant-errors.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20230518
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HandleGrantErrors = void 0;
/**
 * The action taken in the event of errors related to GRANT or REVOKE errors.
 * ABORT - (default) stops the load process and displays an error.
 * DROP_ACCOUNT - deletes the account and continues the load process.
 * IGNORE - ignores the error and continues the load process.
 *
 **/
var HandleGrantErrors;
(function (HandleGrantErrors) {
    HandleGrantErrors["Abort"] = "ABORT";
    HandleGrantErrors["DropAccount"] = "DROP_ACCOUNT";
    HandleGrantErrors["Ignore"] = "IGNORE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    HandleGrantErrors["UnknownValue"] = "UNKNOWN_VALUE";
})(HandleGrantErrors = exports.HandleGrantErrors || (exports.HandleGrantErrors = {}));
(function (HandleGrantErrors) {
    function getJsonObj(obj) {
        return obj;
    }
    HandleGrantErrors.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    HandleGrantErrors.getDeserializedJsonObj = getDeserializedJsonObj;
})(HandleGrantErrors = exports.HandleGrantErrors || (exports.HandleGrantErrors = {}));
