// oci-databasemigration/lib/model/job-mode-oracle.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20230518
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobModeOracle = void 0;
/**
 * job mode.
 **/
var JobModeOracle;
(function (JobModeOracle) {
    JobModeOracle["Full"] = "FULL";
    JobModeOracle["Schema"] = "SCHEMA";
    JobModeOracle["Table"] = "TABLE";
    JobModeOracle["Tablespace"] = "TABLESPACE";
    JobModeOracle["Transportable"] = "TRANSPORTABLE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    JobModeOracle["UnknownValue"] = "UNKNOWN_VALUE";
})(JobModeOracle = exports.JobModeOracle || (exports.JobModeOracle = {}));
(function (JobModeOracle) {
    function getJsonObj(obj) {
        return obj;
    }
    JobModeOracle.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    JobModeOracle.getDeserializedJsonObj = getDeserializedJsonObj;
})(JobModeOracle = exports.JobModeOracle || (exports.JobModeOracle = {}));
