// oci-databasemigration/lib/model/replicat-performance-profile.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20230518
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicatPerformanceProfile = void 0;
/**
 * GoldenGate Replicat performance profile
 *
 **/
var ReplicatPerformanceProfile;
(function (ReplicatPerformanceProfile) {
    ReplicatPerformanceProfile["Low"] = "LOW";
    ReplicatPerformanceProfile["High"] = "HIGH";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ReplicatPerformanceProfile["UnknownValue"] = "UNKNOWN_VALUE";
})(ReplicatPerformanceProfile = exports.ReplicatPerformanceProfile || (exports.ReplicatPerformanceProfile = {}));
(function (ReplicatPerformanceProfile) {
    function getJsonObj(obj) {
        return obj;
    }
    ReplicatPerformanceProfile.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ReplicatPerformanceProfile.getDeserializedJsonObj = getDeserializedJsonObj;
})(ReplicatPerformanceProfile = exports.ReplicatPerformanceProfile || (exports.ReplicatPerformanceProfile = {}));
