// oci-databasemigration/lib/model/technology-type.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20230518
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TechnologyType = void 0;
/**
 * The specific database technology for a type of database connection.
 * Example: OCI_AUTONOMOUS_DATABASE for an ORACLE connection type.
 *
 **/
var TechnologyType;
(function (TechnologyType) {
    TechnologyType["OciAutonomousDatabase"] = "OCI_AUTONOMOUS_DATABASE";
    TechnologyType["OciMysql"] = "OCI_MYSQL";
    TechnologyType["OracleDatabase"] = "ORACLE_DATABASE";
    TechnologyType["OracleExadata"] = "ORACLE_EXADATA";
    TechnologyType["AmazonRdsOracle"] = "AMAZON_RDS_ORACLE";
    TechnologyType["AmazonAuroraMysql"] = "AMAZON_AURORA_MYSQL";
    TechnologyType["AmazonRdsMysql"] = "AMAZON_RDS_MYSQL";
    TechnologyType["AzureMysql"] = "AZURE_MYSQL";
    TechnologyType["GoogleCloudSqlMysql"] = "GOOGLE_CLOUD_SQL_MYSQL";
    TechnologyType["MysqlServer"] = "MYSQL_SERVER";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TechnologyType["UnknownValue"] = "UNKNOWN_VALUE";
})(TechnologyType = exports.TechnologyType || (exports.TechnologyType = {}));
(function (TechnologyType) {
    function getJsonObj(obj) {
        return obj;
    }
    TechnologyType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TechnologyType.getDeserializedJsonObj = getDeserializedJsonObj;
})(TechnologyType = exports.TechnologyType || (exports.TechnologyType = {}));
