// oci-fleetsoftwareupdate/lib/model/collection-lifecycle-states.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CollectionLifecycleStates = void 0;
/**
 * Possible lifecycle states for the Exadata Fleet Update Collection resource.
 *
 **/
var CollectionLifecycleStates;
(function (CollectionLifecycleStates) {
    CollectionLifecycleStates["Creating"] = "CREATING";
    CollectionLifecycleStates["Updating"] = "UPDATING";
    CollectionLifecycleStates["Active"] = "ACTIVE";
    CollectionLifecycleStates["NeedsAttention"] = "NEEDS_ATTENTION";
    CollectionLifecycleStates["Deleting"] = "DELETING";
    CollectionLifecycleStates["Deleted"] = "DELETED";
    CollectionLifecycleStates["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CollectionLifecycleStates["UnknownValue"] = "UNKNOWN_VALUE";
})(CollectionLifecycleStates = exports.CollectionLifecycleStates || (exports.CollectionLifecycleStates = {}));
(function (CollectionLifecycleStates) {
    function getJsonObj(obj) {
        return obj;
    }
    CollectionLifecycleStates.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CollectionLifecycleStates.getDeserializedJsonObj = getDeserializedJsonObj;
})(CollectionLifecycleStates = exports.CollectionLifecycleStates || (exports.CollectionLifecycleStates = {}));
