// oci-fleetsoftwareupdate/lib/model/db-source-major-versions.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbSourceMajorVersions = void 0;
/**
 * Possible source major versions for a DB Exadata Fleet Update Collection.
 *
 **/
var DbSourceMajorVersions;
(function (DbSourceMajorVersions) {
    DbSourceMajorVersions["Db11204"] = "DB_11204";
    DbSourceMajorVersions["Db121"] = "DB_121";
    DbSourceMajorVersions["Db122"] = "DB_122";
    DbSourceMajorVersions["Db18"] = "DB_18";
    DbSourceMajorVersions["Db19"] = "DB_19";
    DbSourceMajorVersions["Db23"] = "DB_23";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DbSourceMajorVersions["UnknownValue"] = "UNKNOWN_VALUE";
})(DbSourceMajorVersions = exports.DbSourceMajorVersions || (exports.DbSourceMajorVersions = {}));
(function (DbSourceMajorVersions) {
    function getJsonObj(obj) {
        return obj;
    }
    DbSourceMajorVersions.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DbSourceMajorVersions.getDeserializedJsonObj = getDeserializedJsonObj;
})(DbSourceMajorVersions = exports.DbSourceMajorVersions || (exports.DbSourceMajorVersions = {}));
