// oci-fleetsoftwareupdate/lib/model/detailed-action-types.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DetailedActionTypes = void 0;
/**
 * Possible Exadata Fleet Update Action types to report in a Exadata Fleet Update Cycle.
 *
 **/
var DetailedActionTypes;
(function (DetailedActionTypes) {
    DetailedActionTypes["Stage"] = "STAGE";
    DetailedActionTypes["PrecheckStage"] = "PRECHECK_STAGE";
    DetailedActionTypes["PrecheckApply"] = "PRECHECK_APPLY";
    DetailedActionTypes["Apply"] = "APPLY";
    DetailedActionTypes["RollbackAndRemoveTarget"] = "ROLLBACK_AND_REMOVE_TARGET";
    DetailedActionTypes["Cleanup"] = "CLEANUP";
    DetailedActionTypes["RollbackMaintenanceCycle"] = "ROLLBACK_MAINTENANCE_CYCLE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DetailedActionTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(DetailedActionTypes = exports.DetailedActionTypes || (exports.DetailedActionTypes = {}));
(function (DetailedActionTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    DetailedActionTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DetailedActionTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(DetailedActionTypes = exports.DetailedActionTypes || (exports.DetailedActionTypes = {}));
