// oci-fleetsoftwareupdate/lib/model/discovery-lifecycle-states.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryLifecycleStates = void 0;
/**
 * Possible lifecycleStates for the Exadata Fleet Update Discovery resource.
 *
 **/
var DiscoveryLifecycleStates;
(function (DiscoveryLifecycleStates) {
    DiscoveryLifecycleStates["Accepted"] = "ACCEPTED";
    DiscoveryLifecycleStates["InProgress"] = "IN_PROGRESS";
    DiscoveryLifecycleStates["Failed"] = "FAILED";
    DiscoveryLifecycleStates["Succeeded"] = "SUCCEEDED";
    DiscoveryLifecycleStates["Canceling"] = "CANCELING";
    DiscoveryLifecycleStates["Canceled"] = "CANCELED";
    DiscoveryLifecycleStates["Deleting"] = "DELETING";
    DiscoveryLifecycleStates["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DiscoveryLifecycleStates["UnknownValue"] = "UNKNOWN_VALUE";
})(DiscoveryLifecycleStates = exports.DiscoveryLifecycleStates || (exports.DiscoveryLifecycleStates = {}));
(function (DiscoveryLifecycleStates) {
    function getJsonObj(obj) {
        return obj;
    }
    DiscoveryLifecycleStates.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DiscoveryLifecycleStates.getDeserializedJsonObj = getDeserializedJsonObj;
})(DiscoveryLifecycleStates = exports.DiscoveryLifecycleStates || (exports.DiscoveryLifecycleStates = {}));
