// oci-fleetsoftwareupdate/lib/model/discovery-criteria.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryCriteria = void 0;
/**
 * Possible types of criteria to use for a Exadata Fleet Update Discovery.
 *
 **/
var DiscoveryCriteria;
(function (DiscoveryCriteria) {
    DiscoveryCriteria["SearchQuery"] = "SEARCH_QUERY";
    DiscoveryCriteria["Filters"] = "FILTERS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DiscoveryCriteria["UnknownValue"] = "UNKNOWN_VALUE";
})(DiscoveryCriteria = exports.DiscoveryCriteria || (exports.DiscoveryCriteria = {}));
(function (DiscoveryCriteria) {
    function getJsonObj(obj) {
        return obj;
    }
    DiscoveryCriteria.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DiscoveryCriteria.getDeserializedJsonObj = getDeserializedJsonObj;
})(DiscoveryCriteria = exports.DiscoveryCriteria || (exports.DiscoveryCriteria = {}));
