// oci-fleetsoftwareupdate/lib/model/gi-fleet-discovery-filter.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GiFleetDiscoveryFilter = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var GiFleetDiscoveryFilter;
(function (GiFleetDiscoveryFilter) {
    let Mode;
    (function (Mode) {
        Mode["Include"] = "INCLUDE";
        Mode["Exclude"] = "EXCLUDE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Mode["UnknownValue"] = "UNKNOWN_VALUE";
    })(Mode = GiFleetDiscoveryFilter.Mode || (GiFleetDiscoveryFilter.Mode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "DEFINED_TAG":
                    return model.GiDefinedTagsFilter.getJsonObj(jsonObj, true);
                case "COMPARTMENT_ID":
                    return model.GiCompartmentIdFilter.getJsonObj(jsonObj, true);
                case "FREEFORM_TAG":
                    return model.GiFreeformTagsFilter.getJsonObj(jsonObj, true);
                case "RESOURCE_ID":
                    return model.GiResourceIdFilter.getJsonObj(jsonObj, true);
                case "VERSION":
                    return model.GiVersionFilter.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    GiFleetDiscoveryFilter.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "DEFINED_TAG":
                    return model.GiDefinedTagsFilter.getDeserializedJsonObj(jsonObj, true);
                case "COMPARTMENT_ID":
                    return model.GiCompartmentIdFilter.getDeserializedJsonObj(jsonObj, true);
                case "FREEFORM_TAG":
                    return model.GiFreeformTagsFilter.getDeserializedJsonObj(jsonObj, true);
                case "RESOURCE_ID":
                    return model.GiResourceIdFilter.getDeserializedJsonObj(jsonObj, true);
                case "VERSION":
                    return model.GiVersionFilter.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    GiFleetDiscoveryFilter.getDeserializedJsonObj = getDeserializedJsonObj;
})(GiFleetDiscoveryFilter = exports.GiFleetDiscoveryFilter || (exports.GiFleetDiscoveryFilter = {}));
