// oci-fleetsoftwareupdate/lib/model/index.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FsuCycleSummaryCollection = exports.FsuCycleSummary = exports.FsuCycle = exports.FsuCollectionTarget = exports.FsuCollectionSummaryCollection = exports.FsuCollectionSummary = exports.FsuCollection = exports.FsuActionSummaryCollection = exports.FsuActionSummary = exports.FsuActionProgressDetails = exports.FsuAction = exports.FreeformTagFilterEntry = exports.FleetDiscoveryStrategies = exports.FleetDiscoveryOperators = exports.DiscoveryTypes = exports.DiscoveryServiceTypes = exports.DiscoveryLifecycleStates = exports.DiscoveryDetailsSummary = exports.DiscoveryDetails = exports.DiscoveryCriteria = exports.DiagnosticsCollectionDetails = exports.DetailedActionTypes = exports.DefinedTagFilterEntry = exports.DbSourceMajorVersions = exports.DbFleetDiscoveryFilter = exports.DbFleetDiscoveryDetails = exports.DataCollectionModes = exports.CycleTypes = exports.CycleLifecycleStates = exports.CreateScheduleDetails = exports.CreateFsuDiscoveryDetails = exports.CreateFsuCycleDetails = exports.CreateFsuCollectionDetails = exports.CreateFsuActionDetails = exports.CreateBatchingStrategyDetails = exports.CollectionTypes = exports.CollectionServiceTypes = exports.CollectionLifecycleStates = exports.CloneFsuCycleDetails = exports.ChangeFsuDiscoveryCompartmentDetails = exports.ChangeFsuCycleCompartmentDetails = exports.ChangeFsuCollectionCompartmentDetails = exports.ChangeFsuActionCompartmentDetails = exports.BatchingStrategyDetails = exports.BatchingStrategies = exports.AddFsuCollectionTargetsDetails = exports.ActiveCycleDetails = exports.ActionTypes = exports.ActionType = exports.ActionLifecycleStates = void 0;
exports.ApplyFsuJob = exports.ApplyActionSummary = exports.ApplyAction = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpgradeDetails = exports.UpdateScheduleDetails = exports.UpdateFsuJobDetails = exports.UpdateFsuDiscoveryDetails = exports.UpdateFsuCycleDetails = exports.UpdateFsuCollectionDetails = exports.UpdateFsuActionDetails = exports.UpdateBatchingStrategyDetails = exports.UpdateBatchingStrategies = exports.TargetSummaryCollection = exports.TargetSummary = exports.TargetProgressSummary = exports.TargetEntry = exports.TargetDetails = exports.SortOrder = exports.ScheduleDetails = exports.RollbackDetails = exports.RollbackCycleStates = exports.RemoveFsuCollectionTargetsDetails = exports.OperationType = exports.OperationStatus = exports.NextActionToExecuteDetails = exports.JobTypes = exports.JobProgressDetails = exports.JobProgress = exports.JobLifecycleStates = exports.GiSourceMajorVersions = exports.GiFleetDiscoveryFilter = exports.GiFleetDiscoveryDetails = exports.FsuJobSummary = exports.FsuJobOutputSummaryCollection = exports.FsuJobOutputSummary = exports.FsuJobCollection = exports.FsuJob = exports.FsuGoalVersionDetails = exports.FsuDiscoverySummaryCollection = exports.FsuDiscoverySummary = exports.FsuDiscovery = void 0;
exports.GiTargetListDiscovery = exports.GiSearchQueryDiscovery = exports.GiResourceIdFilter = exports.GiFsuCollectionSummary = exports.GiFreeformTagsFilter = exports.GiFiltersDiscovery = exports.GiDiscoveryResults = exports.GiDiscoveryDetails = exports.GiDefinedTagsFilter = exports.GiCompartmentIdFilter = exports.GiCollection = exports.FiftyFiftyBatchingStrategyDetails = exports.FailedJobsRollbackDetails = exports.DbVersionFilter = exports.DbUniqueNameFilter = exports.DbTargetListDiscovery = exports.DbSearchQueryDiscovery = exports.DbResourceIdFilter = exports.DbNameFilter = exports.DbHomeNameFilter = exports.DbFsuCollectionSummary = exports.DbFreeformTagsFilter = exports.DbFiltersDiscovery = exports.DbDiscoveryResults = exports.DbDiscoveryDetails = exports.DbDefinedTagsFilter = exports.DbCompartmentIdFilter = exports.DbCollection = exports.DatabaseTargetSummary = exports.CreateUpgradeFsuCycle = exports.CreateStartTimeScheduleDetails = exports.CreateStageActionDetails = exports.CreateServiceAvailabilityFactorBatchingStrategyDetails = exports.CreateSequentialBatchingStrategyDetails = exports.CreateRollbackCycleApplyActionDetails = exports.CreateRollbackActionDetails = exports.CreatePrecheckActionDetails = exports.CreatePatchFsuCycle = exports.CreateNonRollingBatchingStrategyDetails = exports.CreateGiFsuCollectionDetails = exports.CreateFiftyFiftyBatchingStrategyDetails = exports.CreateDbFsuCollectionDetails = exports.CreateCleanupActionDetails = exports.CreateApplyActionDetails = exports.CloudVmClusterTargetSummary = exports.CleanupFsuJobSummary = exports.CleanupFsuJob = exports.CleanupActionSummary = exports.CleanupAction = exports.ApplyFsuJobSummary = void 0;
exports.VmClusterTargetSummary = exports.VersionFsuTargetDetails = exports.UpgradeGiCollectionDetails = exports.UpgradeFsuCycle = exports.UpgradeDbCollectionDetails = exports.UpdateUpgradeFsuCycle = exports.UpdateStartTimeScheduleDetails = exports.UpdateStageActionDetails = exports.UpdateServiceAvailabilityFactorBatchingStrategyDetails = exports.UpdateSequentialBatchingStrategyDetails = exports.UpdateRollbackCycleActionDetails = exports.UpdateRollbackActionDetails = exports.UpdatePrecheckActionDetails = exports.UpdatePatchFsuCycle = exports.UpdateNonRollingBatchingStrategyDetails = exports.UpdateFiftyFiftyBatchingStrategyDetails = exports.UpdateCleanupActionDetails = exports.UpdateApplyActionDetails = exports.TargetIdsRemoveTargetsDetails = exports.StartTimeScheduleDetails = exports.StageFsuJobSummary = exports.StageFsuJob = exports.StageActionSummary = exports.StageAction = exports.ServiceAvailabilityFactorBatchingStrategyDetails = exports.SequentialBatchingStrategyDetails = exports.RollbackFsuJobSummary = exports.RollbackFsuJob = exports.RollbackCycleFsuJobSummary = exports.RollbackCycleFsuJob = exports.RollbackCycleActionSummary = exports.RollbackCycleAction = exports.RollbackActionSummary = exports.RollbackAction = exports.PrecheckFsuJobSummary = exports.PrecheckFsuJob = exports.PrecheckActionSummary = exports.PrecheckAction = exports.PatchFsuCycle = exports.NoneScheduleDetails = exports.NoneBatchingStrategyDetails = exports.NonRollingBatchingStrategyDetails = exports.ListOfTargetsRollbackDetails = exports.ImageIdFsuTargetDetails = exports.GiVersionFilter = void 0;
const ActionLifecycleStates = __importStar(require("./action-lifecycle-states"));
exports.ActionLifecycleStates = ActionLifecycleStates.ActionLifecycleStates;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ActionTypes = __importStar(require("./action-types"));
exports.ActionTypes = ActionTypes.ActionTypes;
const ActiveCycleDetails = __importStar(require("./active-cycle-details"));
exports.ActiveCycleDetails = ActiveCycleDetails.ActiveCycleDetails;
const AddFsuCollectionTargetsDetails = __importStar(require("./add-fsu-collection-targets-details"));
exports.AddFsuCollectionTargetsDetails = AddFsuCollectionTargetsDetails.AddFsuCollectionTargetsDetails;
const BatchingStrategies = __importStar(require("./batching-strategies"));
exports.BatchingStrategies = BatchingStrategies.BatchingStrategies;
const BatchingStrategyDetails = __importStar(require("./batching-strategy-details"));
exports.BatchingStrategyDetails = BatchingStrategyDetails.BatchingStrategyDetails;
const ChangeFsuActionCompartmentDetails = __importStar(require("./change-fsu-action-compartment-details"));
exports.ChangeFsuActionCompartmentDetails = ChangeFsuActionCompartmentDetails.ChangeFsuActionCompartmentDetails;
const ChangeFsuCollectionCompartmentDetails = __importStar(require("./change-fsu-collection-compartment-details"));
exports.ChangeFsuCollectionCompartmentDetails = ChangeFsuCollectionCompartmentDetails.ChangeFsuCollectionCompartmentDetails;
const ChangeFsuCycleCompartmentDetails = __importStar(require("./change-fsu-cycle-compartment-details"));
exports.ChangeFsuCycleCompartmentDetails = ChangeFsuCycleCompartmentDetails.ChangeFsuCycleCompartmentDetails;
const ChangeFsuDiscoveryCompartmentDetails = __importStar(require("./change-fsu-discovery-compartment-details"));
exports.ChangeFsuDiscoveryCompartmentDetails = ChangeFsuDiscoveryCompartmentDetails.ChangeFsuDiscoveryCompartmentDetails;
const CloneFsuCycleDetails = __importStar(require("./clone-fsu-cycle-details"));
exports.CloneFsuCycleDetails = CloneFsuCycleDetails.CloneFsuCycleDetails;
const CollectionLifecycleStates = __importStar(require("./collection-lifecycle-states"));
exports.CollectionLifecycleStates = CollectionLifecycleStates.CollectionLifecycleStates;
const CollectionServiceTypes = __importStar(require("./collection-service-types"));
exports.CollectionServiceTypes = CollectionServiceTypes.CollectionServiceTypes;
const CollectionTypes = __importStar(require("./collection-types"));
exports.CollectionTypes = CollectionTypes.CollectionTypes;
const CreateBatchingStrategyDetails = __importStar(require("./create-batching-strategy-details"));
exports.CreateBatchingStrategyDetails = CreateBatchingStrategyDetails.CreateBatchingStrategyDetails;
const CreateFsuActionDetails = __importStar(require("./create-fsu-action-details"));
exports.CreateFsuActionDetails = CreateFsuActionDetails.CreateFsuActionDetails;
const CreateFsuCollectionDetails = __importStar(require("./create-fsu-collection-details"));
exports.CreateFsuCollectionDetails = CreateFsuCollectionDetails.CreateFsuCollectionDetails;
const CreateFsuCycleDetails = __importStar(require("./create-fsu-cycle-details"));
exports.CreateFsuCycleDetails = CreateFsuCycleDetails.CreateFsuCycleDetails;
const CreateFsuDiscoveryDetails = __importStar(require("./create-fsu-discovery-details"));
exports.CreateFsuDiscoveryDetails = CreateFsuDiscoveryDetails.CreateFsuDiscoveryDetails;
const CreateScheduleDetails = __importStar(require("./create-schedule-details"));
exports.CreateScheduleDetails = CreateScheduleDetails.CreateScheduleDetails;
const CycleLifecycleStates = __importStar(require("./cycle-lifecycle-states"));
exports.CycleLifecycleStates = CycleLifecycleStates.CycleLifecycleStates;
const CycleTypes = __importStar(require("./cycle-types"));
exports.CycleTypes = CycleTypes.CycleTypes;
const DataCollectionModes = __importStar(require("./data-collection-modes"));
exports.DataCollectionModes = DataCollectionModes.DataCollectionModes;
const DbFleetDiscoveryDetails = __importStar(require("./db-fleet-discovery-details"));
exports.DbFleetDiscoveryDetails = DbFleetDiscoveryDetails.DbFleetDiscoveryDetails;
const DbFleetDiscoveryFilter = __importStar(require("./db-fleet-discovery-filter"));
exports.DbFleetDiscoveryFilter = DbFleetDiscoveryFilter.DbFleetDiscoveryFilter;
const DbSourceMajorVersions = __importStar(require("./db-source-major-versions"));
exports.DbSourceMajorVersions = DbSourceMajorVersions.DbSourceMajorVersions;
const DefinedTagFilterEntry = __importStar(require("./defined-tag-filter-entry"));
exports.DefinedTagFilterEntry = DefinedTagFilterEntry.DefinedTagFilterEntry;
const DetailedActionTypes = __importStar(require("./detailed-action-types"));
exports.DetailedActionTypes = DetailedActionTypes.DetailedActionTypes;
const DiagnosticsCollectionDetails = __importStar(require("./diagnostics-collection-details"));
exports.DiagnosticsCollectionDetails = DiagnosticsCollectionDetails.DiagnosticsCollectionDetails;
const DiscoveryCriteria = __importStar(require("./discovery-criteria"));
exports.DiscoveryCriteria = DiscoveryCriteria.DiscoveryCriteria;
const DiscoveryDetails = __importStar(require("./discovery-details"));
exports.DiscoveryDetails = DiscoveryDetails.DiscoveryDetails;
const DiscoveryDetailsSummary = __importStar(require("./discovery-details-summary"));
exports.DiscoveryDetailsSummary = DiscoveryDetailsSummary.DiscoveryDetailsSummary;
const DiscoveryLifecycleStates = __importStar(require("./discovery-lifecycle-states"));
exports.DiscoveryLifecycleStates = DiscoveryLifecycleStates.DiscoveryLifecycleStates;
const DiscoveryServiceTypes = __importStar(require("./discovery-service-types"));
exports.DiscoveryServiceTypes = DiscoveryServiceTypes.DiscoveryServiceTypes;
const DiscoveryTypes = __importStar(require("./discovery-types"));
exports.DiscoveryTypes = DiscoveryTypes.DiscoveryTypes;
const FleetDiscoveryOperators = __importStar(require("./fleet-discovery-operators"));
exports.FleetDiscoveryOperators = FleetDiscoveryOperators.FleetDiscoveryOperators;
const FleetDiscoveryStrategies = __importStar(require("./fleet-discovery-strategies"));
exports.FleetDiscoveryStrategies = FleetDiscoveryStrategies.FleetDiscoveryStrategies;
const FreeformTagFilterEntry = __importStar(require("./freeform-tag-filter-entry"));
exports.FreeformTagFilterEntry = FreeformTagFilterEntry.FreeformTagFilterEntry;
const FsuAction = __importStar(require("./fsu-action"));
exports.FsuAction = FsuAction.FsuAction;
const FsuActionProgressDetails = __importStar(require("./fsu-action-progress-details"));
exports.FsuActionProgressDetails = FsuActionProgressDetails.FsuActionProgressDetails;
const FsuActionSummary = __importStar(require("./fsu-action-summary"));
exports.FsuActionSummary = FsuActionSummary.FsuActionSummary;
const FsuActionSummaryCollection = __importStar(require("./fsu-action-summary-collection"));
exports.FsuActionSummaryCollection = FsuActionSummaryCollection.FsuActionSummaryCollection;
const FsuCollection = __importStar(require("./fsu-collection"));
exports.FsuCollection = FsuCollection.FsuCollection;
const FsuCollectionSummary = __importStar(require("./fsu-collection-summary"));
exports.FsuCollectionSummary = FsuCollectionSummary.FsuCollectionSummary;
const FsuCollectionSummaryCollection = __importStar(require("./fsu-collection-summary-collection"));
exports.FsuCollectionSummaryCollection = FsuCollectionSummaryCollection.FsuCollectionSummaryCollection;
const FsuCollectionTarget = __importStar(require("./fsu-collection-target"));
exports.FsuCollectionTarget = FsuCollectionTarget.FsuCollectionTarget;
const FsuCycle = __importStar(require("./fsu-cycle"));
exports.FsuCycle = FsuCycle.FsuCycle;
const FsuCycleSummary = __importStar(require("./fsu-cycle-summary"));
exports.FsuCycleSummary = FsuCycleSummary.FsuCycleSummary;
const FsuCycleSummaryCollection = __importStar(require("./fsu-cycle-summary-collection"));
exports.FsuCycleSummaryCollection = FsuCycleSummaryCollection.FsuCycleSummaryCollection;
const FsuDiscovery = __importStar(require("./fsu-discovery"));
exports.FsuDiscovery = FsuDiscovery.FsuDiscovery;
const FsuDiscoverySummary = __importStar(require("./fsu-discovery-summary"));
exports.FsuDiscoverySummary = FsuDiscoverySummary.FsuDiscoverySummary;
const FsuDiscoverySummaryCollection = __importStar(require("./fsu-discovery-summary-collection"));
exports.FsuDiscoverySummaryCollection = FsuDiscoverySummaryCollection.FsuDiscoverySummaryCollection;
const FsuGoalVersionDetails = __importStar(require("./fsu-goal-version-details"));
exports.FsuGoalVersionDetails = FsuGoalVersionDetails.FsuGoalVersionDetails;
const FsuJob = __importStar(require("./fsu-job"));
exports.FsuJob = FsuJob.FsuJob;
const FsuJobCollection = __importStar(require("./fsu-job-collection"));
exports.FsuJobCollection = FsuJobCollection.FsuJobCollection;
const FsuJobOutputSummary = __importStar(require("./fsu-job-output-summary"));
exports.FsuJobOutputSummary = FsuJobOutputSummary.FsuJobOutputSummary;
const FsuJobOutputSummaryCollection = __importStar(require("./fsu-job-output-summary-collection"));
exports.FsuJobOutputSummaryCollection = FsuJobOutputSummaryCollection.FsuJobOutputSummaryCollection;
const FsuJobSummary = __importStar(require("./fsu-job-summary"));
exports.FsuJobSummary = FsuJobSummary.FsuJobSummary;
const GiFleetDiscoveryDetails = __importStar(require("./gi-fleet-discovery-details"));
exports.GiFleetDiscoveryDetails = GiFleetDiscoveryDetails.GiFleetDiscoveryDetails;
const GiFleetDiscoveryFilter = __importStar(require("./gi-fleet-discovery-filter"));
exports.GiFleetDiscoveryFilter = GiFleetDiscoveryFilter.GiFleetDiscoveryFilter;
const GiSourceMajorVersions = __importStar(require("./gi-source-major-versions"));
exports.GiSourceMajorVersions = GiSourceMajorVersions.GiSourceMajorVersions;
const JobLifecycleStates = __importStar(require("./job-lifecycle-states"));
exports.JobLifecycleStates = JobLifecycleStates.JobLifecycleStates;
const JobProgress = __importStar(require("./job-progress"));
exports.JobProgress = JobProgress.JobProgress;
const JobProgressDetails = __importStar(require("./job-progress-details"));
exports.JobProgressDetails = JobProgressDetails.JobProgressDetails;
const JobTypes = __importStar(require("./job-types"));
exports.JobTypes = JobTypes.JobTypes;
const NextActionToExecuteDetails = __importStar(require("./next-action-to-execute-details"));
exports.NextActionToExecuteDetails = NextActionToExecuteDetails.NextActionToExecuteDetails;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const RemoveFsuCollectionTargetsDetails = __importStar(require("./remove-fsu-collection-targets-details"));
exports.RemoveFsuCollectionTargetsDetails = RemoveFsuCollectionTargetsDetails.RemoveFsuCollectionTargetsDetails;
const RollbackCycleStates = __importStar(require("./rollback-cycle-states"));
exports.RollbackCycleStates = RollbackCycleStates.RollbackCycleStates;
const RollbackDetails = __importStar(require("./rollback-details"));
exports.RollbackDetails = RollbackDetails.RollbackDetails;
const ScheduleDetails = __importStar(require("./schedule-details"));
exports.ScheduleDetails = ScheduleDetails.ScheduleDetails;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const TargetDetails = __importStar(require("./target-details"));
exports.TargetDetails = TargetDetails.TargetDetails;
const TargetEntry = __importStar(require("./target-entry"));
exports.TargetEntry = TargetEntry.TargetEntry;
const TargetProgressSummary = __importStar(require("./target-progress-summary"));
exports.TargetProgressSummary = TargetProgressSummary.TargetProgressSummary;
const TargetSummary = __importStar(require("./target-summary"));
exports.TargetSummary = TargetSummary.TargetSummary;
const TargetSummaryCollection = __importStar(require("./target-summary-collection"));
exports.TargetSummaryCollection = TargetSummaryCollection.TargetSummaryCollection;
const UpdateBatchingStrategies = __importStar(require("./update-batching-strategies"));
exports.UpdateBatchingStrategies = UpdateBatchingStrategies.UpdateBatchingStrategies;
const UpdateBatchingStrategyDetails = __importStar(require("./update-batching-strategy-details"));
exports.UpdateBatchingStrategyDetails = UpdateBatchingStrategyDetails.UpdateBatchingStrategyDetails;
const UpdateFsuActionDetails = __importStar(require("./update-fsu-action-details"));
exports.UpdateFsuActionDetails = UpdateFsuActionDetails.UpdateFsuActionDetails;
const UpdateFsuCollectionDetails = __importStar(require("./update-fsu-collection-details"));
exports.UpdateFsuCollectionDetails = UpdateFsuCollectionDetails.UpdateFsuCollectionDetails;
const UpdateFsuCycleDetails = __importStar(require("./update-fsu-cycle-details"));
exports.UpdateFsuCycleDetails = UpdateFsuCycleDetails.UpdateFsuCycleDetails;
const UpdateFsuDiscoveryDetails = __importStar(require("./update-fsu-discovery-details"));
exports.UpdateFsuDiscoveryDetails = UpdateFsuDiscoveryDetails.UpdateFsuDiscoveryDetails;
const UpdateFsuJobDetails = __importStar(require("./update-fsu-job-details"));
exports.UpdateFsuJobDetails = UpdateFsuJobDetails.UpdateFsuJobDetails;
const UpdateScheduleDetails = __importStar(require("./update-schedule-details"));
exports.UpdateScheduleDetails = UpdateScheduleDetails.UpdateScheduleDetails;
const UpgradeDetails = __importStar(require("./upgrade-details"));
exports.UpgradeDetails = UpgradeDetails.UpgradeDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const ApplyAction = __importStar(require("./apply-action"));
exports.ApplyAction = ApplyAction.ApplyAction;
const ApplyActionSummary = __importStar(require("./apply-action-summary"));
exports.ApplyActionSummary = ApplyActionSummary.ApplyActionSummary;
const ApplyFsuJob = __importStar(require("./apply-fsu-job"));
exports.ApplyFsuJob = ApplyFsuJob.ApplyFsuJob;
const ApplyFsuJobSummary = __importStar(require("./apply-fsu-job-summary"));
exports.ApplyFsuJobSummary = ApplyFsuJobSummary.ApplyFsuJobSummary;
const CleanupAction = __importStar(require("./cleanup-action"));
exports.CleanupAction = CleanupAction.CleanupAction;
const CleanupActionSummary = __importStar(require("./cleanup-action-summary"));
exports.CleanupActionSummary = CleanupActionSummary.CleanupActionSummary;
const CleanupFsuJob = __importStar(require("./cleanup-fsu-job"));
exports.CleanupFsuJob = CleanupFsuJob.CleanupFsuJob;
const CleanupFsuJobSummary = __importStar(require("./cleanup-fsu-job-summary"));
exports.CleanupFsuJobSummary = CleanupFsuJobSummary.CleanupFsuJobSummary;
const CloudVmClusterTargetSummary = __importStar(require("./cloud-vm-cluster-target-summary"));
exports.CloudVmClusterTargetSummary = CloudVmClusterTargetSummary.CloudVmClusterTargetSummary;
const CreateApplyActionDetails = __importStar(require("./create-apply-action-details"));
exports.CreateApplyActionDetails = CreateApplyActionDetails.CreateApplyActionDetails;
const CreateCleanupActionDetails = __importStar(require("./create-cleanup-action-details"));
exports.CreateCleanupActionDetails = CreateCleanupActionDetails.CreateCleanupActionDetails;
const CreateDbFsuCollectionDetails = __importStar(require("./create-db-fsu-collection-details"));
exports.CreateDbFsuCollectionDetails = CreateDbFsuCollectionDetails.CreateDbFsuCollectionDetails;
const CreateFiftyFiftyBatchingStrategyDetails = __importStar(require("./create-fifty-fifty-batching-strategy-details"));
exports.CreateFiftyFiftyBatchingStrategyDetails = CreateFiftyFiftyBatchingStrategyDetails.CreateFiftyFiftyBatchingStrategyDetails;
const CreateGiFsuCollectionDetails = __importStar(require("./create-gi-fsu-collection-details"));
exports.CreateGiFsuCollectionDetails = CreateGiFsuCollectionDetails.CreateGiFsuCollectionDetails;
const CreateNonRollingBatchingStrategyDetails = __importStar(require("./create-non-rolling-batching-strategy-details"));
exports.CreateNonRollingBatchingStrategyDetails = CreateNonRollingBatchingStrategyDetails.CreateNonRollingBatchingStrategyDetails;
const CreatePatchFsuCycle = __importStar(require("./create-patch-fsu-cycle"));
exports.CreatePatchFsuCycle = CreatePatchFsuCycle.CreatePatchFsuCycle;
const CreatePrecheckActionDetails = __importStar(require("./create-precheck-action-details"));
exports.CreatePrecheckActionDetails = CreatePrecheckActionDetails.CreatePrecheckActionDetails;
const CreateRollbackActionDetails = __importStar(require("./create-rollback-action-details"));
exports.CreateRollbackActionDetails = CreateRollbackActionDetails.CreateRollbackActionDetails;
const CreateRollbackCycleApplyActionDetails = __importStar(require("./create-rollback-cycle-apply-action-details"));
exports.CreateRollbackCycleApplyActionDetails = CreateRollbackCycleApplyActionDetails.CreateRollbackCycleApplyActionDetails;
const CreateSequentialBatchingStrategyDetails = __importStar(require("./create-sequential-batching-strategy-details"));
exports.CreateSequentialBatchingStrategyDetails = CreateSequentialBatchingStrategyDetails.CreateSequentialBatchingStrategyDetails;
const CreateServiceAvailabilityFactorBatchingStrategyDetails = __importStar(require("./create-service-availability-factor-batching-strategy-details"));
exports.CreateServiceAvailabilityFactorBatchingStrategyDetails = CreateServiceAvailabilityFactorBatchingStrategyDetails.CreateServiceAvailabilityFactorBatchingStrategyDetails;
const CreateStageActionDetails = __importStar(require("./create-stage-action-details"));
exports.CreateStageActionDetails = CreateStageActionDetails.CreateStageActionDetails;
const CreateStartTimeScheduleDetails = __importStar(require("./create-start-time-schedule-details"));
exports.CreateStartTimeScheduleDetails = CreateStartTimeScheduleDetails.CreateStartTimeScheduleDetails;
const CreateUpgradeFsuCycle = __importStar(require("./create-upgrade-fsu-cycle"));
exports.CreateUpgradeFsuCycle = CreateUpgradeFsuCycle.CreateUpgradeFsuCycle;
const DatabaseTargetSummary = __importStar(require("./database-target-summary"));
exports.DatabaseTargetSummary = DatabaseTargetSummary.DatabaseTargetSummary;
const DbCollection = __importStar(require("./db-collection"));
exports.DbCollection = DbCollection.DbCollection;
const DbCompartmentIdFilter = __importStar(require("./db-compartment-id-filter"));
exports.DbCompartmentIdFilter = DbCompartmentIdFilter.DbCompartmentIdFilter;
const DbDefinedTagsFilter = __importStar(require("./db-defined-tags-filter"));
exports.DbDefinedTagsFilter = DbDefinedTagsFilter.DbDefinedTagsFilter;
const DbDiscoveryDetails = __importStar(require("./db-discovery-details"));
exports.DbDiscoveryDetails = DbDiscoveryDetails.DbDiscoveryDetails;
const DbDiscoveryResults = __importStar(require("./db-discovery-results"));
exports.DbDiscoveryResults = DbDiscoveryResults.DbDiscoveryResults;
const DbFiltersDiscovery = __importStar(require("./db-filters-discovery"));
exports.DbFiltersDiscovery = DbFiltersDiscovery.DbFiltersDiscovery;
const DbFreeformTagsFilter = __importStar(require("./db-freeform-tags-filter"));
exports.DbFreeformTagsFilter = DbFreeformTagsFilter.DbFreeformTagsFilter;
const DbFsuCollectionSummary = __importStar(require("./db-fsu-collection-summary"));
exports.DbFsuCollectionSummary = DbFsuCollectionSummary.DbFsuCollectionSummary;
const DbHomeNameFilter = __importStar(require("./db-home-name-filter"));
exports.DbHomeNameFilter = DbHomeNameFilter.DbHomeNameFilter;
const DbNameFilter = __importStar(require("./db-name-filter"));
exports.DbNameFilter = DbNameFilter.DbNameFilter;
const DbResourceIdFilter = __importStar(require("./db-resource-id-filter"));
exports.DbResourceIdFilter = DbResourceIdFilter.DbResourceIdFilter;
const DbSearchQueryDiscovery = __importStar(require("./db-search-query-discovery"));
exports.DbSearchQueryDiscovery = DbSearchQueryDiscovery.DbSearchQueryDiscovery;
const DbTargetListDiscovery = __importStar(require("./db-target-list-discovery"));
exports.DbTargetListDiscovery = DbTargetListDiscovery.DbTargetListDiscovery;
const DbUniqueNameFilter = __importStar(require("./db-unique-name-filter"));
exports.DbUniqueNameFilter = DbUniqueNameFilter.DbUniqueNameFilter;
const DbVersionFilter = __importStar(require("./db-version-filter"));
exports.DbVersionFilter = DbVersionFilter.DbVersionFilter;
const FailedJobsRollbackDetails = __importStar(require("./failed-jobs-rollback-details"));
exports.FailedJobsRollbackDetails = FailedJobsRollbackDetails.FailedJobsRollbackDetails;
const FiftyFiftyBatchingStrategyDetails = __importStar(require("./fifty-fifty-batching-strategy-details"));
exports.FiftyFiftyBatchingStrategyDetails = FiftyFiftyBatchingStrategyDetails.FiftyFiftyBatchingStrategyDetails;
const GiCollection = __importStar(require("./gi-collection"));
exports.GiCollection = GiCollection.GiCollection;
const GiCompartmentIdFilter = __importStar(require("./gi-compartment-id-filter"));
exports.GiCompartmentIdFilter = GiCompartmentIdFilter.GiCompartmentIdFilter;
const GiDefinedTagsFilter = __importStar(require("./gi-defined-tags-filter"));
exports.GiDefinedTagsFilter = GiDefinedTagsFilter.GiDefinedTagsFilter;
const GiDiscoveryDetails = __importStar(require("./gi-discovery-details"));
exports.GiDiscoveryDetails = GiDiscoveryDetails.GiDiscoveryDetails;
const GiDiscoveryResults = __importStar(require("./gi-discovery-results"));
exports.GiDiscoveryResults = GiDiscoveryResults.GiDiscoveryResults;
const GiFiltersDiscovery = __importStar(require("./gi-filters-discovery"));
exports.GiFiltersDiscovery = GiFiltersDiscovery.GiFiltersDiscovery;
const GiFreeformTagsFilter = __importStar(require("./gi-freeform-tags-filter"));
exports.GiFreeformTagsFilter = GiFreeformTagsFilter.GiFreeformTagsFilter;
const GiFsuCollectionSummary = __importStar(require("./gi-fsu-collection-summary"));
exports.GiFsuCollectionSummary = GiFsuCollectionSummary.GiFsuCollectionSummary;
const GiResourceIdFilter = __importStar(require("./gi-resource-id-filter"));
exports.GiResourceIdFilter = GiResourceIdFilter.GiResourceIdFilter;
const GiSearchQueryDiscovery = __importStar(require("./gi-search-query-discovery"));
exports.GiSearchQueryDiscovery = GiSearchQueryDiscovery.GiSearchQueryDiscovery;
const GiTargetListDiscovery = __importStar(require("./gi-target-list-discovery"));
exports.GiTargetListDiscovery = GiTargetListDiscovery.GiTargetListDiscovery;
const GiVersionFilter = __importStar(require("./gi-version-filter"));
exports.GiVersionFilter = GiVersionFilter.GiVersionFilter;
const ImageIdFsuTargetDetails = __importStar(require("./image-id-fsu-target-details"));
exports.ImageIdFsuTargetDetails = ImageIdFsuTargetDetails.ImageIdFsuTargetDetails;
const ListOfTargetsRollbackDetails = __importStar(require("./list-of-targets-rollback-details"));
exports.ListOfTargetsRollbackDetails = ListOfTargetsRollbackDetails.ListOfTargetsRollbackDetails;
const NonRollingBatchingStrategyDetails = __importStar(require("./non-rolling-batching-strategy-details"));
exports.NonRollingBatchingStrategyDetails = NonRollingBatchingStrategyDetails.NonRollingBatchingStrategyDetails;
const NoneBatchingStrategyDetails = __importStar(require("./none-batching-strategy-details"));
exports.NoneBatchingStrategyDetails = NoneBatchingStrategyDetails.NoneBatchingStrategyDetails;
const NoneScheduleDetails = __importStar(require("./none-schedule-details"));
exports.NoneScheduleDetails = NoneScheduleDetails.NoneScheduleDetails;
const PatchFsuCycle = __importStar(require("./patch-fsu-cycle"));
exports.PatchFsuCycle = PatchFsuCycle.PatchFsuCycle;
const PrecheckAction = __importStar(require("./precheck-action"));
exports.PrecheckAction = PrecheckAction.PrecheckAction;
const PrecheckActionSummary = __importStar(require("./precheck-action-summary"));
exports.PrecheckActionSummary = PrecheckActionSummary.PrecheckActionSummary;
const PrecheckFsuJob = __importStar(require("./precheck-fsu-job"));
exports.PrecheckFsuJob = PrecheckFsuJob.PrecheckFsuJob;
const PrecheckFsuJobSummary = __importStar(require("./precheck-fsu-job-summary"));
exports.PrecheckFsuJobSummary = PrecheckFsuJobSummary.PrecheckFsuJobSummary;
const RollbackAction = __importStar(require("./rollback-action"));
exports.RollbackAction = RollbackAction.RollbackAction;
const RollbackActionSummary = __importStar(require("./rollback-action-summary"));
exports.RollbackActionSummary = RollbackActionSummary.RollbackActionSummary;
const RollbackCycleAction = __importStar(require("./rollback-cycle-action"));
exports.RollbackCycleAction = RollbackCycleAction.RollbackCycleAction;
const RollbackCycleActionSummary = __importStar(require("./rollback-cycle-action-summary"));
exports.RollbackCycleActionSummary = RollbackCycleActionSummary.RollbackCycleActionSummary;
const RollbackCycleFsuJob = __importStar(require("./rollback-cycle-fsu-job"));
exports.RollbackCycleFsuJob = RollbackCycleFsuJob.RollbackCycleFsuJob;
const RollbackCycleFsuJobSummary = __importStar(require("./rollback-cycle-fsu-job-summary"));
exports.RollbackCycleFsuJobSummary = RollbackCycleFsuJobSummary.RollbackCycleFsuJobSummary;
const RollbackFsuJob = __importStar(require("./rollback-fsu-job"));
exports.RollbackFsuJob = RollbackFsuJob.RollbackFsuJob;
const RollbackFsuJobSummary = __importStar(require("./rollback-fsu-job-summary"));
exports.RollbackFsuJobSummary = RollbackFsuJobSummary.RollbackFsuJobSummary;
const SequentialBatchingStrategyDetails = __importStar(require("./sequential-batching-strategy-details"));
exports.SequentialBatchingStrategyDetails = SequentialBatchingStrategyDetails.SequentialBatchingStrategyDetails;
const ServiceAvailabilityFactorBatchingStrategyDetails = __importStar(require("./service-availability-factor-batching-strategy-details"));
exports.ServiceAvailabilityFactorBatchingStrategyDetails = ServiceAvailabilityFactorBatchingStrategyDetails.ServiceAvailabilityFactorBatchingStrategyDetails;
const StageAction = __importStar(require("./stage-action"));
exports.StageAction = StageAction.StageAction;
const StageActionSummary = __importStar(require("./stage-action-summary"));
exports.StageActionSummary = StageActionSummary.StageActionSummary;
const StageFsuJob = __importStar(require("./stage-fsu-job"));
exports.StageFsuJob = StageFsuJob.StageFsuJob;
const StageFsuJobSummary = __importStar(require("./stage-fsu-job-summary"));
exports.StageFsuJobSummary = StageFsuJobSummary.StageFsuJobSummary;
const StartTimeScheduleDetails = __importStar(require("./start-time-schedule-details"));
exports.StartTimeScheduleDetails = StartTimeScheduleDetails.StartTimeScheduleDetails;
const TargetIdsRemoveTargetsDetails = __importStar(require("./target-ids-remove-targets-details"));
exports.TargetIdsRemoveTargetsDetails = TargetIdsRemoveTargetsDetails.TargetIdsRemoveTargetsDetails;
const UpdateApplyActionDetails = __importStar(require("./update-apply-action-details"));
exports.UpdateApplyActionDetails = UpdateApplyActionDetails.UpdateApplyActionDetails;
const UpdateCleanupActionDetails = __importStar(require("./update-cleanup-action-details"));
exports.UpdateCleanupActionDetails = UpdateCleanupActionDetails.UpdateCleanupActionDetails;
const UpdateFiftyFiftyBatchingStrategyDetails = __importStar(require("./update-fifty-fifty-batching-strategy-details"));
exports.UpdateFiftyFiftyBatchingStrategyDetails = UpdateFiftyFiftyBatchingStrategyDetails.UpdateFiftyFiftyBatchingStrategyDetails;
const UpdateNonRollingBatchingStrategyDetails = __importStar(require("./update-non-rolling-batching-strategy-details"));
exports.UpdateNonRollingBatchingStrategyDetails = UpdateNonRollingBatchingStrategyDetails.UpdateNonRollingBatchingStrategyDetails;
const UpdatePatchFsuCycle = __importStar(require("./update-patch-fsu-cycle"));
exports.UpdatePatchFsuCycle = UpdatePatchFsuCycle.UpdatePatchFsuCycle;
const UpdatePrecheckActionDetails = __importStar(require("./update-precheck-action-details"));
exports.UpdatePrecheckActionDetails = UpdatePrecheckActionDetails.UpdatePrecheckActionDetails;
const UpdateRollbackActionDetails = __importStar(require("./update-rollback-action-details"));
exports.UpdateRollbackActionDetails = UpdateRollbackActionDetails.UpdateRollbackActionDetails;
const UpdateRollbackCycleActionDetails = __importStar(require("./update-rollback-cycle-action-details"));
exports.UpdateRollbackCycleActionDetails = UpdateRollbackCycleActionDetails.UpdateRollbackCycleActionDetails;
const UpdateSequentialBatchingStrategyDetails = __importStar(require("./update-sequential-batching-strategy-details"));
exports.UpdateSequentialBatchingStrategyDetails = UpdateSequentialBatchingStrategyDetails.UpdateSequentialBatchingStrategyDetails;
const UpdateServiceAvailabilityFactorBatchingStrategyDetails = __importStar(require("./update-service-availability-factor-batching-strategy-details"));
exports.UpdateServiceAvailabilityFactorBatchingStrategyDetails = UpdateServiceAvailabilityFactorBatchingStrategyDetails.UpdateServiceAvailabilityFactorBatchingStrategyDetails;
const UpdateStageActionDetails = __importStar(require("./update-stage-action-details"));
exports.UpdateStageActionDetails = UpdateStageActionDetails.UpdateStageActionDetails;
const UpdateStartTimeScheduleDetails = __importStar(require("./update-start-time-schedule-details"));
exports.UpdateStartTimeScheduleDetails = UpdateStartTimeScheduleDetails.UpdateStartTimeScheduleDetails;
const UpdateUpgradeFsuCycle = __importStar(require("./update-upgrade-fsu-cycle"));
exports.UpdateUpgradeFsuCycle = UpdateUpgradeFsuCycle.UpdateUpgradeFsuCycle;
const UpgradeDbCollectionDetails = __importStar(require("./upgrade-db-collection-details"));
exports.UpgradeDbCollectionDetails = UpgradeDbCollectionDetails.UpgradeDbCollectionDetails;
const UpgradeFsuCycle = __importStar(require("./upgrade-fsu-cycle"));
exports.UpgradeFsuCycle = UpgradeFsuCycle.UpgradeFsuCycle;
const UpgradeGiCollectionDetails = __importStar(require("./upgrade-gi-collection-details"));
exports.UpgradeGiCollectionDetails = UpgradeGiCollectionDetails.UpgradeGiCollectionDetails;
const VersionFsuTargetDetails = __importStar(require("./version-fsu-target-details"));
exports.VersionFsuTargetDetails = VersionFsuTargetDetails.VersionFsuTargetDetails;
const VmClusterTargetSummary = __importStar(require("./vm-cluster-target-summary"));
exports.VmClusterTargetSummary = VmClusterTargetSummary.VmClusterTargetSummary;
