// oci-fleetsoftwareupdate/lib/model/job-types.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobTypes = void 0;
/**
 * Fleet Sotware Update Job type.
 *
 **/
var JobTypes;
(function (JobTypes) {
    JobTypes["Stage"] = "STAGE";
    JobTypes["Precheck"] = "PRECHECK";
    JobTypes["Apply"] = "APPLY";
    JobTypes["RollbackAndRemoveTarget"] = "ROLLBACK_AND_REMOVE_TARGET";
    JobTypes["RollbackMaintenanceCycle"] = "ROLLBACK_MAINTENANCE_CYCLE";
    JobTypes["Cleanup"] = "CLEANUP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    JobTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(JobTypes = exports.JobTypes || (exports.JobTypes = {}));
(function (JobTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    JobTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    JobTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(JobTypes = exports.JobTypes || (exports.JobTypes = {}));
