// oci-demandsignal/lib/model/index.js
"use strict";
/**
 * OCI Control Center Demand Signal API
 * Use the OCI Control Center Demand Signal API to manage Demand Signals.
 * OpenAPI spec version: 20240430
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchRequireInstruction = exports.PatchReplaceInstruction = exports.PatchRemoveInstruction = exports.PatchProhibitInstruction = exports.PatchMoveInstruction = exports.PatchMergeInstruction = exports.PatchInsertMultipleInstruction = exports.PatchInsertInstruction = exports.UpdateOccDemandSignalDetails = exports.SortOrder = exports.PatchOccDemandSignalDetails = exports.PatchInstruction = exports.OccDemandSignalValue = exports.OccDemandSignalSummary = exports.OccDemandSignalData = exports.OccDemandSignalCollection = exports.OccDemandSignal = exports.CreateOccDemandSignalDetails = exports.ChangeOccDemandSignalCompartmentDetails = void 0;
const ChangeOccDemandSignalCompartmentDetails = __importStar(require("./change-occ-demand-signal-compartment-details"));
exports.ChangeOccDemandSignalCompartmentDetails = ChangeOccDemandSignalCompartmentDetails.ChangeOccDemandSignalCompartmentDetails;
const CreateOccDemandSignalDetails = __importStar(require("./create-occ-demand-signal-details"));
exports.CreateOccDemandSignalDetails = CreateOccDemandSignalDetails.CreateOccDemandSignalDetails;
const OccDemandSignal = __importStar(require("./occ-demand-signal"));
exports.OccDemandSignal = OccDemandSignal.OccDemandSignal;
const OccDemandSignalCollection = __importStar(require("./occ-demand-signal-collection"));
exports.OccDemandSignalCollection = OccDemandSignalCollection.OccDemandSignalCollection;
const OccDemandSignalData = __importStar(require("./occ-demand-signal-data"));
exports.OccDemandSignalData = OccDemandSignalData.OccDemandSignalData;
const OccDemandSignalSummary = __importStar(require("./occ-demand-signal-summary"));
exports.OccDemandSignalSummary = OccDemandSignalSummary.OccDemandSignalSummary;
const OccDemandSignalValue = __importStar(require("./occ-demand-signal-value"));
exports.OccDemandSignalValue = OccDemandSignalValue.OccDemandSignalValue;
const PatchInstruction = __importStar(require("./patch-instruction"));
exports.PatchInstruction = PatchInstruction.PatchInstruction;
const PatchOccDemandSignalDetails = __importStar(require("./patch-occ-demand-signal-details"));
exports.PatchOccDemandSignalDetails = PatchOccDemandSignalDetails.PatchOccDemandSignalDetails;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateOccDemandSignalDetails = __importStar(require("./update-occ-demand-signal-details"));
exports.UpdateOccDemandSignalDetails = UpdateOccDemandSignalDetails.UpdateOccDemandSignalDetails;
const PatchInsertInstruction = __importStar(require("./patch-insert-instruction"));
exports.PatchInsertInstruction = PatchInsertInstruction.PatchInsertInstruction;
const PatchInsertMultipleInstruction = __importStar(require("./patch-insert-multiple-instruction"));
exports.PatchInsertMultipleInstruction = PatchInsertMultipleInstruction.PatchInsertMultipleInstruction;
const PatchMergeInstruction = __importStar(require("./patch-merge-instruction"));
exports.PatchMergeInstruction = PatchMergeInstruction.PatchMergeInstruction;
const PatchMoveInstruction = __importStar(require("./patch-move-instruction"));
exports.PatchMoveInstruction = PatchMoveInstruction.PatchMoveInstruction;
const PatchProhibitInstruction = __importStar(require("./patch-prohibit-instruction"));
exports.PatchProhibitInstruction = PatchProhibitInstruction.PatchProhibitInstruction;
const PatchRemoveInstruction = __importStar(require("./patch-remove-instruction"));
exports.PatchRemoveInstruction = PatchRemoveInstruction.PatchRemoveInstruction;
const PatchReplaceInstruction = __importStar(require("./patch-replace-instruction"));
exports.PatchReplaceInstruction = PatchReplaceInstruction.PatchReplaceInstruction;
const PatchRequireInstruction = __importStar(require("./patch-require-instruction"));
exports.PatchRequireInstruction = PatchRequireInstruction.PatchRequireInstruction;
