// oci-fleetsoftwareupdate/lib/request/index.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListFsuJobsRequest = exports.ListFsuDiscoveryTargetsRequest = exports.ListFsuDiscoveriesRequest = exports.ListFsuCyclesRequest = exports.ListFsuCollectionsRequest = exports.ListFsuCollectionTargetsRequest = exports.ListFsuActionsRequest = void 0;
const ListFsuActionsRequest = __importStar(require("./list-fsu-actions-request"));
exports.ListFsuActionsRequest = ListFsuActionsRequest.ListFsuActionsRequest;
const ListFsuCollectionTargetsRequest = __importStar(require("./list-fsu-collection-targets-request"));
exports.ListFsuCollectionTargetsRequest = ListFsuCollectionTargetsRequest.ListFsuCollectionTargetsRequest;
const ListFsuCollectionsRequest = __importStar(require("./list-fsu-collections-request"));
exports.ListFsuCollectionsRequest = ListFsuCollectionsRequest.ListFsuCollectionsRequest;
const ListFsuCyclesRequest = __importStar(require("./list-fsu-cycles-request"));
exports.ListFsuCyclesRequest = ListFsuCyclesRequest.ListFsuCyclesRequest;
const ListFsuDiscoveriesRequest = __importStar(require("./list-fsu-discoveries-request"));
exports.ListFsuDiscoveriesRequest = ListFsuDiscoveriesRequest.ListFsuDiscoveriesRequest;
const ListFsuDiscoveryTargetsRequest = __importStar(require("./list-fsu-discovery-targets-request"));
exports.ListFsuDiscoveryTargetsRequest = ListFsuDiscoveryTargetsRequest.ListFsuDiscoveryTargetsRequest;
const ListFsuJobsRequest = __importStar(require("./list-fsu-jobs-request"));
exports.ListFsuJobsRequest = ListFsuJobsRequest.ListFsuJobsRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
