// oci-fleetsoftwareupdate/lib/request/list-fsu-collection-targets-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFsuCollectionTargetsRequest = void 0;
var ListFsuCollectionTargetsRequest;
(function (ListFsuCollectionTargetsRequest) {
    let Status;
    (function (Status) {
        Status["Idle"] = "IDLE";
        Status["ExecutingJob"] = "EXECUTING_JOB";
        Status["JobFailed"] = "JOB_FAILED";
    })(Status = ListFsuCollectionTargetsRequest.Status || (ListFsuCollectionTargetsRequest.Status = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["CurrentVersion"] = "currentVersion";
        SortBy["Status"] = "status";
    })(SortBy = ListFsuCollectionTargetsRequest.SortBy || (ListFsuCollectionTargetsRequest.SortBy = {}));
})(ListFsuCollectionTargetsRequest = exports.ListFsuCollectionTargetsRequest || (exports.ListFsuCollectionTargetsRequest = {}));
