// oci-dbmulticloud/lib/model/oracle-db-azure-connector.js
"use strict";
/**
 * Oracle Database MultiCloud Data plane Integration
 * 1. Oracle Azure Connector Resource: This is for installing Azure Arc Server in ExaCS VM Cluster.
  There are two way to install Azure Arc Server (Azure Identity) in ExaCS VMCluster.
    a. Using Bearer Access Token or
    b. By providing Authentication token

2. Oracle Azure Blob Container Resource: This is for to capture Azure Container details
   and same will be used in multiple ExaCS VMCluster to mount the Azure Container.

3. Oracle Azure Blob Mount Resource: This is for to mount Azure Container in ExaCS VMCluster
   using Oracle Azure Connector and Oracle Azure Blob Container Resource.

 * OpenAPI spec version: 20240501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OracleDbAzureConnector = void 0;
const model = __importStar(require("../model"));
var OracleDbAzureConnector;
(function (OracleDbAzureConnector) {
    let AzureIdentityMechanism;
    (function (AzureIdentityMechanism) {
        AzureIdentityMechanism["ArcAgent"] = "ARC_AGENT";
        AzureIdentityMechanism["ServicePrincipal"] = "SERVICE_PRINCIPAL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AzureIdentityMechanism["UnknownValue"] = "UNKNOWN_VALUE";
    })(AzureIdentityMechanism = OracleDbAzureConnector.AzureIdentityMechanism || (OracleDbAzureConnector.AzureIdentityMechanism = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = OracleDbAzureConnector.LifecycleState || (OracleDbAzureConnector.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "arcAgentNodes": obj.arcAgentNodes
                ? obj.arcAgentNodes.map(item => {
                    return model.ArcAgentNodes.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    OracleDbAzureConnector.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "arcAgentNodes": obj.arcAgentNodes
                ? obj.arcAgentNodes.map(item => {
                    return model.ArcAgentNodes.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    OracleDbAzureConnector.getDeserializedJsonObj = getDeserializedJsonObj;
})(OracleDbAzureConnector = exports.OracleDbAzureConnector || (exports.OracleDbAzureConnector = {}));
