// oci-core/lib/model/compute-gpu-memory-fabric.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](https://docs.oracle.com/iaas/Content/Network/Concepts/overview.htm),
[Compute](https://docs.oracle.com/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](https://docs.oracle.com/iaas/Content/Block/Concepts/overview.htm) services.
The required permissions are documented in the
[Details for the Core Services](https://docs.oracle.com/iaas/Content/Identity/Reference/corepolicyreference.htm) article.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComputeGpuMemoryFabric = void 0;
var ComputeGpuMemoryFabric;
(function (ComputeGpuMemoryFabric) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Occupied"] = "OCCUPIED";
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Degraded"] = "DEGRADED";
        LifecycleState["Unavailable"] = "UNAVAILABLE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ComputeGpuMemoryFabric.LifecycleState || (ComputeGpuMemoryFabric.LifecycleState = {}));
    let FabricHealth;
    (function (FabricHealth) {
        FabricHealth["Healthy"] = "HEALTHY";
        FabricHealth["Unhealthy"] = "UNHEALTHY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        FabricHealth["UnknownValue"] = "UNKNOWN_VALUE";
    })(FabricHealth = ComputeGpuMemoryFabric.FabricHealth || (ComputeGpuMemoryFabric.FabricHealth = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ComputeGpuMemoryFabric.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ComputeGpuMemoryFabric.getDeserializedJsonObj = getDeserializedJsonObj;
})(ComputeGpuMemoryFabric = exports.ComputeGpuMemoryFabric || (exports.ComputeGpuMemoryFabric = {}));
