// oci-ailanguage/lib/model/job.js
"use strict";
/**
 * Language API
 * OCI Language Service solutions can help enterprise customers integrate AI into their products immediately using our proven,
pre-trained and custom models or containers, without a need to set up an house team of AI and ML experts.
This allows enterprises to focus on business drivers and development work rather than AI and ML operations, which shortens the time to market.

 * OpenAPI spec version: 20221001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Job = void 0;
const model = __importStar(require("../model"));
var Job;
(function (Job) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Accepted"] = "ACCEPTED";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        LifecycleState["Succeeded"] = "SUCCEEDED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Canceling"] = "CANCELING";
        LifecycleState["Canceled"] = "CANCELED";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Job.LifecycleState || (Job.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "inputLocation": obj.inputLocation
                ? model.InputLocation.getJsonObj(obj.inputLocation)
                : undefined,
            "inputConfiguration": obj.inputConfiguration
                ? model.InputConfiguration.getJsonObj(obj.inputConfiguration)
                : undefined,
            "modelMetadataDetails": obj.modelMetadataDetails
                ? obj.modelMetadataDetails.map(item => {
                    return model.ModelMetadataDetails.getJsonObj(item);
                })
                : undefined,
            "outputLocation": obj.outputLocation
                ? model.ObjectPrefixOutputLocation.getJsonObj(obj.outputLocation)
                : undefined
        });
        return jsonObj;
    }
    Job.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "inputLocation": obj.inputLocation
                ? model.InputLocation.getDeserializedJsonObj(obj.inputLocation)
                : undefined,
            "inputConfiguration": obj.inputConfiguration
                ? model.InputConfiguration.getDeserializedJsonObj(obj.inputConfiguration)
                : undefined,
            "modelMetadataDetails": obj.modelMetadataDetails
                ? obj.modelMetadataDetails.map(item => {
                    return model.ModelMetadataDetails.getDeserializedJsonObj(item);
                })
                : undefined,
            "outputLocation": obj.outputLocation
                ? model.ObjectPrefixOutputLocation.getDeserializedJsonObj(obj.outputLocation)
                : undefined
        });
        return jsonObj;
    }
    Job.getDeserializedJsonObj = getDeserializedJsonObj;
})(Job = exports.Job || (exports.Job = {}));
