// oci-redis/lib/model/index.js
"use strict";
/**
 * OCI Cache API
 * Use the OCI Cache API to create and manage clusters. A cluster is a memory-based storage solution. For more information, see [OCI Cache](https://docs.oracle.com/iaas/Content/ocicache/home.htm).
 * OpenAPI spec version: 20220315
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PasswordAuthenticationMode = exports.IamAuthenticationMode = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateRedisClusterDetails = exports.UpdateOciCacheUserDetails = exports.SortOrder = exports.RedisNodeCollection = exports.RedisClusterSummary = exports.RedisClusterCollection = exports.RedisCluster = exports.OperationType = exports.OperationStatus = exports.OciCacheUserSummary = exports.OciCacheUserCollection = exports.OciCacheUser = exports.NodeSummary = exports.NodeCollection = exports.Node = exports.IdentityTokenDetailsResponse = exports.DetachOciCacheUsersDetails = exports.CreateRedisClusterDetails = exports.CreateOciCacheUserDetails = exports.CreateIdentityTokenDetails = exports.ChangeRedisClusterCompartmentDetails = exports.ChangeOciCacheUserCompartmentDetails = exports.AuthenticationMode = exports.AttachedOciCacheUser = exports.AttachedOciCacheCluster = exports.AttachOciCacheUsersDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AttachOciCacheUsersDetails = __importStar(require("./attach-oci-cache-users-details"));
exports.AttachOciCacheUsersDetails = AttachOciCacheUsersDetails.AttachOciCacheUsersDetails;
const AttachedOciCacheCluster = __importStar(require("./attached-oci-cache-cluster"));
exports.AttachedOciCacheCluster = AttachedOciCacheCluster.AttachedOciCacheCluster;
const AttachedOciCacheUser = __importStar(require("./attached-oci-cache-user"));
exports.AttachedOciCacheUser = AttachedOciCacheUser.AttachedOciCacheUser;
const AuthenticationMode = __importStar(require("./authentication-mode"));
exports.AuthenticationMode = AuthenticationMode.AuthenticationMode;
const ChangeOciCacheUserCompartmentDetails = __importStar(require("./change-oci-cache-user-compartment-details"));
exports.ChangeOciCacheUserCompartmentDetails = ChangeOciCacheUserCompartmentDetails.ChangeOciCacheUserCompartmentDetails;
const ChangeRedisClusterCompartmentDetails = __importStar(require("./change-redis-cluster-compartment-details"));
exports.ChangeRedisClusterCompartmentDetails = ChangeRedisClusterCompartmentDetails.ChangeRedisClusterCompartmentDetails;
const CreateIdentityTokenDetails = __importStar(require("./create-identity-token-details"));
exports.CreateIdentityTokenDetails = CreateIdentityTokenDetails.CreateIdentityTokenDetails;
const CreateOciCacheUserDetails = __importStar(require("./create-oci-cache-user-details"));
exports.CreateOciCacheUserDetails = CreateOciCacheUserDetails.CreateOciCacheUserDetails;
const CreateRedisClusterDetails = __importStar(require("./create-redis-cluster-details"));
exports.CreateRedisClusterDetails = CreateRedisClusterDetails.CreateRedisClusterDetails;
const DetachOciCacheUsersDetails = __importStar(require("./detach-oci-cache-users-details"));
exports.DetachOciCacheUsersDetails = DetachOciCacheUsersDetails.DetachOciCacheUsersDetails;
const IdentityTokenDetailsResponse = __importStar(require("./identity-token-details-response"));
exports.IdentityTokenDetailsResponse = IdentityTokenDetailsResponse.IdentityTokenDetailsResponse;
const Node = __importStar(require("./node"));
exports.Node = Node.Node;
const NodeCollection = __importStar(require("./node-collection"));
exports.NodeCollection = NodeCollection.NodeCollection;
const NodeSummary = __importStar(require("./node-summary"));
exports.NodeSummary = NodeSummary.NodeSummary;
const OciCacheUser = __importStar(require("./oci-cache-user"));
exports.OciCacheUser = OciCacheUser.OciCacheUser;
const OciCacheUserCollection = __importStar(require("./oci-cache-user-collection"));
exports.OciCacheUserCollection = OciCacheUserCollection.OciCacheUserCollection;
const OciCacheUserSummary = __importStar(require("./oci-cache-user-summary"));
exports.OciCacheUserSummary = OciCacheUserSummary.OciCacheUserSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const RedisCluster = __importStar(require("./redis-cluster"));
exports.RedisCluster = RedisCluster.RedisCluster;
const RedisClusterCollection = __importStar(require("./redis-cluster-collection"));
exports.RedisClusterCollection = RedisClusterCollection.RedisClusterCollection;
const RedisClusterSummary = __importStar(require("./redis-cluster-summary"));
exports.RedisClusterSummary = RedisClusterSummary.RedisClusterSummary;
const RedisNodeCollection = __importStar(require("./redis-node-collection"));
exports.RedisNodeCollection = RedisNodeCollection.RedisNodeCollection;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateOciCacheUserDetails = __importStar(require("./update-oci-cache-user-details"));
exports.UpdateOciCacheUserDetails = UpdateOciCacheUserDetails.UpdateOciCacheUserDetails;
const UpdateRedisClusterDetails = __importStar(require("./update-redis-cluster-details"));
exports.UpdateRedisClusterDetails = UpdateRedisClusterDetails.UpdateRedisClusterDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const IamAuthenticationMode = __importStar(require("./iam-authentication-mode"));
exports.IamAuthenticationMode = IamAuthenticationMode.IamAuthenticationMode;
const PasswordAuthenticationMode = __importStar(require("./password-authentication-mode"));
exports.PasswordAuthenticationMode = PasswordAuthenticationMode.PasswordAuthenticationMode;
