// oci-redis/lib/model/redis-cluster.js
"use strict";
/**
 * OCI Cache API
 * Use the OCI Cache API to create and manage clusters. A cluster is a memory-based storage solution. For more information, see [OCI Cache](https://docs.oracle.com/iaas/Content/ocicache/home.htm).
 * OpenAPI spec version: 20220315
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RedisCluster = void 0;
const model = __importStar(require("../model"));
var RedisCluster;
(function (RedisCluster) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = RedisCluster.LifecycleState || (RedisCluster.LifecycleState = {}));
    let SoftwareVersion;
    (function (SoftwareVersion) {
        SoftwareVersion["V705"] = "V7_0_5";
        SoftwareVersion["Redis70"] = "REDIS_7_0";
        SoftwareVersion["Valkey72"] = "VALKEY_7_2";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SoftwareVersion["UnknownValue"] = "UNKNOWN_VALUE";
    })(SoftwareVersion = RedisCluster.SoftwareVersion || (RedisCluster.SoftwareVersion = {}));
    let ClusterMode;
    (function (ClusterMode) {
        ClusterMode["Sharded"] = "SHARDED";
        ClusterMode["Nonsharded"] = "NONSHARDED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ClusterMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(ClusterMode = RedisCluster.ClusterMode || (RedisCluster.ClusterMode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "nodeCollection": obj.nodeCollection
                ? model.NodeCollection.getJsonObj(obj.nodeCollection)
                : undefined
        });
        return jsonObj;
    }
    RedisCluster.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "nodeCollection": obj.nodeCollection
                ? model.NodeCollection.getDeserializedJsonObj(obj.nodeCollection)
                : undefined
        });
        return jsonObj;
    }
    RedisCluster.getDeserializedJsonObj = getDeserializedJsonObj;
})(RedisCluster = exports.RedisCluster || (exports.RedisCluster = {}));
