// oci-datascience/lib/request/list-work-requests-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = void 0;
var ListWorkRequestsRequest;
(function (ListWorkRequestsRequest) {
    let OperationType;
    (function (OperationType) {
        OperationType["NotebookSessionCreate"] = "NOTEBOOK_SESSION_CREATE";
        OperationType["NotebookSessionDelete"] = "NOTEBOOK_SESSION_DELETE";
        OperationType["NotebookSessionActivate"] = "NOTEBOOK_SESSION_ACTIVATE";
        OperationType["NotebookSessionDeactivate"] = "NOTEBOOK_SESSION_DEACTIVATE";
        OperationType["ModelversionsetDelete"] = "MODELVERSIONSET_DELETE";
        OperationType["ExportModelArtifact"] = "EXPORT_MODEL_ARTIFACT";
        OperationType["ImportModelArtifact"] = "IMPORT_MODEL_ARTIFACT";
        OperationType["ModelDeploymentCreate"] = "MODEL_DEPLOYMENT_CREATE";
        OperationType["ModelDeploymentDelete"] = "MODEL_DEPLOYMENT_DELETE";
        OperationType["ModelDeploymentActivate"] = "MODEL_DEPLOYMENT_ACTIVATE";
        OperationType["ModelDeploymentDeactivate"] = "MODEL_DEPLOYMENT_DEACTIVATE";
        OperationType["ModelDeploymentUpdate"] = "MODEL_DEPLOYMENT_UPDATE";
        OperationType["ProjectDelete"] = "PROJECT_DELETE";
        OperationType["WorkrequestCancel"] = "WORKREQUEST_CANCEL";
        OperationType["JobDelete"] = "JOB_DELETE";
        OperationType["PipelineCreate"] = "PIPELINE_CREATE";
        OperationType["PipelineDelete"] = "PIPELINE_DELETE";
        OperationType["PipelineRunCreate"] = "PIPELINE_RUN_CREATE";
        OperationType["PipelineRunCancel"] = "PIPELINE_RUN_CANCEL";
        OperationType["PipelineRunDelete"] = "PIPELINE_RUN_DELETE";
        OperationType["MlApplicationPackageUpload"] = "ML_APPLICATION_PACKAGE_UPLOAD";
        OperationType["MlApplicationTriggerStart"] = "ML_APPLICATION_TRIGGER_START";
        OperationType["MlApplicationImplementationDelete"] = "ML_APPLICATION_IMPLEMENTATION_DELETE";
        OperationType["MlApplicationImplementationUpdate"] = "ML_APPLICATION_IMPLEMENTATION_UPDATE";
        OperationType["MlApplicationImplementationMove"] = "ML_APPLICATION_IMPLEMENTATION_MOVE";
        OperationType["MlApplicationInstanceCreate"] = "ML_APPLICATION_INSTANCE_CREATE";
        OperationType["MlApplicationInstanceUpdate"] = "ML_APPLICATION_INSTANCE_UPDATE";
        OperationType["MlApplicationInstanceDelete"] = "ML_APPLICATION_INSTANCE_DELETE";
        OperationType["MlApplicationInstanceMove"] = "ML_APPLICATION_INSTANCE_MOVE";
        OperationType["MlApplicationInstanceViewCreate"] = "ML_APPLICATION_INSTANCE_VIEW_CREATE";
        OperationType["MlApplicationInstanceViewUpdate"] = "ML_APPLICATION_INSTANCE_VIEW_UPDATE";
        OperationType["MlApplicationInstanceViewDelete"] = "ML_APPLICATION_INSTANCE_VIEW_DELETE";
        OperationType["MlApplicationInstanceViewUpgrade"] = "ML_APPLICATION_INSTANCE_VIEW_UPGRADE";
        OperationType["MlApplicationInstanceViewMove"] = "ML_APPLICATION_INSTANCE_VIEW_MOVE";
        OperationType["PrivateEndpointCreate"] = "PRIVATE_ENDPOINT_CREATE";
        OperationType["PrivateEndpointDelete"] = "PRIVATE_ENDPOINT_DELETE";
        OperationType["PrivateEndpointMove"] = "PRIVATE_ENDPOINT_MOVE";
        OperationType["PrivateEndpointUpdate"] = "PRIVATE_ENDPOINT_UPDATE";
        OperationType["ScheduleCreate"] = "SCHEDULE_CREATE";
        OperationType["ScheduleUpdate"] = "SCHEDULE_UPDATE";
        OperationType["ScheduleDelete"] = "SCHEDULE_DELETE";
        OperationType["ScheduleMove"] = "SCHEDULE_MOVE";
        OperationType["ScheduleActivate"] = "SCHEDULE_ACTIVATE";
        OperationType["ScheduleDeactivate"] = "SCHEDULE_DEACTIVATE";
        OperationType["RegisterModelArtifact"] = "REGISTER_MODEL_ARTIFACT";
        OperationType["RestoreArchivedModel"] = "RESTORE_ARCHIVED_MODEL";
    })(OperationType = ListWorkRequestsRequest.OperationType || (ListWorkRequestsRequest.OperationType = {}));
    let Status;
    (function (Status) {
        Status["Accepted"] = "ACCEPTED";
        Status["InProgress"] = "IN_PROGRESS";
        Status["Failed"] = "FAILED";
        Status["Succeeded"] = "SUCCEEDED";
        Status["Canceling"] = "CANCELING";
        Status["Canceled"] = "CANCELED";
    })(Status = ListWorkRequestsRequest.Status || (ListWorkRequestsRequest.Status = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListWorkRequestsRequest.SortOrder || (ListWorkRequestsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["OperationType"] = "operationType";
        SortBy["Status"] = "status";
        SortBy["TimeAccepted"] = "timeAccepted";
    })(SortBy = ListWorkRequestsRequest.SortBy || (ListWorkRequestsRequest.SortBy = {}));
})(ListWorkRequestsRequest = exports.ListWorkRequestsRequest || (exports.ListWorkRequestsRequest = {}));
