// oci-datascience/lib/request/list-data-science-private-endpoints-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDataSciencePrivateEndpointsRequest = void 0;
var ListDataSciencePrivateEndpointsRequest;
(function (ListDataSciencePrivateEndpointsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
    })(SortBy = ListDataSciencePrivateEndpointsRequest.SortBy || (ListDataSciencePrivateEndpointsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListDataSciencePrivateEndpointsRequest.SortOrder || (ListDataSciencePrivateEndpointsRequest.SortOrder = {}));
})(ListDataSciencePrivateEndpointsRequest = exports.ListDataSciencePrivateEndpointsRequest || (exports.ListDataSciencePrivateEndpointsRequest = {}));
