// oci-datascience/lib/request/list-ml-application-implementation-versions-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMlApplicationImplementationVersionsRequest = void 0;
var ListMlApplicationImplementationVersionsRequest;
(function (ListMlApplicationImplementationVersionsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListMlApplicationImplementationVersionsRequest.SortOrder || (ListMlApplicationImplementationVersionsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["Name"] = "name";
    })(SortBy = ListMlApplicationImplementationVersionsRequest.SortBy || (ListMlApplicationImplementationVersionsRequest.SortBy = {}));
})(ListMlApplicationImplementationVersionsRequest = exports.ListMlApplicationImplementationVersionsRequest || (exports.ListMlApplicationImplementationVersionsRequest = {}));
