// oci-loganalytics/lib/model/dependent-parser.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DependentParser = void 0;
const model = __importStar(require("../model"));
var DependentParser;
(function (DependentParser) {
    let ParserType;
    (function (ParserType) {
        ParserType["Xml"] = "XML";
        ParserType["Json"] = "JSON";
        ParserType["Regex"] = "REGEX";
        ParserType["Odl"] = "ODL";
        ParserType["Delimited"] = "DELIMITED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ParserType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ParserType = DependentParser.ParserType || (DependentParser.ParserType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dependencies": obj.dependencies
                ? obj.dependencies.map(item => {
                    return model.Dependency.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DependentParser.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dependencies": obj.dependencies
                ? obj.dependencies.map(item => {
                    return model.Dependency.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DependentParser.getDeserializedJsonObj = getDeserializedJsonObj;
})(DependentParser = exports.DependentParser || (exports.DependentParser = {}));
