// oci-loganalytics/lib/model/endpoint-proxy.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointProxy = void 0;
var EndpointProxy;
(function (EndpointProxy) {
    let CredentialType;
    (function (CredentialType) {
        CredentialType["None"] = "NONE";
        CredentialType["BasicAuth"] = "BASIC_AUTH";
        CredentialType["Token"] = "TOKEN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        CredentialType["UnknownValue"] = "UNKNOWN_VALUE";
    })(CredentialType = EndpointProxy.CredentialType || (EndpointProxy.CredentialType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    EndpointProxy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    EndpointProxy.getDeserializedJsonObj = getDeserializedJsonObj;
})(EndpointProxy = exports.EndpointProxy || (exports.EndpointProxy = {}));
