// oci-loganalytics/lib/model/index.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisableAutoAssociationDetail = exports.DependentSource = exports.DependentParser = exports.Dependency = exports.DeleteLogAnalyticsAssociationDetails = exports.DeleteLogAnalyticsAssociation = exports.CredentialEndpoint = exports.CreationSourceType = exports.CreationSource = exports.CreateScheduledTaskDetails = exports.CreateLogAnalyticsObjectCollectionRuleDetails = exports.CreateLogAnalyticsLogGroupDetails = exports.CreateLogAnalyticsEntityTypeDetails = exports.CreateLogAnalyticsEntityDetails = exports.CreateLogAnalyticsEmBridgeDetails = exports.CreateIngestTimeRuleDetails = exports.ConfigLifecycleState = exports.ConditionBlock = exports.CompareLineResult = exports.CompareContentResult = exports.CompareContentDetails = exports.ColumnNameCollection = exports.ColumnName = exports.CharEncodingCollection = exports.ChangeScheduledTaskCompartmentDetails = exports.ChangeLookupCompartmentDetails = exports.ChangeLogAnalyticsObjectCollectionRuleCompartmentDetails = exports.ChangeLogAnalyticsLogGroupCompartmentDetails = exports.ChangeLogAnalyticsEntityCompartmentDetails = exports.ChangeLogAnalyticsEmBridgeCompartmentDetails = exports.ChangeIngestTimeRuleCompartmentDetails = exports.BucketRange = exports.AutoLookups = exports.AutoAssociationState = exports.AutoAssociationCollection = exports.AssociationSummaryReport = exports.AssociationProperty = exports.AssociableEntityCollection = exports.AssociableEntity = exports.AssignEncryptionKeyDetails = exports.Argument = exports.ArchivingConfiguration = exports.AddEntityAssociationDetails = exports.ActionTypes = exports.Action = exports.AbstractParserTestResultLogLine = exports.AbstractParserTestResultLogEntry = exports.AbstractField = exports.AbstractCommandDescriptor = exports.AbstractColumn = void 0;
exports.IngestTimeRuleSummary = exports.IngestTimeRuleResource = exports.IngestTimeRuleCondition = exports.IngestTimeRuleAdditionalFieldCondition = exports.IngestTimeRuleAction = exports.IngestTimeRule = exports.Indexes = exports.GenericConditionBlock = exports.FilterOutput = exports.FilterDetails = exports.Filter = exports.FileValidationResponse = exports.FieldValue = exports.FieldSummaryReport = exports.ExtractLogHeaderResults = exports.ExtractLogHeaderDetails = exports.ExtractLogFieldResults = exports.ExtendedFieldsValidationResult = exports.ExportDetails = exports.ExportContent = exports.EventTypeDetails = exports.EventTypeCollection = exports.EventType = exports.EstimateReleaseDataSizeResult = exports.EstimateReleaseDataSizeDetails = exports.EstimateRecallDataSizeResult = exports.EstimateRecallDataSizeDetails = exports.EstimatePurgeDataSizeResult = exports.EstimatePurgeDataSizeDetails = exports.ErrorDetails = exports.EntityTypeProperty = exports.EntityLifecycleStates = exports.EntityCloudType = exports.EndpointResult = exports.EndpointResponse = exports.EndpointRequest = exports.EndpointProxy = exports.EndpointCredentials = exports.EncryptionKeyType = exports.EncryptionKeySource = exports.EncryptionKeyInfoSummary = exports.EncryptionKeyInfoCollection = exports.EnableAutoAssociationDetails = exports.EnableAutoAssociationDetail = exports.EmBridgeLifecycleStates = exports.EmBridgeLatestImportProcessingStatus = exports.EffectivePropertySummary = exports.EffectivePropertyCollection = exports.EfdRegexResult = exports.DisableAutoAssociationDetails = void 0;
exports.LogAnalyticsLabelCollection = exports.LogAnalyticsLabelAlias = exports.LogAnalyticsLabel = exports.LogAnalyticsImportCustomContent = exports.LogAnalyticsImportCustomChangeList = exports.LogAnalyticsFieldUsages = exports.LogAnalyticsFieldSummary = exports.LogAnalyticsFieldCollection = exports.LogAnalyticsField = exports.LogAnalyticsExtendedField = exports.LogAnalyticsEntityTypeSummary = exports.LogAnalyticsEntityTypeCollection = exports.LogAnalyticsEntityType = exports.LogAnalyticsEntityTopologySummary = exports.LogAnalyticsEntityTopologyLinkCollection = exports.LogAnalyticsEntityTopologyLink = exports.LogAnalyticsEntityTopologyCollection = exports.LogAnalyticsEntitySummaryReport = exports.LogAnalyticsEntitySummary = exports.LogAnalyticsEntityCollection = exports.LogAnalyticsEntity = exports.LogAnalyticsEndpoint = exports.LogAnalyticsEmBridgeSummaryReport = exports.LogAnalyticsEmBridgeSummary = exports.LogAnalyticsEmBridgeCollection = exports.LogAnalyticsEmBridge = exports.LogAnalyticsConfigWorkRequestSummary = exports.LogAnalyticsConfigWorkRequestPayload = exports.LogAnalyticsConfigWorkRequestCollection = exports.LogAnalyticsConfigWorkRequest = exports.LogAnalyticsCategoryCollection = exports.LogAnalyticsCategory = exports.LogAnalyticsAssociationParameterCollection = exports.LogAnalyticsAssociationParameter = exports.LogAnalyticsAssociationCollection = exports.LogAnalyticsAssociation = exports.LogAnalyticsAssociatedEntityCollection = exports.LogAnalyticsAssociatedEntity = exports.LifecycleStates = exports.Level = exports.LastExecutionStatus = exports.LabelSummaryReport = exports.LabelSourceSummary = exports.LabelSourceCollection = exports.LabelPriorityCollection = exports.LabelPriority = exports.LabelNames = exports.JobModeFilter = exports.JobMode = exports.IngestTimeRuleSummaryCollection = void 0;
exports.LogAnalyticsSourceExtendedFieldDefinition = exports.LogAnalyticsSourceEntityType = exports.LogAnalyticsSourceDataFilter = exports.LogAnalyticsSourceCollection = exports.LogAnalyticsSource = exports.LogAnalyticsResourceCategoryDetails = exports.LogAnalyticsResourceCategoryCollection = exports.LogAnalyticsResourceCategory = exports.LogAnalyticsProperty = exports.LogAnalyticsPreferenceDetails = exports.LogAnalyticsPreferenceCollection = exports.LogAnalyticsPreference = exports.LogAnalyticsPatternFilter = exports.LogAnalyticsParserSummary = exports.LogAnalyticsParserMetaPluginParameter = exports.LogAnalyticsParserMetaPluginCollection = exports.LogAnalyticsParserMetaPlugin = exports.LogAnalyticsParserFunctionParameter = exports.LogAnalyticsParserFunctionCollection = exports.LogAnalyticsParserFunction = exports.LogAnalyticsParserFilter = exports.LogAnalyticsParserField = exports.LogAnalyticsParserCollection = exports.LogAnalyticsParser = exports.LogAnalyticsParameter = exports.LogAnalyticsOperationTypes = exports.LogAnalyticsObjectCollectionRuleSummary = exports.LogAnalyticsObjectCollectionRuleCollection = exports.LogAnalyticsObjectCollectionRule = exports.LogAnalyticsMetric = exports.LogAnalyticsMetadataSummary = exports.LogAnalyticsMetadataDetails = exports.LogAnalyticsMetadataCollection = exports.LogAnalyticsMetadata = exports.LogAnalyticsMetaSourceTypeCollection = exports.LogAnalyticsMetaSourceType = exports.LogAnalyticsMetaFunctionCollection = exports.LogAnalyticsMetaFunctionArgument = exports.LogAnalyticsMetaFunction = exports.LogAnalyticsLookupFields = exports.LogAnalyticsLookupCollection = exports.LogAnalyticsLookup = exports.LogAnalyticsLogGroupSummaryCollection = exports.LogAnalyticsLogGroupSummary = exports.LogAnalyticsLogGroup = exports.LogAnalyticsLabelView = exports.LogAnalyticsLabelSummary = exports.LogAnalyticsLabelOperatorCollection = exports.LogAnalyticsLabelOperator = exports.LogAnalyticsLabelDefinition = void 0;
exports.QueryAggregation = exports.PurgeStorageDataDetails = exports.PropertyOverride = exports.PropertyMetadataSummaryCollection = exports.PropertyMetadataSummary = exports.PropertyDefinition = exports.PayloadType = exports.PatternOverride = exports.ParserTestResult = exports.ParserSummaryReport = exports.ParserActionSummaryCollection = exports.ParserActionSummary = exports.ParserAction = exports.ParsedField = exports.ParsedContent = exports.ParseQueryOutput = exports.ParseQueryDetails = exports.OverlappingRecallSummary = exports.OverlappingRecallCollection = exports.OperationStatus = exports.ObjectCollectionRuleLifecycleStates = exports.ObjectCollectionRuleCollectionTypes = exports.NamespaceSummary = exports.NamespaceCollection = exports.Namespace = exports.NameValuePair = exports.MetricExtraction = exports.MatchInfo = exports.LookupSummaryReport = exports.LookupField = exports.LogTypes = exports.LogSetsCount = exports.LogSetKeyTypes = exports.LogSetCollection = exports.LogListEndpoint = exports.LogGroupSummaryReport = exports.LogEndpoint = exports.LogAnalyticsWarningCollection = exports.LogAnalyticsWarning = exports.LogAnalyticsTemplateSummary = exports.LogAnalyticsTemplateCollection = exports.LogAnalyticsTemplate = exports.LogAnalyticsSourceSummary = exports.LogAnalyticsSourcePatternCollection = exports.LogAnalyticsSourcePattern = exports.LogAnalyticsSourceMetric = exports.LogAnalyticsSourceMetadataField = exports.LogAnalyticsSourceLabelCondition = exports.LogAnalyticsSourceFunction = exports.LogAnalyticsSourceExtendedFieldDefinitionCollection = void 0;
exports.TemplateDetails = exports.TaskType = exports.SuggestOutput = exports.SuggestDetails = exports.Success = exports.SubSystemName = exports.StreamCursorTypes = exports.StorageWorkRequestSummary = exports.StorageWorkRequestCollection = exports.StorageWorkRequest = exports.StorageUsage = exports.StorageOperationType = exports.StorageDataType = exports.Storage = exports.StepInfo = exports.StatusSummary = exports.SourceValidateResults = exports.SourceValidateDetails = exports.SourceSummaryReport = exports.SourceMappingResponse = exports.ScopeFilter = exports.SchedulerResource = exports.ScheduledTaskSummary = exports.ScheduledTaskCollection = exports.ScheduledTask = exports.Schedule = exports.RuleSummaryReport = exports.RuleSummaryCollection = exports.RuleSummary = exports.RuleKind = exports.Rule = exports.ResultColumn = exports.RemoveEntityAssociationsDetails = exports.ReleaseRecalledDataDetails = exports.RegexMatchResult = exports.RecalledInfoCollection = exports.RecalledInfo = exports.RecalledDataSize = exports.RecalledDataInfo = exports.RecalledDataCollection = exports.RecalledData = exports.RecallStatus = exports.RecallDefinition = exports.RecallCount = exports.RecallArchivedDataDetails = exports.QueryWorkRequestSummary = exports.QueryWorkRequestCollection = exports.QueryWorkRequest = exports.QueryOperationType = exports.QueryDetails = void 0;
exports.AddInsightsCommandDescriptor = exports.AddFieldsCommandDescriptor = exports.WorkRequestSummary = exports.WorkRequestStatus = exports.WorkRequestResource = exports.WorkRequestLogCollection = exports.WorkRequestLog = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.WarningReferenceDetails = exports.Violation = exports.VerifyOutput = exports.VariableDefinition = exports.ValueType = exports.ValidateLabelConditionResult = exports.ValidateLabelConditionDetails = exports.ValidateEndpointResult = exports.UsageStatusItem = exports.UpsertLogAnalyticsSourceDetails = exports.UpsertLogAnalyticsParserDetails = exports.UpsertLogAnalyticsLabelDetails = exports.UpsertLogAnalyticsFieldDetails = exports.UpsertLogAnalyticsAssociationDetails = exports.UpsertLogAnalyticsAssociation = exports.UploadWarningSummary = exports.UploadWarningCollection = exports.UploadSummary = exports.UploadFileSummary = exports.UploadFileStatus = exports.UploadFileCollection = exports.UploadCollection = exports.Upload = exports.UpdateStorageDetails = exports.UpdateScheduledTaskDetails = exports.UpdateLookupMetadataDetails = exports.UpdateLogAnalyticsObjectCollectionRuleDetails = exports.UpdateLogAnalyticsLogGroupDetails = exports.UpdateLogAnalyticsEntityTypeDetails = exports.UpdateLogAnalyticsEntityDetails = exports.UpdateLogAnalyticsEmBridgeDetails = exports.UnprocessedDataBucket = exports.UiParserTestMetadata = exports.TimezoneCollection = exports.TimeStatsCluster = exports.TimeRange = exports.TestParserPayloadDetails = exports.TemplateParams = exports.TemplateFacet = void 0;
exports.LookupCommandDescriptor = exports.LogTypeEndpoint = exports.LogListTypeEndpoint = exports.LiteralArgument = exports.LinkDetailsCommandDescriptor = exports.LinkCommandDescriptor = exports.JsonExtractCommandDescriptor = exports.IngestTimeRuleMetricExtractionAction = exports.IngestTimeRuleFieldCondition = exports.HighlightRowsCommandDescriptor = exports.HighlightGroupsCommandDescriptor = exports.HighlightCommandDescriptor = exports.HeadCommandDescriptor = exports.GeoStatsCommandDescriptor = exports.FunctionField = exports.FrequentCommandDescriptor = exports.FixedFrequencySchedule = exports.FieldsCommandDescriptor = exports.FieldsAddRemoveField = exports.FieldSummaryCommandDescriptor = exports.FieldArgument = exports.Field = exports.ExtractCommandDescriptor = exports.EventStatsCommandDescriptor = exports.EvalCommandDescriptor = exports.DistinctCommandDescriptor = exports.DemoModeCommandDescriptor = exports.DeltaCommandDescriptor = exports.DeleteCommandDescriptor = exports.DedupCommandDescriptor = exports.CronSchedule = exports.CreateViewCommandDescriptor = exports.CreateTableCommandDescriptor = exports.CreateStandardTaskDetails = exports.CreateAccelerationTaskDetails = exports.CompareCommandDescriptor = exports.CommandDescriptor = exports.Column = exports.ClusterSplitCommandDescriptor = exports.ClusterDetailsCommandDescriptor = exports.ClusterCompareCommandDescriptor = exports.ClusterCommandDescriptor = exports.ClassifyCommandDescriptor = exports.ClassifyColumn = exports.ChartDataColumn = exports.ChartColumn = exports.BucketCommandDescriptor = exports.BottomCommandDescriptor = exports.AutoSchedule = exports.AnomalyCommandDescriptor = void 0;
exports.XmlExtractCommandDescriptor = exports.WhereCommandDescriptor = exports.UpdateTableCommandDescriptor = exports.UpdateStandardTaskDetails = exports.TrendColumn = exports.TopCommandDescriptor = exports.TimeStatsDataColumn = exports.TimeStatsCommandDescriptor = exports.TimeStatsColumn = exports.TimeCompareCommandDescriptor = exports.TimeColumn = exports.TimeClusterDataColumn = exports.TimeClusterCommandDescriptor = exports.TimeClusterColumn = exports.TailCommandDescriptor = exports.TableColumn = exports.StreamAction = exports.StatsCommandDescriptor = exports.StandardTask = exports.SortField = exports.SortCommandDescriptor = exports.SequenceCommandDescriptor = exports.SearchLookupCommandDescriptor = exports.SearchCommandDescriptor = exports.RenameCommandDescriptor = exports.RegexCommandDescriptor = exports.RareCommandDescriptor = exports.PurgeAction = exports.OutlierCommandDescriptor = exports.NlpCommandDescriptor = exports.MultiSearchCommandDescriptor = exports.ModuleCommandDescriptor = exports.MapCommandDescriptor = exports.MacroCommandDescriptor = void 0;
const AbstractColumn = __importStar(require("./abstract-column"));
exports.AbstractColumn = AbstractColumn.AbstractColumn;
const AbstractCommandDescriptor = __importStar(require("./abstract-command-descriptor"));
exports.AbstractCommandDescriptor = AbstractCommandDescriptor.AbstractCommandDescriptor;
const AbstractField = __importStar(require("./abstract-field"));
exports.AbstractField = AbstractField.AbstractField;
const AbstractParserTestResultLogEntry = __importStar(require("./abstract-parser-test-result-log-entry"));
exports.AbstractParserTestResultLogEntry = AbstractParserTestResultLogEntry.AbstractParserTestResultLogEntry;
const AbstractParserTestResultLogLine = __importStar(require("./abstract-parser-test-result-log-line"));
exports.AbstractParserTestResultLogLine = AbstractParserTestResultLogLine.AbstractParserTestResultLogLine;
const Action = __importStar(require("./action"));
exports.Action = Action.Action;
const ActionTypes = __importStar(require("./action-types"));
exports.ActionTypes = ActionTypes.ActionTypes;
const AddEntityAssociationDetails = __importStar(require("./add-entity-association-details"));
exports.AddEntityAssociationDetails = AddEntityAssociationDetails.AddEntityAssociationDetails;
const ArchivingConfiguration = __importStar(require("./archiving-configuration"));
exports.ArchivingConfiguration = ArchivingConfiguration.ArchivingConfiguration;
const Argument = __importStar(require("./argument"));
exports.Argument = Argument.Argument;
const AssignEncryptionKeyDetails = __importStar(require("./assign-encryption-key-details"));
exports.AssignEncryptionKeyDetails = AssignEncryptionKeyDetails.AssignEncryptionKeyDetails;
const AssociableEntity = __importStar(require("./associable-entity"));
exports.AssociableEntity = AssociableEntity.AssociableEntity;
const AssociableEntityCollection = __importStar(require("./associable-entity-collection"));
exports.AssociableEntityCollection = AssociableEntityCollection.AssociableEntityCollection;
const AssociationProperty = __importStar(require("./association-property"));
exports.AssociationProperty = AssociationProperty.AssociationProperty;
const AssociationSummaryReport = __importStar(require("./association-summary-report"));
exports.AssociationSummaryReport = AssociationSummaryReport.AssociationSummaryReport;
const AutoAssociationCollection = __importStar(require("./auto-association-collection"));
exports.AutoAssociationCollection = AutoAssociationCollection.AutoAssociationCollection;
const AutoAssociationState = __importStar(require("./auto-association-state"));
exports.AutoAssociationState = AutoAssociationState.AutoAssociationState;
const AutoLookups = __importStar(require("./auto-lookups"));
exports.AutoLookups = AutoLookups.AutoLookups;
const BucketRange = __importStar(require("./bucket-range"));
exports.BucketRange = BucketRange.BucketRange;
const ChangeIngestTimeRuleCompartmentDetails = __importStar(require("./change-ingest-time-rule-compartment-details"));
exports.ChangeIngestTimeRuleCompartmentDetails = ChangeIngestTimeRuleCompartmentDetails.ChangeIngestTimeRuleCompartmentDetails;
const ChangeLogAnalyticsEmBridgeCompartmentDetails = __importStar(require("./change-log-analytics-em-bridge-compartment-details"));
exports.ChangeLogAnalyticsEmBridgeCompartmentDetails = ChangeLogAnalyticsEmBridgeCompartmentDetails.ChangeLogAnalyticsEmBridgeCompartmentDetails;
const ChangeLogAnalyticsEntityCompartmentDetails = __importStar(require("./change-log-analytics-entity-compartment-details"));
exports.ChangeLogAnalyticsEntityCompartmentDetails = ChangeLogAnalyticsEntityCompartmentDetails.ChangeLogAnalyticsEntityCompartmentDetails;
const ChangeLogAnalyticsLogGroupCompartmentDetails = __importStar(require("./change-log-analytics-log-group-compartment-details"));
exports.ChangeLogAnalyticsLogGroupCompartmentDetails = ChangeLogAnalyticsLogGroupCompartmentDetails.ChangeLogAnalyticsLogGroupCompartmentDetails;
const ChangeLogAnalyticsObjectCollectionRuleCompartmentDetails = __importStar(require("./change-log-analytics-object-collection-rule-compartment-details"));
exports.ChangeLogAnalyticsObjectCollectionRuleCompartmentDetails = ChangeLogAnalyticsObjectCollectionRuleCompartmentDetails.ChangeLogAnalyticsObjectCollectionRuleCompartmentDetails;
const ChangeLookupCompartmentDetails = __importStar(require("./change-lookup-compartment-details"));
exports.ChangeLookupCompartmentDetails = ChangeLookupCompartmentDetails.ChangeLookupCompartmentDetails;
const ChangeScheduledTaskCompartmentDetails = __importStar(require("./change-scheduled-task-compartment-details"));
exports.ChangeScheduledTaskCompartmentDetails = ChangeScheduledTaskCompartmentDetails.ChangeScheduledTaskCompartmentDetails;
const CharEncodingCollection = __importStar(require("./char-encoding-collection"));
exports.CharEncodingCollection = CharEncodingCollection.CharEncodingCollection;
const ColumnName = __importStar(require("./column-name"));
exports.ColumnName = ColumnName.ColumnName;
const ColumnNameCollection = __importStar(require("./column-name-collection"));
exports.ColumnNameCollection = ColumnNameCollection.ColumnNameCollection;
const CompareContentDetails = __importStar(require("./compare-content-details"));
exports.CompareContentDetails = CompareContentDetails.CompareContentDetails;
const CompareContentResult = __importStar(require("./compare-content-result"));
exports.CompareContentResult = CompareContentResult.CompareContentResult;
const CompareLineResult = __importStar(require("./compare-line-result"));
exports.CompareLineResult = CompareLineResult.CompareLineResult;
const ConditionBlock = __importStar(require("./condition-block"));
exports.ConditionBlock = ConditionBlock.ConditionBlock;
const ConfigLifecycleState = __importStar(require("./config-lifecycle-state"));
exports.ConfigLifecycleState = ConfigLifecycleState.ConfigLifecycleState;
const CreateIngestTimeRuleDetails = __importStar(require("./create-ingest-time-rule-details"));
exports.CreateIngestTimeRuleDetails = CreateIngestTimeRuleDetails.CreateIngestTimeRuleDetails;
const CreateLogAnalyticsEmBridgeDetails = __importStar(require("./create-log-analytics-em-bridge-details"));
exports.CreateLogAnalyticsEmBridgeDetails = CreateLogAnalyticsEmBridgeDetails.CreateLogAnalyticsEmBridgeDetails;
const CreateLogAnalyticsEntityDetails = __importStar(require("./create-log-analytics-entity-details"));
exports.CreateLogAnalyticsEntityDetails = CreateLogAnalyticsEntityDetails.CreateLogAnalyticsEntityDetails;
const CreateLogAnalyticsEntityTypeDetails = __importStar(require("./create-log-analytics-entity-type-details"));
exports.CreateLogAnalyticsEntityTypeDetails = CreateLogAnalyticsEntityTypeDetails.CreateLogAnalyticsEntityTypeDetails;
const CreateLogAnalyticsLogGroupDetails = __importStar(require("./create-log-analytics-log-group-details"));
exports.CreateLogAnalyticsLogGroupDetails = CreateLogAnalyticsLogGroupDetails.CreateLogAnalyticsLogGroupDetails;
const CreateLogAnalyticsObjectCollectionRuleDetails = __importStar(require("./create-log-analytics-object-collection-rule-details"));
exports.CreateLogAnalyticsObjectCollectionRuleDetails = CreateLogAnalyticsObjectCollectionRuleDetails.CreateLogAnalyticsObjectCollectionRuleDetails;
const CreateScheduledTaskDetails = __importStar(require("./create-scheduled-task-details"));
exports.CreateScheduledTaskDetails = CreateScheduledTaskDetails.CreateScheduledTaskDetails;
const CreationSource = __importStar(require("./creation-source"));
exports.CreationSource = CreationSource.CreationSource;
const CreationSourceType = __importStar(require("./creation-source-type"));
exports.CreationSourceType = CreationSourceType.CreationSourceType;
const CredentialEndpoint = __importStar(require("./credential-endpoint"));
exports.CredentialEndpoint = CredentialEndpoint.CredentialEndpoint;
const DeleteLogAnalyticsAssociation = __importStar(require("./delete-log-analytics-association"));
exports.DeleteLogAnalyticsAssociation = DeleteLogAnalyticsAssociation.DeleteLogAnalyticsAssociation;
const DeleteLogAnalyticsAssociationDetails = __importStar(require("./delete-log-analytics-association-details"));
exports.DeleteLogAnalyticsAssociationDetails = DeleteLogAnalyticsAssociationDetails.DeleteLogAnalyticsAssociationDetails;
const Dependency = __importStar(require("./dependency"));
exports.Dependency = Dependency.Dependency;
const DependentParser = __importStar(require("./dependent-parser"));
exports.DependentParser = DependentParser.DependentParser;
const DependentSource = __importStar(require("./dependent-source"));
exports.DependentSource = DependentSource.DependentSource;
const DisableAutoAssociationDetail = __importStar(require("./disable-auto-association-detail"));
exports.DisableAutoAssociationDetail = DisableAutoAssociationDetail.DisableAutoAssociationDetail;
const DisableAutoAssociationDetails = __importStar(require("./disable-auto-association-details"));
exports.DisableAutoAssociationDetails = DisableAutoAssociationDetails.DisableAutoAssociationDetails;
const EfdRegexResult = __importStar(require("./efd-regex-result"));
exports.EfdRegexResult = EfdRegexResult.EfdRegexResult;
const EffectivePropertyCollection = __importStar(require("./effective-property-collection"));
exports.EffectivePropertyCollection = EffectivePropertyCollection.EffectivePropertyCollection;
const EffectivePropertySummary = __importStar(require("./effective-property-summary"));
exports.EffectivePropertySummary = EffectivePropertySummary.EffectivePropertySummary;
const EmBridgeLatestImportProcessingStatus = __importStar(require("./em-bridge-latest-import-processing-status"));
exports.EmBridgeLatestImportProcessingStatus = EmBridgeLatestImportProcessingStatus.EmBridgeLatestImportProcessingStatus;
const EmBridgeLifecycleStates = __importStar(require("./em-bridge-lifecycle-states"));
exports.EmBridgeLifecycleStates = EmBridgeLifecycleStates.EmBridgeLifecycleStates;
const EnableAutoAssociationDetail = __importStar(require("./enable-auto-association-detail"));
exports.EnableAutoAssociationDetail = EnableAutoAssociationDetail.EnableAutoAssociationDetail;
const EnableAutoAssociationDetails = __importStar(require("./enable-auto-association-details"));
exports.EnableAutoAssociationDetails = EnableAutoAssociationDetails.EnableAutoAssociationDetails;
const EncryptionKeyInfoCollection = __importStar(require("./encryption-key-info-collection"));
exports.EncryptionKeyInfoCollection = EncryptionKeyInfoCollection.EncryptionKeyInfoCollection;
const EncryptionKeyInfoSummary = __importStar(require("./encryption-key-info-summary"));
exports.EncryptionKeyInfoSummary = EncryptionKeyInfoSummary.EncryptionKeyInfoSummary;
const EncryptionKeySource = __importStar(require("./encryption-key-source"));
exports.EncryptionKeySource = EncryptionKeySource.EncryptionKeySource;
const EncryptionKeyType = __importStar(require("./encryption-key-type"));
exports.EncryptionKeyType = EncryptionKeyType.EncryptionKeyType;
const EndpointCredentials = __importStar(require("./endpoint-credentials"));
exports.EndpointCredentials = EndpointCredentials.EndpointCredentials;
const EndpointProxy = __importStar(require("./endpoint-proxy"));
exports.EndpointProxy = EndpointProxy.EndpointProxy;
const EndpointRequest = __importStar(require("./endpoint-request"));
exports.EndpointRequest = EndpointRequest.EndpointRequest;
const EndpointResponse = __importStar(require("./endpoint-response"));
exports.EndpointResponse = EndpointResponse.EndpointResponse;
const EndpointResult = __importStar(require("./endpoint-result"));
exports.EndpointResult = EndpointResult.EndpointResult;
const EntityCloudType = __importStar(require("./entity-cloud-type"));
exports.EntityCloudType = EntityCloudType.EntityCloudType;
const EntityLifecycleStates = __importStar(require("./entity-lifecycle-states"));
exports.EntityLifecycleStates = EntityLifecycleStates.EntityLifecycleStates;
const EntityTypeProperty = __importStar(require("./entity-type-property"));
exports.EntityTypeProperty = EntityTypeProperty.EntityTypeProperty;
const ErrorDetails = __importStar(require("./error-details"));
exports.ErrorDetails = ErrorDetails.ErrorDetails;
const EstimatePurgeDataSizeDetails = __importStar(require("./estimate-purge-data-size-details"));
exports.EstimatePurgeDataSizeDetails = EstimatePurgeDataSizeDetails.EstimatePurgeDataSizeDetails;
const EstimatePurgeDataSizeResult = __importStar(require("./estimate-purge-data-size-result"));
exports.EstimatePurgeDataSizeResult = EstimatePurgeDataSizeResult.EstimatePurgeDataSizeResult;
const EstimateRecallDataSizeDetails = __importStar(require("./estimate-recall-data-size-details"));
exports.EstimateRecallDataSizeDetails = EstimateRecallDataSizeDetails.EstimateRecallDataSizeDetails;
const EstimateRecallDataSizeResult = __importStar(require("./estimate-recall-data-size-result"));
exports.EstimateRecallDataSizeResult = EstimateRecallDataSizeResult.EstimateRecallDataSizeResult;
const EstimateReleaseDataSizeDetails = __importStar(require("./estimate-release-data-size-details"));
exports.EstimateReleaseDataSizeDetails = EstimateReleaseDataSizeDetails.EstimateReleaseDataSizeDetails;
const EstimateReleaseDataSizeResult = __importStar(require("./estimate-release-data-size-result"));
exports.EstimateReleaseDataSizeResult = EstimateReleaseDataSizeResult.EstimateReleaseDataSizeResult;
const EventType = __importStar(require("./event-type"));
exports.EventType = EventType.EventType;
const EventTypeCollection = __importStar(require("./event-type-collection"));
exports.EventTypeCollection = EventTypeCollection.EventTypeCollection;
const EventTypeDetails = __importStar(require("./event-type-details"));
exports.EventTypeDetails = EventTypeDetails.EventTypeDetails;
const ExportContent = __importStar(require("./export-content"));
exports.ExportContent = ExportContent.ExportContent;
const ExportDetails = __importStar(require("./export-details"));
exports.ExportDetails = ExportDetails.ExportDetails;
const ExtendedFieldsValidationResult = __importStar(require("./extended-fields-validation-result"));
exports.ExtendedFieldsValidationResult = ExtendedFieldsValidationResult.ExtendedFieldsValidationResult;
const ExtractLogFieldResults = __importStar(require("./extract-log-field-results"));
exports.ExtractLogFieldResults = ExtractLogFieldResults.ExtractLogFieldResults;
const ExtractLogHeaderDetails = __importStar(require("./extract-log-header-details"));
exports.ExtractLogHeaderDetails = ExtractLogHeaderDetails.ExtractLogHeaderDetails;
const ExtractLogHeaderResults = __importStar(require("./extract-log-header-results"));
exports.ExtractLogHeaderResults = ExtractLogHeaderResults.ExtractLogHeaderResults;
const FieldSummaryReport = __importStar(require("./field-summary-report"));
exports.FieldSummaryReport = FieldSummaryReport.FieldSummaryReport;
const FieldValue = __importStar(require("./field-value"));
exports.FieldValue = FieldValue.FieldValue;
const FileValidationResponse = __importStar(require("./file-validation-response"));
exports.FileValidationResponse = FileValidationResponse.FileValidationResponse;
const Filter = __importStar(require("./filter"));
exports.Filter = Filter.Filter;
const FilterDetails = __importStar(require("./filter-details"));
exports.FilterDetails = FilterDetails.FilterDetails;
const FilterOutput = __importStar(require("./filter-output"));
exports.FilterOutput = FilterOutput.FilterOutput;
const GenericConditionBlock = __importStar(require("./generic-condition-block"));
exports.GenericConditionBlock = GenericConditionBlock.GenericConditionBlock;
const Indexes = __importStar(require("./indexes"));
exports.Indexes = Indexes.Indexes;
const IngestTimeRule = __importStar(require("./ingest-time-rule"));
exports.IngestTimeRule = IngestTimeRule.IngestTimeRule;
const IngestTimeRuleAction = __importStar(require("./ingest-time-rule-action"));
exports.IngestTimeRuleAction = IngestTimeRuleAction.IngestTimeRuleAction;
const IngestTimeRuleAdditionalFieldCondition = __importStar(require("./ingest-time-rule-additional-field-condition"));
exports.IngestTimeRuleAdditionalFieldCondition = IngestTimeRuleAdditionalFieldCondition.IngestTimeRuleAdditionalFieldCondition;
const IngestTimeRuleCondition = __importStar(require("./ingest-time-rule-condition"));
exports.IngestTimeRuleCondition = IngestTimeRuleCondition.IngestTimeRuleCondition;
const IngestTimeRuleResource = __importStar(require("./ingest-time-rule-resource"));
exports.IngestTimeRuleResource = IngestTimeRuleResource.IngestTimeRuleResource;
const IngestTimeRuleSummary = __importStar(require("./ingest-time-rule-summary"));
exports.IngestTimeRuleSummary = IngestTimeRuleSummary.IngestTimeRuleSummary;
const IngestTimeRuleSummaryCollection = __importStar(require("./ingest-time-rule-summary-collection"));
exports.IngestTimeRuleSummaryCollection = IngestTimeRuleSummaryCollection.IngestTimeRuleSummaryCollection;
const JobMode = __importStar(require("./job-mode"));
exports.JobMode = JobMode.JobMode;
const JobModeFilter = __importStar(require("./job-mode-filter"));
exports.JobModeFilter = JobModeFilter.JobModeFilter;
const LabelNames = __importStar(require("./label-names"));
exports.LabelNames = LabelNames.LabelNames;
const LabelPriority = __importStar(require("./label-priority"));
exports.LabelPriority = LabelPriority.LabelPriority;
const LabelPriorityCollection = __importStar(require("./label-priority-collection"));
exports.LabelPriorityCollection = LabelPriorityCollection.LabelPriorityCollection;
const LabelSourceCollection = __importStar(require("./label-source-collection"));
exports.LabelSourceCollection = LabelSourceCollection.LabelSourceCollection;
const LabelSourceSummary = __importStar(require("./label-source-summary"));
exports.LabelSourceSummary = LabelSourceSummary.LabelSourceSummary;
const LabelSummaryReport = __importStar(require("./label-summary-report"));
exports.LabelSummaryReport = LabelSummaryReport.LabelSummaryReport;
const LastExecutionStatus = __importStar(require("./last-execution-status"));
exports.LastExecutionStatus = LastExecutionStatus.LastExecutionStatus;
const Level = __importStar(require("./level"));
exports.Level = Level.Level;
const LifecycleStates = __importStar(require("./lifecycle-states"));
exports.LifecycleStates = LifecycleStates.LifecycleStates;
const LogAnalyticsAssociatedEntity = __importStar(require("./log-analytics-associated-entity"));
exports.LogAnalyticsAssociatedEntity = LogAnalyticsAssociatedEntity.LogAnalyticsAssociatedEntity;
const LogAnalyticsAssociatedEntityCollection = __importStar(require("./log-analytics-associated-entity-collection"));
exports.LogAnalyticsAssociatedEntityCollection = LogAnalyticsAssociatedEntityCollection.LogAnalyticsAssociatedEntityCollection;
const LogAnalyticsAssociation = __importStar(require("./log-analytics-association"));
exports.LogAnalyticsAssociation = LogAnalyticsAssociation.LogAnalyticsAssociation;
const LogAnalyticsAssociationCollection = __importStar(require("./log-analytics-association-collection"));
exports.LogAnalyticsAssociationCollection = LogAnalyticsAssociationCollection.LogAnalyticsAssociationCollection;
const LogAnalyticsAssociationParameter = __importStar(require("./log-analytics-association-parameter"));
exports.LogAnalyticsAssociationParameter = LogAnalyticsAssociationParameter.LogAnalyticsAssociationParameter;
const LogAnalyticsAssociationParameterCollection = __importStar(require("./log-analytics-association-parameter-collection"));
exports.LogAnalyticsAssociationParameterCollection = LogAnalyticsAssociationParameterCollection.LogAnalyticsAssociationParameterCollection;
const LogAnalyticsCategory = __importStar(require("./log-analytics-category"));
exports.LogAnalyticsCategory = LogAnalyticsCategory.LogAnalyticsCategory;
const LogAnalyticsCategoryCollection = __importStar(require("./log-analytics-category-collection"));
exports.LogAnalyticsCategoryCollection = LogAnalyticsCategoryCollection.LogAnalyticsCategoryCollection;
const LogAnalyticsConfigWorkRequest = __importStar(require("./log-analytics-config-work-request"));
exports.LogAnalyticsConfigWorkRequest = LogAnalyticsConfigWorkRequest.LogAnalyticsConfigWorkRequest;
const LogAnalyticsConfigWorkRequestCollection = __importStar(require("./log-analytics-config-work-request-collection"));
exports.LogAnalyticsConfigWorkRequestCollection = LogAnalyticsConfigWorkRequestCollection.LogAnalyticsConfigWorkRequestCollection;
const LogAnalyticsConfigWorkRequestPayload = __importStar(require("./log-analytics-config-work-request-payload"));
exports.LogAnalyticsConfigWorkRequestPayload = LogAnalyticsConfigWorkRequestPayload.LogAnalyticsConfigWorkRequestPayload;
const LogAnalyticsConfigWorkRequestSummary = __importStar(require("./log-analytics-config-work-request-summary"));
exports.LogAnalyticsConfigWorkRequestSummary = LogAnalyticsConfigWorkRequestSummary.LogAnalyticsConfigWorkRequestSummary;
const LogAnalyticsEmBridge = __importStar(require("./log-analytics-em-bridge"));
exports.LogAnalyticsEmBridge = LogAnalyticsEmBridge.LogAnalyticsEmBridge;
const LogAnalyticsEmBridgeCollection = __importStar(require("./log-analytics-em-bridge-collection"));
exports.LogAnalyticsEmBridgeCollection = LogAnalyticsEmBridgeCollection.LogAnalyticsEmBridgeCollection;
const LogAnalyticsEmBridgeSummary = __importStar(require("./log-analytics-em-bridge-summary"));
exports.LogAnalyticsEmBridgeSummary = LogAnalyticsEmBridgeSummary.LogAnalyticsEmBridgeSummary;
const LogAnalyticsEmBridgeSummaryReport = __importStar(require("./log-analytics-em-bridge-summary-report"));
exports.LogAnalyticsEmBridgeSummaryReport = LogAnalyticsEmBridgeSummaryReport.LogAnalyticsEmBridgeSummaryReport;
const LogAnalyticsEndpoint = __importStar(require("./log-analytics-endpoint"));
exports.LogAnalyticsEndpoint = LogAnalyticsEndpoint.LogAnalyticsEndpoint;
const LogAnalyticsEntity = __importStar(require("./log-analytics-entity"));
exports.LogAnalyticsEntity = LogAnalyticsEntity.LogAnalyticsEntity;
const LogAnalyticsEntityCollection = __importStar(require("./log-analytics-entity-collection"));
exports.LogAnalyticsEntityCollection = LogAnalyticsEntityCollection.LogAnalyticsEntityCollection;
const LogAnalyticsEntitySummary = __importStar(require("./log-analytics-entity-summary"));
exports.LogAnalyticsEntitySummary = LogAnalyticsEntitySummary.LogAnalyticsEntitySummary;
const LogAnalyticsEntitySummaryReport = __importStar(require("./log-analytics-entity-summary-report"));
exports.LogAnalyticsEntitySummaryReport = LogAnalyticsEntitySummaryReport.LogAnalyticsEntitySummaryReport;
const LogAnalyticsEntityTopologyCollection = __importStar(require("./log-analytics-entity-topology-collection"));
exports.LogAnalyticsEntityTopologyCollection = LogAnalyticsEntityTopologyCollection.LogAnalyticsEntityTopologyCollection;
const LogAnalyticsEntityTopologyLink = __importStar(require("./log-analytics-entity-topology-link"));
exports.LogAnalyticsEntityTopologyLink = LogAnalyticsEntityTopologyLink.LogAnalyticsEntityTopologyLink;
const LogAnalyticsEntityTopologyLinkCollection = __importStar(require("./log-analytics-entity-topology-link-collection"));
exports.LogAnalyticsEntityTopologyLinkCollection = LogAnalyticsEntityTopologyLinkCollection.LogAnalyticsEntityTopologyLinkCollection;
const LogAnalyticsEntityTopologySummary = __importStar(require("./log-analytics-entity-topology-summary"));
exports.LogAnalyticsEntityTopologySummary = LogAnalyticsEntityTopologySummary.LogAnalyticsEntityTopologySummary;
const LogAnalyticsEntityType = __importStar(require("./log-analytics-entity-type"));
exports.LogAnalyticsEntityType = LogAnalyticsEntityType.LogAnalyticsEntityType;
const LogAnalyticsEntityTypeCollection = __importStar(require("./log-analytics-entity-type-collection"));
exports.LogAnalyticsEntityTypeCollection = LogAnalyticsEntityTypeCollection.LogAnalyticsEntityTypeCollection;
const LogAnalyticsEntityTypeSummary = __importStar(require("./log-analytics-entity-type-summary"));
exports.LogAnalyticsEntityTypeSummary = LogAnalyticsEntityTypeSummary.LogAnalyticsEntityTypeSummary;
const LogAnalyticsExtendedField = __importStar(require("./log-analytics-extended-field"));
exports.LogAnalyticsExtendedField = LogAnalyticsExtendedField.LogAnalyticsExtendedField;
const LogAnalyticsField = __importStar(require("./log-analytics-field"));
exports.LogAnalyticsField = LogAnalyticsField.LogAnalyticsField;
const LogAnalyticsFieldCollection = __importStar(require("./log-analytics-field-collection"));
exports.LogAnalyticsFieldCollection = LogAnalyticsFieldCollection.LogAnalyticsFieldCollection;
const LogAnalyticsFieldSummary = __importStar(require("./log-analytics-field-summary"));
exports.LogAnalyticsFieldSummary = LogAnalyticsFieldSummary.LogAnalyticsFieldSummary;
const LogAnalyticsFieldUsages = __importStar(require("./log-analytics-field-usages"));
exports.LogAnalyticsFieldUsages = LogAnalyticsFieldUsages.LogAnalyticsFieldUsages;
const LogAnalyticsImportCustomChangeList = __importStar(require("./log-analytics-import-custom-change-list"));
exports.LogAnalyticsImportCustomChangeList = LogAnalyticsImportCustomChangeList.LogAnalyticsImportCustomChangeList;
const LogAnalyticsImportCustomContent = __importStar(require("./log-analytics-import-custom-content"));
exports.LogAnalyticsImportCustomContent = LogAnalyticsImportCustomContent.LogAnalyticsImportCustomContent;
const LogAnalyticsLabel = __importStar(require("./log-analytics-label"));
exports.LogAnalyticsLabel = LogAnalyticsLabel.LogAnalyticsLabel;
const LogAnalyticsLabelAlias = __importStar(require("./log-analytics-label-alias"));
exports.LogAnalyticsLabelAlias = LogAnalyticsLabelAlias.LogAnalyticsLabelAlias;
const LogAnalyticsLabelCollection = __importStar(require("./log-analytics-label-collection"));
exports.LogAnalyticsLabelCollection = LogAnalyticsLabelCollection.LogAnalyticsLabelCollection;
const LogAnalyticsLabelDefinition = __importStar(require("./log-analytics-label-definition"));
exports.LogAnalyticsLabelDefinition = LogAnalyticsLabelDefinition.LogAnalyticsLabelDefinition;
const LogAnalyticsLabelOperator = __importStar(require("./log-analytics-label-operator"));
exports.LogAnalyticsLabelOperator = LogAnalyticsLabelOperator.LogAnalyticsLabelOperator;
const LogAnalyticsLabelOperatorCollection = __importStar(require("./log-analytics-label-operator-collection"));
exports.LogAnalyticsLabelOperatorCollection = LogAnalyticsLabelOperatorCollection.LogAnalyticsLabelOperatorCollection;
const LogAnalyticsLabelSummary = __importStar(require("./log-analytics-label-summary"));
exports.LogAnalyticsLabelSummary = LogAnalyticsLabelSummary.LogAnalyticsLabelSummary;
const LogAnalyticsLabelView = __importStar(require("./log-analytics-label-view"));
exports.LogAnalyticsLabelView = LogAnalyticsLabelView.LogAnalyticsLabelView;
const LogAnalyticsLogGroup = __importStar(require("./log-analytics-log-group"));
exports.LogAnalyticsLogGroup = LogAnalyticsLogGroup.LogAnalyticsLogGroup;
const LogAnalyticsLogGroupSummary = __importStar(require("./log-analytics-log-group-summary"));
exports.LogAnalyticsLogGroupSummary = LogAnalyticsLogGroupSummary.LogAnalyticsLogGroupSummary;
const LogAnalyticsLogGroupSummaryCollection = __importStar(require("./log-analytics-log-group-summary-collection"));
exports.LogAnalyticsLogGroupSummaryCollection = LogAnalyticsLogGroupSummaryCollection.LogAnalyticsLogGroupSummaryCollection;
const LogAnalyticsLookup = __importStar(require("./log-analytics-lookup"));
exports.LogAnalyticsLookup = LogAnalyticsLookup.LogAnalyticsLookup;
const LogAnalyticsLookupCollection = __importStar(require("./log-analytics-lookup-collection"));
exports.LogAnalyticsLookupCollection = LogAnalyticsLookupCollection.LogAnalyticsLookupCollection;
const LogAnalyticsLookupFields = __importStar(require("./log-analytics-lookup-fields"));
exports.LogAnalyticsLookupFields = LogAnalyticsLookupFields.LogAnalyticsLookupFields;
const LogAnalyticsMetaFunction = __importStar(require("./log-analytics-meta-function"));
exports.LogAnalyticsMetaFunction = LogAnalyticsMetaFunction.LogAnalyticsMetaFunction;
const LogAnalyticsMetaFunctionArgument = __importStar(require("./log-analytics-meta-function-argument"));
exports.LogAnalyticsMetaFunctionArgument = LogAnalyticsMetaFunctionArgument.LogAnalyticsMetaFunctionArgument;
const LogAnalyticsMetaFunctionCollection = __importStar(require("./log-analytics-meta-function-collection"));
exports.LogAnalyticsMetaFunctionCollection = LogAnalyticsMetaFunctionCollection.LogAnalyticsMetaFunctionCollection;
const LogAnalyticsMetaSourceType = __importStar(require("./log-analytics-meta-source-type"));
exports.LogAnalyticsMetaSourceType = LogAnalyticsMetaSourceType.LogAnalyticsMetaSourceType;
const LogAnalyticsMetaSourceTypeCollection = __importStar(require("./log-analytics-meta-source-type-collection"));
exports.LogAnalyticsMetaSourceTypeCollection = LogAnalyticsMetaSourceTypeCollection.LogAnalyticsMetaSourceTypeCollection;
const LogAnalyticsMetadata = __importStar(require("./log-analytics-metadata"));
exports.LogAnalyticsMetadata = LogAnalyticsMetadata.LogAnalyticsMetadata;
const LogAnalyticsMetadataCollection = __importStar(require("./log-analytics-metadata-collection"));
exports.LogAnalyticsMetadataCollection = LogAnalyticsMetadataCollection.LogAnalyticsMetadataCollection;
const LogAnalyticsMetadataDetails = __importStar(require("./log-analytics-metadata-details"));
exports.LogAnalyticsMetadataDetails = LogAnalyticsMetadataDetails.LogAnalyticsMetadataDetails;
const LogAnalyticsMetadataSummary = __importStar(require("./log-analytics-metadata-summary"));
exports.LogAnalyticsMetadataSummary = LogAnalyticsMetadataSummary.LogAnalyticsMetadataSummary;
const LogAnalyticsMetric = __importStar(require("./log-analytics-metric"));
exports.LogAnalyticsMetric = LogAnalyticsMetric.LogAnalyticsMetric;
const LogAnalyticsObjectCollectionRule = __importStar(require("./log-analytics-object-collection-rule"));
exports.LogAnalyticsObjectCollectionRule = LogAnalyticsObjectCollectionRule.LogAnalyticsObjectCollectionRule;
const LogAnalyticsObjectCollectionRuleCollection = __importStar(require("./log-analytics-object-collection-rule-collection"));
exports.LogAnalyticsObjectCollectionRuleCollection = LogAnalyticsObjectCollectionRuleCollection.LogAnalyticsObjectCollectionRuleCollection;
const LogAnalyticsObjectCollectionRuleSummary = __importStar(require("./log-analytics-object-collection-rule-summary"));
exports.LogAnalyticsObjectCollectionRuleSummary = LogAnalyticsObjectCollectionRuleSummary.LogAnalyticsObjectCollectionRuleSummary;
const LogAnalyticsOperationTypes = __importStar(require("./log-analytics-operation-types"));
exports.LogAnalyticsOperationTypes = LogAnalyticsOperationTypes.LogAnalyticsOperationTypes;
const LogAnalyticsParameter = __importStar(require("./log-analytics-parameter"));
exports.LogAnalyticsParameter = LogAnalyticsParameter.LogAnalyticsParameter;
const LogAnalyticsParser = __importStar(require("./log-analytics-parser"));
exports.LogAnalyticsParser = LogAnalyticsParser.LogAnalyticsParser;
const LogAnalyticsParserCollection = __importStar(require("./log-analytics-parser-collection"));
exports.LogAnalyticsParserCollection = LogAnalyticsParserCollection.LogAnalyticsParserCollection;
const LogAnalyticsParserField = __importStar(require("./log-analytics-parser-field"));
exports.LogAnalyticsParserField = LogAnalyticsParserField.LogAnalyticsParserField;
const LogAnalyticsParserFilter = __importStar(require("./log-analytics-parser-filter"));
exports.LogAnalyticsParserFilter = LogAnalyticsParserFilter.LogAnalyticsParserFilter;
const LogAnalyticsParserFunction = __importStar(require("./log-analytics-parser-function"));
exports.LogAnalyticsParserFunction = LogAnalyticsParserFunction.LogAnalyticsParserFunction;
const LogAnalyticsParserFunctionCollection = __importStar(require("./log-analytics-parser-function-collection"));
exports.LogAnalyticsParserFunctionCollection = LogAnalyticsParserFunctionCollection.LogAnalyticsParserFunctionCollection;
const LogAnalyticsParserFunctionParameter = __importStar(require("./log-analytics-parser-function-parameter"));
exports.LogAnalyticsParserFunctionParameter = LogAnalyticsParserFunctionParameter.LogAnalyticsParserFunctionParameter;
const LogAnalyticsParserMetaPlugin = __importStar(require("./log-analytics-parser-meta-plugin"));
exports.LogAnalyticsParserMetaPlugin = LogAnalyticsParserMetaPlugin.LogAnalyticsParserMetaPlugin;
const LogAnalyticsParserMetaPluginCollection = __importStar(require("./log-analytics-parser-meta-plugin-collection"));
exports.LogAnalyticsParserMetaPluginCollection = LogAnalyticsParserMetaPluginCollection.LogAnalyticsParserMetaPluginCollection;
const LogAnalyticsParserMetaPluginParameter = __importStar(require("./log-analytics-parser-meta-plugin-parameter"));
exports.LogAnalyticsParserMetaPluginParameter = LogAnalyticsParserMetaPluginParameter.LogAnalyticsParserMetaPluginParameter;
const LogAnalyticsParserSummary = __importStar(require("./log-analytics-parser-summary"));
exports.LogAnalyticsParserSummary = LogAnalyticsParserSummary.LogAnalyticsParserSummary;
const LogAnalyticsPatternFilter = __importStar(require("./log-analytics-pattern-filter"));
exports.LogAnalyticsPatternFilter = LogAnalyticsPatternFilter.LogAnalyticsPatternFilter;
const LogAnalyticsPreference = __importStar(require("./log-analytics-preference"));
exports.LogAnalyticsPreference = LogAnalyticsPreference.LogAnalyticsPreference;
const LogAnalyticsPreferenceCollection = __importStar(require("./log-analytics-preference-collection"));
exports.LogAnalyticsPreferenceCollection = LogAnalyticsPreferenceCollection.LogAnalyticsPreferenceCollection;
const LogAnalyticsPreferenceDetails = __importStar(require("./log-analytics-preference-details"));
exports.LogAnalyticsPreferenceDetails = LogAnalyticsPreferenceDetails.LogAnalyticsPreferenceDetails;
const LogAnalyticsProperty = __importStar(require("./log-analytics-property"));
exports.LogAnalyticsProperty = LogAnalyticsProperty.LogAnalyticsProperty;
const LogAnalyticsResourceCategory = __importStar(require("./log-analytics-resource-category"));
exports.LogAnalyticsResourceCategory = LogAnalyticsResourceCategory.LogAnalyticsResourceCategory;
const LogAnalyticsResourceCategoryCollection = __importStar(require("./log-analytics-resource-category-collection"));
exports.LogAnalyticsResourceCategoryCollection = LogAnalyticsResourceCategoryCollection.LogAnalyticsResourceCategoryCollection;
const LogAnalyticsResourceCategoryDetails = __importStar(require("./log-analytics-resource-category-details"));
exports.LogAnalyticsResourceCategoryDetails = LogAnalyticsResourceCategoryDetails.LogAnalyticsResourceCategoryDetails;
const LogAnalyticsSource = __importStar(require("./log-analytics-source"));
exports.LogAnalyticsSource = LogAnalyticsSource.LogAnalyticsSource;
const LogAnalyticsSourceCollection = __importStar(require("./log-analytics-source-collection"));
exports.LogAnalyticsSourceCollection = LogAnalyticsSourceCollection.LogAnalyticsSourceCollection;
const LogAnalyticsSourceDataFilter = __importStar(require("./log-analytics-source-data-filter"));
exports.LogAnalyticsSourceDataFilter = LogAnalyticsSourceDataFilter.LogAnalyticsSourceDataFilter;
const LogAnalyticsSourceEntityType = __importStar(require("./log-analytics-source-entity-type"));
exports.LogAnalyticsSourceEntityType = LogAnalyticsSourceEntityType.LogAnalyticsSourceEntityType;
const LogAnalyticsSourceExtendedFieldDefinition = __importStar(require("./log-analytics-source-extended-field-definition"));
exports.LogAnalyticsSourceExtendedFieldDefinition = LogAnalyticsSourceExtendedFieldDefinition.LogAnalyticsSourceExtendedFieldDefinition;
const LogAnalyticsSourceExtendedFieldDefinitionCollection = __importStar(require("./log-analytics-source-extended-field-definition-collection"));
exports.LogAnalyticsSourceExtendedFieldDefinitionCollection = LogAnalyticsSourceExtendedFieldDefinitionCollection.LogAnalyticsSourceExtendedFieldDefinitionCollection;
const LogAnalyticsSourceFunction = __importStar(require("./log-analytics-source-function"));
exports.LogAnalyticsSourceFunction = LogAnalyticsSourceFunction.LogAnalyticsSourceFunction;
const LogAnalyticsSourceLabelCondition = __importStar(require("./log-analytics-source-label-condition"));
exports.LogAnalyticsSourceLabelCondition = LogAnalyticsSourceLabelCondition.LogAnalyticsSourceLabelCondition;
const LogAnalyticsSourceMetadataField = __importStar(require("./log-analytics-source-metadata-field"));
exports.LogAnalyticsSourceMetadataField = LogAnalyticsSourceMetadataField.LogAnalyticsSourceMetadataField;
const LogAnalyticsSourceMetric = __importStar(require("./log-analytics-source-metric"));
exports.LogAnalyticsSourceMetric = LogAnalyticsSourceMetric.LogAnalyticsSourceMetric;
const LogAnalyticsSourcePattern = __importStar(require("./log-analytics-source-pattern"));
exports.LogAnalyticsSourcePattern = LogAnalyticsSourcePattern.LogAnalyticsSourcePattern;
const LogAnalyticsSourcePatternCollection = __importStar(require("./log-analytics-source-pattern-collection"));
exports.LogAnalyticsSourcePatternCollection = LogAnalyticsSourcePatternCollection.LogAnalyticsSourcePatternCollection;
const LogAnalyticsSourceSummary = __importStar(require("./log-analytics-source-summary"));
exports.LogAnalyticsSourceSummary = LogAnalyticsSourceSummary.LogAnalyticsSourceSummary;
const LogAnalyticsTemplate = __importStar(require("./log-analytics-template"));
exports.LogAnalyticsTemplate = LogAnalyticsTemplate.LogAnalyticsTemplate;
const LogAnalyticsTemplateCollection = __importStar(require("./log-analytics-template-collection"));
exports.LogAnalyticsTemplateCollection = LogAnalyticsTemplateCollection.LogAnalyticsTemplateCollection;
const LogAnalyticsTemplateSummary = __importStar(require("./log-analytics-template-summary"));
exports.LogAnalyticsTemplateSummary = LogAnalyticsTemplateSummary.LogAnalyticsTemplateSummary;
const LogAnalyticsWarning = __importStar(require("./log-analytics-warning"));
exports.LogAnalyticsWarning = LogAnalyticsWarning.LogAnalyticsWarning;
const LogAnalyticsWarningCollection = __importStar(require("./log-analytics-warning-collection"));
exports.LogAnalyticsWarningCollection = LogAnalyticsWarningCollection.LogAnalyticsWarningCollection;
const LogEndpoint = __importStar(require("./log-endpoint"));
exports.LogEndpoint = LogEndpoint.LogEndpoint;
const LogGroupSummaryReport = __importStar(require("./log-group-summary-report"));
exports.LogGroupSummaryReport = LogGroupSummaryReport.LogGroupSummaryReport;
const LogListEndpoint = __importStar(require("./log-list-endpoint"));
exports.LogListEndpoint = LogListEndpoint.LogListEndpoint;
const LogSetCollection = __importStar(require("./log-set-collection"));
exports.LogSetCollection = LogSetCollection.LogSetCollection;
const LogSetKeyTypes = __importStar(require("./log-set-key-types"));
exports.LogSetKeyTypes = LogSetKeyTypes.LogSetKeyTypes;
const LogSetsCount = __importStar(require("./log-sets-count"));
exports.LogSetsCount = LogSetsCount.LogSetsCount;
const LogTypes = __importStar(require("./log-types"));
exports.LogTypes = LogTypes.LogTypes;
const LookupField = __importStar(require("./lookup-field"));
exports.LookupField = LookupField.LookupField;
const LookupSummaryReport = __importStar(require("./lookup-summary-report"));
exports.LookupSummaryReport = LookupSummaryReport.LookupSummaryReport;
const MatchInfo = __importStar(require("./match-info"));
exports.MatchInfo = MatchInfo.MatchInfo;
const MetricExtraction = __importStar(require("./metric-extraction"));
exports.MetricExtraction = MetricExtraction.MetricExtraction;
const NameValuePair = __importStar(require("./name-value-pair"));
exports.NameValuePair = NameValuePair.NameValuePair;
const Namespace = __importStar(require("./namespace"));
exports.Namespace = Namespace.Namespace;
const NamespaceCollection = __importStar(require("./namespace-collection"));
exports.NamespaceCollection = NamespaceCollection.NamespaceCollection;
const NamespaceSummary = __importStar(require("./namespace-summary"));
exports.NamespaceSummary = NamespaceSummary.NamespaceSummary;
const ObjectCollectionRuleCollectionTypes = __importStar(require("./object-collection-rule-collection-types"));
exports.ObjectCollectionRuleCollectionTypes = ObjectCollectionRuleCollectionTypes.ObjectCollectionRuleCollectionTypes;
const ObjectCollectionRuleLifecycleStates = __importStar(require("./object-collection-rule-lifecycle-states"));
exports.ObjectCollectionRuleLifecycleStates = ObjectCollectionRuleLifecycleStates.ObjectCollectionRuleLifecycleStates;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OverlappingRecallCollection = __importStar(require("./overlapping-recall-collection"));
exports.OverlappingRecallCollection = OverlappingRecallCollection.OverlappingRecallCollection;
const OverlappingRecallSummary = __importStar(require("./overlapping-recall-summary"));
exports.OverlappingRecallSummary = OverlappingRecallSummary.OverlappingRecallSummary;
const ParseQueryDetails = __importStar(require("./parse-query-details"));
exports.ParseQueryDetails = ParseQueryDetails.ParseQueryDetails;
const ParseQueryOutput = __importStar(require("./parse-query-output"));
exports.ParseQueryOutput = ParseQueryOutput.ParseQueryOutput;
const ParsedContent = __importStar(require("./parsed-content"));
exports.ParsedContent = ParsedContent.ParsedContent;
const ParsedField = __importStar(require("./parsed-field"));
exports.ParsedField = ParsedField.ParsedField;
const ParserAction = __importStar(require("./parser-action"));
exports.ParserAction = ParserAction.ParserAction;
const ParserActionSummary = __importStar(require("./parser-action-summary"));
exports.ParserActionSummary = ParserActionSummary.ParserActionSummary;
const ParserActionSummaryCollection = __importStar(require("./parser-action-summary-collection"));
exports.ParserActionSummaryCollection = ParserActionSummaryCollection.ParserActionSummaryCollection;
const ParserSummaryReport = __importStar(require("./parser-summary-report"));
exports.ParserSummaryReport = ParserSummaryReport.ParserSummaryReport;
const ParserTestResult = __importStar(require("./parser-test-result"));
exports.ParserTestResult = ParserTestResult.ParserTestResult;
const PatternOverride = __importStar(require("./pattern-override"));
exports.PatternOverride = PatternOverride.PatternOverride;
const PayloadType = __importStar(require("./payload-type"));
exports.PayloadType = PayloadType.PayloadType;
const PropertyDefinition = __importStar(require("./property-definition"));
exports.PropertyDefinition = PropertyDefinition.PropertyDefinition;
const PropertyMetadataSummary = __importStar(require("./property-metadata-summary"));
exports.PropertyMetadataSummary = PropertyMetadataSummary.PropertyMetadataSummary;
const PropertyMetadataSummaryCollection = __importStar(require("./property-metadata-summary-collection"));
exports.PropertyMetadataSummaryCollection = PropertyMetadataSummaryCollection.PropertyMetadataSummaryCollection;
const PropertyOverride = __importStar(require("./property-override"));
exports.PropertyOverride = PropertyOverride.PropertyOverride;
const PurgeStorageDataDetails = __importStar(require("./purge-storage-data-details"));
exports.PurgeStorageDataDetails = PurgeStorageDataDetails.PurgeStorageDataDetails;
const QueryAggregation = __importStar(require("./query-aggregation"));
exports.QueryAggregation = QueryAggregation.QueryAggregation;
const QueryDetails = __importStar(require("./query-details"));
exports.QueryDetails = QueryDetails.QueryDetails;
const QueryOperationType = __importStar(require("./query-operation-type"));
exports.QueryOperationType = QueryOperationType.QueryOperationType;
const QueryWorkRequest = __importStar(require("./query-work-request"));
exports.QueryWorkRequest = QueryWorkRequest.QueryWorkRequest;
const QueryWorkRequestCollection = __importStar(require("./query-work-request-collection"));
exports.QueryWorkRequestCollection = QueryWorkRequestCollection.QueryWorkRequestCollection;
const QueryWorkRequestSummary = __importStar(require("./query-work-request-summary"));
exports.QueryWorkRequestSummary = QueryWorkRequestSummary.QueryWorkRequestSummary;
const RecallArchivedDataDetails = __importStar(require("./recall-archived-data-details"));
exports.RecallArchivedDataDetails = RecallArchivedDataDetails.RecallArchivedDataDetails;
const RecallCount = __importStar(require("./recall-count"));
exports.RecallCount = RecallCount.RecallCount;
const RecallDefinition = __importStar(require("./recall-definition"));
exports.RecallDefinition = RecallDefinition.RecallDefinition;
const RecallStatus = __importStar(require("./recall-status"));
exports.RecallStatus = RecallStatus.RecallStatus;
const RecalledData = __importStar(require("./recalled-data"));
exports.RecalledData = RecalledData.RecalledData;
const RecalledDataCollection = __importStar(require("./recalled-data-collection"));
exports.RecalledDataCollection = RecalledDataCollection.RecalledDataCollection;
const RecalledDataInfo = __importStar(require("./recalled-data-info"));
exports.RecalledDataInfo = RecalledDataInfo.RecalledDataInfo;
const RecalledDataSize = __importStar(require("./recalled-data-size"));
exports.RecalledDataSize = RecalledDataSize.RecalledDataSize;
const RecalledInfo = __importStar(require("./recalled-info"));
exports.RecalledInfo = RecalledInfo.RecalledInfo;
const RecalledInfoCollection = __importStar(require("./recalled-info-collection"));
exports.RecalledInfoCollection = RecalledInfoCollection.RecalledInfoCollection;
const RegexMatchResult = __importStar(require("./regex-match-result"));
exports.RegexMatchResult = RegexMatchResult.RegexMatchResult;
const ReleaseRecalledDataDetails = __importStar(require("./release-recalled-data-details"));
exports.ReleaseRecalledDataDetails = ReleaseRecalledDataDetails.ReleaseRecalledDataDetails;
const RemoveEntityAssociationsDetails = __importStar(require("./remove-entity-associations-details"));
exports.RemoveEntityAssociationsDetails = RemoveEntityAssociationsDetails.RemoveEntityAssociationsDetails;
const ResultColumn = __importStar(require("./result-column"));
exports.ResultColumn = ResultColumn.ResultColumn;
const Rule = __importStar(require("./rule"));
exports.Rule = Rule.Rule;
const RuleKind = __importStar(require("./rule-kind"));
exports.RuleKind = RuleKind.RuleKind;
const RuleSummary = __importStar(require("./rule-summary"));
exports.RuleSummary = RuleSummary.RuleSummary;
const RuleSummaryCollection = __importStar(require("./rule-summary-collection"));
exports.RuleSummaryCollection = RuleSummaryCollection.RuleSummaryCollection;
const RuleSummaryReport = __importStar(require("./rule-summary-report"));
exports.RuleSummaryReport = RuleSummaryReport.RuleSummaryReport;
const Schedule = __importStar(require("./schedule"));
exports.Schedule = Schedule.Schedule;
const ScheduledTask = __importStar(require("./scheduled-task"));
exports.ScheduledTask = ScheduledTask.ScheduledTask;
const ScheduledTaskCollection = __importStar(require("./scheduled-task-collection"));
exports.ScheduledTaskCollection = ScheduledTaskCollection.ScheduledTaskCollection;
const ScheduledTaskSummary = __importStar(require("./scheduled-task-summary"));
exports.ScheduledTaskSummary = ScheduledTaskSummary.ScheduledTaskSummary;
const SchedulerResource = __importStar(require("./scheduler-resource"));
exports.SchedulerResource = SchedulerResource.SchedulerResource;
const ScopeFilter = __importStar(require("./scope-filter"));
exports.ScopeFilter = ScopeFilter.ScopeFilter;
const SourceMappingResponse = __importStar(require("./source-mapping-response"));
exports.SourceMappingResponse = SourceMappingResponse.SourceMappingResponse;
const SourceSummaryReport = __importStar(require("./source-summary-report"));
exports.SourceSummaryReport = SourceSummaryReport.SourceSummaryReport;
const SourceValidateDetails = __importStar(require("./source-validate-details"));
exports.SourceValidateDetails = SourceValidateDetails.SourceValidateDetails;
const SourceValidateResults = __importStar(require("./source-validate-results"));
exports.SourceValidateResults = SourceValidateResults.SourceValidateResults;
const StatusSummary = __importStar(require("./status-summary"));
exports.StatusSummary = StatusSummary.StatusSummary;
const StepInfo = __importStar(require("./step-info"));
exports.StepInfo = StepInfo.StepInfo;
const Storage = __importStar(require("./storage"));
exports.Storage = Storage.Storage;
const StorageDataType = __importStar(require("./storage-data-type"));
exports.StorageDataType = StorageDataType.StorageDataType;
const StorageOperationType = __importStar(require("./storage-operation-type"));
exports.StorageOperationType = StorageOperationType.StorageOperationType;
const StorageUsage = __importStar(require("./storage-usage"));
exports.StorageUsage = StorageUsage.StorageUsage;
const StorageWorkRequest = __importStar(require("./storage-work-request"));
exports.StorageWorkRequest = StorageWorkRequest.StorageWorkRequest;
const StorageWorkRequestCollection = __importStar(require("./storage-work-request-collection"));
exports.StorageWorkRequestCollection = StorageWorkRequestCollection.StorageWorkRequestCollection;
const StorageWorkRequestSummary = __importStar(require("./storage-work-request-summary"));
exports.StorageWorkRequestSummary = StorageWorkRequestSummary.StorageWorkRequestSummary;
const StreamCursorTypes = __importStar(require("./stream-cursor-types"));
exports.StreamCursorTypes = StreamCursorTypes.StreamCursorTypes;
const SubSystemName = __importStar(require("./sub-system-name"));
exports.SubSystemName = SubSystemName.SubSystemName;
const Success = __importStar(require("./success"));
exports.Success = Success.Success;
const SuggestDetails = __importStar(require("./suggest-details"));
exports.SuggestDetails = SuggestDetails.SuggestDetails;
const SuggestOutput = __importStar(require("./suggest-output"));
exports.SuggestOutput = SuggestOutput.SuggestOutput;
const TaskType = __importStar(require("./task-type"));
exports.TaskType = TaskType.TaskType;
const TemplateDetails = __importStar(require("./template-details"));
exports.TemplateDetails = TemplateDetails.TemplateDetails;
const TemplateFacet = __importStar(require("./template-facet"));
exports.TemplateFacet = TemplateFacet.TemplateFacet;
const TemplateParams = __importStar(require("./template-params"));
exports.TemplateParams = TemplateParams.TemplateParams;
const TestParserPayloadDetails = __importStar(require("./test-parser-payload-details"));
exports.TestParserPayloadDetails = TestParserPayloadDetails.TestParserPayloadDetails;
const TimeRange = __importStar(require("./time-range"));
exports.TimeRange = TimeRange.TimeRange;
const TimeStatsCluster = __importStar(require("./time-stats-cluster"));
exports.TimeStatsCluster = TimeStatsCluster.TimeStatsCluster;
const TimezoneCollection = __importStar(require("./timezone-collection"));
exports.TimezoneCollection = TimezoneCollection.TimezoneCollection;
const UiParserTestMetadata = __importStar(require("./ui-parser-test-metadata"));
exports.UiParserTestMetadata = UiParserTestMetadata.UiParserTestMetadata;
const UnprocessedDataBucket = __importStar(require("./unprocessed-data-bucket"));
exports.UnprocessedDataBucket = UnprocessedDataBucket.UnprocessedDataBucket;
const UpdateLogAnalyticsEmBridgeDetails = __importStar(require("./update-log-analytics-em-bridge-details"));
exports.UpdateLogAnalyticsEmBridgeDetails = UpdateLogAnalyticsEmBridgeDetails.UpdateLogAnalyticsEmBridgeDetails;
const UpdateLogAnalyticsEntityDetails = __importStar(require("./update-log-analytics-entity-details"));
exports.UpdateLogAnalyticsEntityDetails = UpdateLogAnalyticsEntityDetails.UpdateLogAnalyticsEntityDetails;
const UpdateLogAnalyticsEntityTypeDetails = __importStar(require("./update-log-analytics-entity-type-details"));
exports.UpdateLogAnalyticsEntityTypeDetails = UpdateLogAnalyticsEntityTypeDetails.UpdateLogAnalyticsEntityTypeDetails;
const UpdateLogAnalyticsLogGroupDetails = __importStar(require("./update-log-analytics-log-group-details"));
exports.UpdateLogAnalyticsLogGroupDetails = UpdateLogAnalyticsLogGroupDetails.UpdateLogAnalyticsLogGroupDetails;
const UpdateLogAnalyticsObjectCollectionRuleDetails = __importStar(require("./update-log-analytics-object-collection-rule-details"));
exports.UpdateLogAnalyticsObjectCollectionRuleDetails = UpdateLogAnalyticsObjectCollectionRuleDetails.UpdateLogAnalyticsObjectCollectionRuleDetails;
const UpdateLookupMetadataDetails = __importStar(require("./update-lookup-metadata-details"));
exports.UpdateLookupMetadataDetails = UpdateLookupMetadataDetails.UpdateLookupMetadataDetails;
const UpdateScheduledTaskDetails = __importStar(require("./update-scheduled-task-details"));
exports.UpdateScheduledTaskDetails = UpdateScheduledTaskDetails.UpdateScheduledTaskDetails;
const UpdateStorageDetails = __importStar(require("./update-storage-details"));
exports.UpdateStorageDetails = UpdateStorageDetails.UpdateStorageDetails;
const Upload = __importStar(require("./upload"));
exports.Upload = Upload.Upload;
const UploadCollection = __importStar(require("./upload-collection"));
exports.UploadCollection = UploadCollection.UploadCollection;
const UploadFileCollection = __importStar(require("./upload-file-collection"));
exports.UploadFileCollection = UploadFileCollection.UploadFileCollection;
const UploadFileStatus = __importStar(require("./upload-file-status"));
exports.UploadFileStatus = UploadFileStatus.UploadFileStatus;
const UploadFileSummary = __importStar(require("./upload-file-summary"));
exports.UploadFileSummary = UploadFileSummary.UploadFileSummary;
const UploadSummary = __importStar(require("./upload-summary"));
exports.UploadSummary = UploadSummary.UploadSummary;
const UploadWarningCollection = __importStar(require("./upload-warning-collection"));
exports.UploadWarningCollection = UploadWarningCollection.UploadWarningCollection;
const UploadWarningSummary = __importStar(require("./upload-warning-summary"));
exports.UploadWarningSummary = UploadWarningSummary.UploadWarningSummary;
const UpsertLogAnalyticsAssociation = __importStar(require("./upsert-log-analytics-association"));
exports.UpsertLogAnalyticsAssociation = UpsertLogAnalyticsAssociation.UpsertLogAnalyticsAssociation;
const UpsertLogAnalyticsAssociationDetails = __importStar(require("./upsert-log-analytics-association-details"));
exports.UpsertLogAnalyticsAssociationDetails = UpsertLogAnalyticsAssociationDetails.UpsertLogAnalyticsAssociationDetails;
const UpsertLogAnalyticsFieldDetails = __importStar(require("./upsert-log-analytics-field-details"));
exports.UpsertLogAnalyticsFieldDetails = UpsertLogAnalyticsFieldDetails.UpsertLogAnalyticsFieldDetails;
const UpsertLogAnalyticsLabelDetails = __importStar(require("./upsert-log-analytics-label-details"));
exports.UpsertLogAnalyticsLabelDetails = UpsertLogAnalyticsLabelDetails.UpsertLogAnalyticsLabelDetails;
const UpsertLogAnalyticsParserDetails = __importStar(require("./upsert-log-analytics-parser-details"));
exports.UpsertLogAnalyticsParserDetails = UpsertLogAnalyticsParserDetails.UpsertLogAnalyticsParserDetails;
const UpsertLogAnalyticsSourceDetails = __importStar(require("./upsert-log-analytics-source-details"));
exports.UpsertLogAnalyticsSourceDetails = UpsertLogAnalyticsSourceDetails.UpsertLogAnalyticsSourceDetails;
const UsageStatusItem = __importStar(require("./usage-status-item"));
exports.UsageStatusItem = UsageStatusItem.UsageStatusItem;
const ValidateEndpointResult = __importStar(require("./validate-endpoint-result"));
exports.ValidateEndpointResult = ValidateEndpointResult.ValidateEndpointResult;
const ValidateLabelConditionDetails = __importStar(require("./validate-label-condition-details"));
exports.ValidateLabelConditionDetails = ValidateLabelConditionDetails.ValidateLabelConditionDetails;
const ValidateLabelConditionResult = __importStar(require("./validate-label-condition-result"));
exports.ValidateLabelConditionResult = ValidateLabelConditionResult.ValidateLabelConditionResult;
const ValueType = __importStar(require("./value-type"));
exports.ValueType = ValueType.ValueType;
const VariableDefinition = __importStar(require("./variable-definition"));
exports.VariableDefinition = VariableDefinition.VariableDefinition;
const VerifyOutput = __importStar(require("./verify-output"));
exports.VerifyOutput = VerifyOutput.VerifyOutput;
const Violation = __importStar(require("./violation"));
exports.Violation = Violation.Violation;
const WarningReferenceDetails = __importStar(require("./warning-reference-details"));
exports.WarningReferenceDetails = WarningReferenceDetails.WarningReferenceDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLog = __importStar(require("./work-request-log"));
exports.WorkRequestLog = WorkRequestLog.WorkRequestLog;
const WorkRequestLogCollection = __importStar(require("./work-request-log-collection"));
exports.WorkRequestLogCollection = WorkRequestLogCollection.WorkRequestLogCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestStatus = __importStar(require("./work-request-status"));
exports.WorkRequestStatus = WorkRequestStatus.WorkRequestStatus;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const AddFieldsCommandDescriptor = __importStar(require("./add-fields-command-descriptor"));
exports.AddFieldsCommandDescriptor = AddFieldsCommandDescriptor.AddFieldsCommandDescriptor;
const AddInsightsCommandDescriptor = __importStar(require("./add-insights-command-descriptor"));
exports.AddInsightsCommandDescriptor = AddInsightsCommandDescriptor.AddInsightsCommandDescriptor;
const AnomalyCommandDescriptor = __importStar(require("./anomaly-command-descriptor"));
exports.AnomalyCommandDescriptor = AnomalyCommandDescriptor.AnomalyCommandDescriptor;
const AutoSchedule = __importStar(require("./auto-schedule"));
exports.AutoSchedule = AutoSchedule.AutoSchedule;
const BottomCommandDescriptor = __importStar(require("./bottom-command-descriptor"));
exports.BottomCommandDescriptor = BottomCommandDescriptor.BottomCommandDescriptor;
const BucketCommandDescriptor = __importStar(require("./bucket-command-descriptor"));
exports.BucketCommandDescriptor = BucketCommandDescriptor.BucketCommandDescriptor;
const ChartColumn = __importStar(require("./chart-column"));
exports.ChartColumn = ChartColumn.ChartColumn;
const ChartDataColumn = __importStar(require("./chart-data-column"));
exports.ChartDataColumn = ChartDataColumn.ChartDataColumn;
const ClassifyColumn = __importStar(require("./classify-column"));
exports.ClassifyColumn = ClassifyColumn.ClassifyColumn;
const ClassifyCommandDescriptor = __importStar(require("./classify-command-descriptor"));
exports.ClassifyCommandDescriptor = ClassifyCommandDescriptor.ClassifyCommandDescriptor;
const ClusterCommandDescriptor = __importStar(require("./cluster-command-descriptor"));
exports.ClusterCommandDescriptor = ClusterCommandDescriptor.ClusterCommandDescriptor;
const ClusterCompareCommandDescriptor = __importStar(require("./cluster-compare-command-descriptor"));
exports.ClusterCompareCommandDescriptor = ClusterCompareCommandDescriptor.ClusterCompareCommandDescriptor;
const ClusterDetailsCommandDescriptor = __importStar(require("./cluster-details-command-descriptor"));
exports.ClusterDetailsCommandDescriptor = ClusterDetailsCommandDescriptor.ClusterDetailsCommandDescriptor;
const ClusterSplitCommandDescriptor = __importStar(require("./cluster-split-command-descriptor"));
exports.ClusterSplitCommandDescriptor = ClusterSplitCommandDescriptor.ClusterSplitCommandDescriptor;
const Column = __importStar(require("./column"));
exports.Column = Column.Column;
const CommandDescriptor = __importStar(require("./command-descriptor"));
exports.CommandDescriptor = CommandDescriptor.CommandDescriptor;
const CompareCommandDescriptor = __importStar(require("./compare-command-descriptor"));
exports.CompareCommandDescriptor = CompareCommandDescriptor.CompareCommandDescriptor;
const CreateAccelerationTaskDetails = __importStar(require("./create-acceleration-task-details"));
exports.CreateAccelerationTaskDetails = CreateAccelerationTaskDetails.CreateAccelerationTaskDetails;
const CreateStandardTaskDetails = __importStar(require("./create-standard-task-details"));
exports.CreateStandardTaskDetails = CreateStandardTaskDetails.CreateStandardTaskDetails;
const CreateTableCommandDescriptor = __importStar(require("./create-table-command-descriptor"));
exports.CreateTableCommandDescriptor = CreateTableCommandDescriptor.CreateTableCommandDescriptor;
const CreateViewCommandDescriptor = __importStar(require("./create-view-command-descriptor"));
exports.CreateViewCommandDescriptor = CreateViewCommandDescriptor.CreateViewCommandDescriptor;
const CronSchedule = __importStar(require("./cron-schedule"));
exports.CronSchedule = CronSchedule.CronSchedule;
const DedupCommandDescriptor = __importStar(require("./dedup-command-descriptor"));
exports.DedupCommandDescriptor = DedupCommandDescriptor.DedupCommandDescriptor;
const DeleteCommandDescriptor = __importStar(require("./delete-command-descriptor"));
exports.DeleteCommandDescriptor = DeleteCommandDescriptor.DeleteCommandDescriptor;
const DeltaCommandDescriptor = __importStar(require("./delta-command-descriptor"));
exports.DeltaCommandDescriptor = DeltaCommandDescriptor.DeltaCommandDescriptor;
const DemoModeCommandDescriptor = __importStar(require("./demo-mode-command-descriptor"));
exports.DemoModeCommandDescriptor = DemoModeCommandDescriptor.DemoModeCommandDescriptor;
const DistinctCommandDescriptor = __importStar(require("./distinct-command-descriptor"));
exports.DistinctCommandDescriptor = DistinctCommandDescriptor.DistinctCommandDescriptor;
const EvalCommandDescriptor = __importStar(require("./eval-command-descriptor"));
exports.EvalCommandDescriptor = EvalCommandDescriptor.EvalCommandDescriptor;
const EventStatsCommandDescriptor = __importStar(require("./event-stats-command-descriptor"));
exports.EventStatsCommandDescriptor = EventStatsCommandDescriptor.EventStatsCommandDescriptor;
const ExtractCommandDescriptor = __importStar(require("./extract-command-descriptor"));
exports.ExtractCommandDescriptor = ExtractCommandDescriptor.ExtractCommandDescriptor;
const Field = __importStar(require("./field"));
exports.Field = Field.Field;
const FieldArgument = __importStar(require("./field-argument"));
exports.FieldArgument = FieldArgument.FieldArgument;
const FieldSummaryCommandDescriptor = __importStar(require("./field-summary-command-descriptor"));
exports.FieldSummaryCommandDescriptor = FieldSummaryCommandDescriptor.FieldSummaryCommandDescriptor;
const FieldsAddRemoveField = __importStar(require("./fields-add-remove-field"));
exports.FieldsAddRemoveField = FieldsAddRemoveField.FieldsAddRemoveField;
const FieldsCommandDescriptor = __importStar(require("./fields-command-descriptor"));
exports.FieldsCommandDescriptor = FieldsCommandDescriptor.FieldsCommandDescriptor;
const FixedFrequencySchedule = __importStar(require("./fixed-frequency-schedule"));
exports.FixedFrequencySchedule = FixedFrequencySchedule.FixedFrequencySchedule;
const FrequentCommandDescriptor = __importStar(require("./frequent-command-descriptor"));
exports.FrequentCommandDescriptor = FrequentCommandDescriptor.FrequentCommandDescriptor;
const FunctionField = __importStar(require("./function-field"));
exports.FunctionField = FunctionField.FunctionField;
const GeoStatsCommandDescriptor = __importStar(require("./geo-stats-command-descriptor"));
exports.GeoStatsCommandDescriptor = GeoStatsCommandDescriptor.GeoStatsCommandDescriptor;
const HeadCommandDescriptor = __importStar(require("./head-command-descriptor"));
exports.HeadCommandDescriptor = HeadCommandDescriptor.HeadCommandDescriptor;
const HighlightCommandDescriptor = __importStar(require("./highlight-command-descriptor"));
exports.HighlightCommandDescriptor = HighlightCommandDescriptor.HighlightCommandDescriptor;
const HighlightGroupsCommandDescriptor = __importStar(require("./highlight-groups-command-descriptor"));
exports.HighlightGroupsCommandDescriptor = HighlightGroupsCommandDescriptor.HighlightGroupsCommandDescriptor;
const HighlightRowsCommandDescriptor = __importStar(require("./highlight-rows-command-descriptor"));
exports.HighlightRowsCommandDescriptor = HighlightRowsCommandDescriptor.HighlightRowsCommandDescriptor;
const IngestTimeRuleFieldCondition = __importStar(require("./ingest-time-rule-field-condition"));
exports.IngestTimeRuleFieldCondition = IngestTimeRuleFieldCondition.IngestTimeRuleFieldCondition;
const IngestTimeRuleMetricExtractionAction = __importStar(require("./ingest-time-rule-metric-extraction-action"));
exports.IngestTimeRuleMetricExtractionAction = IngestTimeRuleMetricExtractionAction.IngestTimeRuleMetricExtractionAction;
const JsonExtractCommandDescriptor = __importStar(require("./json-extract-command-descriptor"));
exports.JsonExtractCommandDescriptor = JsonExtractCommandDescriptor.JsonExtractCommandDescriptor;
const LinkCommandDescriptor = __importStar(require("./link-command-descriptor"));
exports.LinkCommandDescriptor = LinkCommandDescriptor.LinkCommandDescriptor;
const LinkDetailsCommandDescriptor = __importStar(require("./link-details-command-descriptor"));
exports.LinkDetailsCommandDescriptor = LinkDetailsCommandDescriptor.LinkDetailsCommandDescriptor;
const LiteralArgument = __importStar(require("./literal-argument"));
exports.LiteralArgument = LiteralArgument.LiteralArgument;
const LogListTypeEndpoint = __importStar(require("./log-list-type-endpoint"));
exports.LogListTypeEndpoint = LogListTypeEndpoint.LogListTypeEndpoint;
const LogTypeEndpoint = __importStar(require("./log-type-endpoint"));
exports.LogTypeEndpoint = LogTypeEndpoint.LogTypeEndpoint;
const LookupCommandDescriptor = __importStar(require("./lookup-command-descriptor"));
exports.LookupCommandDescriptor = LookupCommandDescriptor.LookupCommandDescriptor;
const MacroCommandDescriptor = __importStar(require("./macro-command-descriptor"));
exports.MacroCommandDescriptor = MacroCommandDescriptor.MacroCommandDescriptor;
const MapCommandDescriptor = __importStar(require("./map-command-descriptor"));
exports.MapCommandDescriptor = MapCommandDescriptor.MapCommandDescriptor;
const ModuleCommandDescriptor = __importStar(require("./module-command-descriptor"));
exports.ModuleCommandDescriptor = ModuleCommandDescriptor.ModuleCommandDescriptor;
const MultiSearchCommandDescriptor = __importStar(require("./multi-search-command-descriptor"));
exports.MultiSearchCommandDescriptor = MultiSearchCommandDescriptor.MultiSearchCommandDescriptor;
const NlpCommandDescriptor = __importStar(require("./nlp-command-descriptor"));
exports.NlpCommandDescriptor = NlpCommandDescriptor.NlpCommandDescriptor;
const OutlierCommandDescriptor = __importStar(require("./outlier-command-descriptor"));
exports.OutlierCommandDescriptor = OutlierCommandDescriptor.OutlierCommandDescriptor;
const PurgeAction = __importStar(require("./purge-action"));
exports.PurgeAction = PurgeAction.PurgeAction;
const RareCommandDescriptor = __importStar(require("./rare-command-descriptor"));
exports.RareCommandDescriptor = RareCommandDescriptor.RareCommandDescriptor;
const RegexCommandDescriptor = __importStar(require("./regex-command-descriptor"));
exports.RegexCommandDescriptor = RegexCommandDescriptor.RegexCommandDescriptor;
const RenameCommandDescriptor = __importStar(require("./rename-command-descriptor"));
exports.RenameCommandDescriptor = RenameCommandDescriptor.RenameCommandDescriptor;
const SearchCommandDescriptor = __importStar(require("./search-command-descriptor"));
exports.SearchCommandDescriptor = SearchCommandDescriptor.SearchCommandDescriptor;
const SearchLookupCommandDescriptor = __importStar(require("./search-lookup-command-descriptor"));
exports.SearchLookupCommandDescriptor = SearchLookupCommandDescriptor.SearchLookupCommandDescriptor;
const SequenceCommandDescriptor = __importStar(require("./sequence-command-descriptor"));
exports.SequenceCommandDescriptor = SequenceCommandDescriptor.SequenceCommandDescriptor;
const SortCommandDescriptor = __importStar(require("./sort-command-descriptor"));
exports.SortCommandDescriptor = SortCommandDescriptor.SortCommandDescriptor;
const SortField = __importStar(require("./sort-field"));
exports.SortField = SortField.SortField;
const StandardTask = __importStar(require("./standard-task"));
exports.StandardTask = StandardTask.StandardTask;
const StatsCommandDescriptor = __importStar(require("./stats-command-descriptor"));
exports.StatsCommandDescriptor = StatsCommandDescriptor.StatsCommandDescriptor;
const StreamAction = __importStar(require("./stream-action"));
exports.StreamAction = StreamAction.StreamAction;
const TableColumn = __importStar(require("./table-column"));
exports.TableColumn = TableColumn.TableColumn;
const TailCommandDescriptor = __importStar(require("./tail-command-descriptor"));
exports.TailCommandDescriptor = TailCommandDescriptor.TailCommandDescriptor;
const TimeClusterColumn = __importStar(require("./time-cluster-column"));
exports.TimeClusterColumn = TimeClusterColumn.TimeClusterColumn;
const TimeClusterCommandDescriptor = __importStar(require("./time-cluster-command-descriptor"));
exports.TimeClusterCommandDescriptor = TimeClusterCommandDescriptor.TimeClusterCommandDescriptor;
const TimeClusterDataColumn = __importStar(require("./time-cluster-data-column"));
exports.TimeClusterDataColumn = TimeClusterDataColumn.TimeClusterDataColumn;
const TimeColumn = __importStar(require("./time-column"));
exports.TimeColumn = TimeColumn.TimeColumn;
const TimeCompareCommandDescriptor = __importStar(require("./time-compare-command-descriptor"));
exports.TimeCompareCommandDescriptor = TimeCompareCommandDescriptor.TimeCompareCommandDescriptor;
const TimeStatsColumn = __importStar(require("./time-stats-column"));
exports.TimeStatsColumn = TimeStatsColumn.TimeStatsColumn;
const TimeStatsCommandDescriptor = __importStar(require("./time-stats-command-descriptor"));
exports.TimeStatsCommandDescriptor = TimeStatsCommandDescriptor.TimeStatsCommandDescriptor;
const TimeStatsDataColumn = __importStar(require("./time-stats-data-column"));
exports.TimeStatsDataColumn = TimeStatsDataColumn.TimeStatsDataColumn;
const TopCommandDescriptor = __importStar(require("./top-command-descriptor"));
exports.TopCommandDescriptor = TopCommandDescriptor.TopCommandDescriptor;
const TrendColumn = __importStar(require("./trend-column"));
exports.TrendColumn = TrendColumn.TrendColumn;
const UpdateStandardTaskDetails = __importStar(require("./update-standard-task-details"));
exports.UpdateStandardTaskDetails = UpdateStandardTaskDetails.UpdateStandardTaskDetails;
const UpdateTableCommandDescriptor = __importStar(require("./update-table-command-descriptor"));
exports.UpdateTableCommandDescriptor = UpdateTableCommandDescriptor.UpdateTableCommandDescriptor;
const WhereCommandDescriptor = __importStar(require("./where-command-descriptor"));
exports.WhereCommandDescriptor = WhereCommandDescriptor.WhereCommandDescriptor;
const XmlExtractCommandDescriptor = __importStar(require("./xml-extract-command-descriptor"));
exports.XmlExtractCommandDescriptor = XmlExtractCommandDescriptor.XmlExtractCommandDescriptor;
