// oci-loganalytics/lib/model/property-definition.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertyDefinition = void 0;
var PropertyDefinition;
(function (PropertyDefinition) {
    let Type;
    (function (Type) {
        Type["String"] = "STRING";
        Type["Double"] = "DOUBLE";
        Type["Float"] = "FLOAT";
        Type["Long"] = "LONG";
        Type["Integer"] = "INTEGER";
        Type["Timestamp"] = "TIMESTAMP";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = PropertyDefinition.Type || (PropertyDefinition.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PropertyDefinition.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PropertyDefinition.getDeserializedJsonObj = getDeserializedJsonObj;
})(PropertyDefinition = exports.PropertyDefinition || (exports.PropertyDefinition = {}));
