// oci-loganalytics/lib/model/recall-status.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecallStatus = void 0;
/**
 * These are the possible statuses for recall
 *
 **/
var RecallStatus;
(function (RecallStatus) {
    RecallStatus["Recalled"] = "RECALLED";
    RecallStatus["Pending"] = "PENDING";
    RecallStatus["Failed"] = "FAILED";
    RecallStatus["PartialRecalled"] = "PARTIAL_RECALLED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RecallStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(RecallStatus = exports.RecallStatus || (exports.RecallStatus = {}));
(function (RecallStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    RecallStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RecallStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(RecallStatus = exports.RecallStatus || (exports.RecallStatus = {}));
