// oci-loganalytics/lib/model/stream-cursor-types.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamCursorTypes = void 0;
/**
 * Allowed values for stream cursor types.
 *
 **/
var StreamCursorTypes;
(function (StreamCursorTypes) {
    StreamCursorTypes["Default"] = "DEFAULT";
    StreamCursorTypes["TrimHorizon"] = "TRIM_HORIZON";
    StreamCursorTypes["Latest"] = "LATEST";
    StreamCursorTypes["AtTime"] = "AT_TIME";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    StreamCursorTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(StreamCursorTypes = exports.StreamCursorTypes || (exports.StreamCursorTypes = {}));
(function (StreamCursorTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    StreamCursorTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    StreamCursorTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(StreamCursorTypes = exports.StreamCursorTypes || (exports.StreamCursorTypes = {}));
