// oci-datalabelingservicedataplane/lib/model/import-format.js
"use strict";
/**
 * Data Labeling API
 * Use Data Labeling API to create Annotations on Images, Texts & Documents, and generate snapshots.
 * OpenAPI spec version: 20211001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportFormat = void 0;
var ImportFormat;
(function (ImportFormat) {
    let Name;
    (function (Name) {
        Name["JsonlConsolidated"] = "JSONL_CONSOLIDATED";
        Name["JsonlCompactPlusContent"] = "JSONL_COMPACT_PLUS_CONTENT";
        Name["Conll"] = "CONLL";
        Name["Spacy"] = "SPACY";
        Name["Coco"] = "COCO";
        Name["Yolo"] = "YOLO";
        Name["PascalVoc"] = "PASCAL_VOC";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Name["UnknownValue"] = "UNKNOWN_VALUE";
    })(Name = ImportFormat.Name || (ImportFormat.Name = {}));
    let Version;
    (function (Version) {
        Version["V2003"] = "V2003";
        Version["V5"] = "V5";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Version["UnknownValue"] = "UNKNOWN_VALUE";
    })(Version = ImportFormat.Version || (ImportFormat.Version = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ImportFormat.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ImportFormat.getDeserializedJsonObj = getDeserializedJsonObj;
})(ImportFormat = exports.ImportFormat || (exports.ImportFormat = {}));
