// oci-disasterrecovery/lib/model/dr-plan-execution-lifecycle-state.js
"use strict";
/**
 * Full Stack Disaster Recovery API
 * Use the Full Stack Disaster Recovery (DR) API to manage disaster recovery for business applications.
Full Stack DR is an OCI disaster recovery orchestration and management service that provides comprehensive disaster
recovery capabilities for all layers of an application stack, including infrastructure, middleware, database,
and application.

 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrPlanExecutionLifecycleState = void 0;
/**
 * The lifecycle states of a DR plan execution.
 * - ACCEPTED - DR plan execution has been accepted.
 * - IN_PROGRESS - DR plan execution is in progress.
 * - WAITING - DR plan execution is waiting for user action.
 * - CANCELING - DR plan execution is being canceled.
 * - CANCELED - DR plan execution has been canceled.
 * - SUCCEEDED - DR plan execution has succeeded.
 * - FAILED - DR plan execution has failed.
 * - DELETING - DR plan execution is being deleted.
 * - DELETED - DR plan execution has been deleted.
 * - PAUSING - DR plan execution is being paused.
 * - PAUSED - DR plan execution has been paused.
 * - RESUMING - DR plan execution is resuming.
 *
 **/
var DrPlanExecutionLifecycleState;
(function (DrPlanExecutionLifecycleState) {
    DrPlanExecutionLifecycleState["Accepted"] = "ACCEPTED";
    DrPlanExecutionLifecycleState["InProgress"] = "IN_PROGRESS";
    DrPlanExecutionLifecycleState["Waiting"] = "WAITING";
    DrPlanExecutionLifecycleState["Canceling"] = "CANCELING";
    DrPlanExecutionLifecycleState["Canceled"] = "CANCELED";
    DrPlanExecutionLifecycleState["Succeeded"] = "SUCCEEDED";
    DrPlanExecutionLifecycleState["Failed"] = "FAILED";
    DrPlanExecutionLifecycleState["Deleting"] = "DELETING";
    DrPlanExecutionLifecycleState["Deleted"] = "DELETED";
    DrPlanExecutionLifecycleState["Pausing"] = "PAUSING";
    DrPlanExecutionLifecycleState["Paused"] = "PAUSED";
    DrPlanExecutionLifecycleState["Resuming"] = "RESUMING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DrPlanExecutionLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(DrPlanExecutionLifecycleState = exports.DrPlanExecutionLifecycleState || (exports.DrPlanExecutionLifecycleState = {}));
(function (DrPlanExecutionLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    DrPlanExecutionLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DrPlanExecutionLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(DrPlanExecutionLifecycleState = exports.DrPlanExecutionLifecycleState || (exports.DrPlanExecutionLifecycleState = {}));
