// oci-disasterrecovery/lib/model/dr-plan-group-refresh-status.js
"use strict";
/**
 * Full Stack Disaster Recovery API
 * Use the Full Stack Disaster Recovery (DR) API to manage disaster recovery for business applications.
Full Stack DR is an OCI disaster recovery orchestration and management service that provides comprehensive disaster
recovery capabilities for all layers of an application stack, including infrastructure, middleware, database,
and application.

 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrPlanGroupRefreshStatus = void 0;
/**
 * The refresh status of a DR plan group.
 * - GROUP_ADDED - DR plan group was added to the plan.
 * - GROUP_DELETED - DR plan group was deleted from the plan.
 * - GROUP_MODIFIED - DR plan group was modified.
 *
 **/
var DrPlanGroupRefreshStatus;
(function (DrPlanGroupRefreshStatus) {
    DrPlanGroupRefreshStatus["GroupAdded"] = "GROUP_ADDED";
    DrPlanGroupRefreshStatus["GroupDeleted"] = "GROUP_DELETED";
    DrPlanGroupRefreshStatus["GroupModified"] = "GROUP_MODIFIED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DrPlanGroupRefreshStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(DrPlanGroupRefreshStatus = exports.DrPlanGroupRefreshStatus || (exports.DrPlanGroupRefreshStatus = {}));
(function (DrPlanGroupRefreshStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    DrPlanGroupRefreshStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DrPlanGroupRefreshStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(DrPlanGroupRefreshStatus = exports.DrPlanGroupRefreshStatus || (exports.DrPlanGroupRefreshStatus = {}));
