// oci-disasterrecovery/lib/model/dr-protection-group-lifecycle-sub-state.js
"use strict";
/**
 * Full Stack Disaster Recovery API
 * Use the Full Stack Disaster Recovery (DR) API to manage disaster recovery for business applications.
Full Stack DR is an OCI disaster recovery orchestration and management service that provides comprehensive disaster
recovery capabilities for all layers of an application stack, including infrastructure, middleware, database,
and application.

 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrProtectionGroupLifecycleSubState = void 0;
/**
 * The secondary lifecycle states of a DR protection group.  Provides information in addition to the lifecycle state.
 * - DR_DRILL_IN_PROGRESS - A DR Drill is in progress.
 *
 **/
var DrProtectionGroupLifecycleSubState;
(function (DrProtectionGroupLifecycleSubState) {
    DrProtectionGroupLifecycleSubState["DrDrillInProgress"] = "DR_DRILL_IN_PROGRESS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DrProtectionGroupLifecycleSubState["UnknownValue"] = "UNKNOWN_VALUE";
})(DrProtectionGroupLifecycleSubState = exports.DrProtectionGroupLifecycleSubState || (exports.DrProtectionGroupLifecycleSubState = {}));
(function (DrProtectionGroupLifecycleSubState) {
    function getJsonObj(obj) {
        return obj;
    }
    DrProtectionGroupLifecycleSubState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DrProtectionGroupLifecycleSubState.getDeserializedJsonObj = getDeserializedJsonObj;
})(DrProtectionGroupLifecycleSubState = exports.DrProtectionGroupLifecycleSubState || (exports.DrProtectionGroupLifecycleSubState = {}));
