// oci-disasterrecovery/lib/model/dr-plan-execution-type.js
"use strict";
/**
 * Full Stack Disaster Recovery API
 * Use the Full Stack Disaster Recovery (DR) API to manage disaster recovery for business applications.
Full Stack DR is an OCI disaster recovery orchestration and management service that provides comprehensive disaster
recovery capabilities for all layers of an application stack, including infrastructure, middleware, database,
and application.

 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrPlanExecutionType = void 0;
/**
 * The types of DR plan execution that can be created, updated, or deleted by the user.  Executions of type *precheck* cannot be created separately by the user.
 * They are automatically generated as part of the plan, however prechecks can be executed separately from a plan.
 * - SWITCHOVER - Switchover from primary to standby.
 * - SWITCHOVER_PRECHECK - Precheck for switchover DR plan.
 * - FAILOVER - Failover to standby.
 * - FAILOVER_PRECHECK - Precheck for failover DR plan.
 * - START_DRILL - Start DR Drill.
 * - START_DRILL_PRECHECK - Precheck for a start drill DR plan.
 * - STOP_DRILL - Stop DR Drill.
 * - STOP_DRILL_PRECHECK - Precheck for a stop drill DR plan.
 *
 **/
var DrPlanExecutionType;
(function (DrPlanExecutionType) {
    DrPlanExecutionType["Switchover"] = "SWITCHOVER";
    DrPlanExecutionType["SwitchoverPrecheck"] = "SWITCHOVER_PRECHECK";
    DrPlanExecutionType["Failover"] = "FAILOVER";
    DrPlanExecutionType["FailoverPrecheck"] = "FAILOVER_PRECHECK";
    DrPlanExecutionType["StartDrill"] = "START_DRILL";
    DrPlanExecutionType["StartDrillPrecheck"] = "START_DRILL_PRECHECK";
    DrPlanExecutionType["StopDrill"] = "STOP_DRILL";
    DrPlanExecutionType["StopDrillPrecheck"] = "STOP_DRILL_PRECHECK";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DrPlanExecutionType["UnknownValue"] = "UNKNOWN_VALUE";
})(DrPlanExecutionType = exports.DrPlanExecutionType || (exports.DrPlanExecutionType = {}));
(function (DrPlanExecutionType) {
    function getJsonObj(obj) {
        return obj;
    }
    DrPlanExecutionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DrPlanExecutionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(DrPlanExecutionType = exports.DrPlanExecutionType || (exports.DrPlanExecutionType = {}));
