// oci-disasterrecovery/lib/model/dr-protection-group-member-type.js
"use strict";
/**
 * Full Stack Disaster Recovery API
 * Use the Full Stack Disaster Recovery (DR) API to manage disaster recovery for business applications.
Full Stack DR is an OCI disaster recovery orchestration and management service that provides comprehensive disaster
recovery capabilities for all layers of an application stack, including infrastructure, middleware, database,
and application.

 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrProtectionGroupMemberType = void 0;
/**
 * Enumeration of different types of Oracle Cloud Infrastructure (OCI) resources that are members in a DR protection group.
 * - COMPUTE_INSTANCE - Deprecated. A compute instance.
 * - COMPUTE_INSTANCE_MOVABLE - A compute instance that moves across regions or ADs during DR.
 * - COMPUTE_INSTANCE_NON_MOVABLE - A compute instance that does not move across regions or ADs during DR.
 * - VOLUME_GROUP - A volume group.
 * - DATABASE - An Oracle Database instance from Base Database service, or Exadata Database service.
 * - AUTONOMOUS_DATABASE - An Oracle Autonomous Database Serverless.
 * - AUTONOMOUS_CONTAINER_DATABASE - An Oracle Autonomous Container Database.
 * - OKE_CLUSTER - OCI Container Engine for Kubernetes(OKE) cluster.
 * - LOAD_BALANCER - A load balancer.
 * - NETWORK_LOAD_BALANCER - A network load balancer.
 * - FILE_SYSTEM - A file system.
 *
 **/
var DrProtectionGroupMemberType;
(function (DrProtectionGroupMemberType) {
    DrProtectionGroupMemberType["ComputeInstance"] = "COMPUTE_INSTANCE";
    DrProtectionGroupMemberType["ComputeInstanceMovable"] = "COMPUTE_INSTANCE_MOVABLE";
    DrProtectionGroupMemberType["ComputeInstanceNonMovable"] = "COMPUTE_INSTANCE_NON_MOVABLE";
    DrProtectionGroupMemberType["VolumeGroup"] = "VOLUME_GROUP";
    DrProtectionGroupMemberType["Database"] = "DATABASE";
    DrProtectionGroupMemberType["AutonomousDatabase"] = "AUTONOMOUS_DATABASE";
    DrProtectionGroupMemberType["AutonomousContainerDatabase"] = "AUTONOMOUS_CONTAINER_DATABASE";
    DrProtectionGroupMemberType["LoadBalancer"] = "LOAD_BALANCER";
    DrProtectionGroupMemberType["NetworkLoadBalancer"] = "NETWORK_LOAD_BALANCER";
    DrProtectionGroupMemberType["FileSystem"] = "FILE_SYSTEM";
    DrProtectionGroupMemberType["OkeCluster"] = "OKE_CLUSTER";
    DrProtectionGroupMemberType["ObjectStorageBucket"] = "OBJECT_STORAGE_BUCKET";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DrProtectionGroupMemberType["UnknownValue"] = "UNKNOWN_VALUE";
})(DrProtectionGroupMemberType = exports.DrProtectionGroupMemberType || (exports.DrProtectionGroupMemberType = {}));
(function (DrProtectionGroupMemberType) {
    function getJsonObj(obj) {
        return obj;
    }
    DrProtectionGroupMemberType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DrProtectionGroupMemberType.getDeserializedJsonObj = getDeserializedJsonObj;
})(DrProtectionGroupMemberType = exports.DrProtectionGroupMemberType || (exports.DrProtectionGroupMemberType = {}));
