// oci-disasterrecovery/lib/model/oke-cluster-image-replication.js
"use strict";
/**
 * Full Stack Disaster Recovery API
 * Use the Full Stack Disaster Recovery (DR) API to manage disaster recovery for business applications.
Full Stack DR is an OCI disaster recovery orchestration and management service that provides comprehensive disaster
recovery capabilities for all layers of an application stack, including infrastructure, middleware, database,
and application.

 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OkeClusterImageReplication = void 0;
/**
 * Controls the behavior of image replication across regions.
 * Image replication is enabled by default for DR Protection Groups with a primary role.
 *
 **/
var OkeClusterImageReplication;
(function (OkeClusterImageReplication) {
    OkeClusterImageReplication["Enable"] = "ENABLE";
    OkeClusterImageReplication["Disable"] = "DISABLE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OkeClusterImageReplication["UnknownValue"] = "UNKNOWN_VALUE";
})(OkeClusterImageReplication = exports.OkeClusterImageReplication || (exports.OkeClusterImageReplication = {}));
(function (OkeClusterImageReplication) {
    function getJsonObj(obj) {
        return obj;
    }
    OkeClusterImageReplication.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OkeClusterImageReplication.getDeserializedJsonObj = getDeserializedJsonObj;
})(OkeClusterImageReplication = exports.OkeClusterImageReplication || (exports.OkeClusterImageReplication = {}));
