// oci-devops/lib/model/metric-name.js
"use strict";
/**
 * DevOps API
 * Use the DevOps API to create DevOps projects, configure code repositories,  add artifacts to deploy, build and test software applications, configure  target deployment environments, and deploy software applications.  For more information, see [DevOps](https://docs.oracle.com/iaas/Content/devops/using/home.htm).
 * OpenAPI spec version: 20210630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricName = void 0;
/**
 * The metric type:
 * COMMITS - Represents the number of commits.
 * LINES_ADDED - Represents the number of lines of code added.
 * LINES_DELETED - Represents the number of lines of code deleted.
 * PULL_REQUEST_CREATED - Represents the number of pull requests which is in created/ open state.
 * PULL_REQUEST_MERGED - Represents the number of pull requests which is in merging/ merged state.
 * PULL_REQUEST_DECLINED - Represents the number of pull requests which is in declined state.
 * PULL_REQUEST_REVIEW_START_DURATION_IN_DAYS - Represents the time from when a pull request is created to the first comment on the pull request.
 * PULL_REQUEST_REVIEW_DURATION_IN_DAYS - Represents the time from the first comment on the pull request to when the pull request is merged or closed.
 * PULL_REQUEST_APPROVED - Represents the number of pull requests which is approved by the user when the pull request is merged.
 * PULL_REQUEST_REVIEWED - Represents the number of pull requests which has been commented on.
 * PULL_REQUEST_COMMENTS - Represents the total number of comments given across different pull requests.
 *
 **/
var MetricName;
(function (MetricName) {
    MetricName["Commits"] = "COMMITS";
    MetricName["LinesAdded"] = "LINES_ADDED";
    MetricName["LinesDeleted"] = "LINES_DELETED";
    MetricName["PullRequestCreated"] = "PULL_REQUEST_CREATED";
    MetricName["PullRequestMerged"] = "PULL_REQUEST_MERGED";
    MetricName["PullRequestDeclined"] = "PULL_REQUEST_DECLINED";
    MetricName["PullRequestReviewStartDurationInDays"] = "PULL_REQUEST_REVIEW_START_DURATION_IN_DAYS";
    MetricName["PullRequestReviewDurationInDays"] = "PULL_REQUEST_REVIEW_DURATION_IN_DAYS";
    MetricName["PullRequestApproved"] = "PULL_REQUEST_APPROVED";
    MetricName["PullRequestReviewed"] = "PULL_REQUEST_REVIEWED";
    MetricName["PullRequestComments"] = "PULL_REQUEST_COMMENTS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MetricName["UnknownValue"] = "UNKNOWN_VALUE";
})(MetricName = exports.MetricName || (exports.MetricName = {}));
(function (MetricName) {
    function getJsonObj(obj) {
        return obj;
    }
    MetricName.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MetricName.getDeserializedJsonObj = getDeserializedJsonObj;
})(MetricName = exports.MetricName || (exports.MetricName = {}));
