// oci-devops/lib/model/reviewer.js
"use strict";
/**
 * DevOps API
 * Use the DevOps API to create DevOps projects, configure code repositories,  add artifacts to deploy, build and test software applications, configure  target deployment environments, and deploy software applications.  For more information, see [DevOps](https://docs.oracle.com/iaas/Content/devops/using/home.htm).
 * OpenAPI spec version: 20210630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Reviewer = void 0;
var Reviewer;
(function (Reviewer) {
    let PrincipalType;
    (function (PrincipalType) {
        PrincipalType["Service"] = "SERVICE";
        PrincipalType["User"] = "USER";
        PrincipalType["Instance"] = "INSTANCE";
        PrincipalType["Resource"] = "RESOURCE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PrincipalType["UnknownValue"] = "UNKNOWN_VALUE";
    })(PrincipalType = Reviewer.PrincipalType || (Reviewer.PrincipalType = {}));
    let PrincipalState;
    (function (PrincipalState) {
        PrincipalState["Active"] = "ACTIVE";
        PrincipalState["Inactive"] = "INACTIVE";
        PrincipalState["Suppressed"] = "SUPPRESSED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PrincipalState["UnknownValue"] = "UNKNOWN_VALUE";
    })(PrincipalState = Reviewer.PrincipalState || (Reviewer.PrincipalState = {}));
    let Status;
    (function (Status) {
        Status["Approved"] = "APPROVED";
        Status["ReviewPending"] = "REVIEW_PENDING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = Reviewer.Status || (Reviewer.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Reviewer.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Reviewer.getDeserializedJsonObj = getDeserializedJsonObj;
})(Reviewer = exports.Reviewer || (exports.Reviewer = {}));
