// oci-clusterplacementgroups/lib/model/index.js
"use strict";
/**
 * Cluster Placement Groups API
 * API for managing cluster placement groups.
 * OpenAPI spec version: 20230801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateClusterPlacementGroupDetails = exports.SortOrder = exports.PlacementInstructionDetails = exports.OperationType = exports.OperationStatus = exports.CreateClusterPlacementGroupDetails = exports.ClusterPlacementGroupType = exports.ClusterPlacementGroupSummary = exports.ClusterPlacementGroupCollection = exports.ClusterPlacementGroup = exports.ChangeClusterPlacementGroupCompartmentDetails = exports.CapabilityDetails = exports.CapabilitiesCollection = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const CapabilitiesCollection = __importStar(require("./capabilities-collection"));
exports.CapabilitiesCollection = CapabilitiesCollection.CapabilitiesCollection;
const CapabilityDetails = __importStar(require("./capability-details"));
exports.CapabilityDetails = CapabilityDetails.CapabilityDetails;
const ChangeClusterPlacementGroupCompartmentDetails = __importStar(require("./change-cluster-placement-group-compartment-details"));
exports.ChangeClusterPlacementGroupCompartmentDetails = ChangeClusterPlacementGroupCompartmentDetails.ChangeClusterPlacementGroupCompartmentDetails;
const ClusterPlacementGroup = __importStar(require("./cluster-placement-group"));
exports.ClusterPlacementGroup = ClusterPlacementGroup.ClusterPlacementGroup;
const ClusterPlacementGroupCollection = __importStar(require("./cluster-placement-group-collection"));
exports.ClusterPlacementGroupCollection = ClusterPlacementGroupCollection.ClusterPlacementGroupCollection;
const ClusterPlacementGroupSummary = __importStar(require("./cluster-placement-group-summary"));
exports.ClusterPlacementGroupSummary = ClusterPlacementGroupSummary.ClusterPlacementGroupSummary;
const ClusterPlacementGroupType = __importStar(require("./cluster-placement-group-type"));
exports.ClusterPlacementGroupType = ClusterPlacementGroupType.ClusterPlacementGroupType;
const CreateClusterPlacementGroupDetails = __importStar(require("./create-cluster-placement-group-details"));
exports.CreateClusterPlacementGroupDetails = CreateClusterPlacementGroupDetails.CreateClusterPlacementGroupDetails;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PlacementInstructionDetails = __importStar(require("./placement-instruction-details"));
exports.PlacementInstructionDetails = PlacementInstructionDetails.PlacementInstructionDetails;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateClusterPlacementGroupDetails = __importStar(require("./update-cluster-placement-group-details"));
exports.UpdateClusterPlacementGroupDetails = UpdateClusterPlacementGroupDetails.UpdateClusterPlacementGroupDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
