// oci-desktops/lib/request/list-desktops-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20220618
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDesktopsRequest = void 0;
var ListDesktopsRequest;
(function (ListDesktopsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListDesktopsRequest.SortBy || (ListDesktopsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListDesktopsRequest.SortOrder || (ListDesktopsRequest.SortOrder = {}));
})(ListDesktopsRequest = exports.ListDesktopsRequest || (exports.ListDesktopsRequest = {}));
