// oci-dataflow/lib/model/ip-notation-type.js
"use strict";
/**
 * Data Flow API
 * Use the Data Flow APIs to run any Apache Spark application at any scale without deploying or managing any infrastructure.

 * OpenAPI spec version: 20200129
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IpNotationType = void 0;
/**
 * The IP notation types for secure access.
 **/
var IpNotationType;
(function (IpNotationType) {
    IpNotationType["IpAddress"] = "IP_ADDRESS";
    IpNotationType["Cidr"] = "CIDR";
    IpNotationType["Vcn"] = "VCN";
    IpNotationType["VcnOcid"] = "VCN_OCID";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    IpNotationType["UnknownValue"] = "UNKNOWN_VALUE";
})(IpNotationType = exports.IpNotationType || (exports.IpNotationType = {}));
(function (IpNotationType) {
    function getJsonObj(obj) {
        return obj;
    }
    IpNotationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    IpNotationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(IpNotationType = exports.IpNotationType || (exports.IpNotationType = {}));
