// oci-dataflow/lib/model/network-type.js
"use strict";
/**
 * Data Flow API
 * Use the Data Flow APIs to run any Apache Spark application at any scale without deploying or managing any infrastructure.

 * OpenAPI spec version: 20200129
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkType = void 0;
/**
 * The possible network types.
 **/
var NetworkType;
(function (NetworkType) {
    NetworkType["Vcn"] = "VCN";
    NetworkType["SecureAccess"] = "SECURE_ACCESS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NetworkType["UnknownValue"] = "UNKNOWN_VALUE";
})(NetworkType = exports.NetworkType || (exports.NetworkType = {}));
(function (NetworkType) {
    function getJsonObj(obj) {
        return obj;
    }
    NetworkType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    NetworkType.getDeserializedJsonObj = getDeserializedJsonObj;
})(NetworkType = exports.NetworkType || (exports.NetworkType = {}));
