// oci-dataflow/lib/model/pool-lifecycle-state.js
"use strict";
/**
 * Data Flow API
 * Use the Data Flow APIs to run any Apache Spark application at any scale without deploying or managing any infrastructure.

 * OpenAPI spec version: 20200129
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PoolLifecycleState = void 0;
/**
 * The lifecycle state of a pool.
 *
 **/
var PoolLifecycleState;
(function (PoolLifecycleState) {
    PoolLifecycleState["Accepted"] = "ACCEPTED";
    PoolLifecycleState["Scheduled"] = "SCHEDULED";
    PoolLifecycleState["Creating"] = "CREATING";
    PoolLifecycleState["Active"] = "ACTIVE";
    PoolLifecycleState["Stopping"] = "STOPPING";
    PoolLifecycleState["Stopped"] = "STOPPED";
    PoolLifecycleState["Updating"] = "UPDATING";
    PoolLifecycleState["Deleting"] = "DELETING";
    PoolLifecycleState["Deleted"] = "DELETED";
    PoolLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PoolLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(PoolLifecycleState = exports.PoolLifecycleState || (exports.PoolLifecycleState = {}));
(function (PoolLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    PoolLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PoolLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(PoolLifecycleState = exports.PoolLifecycleState || (exports.PoolLifecycleState = {}));
