// oci-dataflow/lib/model/sql-endpoint-lifecycle-state.js
"use strict";
/**
 * Data Flow API
 * Use the Data Flow APIs to run any Apache Spark application at any scale without deploying or managing any infrastructure.

 * OpenAPI spec version: 20200129
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlEndpointLifecycleState = void 0;
/**
 * Common lifecycle states for resources in a Sql Endpoint:
 * CREATING        - The resource is being created and might not be usable until the entire metadata is defined.
 * ACTIVE          - The resource is valid and available for access.
 * DELETING        - The resource is being deleted, and might require a deep clean of any children.
 * DELETED         - The resource has been deleted, and isn't available.
 * FAILED          - The resource is in a failed state due to validation or other errors.
 * UPDATING        - The resource is being updated and might not be usable until all changes are commited.
 * NEEDS_ATTENTION - The resource needs customer/operator intervention.
 *
 **/
var SqlEndpointLifecycleState;
(function (SqlEndpointLifecycleState) {
    SqlEndpointLifecycleState["Creating"] = "CREATING";
    SqlEndpointLifecycleState["Active"] = "ACTIVE";
    SqlEndpointLifecycleState["Deleting"] = "DELETING";
    SqlEndpointLifecycleState["Deleted"] = "DELETED";
    SqlEndpointLifecycleState["Failed"] = "FAILED";
    SqlEndpointLifecycleState["Updating"] = "UPDATING";
    SqlEndpointLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    SqlEndpointLifecycleState["Inactive"] = "INACTIVE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SqlEndpointLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SqlEndpointLifecycleState = exports.SqlEndpointLifecycleState || (exports.SqlEndpointLifecycleState = {}));
(function (SqlEndpointLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SqlEndpointLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SqlEndpointLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlEndpointLifecycleState = exports.SqlEndpointLifecycleState || (exports.SqlEndpointLifecycleState = {}));
