// oci-zpr/lib/request/index.js
"use strict";
/**
 * Zero Trust Packet Routing Control Plane API
 * Use the Zero Trust Packet Routing Control Plane API to manage ZPR configuration and policy. See the [Zero Trust Packet Routing](https://docs.oracle.com/iaas/Content/zero-trust-packet-routing/home.htm) documentation for more information.
 * OpenAPI spec version: 20240301
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListZprPolicyWorkRequestsRequest = exports.ListZprPolicyWorkRequestLogsRequest = exports.ListZprPolicyWorkRequestErrorsRequest = exports.ListZprPoliciesRequest = exports.ListZprConfigurationWorkRequestsRequest = exports.ListZprConfigurationWorkRequestLogsRequest = exports.ListZprConfigurationWorkRequestErrorsRequest = void 0;
const ListZprConfigurationWorkRequestErrorsRequest = __importStar(require("./list-zpr-configuration-work-request-errors-request"));
exports.ListZprConfigurationWorkRequestErrorsRequest = ListZprConfigurationWorkRequestErrorsRequest.ListZprConfigurationWorkRequestErrorsRequest;
const ListZprConfigurationWorkRequestLogsRequest = __importStar(require("./list-zpr-configuration-work-request-logs-request"));
exports.ListZprConfigurationWorkRequestLogsRequest = ListZprConfigurationWorkRequestLogsRequest.ListZprConfigurationWorkRequestLogsRequest;
const ListZprConfigurationWorkRequestsRequest = __importStar(require("./list-zpr-configuration-work-requests-request"));
exports.ListZprConfigurationWorkRequestsRequest = ListZprConfigurationWorkRequestsRequest.ListZprConfigurationWorkRequestsRequest;
const ListZprPoliciesRequest = __importStar(require("./list-zpr-policies-request"));
exports.ListZprPoliciesRequest = ListZprPoliciesRequest.ListZprPoliciesRequest;
const ListZprPolicyWorkRequestErrorsRequest = __importStar(require("./list-zpr-policy-work-request-errors-request"));
exports.ListZprPolicyWorkRequestErrorsRequest = ListZprPolicyWorkRequestErrorsRequest.ListZprPolicyWorkRequestErrorsRequest;
const ListZprPolicyWorkRequestLogsRequest = __importStar(require("./list-zpr-policy-work-request-logs-request"));
exports.ListZprPolicyWorkRequestLogsRequest = ListZprPolicyWorkRequestLogsRequest.ListZprPolicyWorkRequestLogsRequest;
const ListZprPolicyWorkRequestsRequest = __importStar(require("./list-zpr-policy-work-requests-request"));
exports.ListZprPolicyWorkRequestsRequest = ListZprPolicyWorkRequestsRequest.ListZprPolicyWorkRequestsRequest;
