// oci-datascience/lib/model/container-summary.js
"use strict";
/**
 * Data Science API
 * Use the Data Science API to organize your data science work, access data and computing resources, and build, train, deploy and manage models and model deployments. For more information, see [Data Science](https://docs.oracle.com/iaas/data-science/using/data-science.htm).

 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerSummary = void 0;
const model = __importStar(require("../model"));
var ContainerSummary;
(function (ContainerSummary) {
    let TargetWorkloads;
    (function (TargetWorkloads) {
        TargetWorkloads["ModelDeployment"] = "MODEL_DEPLOYMENT";
        TargetWorkloads["JobRun"] = "JOB_RUN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TargetWorkloads["UnknownValue"] = "UNKNOWN_VALUE";
    })(TargetWorkloads = ContainerSummary.TargetWorkloads || (ContainerSummary.TargetWorkloads = {}));
    let Usages;
    (function (Usages) {
        Usages["Inference"] = "INFERENCE";
        Usages["FineTune"] = "FINE_TUNE";
        Usages["Evaluation"] = "EVALUATION";
        Usages["BatchInference"] = "BATCH_INFERENCE";
        Usages["Other"] = "OTHER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Usages["UnknownValue"] = "UNKNOWN_VALUE";
    })(Usages = ContainerSummary.Usages || (ContainerSummary.Usages = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "workloadConfigurationDetailsList": obj.workloadConfigurationDetailsList
                ? obj.workloadConfigurationDetailsList.map(item => {
                    return model.WorkloadConfigurationDetails.getJsonObj(item);
                })
                : undefined,
            "tagConfigurationList": obj.tagConfigurationList
                ? obj.tagConfigurationList.map(item => {
                    return model.TagConfiguration.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ContainerSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "workloadConfigurationDetailsList": obj.workloadConfigurationDetailsList
                ? obj.workloadConfigurationDetailsList.map(item => {
                    return model.WorkloadConfigurationDetails.getDeserializedJsonObj(item);
                })
                : undefined,
            "tagConfigurationList": obj.tagConfigurationList
                ? obj.tagConfigurationList.map(item => {
                    return model.TagConfiguration.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ContainerSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(ContainerSummary = exports.ContainerSummary || (exports.ContainerSummary = {}));
