// oci-datascience/lib/model/ml-application-instance.js
"use strict";
/**
 * Data Science API
 * Use the Data Science API to organize your data science work, access data and computing resources, and build, train, deploy and manage models and model deployments. For more information, see [Data Science](https://docs.oracle.com/iaas/data-science/using/data-science.htm).

 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MlApplicationInstance = void 0;
const model = __importStar(require("../model"));
var MlApplicationInstance;
(function (MlApplicationInstance) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = MlApplicationInstance.LifecycleState || (MlApplicationInstance.LifecycleState = {}));
    let LifecycleSubstate;
    (function (LifecycleSubstate) {
        LifecycleSubstate["Creating"] = "CREATING";
        LifecycleSubstate["Updating"] = "UPDATING";
        LifecycleSubstate["Upgrading"] = "UPGRADING";
        LifecycleSubstate["Active"] = "ACTIVE";
        LifecycleSubstate["Inactive"] = "INACTIVE";
        LifecycleSubstate["Deleting"] = "DELETING";
        LifecycleSubstate["Deleted"] = "DELETED";
        LifecycleSubstate["NeedsAttention"] = "NEEDS_ATTENTION";
        LifecycleSubstate["Failed"] = "FAILED";
        LifecycleSubstate["NonRecoverableProviderIssue"] = "NON_RECOVERABLE_PROVIDER_ISSUE";
        LifecycleSubstate["RecoverableProviderIssue"] = "RECOVERABLE_PROVIDER_ISSUE";
        LifecycleSubstate["NonRecoverableServiceIssue"] = "NON_RECOVERABLE_SERVICE_ISSUE";
        LifecycleSubstate["RecoverableServiceIssue"] = "RECOVERABLE_SERVICE_ISSUE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleSubstate["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleSubstate = MlApplicationInstance.LifecycleSubstate || (MlApplicationInstance.LifecycleSubstate = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "authConfiguration": obj.authConfiguration
                ? model.AuthConfiguration.getJsonObj(obj.authConfiguration)
                : undefined,
            "configuration": obj.configuration
                ? obj.configuration.map(item => {
                    return model.ConfigurationProperty.getJsonObj(item);
                })
                : undefined,
            "predictionEndpointDetails": obj.predictionEndpointDetails
                ? model.PredictionEndpointDetails.getJsonObj(obj.predictionEndpointDetails)
                : undefined
        });
        return jsonObj;
    }
    MlApplicationInstance.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "authConfiguration": obj.authConfiguration
                ? model.AuthConfiguration.getDeserializedJsonObj(obj.authConfiguration)
                : undefined,
            "configuration": obj.configuration
                ? obj.configuration.map(item => {
                    return model.ConfigurationProperty.getDeserializedJsonObj(item);
                })
                : undefined,
            "predictionEndpointDetails": obj.predictionEndpointDetails
                ? model.PredictionEndpointDetails.getDeserializedJsonObj(obj.predictionEndpointDetails)
                : undefined
        });
        return jsonObj;
    }
    MlApplicationInstance.getDeserializedJsonObj = getDeserializedJsonObj;
})(MlApplicationInstance = exports.MlApplicationInstance || (exports.MlApplicationInstance = {}));
