// oci-datascience/lib/model/work-request-operation-type.js
"use strict";
/**
 * Data Science API
 * Use the Data Science API to organize your data science work, access data and computing resources, and build, train, deploy and manage models and model deployments. For more information, see [Data Science](https://docs.oracle.com/iaas/data-science/using/data-science.htm).

 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestOperationType = void 0;
/**
 * The types of work request operations.
 **/
var WorkRequestOperationType;
(function (WorkRequestOperationType) {
    WorkRequestOperationType["NotebookSessionCreate"] = "NOTEBOOK_SESSION_CREATE";
    WorkRequestOperationType["NotebookSessionDelete"] = "NOTEBOOK_SESSION_DELETE";
    WorkRequestOperationType["NotebookSessionActivate"] = "NOTEBOOK_SESSION_ACTIVATE";
    WorkRequestOperationType["NotebookSessionDeactivate"] = "NOTEBOOK_SESSION_DEACTIVATE";
    WorkRequestOperationType["ModelversionsetDelete"] = "MODELVERSIONSET_DELETE";
    WorkRequestOperationType["ExportModelArtifact"] = "EXPORT_MODEL_ARTIFACT";
    WorkRequestOperationType["ImportModelArtifact"] = "IMPORT_MODEL_ARTIFACT";
    WorkRequestOperationType["ModelDeploymentCreate"] = "MODEL_DEPLOYMENT_CREATE";
    WorkRequestOperationType["ModelDeploymentDelete"] = "MODEL_DEPLOYMENT_DELETE";
    WorkRequestOperationType["ModelDeploymentActivate"] = "MODEL_DEPLOYMENT_ACTIVATE";
    WorkRequestOperationType["ModelDeploymentDeactivate"] = "MODEL_DEPLOYMENT_DEACTIVATE";
    WorkRequestOperationType["ModelDeploymentUpdate"] = "MODEL_DEPLOYMENT_UPDATE";
    WorkRequestOperationType["ProjectDelete"] = "PROJECT_DELETE";
    WorkRequestOperationType["WorkrequestCancel"] = "WORKREQUEST_CANCEL";
    WorkRequestOperationType["JobDelete"] = "JOB_DELETE";
    WorkRequestOperationType["PipelineCreate"] = "PIPELINE_CREATE";
    WorkRequestOperationType["PipelineDelete"] = "PIPELINE_DELETE";
    WorkRequestOperationType["PipelineRunCreate"] = "PIPELINE_RUN_CREATE";
    WorkRequestOperationType["PipelineRunCancel"] = "PIPELINE_RUN_CANCEL";
    WorkRequestOperationType["PipelineRunDelete"] = "PIPELINE_RUN_DELETE";
    WorkRequestOperationType["MlApplicationPackageUpload"] = "ML_APPLICATION_PACKAGE_UPLOAD";
    WorkRequestOperationType["MlApplicationTriggerStart"] = "ML_APPLICATION_TRIGGER_START";
    WorkRequestOperationType["MlApplicationImplementationDelete"] = "ML_APPLICATION_IMPLEMENTATION_DELETE";
    WorkRequestOperationType["MlApplicationImplementationUpdate"] = "ML_APPLICATION_IMPLEMENTATION_UPDATE";
    WorkRequestOperationType["MlApplicationImplementationMove"] = "ML_APPLICATION_IMPLEMENTATION_MOVE";
    WorkRequestOperationType["MlApplicationInstanceCreate"] = "ML_APPLICATION_INSTANCE_CREATE";
    WorkRequestOperationType["MlApplicationInstanceUpdate"] = "ML_APPLICATION_INSTANCE_UPDATE";
    WorkRequestOperationType["MlApplicationInstanceDelete"] = "ML_APPLICATION_INSTANCE_DELETE";
    WorkRequestOperationType["MlApplicationInstanceMove"] = "ML_APPLICATION_INSTANCE_MOVE";
    WorkRequestOperationType["MlApplicationInstanceViewCreate"] = "ML_APPLICATION_INSTANCE_VIEW_CREATE";
    WorkRequestOperationType["MlApplicationInstanceViewUpdate"] = "ML_APPLICATION_INSTANCE_VIEW_UPDATE";
    WorkRequestOperationType["MlApplicationInstanceViewDelete"] = "ML_APPLICATION_INSTANCE_VIEW_DELETE";
    WorkRequestOperationType["MlApplicationInstanceViewUpgrade"] = "ML_APPLICATION_INSTANCE_VIEW_UPGRADE";
    WorkRequestOperationType["MlApplicationInstanceViewMove"] = "ML_APPLICATION_INSTANCE_VIEW_MOVE";
    WorkRequestOperationType["PrivateEndpointCreate"] = "PRIVATE_ENDPOINT_CREATE";
    WorkRequestOperationType["PrivateEndpointDelete"] = "PRIVATE_ENDPOINT_DELETE";
    WorkRequestOperationType["PrivateEndpointMove"] = "PRIVATE_ENDPOINT_MOVE";
    WorkRequestOperationType["PrivateEndpointUpdate"] = "PRIVATE_ENDPOINT_UPDATE";
    WorkRequestOperationType["ScheduleCreate"] = "SCHEDULE_CREATE";
    WorkRequestOperationType["ScheduleUpdate"] = "SCHEDULE_UPDATE";
    WorkRequestOperationType["ScheduleDelete"] = "SCHEDULE_DELETE";
    WorkRequestOperationType["ScheduleMove"] = "SCHEDULE_MOVE";
    WorkRequestOperationType["ScheduleActivate"] = "SCHEDULE_ACTIVATE";
    WorkRequestOperationType["ScheduleDeactivate"] = "SCHEDULE_DEACTIVATE";
    WorkRequestOperationType["RegisterModelArtifact"] = "REGISTER_MODEL_ARTIFACT";
    WorkRequestOperationType["RestoreArchivedModel"] = "RESTORE_ARCHIVED_MODEL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestOperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestOperationType = exports.WorkRequestOperationType || (exports.WorkRequestOperationType = {}));
(function (WorkRequestOperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    WorkRequestOperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WorkRequestOperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkRequestOperationType = exports.WorkRequestOperationType || (exports.WorkRequestOperationType = {}));
