// oci-apiaccesscontrol/lib/model/index.js
"use strict";
/**
 * Oracle API Access Control
 * This service is used to restrict the control plane service apis; so that everybody won't be
able to access those apis.
There are two main resouces defined as a part of this service
1. PrivilegedApiControl: This is created by the customer which defines which service apis are
   controlled and who can access it.
2. PrivilegedApiRequest: This is a request object again created by the customer operators who           seek access to those privileged apis. After a request is obtained based on the                       PrivilegedAccessControl for which the api belongs to, either it can be approved so that the          requested person can execute the service apis or it will wait for the customer to approve it.

 * OpenAPI spec version: 20241130
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdatePrivilegedApiControlDetails = exports.SortOrder = exports.RevokePrivilegedApiRequestDetails = exports.RejectPrivilegedApiRequestDetails = exports.PrivilegedApiRequestSummary = exports.PrivilegedApiRequestSeverity = exports.PrivilegedApiRequestOperationDetails = exports.PrivilegedApiRequestCollection = exports.PrivilegedApiRequest = exports.PrivilegedApiDetails = exports.PrivilegedApiControlSummary = exports.PrivilegedApiControlCollection = exports.PrivilegedApiControl = exports.OperationType = exports.OperationStatus = exports.CreatePrivilegedApiRequestDetails = exports.CreatePrivilegedApiControlDetails = exports.ClosePrivilegedApiRequestDetails = exports.ChangePrivilegedApiControlCompartmentDetails = exports.ApproverDetail = exports.ApprovePrivilegedApiRequestDetails = exports.ApiMetadataSummary = exports.ApiMetadataCollection = exports.ApiMetadataByEntityTypeSummary = exports.ApiMetadataByEntityTypeCollection = exports.ApiMetadata = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ApiMetadata = __importStar(require("./api-metadata"));
exports.ApiMetadata = ApiMetadata.ApiMetadata;
const ApiMetadataByEntityTypeCollection = __importStar(require("./api-metadata-by-entity-type-collection"));
exports.ApiMetadataByEntityTypeCollection = ApiMetadataByEntityTypeCollection.ApiMetadataByEntityTypeCollection;
const ApiMetadataByEntityTypeSummary = __importStar(require("./api-metadata-by-entity-type-summary"));
exports.ApiMetadataByEntityTypeSummary = ApiMetadataByEntityTypeSummary.ApiMetadataByEntityTypeSummary;
const ApiMetadataCollection = __importStar(require("./api-metadata-collection"));
exports.ApiMetadataCollection = ApiMetadataCollection.ApiMetadataCollection;
const ApiMetadataSummary = __importStar(require("./api-metadata-summary"));
exports.ApiMetadataSummary = ApiMetadataSummary.ApiMetadataSummary;
const ApprovePrivilegedApiRequestDetails = __importStar(require("./approve-privileged-api-request-details"));
exports.ApprovePrivilegedApiRequestDetails = ApprovePrivilegedApiRequestDetails.ApprovePrivilegedApiRequestDetails;
const ApproverDetail = __importStar(require("./approver-detail"));
exports.ApproverDetail = ApproverDetail.ApproverDetail;
const ChangePrivilegedApiControlCompartmentDetails = __importStar(require("./change-privileged-api-control-compartment-details"));
exports.ChangePrivilegedApiControlCompartmentDetails = ChangePrivilegedApiControlCompartmentDetails.ChangePrivilegedApiControlCompartmentDetails;
const ClosePrivilegedApiRequestDetails = __importStar(require("./close-privileged-api-request-details"));
exports.ClosePrivilegedApiRequestDetails = ClosePrivilegedApiRequestDetails.ClosePrivilegedApiRequestDetails;
const CreatePrivilegedApiControlDetails = __importStar(require("./create-privileged-api-control-details"));
exports.CreatePrivilegedApiControlDetails = CreatePrivilegedApiControlDetails.CreatePrivilegedApiControlDetails;
const CreatePrivilegedApiRequestDetails = __importStar(require("./create-privileged-api-request-details"));
exports.CreatePrivilegedApiRequestDetails = CreatePrivilegedApiRequestDetails.CreatePrivilegedApiRequestDetails;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PrivilegedApiControl = __importStar(require("./privileged-api-control"));
exports.PrivilegedApiControl = PrivilegedApiControl.PrivilegedApiControl;
const PrivilegedApiControlCollection = __importStar(require("./privileged-api-control-collection"));
exports.PrivilegedApiControlCollection = PrivilegedApiControlCollection.PrivilegedApiControlCollection;
const PrivilegedApiControlSummary = __importStar(require("./privileged-api-control-summary"));
exports.PrivilegedApiControlSummary = PrivilegedApiControlSummary.PrivilegedApiControlSummary;
const PrivilegedApiDetails = __importStar(require("./privileged-api-details"));
exports.PrivilegedApiDetails = PrivilegedApiDetails.PrivilegedApiDetails;
const PrivilegedApiRequest = __importStar(require("./privileged-api-request"));
exports.PrivilegedApiRequest = PrivilegedApiRequest.PrivilegedApiRequest;
const PrivilegedApiRequestCollection = __importStar(require("./privileged-api-request-collection"));
exports.PrivilegedApiRequestCollection = PrivilegedApiRequestCollection.PrivilegedApiRequestCollection;
const PrivilegedApiRequestOperationDetails = __importStar(require("./privileged-api-request-operation-details"));
exports.PrivilegedApiRequestOperationDetails = PrivilegedApiRequestOperationDetails.PrivilegedApiRequestOperationDetails;
const PrivilegedApiRequestSeverity = __importStar(require("./privileged-api-request-severity"));
exports.PrivilegedApiRequestSeverity = PrivilegedApiRequestSeverity.PrivilegedApiRequestSeverity;
const PrivilegedApiRequestSummary = __importStar(require("./privileged-api-request-summary"));
exports.PrivilegedApiRequestSummary = PrivilegedApiRequestSummary.PrivilegedApiRequestSummary;
const RejectPrivilegedApiRequestDetails = __importStar(require("./reject-privileged-api-request-details"));
exports.RejectPrivilegedApiRequestDetails = RejectPrivilegedApiRequestDetails.RejectPrivilegedApiRequestDetails;
const RevokePrivilegedApiRequestDetails = __importStar(require("./revoke-privileged-api-request-details"));
exports.RevokePrivilegedApiRequestDetails = RevokePrivilegedApiRequestDetails.RevokePrivilegedApiRequestDetails;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdatePrivilegedApiControlDetails = __importStar(require("./update-privileged-api-control-details"));
exports.UpdatePrivilegedApiControlDetails = UpdatePrivilegedApiControlDetails.UpdatePrivilegedApiControlDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
