// oci-apiaccesscontrol/lib/model/operation-type.js
"use strict";
/**
 * Oracle API Access Control
 * This service is used to restrict the control plane service apis; so that everybody won't be
able to access those apis.
There are two main resouces defined as a part of this service
1. PrivilegedApiControl: This is created by the customer which defines which service apis are
   controlled and who can access it.
2. PrivilegedApiRequest: This is a request object again created by the customer operators who           seek access to those privileged apis. After a request is obtained based on the                       PrivilegedAccessControl for which the api belongs to, either it can be approved so that the          requested person can execute the service apis or it will wait for the customer to approve it.

 * OpenAPI spec version: 20241130
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * The asynchronous operation tracked by the work request.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreatePrivilegedApiControl"] = "CREATE_PRIVILEGED_API_CONTROL";
    OperationType["GetPrivilegedApiControl"] = "GET_PRIVILEGED_API_CONTROL";
    OperationType["ListPrivilegedApiControl"] = "LIST_PRIVILEGED_API_CONTROL";
    OperationType["UpdatePrivilegedApiControl"] = "UPDATE_PRIVILEGED_API_CONTROL";
    OperationType["DeletePrivilegedApiControl"] = "DELETE_PRIVILEGED_API_CONTROL";
    OperationType["MovePrivilegedApiControl"] = "MOVE_PRIVILEGED_API_CONTROL";
    OperationType["CreatePrivilegedApiRequest"] = "CREATE_PRIVILEGED_API_REQUEST";
    OperationType["GetPrivilegedApiRequest"] = "GET_PRIVILEGED_API_REQUEST";
    OperationType["ListPrivilegedApiRequest"] = "LIST_PRIVILEGED_API_REQUEST";
    OperationType["ApprovePrivilegedApiRequest"] = "APPROVE_PRIVILEGED_API_REQUEST";
    OperationType["RejectPrivilegedApiRequest"] = "REJECT_PRIVILEGED_API_REQUEST";
    OperationType["RevokePrivilegedApiRequest"] = "REVOKE_PRIVILEGED_API_REQUEST";
    OperationType["ClosePrivilegedApiRequest"] = "CLOSE_PRIVILEGED_API_REQUEST";
    OperationType["TimeoutPrivilegedApiRequest"] = "TIMEOUT_PRIVILEGED_API_REQUEST";
    OperationType["CustomerApprovalCheckPrivilegedApiRequest"] = "CUSTOMER_APPROVAL_CHECK_PRIVILEGED_API_REQUEST";
    OperationType["CheckPrivilegedApiRequestStatus"] = "CHECK_PRIVILEGED_API_REQUEST_STATUS";
    OperationType["GetPrivilegedApiWorkRequest"] = "GET_PRIVILEGED_API_WORK_REQUEST";
    OperationType["ListPrivilegedApiWorkRequest"] = "LIST_PRIVILEGED_API_WORK_REQUEST";
    OperationType["ListPrivilegedApiWorkRequestErrors"] = "LIST_PRIVILEGED_API_WORK_REQUEST_ERRORS";
    OperationType["ListPrivilegedApiWorkRequestLogs"] = "LIST_PRIVILEGED_API_WORK_REQUEST_LOGS";
    OperationType["CancelPrivilegedApiWorkRequest"] = "CANCEL_PRIVILEGED_API_WORK_REQUEST";
    OperationType["ListApiMetadata"] = "LIST_API_METADATA";
    OperationType["ListApiMetadataByEntityType"] = "LIST_API_METADATA_BY_ENTITY_TYPE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
