// oci-apiaccesscontrol/lib/model/privileged-api-request.js
"use strict";
/**
 * Oracle API Access Control
 * This service is used to restrict the control plane service apis; so that everybody won't be
able to access those apis.
There are two main resouces defined as a part of this service
1. PrivilegedApiControl: This is created by the customer which defines which service apis are
   controlled and who can access it.
2. PrivilegedApiRequest: This is a request object again created by the customer operators who           seek access to those privileged apis. After a request is obtained based on the                       PrivilegedAccessControl for which the api belongs to, either it can be approved so that the          requested person can execute the service apis or it will wait for the customer to approve it.

 * OpenAPI spec version: 20241130
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivilegedApiRequest = void 0;
const model = __importStar(require("../model"));
var PrivilegedApiRequest;
(function (PrivilegedApiRequest) {
    let State;
    (function (State) {
        State["Created"] = "CREATED";
        State["ApprovalWaiting"] = "APPROVAL_WAITING";
        State["Approved"] = "APPROVED";
        State["ApprovedForFuture"] = "APPROVED_FOR_FUTURE";
        State["Rejected"] = "REJECTED";
        State["Deployed"] = "DEPLOYED";
        State["DeployFailed"] = "DEPLOY_FAILED";
        State["CloseFailed"] = "CLOSE_FAILED";
        State["RevokeFailed"] = "REVOKE_FAILED";
        State["ExpiryFailed"] = "EXPIRY_FAILED";
        State["Revoking"] = "REVOKING";
        State["Revoked"] = "REVOKED";
        State["Closing"] = "CLOSING";
        State["Closed"] = "CLOSED";
        State["Expired"] = "EXPIRED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        State["UnknownValue"] = "UNKNOWN_VALUE";
    })(State = PrivilegedApiRequest.State || (PrivilegedApiRequest.State = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Accepted"] = "ACCEPTED";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        LifecycleState["Waiting"] = "WAITING";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Succeeded"] = "SUCCEEDED";
        LifecycleState["Canceling"] = "CANCELING";
        LifecycleState["Canceled"] = "CANCELED";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = PrivilegedApiRequest.LifecycleState || (PrivilegedApiRequest.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "privilegedOperationList": obj.privilegedOperationList
                ? obj.privilegedOperationList.map(item => {
                    return model.PrivilegedApiRequestOperationDetails.getJsonObj(item);
                })
                : undefined,
            "approverDetails": obj.approverDetails
                ? obj.approverDetails.map(item => {
                    return model.ApproverDetail.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    PrivilegedApiRequest.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "privilegedOperationList": obj.privilegedOperationList
                ? obj.privilegedOperationList.map(item => {
                    return model.PrivilegedApiRequestOperationDetails.getDeserializedJsonObj(item);
                })
                : undefined,
            "approverDetails": obj.approverDetails
                ? obj.approverDetails.map(item => {
                    return model.ApproverDetail.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    PrivilegedApiRequest.getDeserializedJsonObj = getDeserializedJsonObj;
})(PrivilegedApiRequest = exports.PrivilegedApiRequest || (exports.PrivilegedApiRequest = {}));
