// oci-email/lib/model/domain-verification-status-type.js
"use strict";
/**
 * Email Delivery API
 * Use the Email Delivery API to do the necessary set up to send high-volume and application-generated emails through the OCI Email Delivery service.
For more information, see [Overview of the Email Delivery Service](https://docs.oracle.com/iaas/Content/Email/Concepts/overview.htm).

 **Note:** Write actions (POST, UPDATE, DELETE) may take several minutes to propagate and be reflected by the API.
 If a subsequent read request fails to reflect your changes, wait a few minutes and try again.

 * OpenAPI spec version: 20170907
 * Contact: email-dev_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainVerificationStatusType = void 0;
/**
 * The current domain verification status.
 **/
var DomainVerificationStatusType;
(function (DomainVerificationStatusType) {
    DomainVerificationStatusType["None"] = "NONE";
    DomainVerificationStatusType["Domainid"] = "DOMAINID";
    DomainVerificationStatusType["Other"] = "OTHER";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DomainVerificationStatusType["UnknownValue"] = "UNKNOWN_VALUE";
})(DomainVerificationStatusType = exports.DomainVerificationStatusType || (exports.DomainVerificationStatusType = {}));
(function (DomainVerificationStatusType) {
    function getJsonObj(obj) {
        return obj;
    }
    DomainVerificationStatusType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DomainVerificationStatusType.getDeserializedJsonObj = getDeserializedJsonObj;
})(DomainVerificationStatusType = exports.DomainVerificationStatusType || (exports.DomainVerificationStatusType = {}));
