// oci-email/lib/model/suppression-reason.js
"use strict";
/**
 * Email Delivery API
 * Use the Email Delivery API to do the necessary set up to send high-volume and application-generated emails through the OCI Email Delivery service.
For more information, see [Overview of the Email Delivery Service](https://docs.oracle.com/iaas/Content/Email/Concepts/overview.htm).

 **Note:** Write actions (POST, UPDATE, DELETE) may take several minutes to propagate and be reflected by the API.
 If a subsequent read request fails to reflect your changes, wait a few minutes and try again.

 * OpenAPI spec version: 20170907
 * Contact: email-dev_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SuppressionReason = void 0;
/**
 * The reason that the email address was suppressed. For more information
 * on the types of bounces, see [Suppression List](https://docs.oracle.com/iaas/Content/Email/Concepts/overview.htm#components).
 *
 **/
var SuppressionReason;
(function (SuppressionReason) {
    SuppressionReason["Unknown"] = "UNKNOWN";
    SuppressionReason["Hardbounce"] = "HARDBOUNCE";
    SuppressionReason["Complaint"] = "COMPLAINT";
    SuppressionReason["Manual"] = "MANUAL";
    SuppressionReason["Softbounce"] = "SOFTBOUNCE";
    SuppressionReason["Unsubscribe"] = "UNSUBSCRIBE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SuppressionReason["UnknownValue"] = "UNKNOWN_VALUE";
})(SuppressionReason = exports.SuppressionReason || (exports.SuppressionReason = {}));
(function (SuppressionReason) {
    function getJsonObj(obj) {
        return obj;
    }
    SuppressionReason.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SuppressionReason.getDeserializedJsonObj = getDeserializedJsonObj;
})(SuppressionReason = exports.SuppressionReason || (exports.SuppressionReason = {}));
