// oci-networkloadbalancer/lib/model/backend-operational-status.js
"use strict";
/**
 * NetworkLoadBalancer API
 * This describes the network load balancer API.
 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackendOperationalStatus = void 0;
var BackendOperationalStatus;
(function (BackendOperationalStatus) {
    let Status;
    (function (Status) {
        Status["Active"] = "ACTIVE";
        Status["Standby"] = "STANDBY";
        Status["Unknown"] = "UNKNOWN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = BackendOperationalStatus.Status || (BackendOperationalStatus.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BackendOperationalStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BackendOperationalStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(BackendOperationalStatus = exports.BackendOperationalStatus || (exports.BackendOperationalStatus = {}));
