// oci-networkloadbalancer/lib/model/dns-health-check-query-types.js
"use strict";
/**
 * NetworkLoadBalancer API
 * This describes the network load balancer API.
 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DnsHealthCheckQueryTypes = void 0;
/**
 * Query type values for the DNS health check probe.
 **/
var DnsHealthCheckQueryTypes;
(function (DnsHealthCheckQueryTypes) {
    DnsHealthCheckQueryTypes["A"] = "A";
    DnsHealthCheckQueryTypes["Txt"] = "TXT";
    DnsHealthCheckQueryTypes["Aaaa"] = "AAAA";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DnsHealthCheckQueryTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(DnsHealthCheckQueryTypes = exports.DnsHealthCheckQueryTypes || (exports.DnsHealthCheckQueryTypes = {}));
(function (DnsHealthCheckQueryTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    DnsHealthCheckQueryTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DnsHealthCheckQueryTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(DnsHealthCheckQueryTypes = exports.DnsHealthCheckQueryTypes || (exports.DnsHealthCheckQueryTypes = {}));
