// oci-networkloadbalancer/lib/model/dns-health-check-rcodes.js
"use strict";
/**
 * NetworkLoadBalancer API
 * This describes the network load balancer API.
 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DnsHealthCheckRCodes = void 0;
/**
 * Return code value to validate the DNS response in healthcheck.
 **/
var DnsHealthCheckRCodes;
(function (DnsHealthCheckRCodes) {
    DnsHealthCheckRCodes["Noerror"] = "NOERROR";
    DnsHealthCheckRCodes["Servfail"] = "SERVFAIL";
    DnsHealthCheckRCodes["Nxdomain"] = "NXDOMAIN";
    DnsHealthCheckRCodes["Refused"] = "REFUSED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DnsHealthCheckRCodes["UnknownValue"] = "UNKNOWN_VALUE";
})(DnsHealthCheckRCodes = exports.DnsHealthCheckRCodes || (exports.DnsHealthCheckRCodes = {}));
(function (DnsHealthCheckRCodes) {
    function getJsonObj(obj) {
        return obj;
    }
    DnsHealthCheckRCodes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DnsHealthCheckRCodes.getDeserializedJsonObj = getDeserializedJsonObj;
})(DnsHealthCheckRCodes = exports.DnsHealthCheckRCodes || (exports.DnsHealthCheckRCodes = {}));
