// oci-rover/lib/model/cert-key-algorithm.js
"use strict";
/**
 * RoverCloudService API
 * A description of the RoverCloudService API.
 * OpenAPI spec version: 20201210
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertKeyAlgorithm = void 0;
/**
 * Possible key algorithms for leaf certificate.
 **/
var CertKeyAlgorithm;
(function (CertKeyAlgorithm) {
    CertKeyAlgorithm["Rsa2048"] = "RSA2048";
    CertKeyAlgorithm["Rsa4096"] = "RSA4096";
    CertKeyAlgorithm["EcdsaP256"] = "ECDSA_P256";
    CertKeyAlgorithm["EcdsaP384"] = "ECDSA_P384";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CertKeyAlgorithm["UnknownValue"] = "UNKNOWN_VALUE";
})(CertKeyAlgorithm = exports.CertKeyAlgorithm || (exports.CertKeyAlgorithm = {}));
(function (CertKeyAlgorithm) {
    function getJsonObj(obj) {
        return obj;
    }
    CertKeyAlgorithm.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CertKeyAlgorithm.getDeserializedJsonObj = getDeserializedJsonObj;
})(CertKeyAlgorithm = exports.CertKeyAlgorithm || (exports.CertKeyAlgorithm = {}));
