// oci-rover/lib/model/cert-signature-algorithm.js
"use strict";
/**
 * RoverCloudService API
 * A description of the RoverCloudService API.
 * OpenAPI spec version: 20201210
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertSignatureAlgorithm = void 0;
/**
 * Possible signature algorithms for leaf certificate.
 **/
var CertSignatureAlgorithm;
(function (CertSignatureAlgorithm) {
    CertSignatureAlgorithm["Sha256WithRsa"] = "SHA256_WITH_RSA";
    CertSignatureAlgorithm["Sha384WithRsa"] = "SHA384_WITH_RSA";
    CertSignatureAlgorithm["Sha512WithRsa"] = "SHA512_WITH_RSA";
    CertSignatureAlgorithm["Sha256WithEcdsa"] = "SHA256_WITH_ECDSA";
    CertSignatureAlgorithm["Sha384WithEcdsa"] = "SHA384_WITH_ECDSA";
    CertSignatureAlgorithm["Sha512WithEcdsa"] = "SHA512_WITH_ECDSA";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CertSignatureAlgorithm["UnknownValue"] = "UNKNOWN_VALUE";
})(CertSignatureAlgorithm = exports.CertSignatureAlgorithm || (exports.CertSignatureAlgorithm = {}));
(function (CertSignatureAlgorithm) {
    function getJsonObj(obj) {
        return obj;
    }
    CertSignatureAlgorithm.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CertSignatureAlgorithm.getDeserializedJsonObj = getDeserializedJsonObj;
})(CertSignatureAlgorithm = exports.CertSignatureAlgorithm || (exports.CertSignatureAlgorithm = {}));
