// oci-rover/lib/model/work-request-resource-action-type.js
"use strict";
/**
 * RoverCloudService API
 * A description of the RoverCloudService API.
 * OpenAPI spec version: 20201210
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestResourceActionType = void 0;
/**
 * Possible types of actions on roverCluster resource.
 **/
var WorkRequestResourceActionType;
(function (WorkRequestResourceActionType) {
    WorkRequestResourceActionType["InProgress"] = "IN_PROGRESS";
    WorkRequestResourceActionType["Failed"] = "FAILED";
    WorkRequestResourceActionType["Created"] = "CREATED";
    WorkRequestResourceActionType["Updated"] = "UPDATED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestResourceActionType["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestResourceActionType = exports.WorkRequestResourceActionType || (exports.WorkRequestResourceActionType = {}));
(function (WorkRequestResourceActionType) {
    function getJsonObj(obj) {
        return obj;
    }
    WorkRequestResourceActionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WorkRequestResourceActionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkRequestResourceActionType = exports.WorkRequestResourceActionType || (exports.WorkRequestResourceActionType = {}));
