// oci-marketplaceprivateoffer/lib/model/index.js
"use strict";
/**
 * MarketplacePrivateOffer API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UploadData = exports.UpdateOfferDetails = exports.SortOrder = exports.SellerInformation = exports.ResourceBundle = exports.PublisherSummary = exports.Pricing = exports.PackageType = exports.OfferSummary = exports.OfferInternalDetail = exports.OfferCollection = exports.Offer = exports.CustomField = exports.CreateOfferDetails = exports.CreateAttachmentDetails = exports.Contact = exports.BuyerInformation = exports.AttachmentSummary = exports.AttachmentCollection = exports.Attachment = void 0;
const Attachment = __importStar(require("./attachment"));
exports.Attachment = Attachment.Attachment;
const AttachmentCollection = __importStar(require("./attachment-collection"));
exports.AttachmentCollection = AttachmentCollection.AttachmentCollection;
const AttachmentSummary = __importStar(require("./attachment-summary"));
exports.AttachmentSummary = AttachmentSummary.AttachmentSummary;
const BuyerInformation = __importStar(require("./buyer-information"));
exports.BuyerInformation = BuyerInformation.BuyerInformation;
const Contact = __importStar(require("./contact"));
exports.Contact = Contact.Contact;
const CreateAttachmentDetails = __importStar(require("./create-attachment-details"));
exports.CreateAttachmentDetails = CreateAttachmentDetails.CreateAttachmentDetails;
const CreateOfferDetails = __importStar(require("./create-offer-details"));
exports.CreateOfferDetails = CreateOfferDetails.CreateOfferDetails;
const CustomField = __importStar(require("./custom-field"));
exports.CustomField = CustomField.CustomField;
const Offer = __importStar(require("./offer"));
exports.Offer = Offer.Offer;
const OfferCollection = __importStar(require("./offer-collection"));
exports.OfferCollection = OfferCollection.OfferCollection;
const OfferInternalDetail = __importStar(require("./offer-internal-detail"));
exports.OfferInternalDetail = OfferInternalDetail.OfferInternalDetail;
const OfferSummary = __importStar(require("./offer-summary"));
exports.OfferSummary = OfferSummary.OfferSummary;
const PackageType = __importStar(require("./package-type"));
exports.PackageType = PackageType.PackageType;
const Pricing = __importStar(require("./pricing"));
exports.Pricing = Pricing.Pricing;
const PublisherSummary = __importStar(require("./publisher-summary"));
exports.PublisherSummary = PublisherSummary.PublisherSummary;
const ResourceBundle = __importStar(require("./resource-bundle"));
exports.ResourceBundle = ResourceBundle.ResourceBundle;
const SellerInformation = __importStar(require("./seller-information"));
exports.SellerInformation = SellerInformation.SellerInformation;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateOfferDetails = __importStar(require("./update-offer-details"));
exports.UpdateOfferDetails = UpdateOfferDetails.UpdateOfferDetails;
const UploadData = __importStar(require("./upload-data"));
exports.UploadData = UploadData.UploadData;
