// oci-marketplaceprivateoffer/lib/model/package-type.js
"use strict";
/**
 * MarketplacePrivateOffer API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackageType = void 0;
/**
 * Possible values for a package type.
 **/
var PackageType;
(function (PackageType) {
    PackageType["ContainerImage"] = "CONTAINER_IMAGE";
    PackageType["HelmChart"] = "HELM_CHART";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PackageType["UnknownValue"] = "UNKNOWN_VALUE";
})(PackageType = exports.PackageType || (exports.PackageType = {}));
(function (PackageType) {
    function getJsonObj(obj) {
        return obj;
    }
    PackageType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PackageType.getDeserializedJsonObj = getDeserializedJsonObj;
})(PackageType = exports.PackageType || (exports.PackageType = {}));
