// oci-keymanagement/lib/model/index.js
"use strict";
/**
 * Vault Key Management API
 * Use the Key Management API to manage vaults and keys. For more information, see [Managing Vaults](https://docs.oracle.com/iaas/Content/KeyManagement/Tasks/managingvaults.htm) and [Managing Keys](https://docs.oracle.com/iaas/Content/KeyManagement/Tasks/managingkeys.htm).

 * OpenAPI spec version: release
 * Contact: sparta_kms_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationStatusDetails = exports.ReplicaVaultMetadata = exports.ReplicaDetails = exports.PreCoUserCredentials = exports.PortInformation = exports.OauthMetadataSummary = exports.OauthMetadata = exports.KeyVersionSummary = exports.KeyVersionReplicaDetails = exports.KeyVersion = exports.KeySummary = exports.KeyShape = exports.KeyReplicaDetails = exports.Key = exports.ImportKeyVersionDetails = exports.ImportKeyDetails = exports.HsmPartitionSummary = exports.HsmPartitionCollection = exports.HsmPartition = exports.HsmClusterSummary = exports.HsmClusterCollection = exports.HsmCluster = exports.GeneratedKey = exports.GenerateKeyDetails = exports.ExternalKeyVersionReference = exports.ExternalKeyReferenceDetails = exports.ExternalKeyReference = exports.ExternalKeyManagerMetadataSummary = exports.ExternalKeyManagerMetadata = exports.ExportedKeyData = exports.ExportKeyDetails = exports.EncryptedData = exports.EncryptDataDetails = exports.EkmsPrivateEndpointSummary = exports.EkmsPrivateEndpoint = exports.DeleteVaultReplicaDetails = exports.DecryptedData = exports.DecryptDataDetails = exports.CreateVaultReplicaDetails = exports.CreateVaultDetails = exports.CreateKeyDetails = exports.CreateHsmClusterDetails = exports.CreateEkmsPrivateEndpointDetails = exports.ChangeVaultCompartmentDetails = exports.ChangeKeyCompartmentDetails = exports.ChangeHsmClusterCompartmentDetails = exports.BackupVaultDetails = exports.BackupLocation = exports.BackupKeyDetails = exports.AutoKeyRotationDetails = void 0;
exports.ReplicaExternalVaultMetadata = exports.BackupLocationURI = exports.BackupLocationBucket = exports.WrappingKey = exports.WrappedImportKey = exports.VerifyDataDetails = exports.VerifiedData = exports.VaultUsage = exports.VaultSummary = exports.VaultReplicaSummary = exports.VaultReplicaDetails = exports.Vault = exports.UploadPartitionCertificatesDetails = exports.UpdateVaultDetails = exports.UpdateKeyDetails = exports.UpdateHsmClusterDetails = exports.UpdateEkmsPrivateEndpointDetails = exports.SignedData = exports.SignDataDetails = exports.ScheduleVaultDeletionDetails = exports.ScheduleKeyVersionDeletionDetails = exports.ScheduleKeyDeletionDetails = exports.ScheduleHsmClusterDeletionDetails = exports.RestoreVaultFromObjectStoreDetails = exports.RestoreKeyFromObjectStoreDetails = void 0;
const AutoKeyRotationDetails = __importStar(require("./auto-key-rotation-details"));
exports.AutoKeyRotationDetails = AutoKeyRotationDetails.AutoKeyRotationDetails;
const BackupKeyDetails = __importStar(require("./backup-key-details"));
exports.BackupKeyDetails = BackupKeyDetails.BackupKeyDetails;
const BackupLocation = __importStar(require("./backup-location"));
exports.BackupLocation = BackupLocation.BackupLocation;
const BackupVaultDetails = __importStar(require("./backup-vault-details"));
exports.BackupVaultDetails = BackupVaultDetails.BackupVaultDetails;
const ChangeHsmClusterCompartmentDetails = __importStar(require("./change-hsm-cluster-compartment-details"));
exports.ChangeHsmClusterCompartmentDetails = ChangeHsmClusterCompartmentDetails.ChangeHsmClusterCompartmentDetails;
const ChangeKeyCompartmentDetails = __importStar(require("./change-key-compartment-details"));
exports.ChangeKeyCompartmentDetails = ChangeKeyCompartmentDetails.ChangeKeyCompartmentDetails;
const ChangeVaultCompartmentDetails = __importStar(require("./change-vault-compartment-details"));
exports.ChangeVaultCompartmentDetails = ChangeVaultCompartmentDetails.ChangeVaultCompartmentDetails;
const CreateEkmsPrivateEndpointDetails = __importStar(require("./create-ekms-private-endpoint-details"));
exports.CreateEkmsPrivateEndpointDetails = CreateEkmsPrivateEndpointDetails.CreateEkmsPrivateEndpointDetails;
const CreateHsmClusterDetails = __importStar(require("./create-hsm-cluster-details"));
exports.CreateHsmClusterDetails = CreateHsmClusterDetails.CreateHsmClusterDetails;
const CreateKeyDetails = __importStar(require("./create-key-details"));
exports.CreateKeyDetails = CreateKeyDetails.CreateKeyDetails;
const CreateVaultDetails = __importStar(require("./create-vault-details"));
exports.CreateVaultDetails = CreateVaultDetails.CreateVaultDetails;
const CreateVaultReplicaDetails = __importStar(require("./create-vault-replica-details"));
exports.CreateVaultReplicaDetails = CreateVaultReplicaDetails.CreateVaultReplicaDetails;
const DecryptDataDetails = __importStar(require("./decrypt-data-details"));
exports.DecryptDataDetails = DecryptDataDetails.DecryptDataDetails;
const DecryptedData = __importStar(require("./decrypted-data"));
exports.DecryptedData = DecryptedData.DecryptedData;
const DeleteVaultReplicaDetails = __importStar(require("./delete-vault-replica-details"));
exports.DeleteVaultReplicaDetails = DeleteVaultReplicaDetails.DeleteVaultReplicaDetails;
const EkmsPrivateEndpoint = __importStar(require("./ekms-private-endpoint"));
exports.EkmsPrivateEndpoint = EkmsPrivateEndpoint.EkmsPrivateEndpoint;
const EkmsPrivateEndpointSummary = __importStar(require("./ekms-private-endpoint-summary"));
exports.EkmsPrivateEndpointSummary = EkmsPrivateEndpointSummary.EkmsPrivateEndpointSummary;
const EncryptDataDetails = __importStar(require("./encrypt-data-details"));
exports.EncryptDataDetails = EncryptDataDetails.EncryptDataDetails;
const EncryptedData = __importStar(require("./encrypted-data"));
exports.EncryptedData = EncryptedData.EncryptedData;
const ExportKeyDetails = __importStar(require("./export-key-details"));
exports.ExportKeyDetails = ExportKeyDetails.ExportKeyDetails;
const ExportedKeyData = __importStar(require("./exported-key-data"));
exports.ExportedKeyData = ExportedKeyData.ExportedKeyData;
const ExternalKeyManagerMetadata = __importStar(require("./external-key-manager-metadata"));
exports.ExternalKeyManagerMetadata = ExternalKeyManagerMetadata.ExternalKeyManagerMetadata;
const ExternalKeyManagerMetadataSummary = __importStar(require("./external-key-manager-metadata-summary"));
exports.ExternalKeyManagerMetadataSummary = ExternalKeyManagerMetadataSummary.ExternalKeyManagerMetadataSummary;
const ExternalKeyReference = __importStar(require("./external-key-reference"));
exports.ExternalKeyReference = ExternalKeyReference.ExternalKeyReference;
const ExternalKeyReferenceDetails = __importStar(require("./external-key-reference-details"));
exports.ExternalKeyReferenceDetails = ExternalKeyReferenceDetails.ExternalKeyReferenceDetails;
const ExternalKeyVersionReference = __importStar(require("./external-key-version-reference"));
exports.ExternalKeyVersionReference = ExternalKeyVersionReference.ExternalKeyVersionReference;
const GenerateKeyDetails = __importStar(require("./generate-key-details"));
exports.GenerateKeyDetails = GenerateKeyDetails.GenerateKeyDetails;
const GeneratedKey = __importStar(require("./generated-key"));
exports.GeneratedKey = GeneratedKey.GeneratedKey;
const HsmCluster = __importStar(require("./hsm-cluster"));
exports.HsmCluster = HsmCluster.HsmCluster;
const HsmClusterCollection = __importStar(require("./hsm-cluster-collection"));
exports.HsmClusterCollection = HsmClusterCollection.HsmClusterCollection;
const HsmClusterSummary = __importStar(require("./hsm-cluster-summary"));
exports.HsmClusterSummary = HsmClusterSummary.HsmClusterSummary;
const HsmPartition = __importStar(require("./hsm-partition"));
exports.HsmPartition = HsmPartition.HsmPartition;
const HsmPartitionCollection = __importStar(require("./hsm-partition-collection"));
exports.HsmPartitionCollection = HsmPartitionCollection.HsmPartitionCollection;
const HsmPartitionSummary = __importStar(require("./hsm-partition-summary"));
exports.HsmPartitionSummary = HsmPartitionSummary.HsmPartitionSummary;
const ImportKeyDetails = __importStar(require("./import-key-details"));
exports.ImportKeyDetails = ImportKeyDetails.ImportKeyDetails;
const ImportKeyVersionDetails = __importStar(require("./import-key-version-details"));
exports.ImportKeyVersionDetails = ImportKeyVersionDetails.ImportKeyVersionDetails;
const Key = __importStar(require("./key"));
exports.Key = Key.Key;
const KeyReplicaDetails = __importStar(require("./key-replica-details"));
exports.KeyReplicaDetails = KeyReplicaDetails.KeyReplicaDetails;
const KeyShape = __importStar(require("./key-shape"));
exports.KeyShape = KeyShape.KeyShape;
const KeySummary = __importStar(require("./key-summary"));
exports.KeySummary = KeySummary.KeySummary;
const KeyVersion = __importStar(require("./key-version"));
exports.KeyVersion = KeyVersion.KeyVersion;
const KeyVersionReplicaDetails = __importStar(require("./key-version-replica-details"));
exports.KeyVersionReplicaDetails = KeyVersionReplicaDetails.KeyVersionReplicaDetails;
const KeyVersionSummary = __importStar(require("./key-version-summary"));
exports.KeyVersionSummary = KeyVersionSummary.KeyVersionSummary;
const OauthMetadata = __importStar(require("./oauth-metadata"));
exports.OauthMetadata = OauthMetadata.OauthMetadata;
const OauthMetadataSummary = __importStar(require("./oauth-metadata-summary"));
exports.OauthMetadataSummary = OauthMetadataSummary.OauthMetadataSummary;
const PortInformation = __importStar(require("./port-information"));
exports.PortInformation = PortInformation.PortInformation;
const PreCoUserCredentials = __importStar(require("./pre-co-user-credentials"));
exports.PreCoUserCredentials = PreCoUserCredentials.PreCoUserCredentials;
const ReplicaDetails = __importStar(require("./replica-details"));
exports.ReplicaDetails = ReplicaDetails.ReplicaDetails;
const ReplicaVaultMetadata = __importStar(require("./replica-vault-metadata"));
exports.ReplicaVaultMetadata = ReplicaVaultMetadata.ReplicaVaultMetadata;
const ReplicationStatusDetails = __importStar(require("./replication-status-details"));
exports.ReplicationStatusDetails = ReplicationStatusDetails.ReplicationStatusDetails;
const RestoreKeyFromObjectStoreDetails = __importStar(require("./restore-key-from-object-store-details"));
exports.RestoreKeyFromObjectStoreDetails = RestoreKeyFromObjectStoreDetails.RestoreKeyFromObjectStoreDetails;
const RestoreVaultFromObjectStoreDetails = __importStar(require("./restore-vault-from-object-store-details"));
exports.RestoreVaultFromObjectStoreDetails = RestoreVaultFromObjectStoreDetails.RestoreVaultFromObjectStoreDetails;
const ScheduleHsmClusterDeletionDetails = __importStar(require("./schedule-hsm-cluster-deletion-details"));
exports.ScheduleHsmClusterDeletionDetails = ScheduleHsmClusterDeletionDetails.ScheduleHsmClusterDeletionDetails;
const ScheduleKeyDeletionDetails = __importStar(require("./schedule-key-deletion-details"));
exports.ScheduleKeyDeletionDetails = ScheduleKeyDeletionDetails.ScheduleKeyDeletionDetails;
const ScheduleKeyVersionDeletionDetails = __importStar(require("./schedule-key-version-deletion-details"));
exports.ScheduleKeyVersionDeletionDetails = ScheduleKeyVersionDeletionDetails.ScheduleKeyVersionDeletionDetails;
const ScheduleVaultDeletionDetails = __importStar(require("./schedule-vault-deletion-details"));
exports.ScheduleVaultDeletionDetails = ScheduleVaultDeletionDetails.ScheduleVaultDeletionDetails;
const SignDataDetails = __importStar(require("./sign-data-details"));
exports.SignDataDetails = SignDataDetails.SignDataDetails;
const SignedData = __importStar(require("./signed-data"));
exports.SignedData = SignedData.SignedData;
const UpdateEkmsPrivateEndpointDetails = __importStar(require("./update-ekms-private-endpoint-details"));
exports.UpdateEkmsPrivateEndpointDetails = UpdateEkmsPrivateEndpointDetails.UpdateEkmsPrivateEndpointDetails;
const UpdateHsmClusterDetails = __importStar(require("./update-hsm-cluster-details"));
exports.UpdateHsmClusterDetails = UpdateHsmClusterDetails.UpdateHsmClusterDetails;
const UpdateKeyDetails = __importStar(require("./update-key-details"));
exports.UpdateKeyDetails = UpdateKeyDetails.UpdateKeyDetails;
const UpdateVaultDetails = __importStar(require("./update-vault-details"));
exports.UpdateVaultDetails = UpdateVaultDetails.UpdateVaultDetails;
const UploadPartitionCertificatesDetails = __importStar(require("./upload-partition-certificates-details"));
exports.UploadPartitionCertificatesDetails = UploadPartitionCertificatesDetails.UploadPartitionCertificatesDetails;
const Vault = __importStar(require("./vault"));
exports.Vault = Vault.Vault;
const VaultReplicaDetails = __importStar(require("./vault-replica-details"));
exports.VaultReplicaDetails = VaultReplicaDetails.VaultReplicaDetails;
const VaultReplicaSummary = __importStar(require("./vault-replica-summary"));
exports.VaultReplicaSummary = VaultReplicaSummary.VaultReplicaSummary;
const VaultSummary = __importStar(require("./vault-summary"));
exports.VaultSummary = VaultSummary.VaultSummary;
const VaultUsage = __importStar(require("./vault-usage"));
exports.VaultUsage = VaultUsage.VaultUsage;
const VerifiedData = __importStar(require("./verified-data"));
exports.VerifiedData = VerifiedData.VerifiedData;
const VerifyDataDetails = __importStar(require("./verify-data-details"));
exports.VerifyDataDetails = VerifyDataDetails.VerifyDataDetails;
const WrappedImportKey = __importStar(require("./wrapped-import-key"));
exports.WrappedImportKey = WrappedImportKey.WrappedImportKey;
const WrappingKey = __importStar(require("./wrapping-key"));
exports.WrappingKey = WrappingKey.WrappingKey;
const BackupLocationBucket = __importStar(require("./backup-location-bucket"));
exports.BackupLocationBucket = BackupLocationBucket.BackupLocationBucket;
const BackupLocationURI = __importStar(require("./backup-location-ur-i"));
exports.BackupLocationURI = BackupLocationURI.BackupLocationURI;
const ReplicaExternalVaultMetadata = __importStar(require("./replica-external-vault-metadata"));
exports.ReplicaExternalVaultMetadata = ReplicaExternalVaultMetadata.ReplicaExternalVaultMetadata;
