// oci-keymanagement/lib/model/key-shape.js
"use strict";
/**
 * Vault Key Management API
 * Use the Key Management API to manage vaults and keys. For more information, see [Managing Vaults](https://docs.oracle.com/iaas/Content/KeyManagement/Tasks/managingvaults.htm) and [Managing Keys](https://docs.oracle.com/iaas/Content/KeyManagement/Tasks/managingkeys.htm).

 * OpenAPI spec version: release
 * Contact: sparta_kms_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeyShape = void 0;
var KeyShape;
(function (KeyShape) {
    let Algorithm;
    (function (Algorithm) {
        Algorithm["Aes"] = "AES";
        Algorithm["Rsa"] = "RSA";
        Algorithm["Ecdsa"] = "ECDSA";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Algorithm["UnknownValue"] = "UNKNOWN_VALUE";
    })(Algorithm = KeyShape.Algorithm || (KeyShape.Algorithm = {}));
    let CurveId;
    (function (CurveId) {
        CurveId["NistP256"] = "NIST_P256";
        CurveId["NistP384"] = "NIST_P384";
        CurveId["NistP521"] = "NIST_P521";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        CurveId["UnknownValue"] = "UNKNOWN_VALUE";
    })(CurveId = KeyShape.CurveId || (KeyShape.CurveId = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    KeyShape.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    KeyShape.getDeserializedJsonObj = getDeserializedJsonObj;
})(KeyShape = exports.KeyShape || (exports.KeyShape = {}));
