// oci-integration/lib/model/cross-region-integration-instance-details.js
"use strict";
/**
 * Oracle Integration API
 * Oracle Integration API.
 * OpenAPI spec version: 20190131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CrossRegionIntegrationInstanceDetails = void 0;
var CrossRegionIntegrationInstanceDetails;
(function (CrossRegionIntegrationInstanceDetails) {
    let Role;
    (function (Role) {
        Role["Primary"] = "PRIMARY";
        Role["Secondary"] = "SECONDARY";
        Role["Unknown"] = "UNKNOWN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Role["UnknownValue"] = "UNKNOWN_VALUE";
    })(Role = CrossRegionIntegrationInstanceDetails.Role || (CrossRegionIntegrationInstanceDetails.Role = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CrossRegionIntegrationInstanceDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CrossRegionIntegrationInstanceDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CrossRegionIntegrationInstanceDetails = exports.CrossRegionIntegrationInstanceDetails || (exports.CrossRegionIntegrationInstanceDetails = {}));
