// oci-integration/lib/model/extend-data-retention-details.js
"use strict";
/**
 * Oracle Integration API
 * Oracle Integration API.
 * OpenAPI spec version: 20190131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtendDataRetentionDetails = void 0;
var ExtendDataRetentionDetails;
(function (ExtendDataRetentionDetails) {
    let DataRetentionPeriod;
    (function (DataRetentionPeriod) {
        DataRetentionPeriod["Months1"] = "MONTHS_1";
        DataRetentionPeriod["Months3"] = "MONTHS_3";
        DataRetentionPeriod["Months6"] = "MONTHS_6";
    })(DataRetentionPeriod = ExtendDataRetentionDetails.DataRetentionPeriod || (ExtendDataRetentionDetails.DataRetentionPeriod = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExtendDataRetentionDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExtendDataRetentionDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExtendDataRetentionDetails = exports.ExtendDataRetentionDetails || (exports.ExtendDataRetentionDetails = {}));
