// oci-datasafe/lib/request/list-database-table-access-entries-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDatabaseTableAccessEntriesRequest = void 0;
var ListDatabaseTableAccessEntriesRequest;
(function (ListDatabaseTableAccessEntriesRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Key"] = "key";
        SortBy["Grantee"] = "grantee";
        SortBy["AccessType"] = "accessType";
        SortBy["TableSchema"] = "tableSchema";
        SortBy["TableName"] = "tableName";
        SortBy["PrivilegeType"] = "privilegeType";
        SortBy["Privilege"] = "privilege";
        SortBy["PrivilegeGrantable"] = "privilegeGrantable";
        SortBy["GrantFromRole"] = "grantFromRole";
        SortBy["AccessThroughObject"] = "accessThroughObject";
        SortBy["ColumnName"] = "columnName";
        SortBy["Grantor"] = "grantor";
        SortBy["AreAllTablesAccessible"] = "areAllTablesAccessible";
        SortBy["IsAccessConstrainedByView"] = "isAccessConstrainedByView";
        SortBy["IsAccessConstrainedByLabelSecurity"] = "isAccessConstrainedByLabelSecurity";
        SortBy["IsAccessConstrainedByDatabaseVault"] = "isAccessConstrainedByDatabaseVault";
        SortBy["IsAccessConstrainedByVirtualPrivateDatabase"] = "isAccessConstrainedByVirtualPrivateDatabase";
        SortBy["IsAccessConstrainedByRedaction"] = "isAccessConstrainedByRedaction";
        SortBy["IsAccessConstrainedByRealApplicationSecurity"] = "isAccessConstrainedByRealApplicationSecurity";
        SortBy["IsAccessConstrainedBySqlFirewall"] = "isAccessConstrainedBySqlFirewall";
        SortBy["IsSensitive"] = "isSensitive";
    })(SortBy = ListDatabaseTableAccessEntriesRequest.SortBy || (ListDatabaseTableAccessEntriesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListDatabaseTableAccessEntriesRequest.SortOrder || (ListDatabaseTableAccessEntriesRequest.SortOrder = {}));
})(ListDatabaseTableAccessEntriesRequest = exports.ListDatabaseTableAccessEntriesRequest || (exports.ListDatabaseTableAccessEntriesRequest = {}));
