// oci-datasafe/lib/request/list-findings-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFindingsRequest = void 0;
var ListFindingsRequest;
(function (ListFindingsRequest) {
    let Severity;
    (function (Severity) {
        Severity["High"] = "HIGH";
        Severity["Medium"] = "MEDIUM";
        Severity["Low"] = "LOW";
        Severity["Evaluate"] = "EVALUATE";
        Severity["Advisory"] = "ADVISORY";
        Severity["Pass"] = "PASS";
        Severity["Deferred"] = "DEFERRED";
    })(Severity = ListFindingsRequest.Severity || (ListFindingsRequest.Severity = {}));
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListFindingsRequest.AccessLevel || (ListFindingsRequest.AccessLevel = {}));
    let Field;
    (function (Field) {
        Field["Severity"] = "severity";
        Field["FindingKey"] = "findingKey";
        Field["Reference"] = "reference";
        Field["TargetId"] = "targetId";
        Field["IsTopFinding"] = "isTopFinding";
        Field["Title"] = "title";
        Field["Category"] = "category";
        Field["Remarks"] = "remarks";
        Field["Details"] = "details";
        Field["Summary"] = "summary";
        Field["IsRiskModified"] = "isRiskModified";
    })(Field = ListFindingsRequest.Field || (ListFindingsRequest.Field = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Category"] = "category";
        SortBy["FindingKey"] = "findingKey";
        SortBy["Severity"] = "severity";
    })(SortBy = ListFindingsRequest.SortBy || (ListFindingsRequest.SortBy = {}));
})(ListFindingsRequest = exports.ListFindingsRequest || (exports.ListFindingsRequest = {}));
