// oci-datasafe/lib/request/list-sql-firewall-violations-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSqlFirewallViolationsRequest = void 0;
var ListSqlFirewallViolationsRequest;
(function (ListSqlFirewallViolationsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListSqlFirewallViolationsRequest.AccessLevel || (ListSqlFirewallViolationsRequest.AccessLevel = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListSqlFirewallViolationsRequest.SortOrder || (ListSqlFirewallViolationsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["DbUserName"] = "dbUserName";
        SortBy["TargetId"] = "targetId";
        SortBy["TargetName"] = "targetName";
        SortBy["OperationTime"] = "operationTime";
        SortBy["TimeCollected"] = "timeCollected";
        SortBy["ClientOsUserName"] = "clientOsUserName";
        SortBy["Operation"] = "operation";
        SortBy["CurrentDbUserName"] = "currentDbUserName";
        SortBy["SqlLevel"] = "sqlLevel";
        SortBy["ClientIp"] = "clientIp";
        SortBy["ClientProgram"] = "clientProgram";
        SortBy["ViolationCause"] = "violationCause";
        SortBy["ViolationAction"] = "violationAction";
    })(SortBy = ListSqlFirewallViolationsRequest.SortBy || (ListSqlFirewallViolationsRequest.SortBy = {}));
})(ListSqlFirewallViolationsRequest = exports.ListSqlFirewallViolationsRequest || (exports.ListSqlFirewallViolationsRequest = {}));
