// oci-datasafe/lib/request/list-masking-errors-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMaskingErrorsRequest = void 0;
var ListMaskingErrorsRequest;
(function (ListMaskingErrorsRequest) {
    let StepName;
    (function (StepName) {
        StepName["ExecuteMasking"] = "EXECUTE_MASKING";
        StepName["PreMasking"] = "PRE_MASKING";
        StepName["PostMasking"] = "POST_MASKING";
    })(StepName = ListMaskingErrorsRequest.StepName || (ListMaskingErrorsRequest.StepName = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["StepName"] = "stepName";
        SortBy["TimeCreated"] = "timeCreated";
    })(SortBy = ListMaskingErrorsRequest.SortBy || (ListMaskingErrorsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListMaskingErrorsRequest.SortOrder || (ListMaskingErrorsRequest.SortOrder = {}));
})(ListMaskingErrorsRequest = exports.ListMaskingErrorsRequest || (exports.ListMaskingErrorsRequest = {}));
