// oci-datasafe/lib/request/list-masking-policy-referential-relations-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMaskingPolicyReferentialRelationsRequest = void 0;
var ListMaskingPolicyReferentialRelationsRequest;
(function (ListMaskingPolicyReferentialRelationsRequest) {
    let RelationType;
    (function (RelationType) {
        RelationType["None"] = "NONE";
        RelationType["AppDefined"] = "APP_DEFINED";
        RelationType["DbDefined"] = "DB_DEFINED";
    })(RelationType = ListMaskingPolicyReferentialRelationsRequest.RelationType || (ListMaskingPolicyReferentialRelationsRequest.RelationType = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListMaskingPolicyReferentialRelationsRequest.SortOrder || (ListMaskingPolicyReferentialRelationsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["RelationType"] = "relationType";
        SortBy["SchemaName"] = "schemaName";
        SortBy["TableName"] = "tableName";
        SortBy["ColumnName"] = "columnName";
    })(SortBy = ListMaskingPolicyReferentialRelationsRequest.SortBy || (ListMaskingPolicyReferentialRelationsRequest.SortBy = {}));
})(ListMaskingPolicyReferentialRelationsRequest = exports.ListMaskingPolicyReferentialRelationsRequest || (exports.ListMaskingPolicyReferentialRelationsRequest = {}));
