// oci-datasafe/lib/request/list-masking-policy-health-report-logs-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMaskingPolicyHealthReportLogsRequest = void 0;
var ListMaskingPolicyHealthReportLogsRequest;
(function (ListMaskingPolicyHealthReportLogsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListMaskingPolicyHealthReportLogsRequest.SortOrder || (ListMaskingPolicyHealthReportLogsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["LogType"] = "logType";
    })(SortBy = ListMaskingPolicyHealthReportLogsRequest.SortBy || (ListMaskingPolicyHealthReportLogsRequest.SortBy = {}));
    let MessageType;
    (function (MessageType) {
        MessageType["Pass"] = "PASS";
        MessageType["Warning"] = "WARNING";
        MessageType["Error"] = "ERROR";
    })(MessageType = ListMaskingPolicyHealthReportLogsRequest.MessageType || (ListMaskingPolicyHealthReportLogsRequest.MessageType = {}));
})(ListMaskingPolicyHealthReportLogsRequest = exports.ListMaskingPolicyHealthReportLogsRequest || (exports.ListMaskingPolicyHealthReportLogsRequest = {}));
