// oci-datasafe/lib/request/list-findings-change-audit-logs-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFindingsChangeAuditLogsRequest = void 0;
var ListFindingsChangeAuditLogsRequest;
(function (ListFindingsChangeAuditLogsRequest) {
    let Severity;
    (function (Severity) {
        Severity["High"] = "HIGH";
        Severity["Medium"] = "MEDIUM";
        Severity["Low"] = "LOW";
        Severity["Evaluate"] = "EVALUATE";
        Severity["Advisory"] = "ADVISORY";
        Severity["Pass"] = "PASS";
        Severity["Deferred"] = "DEFERRED";
    })(Severity = ListFindingsChangeAuditLogsRequest.Severity || (ListFindingsChangeAuditLogsRequest.Severity = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListFindingsChangeAuditLogsRequest.SortOrder || (ListFindingsChangeAuditLogsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeUpdated"] = "timeUpdated";
        SortBy["ModifiedBy"] = "modifiedBy";
        SortBy["IsRiskDeferred"] = "isRiskDeferred";
        SortBy["TimeValidUntil"] = "timeValidUntil";
    })(SortBy = ListFindingsChangeAuditLogsRequest.SortBy || (ListFindingsChangeAuditLogsRequest.SortBy = {}));
})(ListFindingsChangeAuditLogsRequest = exports.ListFindingsChangeAuditLogsRequest || (exports.ListFindingsChangeAuditLogsRequest = {}));
