// oci-datasafe/lib/request/list-sql-firewall-violation-analytics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSqlFirewallViolationAnalyticsRequest = void 0;
var ListSqlFirewallViolationAnalyticsRequest;
(function (ListSqlFirewallViolationAnalyticsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListSqlFirewallViolationAnalyticsRequest.AccessLevel || (ListSqlFirewallViolationAnalyticsRequest.AccessLevel = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListSqlFirewallViolationAnalyticsRequest.SortOrder || (ListSqlFirewallViolationAnalyticsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["DbUserName"] = "dbUserName";
        SortBy["TargetId"] = "targetId";
        SortBy["TargetName"] = "targetName";
        SortBy["OperationTime"] = "operationTime";
        SortBy["TimeCollected"] = "timeCollected";
        SortBy["ClientOsUserName"] = "clientOsUserName";
        SortBy["Operation"] = "operation";
        SortBy["CurrentDbUserName"] = "currentDbUserName";
        SortBy["SqlLevel"] = "sqlLevel";
        SortBy["ClientIp"] = "clientIp";
        SortBy["ClientProgram"] = "clientProgram";
        SortBy["ViolationCause"] = "violationCause";
        SortBy["ViolationAction"] = "violationAction";
        SortBy["ViolationCount"] = "violationCount";
    })(SortBy = ListSqlFirewallViolationAnalyticsRequest.SortBy || (ListSqlFirewallViolationAnalyticsRequest.SortBy = {}));
    let SummaryField;
    (function (SummaryField) {
        SummaryField["DbUserName"] = "dbUserName";
        SummaryField["TargetName"] = "targetName";
        SummaryField["ClientOsUserName"] = "clientOsUserName";
        SummaryField["Operation"] = "operation";
        SummaryField["SqlText"] = "sqlText";
        SummaryField["CurrentDbUserName"] = "currentDbUserName";
        SummaryField["SqlLevel"] = "sqlLevel";
        SummaryField["ClientIp"] = "clientIp";
        SummaryField["ClientProgram"] = "clientProgram";
        SummaryField["ViolationCause"] = "violationCause";
        SummaryField["ViolationAction"] = "violationAction";
        SummaryField["Selects"] = "selects";
        SummaryField["Creates"] = "creates";
        SummaryField["Alters"] = "alters";
        SummaryField["Drops"] = "drops";
        SummaryField["Grants"] = "grants";
        SummaryField["Revokes"] = "revokes";
    })(SummaryField = ListSqlFirewallViolationAnalyticsRequest.SummaryField || (ListSqlFirewallViolationAnalyticsRequest.SummaryField = {}));
    let GroupBy;
    (function (GroupBy) {
        GroupBy["DbUserName"] = "dbUserName";
        GroupBy["TargetName"] = "targetName";
        GroupBy["OperationTime"] = "operationTime";
        GroupBy["TimeCollected"] = "timeCollected";
        GroupBy["ClientOsUserName"] = "clientOsUserName";
        GroupBy["Operation"] = "operation";
        GroupBy["SqlText"] = "sqlText";
        GroupBy["CurrentDbUserName"] = "currentDbUserName";
        GroupBy["SqlLevel"] = "sqlLevel";
        GroupBy["ClientIp"] = "clientIp";
        GroupBy["ClientProgram"] = "clientProgram";
        GroupBy["ViolationCause"] = "violationCause";
        GroupBy["ViolationAction"] = "violationAction";
    })(GroupBy = ListSqlFirewallViolationAnalyticsRequest.GroupBy || (ListSqlFirewallViolationAnalyticsRequest.GroupBy = {}));
})(ListSqlFirewallViolationAnalyticsRequest = exports.ListSqlFirewallViolationAnalyticsRequest || (exports.ListSqlFirewallViolationAnalyticsRequest = {}));
