// oci-datasafe/lib/request/list-sql-firewall-policy-analytics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSqlFirewallPolicyAnalyticsRequest = void 0;
var ListSqlFirewallPolicyAnalyticsRequest;
(function (ListSqlFirewallPolicyAnalyticsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListSqlFirewallPolicyAnalyticsRequest.AccessLevel || (ListSqlFirewallPolicyAnalyticsRequest.AccessLevel = {}));
    let GroupBy;
    (function (GroupBy) {
        GroupBy["ViolationAction"] = "violationAction";
        GroupBy["EnforcementScope"] = "enforcementScope";
        GroupBy["SecurityPolicyId"] = "securityPolicyId";
        GroupBy["LifecycleState"] = "lifecycleState";
    })(GroupBy = ListSqlFirewallPolicyAnalyticsRequest.GroupBy || (ListSqlFirewallPolicyAnalyticsRequest.GroupBy = {}));
})(ListSqlFirewallPolicyAnalyticsRequest = exports.ListSqlFirewallPolicyAnalyticsRequest || (exports.ListSqlFirewallPolicyAnalyticsRequest = {}));
