// oci-datasafe/lib/request/list-masking-policy-health-reports-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMaskingPolicyHealthReportsRequest = void 0;
var ListMaskingPolicyHealthReportsRequest;
(function (ListMaskingPolicyHealthReportsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListMaskingPolicyHealthReportsRequest.AccessLevel || (ListMaskingPolicyHealthReportsRequest.AccessLevel = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["DisplayName"] = "displayName";
        SortBy["TimeCreated"] = "timeCreated";
    })(SortBy = ListMaskingPolicyHealthReportsRequest.SortBy || (ListMaskingPolicyHealthReportsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListMaskingPolicyHealthReportsRequest.SortOrder || (ListMaskingPolicyHealthReportsRequest.SortOrder = {}));
})(ListMaskingPolicyHealthReportsRequest = exports.ListMaskingPolicyHealthReportsRequest || (exports.ListMaskingPolicyHealthReportsRequest = {}));
