// oci-datasafe/lib/request/list-finding-analytics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFindingAnalyticsRequest = void 0;
var ListFindingAnalyticsRequest;
(function (ListFindingAnalyticsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListFindingAnalyticsRequest.AccessLevel || (ListFindingAnalyticsRequest.AccessLevel = {}));
    let GroupBy;
    (function (GroupBy) {
        GroupBy["FindingKeyAndTopFindingStatus"] = "findingKeyAndTopFindingStatus";
        GroupBy["FindingKeyAndSeverity"] = "findingKeyAndSeverity";
    })(GroupBy = ListFindingAnalyticsRequest.GroupBy || (ListFindingAnalyticsRequest.GroupBy = {}));
    let Severity;
    (function (Severity) {
        Severity["High"] = "HIGH";
        Severity["Medium"] = "MEDIUM";
        Severity["Low"] = "LOW";
        Severity["Evaluate"] = "EVALUATE";
        Severity["Advisory"] = "ADVISORY";
        Severity["Pass"] = "PASS";
        Severity["Deferred"] = "DEFERRED";
    })(Severity = ListFindingAnalyticsRequest.Severity || (ListFindingAnalyticsRequest.Severity = {}));
})(ListFindingAnalyticsRequest = exports.ListFindingAnalyticsRequest || (exports.ListFindingAnalyticsRequest = {}));
