// oci-aidocument/lib/model/add-model-lock-details.js
"use strict";
/**
 * Document Understanding API
 * Document AI helps customers perform various analysis on their documents. If a customer has lots of documents, they can process them in batch using asynchronous API endpoints.
 * OpenAPI spec version: 20221109
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddModelLockDetails = void 0;
var AddModelLockDetails;
(function (AddModelLockDetails) {
    let Type;
    (function (Type) {
        Type["Full"] = "FULL";
        Type["Delete"] = "DELETE";
    })(Type = AddModelLockDetails.Type || (AddModelLockDetails.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AddModelLockDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AddModelLockDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(AddModelLockDetails = exports.AddModelLockDetails || (exports.AddModelLockDetails = {}));
