// oci-aidocument/lib/model/index.js
"use strict";
/**
 * Document Understanding API
 * Document AI helps customers perform various analysis on their documents. If a customer has lots of documents, they can process them in batch using asynchronous API endpoints.
 * OpenAPI spec version: 20221109
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProcessorJob = exports.ProcessorConfig = exports.ProcessingError = exports.PatchResponseMessage = exports.PatchModelOperation = exports.PatchModelDetails = exports.Page = exports.OutputLocation = exports.OperationType = exports.OperationStatus = exports.ObjectLocation = exports.NormalizedVertex = exports.ModelSummary = exports.ModelMetrics = exports.ModelCollection = exports.Model = exports.Line = exports.KeyValueDetectionOverallMetricsReport = exports.KeyValueDetectionLabelMetricsReport = exports.KeyValueDetectionConfidenceEntry = exports.InputLocation = exports.FieldValue = exports.FieldName = exports.FieldLabel = exports.DocumentType = exports.DocumentMetadata = exports.DocumentField = exports.DocumentFeature = exports.DocumentDetails = exports.DocumentClassificationOverallMetricsReport = exports.DocumentClassificationLabelMetricsReport = exports.DocumentClassificationConfidenceEntry = exports.Dimensions = exports.DetectedLanguage = exports.DetectedDocumentType = exports.DatasetSummary = exports.Dataset = exports.CreateProjectDetails = exports.CreateProcessorJobDetails = exports.CreateModelDetails = exports.ComponentModel = exports.ChangeProjectCompartmentDetails = exports.ChangeModelCompartmentDetails = exports.Cell = exports.BoundingPolygon = exports.AnalyzeDocumentResult = exports.AnalyzeDocumentDetails = exports.AddProjectLockDetails = exports.AddModelLockDetails = exports.ActionType = void 0;
exports.ValueTime = exports.ValueString = exports.ValuePhoneNumber = exports.ValueNumber = exports.ValueInteger = exports.ValueDate = exports.ValueArray = exports.ObjectStorageLocations = exports.ObjectStorageDocumentDetails = exports.ObjectStorageDataset = exports.KeyValueDetectionModelMetrics = exports.InlineDocumentDetails = exports.InlineDocumentContent = exports.GeneralProcessorConfig = exports.DocumentTextExtractionFeature = exports.DocumentTableExtractionFeature = exports.DocumentLanguageClassificationFeature = exports.DocumentKeyValueExtractionFeature = exports.DocumentClassificationModelMetrics = exports.DocumentClassificationFeature = exports.DataScienceLabelingDataset = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.Word = exports.UpdateProjectDetails = exports.UpdateModelDetails = exports.TableRow = exports.Table = exports.SortOrder = exports.ResourceLock = exports.RemoveProjectLockDetails = exports.RemoveModelLockDetails = exports.ProjectSummary = exports.ProjectCollection = exports.Project = exports.ProcessorType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AddModelLockDetails = __importStar(require("./add-model-lock-details"));
exports.AddModelLockDetails = AddModelLockDetails.AddModelLockDetails;
const AddProjectLockDetails = __importStar(require("./add-project-lock-details"));
exports.AddProjectLockDetails = AddProjectLockDetails.AddProjectLockDetails;
const AnalyzeDocumentDetails = __importStar(require("./analyze-document-details"));
exports.AnalyzeDocumentDetails = AnalyzeDocumentDetails.AnalyzeDocumentDetails;
const AnalyzeDocumentResult = __importStar(require("./analyze-document-result"));
exports.AnalyzeDocumentResult = AnalyzeDocumentResult.AnalyzeDocumentResult;
const BoundingPolygon = __importStar(require("./bounding-polygon"));
exports.BoundingPolygon = BoundingPolygon.BoundingPolygon;
const Cell = __importStar(require("./cell"));
exports.Cell = Cell.Cell;
const ChangeModelCompartmentDetails = __importStar(require("./change-model-compartment-details"));
exports.ChangeModelCompartmentDetails = ChangeModelCompartmentDetails.ChangeModelCompartmentDetails;
const ChangeProjectCompartmentDetails = __importStar(require("./change-project-compartment-details"));
exports.ChangeProjectCompartmentDetails = ChangeProjectCompartmentDetails.ChangeProjectCompartmentDetails;
const ComponentModel = __importStar(require("./component-model"));
exports.ComponentModel = ComponentModel.ComponentModel;
const CreateModelDetails = __importStar(require("./create-model-details"));
exports.CreateModelDetails = CreateModelDetails.CreateModelDetails;
const CreateProcessorJobDetails = __importStar(require("./create-processor-job-details"));
exports.CreateProcessorJobDetails = CreateProcessorJobDetails.CreateProcessorJobDetails;
const CreateProjectDetails = __importStar(require("./create-project-details"));
exports.CreateProjectDetails = CreateProjectDetails.CreateProjectDetails;
const Dataset = __importStar(require("./dataset"));
exports.Dataset = Dataset.Dataset;
const DatasetSummary = __importStar(require("./dataset-summary"));
exports.DatasetSummary = DatasetSummary.DatasetSummary;
const DetectedDocumentType = __importStar(require("./detected-document-type"));
exports.DetectedDocumentType = DetectedDocumentType.DetectedDocumentType;
const DetectedLanguage = __importStar(require("./detected-language"));
exports.DetectedLanguage = DetectedLanguage.DetectedLanguage;
const Dimensions = __importStar(require("./dimensions"));
exports.Dimensions = Dimensions.Dimensions;
const DocumentClassificationConfidenceEntry = __importStar(require("./document-classification-confidence-entry"));
exports.DocumentClassificationConfidenceEntry = DocumentClassificationConfidenceEntry.DocumentClassificationConfidenceEntry;
const DocumentClassificationLabelMetricsReport = __importStar(require("./document-classification-label-metrics-report"));
exports.DocumentClassificationLabelMetricsReport = DocumentClassificationLabelMetricsReport.DocumentClassificationLabelMetricsReport;
const DocumentClassificationOverallMetricsReport = __importStar(require("./document-classification-overall-metrics-report"));
exports.DocumentClassificationOverallMetricsReport = DocumentClassificationOverallMetricsReport.DocumentClassificationOverallMetricsReport;
const DocumentDetails = __importStar(require("./document-details"));
exports.DocumentDetails = DocumentDetails.DocumentDetails;
const DocumentFeature = __importStar(require("./document-feature"));
exports.DocumentFeature = DocumentFeature.DocumentFeature;
const DocumentField = __importStar(require("./document-field"));
exports.DocumentField = DocumentField.DocumentField;
const DocumentMetadata = __importStar(require("./document-metadata"));
exports.DocumentMetadata = DocumentMetadata.DocumentMetadata;
const DocumentType = __importStar(require("./document-type"));
exports.DocumentType = DocumentType.DocumentType;
const FieldLabel = __importStar(require("./field-label"));
exports.FieldLabel = FieldLabel.FieldLabel;
const FieldName = __importStar(require("./field-name"));
exports.FieldName = FieldName.FieldName;
const FieldValue = __importStar(require("./field-value"));
exports.FieldValue = FieldValue.FieldValue;
const InputLocation = __importStar(require("./input-location"));
exports.InputLocation = InputLocation.InputLocation;
const KeyValueDetectionConfidenceEntry = __importStar(require("./key-value-detection-confidence-entry"));
exports.KeyValueDetectionConfidenceEntry = KeyValueDetectionConfidenceEntry.KeyValueDetectionConfidenceEntry;
const KeyValueDetectionLabelMetricsReport = __importStar(require("./key-value-detection-label-metrics-report"));
exports.KeyValueDetectionLabelMetricsReport = KeyValueDetectionLabelMetricsReport.KeyValueDetectionLabelMetricsReport;
const KeyValueDetectionOverallMetricsReport = __importStar(require("./key-value-detection-overall-metrics-report"));
exports.KeyValueDetectionOverallMetricsReport = KeyValueDetectionOverallMetricsReport.KeyValueDetectionOverallMetricsReport;
const Line = __importStar(require("./line"));
exports.Line = Line.Line;
const Model = __importStar(require("./model"));
exports.Model = Model.Model;
const ModelCollection = __importStar(require("./model-collection"));
exports.ModelCollection = ModelCollection.ModelCollection;
const ModelMetrics = __importStar(require("./model-metrics"));
exports.ModelMetrics = ModelMetrics.ModelMetrics;
const ModelSummary = __importStar(require("./model-summary"));
exports.ModelSummary = ModelSummary.ModelSummary;
const NormalizedVertex = __importStar(require("./normalized-vertex"));
exports.NormalizedVertex = NormalizedVertex.NormalizedVertex;
const ObjectLocation = __importStar(require("./object-location"));
exports.ObjectLocation = ObjectLocation.ObjectLocation;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OutputLocation = __importStar(require("./output-location"));
exports.OutputLocation = OutputLocation.OutputLocation;
const Page = __importStar(require("./page"));
exports.Page = Page.Page;
const PatchModelDetails = __importStar(require("./patch-model-details"));
exports.PatchModelDetails = PatchModelDetails.PatchModelDetails;
const PatchModelOperation = __importStar(require("./patch-model-operation"));
exports.PatchModelOperation = PatchModelOperation.PatchModelOperation;
const PatchResponseMessage = __importStar(require("./patch-response-message"));
exports.PatchResponseMessage = PatchResponseMessage.PatchResponseMessage;
const ProcessingError = __importStar(require("./processing-error"));
exports.ProcessingError = ProcessingError.ProcessingError;
const ProcessorConfig = __importStar(require("./processor-config"));
exports.ProcessorConfig = ProcessorConfig.ProcessorConfig;
const ProcessorJob = __importStar(require("./processor-job"));
exports.ProcessorJob = ProcessorJob.ProcessorJob;
const ProcessorType = __importStar(require("./processor-type"));
exports.ProcessorType = ProcessorType.ProcessorType;
const Project = __importStar(require("./project"));
exports.Project = Project.Project;
const ProjectCollection = __importStar(require("./project-collection"));
exports.ProjectCollection = ProjectCollection.ProjectCollection;
const ProjectSummary = __importStar(require("./project-summary"));
exports.ProjectSummary = ProjectSummary.ProjectSummary;
const RemoveModelLockDetails = __importStar(require("./remove-model-lock-details"));
exports.RemoveModelLockDetails = RemoveModelLockDetails.RemoveModelLockDetails;
const RemoveProjectLockDetails = __importStar(require("./remove-project-lock-details"));
exports.RemoveProjectLockDetails = RemoveProjectLockDetails.RemoveProjectLockDetails;
const ResourceLock = __importStar(require("./resource-lock"));
exports.ResourceLock = ResourceLock.ResourceLock;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Table = __importStar(require("./table"));
exports.Table = Table.Table;
const TableRow = __importStar(require("./table-row"));
exports.TableRow = TableRow.TableRow;
const UpdateModelDetails = __importStar(require("./update-model-details"));
exports.UpdateModelDetails = UpdateModelDetails.UpdateModelDetails;
const UpdateProjectDetails = __importStar(require("./update-project-details"));
exports.UpdateProjectDetails = UpdateProjectDetails.UpdateProjectDetails;
const Word = __importStar(require("./word"));
exports.Word = Word.Word;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const DataScienceLabelingDataset = __importStar(require("./data-science-labeling-dataset"));
exports.DataScienceLabelingDataset = DataScienceLabelingDataset.DataScienceLabelingDataset;
const DocumentClassificationFeature = __importStar(require("./document-classification-feature"));
exports.DocumentClassificationFeature = DocumentClassificationFeature.DocumentClassificationFeature;
const DocumentClassificationModelMetrics = __importStar(require("./document-classification-model-metrics"));
exports.DocumentClassificationModelMetrics = DocumentClassificationModelMetrics.DocumentClassificationModelMetrics;
const DocumentKeyValueExtractionFeature = __importStar(require("./document-key-value-extraction-feature"));
exports.DocumentKeyValueExtractionFeature = DocumentKeyValueExtractionFeature.DocumentKeyValueExtractionFeature;
const DocumentLanguageClassificationFeature = __importStar(require("./document-language-classification-feature"));
exports.DocumentLanguageClassificationFeature = DocumentLanguageClassificationFeature.DocumentLanguageClassificationFeature;
const DocumentTableExtractionFeature = __importStar(require("./document-table-extraction-feature"));
exports.DocumentTableExtractionFeature = DocumentTableExtractionFeature.DocumentTableExtractionFeature;
const DocumentTextExtractionFeature = __importStar(require("./document-text-extraction-feature"));
exports.DocumentTextExtractionFeature = DocumentTextExtractionFeature.DocumentTextExtractionFeature;
const GeneralProcessorConfig = __importStar(require("./general-processor-config"));
exports.GeneralProcessorConfig = GeneralProcessorConfig.GeneralProcessorConfig;
const InlineDocumentContent = __importStar(require("./inline-document-content"));
exports.InlineDocumentContent = InlineDocumentContent.InlineDocumentContent;
const InlineDocumentDetails = __importStar(require("./inline-document-details"));
exports.InlineDocumentDetails = InlineDocumentDetails.InlineDocumentDetails;
const KeyValueDetectionModelMetrics = __importStar(require("./key-value-detection-model-metrics"));
exports.KeyValueDetectionModelMetrics = KeyValueDetectionModelMetrics.KeyValueDetectionModelMetrics;
const ObjectStorageDataset = __importStar(require("./object-storage-dataset"));
exports.ObjectStorageDataset = ObjectStorageDataset.ObjectStorageDataset;
const ObjectStorageDocumentDetails = __importStar(require("./object-storage-document-details"));
exports.ObjectStorageDocumentDetails = ObjectStorageDocumentDetails.ObjectStorageDocumentDetails;
const ObjectStorageLocations = __importStar(require("./object-storage-locations"));
exports.ObjectStorageLocations = ObjectStorageLocations.ObjectStorageLocations;
const ValueArray = __importStar(require("./value-array"));
exports.ValueArray = ValueArray.ValueArray;
const ValueDate = __importStar(require("./value-date"));
exports.ValueDate = ValueDate.ValueDate;
const ValueInteger = __importStar(require("./value-integer"));
exports.ValueInteger = ValueInteger.ValueInteger;
const ValueNumber = __importStar(require("./value-number"));
exports.ValueNumber = ValueNumber.ValueNumber;
const ValuePhoneNumber = __importStar(require("./value-phone-number"));
exports.ValuePhoneNumber = ValuePhoneNumber.ValuePhoneNumber;
const ValueString = __importStar(require("./value-string"));
exports.ValueString = ValueString.ValueString;
const ValueTime = __importStar(require("./value-time"));
exports.ValueTime = ValueTime.ValueTime;
