// oci-filestorage/lib/model/create-quota-rule-details.js
"use strict";
/**
 * File Storage API
 * Use the File Storage service API to manage file systems, mount targets, and snapshots.
For more information, see [Overview of File Storage](https://docs.oracle.com/iaas/Content/File/Concepts/filestorageoverview.htm).

 * OpenAPI spec version: 20171215
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateQuotaRuleDetails = void 0;
var CreateQuotaRuleDetails;
(function (CreateQuotaRuleDetails) {
    let PrincipalType;
    (function (PrincipalType) {
        PrincipalType["FileSystemLevel"] = "FILE_SYSTEM_LEVEL";
        PrincipalType["DefaultGroup"] = "DEFAULT_GROUP";
        PrincipalType["DefaultUser"] = "DEFAULT_USER";
        PrincipalType["IndividualGroup"] = "INDIVIDUAL_GROUP";
        PrincipalType["IndividualUser"] = "INDIVIDUAL_USER";
    })(PrincipalType = CreateQuotaRuleDetails.PrincipalType || (CreateQuotaRuleDetails.PrincipalType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateQuotaRuleDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateQuotaRuleDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateQuotaRuleDetails = exports.CreateQuotaRuleDetails || (exports.CreateQuotaRuleDetails = {}));
