// oci-filestorage/lib/model/index.js
"use strict";
/**
 * File Storage API
 * Use the File Storage service API to manage file systems, mount targets, and snapshots.
For more information, see [Overview of File Storage](https://docs.oracle.com/iaas/Content/File/Concepts/filestorageoverview.htm).

 * OpenAPI spec version: 20171215
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateFileSystemDetails = exports.UpdateExportSetDetails = exports.UpdateExportDetails = exports.ToggleQuotaRulesDetails = exports.SourceDetails = exports.SnapshotSummary = exports.SnapshotSchedule = exports.Snapshot = exports.ScheduleDowngradeShapeMountTargetDetails = exports.ResourceLock = exports.ReplicationTargetSummary = exports.ReplicationTarget = exports.ReplicationSummary = exports.ReplicationEstimate = exports.Replication = exports.QuotaRuleSummary = exports.QuotaRule = exports.OutboundConnectorSummary = exports.OutboundConnector = exports.MountTargetSummary = exports.MountTarget = exports.LdapIdmap = exports.KeyTabSecretDetails = exports.KerberosKeytabEntry = exports.Kerberos = exports.FilesystemSnapshotPolicySummary = exports.FilesystemSnapshotPolicy = exports.FileSystemSummary = exports.FileSystem = exports.ExportSummary = exports.ExportSetSummary = exports.ExportSet = exports.Export = exports.Endpoint = exports.CreateSnapshotDetails = exports.CreateReplicationDetails = exports.CreateQuotaRuleDetails = exports.CreateOutboundConnectorDetails = exports.CreateMountTargetDetails = exports.CreateLdapIdmapDetails = exports.CreateKerberosDetails = exports.CreateFilesystemSnapshotPolicyDetails = exports.CreateFileSystemDetails = exports.CreateExportDetails = exports.ClientOptions = exports.ChangeReplicationCompartmentDetails = exports.ChangeOutboundConnectorCompartmentDetails = exports.ChangeMountTargetCompartmentDetails = exports.ChangeFilesystemSnapshotPolicyCompartmentDetails = exports.ChangeFileSystemCompartmentDetails = void 0;
exports.LdapBindAccountSummary = exports.LdapBindAccount = exports.CreateLdapBindAccountDetails = exports.ValidateKeyTabsResponseDetails = exports.ValidateKeyTabsDetails = exports.UpgradeShapeMountTargetDetails = exports.UpdateSnapshotDetails = exports.UpdateReplicationDetails = exports.UpdateQuotaRuleDetails = exports.UpdateOutboundConnectorDetails = exports.UpdateMountTargetDetails = exports.UpdateLdapIdmapDetails = exports.UpdateKerberosDetails = exports.UpdateFilesystemSnapshotPolicyDetails = void 0;
const ChangeFileSystemCompartmentDetails = __importStar(require("./change-file-system-compartment-details"));
exports.ChangeFileSystemCompartmentDetails = ChangeFileSystemCompartmentDetails.ChangeFileSystemCompartmentDetails;
const ChangeFilesystemSnapshotPolicyCompartmentDetails = __importStar(require("./change-filesystem-snapshot-policy-compartment-details"));
exports.ChangeFilesystemSnapshotPolicyCompartmentDetails = ChangeFilesystemSnapshotPolicyCompartmentDetails.ChangeFilesystemSnapshotPolicyCompartmentDetails;
const ChangeMountTargetCompartmentDetails = __importStar(require("./change-mount-target-compartment-details"));
exports.ChangeMountTargetCompartmentDetails = ChangeMountTargetCompartmentDetails.ChangeMountTargetCompartmentDetails;
const ChangeOutboundConnectorCompartmentDetails = __importStar(require("./change-outbound-connector-compartment-details"));
exports.ChangeOutboundConnectorCompartmentDetails = ChangeOutboundConnectorCompartmentDetails.ChangeOutboundConnectorCompartmentDetails;
const ChangeReplicationCompartmentDetails = __importStar(require("./change-replication-compartment-details"));
exports.ChangeReplicationCompartmentDetails = ChangeReplicationCompartmentDetails.ChangeReplicationCompartmentDetails;
const ClientOptions = __importStar(require("./client-options"));
exports.ClientOptions = ClientOptions.ClientOptions;
const CreateExportDetails = __importStar(require("./create-export-details"));
exports.CreateExportDetails = CreateExportDetails.CreateExportDetails;
const CreateFileSystemDetails = __importStar(require("./create-file-system-details"));
exports.CreateFileSystemDetails = CreateFileSystemDetails.CreateFileSystemDetails;
const CreateFilesystemSnapshotPolicyDetails = __importStar(require("./create-filesystem-snapshot-policy-details"));
exports.CreateFilesystemSnapshotPolicyDetails = CreateFilesystemSnapshotPolicyDetails.CreateFilesystemSnapshotPolicyDetails;
const CreateKerberosDetails = __importStar(require("./create-kerberos-details"));
exports.CreateKerberosDetails = CreateKerberosDetails.CreateKerberosDetails;
const CreateLdapIdmapDetails = __importStar(require("./create-ldap-idmap-details"));
exports.CreateLdapIdmapDetails = CreateLdapIdmapDetails.CreateLdapIdmapDetails;
const CreateMountTargetDetails = __importStar(require("./create-mount-target-details"));
exports.CreateMountTargetDetails = CreateMountTargetDetails.CreateMountTargetDetails;
const CreateOutboundConnectorDetails = __importStar(require("./create-outbound-connector-details"));
exports.CreateOutboundConnectorDetails = CreateOutboundConnectorDetails.CreateOutboundConnectorDetails;
const CreateQuotaRuleDetails = __importStar(require("./create-quota-rule-details"));
exports.CreateQuotaRuleDetails = CreateQuotaRuleDetails.CreateQuotaRuleDetails;
const CreateReplicationDetails = __importStar(require("./create-replication-details"));
exports.CreateReplicationDetails = CreateReplicationDetails.CreateReplicationDetails;
const CreateSnapshotDetails = __importStar(require("./create-snapshot-details"));
exports.CreateSnapshotDetails = CreateSnapshotDetails.CreateSnapshotDetails;
const Endpoint = __importStar(require("./endpoint"));
exports.Endpoint = Endpoint.Endpoint;
const Export = __importStar(require("./export"));
exports.Export = Export.Export;
const ExportSet = __importStar(require("./export-set"));
exports.ExportSet = ExportSet.ExportSet;
const ExportSetSummary = __importStar(require("./export-set-summary"));
exports.ExportSetSummary = ExportSetSummary.ExportSetSummary;
const ExportSummary = __importStar(require("./export-summary"));
exports.ExportSummary = ExportSummary.ExportSummary;
const FileSystem = __importStar(require("./file-system"));
exports.FileSystem = FileSystem.FileSystem;
const FileSystemSummary = __importStar(require("./file-system-summary"));
exports.FileSystemSummary = FileSystemSummary.FileSystemSummary;
const FilesystemSnapshotPolicy = __importStar(require("./filesystem-snapshot-policy"));
exports.FilesystemSnapshotPolicy = FilesystemSnapshotPolicy.FilesystemSnapshotPolicy;
const FilesystemSnapshotPolicySummary = __importStar(require("./filesystem-snapshot-policy-summary"));
exports.FilesystemSnapshotPolicySummary = FilesystemSnapshotPolicySummary.FilesystemSnapshotPolicySummary;
const Kerberos = __importStar(require("./kerberos"));
exports.Kerberos = Kerberos.Kerberos;
const KerberosKeytabEntry = __importStar(require("./kerberos-keytab-entry"));
exports.KerberosKeytabEntry = KerberosKeytabEntry.KerberosKeytabEntry;
const KeyTabSecretDetails = __importStar(require("./key-tab-secret-details"));
exports.KeyTabSecretDetails = KeyTabSecretDetails.KeyTabSecretDetails;
const LdapIdmap = __importStar(require("./ldap-idmap"));
exports.LdapIdmap = LdapIdmap.LdapIdmap;
const MountTarget = __importStar(require("./mount-target"));
exports.MountTarget = MountTarget.MountTarget;
const MountTargetSummary = __importStar(require("./mount-target-summary"));
exports.MountTargetSummary = MountTargetSummary.MountTargetSummary;
const OutboundConnector = __importStar(require("./outbound-connector"));
exports.OutboundConnector = OutboundConnector.OutboundConnector;
const OutboundConnectorSummary = __importStar(require("./outbound-connector-summary"));
exports.OutboundConnectorSummary = OutboundConnectorSummary.OutboundConnectorSummary;
const QuotaRule = __importStar(require("./quota-rule"));
exports.QuotaRule = QuotaRule.QuotaRule;
const QuotaRuleSummary = __importStar(require("./quota-rule-summary"));
exports.QuotaRuleSummary = QuotaRuleSummary.QuotaRuleSummary;
const Replication = __importStar(require("./replication"));
exports.Replication = Replication.Replication;
const ReplicationEstimate = __importStar(require("./replication-estimate"));
exports.ReplicationEstimate = ReplicationEstimate.ReplicationEstimate;
const ReplicationSummary = __importStar(require("./replication-summary"));
exports.ReplicationSummary = ReplicationSummary.ReplicationSummary;
const ReplicationTarget = __importStar(require("./replication-target"));
exports.ReplicationTarget = ReplicationTarget.ReplicationTarget;
const ReplicationTargetSummary = __importStar(require("./replication-target-summary"));
exports.ReplicationTargetSummary = ReplicationTargetSummary.ReplicationTargetSummary;
const ResourceLock = __importStar(require("./resource-lock"));
exports.ResourceLock = ResourceLock.ResourceLock;
const ScheduleDowngradeShapeMountTargetDetails = __importStar(require("./schedule-downgrade-shape-mount-target-details"));
exports.ScheduleDowngradeShapeMountTargetDetails = ScheduleDowngradeShapeMountTargetDetails.ScheduleDowngradeShapeMountTargetDetails;
const Snapshot = __importStar(require("./snapshot"));
exports.Snapshot = Snapshot.Snapshot;
const SnapshotSchedule = __importStar(require("./snapshot-schedule"));
exports.SnapshotSchedule = SnapshotSchedule.SnapshotSchedule;
const SnapshotSummary = __importStar(require("./snapshot-summary"));
exports.SnapshotSummary = SnapshotSummary.SnapshotSummary;
const SourceDetails = __importStar(require("./source-details"));
exports.SourceDetails = SourceDetails.SourceDetails;
const ToggleQuotaRulesDetails = __importStar(require("./toggle-quota-rules-details"));
exports.ToggleQuotaRulesDetails = ToggleQuotaRulesDetails.ToggleQuotaRulesDetails;
const UpdateExportDetails = __importStar(require("./update-export-details"));
exports.UpdateExportDetails = UpdateExportDetails.UpdateExportDetails;
const UpdateExportSetDetails = __importStar(require("./update-export-set-details"));
exports.UpdateExportSetDetails = UpdateExportSetDetails.UpdateExportSetDetails;
const UpdateFileSystemDetails = __importStar(require("./update-file-system-details"));
exports.UpdateFileSystemDetails = UpdateFileSystemDetails.UpdateFileSystemDetails;
const UpdateFilesystemSnapshotPolicyDetails = __importStar(require("./update-filesystem-snapshot-policy-details"));
exports.UpdateFilesystemSnapshotPolicyDetails = UpdateFilesystemSnapshotPolicyDetails.UpdateFilesystemSnapshotPolicyDetails;
const UpdateKerberosDetails = __importStar(require("./update-kerberos-details"));
exports.UpdateKerberosDetails = UpdateKerberosDetails.UpdateKerberosDetails;
const UpdateLdapIdmapDetails = __importStar(require("./update-ldap-idmap-details"));
exports.UpdateLdapIdmapDetails = UpdateLdapIdmapDetails.UpdateLdapIdmapDetails;
const UpdateMountTargetDetails = __importStar(require("./update-mount-target-details"));
exports.UpdateMountTargetDetails = UpdateMountTargetDetails.UpdateMountTargetDetails;
const UpdateOutboundConnectorDetails = __importStar(require("./update-outbound-connector-details"));
exports.UpdateOutboundConnectorDetails = UpdateOutboundConnectorDetails.UpdateOutboundConnectorDetails;
const UpdateQuotaRuleDetails = __importStar(require("./update-quota-rule-details"));
exports.UpdateQuotaRuleDetails = UpdateQuotaRuleDetails.UpdateQuotaRuleDetails;
const UpdateReplicationDetails = __importStar(require("./update-replication-details"));
exports.UpdateReplicationDetails = UpdateReplicationDetails.UpdateReplicationDetails;
const UpdateSnapshotDetails = __importStar(require("./update-snapshot-details"));
exports.UpdateSnapshotDetails = UpdateSnapshotDetails.UpdateSnapshotDetails;
const UpgradeShapeMountTargetDetails = __importStar(require("./upgrade-shape-mount-target-details"));
exports.UpgradeShapeMountTargetDetails = UpgradeShapeMountTargetDetails.UpgradeShapeMountTargetDetails;
const ValidateKeyTabsDetails = __importStar(require("./validate-key-tabs-details"));
exports.ValidateKeyTabsDetails = ValidateKeyTabsDetails.ValidateKeyTabsDetails;
const ValidateKeyTabsResponseDetails = __importStar(require("./validate-key-tabs-response-details"));
exports.ValidateKeyTabsResponseDetails = ValidateKeyTabsResponseDetails.ValidateKeyTabsResponseDetails;
const CreateLdapBindAccountDetails = __importStar(require("./create-ldap-bind-account-details"));
exports.CreateLdapBindAccountDetails = CreateLdapBindAccountDetails.CreateLdapBindAccountDetails;
const LdapBindAccount = __importStar(require("./ldap-bind-account"));
exports.LdapBindAccount = LdapBindAccount.LdapBindAccount;
const LdapBindAccountSummary = __importStar(require("./ldap-bind-account-summary"));
exports.LdapBindAccountSummary = LdapBindAccountSummary.LdapBindAccountSummary;
