// oci-filestorage/lib/model/snapshot-summary.js
"use strict";
/**
 * File Storage API
 * Use the File Storage service API to manage file systems, mount targets, and snapshots.
For more information, see [Overview of File Storage](https://docs.oracle.com/iaas/Content/File/Concepts/filestorageoverview.htm).

 * OpenAPI spec version: 20171215
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SnapshotSummary = void 0;
const model = __importStar(require("../model"));
var SnapshotSummary;
(function (SnapshotSummary) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = SnapshotSummary.LifecycleState || (SnapshotSummary.LifecycleState = {}));
    let SnapshotType;
    (function (SnapshotType) {
        SnapshotType["User"] = "USER";
        SnapshotType["PolicyBased"] = "POLICY_BASED";
        SnapshotType["Replication"] = "REPLICATION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SnapshotType["UnknownValue"] = "UNKNOWN_VALUE";
    })(SnapshotType = SnapshotSummary.SnapshotType || (SnapshotSummary.SnapshotType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "locks": obj.locks
                ? obj.locks.map(item => {
                    return model.ResourceLock.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    SnapshotSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "locks": obj.locks
                ? obj.locks.map(item => {
                    return model.ResourceLock.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    SnapshotSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(SnapshotSummary = exports.SnapshotSummary || (exports.SnapshotSummary = {}));
