// oci-capacitymanagement/lib/model/metadata-details.js
"use strict";
/**
 * OCI Control Center Capacity Management API
 * OCI Control Center (OCC) Capacity Management enables you to manage capacity requests in realms where OCI Control Center Capacity Management is available. For more information, see [OCI Control Center](https://docs.oracle.com/iaas/Content/control-center/home.htm).
 * OpenAPI spec version: 20231107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetadataDetails = void 0;
var MetadataDetails;
(function (MetadataDetails) {
    let FormatVersion;
    (function (FormatVersion) {
        FormatVersion["V1"] = "V1";
        FormatVersion["V2"] = "V2";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        FormatVersion["UnknownValue"] = "UNKNOWN_VALUE";
    })(FormatVersion = MetadataDetails.FormatVersion || (MetadataDetails.FormatVersion = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MetadataDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MetadataDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(MetadataDetails = exports.MetadataDetails || (exports.MetadataDetails = {}));
