// oci-capacitymanagement/lib/model/occm-demand-signal.js
"use strict";
/**
 * OCI Control Center Capacity Management API
 * OCI Control Center (OCC) Capacity Management enables you to manage capacity requests in realms where OCI Control Center Capacity Management is available. For more information, see [OCI Control Center](https://docs.oracle.com/iaas/Content/control-center/home.htm).
 * OpenAPI spec version: 20231107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OccmDemandSignal = void 0;
var OccmDemandSignal;
(function (OccmDemandSignal) {
    let LifecycleDetails;
    (function (LifecycleDetails) {
        LifecycleDetails["Created"] = "CREATED";
        LifecycleDetails["Submitted"] = "SUBMITTED";
        LifecycleDetails["Deleted"] = "DELETED";
        LifecycleDetails["InProgress"] = "IN_PROGRESS";
        LifecycleDetails["Rejected"] = "REJECTED";
        LifecycleDetails["Completed"] = "COMPLETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleDetails["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleDetails = OccmDemandSignal.LifecycleDetails || (OccmDemandSignal.LifecycleDetails = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = OccmDemandSignal.LifecycleState || (OccmDemandSignal.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    OccmDemandSignal.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    OccmDemandSignal.getDeserializedJsonObj = getDeserializedJsonObj;
})(OccmDemandSignal = exports.OccmDemandSignal || (exports.OccmDemandSignal = {}));
