// oci-capacitymanagement/lib/model/occm-demand-signal-item-request-type.js
"use strict";
/**
 * OCI Control Center Capacity Management API
 * OCI Control Center (OCC) Capacity Management enables you to manage capacity requests in realms where OCI Control Center Capacity Management is available. For more information, see [OCI Control Center](https://docs.oracle.com/iaas/Content/control-center/home.htm).
 * OpenAPI spec version: 20231107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OccmDemandSignalItemRequestType = void 0;
/**
 * The type of request made against a particular demand signal item.
 * DEMAND: This type of request signifies that you are demanding a resource from OCI.
 *
 **/
var OccmDemandSignalItemRequestType;
(function (OccmDemandSignalItemRequestType) {
    OccmDemandSignalItemRequestType["Demand"] = "DEMAND";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OccmDemandSignalItemRequestType["UnknownValue"] = "UNKNOWN_VALUE";
})(OccmDemandSignalItemRequestType = exports.OccmDemandSignalItemRequestType || (exports.OccmDemandSignalItemRequestType = {}));
(function (OccmDemandSignalItemRequestType) {
    function getJsonObj(obj) {
        return obj;
    }
    OccmDemandSignalItemRequestType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OccmDemandSignalItemRequestType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OccmDemandSignalItemRequestType = exports.OccmDemandSignalItemRequestType || (exports.OccmDemandSignalItemRequestType = {}));
