// oci-capacitymanagement/lib/model/update-internal-occ-capacity-request-details.js
"use strict";
/**
 * OCI Control Center Capacity Management API
 * OCI Control Center (OCC) Capacity Management enables you to manage capacity requests in realms where OCI Control Center Capacity Management is available. For more information, see [OCI Control Center](https://docs.oracle.com/iaas/Content/control-center/home.htm).
 * OpenAPI spec version: 20231107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateInternalOccCapacityRequestDetails = void 0;
var UpdateInternalOccCapacityRequestDetails;
(function (UpdateInternalOccCapacityRequestDetails) {
    let RequestState;
    (function (RequestState) {
        RequestState["Resolved"] = "RESOLVED";
        RequestState["Rejected"] = "REJECTED";
        RequestState["InProgress"] = "IN_PROGRESS";
    })(RequestState = UpdateInternalOccCapacityRequestDetails.RequestState || (UpdateInternalOccCapacityRequestDetails.RequestState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateInternalOccCapacityRequestDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateInternalOccCapacityRequestDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateInternalOccCapacityRequestDetails = exports.UpdateInternalOccCapacityRequestDetails || (exports.UpdateInternalOccCapacityRequestDetails = {}));
