// oci-adm/lib/request/index.js
"use strict";
/**
 * Application Dependency Management API
 * Use the Application Dependency Management API to create knowledge bases and vulnerability audits.  For more information, see [ADM](https://docs.oracle.com/iaas/Content/application-dependency-management/home.htm).
 * OpenAPI spec version: 20220421
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListVulnerabilityAuditsRequest = exports.ListStagesRequest = exports.ListRemediationRunsRequest = exports.ListRemediationRecipesRequest = exports.ListKnowledgeBasesRequest = exports.ListApplicationDependencyVulnerabilitiesRequest = exports.ListApplicationDependencyRecommendationsRequest = void 0;
const ListApplicationDependencyRecommendationsRequest = __importStar(require("./list-application-dependency-recommendations-request"));
exports.ListApplicationDependencyRecommendationsRequest = ListApplicationDependencyRecommendationsRequest.ListApplicationDependencyRecommendationsRequest;
const ListApplicationDependencyVulnerabilitiesRequest = __importStar(require("./list-application-dependency-vulnerabilities-request"));
exports.ListApplicationDependencyVulnerabilitiesRequest = ListApplicationDependencyVulnerabilitiesRequest.ListApplicationDependencyVulnerabilitiesRequest;
const ListKnowledgeBasesRequest = __importStar(require("./list-knowledge-bases-request"));
exports.ListKnowledgeBasesRequest = ListKnowledgeBasesRequest.ListKnowledgeBasesRequest;
const ListRemediationRecipesRequest = __importStar(require("./list-remediation-recipes-request"));
exports.ListRemediationRecipesRequest = ListRemediationRecipesRequest.ListRemediationRecipesRequest;
const ListRemediationRunsRequest = __importStar(require("./list-remediation-runs-request"));
exports.ListRemediationRunsRequest = ListRemediationRunsRequest.ListRemediationRunsRequest;
const ListStagesRequest = __importStar(require("./list-stages-request"));
exports.ListStagesRequest = ListStagesRequest.ListStagesRequest;
const ListVulnerabilityAuditsRequest = __importStar(require("./list-vulnerability-audits-request"));
exports.ListVulnerabilityAuditsRequest = ListVulnerabilityAuditsRequest.ListVulnerabilityAuditsRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
