// oci-distributeddatabase/lib/model/create-distributed-database-details.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage the Globally distributed databases.
 * OpenAPI spec version: 20250101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDistributedDatabaseDetails = void 0;
const model = __importStar(require("../model"));
var CreateDistributedDatabaseDetails;
(function (CreateDistributedDatabaseDetails) {
    let ShardingMethod;
    (function (ShardingMethod) {
        ShardingMethod["User"] = "USER";
        ShardingMethod["System"] = "SYSTEM";
    })(ShardingMethod = CreateDistributedDatabaseDetails.ShardingMethod || (CreateDistributedDatabaseDetails.ShardingMethod = {}));
    let ReplicationMethod;
    (function (ReplicationMethod) {
        ReplicationMethod["Raft"] = "RAFT";
        ReplicationMethod["Dg"] = "DG";
    })(ReplicationMethod = CreateDistributedDatabaseDetails.ReplicationMethod || (CreateDistributedDatabaseDetails.ReplicationMethod = {}));
    let DbDeploymentType;
    (function (DbDeploymentType) {
        DbDeploymentType["ExadbXs"] = "EXADB_XS";
    })(DbDeploymentType = CreateDistributedDatabaseDetails.DbDeploymentType || (CreateDistributedDatabaseDetails.DbDeploymentType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "shardDetails": obj.shardDetails
                ? obj.shardDetails.map(item => {
                    return model.CreateDistributedDatabaseShardDetails.getJsonObj(item);
                })
                : undefined,
            "catalogDetails": obj.catalogDetails
                ? obj.catalogDetails.map(item => {
                    return model.CreateDistributedDatabaseCatalogDetails.getJsonObj(item);
                })
                : undefined,
            "dbBackupConfig": obj.dbBackupConfig
                ? model.DistributedDbBackupConfig.getJsonObj(obj.dbBackupConfig)
                : undefined
        });
        return jsonObj;
    }
    CreateDistributedDatabaseDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "shardDetails": obj.shardDetails
                ? obj.shardDetails.map(item => {
                    return model.CreateDistributedDatabaseShardDetails.getDeserializedJsonObj(item);
                })
                : undefined,
            "catalogDetails": obj.catalogDetails
                ? obj.catalogDetails.map(item => {
                    return model.CreateDistributedDatabaseCatalogDetails.getDeserializedJsonObj(item);
                })
                : undefined,
            "dbBackupConfig": obj.dbBackupConfig
                ? model.DistributedDbBackupConfig.getDeserializedJsonObj(obj.dbBackupConfig)
                : undefined
        });
        return jsonObj;
    }
    CreateDistributedDatabaseDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateDistributedDatabaseDetails = exports.CreateDistributedDatabaseDetails || (exports.CreateDistributedDatabaseDetails = {}));
