// oci-distributeddatabase/lib/model/distributed-database-private-endpoint.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage the Globally distributed databases.
 * OpenAPI spec version: 20250101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DistributedDatabasePrivateEndpoint = void 0;
const model = __importStar(require("../model"));
var DistributedDatabasePrivateEndpoint;
(function (DistributedDatabasePrivateEndpoint) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Creating"] = "CREATING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = DistributedDatabasePrivateEndpoint.LifecycleState || (DistributedDatabasePrivateEndpoint.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "globallyDistributedDatabases": obj.globallyDistributedDatabases
                ? obj.globallyDistributedDatabases.map(item => {
                    return model.DistributedDatabaseAssociatedWithPrivateEndpoint.getJsonObj(item);
                })
                : undefined,
            "globallyDistributedAutonomousDatabases": obj.globallyDistributedAutonomousDatabases
                ? obj.globallyDistributedAutonomousDatabases.map(item => {
                    return model.DistributedAutonomousDatabaseAssociatedWithPrivateEndpoint.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DistributedDatabasePrivateEndpoint.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "globallyDistributedDatabases": obj.globallyDistributedDatabases
                ? obj.globallyDistributedDatabases.map(item => {
                    return model.DistributedDatabaseAssociatedWithPrivateEndpoint.getDeserializedJsonObj(item);
                })
                : undefined,
            "globallyDistributedAutonomousDatabases": obj.globallyDistributedAutonomousDatabases
                ? obj.globallyDistributedAutonomousDatabases.map(item => {
                    return model.DistributedAutonomousDatabaseAssociatedWithPrivateEndpoint.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DistributedDatabasePrivateEndpoint.getDeserializedJsonObj = getDeserializedJsonObj;
})(DistributedDatabasePrivateEndpoint = exports.DistributedDatabasePrivateEndpoint || (exports.DistributedDatabasePrivateEndpoint = {}));
