// oci-distributeddatabase/lib/model/distributed-database.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage the Globally distributed databases.
 * OpenAPI spec version: 20250101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DistributedDatabase = void 0;
const model = __importStar(require("../model"));
var DistributedDatabase;
(function (DistributedDatabase) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Creating"] = "CREATING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = DistributedDatabase.LifecycleState || (DistributedDatabase.LifecycleState = {}));
    let ShardingMethod;
    (function (ShardingMethod) {
        ShardingMethod["User"] = "USER";
        ShardingMethod["System"] = "SYSTEM";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ShardingMethod["UnknownValue"] = "UNKNOWN_VALUE";
    })(ShardingMethod = DistributedDatabase.ShardingMethod || (DistributedDatabase.ShardingMethod = {}));
    let ReplicationMethod;
    (function (ReplicationMethod) {
        ReplicationMethod["Raft"] = "RAFT";
        ReplicationMethod["Dg"] = "DG";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ReplicationMethod["UnknownValue"] = "UNKNOWN_VALUE";
    })(ReplicationMethod = DistributedDatabase.ReplicationMethod || (DistributedDatabase.ReplicationMethod = {}));
    let DbDeploymentType;
    (function (DbDeploymentType) {
        DbDeploymentType["ExadbXs"] = "EXADB_XS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DbDeploymentType["UnknownValue"] = "UNKNOWN_VALUE";
    })(DbDeploymentType = DistributedDatabase.DbDeploymentType || (DistributedDatabase.DbDeploymentType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "connectionStrings": obj.connectionStrings
                ? model.DistributedDbConnectionString.getJsonObj(obj.connectionStrings)
                : undefined,
            "latestGsmImageDetails": obj.latestGsmImageDetails
                ? model.DistributedDbGsmImage.getJsonObj(obj.latestGsmImageDetails)
                : undefined,
            "shardDetails": obj.shardDetails
                ? obj.shardDetails.map(item => {
                    return model.DistributedDatabaseShard.getJsonObj(item);
                })
                : undefined,
            "catalogDetails": obj.catalogDetails
                ? obj.catalogDetails.map(item => {
                    return model.DistributedDatabaseCatalog.getJsonObj(item);
                })
                : undefined,
            "gsmDetails": obj.gsmDetails
                ? obj.gsmDetails.map(item => {
                    return model.DistributedDatabaseGsm.getJsonObj(item);
                })
                : undefined,
            "dbBackupConfig": obj.dbBackupConfig
                ? model.DistributedDbBackupConfig.getJsonObj(obj.dbBackupConfig)
                : undefined,
            "metadata": obj.metadata ? model.DistributedDbMetadata.getJsonObj(obj.metadata) : undefined
        });
        return jsonObj;
    }
    DistributedDatabase.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "connectionStrings": obj.connectionStrings
                ? model.DistributedDbConnectionString.getDeserializedJsonObj(obj.connectionStrings)
                : undefined,
            "latestGsmImageDetails": obj.latestGsmImageDetails
                ? model.DistributedDbGsmImage.getDeserializedJsonObj(obj.latestGsmImageDetails)
                : undefined,
            "shardDetails": obj.shardDetails
                ? obj.shardDetails.map(item => {
                    return model.DistributedDatabaseShard.getDeserializedJsonObj(item);
                })
                : undefined,
            "catalogDetails": obj.catalogDetails
                ? obj.catalogDetails.map(item => {
                    return model.DistributedDatabaseCatalog.getDeserializedJsonObj(item);
                })
                : undefined,
            "gsmDetails": obj.gsmDetails
                ? obj.gsmDetails.map(item => {
                    return model.DistributedDatabaseGsm.getDeserializedJsonObj(item);
                })
                : undefined,
            "dbBackupConfig": obj.dbBackupConfig
                ? model.DistributedDbBackupConfig.getDeserializedJsonObj(obj.dbBackupConfig)
                : undefined,
            "metadata": obj.metadata
                ? model.DistributedDbMetadata.getDeserializedJsonObj(obj.metadata)
                : undefined
        });
        return jsonObj;
    }
    DistributedDatabase.getDeserializedJsonObj = getDeserializedJsonObj;
})(DistributedDatabase = exports.DistributedDatabase || (exports.DistributedDatabase = {}));
