// oci-cloudguard/lib/request/list-adhoc-queries-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAdhocQueriesRequest = void 0;
var ListAdhocQueriesRequest;
(function (ListAdhocQueriesRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListAdhocQueriesRequest.AccessLevel || (ListAdhocQueriesRequest.AccessLevel = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListAdhocQueriesRequest.SortBy || (ListAdhocQueriesRequest.SortBy = {}));
})(ListAdhocQueriesRequest = exports.ListAdhocQueriesRequest || (exports.ListAdhocQueriesRequest = {}));
